// @notificationapi/react/dist/components/Provider/index.js
import { jsx as M, jsxs as He } from "react/jsx-runtime";
import * as w from "react";
import { useMemo as pe, useState as F, useCallback as B, useRef as Se, useEffect as ce, useContext as Ge } from "react";
import { NotificationAPIContext as je } from "./context.js";
import { createDebugLogger as ze, formatApiCall as ae } from "../../utils/debug.js";
import { createNotificationAPITheme as Oe } from "../../utils/theme.js";
import { a as Be, D as Je, T as J } from "../../assets/DefaultPropsProvider2.js";
import { P as I, c as Ye, t as Qe, s as Xe } from "../../assets/createTheme.js";
import { R as Ze } from "../../assets/index.js";
import { u as et } from "../../assets/useThemeWithoutDefault.js";
import { e as Ve } from "../../assets/exactProp.js";
import { G as tt } from "../../assets/GlobalStyles.js";
const we = /* @__PURE__ */ w.createContext(null);
process.env.NODE_ENV !== "production" && (we.displayName = "ThemeContext");
function be() {
  const t = w.useContext(we);
  return process.env.NODE_ENV !== "production" && w.useDebugValue(t), t;
}
const ot = typeof Symbol == "function" && Symbol.for, nt = ot ? Symbol.for("mui.nested") : "__THEME_NESTED__";
function it(t, e) {
  if (typeof e == "function") {
    const c = e(t);
    return process.env.NODE_ENV !== "production" && (c || console.error(["MUI: You should return an object from your theme function, i.e.", "<ThemeProvider theme={() => ({})} />"].join(`
`))), c;
  }
  return {
    ...t,
    ...e
  };
}
function ge(t) {
  const {
    children: e,
    theme: c
  } = t, n = be();
  process.env.NODE_ENV !== "production" && n === null && typeof c == "function" && console.error(["MUI: You are providing a theme function prop to the ThemeProvider component:", "<ThemeProvider theme={outerTheme => outerTheme} />", "", "However, no outer theme is present.", "Make sure a theme is already injected higher in the React tree or provide a theme object."].join(`
`));
  const l = w.useMemo(() => {
    const u = n === null ? {
      ...c
    } : it(n, c);
    return u != null && (u[nt] = n !== null), u;
  }, [c, n]);
  return /* @__PURE__ */ M(we.Provider, {
    value: l,
    children: e
  });
}
process.env.NODE_ENV !== "production" && (ge.propTypes = {
  /**
   * Your component tree.
   */
  children: I.node,
  /**
   * A theme object. You can provide a function to extend the outer theme.
   */
  theme: I.oneOfType([I.object, I.func]).isRequired
});
process.env.NODE_ENV !== "production" && process.env.NODE_ENV !== "production" && (ge.propTypes = Ve(ge.propTypes));
const $e = {};
function Le(t, e, c, n = !1) {
  return w.useMemo(() => {
    const l = t && e[t] || e;
    if (typeof c == "function") {
      const u = c(l), d = t ? {
        ...e,
        [t]: u
      } : u;
      return n ? () => d : d;
    }
    return t ? {
      ...e,
      [t]: c
    } : {
      ...e,
      ...c
    };
  }, [t, e, c, n]);
}
function de(t) {
  const {
    children: e,
    theme: c,
    themeId: n
  } = t, l = et($e), u = be() || $e;
  process.env.NODE_ENV !== "production" && (l === null && typeof c == "function" || n && l && !l[n] && typeof c == "function") && console.error(["MUI: You are providing a theme function prop to the ThemeProvider component:", "<ThemeProvider theme={outerTheme => outerTheme} />", "", "However, no outer theme is present.", "Make sure a theme is already injected higher in the React tree or provide a theme object."].join(`
`));
  const d = Le(n, l, c), f = Le(n, u, c, !0), k = (n ? d[n] : d).direction === "rtl";
  return /* @__PURE__ */ M(ge, {
    theme: f,
    children: /* @__PURE__ */ M(Be.Provider, {
      value: d,
      children: /* @__PURE__ */ M(Ze, {
        value: k,
        children: /* @__PURE__ */ M(Je, {
          value: n ? d[n].components : d.components,
          children: e
        })
      })
    })
  });
}
process.env.NODE_ENV !== "production" && (de.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * Your component tree.
   */
  children: I.node,
  /**
   * A theme object. You can provide a function to extend the outer theme.
   */
  theme: I.oneOfType([I.func, I.object]).isRequired,
  /**
   * The design system's unique id for getting the corresponded theme when there are multiple design systems.
   */
  themeId: I.string
});
process.env.NODE_ENV !== "production" && process.env.NODE_ENV !== "production" && (de.propTypes = Ve(de.propTypes));
const ke = "mode", Ce = "color-scheme", rt = "data-color-scheme";
function st(t) {
  const {
    defaultMode: e = "system",
    defaultLightColorScheme: c = "light",
    defaultDarkColorScheme: n = "dark",
    modeStorageKey: l = ke,
    colorSchemeStorageKey: u = Ce,
    attribute: d = rt,
    colorSchemeNode: f = "document.documentElement",
    nonce: k
  } = t || {};
  let S = "", y = d;
  if (d === "class" && (y = ".%s"), d === "data" && (y = "[data-%s]"), y.startsWith(".")) {
    const b = y.substring(1);
    S += `${f}.classList.remove('${b}'.replace('%s', light), '${b}'.replace('%s', dark));
      ${f}.classList.add('${b}'.replace('%s', colorScheme));`;
  }
  const P = y.match(/\[([^\]]+)\]/);
  if (P) {
    const [b, N] = P[1].split("=");
    N || (S += `${f}.removeAttribute('${b}'.replace('%s', light));
      ${f}.removeAttribute('${b}'.replace('%s', dark));`), S += `
      ${f}.setAttribute('${b}'.replace('%s', colorScheme), ${N ? `${N}.replace('%s', colorScheme)` : '""'});`;
  } else
    S += `${f}.setAttribute('${y}', colorScheme);`;
  return /* @__PURE__ */ M("script", {
    suppressHydrationWarning: !0,
    nonce: typeof window > "u" ? k : "",
    dangerouslySetInnerHTML: {
      __html: `(function() {
try {
  let colorScheme = '';
  const mode = localStorage.getItem('${l}') || '${e}';
  const dark = localStorage.getItem('${u}-dark') || '${n}';
  const light = localStorage.getItem('${u}-light') || '${c}';
  if (mode === 'system') {
    // handle system mode
    const mql = window.matchMedia('(prefers-color-scheme: dark)');
    if (mql.matches) {
      colorScheme = dark
    } else {
      colorScheme = light
    }
  }
  if (mode === 'light') {
    colorScheme = light;
  }
  if (mode === 'dark') {
    colorScheme = dark;
  }
  if (colorScheme) {
    ${S}
  }
} catch(e){}})();`
    }
  }, "mui-color-scheme-init");
}
function Ue(t) {
  if (typeof window < "u" && typeof window.matchMedia == "function" && t === "system")
    return window.matchMedia("(prefers-color-scheme: dark)").matches ? "dark" : "light";
}
function _e(t, e) {
  if (t.mode === "light" || t.mode === "system" && t.systemMode === "light")
    return e("light");
  if (t.mode === "dark" || t.mode === "system" && t.systemMode === "dark")
    return e("dark");
}
function ct(t) {
  return _e(t, (e) => {
    if (e === "light")
      return t.lightColorScheme;
    if (e === "dark")
      return t.darkColorScheme;
  });
}
function ye(t, e) {
  if (typeof window > "u")
    return;
  let c;
  try {
    c = localStorage.getItem(t) || void 0, c || localStorage.setItem(t, e);
  } catch {
  }
  return c || e;
}
function at(t) {
  const {
    defaultMode: e = "light",
    defaultLightColorScheme: c,
    defaultDarkColorScheme: n,
    supportedColorSchemes: l = [],
    modeStorageKey: u = ke,
    colorSchemeStorageKey: d = Ce,
    storageWindow: f = typeof window > "u" ? void 0 : window,
    noSsr: k = !1
  } = t, S = l.join(","), y = l.length > 1, [P, b] = w.useState(() => {
    const h = ye(u, e), g = ye(`${d}-light`, c), m = ye(`${d}-dark`, n);
    return {
      mode: h,
      systemMode: Ue(h),
      lightColorScheme: g,
      darkColorScheme: m
    };
  }), [N, K] = w.useState(k || !y);
  w.useEffect(() => {
    K(!0);
  }, []);
  const G = ct(P), R = w.useCallback((h) => {
    b((g) => {
      if (h === g.mode)
        return g;
      const m = h ?? e;
      try {
        localStorage.setItem(u, m);
      } catch {
      }
      return {
        ...g,
        mode: m,
        systemMode: Ue(m)
      };
    });
  }, [u, e]), A = w.useCallback((h) => {
    h ? typeof h == "string" ? h && !S.includes(h) ? console.error(`\`${h}\` does not exist in \`theme.colorSchemes\`.`) : b((g) => {
      const m = {
        ...g
      };
      return _e(g, (D) => {
        try {
          localStorage.setItem(`${d}-${D}`, h);
        } catch {
        }
        D === "light" && (m.lightColorScheme = h), D === "dark" && (m.darkColorScheme = h);
      }), m;
    }) : b((g) => {
      const m = {
        ...g
      }, D = h.light === null ? c : h.light, p = h.dark === null ? n : h.dark;
      if (D)
        if (!S.includes(D))
          console.error(`\`${D}\` does not exist in \`theme.colorSchemes\`.`);
        else {
          m.lightColorScheme = D;
          try {
            localStorage.setItem(`${d}-light`, D);
          } catch {
          }
        }
      if (p)
        if (!S.includes(p))
          console.error(`\`${p}\` does not exist in \`theme.colorSchemes\`.`);
        else {
          m.darkColorScheme = p;
          try {
            localStorage.setItem(`${d}-dark`, p);
          } catch {
          }
        }
      return m;
    }) : b((g) => {
      try {
        localStorage.setItem(`${d}-light`, c), localStorage.setItem(`${d}-dark`, n);
      } catch {
      }
      return {
        ...g,
        lightColorScheme: c,
        darkColorScheme: n
      };
    });
  }, [S, d, c, n]), j = w.useCallback((h) => {
    P.mode === "system" && b((g) => {
      const m = h != null && h.matches ? "dark" : "light";
      return g.systemMode === m ? g : {
        ...g,
        systemMode: m
      };
    });
  }, [P.mode]), _ = w.useRef(j);
  return _.current = j, w.useEffect(() => {
    if (typeof window.matchMedia != "function" || !y)
      return;
    const h = (...m) => _.current(...m), g = window.matchMedia("(prefers-color-scheme: dark)");
    return g.addListener(h), h(g), () => {
      g.removeListener(h);
    };
  }, [y]), w.useEffect(() => {
    if (f && y) {
      const h = (g) => {
        const m = g.newValue;
        typeof g.key == "string" && g.key.startsWith(d) && (!m || S.match(m)) && (g.key.endsWith("light") && A({
          light: m
        }), g.key.endsWith("dark") && A({
          dark: m
        })), g.key === u && (!m || ["light", "dark", "system"].includes(m)) && R(m || e);
      };
      return f.addEventListener("storage", h), () => {
        f.removeEventListener("storage", h);
      };
    }
  }, [A, R, u, d, S, e, f, y]), {
    ...P,
    mode: N ? P.mode : void 0,
    systemMode: N ? P.systemMode : void 0,
    colorScheme: N ? G : void 0,
    setMode: R,
    setColorScheme: A
  };
}
const dt = "*{-webkit-transition:none!important;-moz-transition:none!important;-o-transition:none!important;-ms-transition:none!important;transition:none!important}";
function lt(t) {
  const {
    themeId: e,
    /**
     * This `theme` object needs to follow a certain structure to
     * be used correctly by the finel `CssVarsProvider`. It should have a
     * `colorSchemes` key with the light and dark (and any other) palette.
     * It should also ideally have a vars object created using `prepareCssVars`.
     */
    theme: c = {},
    modeStorageKey: n = ke,
    colorSchemeStorageKey: l = Ce,
    disableTransitionOnChange: u = !1,
    defaultColorScheme: d,
    resolveTheme: f
  } = t, k = {
    allColorSchemes: [],
    colorScheme: void 0,
    darkColorScheme: void 0,
    lightColorScheme: void 0,
    mode: void 0,
    setColorScheme: () => {
    },
    setMode: () => {
    },
    systemMode: void 0
  }, S = /* @__PURE__ */ w.createContext(void 0);
  process.env.NODE_ENV !== "production" && (S.displayName = "ColorSchemeContext");
  const y = () => w.useContext(S) || k, P = {}, b = {};
  function N(A) {
    var Te, Me, Ae, De;
    const {
      children: j,
      theme: _,
      modeStorageKey: h = n,
      colorSchemeStorageKey: g = l,
      disableTransitionOnChange: m = u,
      storageWindow: D = typeof window > "u" ? void 0 : window,
      documentNode: p = typeof document > "u" ? void 0 : document,
      colorSchemeNode: L = typeof document > "u" ? void 0 : document.documentElement,
      disableNestedContext: Y = !1,
      disableStyleSheetGeneration: Q = !1,
      defaultMode: le = "system",
      noSsr: X
    } = A, Z = w.useRef(!1), ee = be(), te = w.useContext(S), oe = !!te && !Y, ue = w.useMemo(() => _ || (typeof c == "function" ? c() : c), [_]), ne = ue[e], T = ne || ue, {
      colorSchemes: W = P,
      components: fe = b,
      cssVarPrefix: o
    } = T, r = Object.keys(W).filter((E) => !!W[E]).join(","), i = w.useMemo(() => r.split(","), [r]), a = typeof d == "string" ? d : d.light, C = typeof d == "string" ? d : d.dark, v = W[a] && W[C] ? le : ((Me = (Te = W[T.defaultColorScheme]) == null ? void 0 : Te.palette) == null ? void 0 : Me.mode) || ((Ae = T.palette) == null ? void 0 : Ae.mode), {
      mode: q,
      setMode: H,
      systemMode: ie,
      lightColorScheme: Ie,
      darkColorScheme: Ne,
      colorScheme: Ke,
      setColorScheme: ve
    } = at({
      supportedColorSchemes: i,
      defaultLightColorScheme: a,
      defaultDarkColorScheme: C,
      modeStorageKey: h,
      colorSchemeStorageKey: g,
      defaultMode: v,
      storageWindow: D,
      noSsr: X
    });
    let me = q, U = Ke;
    oe && (me = te.mode, U = te.colorScheme);
    const z = w.useMemo(() => {
      var se;
      const E = U || T.defaultColorScheme, O = ((se = T.generateThemeVars) == null ? void 0 : se.call(T)) || T.vars, $ = {
        ...T,
        components: fe,
        colorSchemes: W,
        cssVarPrefix: o,
        vars: O
      };
      if (typeof $.generateSpacing == "function" && ($.spacing = $.generateSpacing()), E) {
        const x = W[E];
        x && typeof x == "object" && Object.keys(x).forEach((V) => {
          x[V] && typeof x[V] == "object" ? $[V] = {
            ...$[V],
            ...x[V]
          } : $[V] = x[V];
        });
      }
      return f ? f($) : $;
    }, [T, U, fe, W, o]), re = T.colorSchemeSelector;
    w.useEffect(() => {
      if (U && L && re && re !== "media") {
        const E = re;
        let O = re;
        if (E === "class" && (O = ".%s"), E === "data" && (O = "[data-%s]"), E != null && E.startsWith("data-") && !E.includes("%s") && (O = `[${E}="%s"]`), O.startsWith("."))
          L.classList.remove(...i.map(($) => O.substring(1).replace("%s", $))), L.classList.add(O.substring(1).replace("%s", U));
        else {
          const $ = O.replace("%s", U).match(/\[([^\]]+)\]/);
          if ($) {
            const [se, x] = $[1].split("=");
            x || i.forEach((V) => {
              L.removeAttribute(se.replace(U, V));
            }), L.setAttribute(se, x ? x.replace(/"|'/g, "") : "");
          } else
            L.setAttribute(O, U);
        }
      }
    }, [U, re, L, i]), w.useEffect(() => {
      let E;
      if (m && Z.current && p) {
        const O = p.createElement("style");
        O.appendChild(p.createTextNode(dt)), p.head.appendChild(O), window.getComputedStyle(p.body), E = setTimeout(() => {
          p.head.removeChild(O);
        }, 1);
      }
      return () => {
        clearTimeout(E);
      };
    }, [U, m, p]), w.useEffect(() => (Z.current = !0, () => {
      Z.current = !1;
    }), []);
    const qe = w.useMemo(() => ({
      allColorSchemes: i,
      colorScheme: U,
      darkColorScheme: Ne,
      lightColorScheme: Ie,
      mode: me,
      setColorScheme: ve,
      setMode: process.env.NODE_ENV === "production" ? H : (E) => {
        z.colorSchemeSelector === "media" && console.error(["MUI: The `setMode` function has no effect if `colorSchemeSelector` is `media` (`media` is the default value).", "To toggle the mode manually, please configure `colorSchemeSelector` to use a class or data attribute.", "To learn more, visit https://mui.com/material-ui/customization/css-theme-variables/configuration/#toggling-dark-mode-manually"].join(`
`)), H(E);
      },
      systemMode: ie
    }), [i, U, Ne, Ie, me, ve, H, ie, z.colorSchemeSelector]);
    let Pe = !0;
    (Q || T.cssVariables === !1 || oe && (ee == null ? void 0 : ee.cssVarPrefix) === o) && (Pe = !1);
    const Ee = /* @__PURE__ */ He(w.Fragment, {
      children: [/* @__PURE__ */ M(de, {
        themeId: ne ? e : void 0,
        theme: z,
        children: j
      }), Pe && /* @__PURE__ */ M(tt, {
        styles: ((De = z.generateStyleSheets) == null ? void 0 : De.call(z)) || []
      })]
    });
    return oe ? Ee : /* @__PURE__ */ M(S.Provider, {
      value: qe,
      children: Ee
    });
  }
  process.env.NODE_ENV !== "production" && (N.propTypes = {
    /**
     * The component tree.
     */
    children: I.node,
    /**
     * The node used to attach the color-scheme attribute
     */
    colorSchemeNode: I.any,
    /**
     * localStorage key used to store `colorScheme`
     */
    colorSchemeStorageKey: I.string,
    /**
     * The default mode when the storage is empty,
     * require the theme to have `colorSchemes` with light and dark.
     */
    defaultMode: I.string,
    /**
     * If `true`, the provider creates its own context and generate stylesheet as if it is a root `CssVarsProvider`.
     */
    disableNestedContext: I.bool,
    /**
     * If `true`, the style sheet won't be generated.
     *
     * This is useful for controlling nested CssVarsProvider behavior.
     */
    disableStyleSheetGeneration: I.bool,
    /**
     * Disable CSS transitions when switching between modes or color schemes.
     */
    disableTransitionOnChange: I.bool,
    /**
     * The document to attach the attribute to.
     */
    documentNode: I.any,
    /**
     * The key in the local storage used to store current color scheme.
     */
    modeStorageKey: I.string,
    /**
     * If `true`, the mode will be the same value as the storage without an extra rerendering after the hydration.
     * You should use this option in conjuction with `InitColorSchemeScript` component.
     */
    noSsr: I.bool,
    /**
     * The window that attaches the 'storage' event listener.
     * @default window
     */
    storageWindow: I.any,
    /**
     * The calculated theme object that will be passed through context.
     */
    theme: I.object
  });
  const K = typeof d == "string" ? d : d.light, G = typeof d == "string" ? d : d.dark;
  return {
    CssVarsProvider: N,
    useColorScheme: y,
    getInitColorSchemeScript: (A) => st({
      colorSchemeStorageKey: l,
      defaultLightColorScheme: K,
      defaultDarkColorScheme: G,
      modeStorageKey: n,
      ...A
    })
  };
}
function xe({
  theme: t,
  ...e
}) {
  const c = J in t ? t[J] : void 0;
  return /* @__PURE__ */ M(de, {
    ...e,
    themeId: c ? J : void 0,
    theme: c || t
  });
}
const he = {
  attribute: "data-mui-color-scheme",
  colorSchemeStorageKey: "mui-color-scheme",
  defaultLightColorScheme: "light",
  defaultDarkColorScheme: "dark",
  modeStorageKey: "mui-mode"
}, {
  CssVarsProvider: ut,
  useColorScheme: At,
  getInitColorSchemeScript: Dt
} = lt({
  themeId: J,
  // @ts-ignore ignore module augmentation tests
  theme: () => Ye({
    cssVariables: !0
  }),
  colorSchemeStorageKey: he.colorSchemeStorageKey,
  modeStorageKey: he.modeStorageKey,
  defaultColorScheme: {
    light: he.defaultLightColorScheme,
    dark: he.defaultDarkColorScheme
  },
  resolveTheme: (t) => {
    const e = {
      ...t,
      typography: Qe(t.palette, t.typography)
    };
    return e.unstable_sx = function(n) {
      return Xe({
        sx: n,
        theme: this
      });
    }, e;
  }
}), ft = ut;
function ht({
  theme: t,
  ...e
}) {
  return typeof t == "function" ? /* @__PURE__ */ M(xe, {
    theme: t,
    ...e
  }) : "colorSchemes" in (J in t ? t[J] : t) ? /* @__PURE__ */ M(ft, {
    theme: t,
    ...e
  }) : /* @__PURE__ */ M(xe, {
    theme: t,
    ...e
  });
}
const gt = async (t, e, c, n, l, u, d, f) => {
  const k = mt(n, l, u), S = `https://${e}/${n}/users/${encodeURIComponent(
    l
  )}/${c}`, y = {
    Authorization: `Basic ${k}`
  };
  f && f.log("HTTP Request:", {
    method: t,
    host: e,
    url: S,
    body: d
  });
  const P = Date.now();
  try {
    const b = await fetch(S, {
      method: t,
      body: JSON.stringify(d),
      headers: y
    });
    f && f.log("HTTP Response:", b);
    try {
      const N = await b.json();
      return f && f.log("Response Data:", N), N;
    } catch (N) {
      f && f.warn("Failed to parse response as JSON:", N);
      return;
    }
  } catch (b) {
    const N = Date.now() - P;
    throw f && f.error("HTTP Request Failed:", {
      error: b,
      url: S,
      headers: y,
      duration: `${N}ms`
    }), b;
  }
}, mt = (t, e, c) => btoa(c ? t + ":" + e + ":" + c : t + ":" + e);
class Re {
  constructor(e = !1) {
    this.debug = e;
  }
  log(...e) {
    this.debug && console.log("[NotificationAPI js core SDK Debug]", ...e);
  }
  warn(...e) {
    this.debug && console.warn("[NotificationAPI js core SDK Debug]", ...e);
  }
  error(...e) {
    this.debug && console.error("[NotificationAPI js core SDK Debug]", ...e);
  }
}
const We = {
  host: "api.notificationapi.com",
  websocketHost: "ws.notificationapi.com",
  userId: "",
  clientId: "",
  hashedUserId: "",
  getInAppDefaultCount: 100,
  getInAppDefaultOldest: new Date(
    Date.now() - 30 * 24 * 60 * 60 * 1e3
  ).toISOString(),
  onNewInAppNotifications: void 0,
  keepWebSocketAliveForSeconds: 24 * 60 * 60,
  // 24 hours
  debug: !1,
  sdkDevMode: !1
}, s = {
  config: We,
  logger: new Re(!1),
  init: function(t) {
    return this.config = { ...We, ...t }, this.logger = new Re(this.config.debug), this.logger.log("Initialized with config:", {
      userId: this.config.userId,
      clientId: this.config.clientId,
      host: this.config.host,
      websocketHost: this.config.websocketHost,
      debug: this.config.debug,
      hasHashedUserId: !!this.config.hashedUserId
    }), {
      ...this
    };
  },
  rest: {
    generic: function(t, e, c) {
      return s.logger.log(
        `API Call: ${t} ${e}`,
        c ? { body: c } : ""
      ), gt(
        t,
        s.config.host,
        e,
        s.config.clientId,
        s.config.userId,
        s.config.hashedUserId,
        c,
        s.logger
      );
    },
    // The functions below are nice wrappers over the generic
    // rest  api function above. They must follow REST API naming:
    // Method + Resource, representing the end-point.
    getNotifications: function(t, e) {
      return s.rest.generic(
        "GET",
        `notifications/INAPP_WEB?count=${e}&before=${t}`
      );
    },
    patchNotifications: function(t) {
      return s.rest.generic(
        "PATCH",
        "notifications/INAPP_WEB",
        t
      );
    },
    getPreferences: function() {
      return s.rest.generic("GET", "preferences");
    },
    postPreferences: function(t) {
      return s.rest.generic(
        "POST",
        "preferences",
        t
      );
    },
    postUser: function(t) {
      return s.rest.generic("POST", "", t);
    },
    getUserAccountMetadata: function() {
      return s.rest.generic("GET", "account_metadata");
    }
  },
  websocket: {
    object: void 0,
    connect: function() {
      let t = `wss://${s.config.websocketHost}?userId=${encodeURIComponent(s.config.userId)}&envId=${s.config.clientId}`;
      return s.config.hashedUserId && (t += `&userIdHash=${encodeURIComponent(s.config.hashedUserId)}`), s.logger.log("WebSocket connecting to:", t), s.websocket.object = new WebSocket(t), s.websocket.object.onopen = () => {
        s.logger.log("WebSocket connection opened");
      }, s.websocket.object.onclose = (e) => {
        s.logger.log("WebSocket connection closed:", {
          code: e.code,
          reason: e.reason,
          wasClean: e.wasClean
        });
      }, s.websocket.object.onerror = (e) => {
        s.logger.error("WebSocket error:", e);
      }, s.websocket.object.onmessage = (e) => {
        s.logger.log(
          "WebSocket message received:",
          e.data
        );
        const c = JSON.parse(e.data);
        if (!(!c || !c.route) && c.route === "inapp_web/new_notifications") {
          const n = c;
          s.logger.log(
            "New notifications received:",
            n.payload.notifications
          ), s.config.onNewInAppNotifications && s.config.onNewInAppNotifications(
            n.payload.notifications
          );
        }
      }, s.websocket.object;
    },
    disconnect: function(t) {
      var e;
      s.websocket.object && (s.logger.log("WebSocket disconnecting"), (e = s.websocket.object) == null || e.close(), t && t(s.websocket.object));
    }
  },
  openWebSocket: function() {
    return s.websocket.connect(() => {
      setTimeout(
        () => {
          this.websocket.disconnect(() => {
            this.websocket.connect();
          });
        },
        54e4
      );
    });
  },
  // These functions are developer friendly wrappers over the rest APIs
  // They may or may not do additional tasks.
  // e.g. identify simply maps to postUsers
  getInAppNotifications: async (t) => {
    s.logger.log(
      "getInAppNotifications called with params:",
      t
    );
    const e = t.maxCountNeeded || s.config.getInAppDefaultCount, c = t.oldestNeeded || s.config.getInAppDefaultOldest;
    s.logger.log("Fetching notifications with:", {
      maxCountNeeded: e,
      oldestNeeded: c,
      before: t.before
    });
    let n = [], l = t.before, u = !0, d = !0;
    for (; d; ) {
      const f = (await s.rest.getNotifications(
        l,
        e
      )).notifications, k = f.filter(
        (S) => !n.find((y) => y.id === S.id)
      );
      s.logger.log(
        `Received ${f.length} notifications, ${k.length} unique`
      ), l = k.reduce(
        (S, y) => S < y.date ? S : y.date,
        t.before
      ), n = [...n, ...k], u = k.length > 0, d = !0, (!u || n.length >= e || l < c) && (d = !1, s.logger.log("Stopping fetch loop:", {
        hasMore: u,
        totalResults: n.length,
        maxCountNeeded: e,
        oldestReceived: l,
        oldestNeeded: c
      }));
    }
    return s.logger.log("getInAppNotifications completed:", {
      totalItems: n.length,
      hasMore: u,
      oldestReceived: l
    }), {
      items: n,
      hasMore: u,
      oldestReceived: l
    };
  },
  updateInAppNotifications: async (t) => {
    s.logger.log(
      "updateInAppNotifications called with params:",
      t
    );
    const e = {
      trackingIds: t.ids
    };
    return t.archived === !0 ? e.archived = (/* @__PURE__ */ new Date()).toISOString() : t.archived === !1 && (e.archived = null), t.clicked === !0 ? e.clicked = (/* @__PURE__ */ new Date()).toISOString() : t.clicked === !1 && (e.clicked = null), t.opened === !0 ? e.opened = (/* @__PURE__ */ new Date()).toISOString() : t.opened === !1 && (e.opened = null), s.logger.log(
      "Updating notifications with body:",
      e
    ), s.rest.patchNotifications(e);
  },
  getPreferences: async () => s.rest.getPreferences(),
  updateDeliveryOption: async (t) => s.rest.postPreferences([t]),
  identify: async (t) => {
    if (t.id && t.id !== s.config.userId)
      throw new Error(
        "The id in the parameters does not match the initialized userId."
      );
    return s.rest.postUser(t);
  },
  getUserAccountMetadata: async () => s.rest.getUserAccountMetadata(),
  user: {
    get: async () => s.rest.generic("GET", "")
  },
  slack: {
    getOAuthUrl: (t) => {
      const e = s.config.sdkDevMode ? "localhost:3001" : s.config.host.replace("api.", "app."), c = (t == null ? void 0 : t.destinationUrl) || window.location.href, n = encodeURIComponent(
        JSON.stringify({
          destination: c,
          userId: s.config.userId,
          clientId: s.config.clientId,
          hashedUserId: s.config.hashedUserId
        })
      );
      return `https://slack.com/oauth/v2/authorize?client_id=1146598856352.8825220259395&scope=chat:write,channels:read,channels:join,chat:write.customize,chat:write.public,groups:read,im:read,users:read&redirect_uri=https://${e}/slack/oauth/callback&state=${n}`;
    },
    getChannels: async () => await s.rest.generic(
      "GET",
      "slack/channels"
    ),
    setChannel: async (t) => s.identify({
      slackChannel: t
    })
  }
}, Fe = typeof window < "u", pt = (t) => {
  const e = pe(
    () => ze(t.debug || !1),
    [t.debug]
  );
  e.log("NotificationAPI Provider initializing", {
    clientId: t.clientId,
    userId: "userId" in t ? t.userId : t.user.id,
    debug: t.debug || !1,
    timestamp: (/* @__PURE__ */ new Date()).toISOString()
  });
  const n = {
    ...{
      apiURL: "api.notificationapi.com",
      wsURL: "ws.notificationapi.com",
      initialLoadMaxCount: 1e3,
      initialLoadMaxAge: new Date((/* @__PURE__ */ new Date()).setMonth((/* @__PURE__ */ new Date()).getMonth() - 3)),
      playSoundOnNewNotification: !1,
      newNotificationSoundPath: "https://proxy.notificationsounds.com/notification-sounds/elegant-notification-sound/download/file-sounds-1233-elegant.mp3",
      webPushOptInMessage: "AUTOMATIC",
      customServiceWorkerPath: "/notificationapi-service-worker.js"
    },
    ...t,
    user: "userId" in t ? { id: t.userId } : t.user
  };
  e.log("Configuration loaded", n);
  const [l, u] = F(), [d, f] = F(), [k, S] = F(), [y, P] = F(!1), [b, N] = F((/* @__PURE__ */ new Date()).toISOString()), [K, G] = F(!0), [R, A] = F(n.webPushOptInMessage), [j, _] = F(!1), h = B(() => {
    n.playSoundOnNewNotification && (e.log("Playing notification sound", {
      soundPath: n.newNotificationSoundPath
    }), new Audio(n.newNotificationSoundPath).play().catch((r) => {
      e.error("Failed to play new notification sound", r), console.log("Failed to play new notification sound:", r);
    }));
  }, [
    n.newNotificationSoundPath,
    n.playSoundOnNewNotification,
    e
  ]), g = B(
    (o) => {
      e.group("Adding notifications to state"), e.log("Received notifications", {
        count: (o == null ? void 0 : o.length) || 0,
        notifications: o
      });
      const r = (/* @__PURE__ */ new Date()).toISOString();
      u((i) => {
        const a = (i == null ? void 0 : i.length) || 0;
        e.log("Current notifications count", a), o = Array.isArray(o) ? o : [], o = o.filter((v) => {
          const q = v.expDate && new Date(v.expDate).toISOString() > r, H = new Date(v.date).getTime() > new Date(r).getTime() + 1e3, ie = !q && !H;
          return ie || e.log("Filtering out notification", {
            id: v.id,
            reason: q ? "expired" : "future",
            expDate: v.expDate,
            date: v.date
          }), ie;
        }), i = Array.isArray(i) ? i : [];
        const C = [
          ...o.filter((v) => {
            const q = i.find((H) => H.id === v.id);
            return q && e.log("Filtering out duplicate notification", { id: v.id }), !q;
          }),
          ...i
        ];
        return e.log("State updated", {
          previousCount: a,
          newCount: C.length,
          addedCount: o.length
        }), e.groupEnd(), C;
      });
    },
    [e]
  ), { onNewNotifications: m } = t, D = B(
    (o) => {
      e.log("Received new in-app notifications via WebSocket", {
        count: (o == null ? void 0 : o.length) || 0,
        notifications: o
      }), h(), g(o), m && (e.log("Calling onNewNotifications callback", {
        count: (o == null ? void 0 : o.length) || 0
      }), m(o));
    },
    [h, g, m, e]
  ), p = pe(() => {
    e.group("Initializing NotificationAPI client");
    const o = {
      clientId: n.clientId,
      userId: n.user.id,
      hashedUserId: n.hashedUserId,
      host: n.apiURL,
      websocketHost: n.wsURL,
      debug: n.debug
    };
    e.log("Client configuration", o);
    const r = t.client ? t.client : s.init({
      ...o,
      onNewInAppNotifications: D
    }), i = {
      email: n.user.email,
      number: n.user.number
    };
    return e.log("Identifying user", i), r.identify(i), e.groupEnd(), r;
  }, [
    n.clientId,
    n.user.id,
    n.user.email,
    n.user.number,
    n.hashedUserId,
    n.debug,
    D,
    t.client,
    n.apiURL,
    n.wsURL,
    e
  ]), L = B(
    async (o, r) => {
      var i;
      e.group("Fetching notifications"), e.log(
        "Fetch parameters",
        ae("GET", "/notifications", { date: o, count: r })
      );
      try {
        const a = await p.rest.getNotifications(o, r);
        e.log("Fetch successful", {
          notificationsCount: ((i = a.notifications) == null ? void 0 : i.length) || 0,
          oldestReceived: a.oldestReceived,
          couldLoadMore: a.couldLoadMore
        }), N(a.oldestReceived), G(a.couldLoadMore), g(a.notifications), e.groupEnd();
      } catch (a) {
        throw e.error("Failed to fetch notifications", a, { date: o, count: r }), e.groupEnd(), a;
      }
    },
    [g, p.rest, e]
  ), Y = Se(K), Q = Se(y), le = Se(b);
  ce(() => {
    Y.current = K, Q.current = y, le.current = b;
  }, [K, y, b]);
  const X = B(
    async (o) => {
      if (e.group(`Loading notifications (${o ? "initial" : "more"})`), e.log("Load conditions", {
        initial: o,
        hasMore: Y.current,
        loading: Q.current
      }), !o && (!Y.current || Q.current)) {
        e.log("Skipping load - conditions not met"), e.groupEnd();
        return;
      }
      P(!0), e.log("Loading started");
      try {
        await L(
          o ? (/* @__PURE__ */ new Date()).toISOString() : le.current,
          o ? n.initialLoadMaxCount : 1e3
        ), e.log("Loading completed successfully");
      } catch (r) {
        throw e.error("Loading failed", r), r;
      } finally {
        P(!1), e.groupEnd();
      }
    },
    [n.initialLoadMaxCount, L, e]
  ), Z = async (o) => {
    if (e.group("Marking notifications as clicked"), e.log("Requested IDs", o), !l) {
      e.warn("No notifications available"), e.groupEnd();
      return;
    }
    const r = (/* @__PURE__ */ new Date()).toISOString(), i = l.filter((a) => o.includes(a.id) && !a.clicked).map((a) => a.id);
    if (e.log("Filtered IDs for update", {
      requestedCount: o.length,
      actualCount: i.length,
      ids: i
    }), i.length === 0) {
      e.log("No notifications to update"), e.groupEnd();
      return;
    }
    try {
      e.log(
        "Updating notifications via API",
        ae("PUT", "/notifications/clicked", { ids: i })
      ), p.updateInAppNotifications({ ids: i, clicked: !0 }), u((a) => {
        if (!a) return [];
        const C = [...a];
        return C.filter((v) => i.includes(v.id)).forEach((v) => {
          v.clicked = r;
        }), e.log("Local state updated", { updatedCount: i.length }), C;
      }), e.groupEnd();
    } catch (a) {
      throw e.error("Failed to mark notifications as clicked", a, { ids: i }), e.groupEnd(), a;
    }
  }, ee = async () => {
    if (e.group("Marking notifications as opened"), !l) {
      e.warn("No notifications available"), e.groupEnd();
      return;
    }
    const o = (/* @__PURE__ */ new Date()).toISOString(), r = l.filter((i) => !i.opened || !i.seen).map((i) => i.id);
    if (e.log("Notifications to mark as opened", { count: r.length, ids: r }), r.length === 0) {
      e.log("All notifications already opened"), e.groupEnd();
      return;
    }
    try {
      e.log(
        "Updating notifications via API",
        ae("PUT", "/notifications/opened", { ids: r })
      ), p.updateInAppNotifications({
        ids: r,
        opened: !0
      }), u((i) => {
        if (!i) return [];
        const a = [...i];
        return a.filter((C) => r.includes(C.id)).forEach((C) => {
          C.opened = o, C.seen = !0;
        }), e.log("Local state updated", { updatedCount: r.length }), a;
      }), e.groupEnd();
    } catch (i) {
      throw e.error("Failed to mark notifications as opened", i, { ids: r }), e.groupEnd(), i;
    }
  }, te = async (o) => {
    if (e.group("Marking notifications as unarchived"), e.log("Requested operation", { ids: o }), !l) {
      e.warn("No notifications available"), e.groupEnd();
      return;
    }
    const r = l.filter((i) => i.archived && (o === "ALL" || o.includes(i.id))).map((i) => i.id);
    if (e.log("Filtered notifications for unarchiving", {
      count: r.length,
      ids: r
    }), r.length === 0) {
      e.log("No archived notifications to unarchive"), e.groupEnd();
      return;
    }
    try {
      e.log(
        "Updating notifications via API",
        ae("PUT", "/notifications/unarchived", { ids: r })
      ), p.updateInAppNotifications({
        ids: r,
        archived: !1
      }), u((i) => {
        if (!i) return [];
        const a = [...i];
        return a.filter((C) => r.includes(C.id)).forEach((C) => {
          C.archived = void 0;
        }), e.log("Local state updated", { unarchivedCount: r.length }), a;
      }), e.groupEnd();
    } catch (i) {
      throw e.error("Failed to unarchive notifications", i, { ids: r }), e.groupEnd(), i;
    }
  }, oe = async (o) => {
    if (e.group("Marking notifications as archived"), e.log("Requested operation", { ids: o }), !l) {
      e.warn("No notifications available"), e.groupEnd();
      return;
    }
    const r = (/* @__PURE__ */ new Date()).toISOString(), i = l.filter((a) => !a.archived && (o === "ALL" || o.includes(a.id))).map((a) => a.id);
    if (e.log("Filtered notifications for archiving", {
      count: i.length,
      ids: i
    }), i.length === 0) {
      e.log("No unarchived notifications to archive"), e.groupEnd();
      return;
    }
    try {
      e.log(
        "Updating notifications via API",
        ae("PUT", "/notifications/archived", { ids: i })
      ), p.updateInAppNotifications({ ids: i, archived: !0 }), u((a) => {
        if (!a) return [];
        const C = [...a];
        return C.filter((v) => i.includes(v.id)).forEach((v) => {
          v.archived = r;
        }), e.log("Local state updated", { archivedCount: i.length }), C;
      }), e.groupEnd();
    } catch (a) {
      throw e.error("Failed to archive notifications", a, { ids: i }), e.groupEnd(), a;
    }
  }, ue = (o, r, i, a) => (e.log("Updating single delivery preference", {
    notificationId: o,
    channel: r,
    delivery: i,
    subNotificationId: a
  }), ne([
    {
      notificationId: o,
      channel: r,
      delivery: i,
      subNotificationId: a
    }
  ])), ne = (o) => {
    e.group("Updating delivery preferences"), e.log("Preference updates", { count: o.length, params: o });
    try {
      p.rest.postPreferences(o).then(() => {
        e.log(
          "Preferences updated successfully, fetching latest preferences"
        ), p.getPreferences().then((r) => {
          var i, a, C;
          e.log("Latest preferences fetched", {
            preferencesCount: ((i = r.preferences) == null ? void 0 : i.length) || 0,
            notificationsCount: ((a = r.notifications) == null ? void 0 : a.length) || 0,
            subNotificationsCount: ((C = r.subNotifications) == null ? void 0 : C.length) || 0
          }), f(r), e.groupEnd();
        }).catch((r) => {
          e.error("Failed to fetch updated preferences", r), e.groupEnd();
        });
      }).catch((r) => {
        e.error("Failed to update preferences", r, { params: o }), e.groupEnd();
      });
    } catch (r) {
      throw e.error("Error in updateDeliveries", r, { params: o }), e.groupEnd(), r;
    }
  }, T = B(() => {
    e.group("Requesting web push permission"), e.log("Service worker support check", {
      supported: "serviceWorker" in navigator,
      customServiceWorkerPath: n.customServiceWorkerPath
    }), "serviceWorker" in navigator ? (e.log("Registering service worker"), navigator.serviceWorker.register(n.customServiceWorkerPath).then(async (o) => {
      e.log("Service worker registered successfully"), A(!1), yt().then(async (r) => {
        e.log("Notification permission result", { permission: r }), r === "granted" ? (e.log("Permission granted, subscribing to push manager"), await o.pushManager.subscribe({
          userVisibleOnly: !0,
          applicationServerKey: k == null ? void 0 : k.userAccountMetadata.environmentVapidPublicKey
        }).then(async (i) => {
          e.log("Push subscription successful");
          const a = {
            webPushTokens: [
              {
                sub: {
                  endpoint: i.toJSON().endpoint,
                  keys: i.toJSON().keys
                }
              }
            ]
          };
          e.log("Identifying user with web push tokens"), await p.identify(a), localStorage.setItem("hideWebPushOptInMessage", "true"), e.log("Web push setup completed successfully"), e.groupEnd();
        })) : r === "denied" && (e.warn("Permission for notifications was denied"), console.log("Permission for notifications was denied"), e.groupEnd());
      });
    }).catch((o) => {
      e.error(
        "Service worker registration or push subscription failed",
        o,
        {
          errorCode: o.code,
          customServiceWorkerPath: n.customServiceWorkerPath
        }
      ), o.code === 18 ? console.error(
        "NotificationAPI guide: Probably you are not setup the service worker correctly. Please check the documentation at https://docs.notificationapi.com/guides/web-push#step-by-step-implementation Step 3: Service Worker Setup."
      ) : o.code === 19 ? console.error(
        "The operation is aborted. This can happen if the user denies the permission request."
      ) : o.code === 20 ? console.error(
        "The operation is invalid. This can occur if the provided application server key is invalid. Please contact NotificationAPI support."
      ) : o.code === 21 ? console.error(
        "The operation is not allowed. This can happen if the user has blocked notifications for the site. Please check your browser site settings Notifications part."
      ) : o.code === 22 ? console.error(
        "The operation is not supported. This can occur if the browser does not support the required features."
      ) : console.error(o), e.groupEnd();
    })) : (e.warn("Service worker not supported in this browser"), e.groupEnd());
  }, [
    p,
    n.customServiceWorkerPath,
    k == null ? void 0 : k.userAccountMetadata.environmentVapidPublicKey,
    e
  ]);
  ce(() => {
    e.group("Provider initialization effect"), e.log("Resetting state and loading initial data"), u([]), P(!1), f(void 0), N((/* @__PURE__ */ new Date()).toISOString()), G(!0), X(!0), e.log("Opening WebSocket connection"), p.openWebSocket(), e.log("Fetching user preferences"), p.getPreferences().then((o) => {
      var r, i, a;
      e.log("Initial preferences loaded", {
        preferencesCount: ((r = o.preferences) == null ? void 0 : r.length) || 0,
        notificationsCount: ((i = o.notifications) == null ? void 0 : i.length) || 0,
        subNotificationsCount: ((a = o.subNotifications) == null ? void 0 : a.length) || 0
      }), f(o), e.groupEnd();
    }).catch((o) => {
      e.error("Failed to fetch initial preferences", o), e.groupEnd();
    });
  }, [p, X, T, e]), ce(() => {
    e.group("Fetching user account metadata"), p.getUserAccountMetadata().then((o) => {
      e.log("User account metadata loaded", {
        hasWebPushEnabled: o.userAccountMetadata.hasWebPushEnabled,
        environmentVapidPublicKey: o.userAccountMetadata.environmentVapidPublicKey ? "present" : "missing"
      }), S(o), Fe && "Notification" in window && typeof Notification.requestPermission == "function" ? (e.log("Browser notification support detected", {
        permission: Notification.permission
      }), Notification.permission !== "default" && (e.log(
        "Setting webPushOptInMessage to false (permission already set)"
      ), A(!1))) : (e.log(
        "Browser notification not supported, using server setting",
        {
          hasWebPushEnabled: o.userAccountMetadata.hasWebPushEnabled
        }
      ), A(o.userAccountMetadata.hasWebPushEnabled)), e.groupEnd();
    }).catch((o) => {
      e.error("Failed to fetch user account metadata", o), e.groupEnd();
    });
  }, [p, e]), ce(() => {
    if (e.group("Handling webPushOptInMessage state"), e.log("webPushOptInMessage value", R), R === "AUTOMATIC") {
      const o = localStorage.getItem("hideWebPushOptInMessage") === "true";
      e.log("Automatic mode - checking localStorage", {
        hideMessage: o,
        shouldShow: !o
      }), A(!o);
    }
    e.groupEnd();
  }, [R, e]), ce(() => {
    e.group("Handling webPushOptIn state"), e.log("webPushOptIn state", j), j && (e.log("User opted in for web push, requesting permission"), T()), e.groupEnd();
  }, [j, T, e]);
  const W = {
    notifications: l,
    preferences: d,
    userAccountMetaData: k,
    webPushOptInMessage: R,
    loadNotifications: X,
    markAsOpened: ee,
    markAsArchived: oe,
    markAsUnarchived: te,
    markAsClicked: Z,
    updateDelivery: ue,
    updateDeliveries: ne,
    getClient: () => p,
    setWebPushOptInMessage: A,
    setWebPushOptIn: _
  };
  e.log("NotificationAPI Provider rendering", {
    notificationsCount: (l == null ? void 0 : l.length) || 0,
    hasPreferences: !!d,
    hasUserAccountMetaData: !!k,
    webPushOptInMessage: R,
    webPushOptIn: j
  });
  const fe = pe(() => t.theme ? Oe(t.theme) : Oe("light"), [t.theme]);
  return /* @__PURE__ */ M(ht, { theme: fe, children: /* @__PURE__ */ M(je.Provider, { value: W, children: t.children }) });
}, St = () => {
  const t = Ge(je);
  if (!t)
    throw new Error("useMyContext must be used within a MyProvider");
  return t;
};
pt.useNotificationAPIContext = St;
const yt = async () => {
  if (Fe && "Notification" in window && typeof Notification.requestPermission == "function")
    try {
      return await Notification.requestPermission();
    } catch (t) {
      return console.error("Error requesting notification permission:", t), "default";
    }
  else
    return console.warn("Web Push Notifications are not supported in this browser."), "default";
};
export {
  pt as NotificationAPIProvider
};
