// @notificationapi/core/dist/main.js
const w = async (t, o, n, i, s, r, g, c) => {
  const h = I(i, s, r), a = `https://${o}/${i}/users/${encodeURIComponent(
    s
  )}/${n}`, d = {
    Authorization: `Basic ${h}`
  };
  c && c.log("HTTP Request:", {
    method: t,
    host: o,
    url: a,
    body: g
  });
  const l = Date.now();
  try {
    const u = await fetch(a, {
      method: t,
      body: JSON.stringify(g),
      headers: d
    });
    c && c.log("HTTP Response:", u);
    try {
      const f = await u.json();
      return c && c.log("Response Data:", f), f;
    } catch (f) {
      c && c.warn("Failed to parse response as JSON:", f);
      return;
    }
  } catch (u) {
    const f = Date.now() - l;
    throw c && c.error("HTTP Request Failed:", {
      error: u,
      url: a,
      headers: d,
      duration: `${f}ms`
    }), u;
  }
}, I = (t, o, n) => btoa(n ? t + ":" + o + ":" + n : t + ":" + o);
class p {
  constructor(o = !1) {
    this.debug = o;
  }
  log(...o) {
    this.debug && console.log("[NotificationAPI js core SDK Debug]", ...o);
  }
  warn(...o) {
    this.debug && console.warn("[NotificationAPI js core SDK Debug]", ...o);
  }
  error(...o) {
    this.debug && console.error("[NotificationAPI js core SDK Debug]", ...o);
  }
}
const b = {
  host: "api.notificationapi.com",
  websocketHost: "ws.notificationapi.com",
  userId: "",
  clientId: "",
  hashedUserId: "",
  getInAppDefaultCount: 100,
  getInAppDefaultOldest: new Date(
    Date.now() - 30 * 24 * 60 * 60 * 1e3
  ).toISOString(),
  onNewInAppNotifications: void 0,
  keepWebSocketAliveForSeconds: 24 * 60 * 60,
  // 24 hours
  debug: !1,
  sdkDevMode: !1
}, e = {
  config: b,
  logger: new p(!1),
  init: function(t) {
    return this.config = { ...b, ...t }, this.logger = new p(this.config.debug), this.logger.log("Initialized with config:", {
      userId: this.config.userId,
      clientId: this.config.clientId,
      host: this.config.host,
      websocketHost: this.config.websocketHost,
      debug: this.config.debug,
      hasHashedUserId: !!this.config.hashedUserId
    }), {
      ...this
    };
  },
  rest: {
    generic: function(t, o, n) {
      return e.logger.log(
        `API Call: ${t} ${o}`,
        n ? { body: n } : ""
      ), w(
        t,
        e.config.host,
        o,
        e.config.clientId,
        e.config.userId,
        e.config.hashedUserId,
        n,
        e.logger
      );
    },
    // The functions below are nice wrappers over the generic
    // rest  api function above. They must follow REST API naming:
    // Method + Resource, representing the end-point.
    getNotifications: function(t, o) {
      return e.rest.generic(
        "GET",
        `notifications/INAPP_WEB?count=${o}&before=${t}`
      );
    },
    patchNotifications: function(t) {
      return e.rest.generic(
        "PATCH",
        "notifications/INAPP_WEB",
        t
      );
    },
    getPreferences: function() {
      return e.rest.generic("GET", "preferences");
    },
    postPreferences: function(t) {
      return e.rest.generic(
        "POST",
        "preferences",
        t
      );
    },
    postUser: function(t) {
      return e.rest.generic("POST", "", t);
    },
    getUserAccountMetadata: function() {
      return e.rest.generic("GET", "account_metadata");
    }
  },
  websocket: {
    object: void 0,
    connect: function() {
      let t = `wss://${e.config.websocketHost}?userId=${encodeURIComponent(e.config.userId)}&envId=${e.config.clientId}`;
      return e.config.hashedUserId && (t += `&userIdHash=${encodeURIComponent(e.config.hashedUserId)}`), e.logger.log("WebSocket connecting to:", t), e.websocket.object = new WebSocket(t), e.websocket.object.onopen = () => {
        e.logger.log("WebSocket connection opened");
      }, e.websocket.object.onclose = (o) => {
        e.logger.log("WebSocket connection closed:", {
          code: o.code,
          reason: o.reason,
          wasClean: o.wasClean
        });
      }, e.websocket.object.onerror = (o) => {
        e.logger.error("WebSocket error:", o);
      }, e.websocket.object.onmessage = (o) => {
        e.logger.log(
          "WebSocket message received:",
          o.data
        );
        const n = JSON.parse(o.data);
        if (!(!n || !n.route) && n.route === "inapp_web/new_notifications") {
          const i = n;
          e.logger.log(
            "New notifications received:",
            i.payload.notifications
          ), e.config.onNewInAppNotifications && e.config.onNewInAppNotifications(
            i.payload.notifications
          );
        }
      }, e.websocket.object;
    },
    disconnect: function(t) {
      e.websocket.object && (e.logger.log("WebSocket disconnecting"), e.websocket.object?.close(), t && t(e.websocket.object));
    }
  },
  openWebSocket: function() {
    return e.websocket.connect(() => {
      setTimeout(
        () => {
          this.websocket.disconnect(() => {
            this.websocket.connect();
          });
        },
        54e4
      );
    });
  },
  // These functions are developer friendly wrappers over the rest APIs
  // They may or may not do additional tasks.
  // e.g. identify simply maps to postUsers
  getInAppNotifications: async (t) => {
    e.logger.log(
      "getInAppNotifications called with params:",
      t
    );
    const o = t.maxCountNeeded || e.config.getInAppDefaultCount, n = t.oldestNeeded || e.config.getInAppDefaultOldest;
    e.logger.log("Fetching notifications with:", {
      maxCountNeeded: o,
      oldestNeeded: n,
      before: t.before
    });
    let i = [], s = t.before, r = !0, g = !0;
    for (; g; ) {
      const h = (await e.rest.getNotifications(
        s,
        o
      )).notifications, a = h.filter(
        (d) => !i.find((l) => l.id === d.id)
      );
      e.logger.log(
        `Received ${h.length} notifications, ${a.length} unique`
      ), s = a.reduce(
        (d, l) => d < l.date ? d : l.date,
        t.before
      ), i = [...i, ...a], r = a.length > 0, g = !0, (!r || i.length >= o || s < n) && (g = !1, e.logger.log("Stopping fetch loop:", {
        hasMore: r,
        totalResults: i.length,
        maxCountNeeded: o,
        oldestReceived: s,
        oldestNeeded: n
      }));
    }
    return e.logger.log("getInAppNotifications completed:", {
      totalItems: i.length,
      hasMore: r,
      oldestReceived: s
    }), {
      items: i,
      hasMore: r,
      oldestReceived: s
    };
  },
  updateInAppNotifications: async (t) => {
    e.logger.log(
      "updateInAppNotifications called with params:",
      t
    );
    const o = {
      trackingIds: t.ids
    };
    return t.archived === !0 ? o.archived = (/* @__PURE__ */ new Date()).toISOString() : t.archived === !1 && (o.archived = null), t.clicked === !0 ? o.clicked = (/* @__PURE__ */ new Date()).toISOString() : t.clicked === !1 && (o.clicked = null), t.opened === !0 ? o.opened = (/* @__PURE__ */ new Date()).toISOString() : t.opened === !1 && (o.opened = null), e.logger.log(
      "Updating notifications with body:",
      o
    ), e.rest.patchNotifications(o);
  },
  getPreferences: async () => e.rest.getPreferences(),
  updateDeliveryOption: async (t) => e.rest.postPreferences([t]),
  identify: async (t) => {
    if (t.id && t.id !== e.config.userId)
      throw new Error(
        "The id in the parameters does not match the initialized userId."
      );
    return e.rest.postUser(t);
  },
  getUserAccountMetadata: async () => e.rest.getUserAccountMetadata(),
  user: {
    get: async () => e.rest.generic("GET", "")
  },
  slack: {
    getOAuthUrl: (t) => {
      const n = e.config.sdkDevMode ? "localhost:3001" : e.config.host.replace("api.", "app."), i = t?.destinationUrl || window.location.href, s = encodeURIComponent(
        JSON.stringify({
          destination: i,
          userId: e.config.userId,
          clientId: e.config.clientId,
          hashedUserId: e.config.hashedUserId
        })
      );
      return `https://slack.com/oauth/v2/authorize?client_id=1146598856352.8825220259395&scope=chat:write,channels:read,channels:join,chat:write.customize,chat:write.public,groups:read,im:read,users:read&redirect_uri=https://${n}/slack/oauth/callback&state=${s}`;
    },
    getChannels: async () => await e.rest.generic(
      "GET",
      "slack/channels"
    ),
    setChannel: async (t) => e.identify({
      slackChannel: t
    })
  }
};
export {
  e as NotificationAPIClientSDK
};
