// @notificationapi/react/dist/components/Notifications/Inbox.js
import { jsx as j, jsxs as $e } from "react/jsx-runtime";
import * as d from "react";
import dr, { isValidElement as Qr, useRef as D, useEffect as en, useState as Ye, useContext as tn } from "react";
import { InboxHeader as Bt } from "./InboxHeader.js";
import { L as rn, N as Wt } from "../../assets/Notification.js";
import { Filter as Yt } from "./interface.js";
import { NotificationAPIContext as nn } from "../Provider/context.js";
import { DefaultEmptyComponent as on } from "./DefaultEmpty.js";
import { g as ze, s as he, u as Je, c as _e, a as Qe, m as et, _ as Ct } from "../../assets/DefaultPropsProvider.js";
import { c as an, e as sn, B as cn, b as We, g as ln, j as un, k as fn, u as dn } from "../../assets/ButtonBase.js";
import { g as tt, P as c, a as we, e as xe, q as vn } from "../../assets/createTheme.js";
import { R as Gt, a as Ut } from "../../assets/utils.js";
import { getThemeColors as pn } from "../../utils/theme.js";
import { b as Ge, a as vr, i as Xt, L as Kt } from "../../assets/List.js";
import { u as hn } from "../../assets/resolveComponentProps.js";
import { a as mn } from "../../assets/dividerClasses.js";
import { i as Ae } from "../../assets/integerPropType.js";
import { u as gn } from "../../assets/index.js";
import { c as qt } from "../../assets/createSimplePaletteValueFilter.js";
import { c as rt } from "../../assets/createSvgIcon.js";
function bn(e) {
  return tt("MuiListItem", e);
}
ze("MuiListItem", ["root", "container", "dense", "alignItemsFlexStart", "divider", "gutters", "padding", "secondaryAction"]);
const yn = ze("MuiListItemButton", ["root", "focusVisible", "dense", "alignItemsFlexStart", "disabled", "divider", "gutters", "selected"]);
function Sn(e) {
  return tt("MuiListItemSecondaryAction", e);
}
ze("MuiListItemSecondaryAction", ["root", "disableGutters"]);
const Rn = (e) => {
  const {
    disableGutters: t,
    classes: r
  } = e;
  return Qe({
    root: ["root", t && "disableGutters"]
  }, Sn, r);
}, En = he("div", {
  name: "MuiListItemSecondaryAction",
  slot: "Root",
  overridesResolver: (e, t) => {
    const {
      ownerState: r
    } = e;
    return [t.root, r.disableGutters && t.disableGutters];
  }
})({
  position: "absolute",
  right: 16,
  top: "50%",
  transform: "translateY(-50%)",
  variants: [{
    props: ({
      ownerState: e
    }) => e.disableGutters,
    style: {
      right: 0
    }
  }]
}), Ot = /* @__PURE__ */ d.forwardRef(function(t, r) {
  const n = Je({
    props: t,
    name: "MuiListItemSecondaryAction"
  }), {
    className: o,
    ...i
  } = n, a = d.useContext(Ge), l = {
    ...n,
    disableGutters: a.disableGutters
  }, u = Rn(l);
  return /* @__PURE__ */ j(En, {
    className: _e(u.root, o),
    ownerState: l,
    ref: r,
    ...i
  });
});
process.env.NODE_ENV !== "production" && (Ot.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component, normally an `IconButton` or selection control.
   */
  children: c.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: c.object,
  /**
   * @ignore
   */
  className: c.string,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: c.oneOfType([c.arrayOf(c.oneOfType([c.func, c.object, c.bool])), c.func, c.object])
});
Ot.muiName = "ListItemSecondaryAction";
const wn = (e, t) => {
  const {
    ownerState: r
  } = e;
  return [t.root, r.dense && t.dense, r.alignItems === "flex-start" && t.alignItemsFlexStart, r.divider && t.divider, !r.disableGutters && t.gutters, !r.disablePadding && t.padding, r.hasSecondaryAction && t.secondaryAction];
}, xn = (e) => {
  const {
    alignItems: t,
    classes: r,
    dense: n,
    disableGutters: o,
    disablePadding: i,
    divider: a,
    hasSecondaryAction: l
  } = e;
  return Qe({
    root: ["root", n && "dense", !o && "gutters", !i && "padding", a && "divider", t === "flex-start" && "alignItemsFlexStart", l && "secondaryAction"],
    container: ["container"]
  }, bn, r);
}, _n = he("div", {
  name: "MuiListItem",
  slot: "Root",
  overridesResolver: wn
})(et(({
  theme: e
}) => ({
  display: "flex",
  justifyContent: "flex-start",
  alignItems: "center",
  position: "relative",
  textDecoration: "none",
  width: "100%",
  boxSizing: "border-box",
  textAlign: "left",
  variants: [{
    props: ({
      ownerState: t
    }) => !t.disablePadding,
    style: {
      paddingTop: 8,
      paddingBottom: 8
    }
  }, {
    props: ({
      ownerState: t
    }) => !t.disablePadding && t.dense,
    style: {
      paddingTop: 4,
      paddingBottom: 4
    }
  }, {
    props: ({
      ownerState: t
    }) => !t.disablePadding && !t.disableGutters,
    style: {
      paddingLeft: 16,
      paddingRight: 16
    }
  }, {
    props: ({
      ownerState: t
    }) => !t.disablePadding && !!t.secondaryAction,
    style: {
      // Add some space to avoid collision as `ListItemSecondaryAction`
      // is absolutely positioned.
      paddingRight: 48
    }
  }, {
    props: ({
      ownerState: t
    }) => !!t.secondaryAction,
    style: {
      [`& > .${yn.root}`]: {
        paddingRight: 48
      }
    }
  }, {
    props: {
      alignItems: "flex-start"
    },
    style: {
      alignItems: "flex-start"
    }
  }, {
    props: ({
      ownerState: t
    }) => t.divider,
    style: {
      borderBottom: `1px solid ${(e.vars || e).palette.divider}`,
      backgroundClip: "padding-box"
    }
  }, {
    props: ({
      ownerState: t
    }) => t.button,
    style: {
      transition: e.transitions.create("background-color", {
        duration: e.transitions.duration.shortest
      }),
      "&:hover": {
        textDecoration: "none",
        backgroundColor: (e.vars || e).palette.action.hover,
        // Reset on touch devices, it doesn't add specificity
        "@media (hover: none)": {
          backgroundColor: "transparent"
        }
      }
    }
  }, {
    props: ({
      ownerState: t
    }) => t.hasSecondaryAction,
    style: {
      // Add some space to avoid collision as `ListItemSecondaryAction`
      // is absolutely positioned.
      paddingRight: 48
    }
  }]
}))), Cn = he("li", {
  name: "MuiListItem",
  slot: "Container",
  overridesResolver: (e, t) => t.container
})({
  position: "relative"
}), mt = /* @__PURE__ */ d.forwardRef(function(t, r) {
  const n = Je({
    props: t,
    name: "MuiListItem"
  }), {
    alignItems: o = "center",
    children: i,
    className: a,
    component: l,
    components: u = {},
    componentsProps: f = {},
    ContainerComponent: m = "li",
    ContainerProps: {
      className: v,
      ...y
    } = {},
    dense: g = !1,
    disableGutters: b = !1,
    disablePadding: p = !1,
    divider: s = !1,
    secondaryAction: x,
    slotProps: S = {},
    slots: P = {},
    ...h
  } = n, R = d.useContext(Ge), L = d.useMemo(() => ({
    dense: g || R.dense || !1,
    alignItems: o,
    disableGutters: b
  }), [o, R.dense, g, b]), O = d.useRef(null), T = d.Children.toArray(i), _ = T.length && vr(T[T.length - 1], ["ListItemSecondaryAction"]), M = {
    ...n,
    alignItems: o,
    dense: L.dense,
    disableGutters: b,
    disablePadding: p,
    divider: s,
    hasSecondaryAction: _
  }, F = xn(M), Y = hn(O, r), B = P.root || u.Root || _n, $ = S.root || f.root || {}, G = {
    className: _e(F.root, $.className, a),
    ...h
  };
  let k = l || "li";
  return _ ? (k = !G.component && !l ? "div" : k, m === "li" && (k === "li" ? k = "div" : G.component === "li" && (G.component = "div")), /* @__PURE__ */ j(Ge.Provider, {
    value: L,
    children: /* @__PURE__ */ $e(Cn, {
      as: m,
      className: _e(F.container, v),
      ref: Y,
      ownerState: M,
      ...y,
      children: [/* @__PURE__ */ j(B, {
        ...$,
        ...!Xt(B) && {
          as: k,
          ownerState: {
            ...M,
            ...$.ownerState
          }
        },
        ...G,
        children: T
      }), T.pop()]
    })
  })) : /* @__PURE__ */ j(Ge.Provider, {
    value: L,
    children: /* @__PURE__ */ $e(B, {
      ...$,
      as: k,
      ref: Y,
      ...!Xt(B) && {
        ownerState: {
          ...M,
          ...$.ownerState
        }
      },
      ...G,
      children: [T, x && /* @__PURE__ */ j(Ot, {
        children: x
      })]
    })
  });
});
process.env.NODE_ENV !== "production" && (mt.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * Defines the `align-items` style property.
   * @default 'center'
   */
  alignItems: c.oneOf(["center", "flex-start"]),
  /**
   * The content of the component if a `ListItemSecondaryAction` is used it must
   * be the last child.
   */
  children: an(c.node, (e) => {
    const t = d.Children.toArray(e.children);
    let r = -1;
    for (let n = t.length - 1; n >= 0; n -= 1) {
      const o = t[n];
      if (vr(o, ["ListItemSecondaryAction"])) {
        r = n;
        break;
      }
    }
    return r !== -1 && r !== t.length - 1 ? new Error("MUI: You used an element after ListItemSecondaryAction. For ListItem to detect that it has a secondary action you must pass it as the last child to ListItem.") : null;
  }),
  /**
   * Override or extend the styles applied to the component.
   */
  classes: c.object,
  /**
   * @ignore
   */
  className: c.string,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: c.elementType,
  /**
   * The components used for each slot inside.
   *
   * @deprecated Use the `slots` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   * @default {}
   */
  components: c.shape({
    Root: c.elementType
  }),
  /**
   * The extra props for the slot components.
   * You can override the existing props or add new ones.
   *
   * @deprecated Use the `slotProps` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   * @default {}
   */
  componentsProps: c.shape({
    root: c.object
  }),
  /**
   * The container component used when a `ListItemSecondaryAction` is the last child.
   * @default 'li'
   * @deprecated Use the `component` or `slots.root` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   */
  ContainerComponent: sn,
  /**
   * Props applied to the container component if used.
   * @default {}
   * @deprecated Use the `slotProps.root` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   */
  ContainerProps: c.object,
  /**
   * If `true`, compact vertical padding designed for keyboard and mouse input is used.
   * The prop defaults to the value inherited from the parent List component.
   * @default false
   */
  dense: c.bool,
  /**
   * If `true`, the left and right padding is removed.
   * @default false
   */
  disableGutters: c.bool,
  /**
   * If `true`, all padding is removed.
   * @default false
   */
  disablePadding: c.bool,
  /**
   * If `true`, a 1px light border is added to the bottom of the list item.
   * @default false
   */
  divider: c.bool,
  /**
   * The element to display at the end of ListItem.
   */
  secondaryAction: c.node,
  /**
   * The extra props for the slot components.
   * You can override the existing props or add new ones.
   *
   * @default {}
   */
  slotProps: c.shape({
    root: c.object
  }),
  /**
   * The components used for each slot inside.
   *
   * @default {}
   */
  slots: c.shape({
    root: c.elementType
  }),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: c.oneOfType([c.arrayOf(c.oneOfType([c.func, c.object, c.bool])), c.func, c.object])
});
function On(e) {
  return tt("MuiPagination", e);
}
ze("MuiPagination", ["root", "ul", "outlined", "text"]);
function Pn(e = {}) {
  const {
    boundaryCount: t = 1,
    componentName: r = "usePagination",
    count: n = 1,
    defaultPage: o = 1,
    disabled: i = !1,
    hideNextButton: a = !1,
    hidePrevButton: l = !1,
    onChange: u,
    page: f,
    showFirstButton: m = !1,
    showLastButton: v = !1,
    siblingCount: y = 1,
    ...g
  } = e, [b, p] = mn({
    controlled: f,
    default: o,
    name: r,
    state: "page"
  }), s = (_, M) => {
    f || p(M), u && u(_, M);
  }, x = (_, M) => {
    const F = M - _ + 1;
    return Array.from({
      length: F
    }, (Y, B) => _ + B);
  }, S = x(1, Math.min(t, n)), P = x(Math.max(n - t + 1, t + 1), n), h = Math.max(
    Math.min(
      // Natural start
      b - y,
      // Lower boundary when page is high
      n - t - y * 2 - 1
    ),
    // Greater than startPages
    t + 2
  ), R = Math.min(
    Math.max(
      // Natural end
      b + y,
      // Upper boundary when page is low
      t + y * 2 + 2
    ),
    // Less than endPages
    n - t - 1
  ), L = [
    ...m ? ["first"] : [],
    ...l ? [] : ["previous"],
    ...S,
    // Start ellipsis
    // eslint-disable-next-line no-nested-ternary
    ...h > t + 2 ? ["start-ellipsis"] : t + 1 < n - t ? [t + 1] : [],
    // Sibling pages
    ...x(h, R),
    // End ellipsis
    // eslint-disable-next-line no-nested-ternary
    ...R < n - t - 1 ? ["end-ellipsis"] : n - t > t ? [n - t] : [],
    ...P,
    ...a ? [] : ["next"],
    ...v ? ["last"] : []
  ], O = (_) => {
    switch (_) {
      case "first":
        return 1;
      case "previous":
        return b - 1;
      case "next":
        return b + 1;
      case "last":
        return n;
      default:
        return null;
    }
  };
  return {
    items: L.map((_) => typeof _ == "number" ? {
      onClick: (M) => {
        s(M, _);
      },
      type: "page",
      page: _,
      selected: _ === b,
      disabled: i,
      "aria-current": _ === b ? "page" : void 0
    } : {
      onClick: (M) => {
        s(M, O(_));
      },
      type: _,
      page: O(_),
      selected: !1,
      disabled: i || !_.includes("ellipsis") && (_ === "next" || _ === "last" ? b >= n : b <= 1)
    }),
    ...g
  };
}
function Mn(e) {
  return tt("MuiPaginationItem", e);
}
const Q = ze("MuiPaginationItem", ["root", "page", "sizeSmall", "sizeLarge", "text", "textPrimary", "textSecondary", "outlined", "outlinedPrimary", "outlinedSecondary", "rounded", "ellipsis", "firstLast", "previousNext", "focusVisible", "disabled", "selected", "icon", "colorPrimary", "colorSecondary"]), Tn = rt(/* @__PURE__ */ j("path", {
  d: "M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
}), "FirstPage"), In = rt(/* @__PURE__ */ j("path", {
  d: "M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
}), "LastPage"), Ln = rt(/* @__PURE__ */ j("path", {
  d: "M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
}), "NavigateBefore"), An = rt(/* @__PURE__ */ j("path", {
  d: "M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
}), "NavigateNext"), pr = (e, t) => {
  const {
    ownerState: r
  } = e;
  return [t.root, t[r.variant], t[`size${xe(r.size)}`], r.variant === "text" && t[`text${xe(r.color)}`], r.variant === "outlined" && t[`outlined${xe(r.color)}`], r.shape === "rounded" && t.rounded, r.type === "page" && t.page, (r.type === "start-ellipsis" || r.type === "end-ellipsis") && t.ellipsis, (r.type === "previous" || r.type === "next") && t.previousNext, (r.type === "first" || r.type === "last") && t.firstLast];
}, Nn = (e) => {
  const {
    classes: t,
    color: r,
    disabled: n,
    selected: o,
    size: i,
    shape: a,
    type: l,
    variant: u
  } = e, f = {
    root: ["root", `size${xe(i)}`, u, a, r !== "standard" && `color${xe(r)}`, r !== "standard" && `${u}${xe(r)}`, n && "disabled", o && "selected", {
      page: "page",
      first: "firstLast",
      last: "firstLast",
      "start-ellipsis": "ellipsis",
      "end-ellipsis": "ellipsis",
      previous: "previousNext",
      next: "previousNext"
    }[l]],
    icon: ["icon"]
  };
  return Qe(f, Mn, t);
}, $n = he("div", {
  name: "MuiPaginationItem",
  slot: "Root",
  overridesResolver: pr
})(et(({
  theme: e
}) => ({
  ...e.typography.body2,
  borderRadius: 32 / 2,
  textAlign: "center",
  boxSizing: "border-box",
  minWidth: 32,
  padding: "0 6px",
  margin: "0 3px",
  color: (e.vars || e).palette.text.primary,
  height: "auto",
  [`&.${Q.disabled}`]: {
    opacity: (e.vars || e).palette.action.disabledOpacity
  },
  variants: [{
    props: {
      size: "small"
    },
    style: {
      minWidth: 26,
      borderRadius: 26 / 2,
      margin: "0 1px",
      padding: "0 4px"
    }
  }, {
    props: {
      size: "large"
    },
    style: {
      minWidth: 40,
      borderRadius: 40 / 2,
      padding: "0 10px",
      fontSize: e.typography.pxToRem(15)
    }
  }]
}))), zn = he(cn, {
  name: "MuiPaginationItem",
  slot: "Root",
  overridesResolver: pr
})(et(({
  theme: e
}) => ({
  ...e.typography.body2,
  borderRadius: 32 / 2,
  textAlign: "center",
  boxSizing: "border-box",
  minWidth: 32,
  height: 32,
  padding: "0 6px",
  margin: "0 3px",
  color: (e.vars || e).palette.text.primary,
  [`&.${Q.focusVisible}`]: {
    backgroundColor: (e.vars || e).palette.action.focus
  },
  [`&.${Q.disabled}`]: {
    opacity: (e.vars || e).palette.action.disabledOpacity
  },
  transition: e.transitions.create(["color", "background-color"], {
    duration: e.transitions.duration.short
  }),
  "&:hover": {
    backgroundColor: (e.vars || e).palette.action.hover,
    // Reset on touch devices, it doesn't add specificity
    "@media (hover: none)": {
      backgroundColor: "transparent"
    }
  },
  [`&.${Q.selected}`]: {
    backgroundColor: (e.vars || e).palette.action.selected,
    "&:hover": {
      backgroundColor: e.vars ? `rgba(${e.vars.palette.action.selectedChannel} / calc(${e.vars.palette.action.selectedOpacity} + ${e.vars.palette.action.hoverOpacity}))` : we(e.palette.action.selected, e.palette.action.selectedOpacity + e.palette.action.hoverOpacity),
      // Reset on touch devices, it doesn't add specificity
      "@media (hover: none)": {
        backgroundColor: (e.vars || e).palette.action.selected
      }
    },
    [`&.${Q.focusVisible}`]: {
      backgroundColor: e.vars ? `rgba(${e.vars.palette.action.selectedChannel} / calc(${e.vars.palette.action.selectedOpacity} + ${e.vars.palette.action.focusOpacity}))` : we(e.palette.action.selected, e.palette.action.selectedOpacity + e.palette.action.focusOpacity)
    },
    [`&.${Q.disabled}`]: {
      opacity: 1,
      color: (e.vars || e).palette.action.disabled,
      backgroundColor: (e.vars || e).palette.action.selected
    }
  },
  variants: [{
    props: {
      size: "small"
    },
    style: {
      minWidth: 26,
      height: 26,
      borderRadius: 26 / 2,
      margin: "0 1px",
      padding: "0 4px"
    }
  }, {
    props: {
      size: "large"
    },
    style: {
      minWidth: 40,
      height: 40,
      borderRadius: 40 / 2,
      padding: "0 10px",
      fontSize: e.typography.pxToRem(15)
    }
  }, {
    props: {
      shape: "rounded"
    },
    style: {
      borderRadius: (e.vars || e).shape.borderRadius
    }
  }, {
    props: {
      variant: "outlined"
    },
    style: {
      border: e.vars ? `1px solid rgba(${e.vars.palette.common.onBackgroundChannel} / 0.23)` : `1px solid ${e.palette.mode === "light" ? "rgba(0, 0, 0, 0.23)" : "rgba(255, 255, 255, 0.23)"}`,
      [`&.${Q.selected}`]: {
        [`&.${Q.disabled}`]: {
          borderColor: (e.vars || e).palette.action.disabledBackground,
          color: (e.vars || e).palette.action.disabled
        }
      }
    }
  }, {
    props: {
      variant: "text"
    },
    style: {
      [`&.${Q.selected}`]: {
        [`&.${Q.disabled}`]: {
          color: (e.vars || e).palette.action.disabled
        }
      }
    }
  }, ...Object.entries(e.palette).filter(qt(["dark", "contrastText"])).map(([t]) => ({
    props: {
      variant: "text",
      color: t
    },
    style: {
      [`&.${Q.selected}`]: {
        color: (e.vars || e).palette[t].contrastText,
        backgroundColor: (e.vars || e).palette[t].main,
        "&:hover": {
          backgroundColor: (e.vars || e).palette[t].dark,
          // Reset on touch devices, it doesn't add specificity
          "@media (hover: none)": {
            backgroundColor: (e.vars || e).palette[t].main
          }
        },
        [`&.${Q.focusVisible}`]: {
          backgroundColor: (e.vars || e).palette[t].dark
        },
        [`&.${Q.disabled}`]: {
          color: (e.vars || e).palette.action.disabled
        }
      }
    }
  })), ...Object.entries(e.palette).filter(qt(["light"])).map(([t]) => ({
    props: {
      variant: "outlined",
      color: t
    },
    style: {
      [`&.${Q.selected}`]: {
        color: (e.vars || e).palette[t].main,
        border: `1px solid ${e.vars ? `rgba(${e.vars.palette[t].mainChannel} / 0.5)` : we(e.palette[t].main, 0.5)}`,
        backgroundColor: e.vars ? `rgba(${e.vars.palette[t].mainChannel} / ${e.vars.palette.action.activatedOpacity})` : we(e.palette[t].main, e.palette.action.activatedOpacity),
        "&:hover": {
          backgroundColor: e.vars ? `rgba(${e.vars.palette[t].mainChannel} / calc(${e.vars.palette.action.activatedOpacity} + ${e.vars.palette.action.focusOpacity}))` : we(e.palette[t].main, e.palette.action.activatedOpacity + e.palette.action.focusOpacity),
          // Reset on touch devices, it doesn't add specificity
          "@media (hover: none)": {
            backgroundColor: "transparent"
          }
        },
        [`&.${Q.focusVisible}`]: {
          backgroundColor: e.vars ? `rgba(${e.vars.palette[t].mainChannel} / calc(${e.vars.palette.action.activatedOpacity} + ${e.vars.palette.action.focusOpacity}))` : we(e.palette[t].main, e.palette.action.activatedOpacity + e.palette.action.focusOpacity)
        }
      }
    }
  }))]
}))), Dn = he("div", {
  name: "MuiPaginationItem",
  slot: "Icon",
  overridesResolver: (e, t) => t.icon
})(et(({
  theme: e
}) => ({
  fontSize: e.typography.pxToRem(20),
  margin: "0 -8px",
  variants: [{
    props: {
      size: "small"
    },
    style: {
      fontSize: e.typography.pxToRem(18)
    }
  }, {
    props: {
      size: "large"
    },
    style: {
      fontSize: e.typography.pxToRem(22)
    }
  }]
}))), hr = /* @__PURE__ */ d.forwardRef(function(t, r) {
  const n = Je({
    props: t,
    name: "MuiPaginationItem"
  }), {
    className: o,
    color: i = "standard",
    component: a,
    components: l = {},
    disabled: u = !1,
    page: f,
    selected: m = !1,
    shape: v = "circular",
    size: y = "medium",
    slots: g = {},
    slotProps: b = {},
    type: p = "page",
    variant: s = "text",
    ...x
  } = n, S = {
    ...n,
    color: i,
    disabled: u,
    selected: m,
    shape: v,
    size: y,
    type: p,
    variant: s
  }, P = gn(), h = Nn(S), R = {
    slots: {
      previous: g.previous ?? l.previous,
      next: g.next ?? l.next,
      first: g.first ?? l.first,
      last: g.last ?? l.last
    },
    slotProps: b
  }, [L, O] = We("previous", {
    elementType: Ln,
    externalForwardedProps: R,
    ownerState: S
  }), [T, _] = We("next", {
    elementType: An,
    externalForwardedProps: R,
    ownerState: S
  }), [M, F] = We("first", {
    elementType: Tn,
    externalForwardedProps: R,
    ownerState: S
  }), [Y, B] = We("last", {
    elementType: In,
    externalForwardedProps: R,
    ownerState: S
  }), $ = P ? {
    previous: "next",
    next: "previous",
    first: "last",
    last: "first"
  }[p] : p, G = {
    previous: L,
    next: T,
    first: M,
    last: Y
  }[$], k = {
    previous: O,
    next: _,
    first: F,
    last: B
  }[$];
  return p === "start-ellipsis" || p === "end-ellipsis" ? /* @__PURE__ */ j($n, {
    ref: r,
    ownerState: S,
    className: _e(h.root, o),
    children: "…"
  }) : /* @__PURE__ */ $e(zn, {
    ref: r,
    ownerState: S,
    component: a,
    disabled: u,
    className: _e(h.root, o),
    ...x,
    children: [p === "page" && f, G ? /* @__PURE__ */ j(Dn, {
      ...k,
      className: h.icon,
      as: G
    }) : null]
  });
});
process.env.NODE_ENV !== "production" && (hr.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * @ignore
   */
  children: c.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: c.object,
  /**
   * @ignore
   */
  className: c.string,
  /**
   * The active color.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   * @default 'standard'
   */
  color: c.oneOfType([c.oneOf(["primary", "secondary", "standard"]), c.string]),
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: c.elementType,
  /**
   * The components used for each slot inside.
   *
   * This prop is an alias for the `slots` prop.
   * It's recommended to use the `slots` prop instead.
   *
   * @default {}
   * @deprecated use the `slots` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   */
  components: c.shape({
    first: c.elementType,
    last: c.elementType,
    next: c.elementType,
    previous: c.elementType
  }),
  /**
   * If `true`, the component is disabled.
   * @default false
   */
  disabled: c.bool,
  /**
   * The current page number.
   */
  page: c.node,
  /**
   * If `true` the pagination item is selected.
   * @default false
   */
  selected: c.bool,
  /**
   * The shape of the pagination item.
   * @default 'circular'
   */
  shape: c.oneOf(["circular", "rounded"]),
  /**
   * The size of the component.
   * @default 'medium'
   */
  size: c.oneOfType([c.oneOf(["small", "medium", "large"]), c.string]),
  /**
   * The props used for each slot inside.
   * @default {}
   */
  slotProps: c.shape({
    first: c.oneOfType([c.func, c.object]),
    last: c.oneOfType([c.func, c.object]),
    next: c.oneOfType([c.func, c.object]),
    previous: c.oneOfType([c.func, c.object])
  }),
  /**
   * The components used for each slot inside.
   * @default {}
   */
  slots: c.shape({
    first: c.elementType,
    last: c.elementType,
    next: c.elementType,
    previous: c.elementType
  }),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: c.oneOfType([c.arrayOf(c.oneOfType([c.func, c.object, c.bool])), c.func, c.object]),
  /**
   * The type of pagination item.
   * @default 'page'
   */
  type: c.oneOf(["end-ellipsis", "first", "last", "next", "page", "previous", "start-ellipsis"]),
  /**
   * The variant to use.
   * @default 'text'
   */
  variant: c.oneOfType([c.oneOf(["outlined", "text"]), c.string])
});
const kn = (e) => {
  const {
    classes: t,
    variant: r
  } = e;
  return Qe({
    root: ["root", r],
    ul: ["ul"]
  }, On, t);
}, jn = he("nav", {
  name: "MuiPagination",
  slot: "Root",
  overridesResolver: (e, t) => {
    const {
      ownerState: r
    } = e;
    return [t.root, t[r.variant]];
  }
})({}), Fn = he("ul", {
  name: "MuiPagination",
  slot: "Ul",
  overridesResolver: (e, t) => t.ul
})({
  display: "flex",
  flexWrap: "wrap",
  alignItems: "center",
  padding: 0,
  margin: 0,
  listStyle: "none"
});
function Hn(e, t, r) {
  return e === "page" ? `${r ? "" : "Go to "}page ${t}` : `Go to ${e} page`;
}
const gt = /* @__PURE__ */ d.forwardRef(function(t, r) {
  const n = Je({
    props: t,
    name: "MuiPagination"
  }), {
    boundaryCount: o = 1,
    className: i,
    color: a = "standard",
    count: l = 1,
    defaultPage: u = 1,
    disabled: f = !1,
    getItemAriaLabel: m = Hn,
    hideNextButton: v = !1,
    hidePrevButton: y = !1,
    onChange: g,
    page: b,
    renderItem: p = (M) => /* @__PURE__ */ j(hr, {
      ...M
    }),
    shape: s = "circular",
    showFirstButton: x = !1,
    showLastButton: S = !1,
    siblingCount: P = 1,
    size: h = "medium",
    variant: R = "text",
    ...L
  } = n, {
    items: O
  } = Pn({
    ...n,
    componentName: "Pagination"
  }), T = {
    ...n,
    boundaryCount: o,
    color: a,
    count: l,
    defaultPage: u,
    disabled: f,
    getItemAriaLabel: m,
    hideNextButton: v,
    hidePrevButton: y,
    renderItem: p,
    shape: s,
    showFirstButton: x,
    showLastButton: S,
    siblingCount: P,
    size: h,
    variant: R
  }, _ = kn(T);
  return /* @__PURE__ */ j(jn, {
    "aria-label": "pagination navigation",
    className: _e(_.root, i),
    ownerState: T,
    ref: r,
    ...L,
    children: /* @__PURE__ */ j(Fn, {
      className: _.ul,
      ownerState: T,
      children: O.map((M, F) => /* @__PURE__ */ j("li", {
        children: p({
          ...M,
          color: a,
          "aria-label": m(M.type, M.page, M.selected),
          shape: s,
          size: h,
          variant: R
        })
      }, F))
    })
  });
});
process.env.NODE_ENV !== "production" && (gt.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * Number of always visible pages at the beginning and end.
   * @default 1
   */
  boundaryCount: Ae,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: c.object,
  /**
   * @ignore
   */
  className: c.string,
  /**
   * The active color.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   * @default 'standard'
   */
  color: c.oneOfType([c.oneOf(["primary", "secondary", "standard"]), c.string]),
  /**
   * The total number of pages.
   * @default 1
   */
  count: Ae,
  /**
   * The page selected by default when the component is uncontrolled.
   * @default 1
   */
  defaultPage: Ae,
  /**
   * If `true`, the component is disabled.
   * @default false
   */
  disabled: c.bool,
  /**
   * Accepts a function which returns a string value that provides a user-friendly name for the current page.
   * This is important for screen reader users.
   *
   * For localization purposes, you can use the provided [translations](https://mui.com/material-ui/guides/localization/).
   * @param {string} type The link or button type to format ('page' | 'first' | 'last' | 'next' | 'previous' | 'start-ellipsis' | 'end-ellipsis'). Defaults to 'page'.
   * @param {number | null} page The page number to format.
   * @param {boolean} selected If true, the current page is selected.
   * @returns {string}
   */
  getItemAriaLabel: c.func,
  /**
   * If `true`, hide the next-page button.
   * @default false
   */
  hideNextButton: c.bool,
  /**
   * If `true`, hide the previous-page button.
   * @default false
   */
  hidePrevButton: c.bool,
  /**
   * Callback fired when the page is changed.
   *
   * @param {React.ChangeEvent<unknown>} event The event source of the callback.
   * @param {number} page The page selected.
   */
  onChange: c.func,
  /**
   * The current page. Unlike `TablePagination`, which starts numbering from `0`, this pagination starts from `1`.
   */
  page: Ae,
  /**
   * Render the item.
   * @param {PaginationRenderItemParams} params The props to spread on a PaginationItem.
   * @returns {ReactNode}
   * @default (item) => <PaginationItem {...item} />
   */
  renderItem: c.func,
  /**
   * The shape of the pagination items.
   * @default 'circular'
   */
  shape: c.oneOf(["circular", "rounded"]),
  /**
   * If `true`, show the first-page button.
   * @default false
   */
  showFirstButton: c.bool,
  /**
   * If `true`, show the last-page button.
   * @default false
   */
  showLastButton: c.bool,
  /**
   * Number of always visible pages before and after the current page.
   * @default 1
   */
  siblingCount: Ae,
  /**
   * The size of the component.
   * @default 'medium'
   */
  size: c.oneOfType([c.oneOf(["small", "medium", "large"]), c.string]),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: c.oneOfType([c.arrayOf(c.oneOfType([c.func, c.object, c.bool])), c.func, c.object]),
  /**
   * The variant to use.
   * @default 'text'
   */
  variant: c.oneOfType([c.oneOf(["outlined", "text"]), c.string])
});
function oe(e) {
  "@babel/helpers - typeof";
  return oe = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(t) {
    return typeof t;
  } : function(t) {
    return t && typeof Symbol == "function" && t.constructor === Symbol && t !== Symbol.prototype ? "symbol" : typeof t;
  }, oe(e);
}
function Vn(e, t) {
  if (oe(e) != "object" || !e) return e;
  var r = e[Symbol.toPrimitive];
  if (r !== void 0) {
    var n = r.call(e, t || "default");
    if (oe(n) != "object") return n;
    throw new TypeError("@@toPrimitive must return a primitive value.");
  }
  return (t === "string" ? String : Number)(e);
}
function mr(e) {
  var t = Vn(e, "string");
  return oe(t) == "symbol" ? t : t + "";
}
function ee(e, t, r) {
  return (t = mr(t)) in e ? Object.defineProperty(e, t, {
    value: r,
    enumerable: !0,
    configurable: !0,
    writable: !0
  }) : e[t] = r, e;
}
function Zt(e, t) {
  var r = Object.keys(e);
  if (Object.getOwnPropertySymbols) {
    var n = Object.getOwnPropertySymbols(e);
    t && (n = n.filter(function(o) {
      return Object.getOwnPropertyDescriptor(e, o).enumerable;
    })), r.push.apply(r, n);
  }
  return r;
}
function q(e) {
  for (var t = 1; t < arguments.length; t++) {
    var r = arguments[t] != null ? arguments[t] : {};
    t % 2 ? Zt(Object(r), !0).forEach(function(n) {
      ee(e, n, r[n]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(r)) : Zt(Object(r)).forEach(function(n) {
      Object.defineProperty(e, n, Object.getOwnPropertyDescriptor(r, n));
    });
  }
  return e;
}
function Bn(e) {
  if (Array.isArray(e)) return e;
}
function Wn(e, t) {
  var r = e == null ? null : typeof Symbol < "u" && e[Symbol.iterator] || e["@@iterator"];
  if (r != null) {
    var n, o, i, a, l = [], u = !0, f = !1;
    try {
      if (i = (r = r.call(e)).next, t === 0) {
        if (Object(r) !== r) return;
        u = !1;
      } else for (; !(u = (n = i.call(r)).done) && (l.push(n.value), l.length !== t); u = !0) ;
    } catch (m) {
      f = !0, o = m;
    } finally {
      try {
        if (!u && r.return != null && (a = r.return(), Object(a) !== a)) return;
      } finally {
        if (f) throw o;
      }
    }
    return l;
  }
}
function Jt(e, t) {
  (t == null || t > e.length) && (t = e.length);
  for (var r = 0, n = Array(t); r < t; r++) n[r] = e[r];
  return n;
}
function Yn(e, t) {
  if (e) {
    if (typeof e == "string") return Jt(e, t);
    var r = {}.toString.call(e).slice(8, -1);
    return r === "Object" && e.constructor && (r = e.constructor.name), r === "Map" || r === "Set" ? Array.from(e) : r === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r) ? Jt(e, t) : void 0;
  }
}
function Gn() {
  throw new TypeError(`Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`);
}
function K(e, t) {
  return Bn(e) || Wn(e, t) || Yn(e, t) || Gn();
}
function Un(e, t) {
  if (e == null) return {};
  var r, n, o = ln(e, t);
  if (Object.getOwnPropertySymbols) {
    var i = Object.getOwnPropertySymbols(e);
    for (n = 0; n < i.length; n++) r = i[n], t.indexOf(r) === -1 && {}.propertyIsEnumerable.call(e, r) && (o[r] = e[r]);
  }
  return o;
}
var gr = { exports: {} };
/*!
	Copyright (c) 2018 Jed Watson.
	Licensed under the MIT License (MIT), see
	http://jedwatson.github.io/classnames
*/
(function(e) {
  (function() {
    var t = {}.hasOwnProperty;
    function r() {
      for (var i = "", a = 0; a < arguments.length; a++) {
        var l = arguments[a];
        l && (i = o(i, n(l)));
      }
      return i;
    }
    function n(i) {
      if (typeof i == "string" || typeof i == "number")
        return i;
      if (typeof i != "object")
        return "";
      if (Array.isArray(i))
        return r.apply(null, i);
      if (i.toString !== Object.prototype.toString && !i.toString.toString().includes("[native code]"))
        return i.toString();
      var a = "";
      for (var l in i)
        t.call(i, l) && i[l] && (a = o(a, l));
      return a;
    }
    function o(i, a) {
      return a ? i ? i + " " + a : i + a : i;
    }
    e.exports ? (r.default = r, e.exports = r) : window.classNames = r;
  })();
})(gr);
var Xn = gr.exports;
const Xe = /* @__PURE__ */ vn(Xn);
var Kn = Symbol.for("react.element"), qn = Symbol.for("react.transitional.element"), Zn = Symbol.for("react.fragment");
function br(e) {
  return (
    // Base object type
    e && oe(e) === "object" && // React Element type
    (e.$$typeof === Kn || e.$$typeof === qn) && // React Fragment type
    e.type === Zn
  );
}
function bt(e) {
  var t = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, r = [];
  return dr.Children.forEach(e, function(n) {
    n == null && !t.keepEmpty || (Array.isArray(n) ? r = r.concat(bt(n)) : br(n) && n.props ? r = r.concat(bt(n.props.children, t)) : r.push(n));
  }), r;
}
var yt = {}, Pt = [], Jn = function(t) {
  Pt.push(t);
};
function St(e, t) {
  if (process.env.NODE_ENV !== "production" && !e && console !== void 0) {
    var r = Pt.reduce(function(n, o) {
      return o(n ?? "", "warning");
    }, t);
    r && console.error("Warning: ".concat(r));
  }
}
function Qn(e, t) {
  if (process.env.NODE_ENV !== "production" && !e && console !== void 0) {
    var r = Pt.reduce(function(n, o) {
      return o(n ?? "", "note");
    }, t);
    r && console.warn("Note: ".concat(r));
  }
}
function eo() {
  yt = {};
}
function yr(e, t, r) {
  !t && !yt[r] && (e(!1, r), yt[r] = !0);
}
function nt(e, t) {
  yr(St, e, t);
}
function to(e, t) {
  yr(Qn, e, t);
}
nt.preMessage = Jn;
nt.resetWarned = eo;
nt.noteOnce = to;
function Qt(e) {
  return e instanceof HTMLElement || e instanceof SVGElement;
}
function ro(e) {
  return e && oe(e) === "object" && Qt(e.nativeElement) ? e.nativeElement : Qt(e) ? e : null;
}
function Ue(e) {
  var t = ro(e);
  if (t)
    return t;
  if (e instanceof dr.Component) {
    var r;
    return (r = Gt.findDOMNode) === null || r === void 0 ? void 0 : r.call(Gt, e);
  }
  return null;
}
var Rt = { exports: {} }, A = {};
/**
 * @license React
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var er;
function no() {
  if (er) return A;
  er = 1;
  var e = Symbol.for("react.element"), t = Symbol.for("react.portal"), r = Symbol.for("react.fragment"), n = Symbol.for("react.strict_mode"), o = Symbol.for("react.profiler"), i = Symbol.for("react.provider"), a = Symbol.for("react.context"), l = Symbol.for("react.server_context"), u = Symbol.for("react.forward_ref"), f = Symbol.for("react.suspense"), m = Symbol.for("react.suspense_list"), v = Symbol.for("react.memo"), y = Symbol.for("react.lazy"), g = Symbol.for("react.offscreen"), b;
  b = Symbol.for("react.module.reference");
  function p(s) {
    if (typeof s == "object" && s !== null) {
      var x = s.$$typeof;
      switch (x) {
        case e:
          switch (s = s.type, s) {
            case r:
            case o:
            case n:
            case f:
            case m:
              return s;
            default:
              switch (s = s && s.$$typeof, s) {
                case l:
                case a:
                case u:
                case y:
                case v:
                case i:
                  return s;
                default:
                  return x;
              }
          }
        case t:
          return x;
      }
    }
  }
  return A.ContextConsumer = a, A.ContextProvider = i, A.Element = e, A.ForwardRef = u, A.Fragment = r, A.Lazy = y, A.Memo = v, A.Portal = t, A.Profiler = o, A.StrictMode = n, A.Suspense = f, A.SuspenseList = m, A.isAsyncMode = function() {
    return !1;
  }, A.isConcurrentMode = function() {
    return !1;
  }, A.isContextConsumer = function(s) {
    return p(s) === a;
  }, A.isContextProvider = function(s) {
    return p(s) === i;
  }, A.isElement = function(s) {
    return typeof s == "object" && s !== null && s.$$typeof === e;
  }, A.isForwardRef = function(s) {
    return p(s) === u;
  }, A.isFragment = function(s) {
    return p(s) === r;
  }, A.isLazy = function(s) {
    return p(s) === y;
  }, A.isMemo = function(s) {
    return p(s) === v;
  }, A.isPortal = function(s) {
    return p(s) === t;
  }, A.isProfiler = function(s) {
    return p(s) === o;
  }, A.isStrictMode = function(s) {
    return p(s) === n;
  }, A.isSuspense = function(s) {
    return p(s) === f;
  }, A.isSuspenseList = function(s) {
    return p(s) === m;
  }, A.isValidElementType = function(s) {
    return typeof s == "string" || typeof s == "function" || s === r || s === o || s === n || s === f || s === m || s === g || typeof s == "object" && s !== null && (s.$$typeof === y || s.$$typeof === v || s.$$typeof === i || s.$$typeof === a || s.$$typeof === u || s.$$typeof === b || s.getModuleId !== void 0);
  }, A.typeOf = p, A;
}
var N = {};
/**
 * @license React
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var tr;
function oo() {
  return tr || (tr = 1, process.env.NODE_ENV !== "production" && function() {
    var e = Symbol.for("react.element"), t = Symbol.for("react.portal"), r = Symbol.for("react.fragment"), n = Symbol.for("react.strict_mode"), o = Symbol.for("react.profiler"), i = Symbol.for("react.provider"), a = Symbol.for("react.context"), l = Symbol.for("react.server_context"), u = Symbol.for("react.forward_ref"), f = Symbol.for("react.suspense"), m = Symbol.for("react.suspense_list"), v = Symbol.for("react.memo"), y = Symbol.for("react.lazy"), g = Symbol.for("react.offscreen"), b = !1, p = !1, s = !1, x = !1, S = !1, P;
    P = Symbol.for("react.module.reference");
    function h(E) {
      return !!(typeof E == "string" || typeof E == "function" || E === r || E === o || S || E === n || E === f || E === m || x || E === g || b || p || s || typeof E == "object" && E !== null && (E.$$typeof === y || E.$$typeof === v || E.$$typeof === i || E.$$typeof === a || E.$$typeof === u || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      E.$$typeof === P || E.getModuleId !== void 0));
    }
    function R(E) {
      if (typeof E == "object" && E !== null) {
        var le = E.$$typeof;
        switch (le) {
          case e:
            var ue = E.type;
            switch (ue) {
              case r:
              case o:
              case n:
              case f:
              case m:
                return ue;
              default:
                var be = ue && ue.$$typeof;
                switch (be) {
                  case l:
                  case a:
                  case u:
                  case y:
                  case v:
                  case i:
                    return be;
                  default:
                    return le;
                }
            }
          case t:
            return le;
        }
      }
    }
    var L = a, O = i, T = e, _ = u, M = r, F = y, Y = v, B = t, $ = o, G = n, k = f, ae = m, Z = !1, W = !1;
    function me(E) {
      return Z || (Z = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 18+.")), !1;
    }
    function H(E) {
      return W || (W = !0, console.warn("The ReactIs.isConcurrentMode() alias has been deprecated, and will be removed in React 18+.")), !1;
    }
    function se(E) {
      return R(E) === a;
    }
    function ve(E) {
      return R(E) === i;
    }
    function pe(E) {
      return typeof E == "object" && E !== null && E.$$typeof === e;
    }
    function Se(E) {
      return R(E) === u;
    }
    function ge(E) {
      return R(E) === r;
    }
    function U(E) {
      return R(E) === y;
    }
    function te(E) {
      return R(E) === v;
    }
    function ie(E) {
      return R(E) === t;
    }
    function X(E) {
      return R(E) === o;
    }
    function z(E) {
      return R(E) === n;
    }
    function re(E) {
      return R(E) === f;
    }
    function Re(E) {
      return R(E) === m;
    }
    N.ContextConsumer = L, N.ContextProvider = O, N.Element = T, N.ForwardRef = _, N.Fragment = M, N.Lazy = F, N.Memo = Y, N.Portal = B, N.Profiler = $, N.StrictMode = G, N.Suspense = k, N.SuspenseList = ae, N.isAsyncMode = me, N.isConcurrentMode = H, N.isContextConsumer = se, N.isContextProvider = ve, N.isElement = pe, N.isForwardRef = Se, N.isFragment = ge, N.isLazy = U, N.isMemo = te, N.isPortal = ie, N.isProfiler = X, N.isStrictMode = z, N.isSuspense = re, N.isSuspenseList = Re, N.isValidElementType = h, N.typeOf = R;
  }()), N;
}
process.env.NODE_ENV === "production" ? Rt.exports = no() : Rt.exports = oo();
var ht = Rt.exports;
function io(e, t, r) {
  var n = d.useRef({});
  return (!("value" in n.current) || r(n.current.condition, t)) && (n.current.value = e(), n.current.condition = t), n.current.value;
}
var ao = function(t, r) {
  typeof t == "function" ? t(r) : oe(t) === "object" && t && "current" in t && (t.current = r);
}, so = function() {
  for (var t = arguments.length, r = new Array(t), n = 0; n < t; n++)
    r[n] = arguments[n];
  var o = r.filter(Boolean);
  return o.length <= 1 ? o[0] : function(i) {
    r.forEach(function(a) {
      ao(a, i);
    });
  };
}, co = function() {
  for (var t = arguments.length, r = new Array(t), n = 0; n < t; n++)
    r[n] = arguments[n];
  return io(function() {
    return so.apply(void 0, r);
  }, r, function(o, i) {
    return o.length !== i.length || o.every(function(a, l) {
      return a !== i[l];
    });
  });
}, lo = function(t) {
  var r, n;
  if (!t)
    return !1;
  if (Sr(t) && t.props.propertyIsEnumerable("ref"))
    return !0;
  var o = ht.isMemo(t) ? t.type.type : t.type;
  return !(typeof o == "function" && !((r = o.prototype) !== null && r !== void 0 && r.render) && o.$$typeof !== ht.ForwardRef || typeof t == "function" && !((n = t.prototype) !== null && n !== void 0 && n.render) && t.$$typeof !== ht.ForwardRef);
};
function Sr(e) {
  return /* @__PURE__ */ Qr(e) && !br(e);
}
var uo = function(t) {
  if (t && Sr(t)) {
    var r = t;
    return r.props.propertyIsEnumerable("ref") ? r.props.ref : r.ref;
  }
  return null;
}, Et = /* @__PURE__ */ d.createContext(null);
function fo(e) {
  var t = e.children, r = e.onBatchResize, n = d.useRef(0), o = d.useRef([]), i = d.useContext(Et), a = d.useCallback(function(l, u, f) {
    n.current += 1;
    var m = n.current;
    o.current.push({
      size: l,
      element: u,
      data: f
    }), Promise.resolve().then(function() {
      m === n.current && (r == null || r(o.current), o.current = []);
    }), i == null || i(l, u, f);
  }, [r, i]);
  return /* @__PURE__ */ d.createElement(Et.Provider, {
    value: a
  }, t);
}
var Rr = function() {
  if (typeof Map < "u")
    return Map;
  function e(t, r) {
    var n = -1;
    return t.some(function(o, i) {
      return o[0] === r ? (n = i, !0) : !1;
    }), n;
  }
  return (
    /** @class */
    function() {
      function t() {
        this.__entries__ = [];
      }
      return Object.defineProperty(t.prototype, "size", {
        /**
         * @returns {boolean}
         */
        get: function() {
          return this.__entries__.length;
        },
        enumerable: !0,
        configurable: !0
      }), t.prototype.get = function(r) {
        var n = e(this.__entries__, r), o = this.__entries__[n];
        return o && o[1];
      }, t.prototype.set = function(r, n) {
        var o = e(this.__entries__, r);
        ~o ? this.__entries__[o][1] = n : this.__entries__.push([r, n]);
      }, t.prototype.delete = function(r) {
        var n = this.__entries__, o = e(n, r);
        ~o && n.splice(o, 1);
      }, t.prototype.has = function(r) {
        return !!~e(this.__entries__, r);
      }, t.prototype.clear = function() {
        this.__entries__.splice(0);
      }, t.prototype.forEach = function(r, n) {
        n === void 0 && (n = null);
        for (var o = 0, i = this.__entries__; o < i.length; o++) {
          var a = i[o];
          r.call(n, a[1], a[0]);
        }
      }, t;
    }()
  );
}(), wt = typeof window < "u" && typeof document < "u" && window.document === document, Ke = function() {
  return typeof global < "u" && global.Math === Math ? global : typeof self < "u" && self.Math === Math ? self : typeof window < "u" && window.Math === Math ? window : Function("return this")();
}(), vo = function() {
  return typeof requestAnimationFrame == "function" ? requestAnimationFrame.bind(Ke) : function(e) {
    return setTimeout(function() {
      return e(Date.now());
    }, 1e3 / 60);
  };
}(), po = 2;
function ho(e, t) {
  var r = !1, n = !1, o = 0;
  function i() {
    r && (r = !1, e()), n && l();
  }
  function a() {
    vo(i);
  }
  function l() {
    var u = Date.now();
    if (r) {
      if (u - o < po)
        return;
      n = !0;
    } else
      r = !0, n = !1, setTimeout(a, t);
    o = u;
  }
  return l;
}
var mo = 20, go = ["top", "right", "bottom", "left", "width", "height", "size", "weight"], bo = typeof MutationObserver < "u", yo = (
  /** @class */
  function() {
    function e() {
      this.connected_ = !1, this.mutationEventsAdded_ = !1, this.mutationsObserver_ = null, this.observers_ = [], this.onTransitionEnd_ = this.onTransitionEnd_.bind(this), this.refresh = ho(this.refresh.bind(this), mo);
    }
    return e.prototype.addObserver = function(t) {
      ~this.observers_.indexOf(t) || this.observers_.push(t), this.connected_ || this.connect_();
    }, e.prototype.removeObserver = function(t) {
      var r = this.observers_, n = r.indexOf(t);
      ~n && r.splice(n, 1), !r.length && this.connected_ && this.disconnect_();
    }, e.prototype.refresh = function() {
      var t = this.updateObservers_();
      t && this.refresh();
    }, e.prototype.updateObservers_ = function() {
      var t = this.observers_.filter(function(r) {
        return r.gatherActive(), r.hasActive();
      });
      return t.forEach(function(r) {
        return r.broadcastActive();
      }), t.length > 0;
    }, e.prototype.connect_ = function() {
      !wt || this.connected_ || (document.addEventListener("transitionend", this.onTransitionEnd_), window.addEventListener("resize", this.refresh), bo ? (this.mutationsObserver_ = new MutationObserver(this.refresh), this.mutationsObserver_.observe(document, {
        attributes: !0,
        childList: !0,
        characterData: !0,
        subtree: !0
      })) : (document.addEventListener("DOMSubtreeModified", this.refresh), this.mutationEventsAdded_ = !0), this.connected_ = !0);
    }, e.prototype.disconnect_ = function() {
      !wt || !this.connected_ || (document.removeEventListener("transitionend", this.onTransitionEnd_), window.removeEventListener("resize", this.refresh), this.mutationsObserver_ && this.mutationsObserver_.disconnect(), this.mutationEventsAdded_ && document.removeEventListener("DOMSubtreeModified", this.refresh), this.mutationsObserver_ = null, this.mutationEventsAdded_ = !1, this.connected_ = !1);
    }, e.prototype.onTransitionEnd_ = function(t) {
      var r = t.propertyName, n = r === void 0 ? "" : r, o = go.some(function(i) {
        return !!~n.indexOf(i);
      });
      o && this.refresh();
    }, e.getInstance = function() {
      return this.instance_ || (this.instance_ = new e()), this.instance_;
    }, e.instance_ = null, e;
  }()
), Er = function(e, t) {
  for (var r = 0, n = Object.keys(t); r < n.length; r++) {
    var o = n[r];
    Object.defineProperty(e, o, {
      value: t[o],
      enumerable: !1,
      writable: !1,
      configurable: !0
    });
  }
  return e;
}, Ce = function(e) {
  var t = e && e.ownerDocument && e.ownerDocument.defaultView;
  return t || Ke;
}, wr = ot(0, 0, 0, 0);
function qe(e) {
  return parseFloat(e) || 0;
}
function rr(e) {
  for (var t = [], r = 1; r < arguments.length; r++)
    t[r - 1] = arguments[r];
  return t.reduce(function(n, o) {
    var i = e["border-" + o + "-width"];
    return n + qe(i);
  }, 0);
}
function So(e) {
  for (var t = ["top", "right", "bottom", "left"], r = {}, n = 0, o = t; n < o.length; n++) {
    var i = o[n], a = e["padding-" + i];
    r[i] = qe(a);
  }
  return r;
}
function Ro(e) {
  var t = e.getBBox();
  return ot(0, 0, t.width, t.height);
}
function Eo(e) {
  var t = e.clientWidth, r = e.clientHeight;
  if (!t && !r)
    return wr;
  var n = Ce(e).getComputedStyle(e), o = So(n), i = o.left + o.right, a = o.top + o.bottom, l = qe(n.width), u = qe(n.height);
  if (n.boxSizing === "border-box" && (Math.round(l + i) !== t && (l -= rr(n, "left", "right") + i), Math.round(u + a) !== r && (u -= rr(n, "top", "bottom") + a)), !xo(e)) {
    var f = Math.round(l + i) - t, m = Math.round(u + a) - r;
    Math.abs(f) !== 1 && (l -= f), Math.abs(m) !== 1 && (u -= m);
  }
  return ot(o.left, o.top, l, u);
}
var wo = /* @__PURE__ */ function() {
  return typeof SVGGraphicsElement < "u" ? function(e) {
    return e instanceof Ce(e).SVGGraphicsElement;
  } : function(e) {
    return e instanceof Ce(e).SVGElement && typeof e.getBBox == "function";
  };
}();
function xo(e) {
  return e === Ce(e).document.documentElement;
}
function _o(e) {
  return wt ? wo(e) ? Ro(e) : Eo(e) : wr;
}
function Co(e) {
  var t = e.x, r = e.y, n = e.width, o = e.height, i = typeof DOMRectReadOnly < "u" ? DOMRectReadOnly : Object, a = Object.create(i.prototype);
  return Er(a, {
    x: t,
    y: r,
    width: n,
    height: o,
    top: r,
    right: t + n,
    bottom: o + r,
    left: t
  }), a;
}
function ot(e, t, r, n) {
  return { x: e, y: t, width: r, height: n };
}
var Oo = (
  /** @class */
  function() {
    function e(t) {
      this.broadcastWidth = 0, this.broadcastHeight = 0, this.contentRect_ = ot(0, 0, 0, 0), this.target = t;
    }
    return e.prototype.isActive = function() {
      var t = _o(this.target);
      return this.contentRect_ = t, t.width !== this.broadcastWidth || t.height !== this.broadcastHeight;
    }, e.prototype.broadcastRect = function() {
      var t = this.contentRect_;
      return this.broadcastWidth = t.width, this.broadcastHeight = t.height, t;
    }, e;
  }()
), Po = (
  /** @class */
  /* @__PURE__ */ function() {
    function e(t, r) {
      var n = Co(r);
      Er(this, { target: t, contentRect: n });
    }
    return e;
  }()
), Mo = (
  /** @class */
  function() {
    function e(t, r, n) {
      if (this.activeObservations_ = [], this.observations_ = new Rr(), typeof t != "function")
        throw new TypeError("The callback provided as parameter 1 is not a function.");
      this.callback_ = t, this.controller_ = r, this.callbackCtx_ = n;
    }
    return e.prototype.observe = function(t) {
      if (!arguments.length)
        throw new TypeError("1 argument required, but only 0 present.");
      if (!(typeof Element > "u" || !(Element instanceof Object))) {
        if (!(t instanceof Ce(t).Element))
          throw new TypeError('parameter 1 is not of type "Element".');
        var r = this.observations_;
        r.has(t) || (r.set(t, new Oo(t)), this.controller_.addObserver(this), this.controller_.refresh());
      }
    }, e.prototype.unobserve = function(t) {
      if (!arguments.length)
        throw new TypeError("1 argument required, but only 0 present.");
      if (!(typeof Element > "u" || !(Element instanceof Object))) {
        if (!(t instanceof Ce(t).Element))
          throw new TypeError('parameter 1 is not of type "Element".');
        var r = this.observations_;
        r.has(t) && (r.delete(t), r.size || this.controller_.removeObserver(this));
      }
    }, e.prototype.disconnect = function() {
      this.clearActive(), this.observations_.clear(), this.controller_.removeObserver(this);
    }, e.prototype.gatherActive = function() {
      var t = this;
      this.clearActive(), this.observations_.forEach(function(r) {
        r.isActive() && t.activeObservations_.push(r);
      });
    }, e.prototype.broadcastActive = function() {
      if (this.hasActive()) {
        var t = this.callbackCtx_, r = this.activeObservations_.map(function(n) {
          return new Po(n.target, n.broadcastRect());
        });
        this.callback_.call(t, r, t), this.clearActive();
      }
    }, e.prototype.clearActive = function() {
      this.activeObservations_.splice(0);
    }, e.prototype.hasActive = function() {
      return this.activeObservations_.length > 0;
    }, e;
  }()
), xr = typeof WeakMap < "u" ? /* @__PURE__ */ new WeakMap() : new Rr(), _r = (
  /** @class */
  /* @__PURE__ */ function() {
    function e(t) {
      if (!(this instanceof e))
        throw new TypeError("Cannot call a class as a function.");
      if (!arguments.length)
        throw new TypeError("1 argument required, but only 0 present.");
      var r = yo.getInstance(), n = new Mo(t, r, this);
      xr.set(this, n);
    }
    return e;
  }()
);
[
  "observe",
  "unobserve",
  "disconnect"
].forEach(function(e) {
  _r.prototype[e] = function() {
    var t;
    return (t = xr.get(this))[e].apply(t, arguments);
  };
});
var To = function() {
  return typeof Ke.ResizeObserver < "u" ? Ke.ResizeObserver : _r;
}(), de = /* @__PURE__ */ new Map();
function Cr(e) {
  e.forEach(function(t) {
    var r, n = t.target;
    (r = de.get(n)) === null || r === void 0 || r.forEach(function(o) {
      return o(n);
    });
  });
}
var Or = new To(Cr);
process.env.NODE_ENV;
process.env.NODE_ENV;
function Io(e, t) {
  de.has(e) || (de.set(e, /* @__PURE__ */ new Set()), Or.observe(e)), de.get(e).add(t);
}
function Lo(e, t) {
  de.has(e) && (de.get(e).delete(t), de.get(e).size || (Or.unobserve(e), de.delete(e)));
}
function Pr(e, t) {
  if (!(e instanceof t)) throw new TypeError("Cannot call a class as a function");
}
function Ao(e, t) {
  for (var r = 0; r < t.length; r++) {
    var n = t[r];
    n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), Object.defineProperty(e, mr(n.key), n);
  }
}
function Mr(e, t, r) {
  return t && Ao(e.prototype, t), Object.defineProperty(e, "prototype", {
    writable: !1
  }), e;
}
function No(e, t) {
  if (typeof t != "function" && t !== null) throw new TypeError("Super expression must either be null or a function");
  e.prototype = Object.create(t && t.prototype, {
    constructor: {
      value: e,
      writable: !0,
      configurable: !0
    }
  }), Object.defineProperty(e, "prototype", {
    writable: !1
  }), t && un(e, t);
}
function Ze(e) {
  return Ze = Object.setPrototypeOf ? Object.getPrototypeOf.bind() : function(t) {
    return t.__proto__ || Object.getPrototypeOf(t);
  }, Ze(e);
}
function Tr() {
  try {
    var e = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function() {
    }));
  } catch {
  }
  return (Tr = function() {
    return !!e;
  })();
}
function $o(e, t) {
  if (t && (oe(t) == "object" || typeof t == "function")) return t;
  if (t !== void 0) throw new TypeError("Derived constructors may only return object or undefined");
  return fn(e);
}
function zo(e) {
  var t = Tr();
  return function() {
    var r, n = Ze(e);
    if (t) {
      var o = Ze(this).constructor;
      r = Reflect.construct(n, arguments, o);
    } else r = n.apply(this, arguments);
    return $o(this, r);
  };
}
var Do = /* @__PURE__ */ function(e) {
  No(r, e);
  var t = zo(r);
  function r() {
    return Pr(this, r), t.apply(this, arguments);
  }
  return Mr(r, [{
    key: "render",
    value: function() {
      return this.props.children;
    }
  }]), r;
}(d.Component);
function ko(e, t) {
  var r = e.children, n = e.disabled, o = d.useRef(null), i = d.useRef(null), a = d.useContext(Et), l = typeof r == "function", u = l ? r(o) : r, f = d.useRef({
    width: -1,
    height: -1,
    offsetWidth: -1,
    offsetHeight: -1
  }), m = !l && /* @__PURE__ */ d.isValidElement(u) && lo(u), v = m ? uo(u) : null, y = co(v, o), g = function() {
    var x;
    return Ue(o.current) || // Support `nativeElement` format
    (o.current && oe(o.current) === "object" ? Ue((x = o.current) === null || x === void 0 ? void 0 : x.nativeElement) : null) || Ue(i.current);
  };
  d.useImperativeHandle(t, function() {
    return g();
  });
  var b = d.useRef(e);
  b.current = e;
  var p = d.useCallback(function(s) {
    var x = b.current, S = x.onResize, P = x.data, h = s.getBoundingClientRect(), R = h.width, L = h.height, O = s.offsetWidth, T = s.offsetHeight, _ = Math.floor(R), M = Math.floor(L);
    if (f.current.width !== _ || f.current.height !== M || f.current.offsetWidth !== O || f.current.offsetHeight !== T) {
      var F = {
        width: _,
        height: M,
        offsetWidth: O,
        offsetHeight: T
      };
      f.current = F;
      var Y = O === Math.round(R) ? R : O, B = T === Math.round(L) ? L : T, $ = q(q({}, F), {}, {
        offsetWidth: Y,
        offsetHeight: B
      });
      a == null || a($, s, P), S && Promise.resolve().then(function() {
        S($, s);
      });
    }
  }, []);
  return d.useEffect(function() {
    var s = g();
    return s && !n && Io(s, p), function() {
      return Lo(s, p);
    };
  }, [o.current, n]), /* @__PURE__ */ d.createElement(Do, {
    ref: i
  }, m ? /* @__PURE__ */ d.cloneElement(u, {
    ref: y
  }) : u);
}
var Ir = /* @__PURE__ */ d.forwardRef(ko);
process.env.NODE_ENV !== "production" && (Ir.displayName = "SingleObserver");
var jo = "rc-observer-key";
function Fo(e, t) {
  var r = e.children, n = typeof r == "function" ? [r] : bt(r);
  return process.env.NODE_ENV !== "production" && (n.length > 1 ? St(!1, "Find more than one child node with `children` in ResizeObserver. Please use ResizeObserver.Collection instead.") : n.length === 0 && St(!1, "`children` of ResizeObserver is empty. Nothing is in observe.")), n.map(function(o, i) {
    var a = (o == null ? void 0 : o.key) || "".concat(jo, "-").concat(i);
    return /* @__PURE__ */ d.createElement(Ir, Ct({}, e, {
      key: a,
      ref: i === 0 ? t : void 0
    }), o);
  });
}
var it = /* @__PURE__ */ d.forwardRef(Fo);
process.env.NODE_ENV !== "production" && (it.displayName = "ResizeObserver");
it.Collection = fo;
function nr(e) {
  var t = d.useRef();
  t.current = e;
  var r = d.useCallback(function() {
    for (var n, o = arguments.length, i = new Array(o), a = 0; a < o; a++)
      i[a] = arguments[a];
    return (n = t.current) === null || n === void 0 ? void 0 : n.call.apply(n, [t].concat(i));
  }, []);
  return r;
}
function Ho() {
  return !!(typeof window < "u" && window.document && window.document.createElement);
}
var or = process.env.NODE_ENV !== "test" && Ho() ? d.useLayoutEffect : d.useEffect, Ne = function(t, r) {
  var n = d.useRef(!0);
  or(function() {
    return t(n.current);
  }, r), or(function() {
    return n.current = !1, function() {
      n.current = !0;
    };
  }, []);
}, Lr = /* @__PURE__ */ d.forwardRef(function(e, t) {
  var r = e.height, n = e.offsetY, o = e.offsetX, i = e.children, a = e.prefixCls, l = e.onInnerResize, u = e.innerProps, f = e.rtl, m = e.extra, v = {}, y = {
    display: "flex",
    flexDirection: "column"
  };
  return n !== void 0 && (v = {
    height: r,
    position: "relative",
    overflow: "hidden"
  }, y = q(q({}, y), {}, ee(ee(ee(ee(ee({
    transform: "translateY(".concat(n, "px)")
  }, f ? "marginRight" : "marginLeft", -o), "position", "absolute"), "left", 0), "right", 0), "top", 0))), /* @__PURE__ */ d.createElement("div", {
    style: v
  }, /* @__PURE__ */ d.createElement(it, {
    onResize: function(b) {
      var p = b.offsetHeight;
      p && l && l();
    }
  }, /* @__PURE__ */ d.createElement("div", Ct({
    style: y,
    className: Xe(ee({}, "".concat(a, "-holder-inner"), a)),
    ref: t
  }, u), i, m)));
});
Lr.displayName = "Filler";
function Vo(e) {
  var t = e.children, r = e.setRef, n = d.useCallback(function(o) {
    r(o);
  }, []);
  return /* @__PURE__ */ d.cloneElement(t, {
    ref: n
  });
}
function Bo(e, t, r, n, o, i, a, l) {
  var u = l.getKey;
  return e.slice(t, r + 1).map(function(f, m) {
    var v = t + m, y = a(f, v, {
      style: {
        width: n
      },
      offsetX: o
    }), g = u(f);
    return /* @__PURE__ */ d.createElement(Vo, {
      key: g,
      setRef: function(p) {
        return i(f, p);
      }
    }, y);
  });
}
function Wo(e, t, r) {
  var n = e.length, o = t.length, i, a;
  if (n === 0 && o === 0)
    return null;
  n < o ? (i = e, a = t) : (i = t, a = e);
  var l = {
    __EMPTY_ITEM__: !0
  };
  function u(b) {
    return b !== void 0 ? r(b) : l;
  }
  for (var f = null, m = Math.abs(n - o) !== 1, v = 0; v < a.length; v += 1) {
    var y = u(i[v]), g = u(a[v]);
    if (y !== g) {
      f = v, m = m || y !== u(a[v + 1]);
      break;
    }
  }
  return f === null ? null : {
    index: f,
    multiple: m
  };
}
function Yo(e, t, r) {
  var n = d.useState(e), o = K(n, 2), i = o[0], a = o[1], l = d.useState(null), u = K(l, 2), f = u[0], m = u[1];
  return d.useEffect(function() {
    var v = Wo(i || [], e || [], t);
    (v == null ? void 0 : v.index) !== void 0 && m(e[v.index]), a(e);
  }, [e]), [f];
}
var Ar = function(t) {
  return +setTimeout(t, 16);
}, Nr = function(t) {
  return clearTimeout(t);
};
typeof window < "u" && "requestAnimationFrame" in window && (Ar = function(t) {
  return window.requestAnimationFrame(t);
}, Nr = function(t) {
  return window.cancelAnimationFrame(t);
});
var ir = 0, at = /* @__PURE__ */ new Map();
function $r(e) {
  at.delete(e);
}
var ne = function(t) {
  var r = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : 1;
  ir += 1;
  var n = ir;
  function o(i) {
    if (i === 0)
      $r(n), t();
    else {
      var a = Ar(function() {
        o(i - 1);
      });
      at.set(n, a);
    }
  }
  return o(r), n;
};
ne.cancel = function(e) {
  var t = at.get(e);
  return $r(e), Nr(t);
};
process.env.NODE_ENV !== "production" && (ne.ids = function() {
  return at;
});
var ar = (typeof navigator > "u" ? "undefined" : oe(navigator)) === "object" && /Firefox/i.test(navigator.userAgent);
const zr = function(e, t, r, n) {
  var o = D(!1), i = D(null);
  function a() {
    clearTimeout(i.current), o.current = !0, i.current = setTimeout(function() {
      o.current = !1;
    }, 50);
  }
  var l = D({
    top: e,
    bottom: t,
    left: r,
    right: n
  });
  return l.current.top = e, l.current.bottom = t, l.current.left = r, l.current.right = n, function(u, f) {
    var m = arguments.length > 2 && arguments[2] !== void 0 ? arguments[2] : !1, v = u ? (
      // Pass origin wheel when on the left
      f < 0 && l.current.left || // Pass origin wheel when on the right
      f > 0 && l.current.right
    ) : f < 0 && l.current.top || // Pass origin wheel when on the bottom
    f > 0 && l.current.bottom;
    return m && v ? (clearTimeout(i.current), o.current = !1) : (!v || o.current) && a(), !o.current && v;
  };
};
function Go(e, t, r, n, o, i, a) {
  var l = D(0), u = D(null), f = D(null), m = D(!1), v = zr(t, r, n, o);
  function y(S, P) {
    if (ne.cancel(u.current), !v(!1, P)) {
      var h = S;
      if (!h._virtualHandled)
        h._virtualHandled = !0;
      else
        return;
      l.current += P, f.current = P, ar || h.preventDefault(), u.current = ne(function() {
        var R = m.current ? 10 : 1;
        a(l.current * R, !1), l.current = 0;
      });
    }
  }
  function g(S, P) {
    a(P, !0), ar || S.preventDefault();
  }
  var b = D(null), p = D(null);
  function s(S) {
    if (e) {
      ne.cancel(p.current), p.current = ne(function() {
        b.current = null;
      }, 2);
      var P = S.deltaX, h = S.deltaY, R = S.shiftKey, L = P, O = h;
      (b.current === "sx" || !b.current && R && h && !P) && (L = h, O = 0, b.current = "sx");
      var T = Math.abs(L), _ = Math.abs(O);
      b.current === null && (b.current = i && T > _ ? "x" : "y"), b.current === "y" ? y(S, O) : g(S, L);
    }
  }
  function x(S) {
    e && (m.current = S.detail === f.current);
  }
  return [s, x];
}
function Uo(e, t, r, n) {
  var o = d.useMemo(function() {
    return [/* @__PURE__ */ new Map(), []];
  }, [e, r.id, n]), i = K(o, 2), a = i[0], l = i[1], u = function(m) {
    var v = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : m, y = a.get(m), g = a.get(v);
    if (y === void 0 || g === void 0)
      for (var b = e.length, p = l.length; p < b; p += 1) {
        var s, x = e[p], S = t(x);
        a.set(S, p);
        var P = (s = r.get(S)) !== null && s !== void 0 ? s : n;
        if (l[p] = (l[p - 1] || 0) + P, S === m && (y = p), S === v && (g = p), y !== void 0 && g !== void 0)
          break;
      }
    return {
      top: l[y - 1] || 0,
      bottom: l[g]
    };
  };
  return u;
}
var Xo = /* @__PURE__ */ function() {
  function e() {
    Pr(this, e), ee(this, "maps", void 0), ee(this, "id", 0), this.maps = /* @__PURE__ */ Object.create(null);
  }
  return Mr(e, [{
    key: "set",
    value: function(r, n) {
      this.maps[r] = n, this.id += 1;
    }
  }, {
    key: "get",
    value: function(r) {
      return this.maps[r];
    }
  }]), e;
}();
function sr(e) {
  var t = parseFloat(e);
  return isNaN(t) ? 0 : t;
}
function Ko(e, t, r) {
  var n = d.useState(0), o = K(n, 2), i = o[0], a = o[1], l = D(/* @__PURE__ */ new Map()), u = D(new Xo()), f = D();
  function m() {
    ne.cancel(f.current);
  }
  function v() {
    var g = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : !1;
    m();
    var b = function() {
      l.current.forEach(function(s, x) {
        if (s && s.offsetParent) {
          var S = Ue(s), P = S.offsetHeight, h = getComputedStyle(S), R = h.marginTop, L = h.marginBottom, O = sr(R), T = sr(L), _ = P + O + T;
          u.current.get(x) !== _ && u.current.set(x, _);
        }
      }), a(function(s) {
        return s + 1;
      });
    };
    g ? b() : f.current = ne(b);
  }
  function y(g, b) {
    var p = e(g);
    l.current.get(p), b ? (l.current.set(p, b), v()) : l.current.delete(p);
  }
  return en(function() {
    return m;
  }, []), [y, v, u.current, i];
}
var cr = 14 / 15;
function qo(e, t, r) {
  var n = D(!1), o = D(0), i = D(0), a = D(null), l = D(null), u, f = function(g) {
    if (n.current) {
      var b = Math.ceil(g.touches[0].pageX), p = Math.ceil(g.touches[0].pageY), s = o.current - b, x = i.current - p, S = Math.abs(s) > Math.abs(x);
      S ? o.current = b : i.current = p;
      var P = r(S, S ? s : x, !1, g);
      P && g.preventDefault(), clearInterval(l.current), P && (l.current = setInterval(function() {
        S ? s *= cr : x *= cr;
        var h = Math.floor(S ? s : x);
        (!r(S, h, !0) || Math.abs(h) <= 0.1) && clearInterval(l.current);
      }, 16));
    }
  }, m = function() {
    n.current = !1, u();
  }, v = function(g) {
    u(), g.touches.length === 1 && !n.current && (n.current = !0, o.current = Math.ceil(g.touches[0].pageX), i.current = Math.ceil(g.touches[0].pageY), a.current = g.target, a.current.addEventListener("touchmove", f, {
      passive: !1
    }), a.current.addEventListener("touchend", m, {
      passive: !0
    }));
  };
  u = function() {
    a.current && (a.current.removeEventListener("touchmove", f), a.current.removeEventListener("touchend", m));
  }, Ne(function() {
    return e && t.current.addEventListener("touchstart", v, {
      passive: !0
    }), function() {
      var y;
      (y = t.current) === null || y === void 0 || y.removeEventListener("touchstart", v), u(), clearInterval(l.current);
    };
  }, [e]);
}
function lr(e) {
  return Math.floor(Math.pow(e, 0.5));
}
function xt(e, t) {
  var r = "touches" in e ? e.touches[0] : e;
  return r[t ? "pageX" : "pageY"] - window[t ? "scrollX" : "scrollY"];
}
function Zo(e, t, r) {
  d.useEffect(function() {
    var n = t.current;
    if (e && n) {
      var o = !1, i, a, l = function() {
        ne.cancel(i);
      }, u = function y() {
        l(), i = ne(function() {
          r(a), y();
        });
      }, f = function(g) {
        var b = g;
        b._virtualHandled || (b._virtualHandled = !0, o = !0);
      }, m = function() {
        o = !1, l();
      }, v = function(g) {
        if (o) {
          var b = xt(g, !1), p = n.getBoundingClientRect(), s = p.top, x = p.bottom;
          if (b <= s) {
            var S = s - b;
            a = -lr(S), u();
          } else if (b >= x) {
            var P = b - x;
            a = lr(P), u();
          } else
            l();
        }
      };
      return n.addEventListener("mousedown", f), n.ownerDocument.addEventListener("mouseup", m), n.ownerDocument.addEventListener("mousemove", v), function() {
        n.removeEventListener("mousedown", f), n.ownerDocument.removeEventListener("mouseup", m), n.ownerDocument.removeEventListener("mousemove", v), l();
      };
    }
  }, [e]);
}
var ur = 10;
function Jo(e, t, r, n, o, i, a, l) {
  var u = d.useRef(), f = d.useState(null), m = K(f, 2), v = m[0], y = m[1];
  return Ne(function() {
    if (v && v.times < ur) {
      if (!e.current) {
        y(function(W) {
          return q({}, W);
        });
        return;
      }
      i();
      var g = v.targetAlign, b = v.originAlign, p = v.index, s = v.offset, x = e.current.clientHeight, S = !1, P = g, h = null;
      if (x) {
        for (var R = g || b, L = 0, O = 0, T = 0, _ = Math.min(t.length - 1, p), M = 0; M <= _; M += 1) {
          var F = o(t[M]);
          O = L;
          var Y = r.get(F);
          T = O + (Y === void 0 ? n : Y), L = T;
        }
        for (var B = R === "top" ? s : x - s, $ = _; $ >= 0; $ -= 1) {
          var G = o(t[$]), k = r.get(G);
          if (k === void 0) {
            S = !0;
            break;
          }
          if (B -= k, B <= 0)
            break;
        }
        switch (R) {
          case "top":
            h = O - s;
            break;
          case "bottom":
            h = T - x + s;
            break;
          default: {
            var ae = e.current.scrollTop, Z = ae + x;
            O < ae ? P = "top" : T > Z && (P = "bottom");
          }
        }
        h !== null && a(h), h !== v.lastTop && (S = !0);
      }
      S && y(q(q({}, v), {}, {
        times: v.times + 1,
        targetAlign: P,
        lastTop: h
      }));
    } else process.env.NODE_ENV !== "production" && (v == null ? void 0 : v.times) === ur && nt(!1, "Seems `scrollTo` with `rc-virtual-list` reach the max limitation. Please fire issue for us. Thanks.");
  }, [v, e.current]), function(g) {
    if (g == null) {
      l();
      return;
    }
    if (ne.cancel(u.current), typeof g == "number")
      a(g);
    else if (g && oe(g) === "object") {
      var b, p = g.align;
      "index" in g ? b = g.index : b = t.findIndex(function(S) {
        return o(S) === g.key;
      });
      var s = g.offset, x = s === void 0 ? 0 : s;
      y({
        times: 0,
        index: b,
        offset: x,
        originAlign: p
      });
    }
  };
}
var _t = /* @__PURE__ */ d.forwardRef(function(e, t) {
  var r = e.prefixCls, n = e.rtl, o = e.scrollOffset, i = e.scrollRange, a = e.onStartMove, l = e.onStopMove, u = e.onScroll, f = e.horizontal, m = e.spinSize, v = e.containerSize, y = e.style, g = e.thumbStyle, b = d.useState(!1), p = K(b, 2), s = p[0], x = p[1], S = d.useState(null), P = K(S, 2), h = P[0], R = P[1], L = d.useState(null), O = K(L, 2), T = O[0], _ = O[1], M = !n, F = d.useRef(), Y = d.useRef(), B = d.useState(!1), $ = K(B, 2), G = $[0], k = $[1], ae = d.useRef(), Z = function() {
    clearTimeout(ae.current), k(!0), ae.current = setTimeout(function() {
      k(!1);
    }, 3e3);
  }, W = i - v || 0, me = v - m || 0, H = d.useMemo(function() {
    if (o === 0 || W === 0)
      return 0;
    var X = o / W;
    return X * me;
  }, [o, W, me]), se = function(z) {
    z.stopPropagation(), z.preventDefault();
  }, ve = d.useRef({
    top: H,
    dragging: s,
    pageY: h,
    startTop: T
  });
  ve.current = {
    top: H,
    dragging: s,
    pageY: h,
    startTop: T
  };
  var pe = function(z) {
    x(!0), R(xt(z, f)), _(ve.current.top), a(), z.stopPropagation(), z.preventDefault();
  };
  d.useEffect(function() {
    var X = function(E) {
      E.preventDefault();
    }, z = F.current, re = Y.current;
    return z.addEventListener("touchstart", X, {
      passive: !1
    }), re.addEventListener("touchstart", pe, {
      passive: !1
    }), function() {
      z.removeEventListener("touchstart", X), re.removeEventListener("touchstart", pe);
    };
  }, []);
  var Se = d.useRef();
  Se.current = W;
  var ge = d.useRef();
  ge.current = me, d.useEffect(function() {
    if (s) {
      var X, z = function(E) {
        var le = ve.current, ue = le.dragging, be = le.pageY, De = le.startTop;
        ne.cancel(X);
        var ke = F.current.getBoundingClientRect(), ye = v / (f ? ke.width : ke.height);
        if (ue) {
          var Oe = (xt(E, f) - be) * ye, Pe = De;
          !M && f ? Pe -= Oe : Pe += Oe;
          var je = Se.current, Fe = ge.current, st = Fe ? Pe / Fe : 0, ce = Math.ceil(st * je);
          ce = Math.max(ce, 0), ce = Math.min(ce, je), X = ne(function() {
            u(ce, f);
          });
        }
      }, re = function() {
        x(!1), l();
      };
      return window.addEventListener("mousemove", z, {
        passive: !0
      }), window.addEventListener("touchmove", z, {
        passive: !0
      }), window.addEventListener("mouseup", re, {
        passive: !0
      }), window.addEventListener("touchend", re, {
        passive: !0
      }), function() {
        window.removeEventListener("mousemove", z), window.removeEventListener("touchmove", z), window.removeEventListener("mouseup", re), window.removeEventListener("touchend", re), ne.cancel(X);
      };
    }
  }, [s]), d.useEffect(function() {
    return Z(), function() {
      clearTimeout(ae.current);
    };
  }, [o]), d.useImperativeHandle(t, function() {
    return {
      delayHidden: Z
    };
  });
  var U = "".concat(r, "-scrollbar"), te = {
    position: "absolute",
    visibility: G ? null : "hidden"
  }, ie = {
    position: "absolute",
    background: "rgba(0, 0, 0, 0.5)",
    borderRadius: 99,
    cursor: "pointer",
    userSelect: "none"
  };
  return f ? (te.height = 8, te.left = 0, te.right = 0, te.bottom = 0, ie.height = "100%", ie.width = m, M ? ie.left = H : ie.right = H) : (te.width = 8, te.top = 0, te.bottom = 0, M ? te.right = 0 : te.left = 0, ie.width = "100%", ie.height = m, ie.top = H), /* @__PURE__ */ d.createElement("div", {
    ref: F,
    className: Xe(U, ee(ee(ee({}, "".concat(U, "-horizontal"), f), "".concat(U, "-vertical"), !f), "".concat(U, "-visible"), G)),
    style: q(q({}, te), y),
    onMouseDown: se,
    onMouseMove: Z
  }, /* @__PURE__ */ d.createElement("div", {
    ref: Y,
    className: Xe("".concat(U, "-thumb"), ee({}, "".concat(U, "-thumb-moving"), s)),
    style: q(q({}, ie), g),
    onMouseDown: pe
  }));
});
process.env.NODE_ENV !== "production" && (_t.displayName = "ScrollBar");
var Qo = 20;
function fr() {
  var e = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : 0, t = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : 0, r = e / t * e;
  return isNaN(r) && (r = 0), r = Math.max(r, Qo), Math.floor(r);
}
var ei = ["prefixCls", "className", "height", "itemHeight", "fullHeight", "style", "data", "children", "itemKey", "virtual", "direction", "scrollWidth", "component", "onScroll", "onVirtualScroll", "onVisibleChange", "innerProps", "extraRender", "styles"], ti = [], ri = {
  overflowY: "auto",
  overflowAnchor: "none"
};
function ni(e, t) {
  var r = e.prefixCls, n = r === void 0 ? "rc-virtual-list" : r, o = e.className, i = e.height, a = e.itemHeight, l = e.fullHeight, u = l === void 0 ? !0 : l, f = e.style, m = e.data, v = e.children, y = e.itemKey, g = e.virtual, b = e.direction, p = e.scrollWidth, s = e.component, x = s === void 0 ? "div" : s, S = e.onScroll, P = e.onVirtualScroll, h = e.onVisibleChange, R = e.innerProps, L = e.extraRender, O = e.styles, T = Un(e, ei), _ = d.useCallback(function(C) {
    return typeof y == "function" ? y(C) : C == null ? void 0 : C[y];
  }, [y]), M = Ko(_), F = K(M, 4), Y = F[0], B = F[1], $ = F[2], G = F[3], k = !!(g !== !1 && i && a), ae = d.useMemo(function() {
    return Object.values($.maps).reduce(function(C, w) {
      return C + w;
    }, 0);
  }, [$.id, $.maps]), Z = k && m && (Math.max(a * m.length, ae) > i || !!p), W = b === "rtl", me = Xe(n, ee({}, "".concat(n, "-rtl"), W), o), H = m || ti, se = D(), ve = D(), pe = D(), Se = Ye(0), ge = K(Se, 2), U = ge[0], te = ge[1], ie = Ye(0), X = K(ie, 2), z = X[0], re = X[1], Re = Ye(!1), E = K(Re, 2), le = E[0], ue = E[1], be = function() {
    ue(!0);
  }, De = function() {
    ue(!1);
  }, ke = {
    getKey: _
  };
  function ye(C) {
    te(function(w) {
      var I;
      typeof C == "function" ? I = C(w) : I = C;
      var V = Br(I);
      return se.current.scrollTop = V, V;
    });
  }
  var Oe = D({
    start: 0,
    end: H.length
  }), Pe = D(), je = Yo(H, _), Fe = K(je, 1), st = Fe[0];
  Pe.current = st;
  var ce = d.useMemo(function() {
    if (!k)
      return {
        scrollHeight: void 0,
        start: 0,
        end: H.length - 1,
        offset: void 0
      };
    if (!Z) {
      var C;
      return {
        scrollHeight: ((C = ve.current) === null || C === void 0 ? void 0 : C.offsetHeight) || 0,
        start: 0,
        end: H.length - 1,
        offset: void 0
      };
    }
    for (var w = 0, I, V, J, Ht = H.length, Le = 0; Le < Ht; Le += 1) {
      var Zr = H[Le], Jr = _(Zr), Vt = $.get(Jr), pt = w + (Vt === void 0 ? a : Vt);
      pt >= U && I === void 0 && (I = Le, V = w), pt > U + i && J === void 0 && (J = Le), w = pt;
    }
    return I === void 0 && (I = 0, V = 0, J = Math.ceil(i / a)), J === void 0 && (J = H.length - 1), J = Math.min(J + 1, H.length - 1), {
      scrollHeight: w,
      start: I,
      end: J,
      offset: V
    };
  }, [Z, k, U, H, G, i]), Ee = ce.scrollHeight, Me = ce.start, Te = ce.end, Mt = ce.offset;
  Oe.current.start = Me, Oe.current.end = Te;
  var kr = d.useState({
    width: 0,
    height: i
  }), Tt = K(kr, 2), fe = Tt[0], jr = Tt[1], Fr = function(w) {
    jr({
      width: w.offsetWidth,
      height: w.offsetHeight
    });
  }, It = D(), Lt = D(), Hr = d.useMemo(function() {
    return fr(fe.width, p);
  }, [fe.width, p]), Vr = d.useMemo(function() {
    return fr(fe.height, Ee);
  }, [fe.height, Ee]), ct = Ee - i, lt = D(ct);
  lt.current = ct;
  function Br(C) {
    var w = C;
    return Number.isNaN(lt.current) || (w = Math.min(w, lt.current)), w = Math.max(w, 0), w;
  }
  var He = U <= 0, Ve = U >= ct, At = z <= 0, Nt = z >= p, Wr = zr(He, Ve, At, Nt), ut = function() {
    return {
      x: W ? -z : z,
      y: U
    };
  }, ft = D(ut()), Be = nr(function(C) {
    if (P) {
      var w = q(q({}, ut()), C);
      (ft.current.x !== w.x || ft.current.y !== w.y) && (P(w), ft.current = w);
    }
  });
  function $t(C, w) {
    var I = C;
    w ? (Ut.flushSync(function() {
      re(I);
    }), Be()) : ye(I);
  }
  function Yr(C) {
    var w = C.currentTarget.scrollTop;
    w !== U && ye(w), S == null || S(C), Be();
  }
  var dt = function(w) {
    var I = w, V = p ? p - fe.width : 0;
    return I = Math.max(I, 0), I = Math.min(I, V), I;
  }, Gr = nr(function(C, w) {
    w ? (Ut.flushSync(function() {
      re(function(I) {
        var V = I + (W ? -C : C);
        return dt(V);
      });
    }), Be()) : ye(function(I) {
      var V = I + C;
      return V;
    });
  }), Ur = Go(k, He, Ve, At, Nt, !!p, Gr), zt = K(Ur, 2), vt = zt[0], Dt = zt[1];
  qo(k, se, function(C, w, I, V) {
    var J = V;
    return Wr(C, w, I) ? !1 : !J || !J._virtualHandled ? (J && (J._virtualHandled = !0), vt({
      preventDefault: function() {
      },
      deltaX: C ? w : 0,
      deltaY: C ? 0 : w
    }), !0) : !1;
  }), Zo(Z, se, function(C) {
    ye(function(w) {
      return w + C;
    });
  }), Ne(function() {
    function C(I) {
      var V = He && I.detail < 0, J = Ve && I.detail > 0;
      k && !V && !J && I.preventDefault();
    }
    var w = se.current;
    return w.addEventListener("wheel", vt, {
      passive: !1
    }), w.addEventListener("DOMMouseScroll", Dt, {
      passive: !0
    }), w.addEventListener("MozMousePixelScroll", C, {
      passive: !1
    }), function() {
      w.removeEventListener("wheel", vt), w.removeEventListener("DOMMouseScroll", Dt), w.removeEventListener("MozMousePixelScroll", C);
    };
  }, [k, He, Ve]), Ne(function() {
    if (p) {
      var C = dt(z);
      re(C), Be({
        x: C
      });
    }
  }, [fe.width, p]);
  var kt = function() {
    var w, I;
    (w = It.current) === null || w === void 0 || w.delayHidden(), (I = Lt.current) === null || I === void 0 || I.delayHidden();
  }, jt = Jo(se, H, $, a, _, function() {
    return B(!0);
  }, ye, kt);
  d.useImperativeHandle(t, function() {
    return {
      nativeElement: pe.current,
      getScrollInfo: ut,
      scrollTo: function(w) {
        function I(V) {
          return V && oe(V) === "object" && ("left" in V || "top" in V);
        }
        I(w) ? (w.left !== void 0 && re(dt(w.left)), jt(w.top)) : jt(w);
      }
    };
  }), Ne(function() {
    if (h) {
      var C = H.slice(Me, Te + 1);
      h(C, H);
    }
  }, [Me, Te, H]);
  var Xr = Uo(H, _, $, a), Kr = L == null ? void 0 : L({
    start: Me,
    end: Te,
    virtual: Z,
    offsetX: z,
    offsetY: Mt,
    rtl: W,
    getSize: Xr
  }), qr = Bo(H, Me, Te, p, z, Y, v, ke), Ie = null;
  i && (Ie = q(ee({}, u ? "height" : "maxHeight", i), ri), k && (Ie.overflowY = "hidden", p && (Ie.overflowX = "hidden"), le && (Ie.pointerEvents = "none")));
  var Ft = {};
  return W && (Ft.dir = "rtl"), /* @__PURE__ */ d.createElement("div", Ct({
    ref: pe,
    style: q(q({}, f), {}, {
      position: "relative"
    }),
    className: me
  }, Ft, T), /* @__PURE__ */ d.createElement(it, {
    onResize: Fr
  }, /* @__PURE__ */ d.createElement(x, {
    className: "".concat(n, "-holder"),
    style: Ie,
    ref: se,
    onScroll: Yr,
    onMouseEnter: kt
  }, /* @__PURE__ */ d.createElement(Lr, {
    prefixCls: n,
    height: Ee,
    offsetX: z,
    offsetY: Mt,
    scrollWidth: p,
    onInnerResize: B,
    ref: ve,
    innerProps: R,
    rtl: W,
    extra: Kr
  }, qr))), Z && Ee > i && /* @__PURE__ */ d.createElement(_t, {
    ref: It,
    prefixCls: n,
    scrollOffset: U,
    scrollRange: Ee,
    rtl: W,
    onScroll: $t,
    onStartMove: be,
    onStopMove: De,
    spinSize: Vr,
    containerSize: fe.height,
    style: O == null ? void 0 : O.verticalScrollBar,
    thumbStyle: O == null ? void 0 : O.verticalScrollBarThumb
  }), Z && p > fe.width && /* @__PURE__ */ d.createElement(_t, {
    ref: Lt,
    prefixCls: n,
    scrollOffset: z,
    scrollRange: p,
    rtl: W,
    onScroll: $t,
    onStartMove: be,
    onStopMove: De,
    spinSize: Hr,
    containerSize: fe.width,
    horizontal: !0,
    style: O == null ? void 0 : O.horizontalScrollBar,
    thumbStyle: O == null ? void 0 : O.horizontalScrollBarThumb
  }));
}
var Dr = /* @__PURE__ */ d.forwardRef(ni);
Dr.displayName = "List";
const wi = (e) => {
  var b, p, s, x, S, P;
  const [t, r] = Ye(1), n = tn(nn), o = dn(), i = pn(o);
  if (!n)
    return null;
  const a = (h) => e.filter === Yt.ALL || !e.filter ? h : e.filter === Yt.UNARCHIVED ? h.filter((R) => !R.archived) : h.filter(e.filter);
  if (n.notifications === void 0) return null;
  const u = a(n.notifications).sort((h, R) => new Date(h.date).getTime() - new Date(R.date).getTime()), f = {}, m = new rn();
  u.forEach((h) => {
    var R, L;
    if ((L = (R = h.deliveryOptions) == null ? void 0 : R.instant) != null && L.batching) {
      const O = h.deliveryOptions.instant.batchingKey, T = O ? m.parseAndRenderSync(`{{${O}}}`, h) : "", _ = `${h.notificationId}-${T}`;
      f[_] ? f[_].push(h) : f[_] = [h];
    } else {
      const O = h.id;
      f[O] = [h];
    }
  });
  const v = Object.values(f).sort(
    (h, R) => new Date(R[R.length - 1].date).getTime() - new Date(h[h.length - 1].date).getTime()
  ), y = e.empty ?? on, g = (h, R) => {
    R >= Math.floor(n.notifications.length / e.pageSize) && n.loadNotifications(), r(R);
  };
  return /* @__PURE__ */ j("div", { children: e.pagination === "INFINITE_SCROLL" ? /* @__PURE__ */ $e(
    Kt,
    {
      sx: {
        backgroundColor: i.paper,
        color: i.text
      },
      subheader: /* @__PURE__ */ j(
        Bt,
        {
          title: (b = e.header) == null ? void 0 : b.title,
          button1ClickHandler: (p = e.header) == null ? void 0 : p.button1ClickHandler,
          button2ClickHandler: (s = e.header) == null ? void 0 : s.button2ClickHandler
        }
      ),
      children: [
        v.length === 0 && y,
        /* @__PURE__ */ j(
          Dr,
          {
            data: v,
            height: e.maxHeight,
            itemHeight: 47,
            itemKey: (h) => h[0].id,
            onScroll: (h) => {
              Math.abs(
                h.currentTarget.scrollHeight - h.currentTarget.scrollTop - e.maxHeight
              ) <= 1 && n.loadNotifications();
            },
            children: (h) => /* @__PURE__ */ j(mt, { style: { padding: 0 }, children: /* @__PURE__ */ j(
              Wt,
              {
                markAsArchived: n.markAsArchived,
                notifications: h,
                markAsClicked: n.markAsClicked,
                renderer: e.notificationRenderer,
                imageShape: e.imageShape,
                newTab: e.newTab
              }
            ) }, h[0].id)
          }
        )
      ]
    }
  ) : /* @__PURE__ */ $e(
    Kt,
    {
      sx: {
        backgroundColor: i.paper,
        color: i.text
      },
      subheader: /* @__PURE__ */ j(
        Bt,
        {
          title: (x = e.header) == null ? void 0 : x.title,
          button1ClickHandler: (S = e.header) == null ? void 0 : S.button1ClickHandler,
          button2ClickHandler: (P = e.header) == null ? void 0 : P.button2ClickHandler
        }
      ),
      children: [
        e.pagePosition === "top" && v.length > 0 && /* @__PURE__ */ j(
          gt,
          {
            count: Math.ceil(v.length / e.pageSize),
            page: t,
            onChange: g
          }
        ),
        v.filter((h, R) => e.pagination === "PAGINATED" ? R >= (t - 1) * e.pageSize && R < t * e.pageSize : !0).map((h) => /* @__PURE__ */ j(mt, { style: { padding: 0 }, children: /* @__PURE__ */ j(
          Wt,
          {
            markAsArchived: n.markAsArchived,
            notifications: h,
            markAsClicked: n.markAsClicked,
            renderer: e.notificationRenderer,
            imageShape: e.imageShape,
            newTab: e.newTab
          }
        ) }, h[0].id)),
        e.pagePosition === "bottom" && v.length > 0 && /* @__PURE__ */ j(
          gt,
          {
            count: Math.ceil(v.length / e.pageSize),
            page: t,
            onChange: g
          }
        ),
        v.length === 0 && y
      ]
    }
  ) });
};
export {
  wi as Inbox
};
