// @notificationapi/react/dist/components/Notifications/InboxHeader.js
import { jsx as k, jsxs as Be, Fragment as mo } from "react/jsx-runtime";
import { T as vo } from "../../assets/Typography.js";
import { B as ho } from "../../assets/Box.js";
import * as M from "react";
import { g as Wt, P as r, e as It, a as Ht } from "../../assets/createTheme.js";
import { g as Vt, a as Ft, s as ot, u as Ut, m as ut, c as xt } from "../../assets/DefaultPropsProvider.js";
import { u as zt } from "../../assets/index.js";
import { c as go, r as qt, d as Tt, u as yo, a as Xe, f as Pt, b as Ge, h as bo, i as Et } from "../../assets/ButtonBase.js";
import { o as Rt, P as wo, H as _e, g as Oo, e as xo } from "../../assets/Portal.js";
import { u as pt } from "../../assets/resolveComponentProps.js";
import { u as To } from "../../assets/useSlotProps.js";
import { a as Po, u as Eo } from "../../assets/dividerClasses.js";
import { G as Ro } from "../../assets/Grow.js";
import { I as At } from "../../assets/IconButton.js";
import { c as Yt } from "../../assets/createSvgIcon.js";
import { D as Ao } from "../../assets/Divider.js";
var I = "top", Y = "bottom", X = "right", H = "left", dt = "auto", Ve = [I, Y, X, H], Te = "start", Ie = "end", Co = "clippingParents", Xt = "viewport", Le = "popper", Mo = "reference", Ct = /* @__PURE__ */ Ve.reduce(function(t, e) {
  return t.concat([e + "-" + Te, e + "-" + Ie]);
}, []), Gt = /* @__PURE__ */ [].concat(Ve, [dt]).reduce(function(t, e) {
  return t.concat([e, e + "-" + Te, e + "-" + Ie]);
}, []), jo = "beforeRead", Do = "read", So = "afterRead", $o = "beforeMain", Lo = "main", ko = "afterMain", Bo = "beforeWrite", No = "write", Wo = "afterWrite", Io = [jo, Do, So, $o, Lo, ko, Bo, No, Wo];
function oe(t) {
  return t ? (t.nodeName || "").toLowerCase() : null;
}
function V(t) {
  if (t == null)
    return window;
  if (t.toString() !== "[object Window]") {
    var e = t.ownerDocument;
    return e && e.defaultView || window;
  }
  return t;
}
function me(t) {
  var e = V(t).Element;
  return t instanceof e || t instanceof Element;
}
function q(t) {
  var e = V(t).HTMLElement;
  return t instanceof e || t instanceof HTMLElement;
}
function mt(t) {
  if (typeof ShadowRoot > "u")
    return !1;
  var e = V(t).ShadowRoot;
  return t instanceof e || t instanceof ShadowRoot;
}
function Ho(t) {
  var e = t.state;
  Object.keys(e.elements).forEach(function(o) {
    var n = e.styles[o] || {}, a = e.attributes[o] || {}, s = e.elements[o];
    !q(s) || !oe(s) || (Object.assign(s.style, n), Object.keys(a).forEach(function(c) {
      var i = a[c];
      i === !1 ? s.removeAttribute(c) : s.setAttribute(c, i === !0 ? "" : i);
    }));
  });
}
function Vo(t) {
  var e = t.state, o = {
    popper: {
      position: e.options.strategy,
      left: "0",
      top: "0",
      margin: "0"
    },
    arrow: {
      position: "absolute"
    },
    reference: {}
  };
  return Object.assign(e.elements.popper.style, o.popper), e.styles = o, e.elements.arrow && Object.assign(e.elements.arrow.style, o.arrow), function() {
    Object.keys(e.elements).forEach(function(n) {
      var a = e.elements[n], s = e.attributes[n] || {}, c = Object.keys(e.styles.hasOwnProperty(n) ? e.styles[n] : o[n]), i = c.reduce(function(p, f) {
        return p[f] = "", p;
      }, {});
      !q(a) || !oe(a) || (Object.assign(a.style, i), Object.keys(s).forEach(function(p) {
        a.removeAttribute(p);
      }));
    });
  };
}
const Fo = {
  name: "applyStyles",
  enabled: !0,
  phase: "write",
  fn: Ho,
  effect: Vo,
  requires: ["computeStyles"]
};
function te(t) {
  return t.split("-")[0];
}
var de = Math.max, et = Math.min, Pe = Math.round;
function lt() {
  var t = navigator.userAgentData;
  return t != null && t.brands && Array.isArray(t.brands) ? t.brands.map(function(e) {
    return e.brand + "/" + e.version;
  }).join(" ") : navigator.userAgent;
}
function Kt() {
  return !/^((?!chrome|android).)*safari/i.test(lt());
}
function Ee(t, e, o) {
  e === void 0 && (e = !1), o === void 0 && (o = !1);
  var n = t.getBoundingClientRect(), a = 1, s = 1;
  e && q(t) && (a = t.offsetWidth > 0 && Pe(n.width) / t.offsetWidth || 1, s = t.offsetHeight > 0 && Pe(n.height) / t.offsetHeight || 1);
  var c = me(t) ? V(t) : window, i = c.visualViewport, p = !Kt() && o, f = (n.left + (p && i ? i.offsetLeft : 0)) / a, l = (n.top + (p && i ? i.offsetTop : 0)) / s, g = n.width / a, x = n.height / s;
  return {
    width: g,
    height: x,
    top: l,
    right: f + g,
    bottom: l + x,
    left: f,
    x: f,
    y: l
  };
}
function vt(t) {
  var e = Ee(t), o = t.offsetWidth, n = t.offsetHeight;
  return Math.abs(e.width - o) <= 1 && (o = e.width), Math.abs(e.height - n) <= 1 && (n = e.height), {
    x: t.offsetLeft,
    y: t.offsetTop,
    width: o,
    height: n
  };
}
function Jt(t, e) {
  var o = e.getRootNode && e.getRootNode();
  if (t.contains(e))
    return !0;
  if (o && mt(o)) {
    var n = e;
    do {
      if (n && t.isSameNode(n))
        return !0;
      n = n.parentNode || n.host;
    } while (n);
  }
  return !1;
}
function ae(t) {
  return V(t).getComputedStyle(t);
}
function Uo(t) {
  return ["table", "td", "th"].indexOf(oe(t)) >= 0;
}
function pe(t) {
  return ((me(t) ? t.ownerDocument : (
    // $FlowFixMe[prop-missing]
    t.document
  )) || window.document).documentElement;
}
function rt(t) {
  return oe(t) === "html" ? t : (
    // this is a quicker (but less type safe) way to save quite some bytes from the bundle
    // $FlowFixMe[incompatible-return]
    // $FlowFixMe[prop-missing]
    t.assignedSlot || // step into the shadow DOM of the parent of a slotted node
    t.parentNode || // DOM Element detected
    (mt(t) ? t.host : null) || // ShadowRoot detected
    // $FlowFixMe[incompatible-call]: HTMLElement is a Node
    pe(t)
  );
}
function Mt(t) {
  return !q(t) || // https://github.com/popperjs/popper-core/issues/837
  ae(t).position === "fixed" ? null : t.offsetParent;
}
function zo(t) {
  var e = /firefox/i.test(lt()), o = /Trident/i.test(lt());
  if (o && q(t)) {
    var n = ae(t);
    if (n.position === "fixed")
      return null;
  }
  var a = rt(t);
  for (mt(a) && (a = a.host); q(a) && ["html", "body"].indexOf(oe(a)) < 0; ) {
    var s = ae(a);
    if (s.transform !== "none" || s.perspective !== "none" || s.contain === "paint" || ["transform", "perspective"].indexOf(s.willChange) !== -1 || e && s.willChange === "filter" || e && s.filter && s.filter !== "none")
      return a;
    a = a.parentNode;
  }
  return null;
}
function Fe(t) {
  for (var e = V(t), o = Mt(t); o && Uo(o) && ae(o).position === "static"; )
    o = Mt(o);
  return o && (oe(o) === "html" || oe(o) === "body" && ae(o).position === "static") ? e : o || zo(t) || e;
}
function ht(t) {
  return ["top", "bottom"].indexOf(t) >= 0 ? "x" : "y";
}
function Ne(t, e, o) {
  return de(t, et(e, o));
}
function qo(t, e, o) {
  var n = Ne(t, e, o);
  return n > o ? o : n;
}
function Qt() {
  return {
    top: 0,
    right: 0,
    bottom: 0,
    left: 0
  };
}
function Zt(t) {
  return Object.assign({}, Qt(), t);
}
function _t(t, e) {
  return e.reduce(function(o, n) {
    return o[n] = t, o;
  }, {});
}
var Yo = function(e, o) {
  return e = typeof e == "function" ? e(Object.assign({}, o.rects, {
    placement: o.placement
  })) : e, Zt(typeof e != "number" ? e : _t(e, Ve));
};
function Xo(t) {
  var e, o = t.state, n = t.name, a = t.options, s = o.elements.arrow, c = o.modifiersData.popperOffsets, i = te(o.placement), p = ht(i), f = [H, X].indexOf(i) >= 0, l = f ? "height" : "width";
  if (!(!s || !c)) {
    var g = Yo(a.padding, o), x = vt(s), u = p === "y" ? I : H, T = p === "y" ? Y : X, m = o.rects.reference[l] + o.rects.reference[p] - c[p] - o.rects.popper[l], h = c[p] - o.rects.reference[p], w = Fe(s), E = w ? p === "y" ? w.clientHeight || 0 : w.clientWidth || 0 : 0, O = m / 2 - h / 2, d = g[u], y = E - x[l] - g[T], b = E / 2 - x[l] / 2 + O, R = Ne(d, b, y), S = p;
    o.modifiersData[n] = (e = {}, e[S] = R, e.centerOffset = R - b, e);
  }
}
function Go(t) {
  var e = t.state, o = t.options, n = o.element, a = n === void 0 ? "[data-popper-arrow]" : n;
  a != null && (typeof a == "string" && (a = e.elements.popper.querySelector(a), !a) || Jt(e.elements.popper, a) && (e.elements.arrow = a));
}
const Ko = {
  name: "arrow",
  enabled: !0,
  phase: "main",
  fn: Xo,
  effect: Go,
  requires: ["popperOffsets"],
  requiresIfExists: ["preventOverflow"]
};
function Re(t) {
  return t.split("-")[1];
}
var Jo = {
  top: "auto",
  right: "auto",
  bottom: "auto",
  left: "auto"
};
function Qo(t, e) {
  var o = t.x, n = t.y, a = e.devicePixelRatio || 1;
  return {
    x: Pe(o * a) / a || 0,
    y: Pe(n * a) / a || 0
  };
}
function jt(t) {
  var e, o = t.popper, n = t.popperRect, a = t.placement, s = t.variation, c = t.offsets, i = t.position, p = t.gpuAcceleration, f = t.adaptive, l = t.roundOffsets, g = t.isFixed, x = c.x, u = x === void 0 ? 0 : x, T = c.y, m = T === void 0 ? 0 : T, h = typeof l == "function" ? l({
    x: u,
    y: m
  }) : {
    x: u,
    y: m
  };
  u = h.x, m = h.y;
  var w = c.hasOwnProperty("x"), E = c.hasOwnProperty("y"), O = H, d = I, y = window;
  if (f) {
    var b = Fe(o), R = "clientHeight", S = "clientWidth";
    if (b === V(o) && (b = pe(o), ae(b).position !== "static" && i === "absolute" && (R = "scrollHeight", S = "scrollWidth")), b = b, a === I || (a === H || a === X) && s === Ie) {
      d = Y;
      var A = g && b === y && y.visualViewport ? y.visualViewport.height : (
        // $FlowFixMe[prop-missing]
        b[R]
      );
      m -= A - n.height, m *= p ? 1 : -1;
    }
    if (a === H || (a === I || a === Y) && s === Ie) {
      O = X;
      var P = g && b === y && y.visualViewport ? y.visualViewport.width : (
        // $FlowFixMe[prop-missing]
        b[S]
      );
      u -= P - n.width, u *= p ? 1 : -1;
    }
  }
  var C = Object.assign({
    position: i
  }, f && Jo), W = l === !0 ? Qo({
    x: u,
    y: m
  }, V(o)) : {
    x: u,
    y: m
  };
  if (u = W.x, m = W.y, p) {
    var j;
    return Object.assign({}, C, (j = {}, j[d] = E ? "0" : "", j[O] = w ? "0" : "", j.transform = (y.devicePixelRatio || 1) <= 1 ? "translate(" + u + "px, " + m + "px)" : "translate3d(" + u + "px, " + m + "px, 0)", j));
  }
  return Object.assign({}, C, (e = {}, e[d] = E ? m + "px" : "", e[O] = w ? u + "px" : "", e.transform = "", e));
}
function Zo(t) {
  var e = t.state, o = t.options, n = o.gpuAcceleration, a = n === void 0 ? !0 : n, s = o.adaptive, c = s === void 0 ? !0 : s, i = o.roundOffsets, p = i === void 0 ? !0 : i, f = {
    placement: te(e.placement),
    variation: Re(e.placement),
    popper: e.elements.popper,
    popperRect: e.rects.popper,
    gpuAcceleration: a,
    isFixed: e.options.strategy === "fixed"
  };
  e.modifiersData.popperOffsets != null && (e.styles.popper = Object.assign({}, e.styles.popper, jt(Object.assign({}, f, {
    offsets: e.modifiersData.popperOffsets,
    position: e.options.strategy,
    adaptive: c,
    roundOffsets: p
  })))), e.modifiersData.arrow != null && (e.styles.arrow = Object.assign({}, e.styles.arrow, jt(Object.assign({}, f, {
    offsets: e.modifiersData.arrow,
    position: "absolute",
    adaptive: !1,
    roundOffsets: p
  })))), e.attributes.popper = Object.assign({}, e.attributes.popper, {
    "data-popper-placement": e.placement
  });
}
const _o = {
  name: "computeStyles",
  enabled: !0,
  phase: "beforeWrite",
  fn: Zo,
  data: {}
};
var Ke = {
  passive: !0
};
function er(t) {
  var e = t.state, o = t.instance, n = t.options, a = n.scroll, s = a === void 0 ? !0 : a, c = n.resize, i = c === void 0 ? !0 : c, p = V(e.elements.popper), f = [].concat(e.scrollParents.reference, e.scrollParents.popper);
  return s && f.forEach(function(l) {
    l.addEventListener("scroll", o.update, Ke);
  }), i && p.addEventListener("resize", o.update, Ke), function() {
    s && f.forEach(function(l) {
      l.removeEventListener("scroll", o.update, Ke);
    }), i && p.removeEventListener("resize", o.update, Ke);
  };
}
const tr = {
  name: "eventListeners",
  enabled: !0,
  phase: "write",
  fn: function() {
  },
  effect: er,
  data: {}
};
var or = {
  left: "right",
  right: "left",
  bottom: "top",
  top: "bottom"
};
function Ze(t) {
  return t.replace(/left|right|bottom|top/g, function(e) {
    return or[e];
  });
}
var rr = {
  start: "end",
  end: "start"
};
function Dt(t) {
  return t.replace(/start|end/g, function(e) {
    return rr[e];
  });
}
function gt(t) {
  var e = V(t), o = e.pageXOffset, n = e.pageYOffset;
  return {
    scrollLeft: o,
    scrollTop: n
  };
}
function yt(t) {
  return Ee(pe(t)).left + gt(t).scrollLeft;
}
function nr(t, e) {
  var o = V(t), n = pe(t), a = o.visualViewport, s = n.clientWidth, c = n.clientHeight, i = 0, p = 0;
  if (a) {
    s = a.width, c = a.height;
    var f = Kt();
    (f || !f && e === "fixed") && (i = a.offsetLeft, p = a.offsetTop);
  }
  return {
    width: s,
    height: c,
    x: i + yt(t),
    y: p
  };
}
function ar(t) {
  var e, o = pe(t), n = gt(t), a = (e = t.ownerDocument) == null ? void 0 : e.body, s = de(o.scrollWidth, o.clientWidth, a ? a.scrollWidth : 0, a ? a.clientWidth : 0), c = de(o.scrollHeight, o.clientHeight, a ? a.scrollHeight : 0, a ? a.clientHeight : 0), i = -n.scrollLeft + yt(t), p = -n.scrollTop;
  return ae(a || o).direction === "rtl" && (i += de(o.clientWidth, a ? a.clientWidth : 0) - s), {
    width: s,
    height: c,
    x: i,
    y: p
  };
}
function bt(t) {
  var e = ae(t), o = e.overflow, n = e.overflowX, a = e.overflowY;
  return /auto|scroll|overlay|hidden/.test(o + a + n);
}
function eo(t) {
  return ["html", "body", "#document"].indexOf(oe(t)) >= 0 ? t.ownerDocument.body : q(t) && bt(t) ? t : eo(rt(t));
}
function We(t, e) {
  var o;
  e === void 0 && (e = []);
  var n = eo(t), a = n === ((o = t.ownerDocument) == null ? void 0 : o.body), s = V(n), c = a ? [s].concat(s.visualViewport || [], bt(n) ? n : []) : n, i = e.concat(c);
  return a ? i : (
    // $FlowFixMe[incompatible-call]: isBody tells us target will be an HTMLElement here
    i.concat(We(rt(c)))
  );
}
function ct(t) {
  return Object.assign({}, t, {
    left: t.x,
    top: t.y,
    right: t.x + t.width,
    bottom: t.y + t.height
  });
}
function ir(t, e) {
  var o = Ee(t, !1, e === "fixed");
  return o.top = o.top + t.clientTop, o.left = o.left + t.clientLeft, o.bottom = o.top + t.clientHeight, o.right = o.left + t.clientWidth, o.width = t.clientWidth, o.height = t.clientHeight, o.x = o.left, o.y = o.top, o;
}
function St(t, e, o) {
  return e === Xt ? ct(nr(t, o)) : me(e) ? ir(e, o) : ct(ar(pe(t)));
}
function sr(t) {
  var e = We(rt(t)), o = ["absolute", "fixed"].indexOf(ae(t).position) >= 0, n = o && q(t) ? Fe(t) : t;
  return me(n) ? e.filter(function(a) {
    return me(a) && Jt(a, n) && oe(a) !== "body";
  }) : [];
}
function pr(t, e, o, n) {
  var a = e === "clippingParents" ? sr(t) : [].concat(e), s = [].concat(a, [o]), c = s[0], i = s.reduce(function(p, f) {
    var l = St(t, f, n);
    return p.top = de(l.top, p.top), p.right = et(l.right, p.right), p.bottom = et(l.bottom, p.bottom), p.left = de(l.left, p.left), p;
  }, St(t, c, n));
  return i.width = i.right - i.left, i.height = i.bottom - i.top, i.x = i.left, i.y = i.top, i;
}
function to(t) {
  var e = t.reference, o = t.element, n = t.placement, a = n ? te(n) : null, s = n ? Re(n) : null, c = e.x + e.width / 2 - o.width / 2, i = e.y + e.height / 2 - o.height / 2, p;
  switch (a) {
    case I:
      p = {
        x: c,
        y: e.y - o.height
      };
      break;
    case Y:
      p = {
        x: c,
        y: e.y + e.height
      };
      break;
    case X:
      p = {
        x: e.x + e.width,
        y: i
      };
      break;
    case H:
      p = {
        x: e.x - o.width,
        y: i
      };
      break;
    default:
      p = {
        x: e.x,
        y: e.y
      };
  }
  var f = a ? ht(a) : null;
  if (f != null) {
    var l = f === "y" ? "height" : "width";
    switch (s) {
      case Te:
        p[f] = p[f] - (e[l] / 2 - o[l] / 2);
        break;
      case Ie:
        p[f] = p[f] + (e[l] / 2 - o[l] / 2);
        break;
    }
  }
  return p;
}
function He(t, e) {
  e === void 0 && (e = {});
  var o = e, n = o.placement, a = n === void 0 ? t.placement : n, s = o.strategy, c = s === void 0 ? t.strategy : s, i = o.boundary, p = i === void 0 ? Co : i, f = o.rootBoundary, l = f === void 0 ? Xt : f, g = o.elementContext, x = g === void 0 ? Le : g, u = o.altBoundary, T = u === void 0 ? !1 : u, m = o.padding, h = m === void 0 ? 0 : m, w = Zt(typeof h != "number" ? h : _t(h, Ve)), E = x === Le ? Mo : Le, O = t.rects.popper, d = t.elements[T ? E : x], y = pr(me(d) ? d : d.contextElement || pe(t.elements.popper), p, l, c), b = Ee(t.elements.reference), R = to({
    reference: b,
    element: O,
    strategy: "absolute",
    placement: a
  }), S = ct(Object.assign({}, O, R)), A = x === Le ? S : b, P = {
    top: y.top - A.top + w.top,
    bottom: A.bottom - y.bottom + w.bottom,
    left: y.left - A.left + w.left,
    right: A.right - y.right + w.right
  }, C = t.modifiersData.offset;
  if (x === Le && C) {
    var W = C[a];
    Object.keys(P).forEach(function(j) {
      var G = [X, Y].indexOf(j) >= 0 ? 1 : -1, K = [I, Y].indexOf(j) >= 0 ? "y" : "x";
      P[j] += W[K] * G;
    });
  }
  return P;
}
function lr(t, e) {
  e === void 0 && (e = {});
  var o = e, n = o.placement, a = o.boundary, s = o.rootBoundary, c = o.padding, i = o.flipVariations, p = o.allowedAutoPlacements, f = p === void 0 ? Gt : p, l = Re(n), g = l ? i ? Ct : Ct.filter(function(T) {
    return Re(T) === l;
  }) : Ve, x = g.filter(function(T) {
    return f.indexOf(T) >= 0;
  });
  x.length === 0 && (x = g);
  var u = x.reduce(function(T, m) {
    return T[m] = He(t, {
      placement: m,
      boundary: a,
      rootBoundary: s,
      padding: c
    })[te(m)], T;
  }, {});
  return Object.keys(u).sort(function(T, m) {
    return u[T] - u[m];
  });
}
function cr(t) {
  if (te(t) === dt)
    return [];
  var e = Ze(t);
  return [Dt(t), e, Dt(e)];
}
function fr(t) {
  var e = t.state, o = t.options, n = t.name;
  if (!e.modifiersData[n]._skip) {
    for (var a = o.mainAxis, s = a === void 0 ? !0 : a, c = o.altAxis, i = c === void 0 ? !0 : c, p = o.fallbackPlacements, f = o.padding, l = o.boundary, g = o.rootBoundary, x = o.altBoundary, u = o.flipVariations, T = u === void 0 ? !0 : u, m = o.allowedAutoPlacements, h = e.options.placement, w = te(h), E = w === h, O = p || (E || !T ? [Ze(h)] : cr(h)), d = [h].concat(O).reduce(function(J, F) {
      return J.concat(te(F) === dt ? lr(e, {
        placement: F,
        boundary: l,
        rootBoundary: g,
        padding: f,
        flipVariations: T,
        allowedAutoPlacements: m
      }) : F);
    }, []), y = e.rects.reference, b = e.rects.popper, R = /* @__PURE__ */ new Map(), S = !0, A = d[0], P = 0; P < d.length; P++) {
      var C = d[P], W = te(C), j = Re(C) === Te, G = [I, Y].indexOf(W) >= 0, K = G ? "width" : "height", $ = He(e, {
        placement: C,
        boundary: l,
        rootBoundary: g,
        altBoundary: x,
        padding: f
      }), D = G ? j ? X : H : j ? Y : I;
      y[K] > b[K] && (D = Ze(D));
      var N = Ze(D), re = [];
      if (s && re.push($[W] <= 0), i && re.push($[D] <= 0, $[N] <= 0), re.every(function(J) {
        return J;
      })) {
        A = C, S = !1;
        break;
      }
      R.set(C, re);
    }
    if (S)
      for (var B = T ? 3 : 1, ve = function(F) {
        var ie = d.find(function(se) {
          var U = R.get(se);
          if (U)
            return U.slice(0, F).every(function(ge) {
              return ge;
            });
        });
        if (ie)
          return A = ie, "break";
      }, ne = B; ne > 0; ne--) {
        var he = ve(ne);
        if (he === "break") break;
      }
    e.placement !== A && (e.modifiersData[n]._skip = !0, e.placement = A, e.reset = !0);
  }
}
const ur = {
  name: "flip",
  enabled: !0,
  phase: "main",
  fn: fr,
  requiresIfExists: ["offset"],
  data: {
    _skip: !1
  }
};
function $t(t, e, o) {
  return o === void 0 && (o = {
    x: 0,
    y: 0
  }), {
    top: t.top - e.height - o.y,
    right: t.right - e.width + o.x,
    bottom: t.bottom - e.height + o.y,
    left: t.left - e.width - o.x
  };
}
function Lt(t) {
  return [I, X, Y, H].some(function(e) {
    return t[e] >= 0;
  });
}
function dr(t) {
  var e = t.state, o = t.name, n = e.rects.reference, a = e.rects.popper, s = e.modifiersData.preventOverflow, c = He(e, {
    elementContext: "reference"
  }), i = He(e, {
    altBoundary: !0
  }), p = $t(c, n), f = $t(i, a, s), l = Lt(p), g = Lt(f);
  e.modifiersData[o] = {
    referenceClippingOffsets: p,
    popperEscapeOffsets: f,
    isReferenceHidden: l,
    hasPopperEscaped: g
  }, e.attributes.popper = Object.assign({}, e.attributes.popper, {
    "data-popper-reference-hidden": l,
    "data-popper-escaped": g
  });
}
const mr = {
  name: "hide",
  enabled: !0,
  phase: "main",
  requiresIfExists: ["preventOverflow"],
  fn: dr
};
function vr(t, e, o) {
  var n = te(t), a = [H, I].indexOf(n) >= 0 ? -1 : 1, s = typeof o == "function" ? o(Object.assign({}, e, {
    placement: t
  })) : o, c = s[0], i = s[1];
  return c = c || 0, i = (i || 0) * a, [H, X].indexOf(n) >= 0 ? {
    x: i,
    y: c
  } : {
    x: c,
    y: i
  };
}
function hr(t) {
  var e = t.state, o = t.options, n = t.name, a = o.offset, s = a === void 0 ? [0, 0] : a, c = Gt.reduce(function(l, g) {
    return l[g] = vr(g, e.rects, s), l;
  }, {}), i = c[e.placement], p = i.x, f = i.y;
  e.modifiersData.popperOffsets != null && (e.modifiersData.popperOffsets.x += p, e.modifiersData.popperOffsets.y += f), e.modifiersData[n] = c;
}
const gr = {
  name: "offset",
  enabled: !0,
  phase: "main",
  requires: ["popperOffsets"],
  fn: hr
};
function yr(t) {
  var e = t.state, o = t.name;
  e.modifiersData[o] = to({
    reference: e.rects.reference,
    element: e.rects.popper,
    strategy: "absolute",
    placement: e.placement
  });
}
const br = {
  name: "popperOffsets",
  enabled: !0,
  phase: "read",
  fn: yr,
  data: {}
};
function wr(t) {
  return t === "x" ? "y" : "x";
}
function Or(t) {
  var e = t.state, o = t.options, n = t.name, a = o.mainAxis, s = a === void 0 ? !0 : a, c = o.altAxis, i = c === void 0 ? !1 : c, p = o.boundary, f = o.rootBoundary, l = o.altBoundary, g = o.padding, x = o.tether, u = x === void 0 ? !0 : x, T = o.tetherOffset, m = T === void 0 ? 0 : T, h = He(e, {
    boundary: p,
    rootBoundary: f,
    padding: g,
    altBoundary: l
  }), w = te(e.placement), E = Re(e.placement), O = !E, d = ht(w), y = wr(d), b = e.modifiersData.popperOffsets, R = e.rects.reference, S = e.rects.popper, A = typeof m == "function" ? m(Object.assign({}, e.rects, {
    placement: e.placement
  })) : m, P = typeof A == "number" ? {
    mainAxis: A,
    altAxis: A
  } : Object.assign({
    mainAxis: 0,
    altAxis: 0
  }, A), C = e.modifiersData.offset ? e.modifiersData.offset[e.placement] : null, W = {
    x: 0,
    y: 0
  };
  if (b) {
    if (s) {
      var j, G = d === "y" ? I : H, K = d === "y" ? Y : X, $ = d === "y" ? "height" : "width", D = b[d], N = D + h[G], re = D - h[K], B = u ? -S[$] / 2 : 0, ve = E === Te ? R[$] : S[$], ne = E === Te ? -S[$] : -R[$], he = e.elements.arrow, J = u && he ? vt(he) : {
        width: 0,
        height: 0
      }, F = e.modifiersData["arrow#persistent"] ? e.modifiersData["arrow#persistent"].padding : Qt(), ie = F[G], se = F[K], U = Ne(0, R[$], J[$]), ge = O ? R[$] / 2 - B - U - ie - P.mainAxis : ve - U - ie - P.mainAxis, at = O ? -R[$] / 2 + B + U + se + P.mainAxis : ne + U + se + P.mainAxis, ye = e.elements.arrow && Fe(e.elements.arrow), Q = ye ? d === "y" ? ye.clientTop || 0 : ye.clientLeft || 0 : 0, be = (j = C == null ? void 0 : C[d]) != null ? j : 0, le = D + ge - be - Q, we = D + at - be, Ae = Ne(u ? et(N, le) : N, D, u ? de(re, we) : re);
      b[d] = Ae, W[d] = Ae - D;
    }
    if (i) {
      var ce, Oe = d === "x" ? I : H, Ce = d === "x" ? Y : X, ee = b[y], fe = y === "y" ? "height" : "width", Me = ee + h[Oe], je = ee - h[Ce], De = [I, H].indexOf(w) !== -1, Ue = (ce = C == null ? void 0 : C[y]) != null ? ce : 0, ze = De ? Me : ee - R[fe] - S[fe] - Ue + P.altAxis, xe = De ? ee + R[fe] + S[fe] - Ue - P.altAxis : je, qe = u && De ? qo(ze, ee, xe) : Ne(u ? ze : Me, ee, u ? xe : je);
      b[y] = qe, W[y] = qe - ee;
    }
    e.modifiersData[n] = W;
  }
}
const xr = {
  name: "preventOverflow",
  enabled: !0,
  phase: "main",
  fn: Or,
  requiresIfExists: ["offset"]
};
function Tr(t) {
  return {
    scrollLeft: t.scrollLeft,
    scrollTop: t.scrollTop
  };
}
function Pr(t) {
  return t === V(t) || !q(t) ? gt(t) : Tr(t);
}
function Er(t) {
  var e = t.getBoundingClientRect(), o = Pe(e.width) / t.offsetWidth || 1, n = Pe(e.height) / t.offsetHeight || 1;
  return o !== 1 || n !== 1;
}
function Rr(t, e, o) {
  o === void 0 && (o = !1);
  var n = q(e), a = q(e) && Er(e), s = pe(e), c = Ee(t, a, o), i = {
    scrollLeft: 0,
    scrollTop: 0
  }, p = {
    x: 0,
    y: 0
  };
  return (n || !n && !o) && ((oe(e) !== "body" || // https://github.com/popperjs/popper-core/issues/1078
  bt(s)) && (i = Pr(e)), q(e) ? (p = Ee(e, !0), p.x += e.clientLeft, p.y += e.clientTop) : s && (p.x = yt(s))), {
    x: c.left + i.scrollLeft - p.x,
    y: c.top + i.scrollTop - p.y,
    width: c.width,
    height: c.height
  };
}
function Ar(t) {
  var e = /* @__PURE__ */ new Map(), o = /* @__PURE__ */ new Set(), n = [];
  t.forEach(function(s) {
    e.set(s.name, s);
  });
  function a(s) {
    o.add(s.name);
    var c = [].concat(s.requires || [], s.requiresIfExists || []);
    c.forEach(function(i) {
      if (!o.has(i)) {
        var p = e.get(i);
        p && a(p);
      }
    }), n.push(s);
  }
  return t.forEach(function(s) {
    o.has(s.name) || a(s);
  }), n;
}
function Cr(t) {
  var e = Ar(t);
  return Io.reduce(function(o, n) {
    return o.concat(e.filter(function(a) {
      return a.phase === n;
    }));
  }, []);
}
function Mr(t) {
  var e;
  return function() {
    return e || (e = new Promise(function(o) {
      Promise.resolve().then(function() {
        e = void 0, o(t());
      });
    })), e;
  };
}
function jr(t) {
  var e = t.reduce(function(o, n) {
    var a = o[n.name];
    return o[n.name] = a ? Object.assign({}, a, n, {
      options: Object.assign({}, a.options, n.options),
      data: Object.assign({}, a.data, n.data)
    }) : n, o;
  }, {});
  return Object.keys(e).map(function(o) {
    return e[o];
  });
}
var kt = {
  placement: "bottom",
  modifiers: [],
  strategy: "absolute"
};
function Bt() {
  for (var t = arguments.length, e = new Array(t), o = 0; o < t; o++)
    e[o] = arguments[o];
  return !e.some(function(n) {
    return !(n && typeof n.getBoundingClientRect == "function");
  });
}
function Dr(t) {
  t === void 0 && (t = {});
  var e = t, o = e.defaultModifiers, n = o === void 0 ? [] : o, a = e.defaultOptions, s = a === void 0 ? kt : a;
  return function(i, p, f) {
    f === void 0 && (f = s);
    var l = {
      placement: "bottom",
      orderedModifiers: [],
      options: Object.assign({}, kt, s),
      modifiersData: {},
      elements: {
        reference: i,
        popper: p
      },
      attributes: {},
      styles: {}
    }, g = [], x = !1, u = {
      state: l,
      setOptions: function(w) {
        var E = typeof w == "function" ? w(l.options) : w;
        m(), l.options = Object.assign({}, s, l.options, E), l.scrollParents = {
          reference: me(i) ? We(i) : i.contextElement ? We(i.contextElement) : [],
          popper: We(p)
        };
        var O = Cr(jr([].concat(n, l.options.modifiers)));
        return l.orderedModifiers = O.filter(function(d) {
          return d.enabled;
        }), T(), u.update();
      },
      // Sync update – it will always be executed, even if not necessary. This
      // is useful for low frequency updates where sync behavior simplifies the
      // logic.
      // For high frequency updates (e.g. `resize` and `scroll` events), always
      // prefer the async Popper#update method
      forceUpdate: function() {
        if (!x) {
          var w = l.elements, E = w.reference, O = w.popper;
          if (Bt(E, O)) {
            l.rects = {
              reference: Rr(E, Fe(O), l.options.strategy === "fixed"),
              popper: vt(O)
            }, l.reset = !1, l.placement = l.options.placement, l.orderedModifiers.forEach(function(P) {
              return l.modifiersData[P.name] = Object.assign({}, P.data);
            });
            for (var d = 0; d < l.orderedModifiers.length; d++) {
              if (l.reset === !0) {
                l.reset = !1, d = -1;
                continue;
              }
              var y = l.orderedModifiers[d], b = y.fn, R = y.options, S = R === void 0 ? {} : R, A = y.name;
              typeof b == "function" && (l = b({
                state: l,
                options: S,
                name: A,
                instance: u
              }) || l);
            }
          }
        }
      },
      // Async and optimistically optimized update – it will not be executed if
      // not necessary (debounced to run at most once-per-tick)
      update: Mr(function() {
        return new Promise(function(h) {
          u.forceUpdate(), h(l);
        });
      }),
      destroy: function() {
        m(), x = !0;
      }
    };
    if (!Bt(i, p))
      return u;
    u.setOptions(f).then(function(h) {
      !x && f.onFirstUpdate && f.onFirstUpdate(h);
    });
    function T() {
      l.orderedModifiers.forEach(function(h) {
        var w = h.name, E = h.options, O = E === void 0 ? {} : E, d = h.effect;
        if (typeof d == "function") {
          var y = d({
            state: l,
            name: w,
            instance: u,
            options: O
          }), b = function() {
          };
          g.push(y || b);
        }
      });
    }
    function m() {
      g.forEach(function(h) {
        return h();
      }), g = [];
    }
    return u;
  };
}
var Sr = [tr, br, _o, Fo, gr, ur, xr, Ko, mr], $r = /* @__PURE__ */ Dr({
  defaultModifiers: Sr
});
function Lr(t) {
  return Wt("MuiPopper", t);
}
Vt("MuiPopper", ["root"]);
function kr(t, e) {
  if (e === "ltr")
    return t;
  switch (t) {
    case "bottom-end":
      return "bottom-start";
    case "bottom-start":
      return "bottom-end";
    case "top-end":
      return "top-start";
    case "top-start":
      return "top-end";
    default:
      return t;
  }
}
function tt(t) {
  return typeof t == "function" ? t() : t;
}
function nt(t) {
  return t.nodeType !== void 0;
}
function Br(t) {
  return !nt(t);
}
const Nr = (t) => {
  const {
    classes: e
  } = t;
  return Ft({
    root: ["root"]
  }, Lr, e);
}, Wr = {}, Ir = /* @__PURE__ */ M.forwardRef(function(e, o) {
  const {
    anchorEl: n,
    children: a,
    direction: s,
    disablePortal: c,
    modifiers: i,
    open: p,
    placement: f,
    popperOptions: l,
    popperRef: g,
    slotProps: x = {},
    slots: u = {},
    TransitionProps: T,
    // @ts-ignore internal logic
    ownerState: m,
    // prevent from spreading to DOM, it can come from the parent component e.g. Select.
    ...h
  } = e, w = M.useRef(null), E = pt(w, o), O = M.useRef(null), d = pt(O, g), y = M.useRef(d);
  Tt(() => {
    y.current = d;
  }, [d]), M.useImperativeHandle(g, () => O.current, []);
  const b = kr(f, s), [R, S] = M.useState(b), [A, P] = M.useState(tt(n));
  M.useEffect(() => {
    O.current && O.current.forceUpdate();
  }), M.useEffect(() => {
    n && P(tt(n));
  }, [n]), Tt(() => {
    if (!A || !p)
      return;
    const K = (N) => {
      S(N.placement);
    };
    if (process.env.NODE_ENV !== "production" && A && nt(A) && A.nodeType === 1) {
      const N = A.getBoundingClientRect();
      process.env.NODE_ENV !== "test" && N.top === 0 && N.left === 0 && N.right === 0 && N.bottom === 0 && console.warn(["MUI: The `anchorEl` prop provided to the component is invalid.", "The anchor element should be part of the document layout.", "Make sure the element is present in the document or that it's not display none."].join(`
`));
    }
    let $ = [{
      name: "preventOverflow",
      options: {
        altBoundary: c
      }
    }, {
      name: "flip",
      options: {
        altBoundary: c
      }
    }, {
      name: "onUpdate",
      enabled: !0,
      phase: "afterWrite",
      fn: ({
        state: N
      }) => {
        K(N);
      }
    }];
    i != null && ($ = $.concat(i)), l && l.modifiers != null && ($ = $.concat(l.modifiers));
    const D = $r(A, w.current, {
      placement: b,
      ...l,
      modifiers: $
    });
    return y.current(D), () => {
      D.destroy(), y.current(null);
    };
  }, [A, c, i, p, l, b]);
  const C = {
    placement: R
  };
  T !== null && (C.TransitionProps = T);
  const W = Nr(e), j = u.root ?? "div", G = To({
    elementType: j,
    externalSlotProps: x.root,
    externalForwardedProps: h,
    additionalProps: {
      role: "tooltip",
      ref: E
    },
    ownerState: e,
    className: W.root
  });
  return /* @__PURE__ */ k(j, {
    ...G,
    children: typeof a == "function" ? a(C) : a
  });
}), oo = /* @__PURE__ */ M.forwardRef(function(e, o) {
  const {
    anchorEl: n,
    children: a,
    container: s,
    direction: c = "ltr",
    disablePortal: i = !1,
    keepMounted: p = !1,
    modifiers: f,
    open: l,
    placement: g = "bottom",
    popperOptions: x = Wr,
    popperRef: u,
    style: T,
    transition: m = !1,
    slotProps: h = {},
    slots: w = {},
    ...E
  } = e, [O, d] = M.useState(!0), y = () => {
    d(!1);
  }, b = () => {
    d(!0);
  };
  if (!p && !l && (!m || O))
    return null;
  let R;
  if (s)
    R = s;
  else if (n) {
    const P = tt(n);
    R = P && nt(P) ? Rt(P).body : Rt(null).body;
  }
  const S = !l && p && (!m || O) ? "none" : void 0, A = m ? {
    in: l,
    onEnter: y,
    onExited: b
  } : void 0;
  return /* @__PURE__ */ k(wo, {
    disablePortal: i,
    container: R,
    children: /* @__PURE__ */ k(Ir, {
      anchorEl: n,
      direction: c,
      disablePortal: i,
      modifiers: f,
      ref: o,
      open: m ? !O : l,
      placement: g,
      popperOptions: x,
      popperRef: u,
      slotProps: h,
      slots: w,
      ...E,
      style: {
        // Prevents scroll issue, waiting for Popper.js to add this style once initiated.
        position: "fixed",
        // Fix Popper.js display issue
        top: 0,
        left: 0,
        display: S,
        ...T
      },
      TransitionProps: A,
      children: a
    })
  });
});
process.env.NODE_ENV !== "production" && (oo.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │ To update them, edit the TypeScript types and run `pnpm proptypes`. │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * An HTML element, [virtualElement](https://popper.js.org/docs/v2/virtual-elements/),
   * or a function that returns either.
   * It's used to set the position of the popper.
   * The return value will passed as the reference object of the Popper instance.
   */
  anchorEl: go(r.oneOfType([_e, r.object, r.func]), (t) => {
    if (t.open) {
      const e = tt(t.anchorEl);
      if (e && nt(e) && e.nodeType === 1) {
        const o = e.getBoundingClientRect();
        if (process.env.NODE_ENV !== "test" && o.top === 0 && o.left === 0 && o.right === 0 && o.bottom === 0)
          return new Error(["MUI: The `anchorEl` prop provided to the component is invalid.", "The anchor element should be part of the document layout.", "Make sure the element is present in the document or that it's not display none."].join(`
`));
      } else if (!e || typeof e.getBoundingClientRect != "function" || Br(e) && e.contextElement != null && e.contextElement.nodeType !== 1)
        return new Error(["MUI: The `anchorEl` prop provided to the component is invalid.", "It should be an HTML element instance or a virtualElement ", "(https://popper.js.org/docs/v2/virtual-elements/)."].join(`
`));
    }
    return null;
  }),
  /**
   * Popper render function or node.
   */
  children: r.oneOfType([r.node, r.func]),
  /**
   * An HTML element or function that returns one.
   * The `container` will have the portal children appended to it.
   *
   * You can also provide a callback, which is called in a React layout effect.
   * This lets you set the container from a ref, and also makes server-side rendering possible.
   *
   * By default, it uses the body of the top-level document object,
   * so it's simply `document.body` most of the time.
   */
  container: r.oneOfType([_e, r.func]),
  /**
   * Direction of the text.
   * @default 'ltr'
   */
  direction: r.oneOf(["ltr", "rtl"]),
  /**
   * The `children` will be under the DOM hierarchy of the parent component.
   * @default false
   */
  disablePortal: r.bool,
  /**
   * Always keep the children in the DOM.
   * This prop can be useful in SEO situation or
   * when you want to maximize the responsiveness of the Popper.
   * @default false
   */
  keepMounted: r.bool,
  /**
   * Popper.js is based on a "plugin-like" architecture,
   * most of its features are fully encapsulated "modifiers".
   *
   * A modifier is a function that is called each time Popper.js needs to
   * compute the position of the popper.
   * For this reason, modifiers should be very performant to avoid bottlenecks.
   * To learn how to create a modifier, [read the modifiers documentation](https://popper.js.org/docs/v2/modifiers/).
   */
  modifiers: r.arrayOf(r.shape({
    data: r.object,
    effect: r.func,
    enabled: r.bool,
    fn: r.func,
    name: r.any,
    options: r.object,
    phase: r.oneOf(["afterMain", "afterRead", "afterWrite", "beforeMain", "beforeRead", "beforeWrite", "main", "read", "write"]),
    requires: r.arrayOf(r.string),
    requiresIfExists: r.arrayOf(r.string)
  })),
  /**
   * If `true`, the component is shown.
   */
  open: r.bool.isRequired,
  /**
   * Popper placement.
   * @default 'bottom'
   */
  placement: r.oneOf(["auto-end", "auto-start", "auto", "bottom-end", "bottom-start", "bottom", "left-end", "left-start", "left", "right-end", "right-start", "right", "top-end", "top-start", "top"]),
  /**
   * Options provided to the [`Popper.js`](https://popper.js.org/docs/v2/constructors/#options) instance.
   * @default {}
   */
  popperOptions: r.shape({
    modifiers: r.array,
    onFirstUpdate: r.func,
    placement: r.oneOf(["auto-end", "auto-start", "auto", "bottom-end", "bottom-start", "bottom", "left-end", "left-start", "left", "right-end", "right-start", "right", "top-end", "top-start", "top"]),
    strategy: r.oneOf(["absolute", "fixed"])
  }),
  /**
   * A ref that points to the used popper instance.
   */
  popperRef: qt,
  /**
   * The props used for each slot inside the Popper.
   * @default {}
   */
  slotProps: r.shape({
    root: r.oneOfType([r.func, r.object])
  }),
  /**
   * The components used for each slot inside the Popper.
   * Either a string to use a HTML element or a component.
   * @default {}
   */
  slots: r.shape({
    root: r.elementType
  }),
  /**
   * Help supporting a react-transition-group/Transition component.
   * @default false
   */
  transition: r.bool
});
const Hr = ot(oo, {
  name: "MuiPopper",
  slot: "Root",
  overridesResolver: (t, e) => e.root
})({}), wt = /* @__PURE__ */ M.forwardRef(function(e, o) {
  const n = zt(), a = Ut({
    props: e,
    name: "MuiPopper"
  }), {
    anchorEl: s,
    component: c,
    components: i,
    componentsProps: p,
    container: f,
    disablePortal: l,
    keepMounted: g,
    modifiers: x,
    open: u,
    placement: T,
    popperOptions: m,
    popperRef: h,
    transition: w,
    slots: E,
    slotProps: O,
    ...d
  } = a, y = (E == null ? void 0 : E.root) ?? (i == null ? void 0 : i.Root), b = {
    anchorEl: s,
    container: f,
    disablePortal: l,
    keepMounted: g,
    modifiers: x,
    open: u,
    placement: T,
    popperOptions: m,
    popperRef: h,
    transition: w,
    ...d
  };
  return /* @__PURE__ */ k(Hr, {
    as: c,
    direction: n ? "rtl" : "ltr",
    slots: {
      root: y
    },
    slotProps: O ?? p,
    ...b,
    ref: o
  });
});
process.env.NODE_ENV !== "production" && (wt.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │ To update them, edit the TypeScript types and run `pnpm proptypes`. │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * An HTML element, [virtualElement](https://popper.js.org/docs/v2/virtual-elements/),
   * or a function that returns either.
   * It's used to set the position of the popper.
   * The return value will passed as the reference object of the Popper instance.
   */
  anchorEl: r.oneOfType([_e, r.object, r.func]),
  /**
   * Popper render function or node.
   */
  children: r.oneOfType([r.node, r.func]),
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: r.elementType,
  /**
   * The components used for each slot inside the Popper.
   * Either a string to use a HTML element or a component.
   *
   * @deprecated use the `slots` prop instead. This prop will be removed in v7. [How to migrate](/material-ui/migration/migrating-from-deprecated-apis/).
   * @default {}
   */
  components: r.shape({
    Root: r.elementType
  }),
  /**
   * The props used for each slot inside the Popper.
   *
   * @deprecated use the `slotProps` prop instead. This prop will be removed in v7. [How to migrate](/material-ui/migration/migrating-from-deprecated-apis/).
   * @default {}
   */
  componentsProps: r.shape({
    root: r.oneOfType([r.func, r.object])
  }),
  /**
   * An HTML element or function that returns one.
   * The `container` will have the portal children appended to it.
   *
   * You can also provide a callback, which is called in a React layout effect.
   * This lets you set the container from a ref, and also makes server-side rendering possible.
   *
   * By default, it uses the body of the top-level document object,
   * so it's simply `document.body` most of the time.
   */
  container: r.oneOfType([_e, r.func]),
  /**
   * The `children` will be under the DOM hierarchy of the parent component.
   * @default false
   */
  disablePortal: r.bool,
  /**
   * Always keep the children in the DOM.
   * This prop can be useful in SEO situation or
   * when you want to maximize the responsiveness of the Popper.
   * @default false
   */
  keepMounted: r.bool,
  /**
   * Popper.js is based on a "plugin-like" architecture,
   * most of its features are fully encapsulated "modifiers".
   *
   * A modifier is a function that is called each time Popper.js needs to
   * compute the position of the popper.
   * For this reason, modifiers should be very performant to avoid bottlenecks.
   * To learn how to create a modifier, [read the modifiers documentation](https://popper.js.org/docs/v2/modifiers/).
   */
  modifiers: r.arrayOf(r.shape({
    data: r.object,
    effect: r.func,
    enabled: r.bool,
    fn: r.func,
    name: r.any,
    options: r.object,
    phase: r.oneOf(["afterMain", "afterRead", "afterWrite", "beforeMain", "beforeRead", "beforeWrite", "main", "read", "write"]),
    requires: r.arrayOf(r.string),
    requiresIfExists: r.arrayOf(r.string)
  })),
  /**
   * If `true`, the component is shown.
   */
  open: r.bool.isRequired,
  /**
   * Popper placement.
   * @default 'bottom'
   */
  placement: r.oneOf(["auto-end", "auto-start", "auto", "bottom-end", "bottom-start", "bottom", "left-end", "left-start", "left", "right-end", "right-start", "right", "top-end", "top-start", "top"]),
  /**
   * Options provided to the [`Popper.js`](https://popper.js.org/docs/v2/constructors/#options) instance.
   * @default {}
   */
  popperOptions: r.shape({
    modifiers: r.array,
    onFirstUpdate: r.func,
    placement: r.oneOf(["auto-end", "auto-start", "auto", "bottom-end", "bottom-start", "bottom", "left-end", "left-start", "left", "right-end", "right-start", "right", "top-end", "top-start", "top"]),
    strategy: r.oneOf(["absolute", "fixed"])
  }),
  /**
   * A ref that points to the used popper instance.
   */
  popperRef: qt,
  /**
   * The props used for each slot inside the Popper.
   * @default {}
   */
  slotProps: r.shape({
    root: r.oneOfType([r.func, r.object])
  }),
  /**
   * The components used for each slot inside the Popper.
   * Either a string to use a HTML element or a component.
   * @default {}
   */
  slots: r.shape({
    root: r.elementType
  }),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: r.oneOfType([r.arrayOf(r.oneOfType([r.func, r.object, r.bool])), r.func, r.object]),
  /**
   * Help supporting a react-transition-group/Transition component.
   * @default false
   */
  transition: r.bool
});
function Vr(t) {
  return Wt("MuiTooltip", t);
}
const L = Vt("MuiTooltip", ["popper", "popperInteractive", "popperArrow", "popperClose", "tooltip", "tooltipArrow", "touch", "tooltipPlacementLeft", "tooltipPlacementRight", "tooltipPlacementTop", "tooltipPlacementBottom", "arrow"]);
function Fr(t) {
  return Math.round(t * 1e5) / 1e5;
}
const Ur = (t) => {
  const {
    classes: e,
    disableInteractive: o,
    arrow: n,
    touch: a,
    placement: s
  } = t, c = {
    popper: ["popper", !o && "popperInteractive", n && "popperArrow"],
    tooltip: ["tooltip", n && "tooltipArrow", a && "touch", `tooltipPlacement${It(s.split("-")[0])}`],
    arrow: ["arrow"]
  };
  return Ft(c, Vr, e);
}, zr = ot(wt, {
  name: "MuiTooltip",
  slot: "Popper",
  overridesResolver: (t, e) => {
    const {
      ownerState: o
    } = t;
    return [e.popper, !o.disableInteractive && e.popperInteractive, o.arrow && e.popperArrow, !o.open && e.popperClose];
  }
})(ut(({
  theme: t
}) => ({
  zIndex: (t.vars || t).zIndex.tooltip,
  pointerEvents: "none",
  variants: [{
    props: ({
      ownerState: e
    }) => !e.disableInteractive,
    style: {
      pointerEvents: "auto"
    }
  }, {
    props: ({
      open: e
    }) => !e,
    style: {
      pointerEvents: "none"
    }
  }, {
    props: ({
      ownerState: e
    }) => e.arrow,
    style: {
      [`&[data-popper-placement*="bottom"] .${L.arrow}`]: {
        top: 0,
        marginTop: "-0.71em",
        "&::before": {
          transformOrigin: "0 100%"
        }
      },
      [`&[data-popper-placement*="top"] .${L.arrow}`]: {
        bottom: 0,
        marginBottom: "-0.71em",
        "&::before": {
          transformOrigin: "100% 0"
        }
      },
      [`&[data-popper-placement*="right"] .${L.arrow}`]: {
        height: "1em",
        width: "0.71em",
        "&::before": {
          transformOrigin: "100% 100%"
        }
      },
      [`&[data-popper-placement*="left"] .${L.arrow}`]: {
        height: "1em",
        width: "0.71em",
        "&::before": {
          transformOrigin: "0 0"
        }
      }
    }
  }, {
    props: ({
      ownerState: e
    }) => e.arrow && !e.isRtl,
    style: {
      [`&[data-popper-placement*="right"] .${L.arrow}`]: {
        left: 0,
        marginLeft: "-0.71em"
      }
    }
  }, {
    props: ({
      ownerState: e
    }) => e.arrow && !!e.isRtl,
    style: {
      [`&[data-popper-placement*="right"] .${L.arrow}`]: {
        right: 0,
        marginRight: "-0.71em"
      }
    }
  }, {
    props: ({
      ownerState: e
    }) => e.arrow && !e.isRtl,
    style: {
      [`&[data-popper-placement*="left"] .${L.arrow}`]: {
        right: 0,
        marginRight: "-0.71em"
      }
    }
  }, {
    props: ({
      ownerState: e
    }) => e.arrow && !!e.isRtl,
    style: {
      [`&[data-popper-placement*="left"] .${L.arrow}`]: {
        left: 0,
        marginLeft: "-0.71em"
      }
    }
  }]
}))), qr = ot("div", {
  name: "MuiTooltip",
  slot: "Tooltip",
  overridesResolver: (t, e) => {
    const {
      ownerState: o
    } = t;
    return [e.tooltip, o.touch && e.touch, o.arrow && e.tooltipArrow, e[`tooltipPlacement${It(o.placement.split("-")[0])}`]];
  }
})(ut(({
  theme: t
}) => ({
  backgroundColor: t.vars ? t.vars.palette.Tooltip.bg : Ht(t.palette.grey[700], 0.92),
  borderRadius: (t.vars || t).shape.borderRadius,
  color: (t.vars || t).palette.common.white,
  fontFamily: t.typography.fontFamily,
  padding: "4px 8px",
  fontSize: t.typography.pxToRem(11),
  maxWidth: 300,
  margin: 2,
  wordWrap: "break-word",
  fontWeight: t.typography.fontWeightMedium,
  [`.${L.popper}[data-popper-placement*="left"] &`]: {
    transformOrigin: "right center"
  },
  [`.${L.popper}[data-popper-placement*="right"] &`]: {
    transformOrigin: "left center"
  },
  [`.${L.popper}[data-popper-placement*="top"] &`]: {
    transformOrigin: "center bottom",
    marginBottom: "14px"
  },
  [`.${L.popper}[data-popper-placement*="bottom"] &`]: {
    transformOrigin: "center top",
    marginTop: "14px"
  },
  variants: [{
    props: ({
      ownerState: e
    }) => e.arrow,
    style: {
      position: "relative",
      margin: 0
    }
  }, {
    props: ({
      ownerState: e
    }) => e.touch,
    style: {
      padding: "8px 16px",
      fontSize: t.typography.pxToRem(14),
      lineHeight: `${Fr(16 / 14)}em`,
      fontWeight: t.typography.fontWeightRegular
    }
  }, {
    props: ({
      ownerState: e
    }) => !e.isRtl,
    style: {
      [`.${L.popper}[data-popper-placement*="left"] &`]: {
        marginRight: "14px"
      },
      [`.${L.popper}[data-popper-placement*="right"] &`]: {
        marginLeft: "14px"
      }
    }
  }, {
    props: ({
      ownerState: e
    }) => !e.isRtl && e.touch,
    style: {
      [`.${L.popper}[data-popper-placement*="left"] &`]: {
        marginRight: "24px"
      },
      [`.${L.popper}[data-popper-placement*="right"] &`]: {
        marginLeft: "24px"
      }
    }
  }, {
    props: ({
      ownerState: e
    }) => !!e.isRtl,
    style: {
      [`.${L.popper}[data-popper-placement*="left"] &`]: {
        marginLeft: "14px"
      },
      [`.${L.popper}[data-popper-placement*="right"] &`]: {
        marginRight: "14px"
      }
    }
  }, {
    props: ({
      ownerState: e
    }) => !!e.isRtl && e.touch,
    style: {
      [`.${L.popper}[data-popper-placement*="left"] &`]: {
        marginLeft: "24px"
      },
      [`.${L.popper}[data-popper-placement*="right"] &`]: {
        marginRight: "24px"
      }
    }
  }, {
    props: ({
      ownerState: e
    }) => e.touch,
    style: {
      [`.${L.popper}[data-popper-placement*="top"] &`]: {
        marginBottom: "24px"
      }
    }
  }, {
    props: ({
      ownerState: e
    }) => e.touch,
    style: {
      [`.${L.popper}[data-popper-placement*="bottom"] &`]: {
        marginTop: "24px"
      }
    }
  }]
}))), Yr = ot("span", {
  name: "MuiTooltip",
  slot: "Arrow",
  overridesResolver: (t, e) => e.arrow
})(ut(({
  theme: t
}) => ({
  overflow: "hidden",
  position: "absolute",
  width: "1em",
  height: "0.71em",
  boxSizing: "border-box",
  color: t.vars ? t.vars.palette.Tooltip.bg : Ht(t.palette.grey[700], 0.9),
  "&::before": {
    content: '""',
    margin: "auto",
    display: "block",
    width: "100%",
    height: "100%",
    backgroundColor: "currentColor",
    transform: "rotate(45deg)"
  }
})));
let Je = !1;
const Nt = new bo();
let ke = {
  x: 0,
  y: 0
};
function Qe(t, e) {
  return (o, ...n) => {
    e && e(o, ...n), t(o, ...n);
  };
}
const ft = /* @__PURE__ */ M.forwardRef(function(e, o) {
  const n = Ut({
    props: e,
    name: "MuiTooltip"
  }), {
    arrow: a = !1,
    children: s,
    classes: c,
    components: i = {},
    componentsProps: p = {},
    describeChild: f = !1,
    disableFocusListener: l = !1,
    disableHoverListener: g = !1,
    disableInteractive: x = !1,
    disableTouchListener: u = !1,
    enterDelay: T = 100,
    enterNextDelay: m = 0,
    enterTouchDelay: h = 700,
    followCursor: w = !1,
    id: E,
    leaveDelay: O = 0,
    leaveTouchDelay: d = 1500,
    onClose: y,
    onOpen: b,
    open: R,
    placement: S = "bottom",
    PopperComponent: A,
    PopperProps: P = {},
    slotProps: C = {},
    slots: W = {},
    title: j,
    TransitionComponent: G,
    TransitionProps: K,
    ...$
  } = n, D = /* @__PURE__ */ M.isValidElement(s) ? s : /* @__PURE__ */ k("span", {
    children: s
  }), N = yo(), re = zt(), [B, ve] = M.useState(), [ne, he] = M.useState(null), J = M.useRef(!1), F = x || w, ie = Xe(), se = Xe(), U = Xe(), ge = Xe(), [at, ye] = Po({
    controlled: R,
    default: !1,
    name: "Tooltip",
    state: "open"
  });
  let Q = at;
  if (process.env.NODE_ENV !== "production") {
    const {
      current: v
    } = M.useRef(R !== void 0);
    M.useEffect(() => {
      B && B.disabled && !v && j !== "" && B.tagName.toLowerCase() === "button" && console.warn(["MUI: You are providing a disabled `button` child to the Tooltip component.", "A disabled element does not fire events.", "Tooltip needs to listen to the child element's events to display the title.", "", "Add a simple wrapper element, such as a `span`."].join(`
`));
    }, [j, B, v]);
  }
  const be = Eo(E), le = M.useRef(), we = Pt(() => {
    le.current !== void 0 && (document.body.style.WebkitUserSelect = le.current, le.current = void 0), ge.clear();
  });
  M.useEffect(() => we, [we]);
  const Ae = (v) => {
    Nt.clear(), Je = !0, ye(!0), b && !Q && b(v);
  }, ce = Pt(
    /**
     * @param {React.SyntheticEvent | Event} event
     */
    (v) => {
      Nt.start(800 + O, () => {
        Je = !1;
      }), ye(!1), y && Q && y(v), ie.start(N.transitions.duration.shortest, () => {
        J.current = !1;
      });
    }
  ), Oe = (v) => {
    J.current && v.type !== "touchstart" || (B && B.removeAttribute("title"), se.clear(), U.clear(), T || Je && m ? se.start(Je ? m : T, () => {
      Ae(v);
    }) : Ae(v));
  }, Ce = (v) => {
    se.clear(), U.start(O, () => {
      ce(v);
    });
  }, [, ee] = M.useState(!1), fe = (v) => {
    Et(v.target) || (ee(!1), Ce(v));
  }, Me = (v) => {
    B || ve(v.currentTarget), Et(v.target) && (ee(!0), Oe(v));
  }, je = (v) => {
    J.current = !0;
    const _ = D.props;
    _.onTouchStart && _.onTouchStart(v);
  }, De = (v) => {
    je(v), U.clear(), ie.clear(), we(), le.current = document.body.style.WebkitUserSelect, document.body.style.WebkitUserSelect = "none", ge.start(h, () => {
      document.body.style.WebkitUserSelect = le.current, Oe(v);
    });
  }, Ue = (v) => {
    D.props.onTouchEnd && D.props.onTouchEnd(v), we(), U.start(d, () => {
      ce(v);
    });
  };
  M.useEffect(() => {
    if (!Q)
      return;
    function v(_) {
      _.key === "Escape" && ce(_);
    }
    return document.addEventListener("keydown", v), () => {
      document.removeEventListener("keydown", v);
    };
  }, [ce, Q]);
  const ze = pt(Oo(D), ve, o);
  !j && j !== 0 && (Q = !1);
  const xe = M.useRef(), qe = (v) => {
    const _ = D.props;
    _.onMouseMove && _.onMouseMove(v), ke = {
      x: v.clientX,
      y: v.clientY
    }, xe.current && xe.current.update();
  }, Se = {}, it = typeof j == "string";
  f ? (Se.title = !Q && it && !g ? j : null, Se["aria-describedby"] = Q ? be : null) : (Se["aria-label"] = it ? j : null, Se["aria-labelledby"] = Q && !it ? be : null);
  const z = {
    ...Se,
    ...$,
    ...D.props,
    className: xt($.className, D.props.className),
    onTouchStart: je,
    ref: ze,
    ...w ? {
      onMouseMove: qe
    } : {}
  };
  process.env.NODE_ENV !== "production" && (z["data-mui-internal-clone-element"] = !0, M.useEffect(() => {
    B && !B.getAttribute("data-mui-internal-clone-element") && console.error(["MUI: The `children` component of the Tooltip is not forwarding its props correctly.", "Please make sure that props are spread on the same element that the ref is applied to."].join(`
`));
  }, [B]));
  const $e = {};
  u || (z.onTouchStart = De, z.onTouchEnd = Ue), g || (z.onMouseOver = Qe(Oe, z.onMouseOver), z.onMouseLeave = Qe(Ce, z.onMouseLeave), F || ($e.onMouseOver = Oe, $e.onMouseLeave = Ce)), l || (z.onFocus = Qe(Me, z.onFocus), z.onBlur = Qe(fe, z.onBlur), F || ($e.onFocus = Me, $e.onBlur = fe)), process.env.NODE_ENV !== "production" && D.props.title && console.error(["MUI: You have provided a `title` prop to the child of <Tooltip />.", `Remove this title prop \`${D.props.title}\` or the Tooltip component.`].join(`
`));
  const ue = {
    ...n,
    isRtl: re,
    arrow: a,
    disableInteractive: F,
    placement: S,
    PopperComponentProp: A,
    touch: J.current
  }, Z = typeof C.popper == "function" ? C.popper(ue) : C.popper, ro = M.useMemo(() => {
    var _, Ot;
    let v = [{
      name: "arrow",
      enabled: !!ne,
      options: {
        element: ne,
        padding: 4
      }
    }];
    return (_ = P.popperOptions) != null && _.modifiers && (v = v.concat(P.popperOptions.modifiers)), (Ot = Z == null ? void 0 : Z.popperOptions) != null && Ot.modifiers && (v = v.concat(Z.popperOptions.modifiers)), {
      ...P.popperOptions,
      ...Z == null ? void 0 : Z.popperOptions,
      modifiers: v
    };
  }, [ne, P.popperOptions, Z == null ? void 0 : Z.popperOptions]), st = Ur(ue), no = typeof C.transition == "function" ? C.transition(ue) : C.transition, Ye = {
    slots: {
      popper: i.Popper,
      transition: i.Transition ?? G,
      tooltip: i.Tooltip,
      arrow: i.Arrow,
      ...W
    },
    slotProps: {
      arrow: C.arrow ?? p.arrow,
      popper: {
        ...P,
        ...Z ?? p.popper
      },
      // resolvedPopperProps can be spread because it's already an object
      tooltip: C.tooltip ?? p.tooltip,
      transition: {
        ...K,
        ...no ?? p.transition
      }
    }
  }, [ao, io] = Ge("popper", {
    elementType: zr,
    externalForwardedProps: Ye,
    ownerState: ue,
    className: xt(st.popper, P == null ? void 0 : P.className)
  }), [so, po] = Ge("transition", {
    elementType: Ro,
    externalForwardedProps: Ye,
    ownerState: ue
  }), [lo, co] = Ge("tooltip", {
    elementType: qr,
    className: st.tooltip,
    externalForwardedProps: Ye,
    ownerState: ue
  }), [fo, uo] = Ge("arrow", {
    elementType: Yr,
    className: st.arrow,
    externalForwardedProps: Ye,
    ownerState: ue,
    ref: he
  });
  return /* @__PURE__ */ Be(M.Fragment, {
    children: [/* @__PURE__ */ M.cloneElement(D, z), /* @__PURE__ */ k(ao, {
      as: A ?? wt,
      placement: S,
      anchorEl: w ? {
        getBoundingClientRect: () => ({
          top: ke.y,
          left: ke.x,
          right: ke.x,
          bottom: ke.y,
          width: 0,
          height: 0
        })
      } : B,
      popperRef: xe,
      open: B ? Q : !1,
      id: be,
      transition: !0,
      ...$e,
      ...io,
      popperOptions: ro,
      children: ({
        TransitionProps: v
      }) => /* @__PURE__ */ k(so, {
        timeout: N.transitions.duration.shorter,
        ...v,
        ...po,
        children: /* @__PURE__ */ Be(lo, {
          ...co,
          children: [j, a ? /* @__PURE__ */ k(fo, {
            ...uo
          }) : null]
        })
      })
    })]
  });
});
process.env.NODE_ENV !== "production" && (ft.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * If `true`, adds an arrow to the tooltip.
   * @default false
   */
  arrow: r.bool,
  /**
   * Tooltip reference element.
   */
  children: xo.isRequired,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: r.object,
  /**
   * @ignore
   */
  className: r.string,
  /**
   * The components used for each slot inside.
   *
   * @deprecated use the `slots` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  components: r.shape({
    Arrow: r.elementType,
    Popper: r.elementType,
    Tooltip: r.elementType,
    Transition: r.elementType
  }),
  /**
   * The extra props for the slot components.
   * You can override the existing props or add new ones.
   *
   * @deprecated use the `slotProps` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  componentsProps: r.shape({
    arrow: r.object,
    popper: r.object,
    tooltip: r.object,
    transition: r.object
  }),
  /**
   * Set to `true` if the `title` acts as an accessible description.
   * By default the `title` acts as an accessible label for the child.
   * @default false
   */
  describeChild: r.bool,
  /**
   * Do not respond to focus-visible events.
   * @default false
   */
  disableFocusListener: r.bool,
  /**
   * Do not respond to hover events.
   * @default false
   */
  disableHoverListener: r.bool,
  /**
   * Makes a tooltip not interactive, i.e. it will close when the user
   * hovers over the tooltip before the `leaveDelay` is expired.
   * @default false
   */
  disableInteractive: r.bool,
  /**
   * Do not respond to long press touch events.
   * @default false
   */
  disableTouchListener: r.bool,
  /**
   * The number of milliseconds to wait before showing the tooltip.
   * This prop won't impact the enter touch delay (`enterTouchDelay`).
   * @default 100
   */
  enterDelay: r.number,
  /**
   * The number of milliseconds to wait before showing the tooltip when one was already recently opened.
   * @default 0
   */
  enterNextDelay: r.number,
  /**
   * The number of milliseconds a user must touch the element before showing the tooltip.
   * @default 700
   */
  enterTouchDelay: r.number,
  /**
   * If `true`, the tooltip follow the cursor over the wrapped element.
   * @default false
   */
  followCursor: r.bool,
  /**
   * This prop is used to help implement the accessibility logic.
   * If you don't provide this prop. It falls back to a randomly generated id.
   */
  id: r.string,
  /**
   * The number of milliseconds to wait before hiding the tooltip.
   * This prop won't impact the leave touch delay (`leaveTouchDelay`).
   * @default 0
   */
  leaveDelay: r.number,
  /**
   * The number of milliseconds after the user stops touching an element before hiding the tooltip.
   * @default 1500
   */
  leaveTouchDelay: r.number,
  /**
   * Callback fired when the component requests to be closed.
   *
   * @param {React.SyntheticEvent} event The event source of the callback.
   */
  onClose: r.func,
  /**
   * Callback fired when the component requests to be open.
   *
   * @param {React.SyntheticEvent} event The event source of the callback.
   */
  onOpen: r.func,
  /**
   * If `true`, the component is shown.
   */
  open: r.bool,
  /**
   * Tooltip placement.
   * @default 'bottom'
   */
  placement: r.oneOf(["bottom-end", "bottom-start", "bottom", "left-end", "left-start", "left", "right-end", "right-start", "right", "top-end", "top-start", "top"]),
  /**
   * The component used for the popper.
   * @deprecated use the `slots.popper` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   */
  PopperComponent: r.elementType,
  /**
   * Props applied to the [`Popper`](https://mui.com/material-ui/api/popper/) element.
   * @deprecated use the `slotProps.popper` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   * @default {}
   */
  PopperProps: r.object,
  /**
   * The props used for each slot inside.
   * @default {}
   */
  slotProps: r.shape({
    arrow: r.oneOfType([r.func, r.object]),
    popper: r.oneOfType([r.func, r.object]),
    tooltip: r.oneOfType([r.func, r.object]),
    transition: r.oneOfType([r.func, r.object])
  }),
  /**
   * The components used for each slot inside.
   * @default {}
   */
  slots: r.shape({
    arrow: r.elementType,
    popper: r.elementType,
    tooltip: r.elementType,
    transition: r.elementType
  }),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: r.oneOfType([r.arrayOf(r.oneOfType([r.func, r.object, r.bool])), r.func, r.object]),
  /**
   * Tooltip title. Zero-length titles string, undefined, null and false are never displayed.
   */
  title: r.node,
  /**
   * The component used for the transition.
   * [Follow this guide](https://mui.com/material-ui/transitions/#transitioncomponent-prop) to learn more about the requirements for this component.
   * @deprecated use the `slots.transition` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   */
  TransitionComponent: r.elementType,
  /**
   * Props applied to the transition element.
   * By default, the element is based on this [`Transition`](https://reactcommunity.org/react-transition-group/transition/) component.
   * @deprecated use the `slotProps.transition` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   * @default {}
   */
  TransitionProps: r.object
});
const Xr = Yt(/* @__PURE__ */ k("path", {
  d: "m18 7-1.41-1.41-6.34 6.34 1.41 1.41zm4.24-1.41L11.66 16.17 7.48 12l-1.41 1.41L11.66 19l12-12zM.41 13.41 6 19l1.41-1.41L1.83 12z"
}), "DoneAll"), Gr = Yt(/* @__PURE__ */ k("path", {
  d: "M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6"
}), "Settings"), fn = (t) => {
  const e = t.title ?? /* @__PURE__ */ k(vo, { variant: "body1", children: "Notifications" });
  return /* @__PURE__ */ Be(mo, { children: [
    /* @__PURE__ */ Be(
      "div",
      {
        style: {
          display: "flex",
          justifyContent: "space-between",
          alignItems: "center",
          padding: "16px 0 12px 0"
        },
        children: [
          e,
          /* @__PURE__ */ Be(ho, { children: [
            /* @__PURE__ */ k(ft, { title: "Resolve all", children: /* @__PURE__ */ k(
              At,
              {
                size: "small",
                onClick: () => {
                  t.button1ClickHandler && t.button1ClickHandler("ALL");
                },
                children: /* @__PURE__ */ k(Xr, { fontSize: "small" })
              }
            ) }),
            /* @__PURE__ */ k(ft, { title: "Notification Preferences", children: /* @__PURE__ */ k(At, { size: "small", onClick: t.button2ClickHandler, children: /* @__PURE__ */ k(Gr, { fontSize: "small" }) }) })
          ] })
        ]
      }
    ),
    /* @__PURE__ */ k(Ao, {})
  ] });
};
export {
  fn as InboxHeader
};
