// @notificationapi/react/dist/components/Notifications/NotificationLauncher.js
import { jsx as o, jsxs as l } from "react/jsx-runtime";
import { useState as c, useContext as y } from "react";
import { Inbox as z } from "./Inbox.js";
import { UnreadBadge as T } from "./UnreadBadge.js";
import { NotificationAPIContext as N } from "../Provider/context.js";
import { NotificationPreferencesPopup as k } from "../Preferences/NotificationPreferencesPopup.js";
import { getThemeColors as v } from "../../utils/theme.js";
import { W as H, L as A } from "../../assets/WebPushOptInMessage.js";
import { u as L } from "../../assets/ButtonBase.js";
import { N as w } from "../../assets/channelUtils.js";
import { I as W } from "../../assets/IconButton.js";
import { P as B } from "../../assets/Popover.js";
import { D as E } from "../../assets/Divider.js";
const K = (e) => {
  var f, u, h, p, g, m, b, I;
  const [x, d] = c(!1), [r, s] = c(!1), i = y(N), [C, P] = c(null), O = L(), n = v(O);
  if (!i)
    return null;
  const t = {
    buttonIcon: e.buttonIcon || /* @__PURE__ */ o(
      w,
      {
        style: {
          fontSize: e.buttonIconSize || 20,
          color: e.iconColor || n.icon
        }
      }
    ),
    buttonStyles: {
      width: 40,
      height: 40,
      ...e.buttonStyles
    },
    popupWidth: e.popupWidth || 400,
    popupHeight: e.popupHeight || 600,
    buttonIconSize: e.buttonIconSize || 20,
    iconColor: e.iconColor || n.icon,
    pagination: e.pagination || "INFINITE_SCROLL",
    pageSize: e.pageSize || 10,
    pagePosition: e.pagePosition || "top",
    popupZIndex: e.popupZIndex || 1030,
    unreadBadgeProps: e.unreadBadgeProps ?? {},
    offsetX: e.offsetX || 16,
    offsetY: e.offsetY || 16,
    position: e.position || "BOTTOM_RIGHT",
    count: e.count || "COUNT_UNOPENED_NOTIFICATIONS",
    filter: e.filter || "ALL",
    header: {
      title: (f = e.header) == null ? void 0 : f.title,
      button1ClickHandler: ((u = e.header) == null ? void 0 : u.button1ClickHandler) ?? i.markAsArchived,
      button2ClickHandler: ((h = e.header) == null ? void 0 : h.button2ClickHandler) ?? (() => d(!0))
    },
    renderers: {
      notification: (p = e.renderers) == null ? void 0 : p.notification
    },
    popoverPosition: {
      anchorOrigin: {
        vertical: ((m = (g = e.popoverPosition) == null ? void 0 : g.anchorOrigin) == null ? void 0 : m.vertical) ?? "top",
        horizontal: ((I = (b = e.popoverPosition) == null ? void 0 : b.anchorOrigin) == null ? void 0 : I.horizontal) ?? "left"
      }
    },
    newTab: e.newTab ?? !1
  }, S = (a) => {
    s(!r), P(a == null ? void 0 : a.currentTarget), r && i.markAsOpened();
  };
  return /* @__PURE__ */ l(
    "div",
    {
      style: {
        position: "fixed",
        right: t.offsetX,
        bottom: t.offsetY,
        zIndex: 999
      },
      children: [
        /* @__PURE__ */ o(
          "div",
          {
            style: {
              display: "inline-block"
            },
            children: /* @__PURE__ */ o(T, { ...e.unreadBadgeProps, count: t.count, children: /* @__PURE__ */ o(W, { style: t.buttonStyles, onClick: S, children: t.buttonIcon }) })
          }
        ),
        /* @__PURE__ */ o(
          B,
          {
            open: r,
            anchorEl: C,
            anchorReference: "anchorEl",
            anchorOrigin: {
              horizontal: "center",
              vertical: "top"
            },
            transformOrigin: {
              vertical: "bottom",
              horizontal: "center"
            },
            onClose: () => s(!1),
            slotProps: {
              paper: {
                style: {
                  borderRadius: 8,
                  backgroundColor: n.paper,
                  color: n.text
                }
              }
            },
            children: /* @__PURE__ */ l(
              "div",
              {
                style: {
                  width: t.popupWidth,
                  padding: "0 16px",
                  zIndex: t.popupZIndex,
                  height: t.popupHeight,
                  backgroundColor: n.paper,
                  color: n.text
                },
                children: [
                  /* @__PURE__ */ o(
                    z,
                    {
                      maxHeight: t.popupHeight - 73,
                      pagination: t.pagination,
                      filter: t.filter,
                      pageSize: t.pageSize,
                      pagePosition: t.pagePosition,
                      notificationRenderer: t.renderers.notification,
                      header: t.header,
                      newTab: t.newTab
                    }
                  ),
                  i.webPushOptInMessage && localStorage.getItem("hideWebPushOptInMessage") !== "true" && /* @__PURE__ */ l("div", { children: [
                    /* @__PURE__ */ o(
                      E,
                      {
                        style: { margin: "10px 0", borderColor: n.divider }
                      }
                    ),
                    /* @__PURE__ */ o(
                      H,
                      {
                        hideAfterInteraction: !0,
                        icon: /* @__PURE__ */ o(
                          A,
                          {
                            type: "text",
                            style: { marginLeft: "9px" }
                          }
                        ),
                        alertContainerStyle: { maxWidth: "345px" }
                      }
                    )
                  ] })
                ]
              }
            )
          }
        ),
        /* @__PURE__ */ o(
          k,
          {
            open: x,
            onClose: () => d(!1)
          }
        )
      ]
    }
  );
};
export {
  K as NotificationLauncher
};
