// @notificationapi/react/dist/components/Notifications/NotificationFeed.js
import { jsx as n, jsxs as f } from "react/jsx-runtime";
import { useState as h, useContext as u, useEffect as p } from "react";
import { Inbox as x } from "./Inbox.js";
import { NotificationAPIContext as b } from "../Provider/context.js";
import { NotificationPreferencesPopup as S } from "../Preferences/NotificationPreferencesPopup.js";
import { Filter as C } from "./interface.js";
import { getThemeColors as I } from "../../utils/theme.js";
import { W as H } from "../../assets/WebPushOptInMessage.js";
import { c as P } from "../../assets/createSvgIcon.js";
import { u as y } from "../../assets/ButtonBase.js";
import { D as v } from "../../assets/Divider.js";
const M = P(/* @__PURE__ */ n("path", {
  d: "M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2m6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56M12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96M4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2s.06 1.34.14 2zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56m2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8M12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96M14.34 14H9.66c-.09-.66-.16-1.32-.16-2s.07-1.35.16-2h4.68c.09.65.16 1.32.16 2s-.07 1.34-.16 2m.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56M16.36 14c.08-.66.14-1.32.14-2s-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2z"
}), "Language"), R = (e) => {
  var a, c, s, l;
  const [m, o] = h(!1), i = u(b), g = y(), r = I(g);
  if (p(() => {
    if (!i) return;
    i.markAsOpened();
    const d = setInterval(() => {
      i.markAsOpened();
    }, 5e3);
    return () => clearInterval(d);
  }, [i]), !i)
    return null;
  const t = {
    pagination: e.pagination || "INFINITE_SCROLL",
    pageSize: e.pageSize || 5,
    pagePosition: e.pagePosition || "top",
    style: e.style || {},
    filter: e.filter || C.ALL,
    infiniteScrollHeight: e.infiniteScrollHeight ? e.infiniteScrollHeight : window.innerHeight * 0.75,
    renderers: {
      notification: (a = e.renderers) == null ? void 0 : a.notification
    },
    header: {
      title: (c = e.header) == null ? void 0 : c.title,
      button1ClickHandler: ((s = e.header) == null ? void 0 : s.button1ClickHandler) ?? i.markAsArchived,
      button2ClickHandler: ((l = e.header) == null ? void 0 : l.button2ClickHandler) ?? (() => o(!0))
    },
    imageShape: e.imageShape || "circle",
    newTab: e.newTab ?? !1
  };
  return /* @__PURE__ */ f(
    "div",
    {
      style: {
        padding: "0 12px",
        boxSizing: "border-box",
        borderRadius: 8,
        background: r.paper,
        border: `1px solid ${r.border}`,
        color: r.text,
        ...e.style
      },
      children: [
        /* @__PURE__ */ n(
          x,
          {
            maxHeight: t.infiniteScrollHeight,
            pagination: t.pagination,
            filter: t.filter,
            pageSize: t.pageSize,
            pagePosition: t.pagePosition,
            notificationRenderer: t.renderers.notification,
            header: t.header,
            imageShape: t.imageShape,
            newTab: t.newTab
          }
        ),
        i.webPushOptInMessage && localStorage.getItem("hideWebPushOptInMessage") !== "true" && /* @__PURE__ */ f("div", { children: [
          /* @__PURE__ */ n(
            v,
            {
              style: { margin: "10px 0", borderColor: r.divider }
            }
          ),
          /* @__PURE__ */ n(
            H,
            {
              hideAfterInteraction: !0,
              icon: /* @__PURE__ */ n(M, { type: "text", style: { marginLeft: "9px" } }),
              descriptionStyle: {
                flexDirection: "column",
                // Stack the elements vertically
                justifyContent: "flex-start",
                // Align items to the left
                fontSize: "14px",
                alignItems: "flex-start"
                // Align items to the left
              },
              buttonContainerStyle: {
                justifyContent: "flex-start",
                // Align buttons to the left
                alignItems: "flex-start",
                // Align buttons to the left
                marginTop: "10px"
                // Add some space between message and buttons
              }
            }
          )
        ] }),
        /* @__PURE__ */ n(
          S,
          {
            open: m,
            onClose: () => o(!1)
          }
        )
      ]
    }
  );
};
export {
  R as NotificationFeed
};
