// @notificationapi/react/dist/components/Notifications/NotificationPopup.js
import { jsx as t, jsxs as a, Fragment as O } from "react/jsx-runtime";
import { useState as b, useContext as S } from "react";
import { Inbox as v } from "./Inbox.js";
import { UnreadBadge as N } from "./UnreadBadge.js";
import { NotificationAPIContext as k } from "../Provider/context.js";
import { NotificationPreferencesPopup as z } from "../Preferences/NotificationPreferencesPopup.js";
import { Filter as T } from "./interface.js";
import { getThemeColors as y } from "../../utils/theme.js";
import { W as H } from "../../assets/WebPushOptInMessage.js";
import { N as A } from "../../assets/channelUtils.js";
import { u as w } from "../../assets/ButtonBase.js";
import { I as B } from "../../assets/IconButton.js";
import { P as W } from "../../assets/Popover.js";
import { D as E } from "../../assets/Divider.js";
const V = (e) => {
  var d, u, h, s, p, g, f, m;
  const [I, l] = b(!1), [r, c] = b(null), i = S(k), P = w(), n = y(P);
  if (!i)
    return null;
  const o = {
    buttonIcon: e.buttonIcon || /* @__PURE__ */ t(
      A,
      {
        style: {
          fontSize: e.buttonIconSize || 20,
          color: e.iconColor || n.icon
        }
      }
    ),
    buttonStyles: {
      width: 40,
      height: 40,
      ...e.buttonStyles
    },
    popupWidth: e.popupWidth || 400,
    popupHeight: e.popupHeight || 600,
    buttonIconSize: e.buttonIconSize || 20,
    iconColor: e.iconColor || n.icon,
    pagination: e.pagination || "INFINITE_SCROLL",
    pageSize: e.pageSize || 10,
    pagePosition: e.pagePosition || "bottom",
    popupZIndex: e.popupZIndex || 1030,
    unreadBadgeProps: e.unreadBadgeProps ?? {},
    count: e.count || "COUNT_UNOPENED_NOTIFICATIONS",
    filter: e.filter || T.ALL,
    header: {
      title: (d = e.header) == null ? void 0 : d.title,
      button1ClickHandler: ((u = e.header) == null ? void 0 : u.button1ClickHandler) ?? i.markAsArchived,
      button2ClickHandler: ((h = e.header) == null ? void 0 : h.button2ClickHandler) ?? (() => l(!0))
    },
    renderers: {
      notification: (s = e.renderers) == null ? void 0 : s.notification
    },
    popoverPosition: {
      anchorOrigin: {
        vertical: ((g = (p = e.popoverPosition) == null ? void 0 : p.anchorOrigin) == null ? void 0 : g.vertical) ?? "top",
        horizontal: ((m = (f = e.popoverPosition) == null ? void 0 : f.anchorOrigin) == null ? void 0 : m.horizontal) ?? "left"
      }
    },
    newTab: e.newTab ?? !1
  }, C = (x) => {
    c(x.currentTarget), i.markAsOpened();
  };
  return /* @__PURE__ */ a(O, { children: [
    /* @__PURE__ */ t(
      "div",
      {
        style: {
          display: "inline-block"
        },
        children: /* @__PURE__ */ t(
          N,
          {
            ...e.unreadBadgeProps,
            count: o.count,
            filter: o.filter,
            children: /* @__PURE__ */ t(B, { onClick: C, style: o.buttonStyles, children: o.buttonIcon })
          }
        )
      }
    ),
    /* @__PURE__ */ t(
      W,
      {
        open: !!r,
        onClose: () => c(null),
        anchorEl: r,
        slotProps: {
          paper: {
            style: {
              borderRadius: 8,
              backgroundColor: n.paper,
              color: n.text
            }
          }
        },
        "aria-hidden": !r,
        anchorOrigin: o.popoverPosition.anchorOrigin,
        children: /* @__PURE__ */ a(
          "div",
          {
            style: {
              width: o.popupWidth,
              padding: "0 16px",
              zIndex: e.popupZIndex,
              height: o.popupHeight,
              backgroundColor: n.paper,
              color: n.text
            },
            children: [
              /* @__PURE__ */ t(
                v,
                {
                  maxHeight: o.popupHeight - 73,
                  pagination: o.pagination,
                  filter: o.filter,
                  pageSize: o.pageSize,
                  pagePosition: o.pagePosition,
                  notificationRenderer: o.renderers.notification,
                  header: o.header,
                  newTab: o.newTab
                }
              ),
              i.webPushOptInMessage && localStorage.getItem("hideWebPushOptInMessage") !== "true" && /* @__PURE__ */ a("div", { children: [
                /* @__PURE__ */ t(
                  E,
                  {
                    style: { margin: "10px 0", borderColor: n.divider }
                  }
                ),
                /* @__PURE__ */ t(H, { hideAfterInteraction: !0 })
              ] })
            ]
          }
        )
      }
    ),
    /* @__PURE__ */ t(
      z,
      {
        open: I,
        onClose: () => l(!1)
      }
    )
  ] });
};
export {
  V as NotificationPopup
};
