// @notificationapi/react/dist/components/Notifications/UnreadBadge.js
import { jsx as c } from "react/jsx-runtime";
import { useContext as d } from "react";
import { NotificationAPIContext as a } from "../Provider/context.js";
import { B as u } from "../../assets/Badge.js";
const b = (t) => {
  var r, i, l, f;
  const o = d(a);
  return /* @__PURE__ */ c(
    u,
    {
      overlap: "circular",
      badgeContent: ((n) => t.count === "COUNT_UNOPENED_NOTIFICATIONS" || t.count === void 0 ? n.filter((e) => !e.opened && !e.seen).length : t.count === "COUNT_UNARCHIVED_NOTIFICATIONS" ? n.filter(
        (e) => !e.archived && !e.clicked && !e.replies && !e.actioned1 && !e.actioned2
      ).length : n.filter(t.count).length)((o == null ? void 0 : o.notifications) || []),
      max: t.max,
      variant: t.dot ? "dot" : "standard",
      showZero: t.showZero,
      sx: {
        "& .MuiBadge-badge": {
          backgroundColor: t.bubbleColor || "#d32f2f",
          color: t.fontColor || "#FFF"
        }
      },
      slotProps: {
        badge: {
          style: {
            top: (r = t.offset) == null ? void 0 : r.top,
            right: (i = t.offset) == null ? void 0 : i.right,
            bottom: (l = t.offset) == null ? void 0 : l.bottom,
            left: (f = t.offset) == null ? void 0 : f.left
          }
        }
      },
      children: t.children
    }
  );
};
export {
  b as UnreadBadge
};
