// @notificationapi/react/dist/components/Slack/SlackConnect.js
import { jsxs as q, jsx as y } from "react/jsx-runtime";
import * as c from "react";
import { useContext as at, useState as ge, useCallback as Ro, useEffect as ko } from "react";
import { NotificationAPIContext as it } from "../Provider/context.js";
import { B as Re } from "../../assets/Box.js";
import { P as e, g as J, e as _, u as Uo, k as bo, a as Ke } from "../../assets/createTheme.js";
import { g as X, s as E, m as K, u as ee, c as Z, a as Q, r as ne, f as lt } from "../../assets/DefaultPropsProvider.js";
import { c as Ue } from "../../assets/createSimplePaletteValueFilter.js";
import { k as Vo, c as Ho } from "../../assets/useThemeWithoutDefault.js";
import { d as We, e as ct, r as Me, c as dt, B as ut } from "../../assets/ButtonBase.js";
import { A as io, B as ke } from "../../assets/Button.js";
import { u as Ve, f as He, F as qo, S as lo } from "../../assets/Stack.js";
import { g as pt, T as Ge } from "../../assets/Typography.js";
import { i as To, a as co, L as ft, b as Mo } from "../../assets/List.js";
import { r as yo } from "../../assets/index2.js";
import { u as mt } from "../../assets/index.js";
import { d as bt, P as yt, a as gt } from "../../assets/Popover.js";
import { o as po, g as ht } from "../../assets/Modal.js";
import { u as Ee } from "../../assets/resolveComponentProps.js";
import { o as _o, H as vt, g as Ct } from "../../assets/Portal.js";
import { u as Eo } from "../../assets/useSlotProps.js";
import { d as No, a as Po, u as St } from "../../assets/dividerClasses.js";
import { c as wt } from "../../assets/createSvgIcon.js";
function Xe(o) {
  return parseInt(o, 10) || 0;
}
const Ot = {
  shadow: {
    // Visibility needed to hide the extra text area on iPads
    visibility: "hidden",
    // Remove from the content flow
    position: "absolute",
    // Ignore the scrollbar width
    overflow: "hidden",
    height: 0,
    top: 0,
    left: 0,
    // Create a new layer, increase the isolation of the computed values
    transform: "translateZ(0)"
  }
};
function xt(o) {
  return o == null || Object.keys(o).length === 0 || o.outerHeightStyle === 0 && !o.overflowing;
}
const Ko = /* @__PURE__ */ c.forwardRef(function(t, n) {
  const {
    onChange: r,
    maxRows: i,
    minRows: u = 1,
    style: a,
    value: d,
    ...m
  } = t, {
    current: v
  } = c.useRef(d != null), l = c.useRef(null), h = Ee(n, l), x = c.useRef(null), g = c.useRef(null), I = c.useCallback(() => {
    const p = l.current, s = po(p).getComputedStyle(p);
    if (s.width === "0px")
      return {
        outerHeightStyle: 0,
        overflowing: !1
      };
    const f = g.current;
    f.style.width = s.width, f.value = p.value || t.placeholder || "x", f.value.slice(-1) === `
` && (f.value += " ");
    const R = s.boxSizing, T = Xe(s.paddingBottom) + Xe(s.paddingTop), $ = Xe(s.borderBottomWidth) + Xe(s.borderTopWidth), k = f.scrollHeight;
    f.value = "x";
    const j = f.scrollHeight;
    let N = k;
    u && (N = Math.max(Number(u) * j, N)), i && (N = Math.min(Number(i) * j, N)), N = Math.max(N, j);
    const B = N + (R === "border-box" ? T + $ : 0), z = Math.abs(N - k) <= 1;
    return {
      outerHeightStyle: B,
      overflowing: z
    };
  }, [i, u, t.placeholder]), S = c.useCallback(() => {
    const p = I();
    if (xt(p))
      return;
    const O = p.outerHeightStyle, s = l.current;
    x.current !== O && (x.current = O, s.style.height = `${O}px`), s.style.overflow = p.overflowing ? "hidden" : "";
  }, [I]);
  We(() => {
    const p = () => {
      S();
    };
    let O;
    const s = () => {
      cancelAnimationFrame(O), O = requestAnimationFrame(() => {
        p();
      });
    }, f = bt(p), R = l.current, T = po(R);
    T.addEventListener("resize", f);
    let $;
    return typeof ResizeObserver < "u" && ($ = new ResizeObserver(process.env.NODE_ENV === "test" ? s : p), $.observe(R)), () => {
      f.clear(), cancelAnimationFrame(O), T.removeEventListener("resize", f), $ && $.disconnect();
    };
  }, [I, S]), We(() => {
    S();
  });
  const w = (p) => {
    v || S(), r && r(p);
  };
  return /* @__PURE__ */ q(c.Fragment, {
    children: [/* @__PURE__ */ y("textarea", {
      value: d,
      onChange: w,
      ref: h,
      rows: u,
      style: a,
      ...m
    }), /* @__PURE__ */ y("textarea", {
      "aria-hidden": !0,
      className: t.className,
      readOnly: !0,
      ref: g,
      tabIndex: -1,
      style: {
        ...Ot.shadow,
        ...a,
        paddingTop: 0,
        paddingBottom: 0
      }
    })]
  });
});
process.env.NODE_ENV !== "production" && (Ko.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │ To update them, edit the TypeScript types and run `pnpm proptypes`. │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * @ignore
   */
  className: e.string,
  /**
   * Maximum number of rows to display.
   */
  maxRows: e.oneOfType([e.number, e.string]),
  /**
   * Minimum number of rows to display.
   * @default 1
   */
  minRows: e.oneOfType([e.number, e.string]),
  /**
   * @ignore
   */
  onChange: e.func,
  /**
   * @ignore
   */
  placeholder: e.string,
  /**
   * @ignore
   */
  style: e.object,
  /**
   * @ignore
   */
  value: e.oneOfType([e.arrayOf(e.string), e.number, e.string])
});
function Fo(o) {
  return o != null && !(Array.isArray(o) && o.length === 0);
}
function Ye(o, t = !1) {
  return o && (Fo(o.value) && o.value !== "" || t && Fo(o.defaultValue) && o.defaultValue !== "");
}
function It(o) {
  return o.startAdornment;
}
function Rt(o) {
  return J("MuiInputBase", o);
}
const Te = X("MuiInputBase", ["root", "formControl", "focused", "disabled", "adornedStart", "adornedEnd", "error", "sizeSmall", "multiline", "colorSecondary", "fullWidth", "hiddenLabel", "readOnly", "input", "inputSizeSmall", "inputMultiline", "inputTypeSearch", "inputAdornedStart", "inputAdornedEnd", "inputHiddenLabel"]);
var $o;
const Ze = (o, t) => {
  const {
    ownerState: n
  } = o;
  return [t.root, n.formControl && t.formControl, n.startAdornment && t.adornedStart, n.endAdornment && t.adornedEnd, n.error && t.error, n.size === "small" && t.sizeSmall, n.multiline && t.multiline, n.color && t[`color${_(n.color)}`], n.fullWidth && t.fullWidth, n.hiddenLabel && t.hiddenLabel];
}, Je = (o, t) => {
  const {
    ownerState: n
  } = o;
  return [t.input, n.size === "small" && t.inputSizeSmall, n.multiline && t.inputMultiline, n.type === "search" && t.inputTypeSearch, n.startAdornment && t.inputAdornedStart, n.endAdornment && t.inputAdornedEnd, n.hiddenLabel && t.inputHiddenLabel];
}, kt = (o) => {
  const {
    classes: t,
    color: n,
    disabled: r,
    error: i,
    endAdornment: u,
    focused: a,
    formControl: d,
    fullWidth: m,
    hiddenLabel: v,
    multiline: l,
    readOnly: h,
    size: x,
    startAdornment: g,
    type: I
  } = o, S = {
    root: ["root", `color${_(n)}`, r && "disabled", i && "error", m && "fullWidth", a && "focused", d && "formControl", x && x !== "medium" && `size${_(x)}`, l && "multiline", g && "adornedStart", u && "adornedEnd", v && "hiddenLabel", h && "readOnly"],
    input: ["input", r && "disabled", I === "search" && "inputTypeSearch", l && "inputMultiline", x === "small" && "inputSizeSmall", v && "inputHiddenLabel", g && "inputAdornedStart", u && "inputAdornedEnd", h && "readOnly"]
  };
  return Q(S, Rt, t);
}, Qe = E("div", {
  name: "MuiInputBase",
  slot: "Root",
  overridesResolver: Ze
})(K(({
  theme: o
}) => ({
  ...o.typography.body1,
  color: (o.vars || o).palette.text.primary,
  lineHeight: "1.4375em",
  // 23px
  boxSizing: "border-box",
  // Prevent padding issue with fullWidth.
  position: "relative",
  cursor: "text",
  display: "inline-flex",
  alignItems: "center",
  [`&.${Te.disabled}`]: {
    color: (o.vars || o).palette.text.disabled,
    cursor: "default"
  },
  variants: [{
    props: ({
      ownerState: t
    }) => t.multiline,
    style: {
      padding: "4px 0 5px"
    }
  }, {
    props: ({
      ownerState: t,
      size: n
    }) => t.multiline && n === "small",
    style: {
      paddingTop: 1
    }
  }, {
    props: ({
      ownerState: t
    }) => t.fullWidth,
    style: {
      width: "100%"
    }
  }]
}))), eo = E("input", {
  name: "MuiInputBase",
  slot: "Input",
  overridesResolver: Je
})(K(({
  theme: o
}) => {
  const t = o.palette.mode === "light", n = {
    color: "currentColor",
    ...o.vars ? {
      opacity: o.vars.opacity.inputPlaceholder
    } : {
      opacity: t ? 0.42 : 0.5
    },
    transition: o.transitions.create("opacity", {
      duration: o.transitions.duration.shorter
    })
  }, r = {
    opacity: "0 !important"
  }, i = o.vars ? {
    opacity: o.vars.opacity.inputPlaceholder
  } : {
    opacity: t ? 0.42 : 0.5
  };
  return {
    font: "inherit",
    letterSpacing: "inherit",
    color: "currentColor",
    padding: "4px 0 5px",
    border: 0,
    boxSizing: "content-box",
    background: "none",
    height: "1.4375em",
    // Reset 23pxthe native input line-height
    margin: 0,
    // Reset for Safari
    WebkitTapHighlightColor: "transparent",
    display: "block",
    // Make the flex item shrink with Firefox
    minWidth: 0,
    width: "100%",
    "&::-webkit-input-placeholder": n,
    "&::-moz-placeholder": n,
    // Firefox 19+
    "&::-ms-input-placeholder": n,
    // Edge
    "&:focus": {
      outline: 0
    },
    // Reset Firefox invalid required input style
    "&:invalid": {
      boxShadow: "none"
    },
    "&::-webkit-search-decoration": {
      // Remove the padding when type=search.
      WebkitAppearance: "none"
    },
    // Show and hide the placeholder logic
    [`label[data-shrink=false] + .${Te.formControl} &`]: {
      "&::-webkit-input-placeholder": r,
      "&::-moz-placeholder": r,
      // Firefox 19+
      "&::-ms-input-placeholder": r,
      // Edge
      "&:focus::-webkit-input-placeholder": i,
      "&:focus::-moz-placeholder": i,
      // Firefox 19+
      "&:focus::-ms-input-placeholder": i
      // Edge
    },
    [`&.${Te.disabled}`]: {
      opacity: 1,
      // Reset iOS opacity
      WebkitTextFillColor: (o.vars || o).palette.text.disabled
      // Fix opacity Safari bug
    },
    variants: [{
      props: ({
        ownerState: u
      }) => !u.disableInjectingGlobalStyles,
      style: {
        animationName: "mui-auto-fill-cancel",
        animationDuration: "10ms",
        "&:-webkit-autofill": {
          animationDuration: "5000s",
          animationName: "mui-auto-fill"
        }
      }
    }, {
      props: {
        size: "small"
      },
      style: {
        paddingTop: 1
      }
    }, {
      props: ({
        ownerState: u
      }) => u.multiline,
      style: {
        height: "auto",
        resize: "none",
        padding: 0,
        paddingTop: 0
      }
    }, {
      props: {
        type: "search"
      },
      style: {
        MozAppearance: "textfield"
        // Improve type search style.
      }
    }]
  };
})), Lo = pt({
  "@keyframes mui-auto-fill": {
    from: {
      display: "block"
    }
  },
  "@keyframes mui-auto-fill-cancel": {
    from: {
      display: "block"
    }
  }
}), oo = /* @__PURE__ */ c.forwardRef(function(t, n) {
  const r = ee({
    props: t,
    name: "MuiInputBase"
  }), {
    "aria-describedby": i,
    autoComplete: u,
    autoFocus: a,
    className: d,
    color: m,
    components: v = {},
    componentsProps: l = {},
    defaultValue: h,
    disabled: x,
    disableInjectingGlobalStyles: g,
    endAdornment: I,
    error: S,
    fullWidth: w = !1,
    id: p,
    inputComponent: O = "input",
    inputProps: s = {},
    inputRef: f,
    margin: R,
    maxRows: T,
    minRows: $,
    multiline: k = !1,
    name: j,
    onBlur: N,
    onChange: B,
    onClick: z,
    onFocus: W,
    onKeyDown: U,
    onKeyUp: V,
    placeholder: b,
    readOnly: F,
    renderSuffix: L,
    rows: P,
    size: ve,
    slotProps: ae = {},
    slots: qe = {},
    startAdornment: ie,
    type: oe = "text",
    value: re,
    ...to
  } = r, le = s.value != null ? s.value : re, {
    current: Ne
  } = c.useRef(le != null), se = c.useRef(), no = c.useCallback((D) => {
    process.env.NODE_ENV !== "production" && D && D.nodeName !== "INPUT" && !D.focus && console.error(["MUI: You have provided a `inputComponent` to the input component", "that does not correctly handle the `ref` prop.", "Make sure the `ref` prop is called with a HTMLInputElement."].join(`
`));
  }, []), ro = Ee(se, f, s.ref, no), [ce, de] = c.useState(!1), A = Ve();
  process.env.NODE_ENV !== "production" && c.useEffect(() => {
    if (A)
      return A.registerEffect();
  }, [A]);
  const Y = He({
    props: r,
    muiFormControl: A,
    states: ["color", "disabled", "error", "hiddenLabel", "size", "required", "filled"]
  });
  Y.focused = A ? A.focused : ce, c.useEffect(() => {
    !A && x && ce && (de(!1), N && N());
  }, [A, x, ce, N]);
  const ue = A && A.onFilled, Pe = A && A.onEmpty, me = c.useCallback((D) => {
    Ye(D) ? ue && ue() : Pe && Pe();
  }, [ue, Pe]);
  We(() => {
    Ne && me({
      value: le
    });
  }, [le, me, Ne]);
  const so = (D) => {
    W && W(D), s.onFocus && s.onFocus(D), A && A.onFocus ? A.onFocus(D) : de(!0);
  }, Ce = (D) => {
    N && N(D), s.onBlur && s.onBlur(D), A && A.onBlur ? A.onBlur(D) : de(!1);
  }, ao = (D, ...Oe) => {
    if (!Ne) {
      const xe = D.target || se.current;
      if (xe == null)
        throw new Error(process.env.NODE_ENV !== "production" ? "MUI: Expected valid input target. Did you use a custom `inputComponent` and forget to forward refs? See https://mui.com/r/input-component-ref-interface for more info." : Uo(1));
      me({
        value: xe.value
      });
    }
    s.onChange && s.onChange(D, ...Oe), B && B(D, ...Oe);
  };
  c.useEffect(() => {
    me(se.current);
  }, []);
  const pe = (D) => {
    se.current && D.currentTarget === D.target && se.current.focus(), z && z(D);
  };
  let Se = O, G = s;
  k && Se === "input" && (P ? (process.env.NODE_ENV !== "production" && ($ || T) && console.warn("MUI: You can not use the `minRows` or `maxRows` props when the input `rows` prop is set."), G = {
    type: void 0,
    minRows: P,
    maxRows: P,
    ...G
  }) : G = {
    type: void 0,
    maxRows: T,
    minRows: $,
    ...G
  }, Se = Ko);
  const we = (D) => {
    me(D.animationName === "mui-auto-fill-cancel" ? se.current : {
      value: "x"
    });
  };
  c.useEffect(() => {
    A && A.setAdornedStart(!!ie);
  }, [A, ie]);
  const be = {
    ...r,
    color: Y.color || "primary",
    disabled: Y.disabled,
    endAdornment: I,
    error: Y.error,
    focused: Y.focused,
    formControl: A,
    fullWidth: w,
    hiddenLabel: Y.hiddenLabel,
    multiline: k,
    size: Y.size,
    startAdornment: ie,
    type: oe
  }, _e = kt(be), Fe = qe.root || v.Root || Qe, ye = ae.root || l.root || {}, $e = qe.input || v.Input || eo;
  return G = {
    ...G,
    ...ae.input ?? l.input
  }, /* @__PURE__ */ q(c.Fragment, {
    children: [!g && typeof Lo == "function" && // For Emotion/Styled-components, InputGlobalStyles will be a function
    // For Pigment CSS, this has no effect because the InputGlobalStyles will be null.
    ($o || ($o = /* @__PURE__ */ y(Lo, {}))), /* @__PURE__ */ q(Fe, {
      ...ye,
      ref: n,
      onClick: pe,
      ...to,
      ...!To(Fe) && {
        ownerState: {
          ...be,
          ...ye.ownerState
        }
      },
      className: Z(_e.root, ye.className, d, F && "MuiInputBase-readOnly"),
      children: [ie, /* @__PURE__ */ y(qo.Provider, {
        value: null,
        children: /* @__PURE__ */ y($e, {
          "aria-invalid": Y.error,
          "aria-describedby": i,
          autoComplete: u,
          autoFocus: a,
          defaultValue: h,
          disabled: Y.disabled,
          id: p,
          onAnimationStart: we,
          name: j,
          placeholder: b,
          readOnly: F,
          required: Y.required,
          rows: P,
          value: le,
          onKeyDown: U,
          onKeyUp: V,
          type: oe,
          ...G,
          ...!To($e) && {
            as: Se,
            ownerState: {
              ...be,
              ...G.ownerState
            }
          },
          ref: ro,
          className: Z(_e.input, G.className, F && "MuiInputBase-readOnly"),
          onBlur: Ce,
          onChange: ao,
          onFocus: so
        })
      }), I, L ? L({
        ...Y,
        startAdornment: ie
      }) : null]
    })]
  });
});
process.env.NODE_ENV !== "production" && (oo.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * @ignore
   */
  "aria-describedby": e.string,
  /**
   * This prop helps users to fill forms faster, especially on mobile devices.
   * The name can be confusing, as it's more like an autofill.
   * You can learn more about it [following the specification](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill).
   */
  autoComplete: e.string,
  /**
   * If `true`, the `input` element is focused during the first mount.
   */
  autoFocus: e.bool,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * The color of the component.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   * The prop defaults to the value (`'primary'`) inherited from the parent FormControl component.
   */
  color: e.oneOfType([e.oneOf(["primary", "secondary", "error", "info", "success", "warning"]), e.string]),
  /**
   * The components used for each slot inside.
   *
   * @deprecated use the `slots` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  components: e.shape({
    Input: e.elementType,
    Root: e.elementType
  }),
  /**
   * The extra props for the slot components.
   * You can override the existing props or add new ones.
   *
   * @deprecated use the `slotProps` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  componentsProps: e.shape({
    input: e.object,
    root: e.object
  }),
  /**
   * The default value. Use when the component is not controlled.
   */
  defaultValue: e.any,
  /**
   * If `true`, the component is disabled.
   * The prop defaults to the value (`false`) inherited from the parent FormControl component.
   */
  disabled: e.bool,
  /**
   * If `true`, GlobalStyles for the auto-fill keyframes will not be injected/removed on mount/unmount. Make sure to inject them at the top of your application.
   * This option is intended to help with boosting the initial rendering performance if you are loading a big amount of Input components at once.
   * @default false
   */
  disableInjectingGlobalStyles: e.bool,
  /**
   * End `InputAdornment` for this component.
   */
  endAdornment: e.node,
  /**
   * If `true`, the `input` will indicate an error.
   * The prop defaults to the value (`false`) inherited from the parent FormControl component.
   */
  error: e.bool,
  /**
   * If `true`, the `input` will take up the full width of its container.
   * @default false
   */
  fullWidth: e.bool,
  /**
   * The id of the `input` element.
   */
  id: e.string,
  /**
   * The component used for the `input` element.
   * Either a string to use a HTML element or a component.
   * @default 'input'
   */
  inputComponent: ct,
  /**
   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes) applied to the `input` element.
   * @default {}
   */
  inputProps: e.object,
  /**
   * Pass a ref to the `input` element.
   */
  inputRef: Me,
  /**
   * If `dense`, will adjust vertical spacing. This is normally obtained via context from
   * FormControl.
   * The prop defaults to the value (`'none'`) inherited from the parent FormControl component.
   */
  margin: e.oneOf(["dense", "none"]),
  /**
   * Maximum number of rows to display when multiline option is set to true.
   */
  maxRows: e.oneOfType([e.number, e.string]),
  /**
   * Minimum number of rows to display when multiline option is set to true.
   */
  minRows: e.oneOfType([e.number, e.string]),
  /**
   * If `true`, a [TextareaAutosize](https://mui.com/material-ui/react-textarea-autosize/) element is rendered.
   * @default false
   */
  multiline: e.bool,
  /**
   * Name attribute of the `input` element.
   */
  name: e.string,
  /**
   * Callback fired when the `input` is blurred.
   *
   * Notice that the first argument (event) might be undefined.
   */
  onBlur: e.func,
  /**
   * Callback fired when the value is changed.
   *
   * @param {React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>} event The event source of the callback.
   * You can pull out the new value by accessing `event.target.value` (string).
   */
  onChange: e.func,
  /**
   * @ignore
   */
  onClick: e.func,
  /**
   * @ignore
   */
  onFocus: e.func,
  /**
   * Callback fired when the `input` doesn't satisfy its constraints.
   */
  onInvalid: e.func,
  /**
   * @ignore
   */
  onKeyDown: e.func,
  /**
   * @ignore
   */
  onKeyUp: e.func,
  /**
   * The short hint displayed in the `input` before the user enters a value.
   */
  placeholder: e.string,
  /**
   * It prevents the user from changing the value of the field
   * (not from interacting with the field).
   */
  readOnly: e.bool,
  /**
   * @ignore
   */
  renderSuffix: e.func,
  /**
   * If `true`, the `input` element is required.
   * The prop defaults to the value (`false`) inherited from the parent FormControl component.
   */
  required: e.bool,
  /**
   * Number of rows to display when multiline option is set to true.
   */
  rows: e.oneOfType([e.number, e.string]),
  /**
   * The size of the component.
   */
  size: e.oneOfType([e.oneOf(["medium", "small"]), e.string]),
  /**
   * The extra props for the slot components.
   * You can override the existing props or add new ones.
   *
   * This prop is an alias for the `componentsProps` prop, which will be deprecated in the future.
   *
   * @default {}
   */
  slotProps: e.shape({
    input: e.object,
    root: e.object
  }),
  /**
   * The components used for each slot inside.
   *
   * This prop is an alias for the `components` prop, which will be deprecated in the future.
   *
   * @default {}
   */
  slots: e.shape({
    input: e.elementType,
    root: e.elementType
  }),
  /**
   * Start `InputAdornment` for this component.
   */
  startAdornment: e.node,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * Type of the `input` element. It should be [a valid HTML5 input type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Form_%3Cinput%3E_types).
   * @default 'text'
   */
  type: e.string,
  /**
   * The value of the `input` element, required for a controlled component.
   */
  value: e.any
});
function Tt(o) {
  return J("MuiInput", o);
}
const Le = {
  ...Te,
  ...X("MuiInput", ["root", "underline", "input"])
};
function Mt(o) {
  return J("MuiOutlinedInput", o);
}
const te = {
  ...Te,
  ...X("MuiOutlinedInput", ["root", "notchedOutline", "input"])
};
function Et(o) {
  return J("MuiFilledInput", o);
}
const he = {
  ...Te,
  ...X("MuiFilledInput", ["root", "underline", "input", "adornedStart", "adornedEnd", "sizeSmall", "multiline", "hiddenLabel"])
}, Nt = wt(/* @__PURE__ */ y("path", {
  d: "M7 10l5 5 5-5z"
}), "ArrowDropDown");
function Pt(o) {
  return J("MuiCircularProgress", o);
}
X("MuiCircularProgress", ["root", "determinate", "indeterminate", "colorPrimary", "colorSecondary", "svg", "circle", "circleDeterminate", "circleIndeterminate", "circleDisableShrink"]);
const fe = 44, fo = Vo`
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
`, mo = Vo`
  0% {
    stroke-dasharray: 1px, 200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px, 200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 100px, 200px;
    stroke-dashoffset: -125px;
  }
`, Ft = typeof fo != "string" ? Ho`
        animation: ${fo} 1.4s linear infinite;
      ` : null, $t = typeof mo != "string" ? Ho`
        animation: ${mo} 1.4s ease-in-out infinite;
      ` : null, Lt = (o) => {
  const {
    classes: t,
    variant: n,
    color: r,
    disableShrink: i
  } = o, u = {
    root: ["root", n, `color${_(r)}`],
    svg: ["svg"],
    circle: ["circle", `circle${_(n)}`, i && "circleDisableShrink"]
  };
  return Q(u, Pt, t);
}, Dt = E("span", {
  name: "MuiCircularProgress",
  slot: "Root",
  overridesResolver: (o, t) => {
    const {
      ownerState: n
    } = o;
    return [t.root, t[n.variant], t[`color${_(n.color)}`]];
  }
})(K(({
  theme: o
}) => ({
  display: "inline-block",
  variants: [{
    props: {
      variant: "determinate"
    },
    style: {
      transition: o.transitions.create("transform")
    }
  }, {
    props: {
      variant: "indeterminate"
    },
    style: Ft || {
      animation: `${fo} 1.4s linear infinite`
    }
  }, ...Object.entries(o.palette).filter(Ue()).map(([t]) => ({
    props: {
      color: t
    },
    style: {
      color: (o.vars || o).palette[t].main
    }
  }))]
}))), At = E("svg", {
  name: "MuiCircularProgress",
  slot: "Svg",
  overridesResolver: (o, t) => t.svg
})({
  display: "block"
  // Keeps the progress centered
}), jt = E("circle", {
  name: "MuiCircularProgress",
  slot: "Circle",
  overridesResolver: (o, t) => {
    const {
      ownerState: n
    } = o;
    return [t.circle, t[`circle${_(n.variant)}`], n.disableShrink && t.circleDisableShrink];
  }
})(K(({
  theme: o
}) => ({
  stroke: "currentColor",
  variants: [{
    props: {
      variant: "determinate"
    },
    style: {
      transition: o.transitions.create("stroke-dashoffset")
    }
  }, {
    props: {
      variant: "indeterminate"
    },
    style: {
      // Some default value that looks fine waiting for the animation to kicks in.
      strokeDasharray: "80px, 200px",
      strokeDashoffset: 0
      // Add the unit to fix a Edge 16 and below bug.
    }
  }, {
    props: ({
      ownerState: t
    }) => t.variant === "indeterminate" && !t.disableShrink,
    style: $t || {
      // At runtime for Pigment CSS, `bufferAnimation` will be null and the generated keyframe will be used.
      animation: `${mo} 1.4s ease-in-out infinite`
    }
  }]
}))), Be = /* @__PURE__ */ c.forwardRef(function(t, n) {
  const r = ee({
    props: t,
    name: "MuiCircularProgress"
  }), {
    className: i,
    color: u = "primary",
    disableShrink: a = !1,
    size: d = 40,
    style: m,
    thickness: v = 3.6,
    value: l = 0,
    variant: h = "indeterminate",
    ...x
  } = r, g = {
    ...r,
    color: u,
    disableShrink: a,
    size: d,
    thickness: v,
    value: l,
    variant: h
  }, I = Lt(g), S = {}, w = {}, p = {};
  if (h === "determinate") {
    const O = 2 * Math.PI * ((fe - v) / 2);
    S.strokeDasharray = O.toFixed(3), p["aria-valuenow"] = Math.round(l), S.strokeDashoffset = `${((100 - l) / 100 * O).toFixed(3)}px`, w.transform = "rotate(-90deg)";
  }
  return /* @__PURE__ */ y(Dt, {
    className: Z(I.root, i),
    style: {
      width: d,
      height: d,
      ...w,
      ...m
    },
    ownerState: g,
    ref: n,
    role: "progressbar",
    ...p,
    ...x,
    children: /* @__PURE__ */ y(At, {
      className: I.svg,
      ownerState: g,
      viewBox: `${fe / 2} ${fe / 2} ${fe} ${fe}`,
      children: /* @__PURE__ */ y(jt, {
        className: I.circle,
        style: S,
        ownerState: g,
        cx: fe,
        cy: fe,
        r: (fe - v) / 2,
        fill: "none",
        strokeWidth: v
      })
    })
  });
});
process.env.NODE_ENV !== "production" && (Be.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * The color of the component.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   * @default 'primary'
   */
  color: e.oneOfType([e.oneOf(["inherit", "primary", "secondary", "error", "info", "success", "warning"]), e.string]),
  /**
   * If `true`, the shrink animation is disabled.
   * This only works if variant is `indeterminate`.
   * @default false
   */
  disableShrink: dt(e.bool, (o) => o.disableShrink && o.variant && o.variant !== "indeterminate" ? new Error("MUI: You have provided the `disableShrink` prop with a variant other than `indeterminate`. This will have no effect.") : null),
  /**
   * The size of the component.
   * If using a number, the pixel unit is assumed.
   * If using a string, you need to provide the CSS unit, for example '3rem'.
   * @default 40
   */
  size: e.oneOfType([e.number, e.string]),
  /**
   * @ignore
   */
  style: e.object,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * The thickness of the circle.
   * @default 3.6
   */
  thickness: e.number,
  /**
   * The value of the progress indicator for the determinate variant.
   * Value between 0 and 100.
   * @default 0
   */
  value: e.number,
  /**
   * The variant to use.
   * Use indeterminate when there is no progress value.
   * @default 'indeterminate'
   */
  variant: e.oneOf(["determinate", "indeterminate"])
});
const Bt = (o) => {
  const {
    classes: t,
    disableUnderline: n,
    startAdornment: r,
    endAdornment: i,
    size: u,
    hiddenLabel: a,
    multiline: d
  } = o, m = {
    root: ["root", !n && "underline", r && "adornedStart", i && "adornedEnd", u === "small" && `size${_(u)}`, a && "hiddenLabel", d && "multiline"],
    input: ["input"]
  }, v = Q(m, Et, t);
  return {
    ...t,
    // forward classes to the InputBase
    ...v
  };
}, zt = E(Qe, {
  shouldForwardProp: (o) => ne(o) || o === "classes",
  name: "MuiFilledInput",
  slot: "Root",
  overridesResolver: (o, t) => {
    const {
      ownerState: n
    } = o;
    return [...Ze(o, t), !n.disableUnderline && t.underline];
  }
})(K(({
  theme: o
}) => {
  const t = o.palette.mode === "light", n = t ? "rgba(0, 0, 0, 0.42)" : "rgba(255, 255, 255, 0.7)", r = t ? "rgba(0, 0, 0, 0.06)" : "rgba(255, 255, 255, 0.09)", i = t ? "rgba(0, 0, 0, 0.09)" : "rgba(255, 255, 255, 0.13)", u = t ? "rgba(0, 0, 0, 0.12)" : "rgba(255, 255, 255, 0.12)";
  return {
    position: "relative",
    backgroundColor: o.vars ? o.vars.palette.FilledInput.bg : r,
    borderTopLeftRadius: (o.vars || o).shape.borderRadius,
    borderTopRightRadius: (o.vars || o).shape.borderRadius,
    transition: o.transitions.create("background-color", {
      duration: o.transitions.duration.shorter,
      easing: o.transitions.easing.easeOut
    }),
    "&:hover": {
      backgroundColor: o.vars ? o.vars.palette.FilledInput.hoverBg : i,
      // Reset on touch devices, it doesn't add specificity
      "@media (hover: none)": {
        backgroundColor: o.vars ? o.vars.palette.FilledInput.bg : r
      }
    },
    [`&.${he.focused}`]: {
      backgroundColor: o.vars ? o.vars.palette.FilledInput.bg : r
    },
    [`&.${he.disabled}`]: {
      backgroundColor: o.vars ? o.vars.palette.FilledInput.disabledBg : u
    },
    variants: [{
      props: ({
        ownerState: a
      }) => !a.disableUnderline,
      style: {
        "&::after": {
          left: 0,
          bottom: 0,
          content: '""',
          position: "absolute",
          right: 0,
          transform: "scaleX(0)",
          transition: o.transitions.create("transform", {
            duration: o.transitions.duration.shorter,
            easing: o.transitions.easing.easeOut
          }),
          pointerEvents: "none"
          // Transparent to the hover style.
        },
        [`&.${he.focused}:after`]: {
          // translateX(0) is a workaround for Safari transform scale bug
          // See https://github.com/mui/material-ui/issues/31766
          transform: "scaleX(1) translateX(0)"
        },
        [`&.${he.error}`]: {
          "&::before, &::after": {
            borderBottomColor: (o.vars || o).palette.error.main
          }
        },
        "&::before": {
          borderBottom: `1px solid ${o.vars ? `rgba(${o.vars.palette.common.onBackgroundChannel} / ${o.vars.opacity.inputUnderline})` : n}`,
          left: 0,
          bottom: 0,
          content: '"\\00a0"',
          position: "absolute",
          right: 0,
          transition: o.transitions.create("border-bottom-color", {
            duration: o.transitions.duration.shorter
          }),
          pointerEvents: "none"
          // Transparent to the hover style.
        },
        [`&:hover:not(.${he.disabled}, .${he.error}):before`]: {
          borderBottom: `1px solid ${(o.vars || o).palette.text.primary}`
        },
        [`&.${he.disabled}:before`]: {
          borderBottomStyle: "dotted"
        }
      }
    }, ...Object.entries(o.palette).filter(Ue()).map(([a]) => {
      var d;
      return {
        props: {
          disableUnderline: !1,
          color: a
        },
        style: {
          "&::after": {
            borderBottom: `2px solid ${(d = (o.vars || o).palette[a]) == null ? void 0 : d.main}`
          }
        }
      };
    }), {
      props: ({
        ownerState: a
      }) => a.startAdornment,
      style: {
        paddingLeft: 12
      }
    }, {
      props: ({
        ownerState: a
      }) => a.endAdornment,
      style: {
        paddingRight: 12
      }
    }, {
      props: ({
        ownerState: a
      }) => a.multiline,
      style: {
        padding: "25px 12px 8px"
      }
    }, {
      props: ({
        ownerState: a,
        size: d
      }) => a.multiline && d === "small",
      style: {
        paddingTop: 21,
        paddingBottom: 4
      }
    }, {
      props: ({
        ownerState: a
      }) => a.multiline && a.hiddenLabel,
      style: {
        paddingTop: 16,
        paddingBottom: 17
      }
    }, {
      props: ({
        ownerState: a
      }) => a.multiline && a.hiddenLabel && a.size === "small",
      style: {
        paddingTop: 8,
        paddingBottom: 9
      }
    }]
  };
})), Wt = E(eo, {
  name: "MuiFilledInput",
  slot: "Input",
  overridesResolver: Je
})(K(({
  theme: o
}) => ({
  paddingTop: 25,
  paddingRight: 12,
  paddingBottom: 8,
  paddingLeft: 12,
  ...!o.vars && {
    "&:-webkit-autofill": {
      WebkitBoxShadow: o.palette.mode === "light" ? null : "0 0 0 100px #266798 inset",
      WebkitTextFillColor: o.palette.mode === "light" ? null : "#fff",
      caretColor: o.palette.mode === "light" ? null : "#fff",
      borderTopLeftRadius: "inherit",
      borderTopRightRadius: "inherit"
    }
  },
  ...o.vars && {
    "&:-webkit-autofill": {
      borderTopLeftRadius: "inherit",
      borderTopRightRadius: "inherit"
    },
    [o.getColorSchemeSelector("dark")]: {
      "&:-webkit-autofill": {
        WebkitBoxShadow: "0 0 0 100px #266798 inset",
        WebkitTextFillColor: "#fff",
        caretColor: "#fff"
      }
    }
  },
  variants: [{
    props: {
      size: "small"
    },
    style: {
      paddingTop: 21,
      paddingBottom: 4
    }
  }, {
    props: ({
      ownerState: t
    }) => t.hiddenLabel,
    style: {
      paddingTop: 16,
      paddingBottom: 17
    }
  }, {
    props: ({
      ownerState: t
    }) => t.startAdornment,
    style: {
      paddingLeft: 0
    }
  }, {
    props: ({
      ownerState: t
    }) => t.endAdornment,
    style: {
      paddingRight: 0
    }
  }, {
    props: ({
      ownerState: t
    }) => t.hiddenLabel && t.size === "small",
    style: {
      paddingTop: 8,
      paddingBottom: 9
    }
  }, {
    props: ({
      ownerState: t
    }) => t.multiline,
    style: {
      paddingTop: 0,
      paddingBottom: 0,
      paddingLeft: 0,
      paddingRight: 0
    }
  }]
}))), go = /* @__PURE__ */ c.forwardRef(function(t, n) {
  const r = ee({
    props: t,
    name: "MuiFilledInput"
  }), {
    disableUnderline: i = !1,
    components: u = {},
    componentsProps: a,
    fullWidth: d = !1,
    hiddenLabel: m,
    // declare here to prevent spreading to DOM
    inputComponent: v = "input",
    multiline: l = !1,
    slotProps: h,
    slots: x = {},
    type: g = "text",
    ...I
  } = r, S = {
    ...r,
    disableUnderline: i,
    fullWidth: d,
    inputComponent: v,
    multiline: l,
    type: g
  }, w = Bt(r), p = {
    root: {
      ownerState: S
    },
    input: {
      ownerState: S
    }
  }, O = h ?? a ? bo(p, h ?? a) : p, s = x.root ?? u.Root ?? zt, f = x.input ?? u.Input ?? Wt;
  return /* @__PURE__ */ y(oo, {
    slots: {
      root: s,
      input: f
    },
    slotProps: O,
    fullWidth: d,
    inputComponent: v,
    multiline: l,
    ref: n,
    type: g,
    ...I,
    classes: w
  });
});
process.env.NODE_ENV !== "production" && (go.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * This prop helps users to fill forms faster, especially on mobile devices.
   * The name can be confusing, as it's more like an autofill.
   * You can learn more about it [following the specification](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill).
   */
  autoComplete: e.string,
  /**
   * If `true`, the `input` element is focused during the first mount.
   */
  autoFocus: e.bool,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * The color of the component.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   * The prop defaults to the value (`'primary'`) inherited from the parent FormControl component.
   */
  color: e.oneOfType([e.oneOf(["primary", "secondary"]), e.string]),
  /**
   * The components used for each slot inside.
   *
   * @deprecated use the `slots` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  components: e.shape({
    Input: e.elementType,
    Root: e.elementType
  }),
  /**
   * The extra props for the slot components.
   * You can override the existing props or add new ones.
   *
   * @deprecated use the `slotProps` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  componentsProps: e.shape({
    input: e.object,
    root: e.object
  }),
  /**
   * The default value. Use when the component is not controlled.
   */
  defaultValue: e.any,
  /**
   * If `true`, the component is disabled.
   * The prop defaults to the value (`false`) inherited from the parent FormControl component.
   */
  disabled: e.bool,
  /**
   * If `true`, the input will not have an underline.
   * @default false
   */
  disableUnderline: e.bool,
  /**
   * End `InputAdornment` for this component.
   */
  endAdornment: e.node,
  /**
   * If `true`, the `input` will indicate an error.
   * The prop defaults to the value (`false`) inherited from the parent FormControl component.
   */
  error: e.bool,
  /**
   * If `true`, the `input` will take up the full width of its container.
   * @default false
   */
  fullWidth: e.bool,
  /**
   * If `true`, the label is hidden.
   * This is used to increase density for a `FilledInput`.
   * Be sure to add `aria-label` to the `input` element.
   * @default false
   */
  hiddenLabel: e.bool,
  /**
   * The id of the `input` element.
   */
  id: e.string,
  /**
   * The component used for the `input` element.
   * Either a string to use a HTML element or a component.
   * @default 'input'
   */
  inputComponent: e.elementType,
  /**
   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes) applied to the `input` element.
   * @default {}
   */
  inputProps: e.object,
  /**
   * Pass a ref to the `input` element.
   */
  inputRef: Me,
  /**
   * If `dense`, will adjust vertical spacing. This is normally obtained via context from
   * FormControl.
   * The prop defaults to the value (`'none'`) inherited from the parent FormControl component.
   */
  margin: e.oneOf(["dense", "none"]),
  /**
   * Maximum number of rows to display when multiline option is set to true.
   */
  maxRows: e.oneOfType([e.number, e.string]),
  /**
   * Minimum number of rows to display when multiline option is set to true.
   */
  minRows: e.oneOfType([e.number, e.string]),
  /**
   * If `true`, a [TextareaAutosize](https://mui.com/material-ui/react-textarea-autosize/) element is rendered.
   * @default false
   */
  multiline: e.bool,
  /**
   * Name attribute of the `input` element.
   */
  name: e.string,
  /**
   * Callback fired when the value is changed.
   *
   * @param {React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>} event The event source of the callback.
   * You can pull out the new value by accessing `event.target.value` (string).
   */
  onChange: e.func,
  /**
   * The short hint displayed in the `input` before the user enters a value.
   */
  placeholder: e.string,
  /**
   * It prevents the user from changing the value of the field
   * (not from interacting with the field).
   */
  readOnly: e.bool,
  /**
   * If `true`, the `input` element is required.
   * The prop defaults to the value (`false`) inherited from the parent FormControl component.
   */
  required: e.bool,
  /**
   * Number of rows to display when multiline option is set to true.
   */
  rows: e.oneOfType([e.number, e.string]),
  /**
   * The extra props for the slot components.
   * You can override the existing props or add new ones.
   *
   * This prop is an alias for the `componentsProps` prop, which will be deprecated in the future.
   *
   * @default {}
   */
  slotProps: e.shape({
    input: e.object,
    root: e.object
  }),
  /**
   * The components used for each slot inside.
   *
   * This prop is an alias for the `components` prop, which will be deprecated in the future.
   *
   * @default {}
   */
  slots: e.shape({
    input: e.elementType,
    root: e.elementType
  }),
  /**
   * Start `InputAdornment` for this component.
   */
  startAdornment: e.node,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * Type of the `input` element. It should be [a valid HTML5 input type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Form_%3Cinput%3E_types).
   * @default 'text'
   */
  type: e.string,
  /**
   * The value of the `input` element, required for a controlled component.
   */
  value: e.any
});
go.muiName = "Input";
function Ut(o) {
  return J("MuiFormControl", o);
}
X("MuiFormControl", ["root", "marginNone", "marginNormal", "marginDense", "fullWidth", "disabled"]);
const Vt = (o) => {
  const {
    classes: t,
    margin: n,
    fullWidth: r
  } = o, i = {
    root: ["root", n !== "none" && `margin${_(n)}`, r && "fullWidth"]
  };
  return Q(i, Ut, t);
}, Ht = E("div", {
  name: "MuiFormControl",
  slot: "Root",
  overridesResolver: (o, t) => {
    const {
      ownerState: n
    } = o;
    return [t.root, t[`margin${_(n.margin)}`], n.fullWidth && t.fullWidth];
  }
})({
  display: "inline-flex",
  flexDirection: "column",
  position: "relative",
  // Reset fieldset default style.
  minWidth: 0,
  padding: 0,
  margin: 0,
  border: 0,
  verticalAlign: "top",
  // Fix alignment issue on Safari.
  variants: [{
    props: {
      margin: "normal"
    },
    style: {
      marginTop: 16,
      marginBottom: 8
    }
  }, {
    props: {
      margin: "dense"
    },
    style: {
      marginTop: 8,
      marginBottom: 4
    }
  }, {
    props: {
      fullWidth: !0
    },
    style: {
      width: "100%"
    }
  }]
}), Go = /* @__PURE__ */ c.forwardRef(function(t, n) {
  const r = ee({
    props: t,
    name: "MuiFormControl"
  }), {
    children: i,
    className: u,
    color: a = "primary",
    component: d = "div",
    disabled: m = !1,
    error: v = !1,
    focused: l,
    fullWidth: h = !1,
    hiddenLabel: x = !1,
    margin: g = "none",
    required: I = !1,
    size: S = "medium",
    variant: w = "outlined",
    ...p
  } = r, O = {
    ...r,
    color: a,
    component: d,
    disabled: m,
    error: v,
    fullWidth: h,
    hiddenLabel: x,
    margin: g,
    required: I,
    size: S,
    variant: w
  }, s = Vt(O), [f, R] = c.useState(() => {
    let U = !1;
    return i && c.Children.forEach(i, (V) => {
      if (!co(V, ["Input", "Select"]))
        return;
      const b = co(V, ["Select"]) ? V.props.input : V;
      b && It(b.props) && (U = !0);
    }), U;
  }), [T, $] = c.useState(() => {
    let U = !1;
    return i && c.Children.forEach(i, (V) => {
      co(V, ["Input", "Select"]) && (Ye(V.props, !0) || Ye(V.props.inputProps, !0)) && (U = !0);
    }), U;
  }), [k, j] = c.useState(!1);
  m && k && j(!1);
  const N = l !== void 0 && !m ? l : k;
  let B;
  const z = c.useRef(!1);
  process.env.NODE_ENV !== "production" && (B = () => (z.current && console.error(["MUI: There are multiple `InputBase` components inside a FormControl.", "This creates visual inconsistencies, only use one `InputBase`."].join(`
`)), z.current = !0, () => {
    z.current = !1;
  }));
  const W = c.useMemo(() => ({
    adornedStart: f,
    setAdornedStart: R,
    color: a,
    disabled: m,
    error: v,
    filled: T,
    focused: N,
    fullWidth: h,
    hiddenLabel: x,
    size: S,
    onBlur: () => {
      j(!1);
    },
    onEmpty: () => {
      $(!1);
    },
    onFilled: () => {
      $(!0);
    },
    onFocus: () => {
      j(!0);
    },
    registerEffect: B,
    required: I,
    variant: w
  }), [f, a, m, v, T, N, h, x, B, I, S, w]);
  return /* @__PURE__ */ y(qo.Provider, {
    value: W,
    children: /* @__PURE__ */ y(Ht, {
      as: d,
      ownerState: O,
      className: Z(s.root, u),
      ref: n,
      ...p,
      children: i
    })
  });
});
process.env.NODE_ENV !== "production" && (Go.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: e.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * The color of the component.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   * @default 'primary'
   */
  color: e.oneOfType([e.oneOf(["primary", "secondary", "error", "info", "success", "warning"]), e.string]),
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: e.elementType,
  /**
   * If `true`, the label, input and helper text should be displayed in a disabled state.
   * @default false
   */
  disabled: e.bool,
  /**
   * If `true`, the label is displayed in an error state.
   * @default false
   */
  error: e.bool,
  /**
   * If `true`, the component is displayed in focused state.
   */
  focused: e.bool,
  /**
   * If `true`, the component will take up the full width of its container.
   * @default false
   */
  fullWidth: e.bool,
  /**
   * If `true`, the label is hidden.
   * This is used to increase density for a `FilledInput`.
   * Be sure to add `aria-label` to the `input` element.
   * @default false
   */
  hiddenLabel: e.bool,
  /**
   * If `dense` or `normal`, will adjust vertical spacing of this and contained components.
   * @default 'none'
   */
  margin: e.oneOf(["dense", "none", "normal"]),
  /**
   * If `true`, the label will indicate that the `input` is required.
   * @default false
   */
  required: e.bool,
  /**
   * The size of the component.
   * @default 'medium'
   */
  size: e.oneOfType([e.oneOf(["medium", "small"]), e.string]),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * The variant to use.
   * @default 'outlined'
   */
  variant: e.oneOf(["filled", "outlined", "standard"])
});
function qt(o) {
  return J("MuiFormLabel", o);
}
const ze = X("MuiFormLabel", ["root", "colorSecondary", "focused", "disabled", "error", "filled", "required", "asterisk"]), _t = (o) => {
  const {
    classes: t,
    color: n,
    focused: r,
    disabled: i,
    error: u,
    filled: a,
    required: d
  } = o, m = {
    root: ["root", `color${_(n)}`, i && "disabled", u && "error", a && "filled", r && "focused", d && "required"],
    asterisk: ["asterisk", u && "error"]
  };
  return Q(m, qt, t);
}, Kt = E("label", {
  name: "MuiFormLabel",
  slot: "Root",
  overridesResolver: (o, t) => {
    const {
      ownerState: n
    } = o;
    return [t.root, n.color === "secondary" && t.colorSecondary, n.filled && t.filled];
  }
})(K(({
  theme: o
}) => ({
  color: (o.vars || o).palette.text.secondary,
  ...o.typography.body1,
  lineHeight: "1.4375em",
  padding: 0,
  position: "relative",
  variants: [...Object.entries(o.palette).filter(Ue()).map(([t]) => ({
    props: {
      color: t
    },
    style: {
      [`&.${ze.focused}`]: {
        color: (o.vars || o).palette[t].main
      }
    }
  })), {
    props: {},
    style: {
      [`&.${ze.disabled}`]: {
        color: (o.vars || o).palette.text.disabled
      },
      [`&.${ze.error}`]: {
        color: (o.vars || o).palette.error.main
      }
    }
  }]
}))), Gt = E("span", {
  name: "MuiFormLabel",
  slot: "Asterisk",
  overridesResolver: (o, t) => t.asterisk
})(K(({
  theme: o
}) => ({
  [`&.${ze.error}`]: {
    color: (o.vars || o).palette.error.main
  }
}))), Xo = /* @__PURE__ */ c.forwardRef(function(t, n) {
  const r = ee({
    props: t,
    name: "MuiFormLabel"
  }), {
    children: i,
    className: u,
    color: a,
    component: d = "label",
    disabled: m,
    error: v,
    filled: l,
    focused: h,
    required: x,
    ...g
  } = r, I = Ve(), S = He({
    props: r,
    muiFormControl: I,
    states: ["color", "required", "focused", "disabled", "error", "filled"]
  }), w = {
    ...r,
    color: S.color || "primary",
    component: d,
    disabled: S.disabled,
    error: S.error,
    filled: S.filled,
    focused: S.focused,
    required: S.required
  }, p = _t(w);
  return /* @__PURE__ */ q(Kt, {
    as: d,
    ownerState: w,
    className: Z(p.root, u),
    ref: n,
    ...g,
    children: [i, S.required && /* @__PURE__ */ q(Gt, {
      ownerState: w,
      "aria-hidden": !0,
      className: p.asterisk,
      children: [" ", "*"]
    })]
  });
});
process.env.NODE_ENV !== "production" && (Xo.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: e.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * The color of the component.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   */
  color: e.oneOfType([e.oneOf(["error", "info", "primary", "secondary", "success", "warning"]), e.string]),
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: e.elementType,
  /**
   * If `true`, the label should be displayed in a disabled state.
   */
  disabled: e.bool,
  /**
   * If `true`, the label is displayed in an error state.
   */
  error: e.bool,
  /**
   * If `true`, the label should use filled classes key.
   */
  filled: e.bool,
  /**
   * If `true`, the input of this label is focused (used by `FormGroup` components).
   */
  focused: e.bool,
  /**
   * If `true`, the label will indicate that the `input` is required.
   */
  required: e.bool,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object])
});
const Xt = (o) => {
  const {
    classes: t,
    disableUnderline: n
  } = o, i = Q({
    root: ["root", !n && "underline"],
    input: ["input"]
  }, Tt, t);
  return {
    ...t,
    // forward classes to the InputBase
    ...i
  };
}, Yt = E(Qe, {
  shouldForwardProp: (o) => ne(o) || o === "classes",
  name: "MuiInput",
  slot: "Root",
  overridesResolver: (o, t) => {
    const {
      ownerState: n
    } = o;
    return [...Ze(o, t), !n.disableUnderline && t.underline];
  }
})(K(({
  theme: o
}) => {
  let n = o.palette.mode === "light" ? "rgba(0, 0, 0, 0.42)" : "rgba(255, 255, 255, 0.7)";
  return o.vars && (n = `rgba(${o.vars.palette.common.onBackgroundChannel} / ${o.vars.opacity.inputUnderline})`), {
    position: "relative",
    variants: [{
      props: ({
        ownerState: r
      }) => r.formControl,
      style: {
        "label + &": {
          marginTop: 16
        }
      }
    }, {
      props: ({
        ownerState: r
      }) => !r.disableUnderline,
      style: {
        "&::after": {
          left: 0,
          bottom: 0,
          content: '""',
          position: "absolute",
          right: 0,
          transform: "scaleX(0)",
          transition: o.transitions.create("transform", {
            duration: o.transitions.duration.shorter,
            easing: o.transitions.easing.easeOut
          }),
          pointerEvents: "none"
          // Transparent to the hover style.
        },
        [`&.${Le.focused}:after`]: {
          // translateX(0) is a workaround for Safari transform scale bug
          // See https://github.com/mui/material-ui/issues/31766
          transform: "scaleX(1) translateX(0)"
        },
        [`&.${Le.error}`]: {
          "&::before, &::after": {
            borderBottomColor: (o.vars || o).palette.error.main
          }
        },
        "&::before": {
          borderBottom: `1px solid ${n}`,
          left: 0,
          bottom: 0,
          content: '"\\00a0"',
          position: "absolute",
          right: 0,
          transition: o.transitions.create("border-bottom-color", {
            duration: o.transitions.duration.shorter
          }),
          pointerEvents: "none"
          // Transparent to the hover style.
        },
        [`&:hover:not(.${Le.disabled}, .${Le.error}):before`]: {
          borderBottom: `2px solid ${(o.vars || o).palette.text.primary}`,
          // Reset on touch devices, it doesn't add specificity
          "@media (hover: none)": {
            borderBottom: `1px solid ${n}`
          }
        },
        [`&.${Le.disabled}:before`]: {
          borderBottomStyle: "dotted"
        }
      }
    }, ...Object.entries(o.palette).filter(Ue()).map(([r]) => ({
      props: {
        color: r,
        disableUnderline: !1
      },
      style: {
        "&::after": {
          borderBottom: `2px solid ${(o.vars || o).palette[r].main}`
        }
      }
    }))]
  };
})), Zt = E(eo, {
  name: "MuiInput",
  slot: "Input",
  overridesResolver: Je
})({}), ho = /* @__PURE__ */ c.forwardRef(function(t, n) {
  const r = ee({
    props: t,
    name: "MuiInput"
  }), {
    disableUnderline: i = !1,
    components: u = {},
    componentsProps: a,
    fullWidth: d = !1,
    inputComponent: m = "input",
    multiline: v = !1,
    slotProps: l,
    slots: h = {},
    type: x = "text",
    ...g
  } = r, I = Xt(r), w = {
    root: {
      ownerState: {
        disableUnderline: i
      }
    }
  }, p = l ?? a ? bo(l ?? a, w) : w, O = h.root ?? u.Root ?? Yt, s = h.input ?? u.Input ?? Zt;
  return /* @__PURE__ */ y(oo, {
    slots: {
      root: O,
      input: s
    },
    slotProps: p,
    fullWidth: d,
    inputComponent: m,
    multiline: v,
    ref: n,
    type: x,
    ...g,
    classes: I
  });
});
process.env.NODE_ENV !== "production" && (ho.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * This prop helps users to fill forms faster, especially on mobile devices.
   * The name can be confusing, as it's more like an autofill.
   * You can learn more about it [following the specification](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill).
   */
  autoComplete: e.string,
  /**
   * If `true`, the `input` element is focused during the first mount.
   */
  autoFocus: e.bool,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * The color of the component.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   * The prop defaults to the value (`'primary'`) inherited from the parent FormControl component.
   */
  color: e.oneOfType([e.oneOf(["primary", "secondary"]), e.string]),
  /**
   * The components used for each slot inside.
   *
   * @deprecated use the `slots` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  components: e.shape({
    Input: e.elementType,
    Root: e.elementType
  }),
  /**
   * The extra props for the slot components.
   * You can override the existing props or add new ones.
   *
   * @deprecated use the `slotProps` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  componentsProps: e.shape({
    input: e.object,
    root: e.object
  }),
  /**
   * The default value. Use when the component is not controlled.
   */
  defaultValue: e.any,
  /**
   * If `true`, the component is disabled.
   * The prop defaults to the value (`false`) inherited from the parent FormControl component.
   */
  disabled: e.bool,
  /**
   * If `true`, the `input` will not have an underline.
   * @default false
   */
  disableUnderline: e.bool,
  /**
   * End `InputAdornment` for this component.
   */
  endAdornment: e.node,
  /**
   * If `true`, the `input` will indicate an error.
   * The prop defaults to the value (`false`) inherited from the parent FormControl component.
   */
  error: e.bool,
  /**
   * If `true`, the `input` will take up the full width of its container.
   * @default false
   */
  fullWidth: e.bool,
  /**
   * The id of the `input` element.
   */
  id: e.string,
  /**
   * The component used for the `input` element.
   * Either a string to use a HTML element or a component.
   * @default 'input'
   */
  inputComponent: e.elementType,
  /**
   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes) applied to the `input` element.
   * @default {}
   */
  inputProps: e.object,
  /**
   * Pass a ref to the `input` element.
   */
  inputRef: Me,
  /**
   * If `dense`, will adjust vertical spacing. This is normally obtained via context from
   * FormControl.
   * The prop defaults to the value (`'none'`) inherited from the parent FormControl component.
   */
  margin: e.oneOf(["dense", "none"]),
  /**
   * Maximum number of rows to display when multiline option is set to true.
   */
  maxRows: e.oneOfType([e.number, e.string]),
  /**
   * Minimum number of rows to display when multiline option is set to true.
   */
  minRows: e.oneOfType([e.number, e.string]),
  /**
   * If `true`, a [TextareaAutosize](https://mui.com/material-ui/react-textarea-autosize/) element is rendered.
   * @default false
   */
  multiline: e.bool,
  /**
   * Name attribute of the `input` element.
   */
  name: e.string,
  /**
   * Callback fired when the value is changed.
   *
   * @param {React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>} event The event source of the callback.
   * You can pull out the new value by accessing `event.target.value` (string).
   */
  onChange: e.func,
  /**
   * The short hint displayed in the `input` before the user enters a value.
   */
  placeholder: e.string,
  /**
   * It prevents the user from changing the value of the field
   * (not from interacting with the field).
   */
  readOnly: e.bool,
  /**
   * If `true`, the `input` element is required.
   * The prop defaults to the value (`false`) inherited from the parent FormControl component.
   */
  required: e.bool,
  /**
   * Number of rows to display when multiline option is set to true.
   */
  rows: e.oneOfType([e.number, e.string]),
  /**
   * The extra props for the slot components.
   * You can override the existing props or add new ones.
   *
   * This prop is an alias for the `componentsProps` prop, which will be deprecated in the future.
   *
   * @default {}
   */
  slotProps: e.shape({
    input: e.object,
    root: e.object
  }),
  /**
   * The components used for each slot inside.
   *
   * This prop is an alias for the `components` prop, which will be deprecated in the future.
   *
   * @default {}
   */
  slots: e.shape({
    input: e.elementType,
    root: e.elementType
  }),
  /**
   * Start `InputAdornment` for this component.
   */
  startAdornment: e.node,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * Type of the `input` element. It should be [a valid HTML5 input type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Form_%3Cinput%3E_types).
   * @default 'text'
   */
  type: e.string,
  /**
   * The value of the `input` element, required for a controlled component.
   */
  value: e.any
});
ho.muiName = "Input";
function Jt(o) {
  return J("MuiInputLabel", o);
}
X("MuiInputLabel", ["root", "focused", "disabled", "error", "required", "asterisk", "formControl", "sizeSmall", "shrink", "animated", "standard", "filled", "outlined"]);
const Qt = (o) => {
  const {
    classes: t,
    formControl: n,
    size: r,
    shrink: i,
    disableAnimation: u,
    variant: a,
    required: d
  } = o, m = {
    root: ["root", n && "formControl", !u && "animated", i && "shrink", r && r !== "normal" && `size${_(r)}`, a],
    asterisk: [d && "asterisk"]
  }, v = Q(m, Jt, t);
  return {
    ...t,
    // forward the focused, disabled, etc. classes to the FormLabel
    ...v
  };
}, en = E(Xo, {
  shouldForwardProp: (o) => ne(o) || o === "classes",
  name: "MuiInputLabel",
  slot: "Root",
  overridesResolver: (o, t) => {
    const {
      ownerState: n
    } = o;
    return [{
      [`& .${ze.asterisk}`]: t.asterisk
    }, t.root, n.formControl && t.formControl, n.size === "small" && t.sizeSmall, n.shrink && t.shrink, !n.disableAnimation && t.animated, n.focused && t.focused, t[n.variant]];
  }
})(K(({
  theme: o
}) => ({
  display: "block",
  transformOrigin: "top left",
  whiteSpace: "nowrap",
  overflow: "hidden",
  textOverflow: "ellipsis",
  maxWidth: "100%",
  variants: [{
    props: ({
      ownerState: t
    }) => t.formControl,
    style: {
      position: "absolute",
      left: 0,
      top: 0,
      // slight alteration to spec spacing to match visual spec result
      transform: "translate(0, 20px) scale(1)"
    }
  }, {
    props: {
      size: "small"
    },
    style: {
      // Compensation for the `Input.inputSizeSmall` style.
      transform: "translate(0, 17px) scale(1)"
    }
  }, {
    props: ({
      ownerState: t
    }) => t.shrink,
    style: {
      transform: "translate(0, -1.5px) scale(0.75)",
      transformOrigin: "top left",
      maxWidth: "133%"
    }
  }, {
    props: ({
      ownerState: t
    }) => !t.disableAnimation,
    style: {
      transition: o.transitions.create(["color", "transform", "max-width"], {
        duration: o.transitions.duration.shorter,
        easing: o.transitions.easing.easeOut
      })
    }
  }, {
    props: {
      variant: "filled"
    },
    style: {
      // Chrome's autofill feature gives the input field a yellow background.
      // Since the input field is behind the label in the HTML tree,
      // the input field is drawn last and hides the label with an opaque background color.
      // zIndex: 1 will raise the label above opaque background-colors of input.
      zIndex: 1,
      pointerEvents: "none",
      transform: "translate(12px, 16px) scale(1)",
      maxWidth: "calc(100% - 24px)"
    }
  }, {
    props: {
      variant: "filled",
      size: "small"
    },
    style: {
      transform: "translate(12px, 13px) scale(1)"
    }
  }, {
    props: ({
      variant: t,
      ownerState: n
    }) => t === "filled" && n.shrink,
    style: {
      userSelect: "none",
      pointerEvents: "auto",
      transform: "translate(12px, 7px) scale(0.75)",
      maxWidth: "calc(133% - 24px)"
    }
  }, {
    props: ({
      variant: t,
      ownerState: n,
      size: r
    }) => t === "filled" && n.shrink && r === "small",
    style: {
      transform: "translate(12px, 4px) scale(0.75)"
    }
  }, {
    props: {
      variant: "outlined"
    },
    style: {
      // see comment above on filled.zIndex
      zIndex: 1,
      pointerEvents: "none",
      transform: "translate(14px, 16px) scale(1)",
      maxWidth: "calc(100% - 24px)"
    }
  }, {
    props: {
      variant: "outlined",
      size: "small"
    },
    style: {
      transform: "translate(14px, 9px) scale(1)"
    }
  }, {
    props: ({
      variant: t,
      ownerState: n
    }) => t === "outlined" && n.shrink,
    style: {
      userSelect: "none",
      pointerEvents: "auto",
      // Theoretically, we should have (8+5)*2/0.75 = 34px
      // but it feels a better when it bleeds a bit on the left, so 32px.
      maxWidth: "calc(133% - 32px)",
      transform: "translate(14px, -9px) scale(0.75)"
    }
  }]
}))), Yo = /* @__PURE__ */ c.forwardRef(function(t, n) {
  const r = ee({
    name: "MuiInputLabel",
    props: t
  }), {
    disableAnimation: i = !1,
    margin: u,
    shrink: a,
    variant: d,
    className: m,
    ...v
  } = r, l = Ve();
  let h = a;
  typeof h > "u" && l && (h = l.filled || l.focused || l.adornedStart);
  const x = He({
    props: r,
    muiFormControl: l,
    states: ["size", "variant", "required", "focused"]
  }), g = {
    ...r,
    disableAnimation: i,
    formControl: l,
    shrink: h,
    size: x.size,
    variant: x.variant,
    required: x.required,
    focused: x.focused
  }, I = Qt(g);
  return /* @__PURE__ */ y(en, {
    "data-shrink": h,
    ref: n,
    className: Z(I.root, m),
    ...v,
    ownerState: g,
    classes: I
  });
});
process.env.NODE_ENV !== "production" && (Yo.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: e.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * The color of the component.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   */
  color: e.oneOfType([e.oneOf(["error", "info", "primary", "secondary", "success", "warning"]), e.string]),
  /**
   * If `true`, the transition animation is disabled.
   * @default false
   */
  disableAnimation: e.bool,
  /**
   * If `true`, the component is disabled.
   */
  disabled: e.bool,
  /**
   * If `true`, the label is displayed in an error state.
   */
  error: e.bool,
  /**
   * If `true`, the `input` of this label is focused.
   */
  focused: e.bool,
  /**
   * If `dense`, will adjust vertical spacing. This is normally obtained via context from
   * FormControl.
   */
  margin: e.oneOf(["dense"]),
  /**
   * if `true`, the label will indicate that the `input` is required.
   */
  required: e.bool,
  /**
   * If `true`, the label is shrunk.
   */
  shrink: e.bool,
  /**
   * The size of the component.
   * @default 'normal'
   */
  size: e.oneOfType([e.oneOf(["normal", "small"]), e.string]),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * The variant to use.
   */
  variant: e.oneOf(["filled", "outlined", "standard"])
});
const Do = X("MuiListItemIcon", ["root", "alignItemsFlexStart"]), Ao = X("MuiListItemText", ["root", "multiline", "dense", "inset", "primary", "secondary"]);
function uo(o, t, n) {
  return o === t ? o.firstChild : t && t.nextElementSibling ? t.nextElementSibling : n ? null : o.firstChild;
}
function jo(o, t, n) {
  return o === t ? n ? o.firstChild : o.lastChild : t && t.previousElementSibling ? t.previousElementSibling : n ? null : o.lastChild;
}
function Zo(o, t) {
  if (t === void 0)
    return !0;
  let n = o.innerText;
  return n === void 0 && (n = o.textContent), n = n.trim().toLowerCase(), n.length === 0 ? !1 : t.repeating ? n[0] === t.keys[0] : n.startsWith(t.keys.join(""));
}
function De(o, t, n, r, i, u) {
  let a = !1, d = i(o, t, t ? n : !1);
  for (; d; ) {
    if (d === o.firstChild) {
      if (a)
        return !1;
      a = !0;
    }
    const m = r ? !1 : d.disabled || d.getAttribute("aria-disabled") === "true";
    if (!d.hasAttribute("tabindex") || !Zo(d, u) || m)
      d = i(o, d, n);
    else
      return d.focus(), !0;
  }
  return !1;
}
const Jo = /* @__PURE__ */ c.forwardRef(function(t, n) {
  const {
    // private
    // eslint-disable-next-line react/prop-types
    actions: r,
    autoFocus: i = !1,
    autoFocusItem: u = !1,
    children: a,
    className: d,
    disabledItemsFocusable: m = !1,
    disableListWrap: v = !1,
    onKeyDown: l,
    variant: h = "selectedMenu",
    ...x
  } = t, g = c.useRef(null), I = c.useRef({
    keys: [],
    repeating: !0,
    previousKeyMatched: !0,
    lastTime: null
  });
  We(() => {
    i && g.current.focus();
  }, [i]), c.useImperativeHandle(r, () => ({
    adjustStyleForScrollbar: (s, {
      direction: f
    }) => {
      const R = !g.current.style.width;
      if (s.clientHeight < g.current.clientHeight && R) {
        const T = `${ht(po(s))}px`;
        g.current.style[f === "rtl" ? "paddingLeft" : "paddingRight"] = T, g.current.style.width = `calc(100% + ${T})`;
      }
      return g.current;
    }
  }), []);
  const S = (s) => {
    const f = g.current, R = s.key;
    if (s.ctrlKey || s.metaKey || s.altKey) {
      l && l(s);
      return;
    }
    const $ = _o(f).activeElement;
    if (R === "ArrowDown")
      s.preventDefault(), De(f, $, v, m, uo);
    else if (R === "ArrowUp")
      s.preventDefault(), De(f, $, v, m, jo);
    else if (R === "Home")
      s.preventDefault(), De(f, null, v, m, uo);
    else if (R === "End")
      s.preventDefault(), De(f, null, v, m, jo);
    else if (R.length === 1) {
      const k = I.current, j = R.toLowerCase(), N = performance.now();
      k.keys.length > 0 && (N - k.lastTime > 500 ? (k.keys = [], k.repeating = !0, k.previousKeyMatched = !0) : k.repeating && j !== k.keys[0] && (k.repeating = !1)), k.lastTime = N, k.keys.push(j);
      const B = $ && !k.repeating && Zo($, k);
      k.previousKeyMatched && (B || De(f, $, !1, m, uo, k)) ? s.preventDefault() : k.previousKeyMatched = !1;
    }
    l && l(s);
  }, w = Ee(g, n);
  let p = -1;
  c.Children.forEach(a, (s, f) => {
    if (!/* @__PURE__ */ c.isValidElement(s)) {
      p === f && (p += 1, p >= a.length && (p = -1));
      return;
    }
    process.env.NODE_ENV !== "production" && yo.isFragment(s) && console.error(["MUI: The Menu component doesn't accept a Fragment as a child.", "Consider providing an array instead."].join(`
`)), s.props.disabled || (h === "selectedMenu" && s.props.selected || p === -1) && (p = f), p === f && (s.props.disabled || s.props.muiSkipListHighlight || s.type.muiSkipListHighlight) && (p += 1, p >= a.length && (p = -1));
  });
  const O = c.Children.map(a, (s, f) => {
    if (f === p) {
      const R = {};
      return u && (R.autoFocus = !0), s.props.tabIndex === void 0 && h === "selectedMenu" && (R.tabIndex = 0), /* @__PURE__ */ c.cloneElement(s, R);
    }
    return s;
  });
  return /* @__PURE__ */ y(ft, {
    role: "menu",
    ref: w,
    className: d,
    onKeyDown: S,
    tabIndex: i ? 0 : -1,
    ...x,
    children: O
  });
});
process.env.NODE_ENV !== "production" && (Jo.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * If `true`, will focus the `[role="menu"]` container and move into tab order.
   * @default false
   */
  autoFocus: e.bool,
  /**
   * If `true`, will focus the first menuitem if `variant="menu"` or selected item
   * if `variant="selectedMenu"`.
   * @default false
   */
  autoFocusItem: e.bool,
  /**
   * MenuList contents, normally `MenuItem`s.
   */
  children: e.node,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * If `true`, will allow focus on disabled items.
   * @default false
   */
  disabledItemsFocusable: e.bool,
  /**
   * If `true`, the menu items will not wrap focus.
   * @default false
   */
  disableListWrap: e.bool,
  /**
   * @ignore
   */
  onKeyDown: e.func,
  /**
   * The variant to use. Use `menu` to prevent selected items from impacting the initial focus
   * and the vertical alignment relative to the anchor element.
   * @default 'selectedMenu'
   */
  variant: e.oneOf(["menu", "selectedMenu"])
});
function on(o) {
  return J("MuiMenu", o);
}
X("MuiMenu", ["root", "paper", "list"]);
const tn = {
  vertical: "top",
  horizontal: "right"
}, nn = {
  vertical: "top",
  horizontal: "left"
}, rn = (o) => {
  const {
    classes: t
  } = o;
  return Q({
    root: ["root"],
    paper: ["paper"],
    list: ["list"]
  }, on, t);
}, sn = E(yt, {
  shouldForwardProp: (o) => ne(o) || o === "classes",
  name: "MuiMenu",
  slot: "Root",
  overridesResolver: (o, t) => t.root
})({}), an = E(gt, {
  name: "MuiMenu",
  slot: "Paper",
  overridesResolver: (o, t) => t.paper
})({
  // specZ: The maximum height of a simple menu should be one or more rows less than the view
  // height. This ensures a tappable area outside of the simple menu with which to dismiss
  // the menu.
  maxHeight: "calc(100% - 96px)",
  // Add iOS momentum scrolling for iOS < 13.0
  WebkitOverflowScrolling: "touch"
}), ln = E(Jo, {
  name: "MuiMenu",
  slot: "List",
  overridesResolver: (o, t) => t.list
})({
  // We disable the focus ring for mouse, touch and keyboard users.
  outline: 0
}), Qo = /* @__PURE__ */ c.forwardRef(function(t, n) {
  const r = ee({
    props: t,
    name: "MuiMenu"
  }), {
    autoFocus: i = !0,
    children: u,
    className: a,
    disableAutoFocusItem: d = !1,
    MenuListProps: m = {},
    onClose: v,
    open: l,
    PaperProps: h = {},
    PopoverClasses: x,
    transitionDuration: g = "auto",
    TransitionProps: {
      onEntering: I,
      ...S
    } = {},
    variant: w = "selectedMenu",
    slots: p = {},
    slotProps: O = {},
    ...s
  } = r, f = mt(), R = {
    ...r,
    autoFocus: i,
    disableAutoFocusItem: d,
    MenuListProps: m,
    onEntering: I,
    PaperProps: h,
    transitionDuration: g,
    TransitionProps: S,
    variant: w
  }, T = rn(R), $ = i && !d && l, k = c.useRef(null), j = (b, F) => {
    k.current && k.current.adjustStyleForScrollbar(b, {
      direction: f ? "rtl" : "ltr"
    }), I && I(b, F);
  }, N = (b) => {
    b.key === "Tab" && (b.preventDefault(), v && v(b, "tabKeyDown"));
  };
  let B = -1;
  c.Children.map(u, (b, F) => {
    /* @__PURE__ */ c.isValidElement(b) && (process.env.NODE_ENV !== "production" && yo.isFragment(b) && console.error(["MUI: The Menu component doesn't accept a Fragment as a child.", "Consider providing an array instead."].join(`
`)), b.props.disabled || (w === "selectedMenu" && b.props.selected || B === -1) && (B = F));
  });
  const z = p.paper ?? an, W = O.paper ?? h, U = Eo({
    elementType: p.root,
    externalSlotProps: O.root,
    ownerState: R,
    className: [T.root, a]
  }), V = Eo({
    elementType: z,
    externalSlotProps: W,
    ownerState: R,
    className: T.paper
  });
  return /* @__PURE__ */ y(sn, {
    onClose: v,
    anchorOrigin: {
      vertical: "bottom",
      horizontal: f ? "right" : "left"
    },
    transformOrigin: f ? tn : nn,
    slots: {
      paper: z,
      root: p.root
    },
    slotProps: {
      root: U,
      paper: V
    },
    open: l,
    ref: n,
    transitionDuration: g,
    TransitionProps: {
      onEntering: j,
      ...S
    },
    ownerState: R,
    ...s,
    classes: x,
    children: /* @__PURE__ */ y(ln, {
      onKeyDown: N,
      actions: k,
      autoFocus: i && (B === -1 || d),
      autoFocusItem: $,
      variant: w,
      ...m,
      className: Z(T.list, m.className),
      children: u
    })
  });
});
process.env.NODE_ENV !== "production" && (Qo.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * An HTML element, or a function that returns one.
   * It's used to set the position of the menu.
   */
  anchorEl: e.oneOfType([vt, e.func]),
  /**
   * If `true` (Default) will focus the `[role="menu"]` if no focusable child is found. Disabled
   * children are not focusable. If you set this prop to `false` focus will be placed
   * on the parent modal container. This has severe accessibility implications
   * and should only be considered if you manage focus otherwise.
   * @default true
   */
  autoFocus: e.bool,
  /**
   * Menu contents, normally `MenuItem`s.
   */
  children: e.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * When opening the menu will not focus the active item but the `[role="menu"]`
   * unless `autoFocus` is also set to `false`. Not using the default means not
   * following WAI-ARIA authoring practices. Please be considerate about possible
   * accessibility implications.
   * @default false
   */
  disableAutoFocusItem: e.bool,
  /**
   * Props applied to the [`MenuList`](https://mui.com/material-ui/api/menu-list/) element.
   * @default {}
   */
  MenuListProps: e.object,
  /**
   * Callback fired when the component requests to be closed.
   *
   * @param {object} event The event source of the callback.
   * @param {string} reason Can be: `"escapeKeyDown"`, `"backdropClick"`, `"tabKeyDown"`.
   */
  onClose: e.func,
  /**
   * If `true`, the component is shown.
   */
  open: e.bool.isRequired,
  /**
   * @ignore
   */
  PaperProps: e.object,
  /**
   * `classes` prop applied to the [`Popover`](https://mui.com/material-ui/api/popover/) element.
   */
  PopoverClasses: e.object,
  /**
   * The props used for each slot inside.
   * @default {}
   */
  slotProps: e.shape({
    paper: e.oneOfType([e.func, e.object]),
    root: e.oneOfType([e.func, e.object])
  }),
  /**
   * The components used for each slot inside.
   * @default {}
   */
  slots: e.shape({
    paper: e.elementType,
    root: e.elementType
  }),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * The length of the transition in `ms`, or 'auto'
   * @default 'auto'
   */
  transitionDuration: e.oneOfType([e.oneOf(["auto"]), e.number, e.shape({
    appear: e.number,
    enter: e.number,
    exit: e.number
  })]),
  /**
   * Props applied to the transition element.
   * By default, the element is based on this [`Transition`](https://reactcommunity.org/react-transition-group/transition/) component.
   * @default {}
   */
  TransitionProps: e.object,
  /**
   * The variant to use. Use `menu` to prevent selected items from impacting the initial focus.
   * @default 'selectedMenu'
   */
  variant: e.oneOf(["menu", "selectedMenu"])
});
function cn(o) {
  return J("MuiMenuItem", o);
}
const Ae = X("MuiMenuItem", ["root", "focusVisible", "dense", "disabled", "divider", "gutters", "selected"]), dn = (o, t) => {
  const {
    ownerState: n
  } = o;
  return [t.root, n.dense && t.dense, n.divider && t.divider, !n.disableGutters && t.gutters];
}, un = (o) => {
  const {
    disabled: t,
    dense: n,
    divider: r,
    disableGutters: i,
    selected: u,
    classes: a
  } = o, m = Q({
    root: ["root", n && "dense", t && "disabled", !i && "gutters", r && "divider", u && "selected"]
  }, cn, a);
  return {
    ...a,
    ...m
  };
}, pn = E(ut, {
  shouldForwardProp: (o) => ne(o) || o === "classes",
  name: "MuiMenuItem",
  slot: "Root",
  overridesResolver: dn
})(K(({
  theme: o
}) => ({
  ...o.typography.body1,
  display: "flex",
  justifyContent: "flex-start",
  alignItems: "center",
  position: "relative",
  textDecoration: "none",
  minHeight: 48,
  paddingTop: 6,
  paddingBottom: 6,
  boxSizing: "border-box",
  whiteSpace: "nowrap",
  "&:hover": {
    textDecoration: "none",
    backgroundColor: (o.vars || o).palette.action.hover,
    // Reset on touch devices, it doesn't add specificity
    "@media (hover: none)": {
      backgroundColor: "transparent"
    }
  },
  [`&.${Ae.selected}`]: {
    backgroundColor: o.vars ? `rgba(${o.vars.palette.primary.mainChannel} / ${o.vars.palette.action.selectedOpacity})` : Ke(o.palette.primary.main, o.palette.action.selectedOpacity),
    [`&.${Ae.focusVisible}`]: {
      backgroundColor: o.vars ? `rgba(${o.vars.palette.primary.mainChannel} / calc(${o.vars.palette.action.selectedOpacity} + ${o.vars.palette.action.focusOpacity}))` : Ke(o.palette.primary.main, o.palette.action.selectedOpacity + o.palette.action.focusOpacity)
    }
  },
  [`&.${Ae.selected}:hover`]: {
    backgroundColor: o.vars ? `rgba(${o.vars.palette.primary.mainChannel} / calc(${o.vars.palette.action.selectedOpacity} + ${o.vars.palette.action.hoverOpacity}))` : Ke(o.palette.primary.main, o.palette.action.selectedOpacity + o.palette.action.hoverOpacity),
    // Reset on touch devices, it doesn't add specificity
    "@media (hover: none)": {
      backgroundColor: o.vars ? `rgba(${o.vars.palette.primary.mainChannel} / ${o.vars.palette.action.selectedOpacity})` : Ke(o.palette.primary.main, o.palette.action.selectedOpacity)
    }
  },
  [`&.${Ae.focusVisible}`]: {
    backgroundColor: (o.vars || o).palette.action.focus
  },
  [`&.${Ae.disabled}`]: {
    opacity: (o.vars || o).palette.action.disabledOpacity
  },
  [`& + .${No.root}`]: {
    marginTop: o.spacing(1),
    marginBottom: o.spacing(1)
  },
  [`& + .${No.inset}`]: {
    marginLeft: 52
  },
  [`& .${Ao.root}`]: {
    marginTop: 0,
    marginBottom: 0
  },
  [`& .${Ao.inset}`]: {
    paddingLeft: 36
  },
  [`& .${Do.root}`]: {
    minWidth: 36
  },
  variants: [{
    props: ({
      ownerState: t
    }) => !t.disableGutters,
    style: {
      paddingLeft: 16,
      paddingRight: 16
    }
  }, {
    props: ({
      ownerState: t
    }) => t.divider,
    style: {
      borderBottom: `1px solid ${(o.vars || o).palette.divider}`,
      backgroundClip: "padding-box"
    }
  }, {
    props: ({
      ownerState: t
    }) => !t.dense,
    style: {
      [o.breakpoints.up("sm")]: {
        minHeight: "auto"
      }
    }
  }, {
    props: ({
      ownerState: t
    }) => t.dense,
    style: {
      minHeight: 32,
      // https://m2.material.io/components/menus#specs > Dense
      paddingTop: 4,
      paddingBottom: 4,
      ...o.typography.body2,
      [`& .${Do.root} svg`]: {
        fontSize: "1.25rem"
      }
    }
  }]
}))), et = /* @__PURE__ */ c.forwardRef(function(t, n) {
  const r = ee({
    props: t,
    name: "MuiMenuItem"
  }), {
    autoFocus: i = !1,
    component: u = "li",
    dense: a = !1,
    divider: d = !1,
    disableGutters: m = !1,
    focusVisibleClassName: v,
    role: l = "menuitem",
    tabIndex: h,
    className: x,
    ...g
  } = r, I = c.useContext(Mo), S = c.useMemo(() => ({
    dense: a || I.dense || !1,
    disableGutters: m
  }), [I.dense, a, m]), w = c.useRef(null);
  We(() => {
    i && (w.current ? w.current.focus() : process.env.NODE_ENV !== "production" && console.error("MUI: Unable to set focus to a MenuItem whose component has not been rendered."));
  }, [i]);
  const p = {
    ...r,
    dense: S.dense,
    divider: d,
    disableGutters: m
  }, O = un(r), s = Ee(w, n);
  let f;
  return r.disabled || (f = h !== void 0 ? h : -1), /* @__PURE__ */ y(Mo.Provider, {
    value: S,
    children: /* @__PURE__ */ y(pn, {
      ref: s,
      role: l,
      tabIndex: f,
      component: u,
      focusVisibleClassName: Z(O.focusVisible, v),
      className: Z(O.root, x),
      ...g,
      ownerState: p,
      classes: O
    })
  });
});
process.env.NODE_ENV !== "production" && (et.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * If `true`, the list item is focused during the first mount.
   * Focus will also be triggered if the value changes from false to true.
   * @default false
   */
  autoFocus: e.bool,
  /**
   * The content of the component.
   */
  children: e.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: e.elementType,
  /**
   * If `true`, compact vertical padding designed for keyboard and mouse input is used.
   * The prop defaults to the value inherited from the parent Menu component.
   * @default false
   */
  dense: e.bool,
  /**
   * @ignore
   */
  disabled: e.bool,
  /**
   * If `true`, the left and right padding is removed.
   * @default false
   */
  disableGutters: e.bool,
  /**
   * If `true`, a 1px light border is added to the bottom of the menu item.
   * @default false
   */
  divider: e.bool,
  /**
   * This prop can help identify which element has keyboard focus.
   * The class name will be applied when the element gains the focus through keyboard interaction.
   * It's a polyfill for the [CSS :focus-visible selector](https://drafts.csswg.org/selectors-4/#the-focus-visible-pseudo).
   * The rationale for using this feature [is explained here](https://github.com/WICG/focus-visible/blob/HEAD/explainer.md).
   * A [polyfill can be used](https://github.com/WICG/focus-visible) to apply a `focus-visible` class to other components
   * if needed.
   */
  focusVisibleClassName: e.string,
  /**
   * @ignore
   */
  role: e.string,
  /**
   * If `true`, the component is selected.
   * @default false
   */
  selected: e.bool,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * @default 0
   */
  tabIndex: e.number
});
function fn(o) {
  return J("MuiNativeSelect", o);
}
const vo = X("MuiNativeSelect", ["root", "select", "multiple", "filled", "outlined", "standard", "disabled", "icon", "iconOpen", "iconFilled", "iconOutlined", "iconStandard", "nativeInput", "error"]), mn = (o) => {
  const {
    classes: t,
    variant: n,
    disabled: r,
    multiple: i,
    open: u,
    error: a
  } = o, d = {
    select: ["select", n, r && "disabled", i && "multiple", a && "error"],
    icon: ["icon", `icon${_(n)}`, u && "iconOpen", r && "disabled"]
  };
  return Q(d, fn, t);
}, ot = E("select")(({
  theme: o
}) => ({
  // Reset
  MozAppearance: "none",
  // Reset
  WebkitAppearance: "none",
  // When interacting quickly, the text can end up selected.
  // Native select can't be selected either.
  userSelect: "none",
  // Reset
  borderRadius: 0,
  cursor: "pointer",
  "&:focus": {
    // Reset Chrome style
    borderRadius: 0
  },
  [`&.${vo.disabled}`]: {
    cursor: "default"
  },
  "&[multiple]": {
    height: "auto"
  },
  "&:not([multiple]) option, &:not([multiple]) optgroup": {
    backgroundColor: (o.vars || o).palette.background.paper
  },
  variants: [{
    props: ({
      ownerState: t
    }) => t.variant !== "filled" && t.variant !== "outlined",
    style: {
      // Bump specificity to allow extending custom inputs
      "&&&": {
        paddingRight: 24,
        minWidth: 16
        // So it doesn't collapse.
      }
    }
  }, {
    props: {
      variant: "filled"
    },
    style: {
      "&&&": {
        paddingRight: 32
      }
    }
  }, {
    props: {
      variant: "outlined"
    },
    style: {
      borderRadius: (o.vars || o).shape.borderRadius,
      "&:focus": {
        borderRadius: (o.vars || o).shape.borderRadius
        // Reset the reset for Chrome style
      },
      "&&&": {
        paddingRight: 32
      }
    }
  }]
})), bn = E(ot, {
  name: "MuiNativeSelect",
  slot: "Select",
  shouldForwardProp: ne,
  overridesResolver: (o, t) => {
    const {
      ownerState: n
    } = o;
    return [t.select, t[n.variant], n.error && t.error, {
      [`&.${vo.multiple}`]: t.multiple
    }];
  }
})({}), tt = E("svg")(({
  theme: o
}) => ({
  // We use a position absolute over a flexbox in order to forward the pointer events
  // to the input and to support wrapping tags..
  position: "absolute",
  right: 0,
  // Center vertically, height is 1em
  top: "calc(50% - .5em)",
  // Don't block pointer events on the select under the icon.
  pointerEvents: "none",
  color: (o.vars || o).palette.action.active,
  [`&.${vo.disabled}`]: {
    color: (o.vars || o).palette.action.disabled
  },
  variants: [{
    props: ({
      ownerState: t
    }) => t.open,
    style: {
      transform: "rotate(180deg)"
    }
  }, {
    props: {
      variant: "filled"
    },
    style: {
      right: 7
    }
  }, {
    props: {
      variant: "outlined"
    },
    style: {
      right: 7
    }
  }]
})), yn = E(tt, {
  name: "MuiNativeSelect",
  slot: "Icon",
  overridesResolver: (o, t) => {
    const {
      ownerState: n
    } = o;
    return [t.icon, n.variant && t[`icon${_(n.variant)}`], n.open && t.iconOpen];
  }
})({}), nt = /* @__PURE__ */ c.forwardRef(function(t, n) {
  const {
    className: r,
    disabled: i,
    error: u,
    IconComponent: a,
    inputRef: d,
    variant: m = "standard",
    ...v
  } = t, l = {
    ...t,
    disabled: i,
    variant: m,
    error: u
  }, h = mn(l);
  return /* @__PURE__ */ q(c.Fragment, {
    children: [/* @__PURE__ */ y(bn, {
      ownerState: l,
      className: Z(h.select, r),
      disabled: i,
      ref: d || n,
      ...v
    }), t.multiple ? null : /* @__PURE__ */ y(yn, {
      as: a,
      ownerState: l,
      className: h.icon
    })]
  });
});
process.env.NODE_ENV !== "production" && (nt.propTypes = {
  /**
   * The option elements to populate the select with.
   * Can be some `<option>` elements.
   */
  children: e.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * The CSS class name of the select element.
   */
  className: e.string,
  /**
   * If `true`, the select is disabled.
   */
  disabled: e.bool,
  /**
   * If `true`, the `select input` will indicate an error.
   */
  error: e.bool,
  /**
   * The icon that displays the arrow.
   */
  IconComponent: e.elementType.isRequired,
  /**
   * Use that prop to pass a ref to the native select element.
   * @deprecated
   */
  inputRef: Me,
  /**
   * @ignore
   */
  multiple: e.bool,
  /**
   * Name attribute of the `select` or hidden `input` element.
   */
  name: e.string,
  /**
   * Callback fired when a menu item is selected.
   *
   * @param {object} event The event source of the callback.
   * You can pull out the new value by accessing `event.target.value` (string).
   */
  onChange: e.func,
  /**
   * The input value.
   */
  value: e.any,
  /**
   * The variant to use.
   */
  variant: e.oneOf(["standard", "outlined", "filled"])
});
var Bo;
const gn = E("fieldset", {
  shouldForwardProp: ne
})({
  textAlign: "left",
  position: "absolute",
  bottom: 0,
  right: 0,
  top: -5,
  left: 0,
  margin: 0,
  padding: "0 8px",
  pointerEvents: "none",
  borderRadius: "inherit",
  borderStyle: "solid",
  borderWidth: 1,
  overflow: "hidden",
  minWidth: "0%"
}), hn = E("legend", {
  shouldForwardProp: ne
})(K(({
  theme: o
}) => ({
  float: "unset",
  // Fix conflict with bootstrap
  width: "auto",
  // Fix conflict with bootstrap
  overflow: "hidden",
  // Fix Horizontal scroll when label too long
  variants: [{
    props: ({
      ownerState: t
    }) => !t.withLabel,
    style: {
      padding: 0,
      lineHeight: "11px",
      // sync with `height` in `legend` styles
      transition: o.transitions.create("width", {
        duration: 150,
        easing: o.transitions.easing.easeOut
      })
    }
  }, {
    props: ({
      ownerState: t
    }) => t.withLabel,
    style: {
      display: "block",
      // Fix conflict with normalize.css and sanitize.css
      padding: 0,
      height: 11,
      // sync with `lineHeight` in `legend` styles
      fontSize: "0.75em",
      visibility: "hidden",
      maxWidth: 0.01,
      transition: o.transitions.create("max-width", {
        duration: 50,
        easing: o.transitions.easing.easeOut
      }),
      whiteSpace: "nowrap",
      "& > span": {
        paddingLeft: 5,
        paddingRight: 5,
        display: "inline-block",
        opacity: 0,
        visibility: "visible"
      }
    }
  }, {
    props: ({
      ownerState: t
    }) => t.withLabel && t.notched,
    style: {
      maxWidth: "100%",
      transition: o.transitions.create("max-width", {
        duration: 100,
        easing: o.transitions.easing.easeOut,
        delay: 50
      })
    }
  }]
})));
function rt(o) {
  const {
    children: t,
    classes: n,
    className: r,
    label: i,
    notched: u,
    ...a
  } = o, d = i != null && i !== "", m = {
    ...o,
    notched: u,
    withLabel: d
  };
  return /* @__PURE__ */ y(gn, {
    "aria-hidden": !0,
    className: r,
    ownerState: m,
    ...a,
    children: /* @__PURE__ */ y(hn, {
      ownerState: m,
      children: d ? /* @__PURE__ */ y("span", {
        children: i
      }) : (
        // notranslate needed while Google Translate will not fix zero-width space issue
        Bo || (Bo = /* @__PURE__ */ y("span", {
          className: "notranslate",
          "aria-hidden": !0,
          children: "​"
        }))
      )
    })
  });
}
process.env.NODE_ENV !== "production" && (rt.propTypes = {
  /**
   * The content of the component.
   */
  children: e.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * The label.
   */
  label: e.node,
  /**
   * If `true`, the outline is notched to accommodate the label.
   */
  notched: e.bool.isRequired,
  /**
   * @ignore
   */
  style: e.object
});
const vn = (o) => {
  const {
    classes: t
  } = o, r = Q({
    root: ["root"],
    notchedOutline: ["notchedOutline"],
    input: ["input"]
  }, Mt, t);
  return {
    ...t,
    // forward classes to the InputBase
    ...r
  };
}, Cn = E(Qe, {
  shouldForwardProp: (o) => ne(o) || o === "classes",
  name: "MuiOutlinedInput",
  slot: "Root",
  overridesResolver: Ze
})(K(({
  theme: o
}) => {
  const t = o.palette.mode === "light" ? "rgba(0, 0, 0, 0.23)" : "rgba(255, 255, 255, 0.23)";
  return {
    position: "relative",
    borderRadius: (o.vars || o).shape.borderRadius,
    [`&:hover .${te.notchedOutline}`]: {
      borderColor: (o.vars || o).palette.text.primary
    },
    // Reset on touch devices, it doesn't add specificity
    "@media (hover: none)": {
      [`&:hover .${te.notchedOutline}`]: {
        borderColor: o.vars ? `rgba(${o.vars.palette.common.onBackgroundChannel} / 0.23)` : t
      }
    },
    [`&.${te.focused} .${te.notchedOutline}`]: {
      borderWidth: 2
    },
    variants: [...Object.entries(o.palette).filter(Ue()).map(([n]) => ({
      props: {
        color: n
      },
      style: {
        [`&.${te.focused} .${te.notchedOutline}`]: {
          borderColor: (o.vars || o).palette[n].main
        }
      }
    })), {
      props: {},
      // to overide the above style
      style: {
        [`&.${te.error} .${te.notchedOutline}`]: {
          borderColor: (o.vars || o).palette.error.main
        },
        [`&.${te.disabled} .${te.notchedOutline}`]: {
          borderColor: (o.vars || o).palette.action.disabled
        }
      }
    }, {
      props: ({
        ownerState: n
      }) => n.startAdornment,
      style: {
        paddingLeft: 14
      }
    }, {
      props: ({
        ownerState: n
      }) => n.endAdornment,
      style: {
        paddingRight: 14
      }
    }, {
      props: ({
        ownerState: n
      }) => n.multiline,
      style: {
        padding: "16.5px 14px"
      }
    }, {
      props: ({
        ownerState: n,
        size: r
      }) => n.multiline && r === "small",
      style: {
        padding: "8.5px 14px"
      }
    }]
  };
})), Sn = E(rt, {
  name: "MuiOutlinedInput",
  slot: "NotchedOutline",
  overridesResolver: (o, t) => t.notchedOutline
})(K(({
  theme: o
}) => {
  const t = o.palette.mode === "light" ? "rgba(0, 0, 0, 0.23)" : "rgba(255, 255, 255, 0.23)";
  return {
    borderColor: o.vars ? `rgba(${o.vars.palette.common.onBackgroundChannel} / 0.23)` : t
  };
})), wn = E(eo, {
  name: "MuiOutlinedInput",
  slot: "Input",
  overridesResolver: Je
})(K(({
  theme: o
}) => ({
  padding: "16.5px 14px",
  ...!o.vars && {
    "&:-webkit-autofill": {
      WebkitBoxShadow: o.palette.mode === "light" ? null : "0 0 0 100px #266798 inset",
      WebkitTextFillColor: o.palette.mode === "light" ? null : "#fff",
      caretColor: o.palette.mode === "light" ? null : "#fff",
      borderRadius: "inherit"
    }
  },
  ...o.vars && {
    "&:-webkit-autofill": {
      borderRadius: "inherit"
    },
    [o.getColorSchemeSelector("dark")]: {
      "&:-webkit-autofill": {
        WebkitBoxShadow: "0 0 0 100px #266798 inset",
        WebkitTextFillColor: "#fff",
        caretColor: "#fff"
      }
    }
  },
  variants: [{
    props: {
      size: "small"
    },
    style: {
      padding: "8.5px 14px"
    }
  }, {
    props: ({
      ownerState: t
    }) => t.multiline,
    style: {
      padding: 0
    }
  }, {
    props: ({
      ownerState: t
    }) => t.startAdornment,
    style: {
      paddingLeft: 0
    }
  }, {
    props: ({
      ownerState: t
    }) => t.endAdornment,
    style: {
      paddingRight: 0
    }
  }]
}))), Co = /* @__PURE__ */ c.forwardRef(function(t, n) {
  var r;
  const i = ee({
    props: t,
    name: "MuiOutlinedInput"
  }), {
    components: u = {},
    fullWidth: a = !1,
    inputComponent: d = "input",
    label: m,
    multiline: v = !1,
    notched: l,
    slots: h = {},
    type: x = "text",
    ...g
  } = i, I = vn(i), S = Ve(), w = He({
    props: i,
    muiFormControl: S,
    states: ["color", "disabled", "error", "focused", "hiddenLabel", "size", "required"]
  }), p = {
    ...i,
    color: w.color || "primary",
    disabled: w.disabled,
    error: w.error,
    focused: w.focused,
    formControl: S,
    fullWidth: a,
    hiddenLabel: w.hiddenLabel,
    multiline: v,
    size: w.size,
    type: x
  }, O = h.root ?? u.Root ?? Cn, s = h.input ?? u.Input ?? wn;
  return /* @__PURE__ */ y(oo, {
    slots: {
      root: O,
      input: s
    },
    renderSuffix: (f) => /* @__PURE__ */ y(Sn, {
      ownerState: p,
      className: I.notchedOutline,
      label: m != null && m !== "" && w.required ? r || (r = /* @__PURE__ */ q(c.Fragment, {
        children: [m, " ", "*"]
      })) : m,
      notched: typeof l < "u" ? l : !!(f.startAdornment || f.filled || f.focused)
    }),
    fullWidth: a,
    inputComponent: d,
    multiline: v,
    ref: n,
    type: x,
    ...g,
    classes: {
      ...I,
      notchedOutline: null
    }
  });
});
process.env.NODE_ENV !== "production" && (Co.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * This prop helps users to fill forms faster, especially on mobile devices.
   * The name can be confusing, as it's more like an autofill.
   * You can learn more about it [following the specification](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill).
   */
  autoComplete: e.string,
  /**
   * If `true`, the `input` element is focused during the first mount.
   */
  autoFocus: e.bool,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * The color of the component.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   * The prop defaults to the value (`'primary'`) inherited from the parent FormControl component.
   */
  color: e.oneOfType([e.oneOf(["primary", "secondary"]), e.string]),
  /**
   * The components used for each slot inside.
   *
   * @deprecated use the `slots` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  components: e.shape({
    Input: e.elementType,
    Root: e.elementType
  }),
  /**
   * The default value. Use when the component is not controlled.
   */
  defaultValue: e.any,
  /**
   * If `true`, the component is disabled.
   * The prop defaults to the value (`false`) inherited from the parent FormControl component.
   */
  disabled: e.bool,
  /**
   * End `InputAdornment` for this component.
   */
  endAdornment: e.node,
  /**
   * If `true`, the `input` will indicate an error.
   * The prop defaults to the value (`false`) inherited from the parent FormControl component.
   */
  error: e.bool,
  /**
   * If `true`, the `input` will take up the full width of its container.
   * @default false
   */
  fullWidth: e.bool,
  /**
   * The id of the `input` element.
   */
  id: e.string,
  /**
   * The component used for the `input` element.
   * Either a string to use a HTML element or a component.
   * @default 'input'
   */
  inputComponent: e.elementType,
  /**
   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes) applied to the `input` element.
   * @default {}
   */
  inputProps: e.object,
  /**
   * Pass a ref to the `input` element.
   */
  inputRef: Me,
  /**
   * The label of the `input`. It is only used for layout. The actual labelling
   * is handled by `InputLabel`.
   */
  label: e.node,
  /**
   * If `dense`, will adjust vertical spacing. This is normally obtained via context from
   * FormControl.
   * The prop defaults to the value (`'none'`) inherited from the parent FormControl component.
   */
  margin: e.oneOf(["dense", "none"]),
  /**
   * Maximum number of rows to display when multiline option is set to true.
   */
  maxRows: e.oneOfType([e.number, e.string]),
  /**
   * Minimum number of rows to display when multiline option is set to true.
   */
  minRows: e.oneOfType([e.number, e.string]),
  /**
   * If `true`, a [TextareaAutosize](https://mui.com/material-ui/react-textarea-autosize/) element is rendered.
   * @default false
   */
  multiline: e.bool,
  /**
   * Name attribute of the `input` element.
   */
  name: e.string,
  /**
   * If `true`, the outline is notched to accommodate the label.
   */
  notched: e.bool,
  /**
   * Callback fired when the value is changed.
   *
   * @param {React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>} event The event source of the callback.
   * You can pull out the new value by accessing `event.target.value` (string).
   */
  onChange: e.func,
  /**
   * The short hint displayed in the `input` before the user enters a value.
   */
  placeholder: e.string,
  /**
   * It prevents the user from changing the value of the field
   * (not from interacting with the field).
   */
  readOnly: e.bool,
  /**
   * If `true`, the `input` element is required.
   * The prop defaults to the value (`false`) inherited from the parent FormControl component.
   */
  required: e.bool,
  /**
   * Number of rows to display when multiline option is set to true.
   */
  rows: e.oneOfType([e.number, e.string]),
  /**
   * The components used for each slot inside.
   *
   * This prop is an alias for the `components` prop, which will be deprecated in the future.
   *
   * @default {}
   */
  slots: e.shape({
    input: e.elementType,
    root: e.elementType
  }),
  /**
   * Start `InputAdornment` for this component.
   */
  startAdornment: e.node,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * Type of the `input` element. It should be [a valid HTML5 input type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Form_%3Cinput%3E_types).
   * @default 'text'
   */
  type: e.string,
  /**
   * The value of the `input` element, required for a controlled component.
   */
  value: e.any
});
Co.muiName = "Input";
function On(o) {
  return J("MuiSelect", o);
}
const je = X("MuiSelect", ["root", "select", "multiple", "filled", "outlined", "standard", "disabled", "focused", "icon", "iconOpen", "iconFilled", "iconOutlined", "iconStandard", "nativeInput", "error"]);
var zo;
const xn = E(ot, {
  name: "MuiSelect",
  slot: "Select",
  overridesResolver: (o, t) => {
    const {
      ownerState: n
    } = o;
    return [
      // Win specificity over the input base
      {
        [`&.${je.select}`]: t.select
      },
      {
        [`&.${je.select}`]: t[n.variant]
      },
      {
        [`&.${je.error}`]: t.error
      },
      {
        [`&.${je.multiple}`]: t.multiple
      }
    ];
  }
})({
  // Win specificity over the input base
  [`&.${je.select}`]: {
    height: "auto",
    // Resets for multiple select with chips
    minHeight: "1.4375em",
    // Required for select\text-field height consistency
    textOverflow: "ellipsis",
    whiteSpace: "nowrap",
    overflow: "hidden"
  }
}), In = E(tt, {
  name: "MuiSelect",
  slot: "Icon",
  overridesResolver: (o, t) => {
    const {
      ownerState: n
    } = o;
    return [t.icon, n.variant && t[`icon${_(n.variant)}`], n.open && t.iconOpen];
  }
})({}), Rn = E("input", {
  shouldForwardProp: (o) => lt(o) && o !== "classes",
  name: "MuiSelect",
  slot: "NativeInput",
  overridesResolver: (o, t) => t.nativeInput
})({
  bottom: 0,
  left: 0,
  position: "absolute",
  opacity: 0,
  pointerEvents: "none",
  width: "100%",
  boxSizing: "border-box"
});
function Wo(o, t) {
  return typeof t == "object" && t !== null ? o === t : String(o) === String(t);
}
function kn(o) {
  return o == null || typeof o == "string" && !o.trim();
}
const Tn = (o) => {
  const {
    classes: t,
    variant: n,
    disabled: r,
    multiple: i,
    open: u,
    error: a
  } = o, d = {
    select: ["select", n, r && "disabled", i && "multiple", a && "error"],
    icon: ["icon", `icon${_(n)}`, u && "iconOpen", r && "disabled"],
    nativeInput: ["nativeInput"]
  };
  return Q(d, On, t);
}, st = /* @__PURE__ */ c.forwardRef(function(t, n) {
  var xo;
  const {
    "aria-describedby": r,
    "aria-label": i,
    autoFocus: u,
    autoWidth: a,
    children: d,
    className: m,
    defaultOpen: v,
    defaultValue: l,
    disabled: h,
    displayEmpty: x,
    error: g = !1,
    IconComponent: I,
    inputRef: S,
    labelId: w,
    MenuProps: p = {},
    multiple: O,
    name: s,
    onBlur: f,
    onChange: R,
    onClose: T,
    onFocus: $,
    onOpen: k,
    open: j,
    readOnly: N,
    renderValue: B,
    required: z,
    SelectDisplayProps: W = {},
    tabIndex: U,
    // catching `type` from Input which makes no sense for SelectInput
    type: V,
    value: b,
    variant: F = "standard",
    ...L
  } = t, [P, ve] = Po({
    controlled: b,
    default: l,
    name: "Select"
  }), [ae, qe] = Po({
    controlled: j,
    default: v,
    name: "Select"
  }), ie = c.useRef(null), oe = c.useRef(null), [re, to] = c.useState(null), {
    current: le
  } = c.useRef(j != null), [Ne, se] = c.useState(), no = Ee(n, S), ro = c.useCallback((C) => {
    oe.current = C, C && to(C);
  }, []), ce = re == null ? void 0 : re.parentNode;
  c.useImperativeHandle(no, () => ({
    focus: () => {
      oe.current.focus();
    },
    node: ie.current,
    value: P
  }), [P]), c.useEffect(() => {
    v && ae && re && !le && (se(a ? null : ce.clientWidth), oe.current.focus());
  }, [re, a]), c.useEffect(() => {
    u && oe.current.focus();
  }, [u]), c.useEffect(() => {
    if (!w)
      return;
    const C = _o(oe.current).getElementById(w);
    if (C) {
      const M = () => {
        getSelection().isCollapsed && oe.current.focus();
      };
      return C.addEventListener("click", M), () => {
        C.removeEventListener("click", M);
      };
    }
  }, [w]);
  const de = (C, M) => {
    C ? k && k(M) : T && T(M), le || (se(a ? null : ce.clientWidth), qe(C));
  }, A = (C) => {
    C.button === 0 && (C.preventDefault(), oe.current.focus(), de(!0, C));
  }, Y = (C) => {
    de(!1, C);
  }, ue = c.Children.toArray(d), Pe = (C) => {
    const M = ue.find((H) => H.props.value === C.target.value);
    M !== void 0 && (ve(M.props.value), R && R(C, M));
  }, me = (C) => (M) => {
    let H;
    if (M.currentTarget.hasAttribute("tabindex")) {
      if (O) {
        H = Array.isArray(P) ? P.slice() : [];
        const Ie = P.indexOf(C.props.value);
        Ie === -1 ? H.push(C.props.value) : H.splice(Ie, 1);
      } else
        H = C.props.value;
      if (C.props.onClick && C.props.onClick(M), P !== H && (ve(H), R)) {
        const Ie = M.nativeEvent || M, Io = new Ie.constructor(Ie.type, Ie);
        Object.defineProperty(Io, "target", {
          writable: !0,
          value: {
            value: H,
            name: s
          }
        }), R(Io, C);
      }
      O || de(!1, M);
    }
  }, so = (C) => {
    N || [
      " ",
      "ArrowUp",
      "ArrowDown",
      // The native select doesn't respond to enter on macOS, but it's recommended by
      // https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/
      "Enter"
    ].includes(C.key) && (C.preventDefault(), de(!0, C));
  }, Ce = re !== null && ae, ao = (C) => {
    !Ce && f && (Object.defineProperty(C, "target", {
      writable: !0,
      value: {
        value: P,
        name: s
      }
    }), f(C));
  };
  delete L["aria-invalid"];
  let pe, Se;
  const G = [];
  let we = !1, be = !1;
  (Ye({
    value: P
  }) || x) && (B ? pe = B(P) : we = !0);
  const _e = ue.map((C) => {
    if (!/* @__PURE__ */ c.isValidElement(C))
      return null;
    process.env.NODE_ENV !== "production" && yo.isFragment(C) && console.error(["MUI: The Select component doesn't accept a Fragment as a child.", "Consider providing an array instead."].join(`
`));
    let M;
    if (O) {
      if (!Array.isArray(P))
        throw new Error(process.env.NODE_ENV !== "production" ? "MUI: The `value` prop must be an array when using the `Select` component with `multiple`." : Uo(2));
      M = P.some((H) => Wo(H, C.props.value)), M && we && G.push(C.props.children);
    } else
      M = Wo(P, C.props.value), M && we && (Se = C.props.children);
    return M && (be = !0), /* @__PURE__ */ c.cloneElement(C, {
      "aria-selected": M ? "true" : "false",
      onClick: me(C),
      onKeyUp: (H) => {
        H.key === " " && H.preventDefault(), C.props.onKeyUp && C.props.onKeyUp(H);
      },
      role: "option",
      selected: M,
      value: void 0,
      // The value is most likely not a valid HTML attribute.
      "data-value": C.props.value
      // Instead, we provide it as a data attribute.
    });
  });
  process.env.NODE_ENV !== "production" && c.useEffect(() => {
    if (!be && !O && P !== "") {
      const C = ue.map((M) => M.props.value);
      console.warn([`MUI: You have provided an out-of-range value \`${P}\` for the select ${s ? `(name="${s}") ` : ""}component.`, "Consider providing a value that matches one of the available options or ''.", `The available values are ${C.filter((M) => M != null).map((M) => `\`${M}\``).join(", ") || '""'}.`].join(`
`));
    }
  }, [be, ue, O, s, P]), we && (O ? G.length === 0 ? pe = null : pe = G.reduce((C, M, H) => (C.push(M), H < G.length - 1 && C.push(", "), C), []) : pe = Se);
  let Fe = Ne;
  !a && le && re && (Fe = ce.clientWidth);
  let ye;
  typeof U < "u" ? ye = U : ye = h ? null : 0;
  const $e = W.id || (s ? `mui-component-select-${s}` : void 0), D = {
    ...t,
    variant: F,
    value: P,
    open: Ce,
    error: g
  }, Oe = Tn(D), xe = {
    ...p.PaperProps,
    ...(xo = p.slotProps) == null ? void 0 : xo.paper
  }, Oo = St();
  return /* @__PURE__ */ q(c.Fragment, {
    children: [/* @__PURE__ */ y(xn, {
      as: "div",
      ref: ro,
      tabIndex: ye,
      role: "combobox",
      "aria-controls": Oo,
      "aria-disabled": h ? "true" : void 0,
      "aria-expanded": Ce ? "true" : "false",
      "aria-haspopup": "listbox",
      "aria-label": i,
      "aria-labelledby": [w, $e].filter(Boolean).join(" ") || void 0,
      "aria-describedby": r,
      "aria-required": z ? "true" : void 0,
      "aria-invalid": g ? "true" : void 0,
      onKeyDown: so,
      onMouseDown: h || N ? null : A,
      onBlur: ao,
      onFocus: $,
      ...W,
      ownerState: D,
      className: Z(W.className, Oe.select, m),
      id: $e,
      children: kn(pe) ? (
        // notranslate needed while Google Translate will not fix zero-width space issue
        zo || (zo = /* @__PURE__ */ y("span", {
          className: "notranslate",
          "aria-hidden": !0,
          children: "​"
        }))
      ) : pe
    }), /* @__PURE__ */ y(Rn, {
      "aria-invalid": g,
      value: Array.isArray(P) ? P.join(",") : P,
      name: s,
      ref: ie,
      "aria-hidden": !0,
      onChange: Pe,
      tabIndex: -1,
      disabled: h,
      className: Oe.nativeInput,
      autoFocus: u,
      required: z,
      ...L,
      ownerState: D
    }), /* @__PURE__ */ y(In, {
      as: I,
      className: Oe.icon,
      ownerState: D
    }), /* @__PURE__ */ y(Qo, {
      id: `menu-${s || ""}`,
      anchorEl: ce,
      open: Ce,
      onClose: Y,
      anchorOrigin: {
        vertical: "bottom",
        horizontal: "center"
      },
      transformOrigin: {
        vertical: "top",
        horizontal: "center"
      },
      ...p,
      MenuListProps: {
        "aria-labelledby": w,
        role: "listbox",
        "aria-multiselectable": O ? "true" : void 0,
        disableListWrap: !0,
        id: Oo,
        ...p.MenuListProps
      },
      slotProps: {
        ...p.slotProps,
        paper: {
          ...xe,
          style: {
            minWidth: Fe,
            ...xe != null ? xe.style : null
          }
        }
      },
      children: _e
    })]
  });
});
process.env.NODE_ENV !== "production" && (st.propTypes = {
  /**
   * @ignore
   */
  "aria-describedby": e.string,
  /**
   * @ignore
   */
  "aria-label": e.string,
  /**
   * @ignore
   */
  autoFocus: e.bool,
  /**
   * If `true`, the width of the popover will automatically be set according to the items inside the
   * menu, otherwise it will be at least the width of the select input.
   */
  autoWidth: e.bool,
  /**
   * The option elements to populate the select with.
   * Can be some `<MenuItem>` elements.
   */
  children: e.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * The CSS class name of the select element.
   */
  className: e.string,
  /**
   * If `true`, the component is toggled on mount. Use when the component open state is not controlled.
   * You can only use it when the `native` prop is `false` (default).
   */
  defaultOpen: e.bool,
  /**
   * The default value. Use when the component is not controlled.
   */
  defaultValue: e.any,
  /**
   * If `true`, the select is disabled.
   */
  disabled: e.bool,
  /**
   * If `true`, the selected item is displayed even if its value is empty.
   */
  displayEmpty: e.bool,
  /**
   * If `true`, the `select input` will indicate an error.
   */
  error: e.bool,
  /**
   * The icon that displays the arrow.
   */
  IconComponent: e.elementType.isRequired,
  /**
   * Imperative handle implementing `{ value: T, node: HTMLElement, focus(): void }`
   * Equivalent to `ref`
   */
  inputRef: Me,
  /**
   * The ID of an element that acts as an additional label. The Select will
   * be labelled by the additional label and the selected value.
   */
  labelId: e.string,
  /**
   * Props applied to the [`Menu`](/material-ui/api/menu/) element.
   */
  MenuProps: e.object,
  /**
   * If `true`, `value` must be an array and the menu will support multiple selections.
   */
  multiple: e.bool,
  /**
   * Name attribute of the `select` or hidden `input` element.
   */
  name: e.string,
  /**
   * @ignore
   */
  onBlur: e.func,
  /**
   * Callback fired when a menu item is selected.
   *
   * @param {object} event The event source of the callback.
   * You can pull out the new value by accessing `event.target.value` (any).
   * @param {object} [child] The react element that was selected.
   */
  onChange: e.func,
  /**
   * Callback fired when the component requests to be closed.
   * Use in controlled mode (see open).
   *
   * @param {object} event The event source of the callback.
   */
  onClose: e.func,
  /**
   * @ignore
   */
  onFocus: e.func,
  /**
   * Callback fired when the component requests to be opened.
   * Use in controlled mode (see open).
   *
   * @param {object} event The event source of the callback.
   */
  onOpen: e.func,
  /**
   * If `true`, the component is shown.
   */
  open: e.bool,
  /**
   * @ignore
   */
  readOnly: e.bool,
  /**
   * Render the selected value.
   *
   * @param {any} value The `value` provided to the component.
   * @returns {ReactNode}
   */
  renderValue: e.func,
  /**
   * If `true`, the component is required.
   */
  required: e.bool,
  /**
   * Props applied to the clickable div element.
   */
  SelectDisplayProps: e.object,
  /**
   * @ignore
   */
  tabIndex: e.oneOfType([e.number, e.string]),
  /**
   * @ignore
   */
  type: e.any,
  /**
   * The input value.
   */
  value: e.any,
  /**
   * The variant to use.
   */
  variant: e.oneOf(["standard", "outlined", "filled"])
});
const Mn = (o) => {
  const {
    classes: t
  } = o;
  return t;
}, So = {
  name: "MuiSelect",
  overridesResolver: (o, t) => t.root,
  shouldForwardProp: (o) => ne(o) && o !== "variant",
  slot: "Root"
}, En = E(ho, So)(""), Nn = E(Co, So)(""), Pn = E(go, So)(""), wo = /* @__PURE__ */ c.forwardRef(function(t, n) {
  const r = ee({
    name: "MuiSelect",
    props: t
  }), {
    autoWidth: i = !1,
    children: u,
    classes: a = {},
    className: d,
    defaultOpen: m = !1,
    displayEmpty: v = !1,
    IconComponent: l = Nt,
    id: h,
    input: x,
    inputProps: g,
    label: I,
    labelId: S,
    MenuProps: w,
    multiple: p = !1,
    native: O = !1,
    onClose: s,
    onOpen: f,
    open: R,
    renderValue: T,
    SelectDisplayProps: $,
    variant: k = "outlined",
    ...j
  } = r, N = O ? nt : st, B = Ve(), z = He({
    props: r,
    muiFormControl: B,
    states: ["variant", "error"]
  }), W = z.variant || k, U = {
    ...r,
    variant: W,
    classes: a
  }, V = Mn(U), {
    root: b,
    ...F
  } = V, L = x || {
    standard: /* @__PURE__ */ y(En, {
      ownerState: U
    }),
    outlined: /* @__PURE__ */ y(Nn, {
      label: I,
      ownerState: U
    }),
    filled: /* @__PURE__ */ y(Pn, {
      ownerState: U
    })
  }[W], P = Ee(n, Ct(L));
  return /* @__PURE__ */ y(c.Fragment, {
    children: /* @__PURE__ */ c.cloneElement(L, {
      // Most of the logic is implemented in `SelectInput`.
      // The `Select` component is a simple API wrapper to expose something better to play with.
      inputComponent: N,
      inputProps: {
        children: u,
        error: z.error,
        IconComponent: l,
        variant: W,
        type: void 0,
        // We render a select. We can ignore the type provided by the `Input`.
        multiple: p,
        ...O ? {
          id: h
        } : {
          autoWidth: i,
          defaultOpen: m,
          displayEmpty: v,
          labelId: S,
          MenuProps: w,
          onClose: s,
          onOpen: f,
          open: R,
          renderValue: T,
          SelectDisplayProps: {
            id: h,
            ...$
          }
        },
        ...g,
        classes: g ? bo(F, g.classes) : F,
        ...x ? x.props.inputProps : {}
      },
      ...(p && O || v) && W === "outlined" ? {
        notched: !0
      } : {},
      ref: P,
      className: Z(L.props.className, d, V.root),
      // If a custom input is provided via 'input' prop, do not allow 'variant' to be propagated to it's root element. See https://github.com/mui/material-ui/issues/33894.
      ...!x && {
        variant: W
      },
      ...j
    })
  });
});
process.env.NODE_ENV !== "production" && (wo.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * If `true`, the width of the popover will automatically be set according to the items inside the
   * menu, otherwise it will be at least the width of the select input.
   * @default false
   */
  autoWidth: e.bool,
  /**
   * The option elements to populate the select with.
   * Can be some `MenuItem` when `native` is false and `option` when `native` is true.
   *
   * ⚠️The `MenuItem` elements **must** be direct descendants when `native` is false.
   */
  children: e.node,
  /**
   * Override or extend the styles applied to the component.
   * @default {}
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * If `true`, the component is initially open. Use when the component open state is not controlled (i.e. the `open` prop is not defined).
   * You can only use it when the `native` prop is `false` (default).
   * @default false
   */
  defaultOpen: e.bool,
  /**
   * The default value. Use when the component is not controlled.
   */
  defaultValue: e.any,
  /**
   * If `true`, a value is displayed even if no items are selected.
   *
   * In order to display a meaningful value, a function can be passed to the `renderValue` prop which
   * returns the value to be displayed when no items are selected.
   *
   * ⚠️ When using this prop, make sure the label doesn't overlap with the empty displayed value.
   * The label should either be hidden or forced to a shrunk state.
   * @default false
   */
  displayEmpty: e.bool,
  /**
   * The icon that displays the arrow.
   * @default ArrowDropDownIcon
   */
  IconComponent: e.elementType,
  /**
   * The `id` of the wrapper element or the `select` element when `native`.
   */
  id: e.string,
  /**
   * An `Input` element; does not have to be a material-ui specific `Input`.
   */
  input: e.element,
  /**
   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes) applied to the `input` element.
   * When `native` is `true`, the attributes are applied on the `select` element.
   */
  inputProps: e.object,
  /**
   * See [OutlinedInput#label](https://mui.com/material-ui/api/outlined-input/#props)
   */
  label: e.node,
  /**
   * The ID of an element that acts as an additional label. The Select will
   * be labelled by the additional label and the selected value.
   */
  labelId: e.string,
  /**
   * Props applied to the [`Menu`](https://mui.com/material-ui/api/menu/) element.
   */
  MenuProps: e.object,
  /**
   * If `true`, `value` must be an array and the menu will support multiple selections.
   * @default false
   */
  multiple: e.bool,
  /**
   * If `true`, the component uses a native `select` element.
   * @default false
   */
  native: e.bool,
  /**
   * Callback fired when a menu item is selected.
   *
   * @param {SelectChangeEvent<Value>} event The event source of the callback.
   * You can pull out the new value by accessing `event.target.value` (any).
   * **Warning**: This is a generic event, not a change event, unless the change event is caused by browser autofill.
   * @param {object} [child] The react element that was selected when `native` is `false` (default).
   */
  onChange: e.func,
  /**
   * Callback fired when the component requests to be closed.
   * Use it in either controlled (see the `open` prop), or uncontrolled mode (to detect when the Select collapses).
   *
   * @param {object} event The event source of the callback.
   */
  onClose: e.func,
  /**
   * Callback fired when the component requests to be opened.
   * Use it in either controlled (see the `open` prop), or uncontrolled mode (to detect when the Select expands).
   *
   * @param {object} event The event source of the callback.
   */
  onOpen: e.func,
  /**
   * If `true`, the component is shown.
   * You can only use it when the `native` prop is `false` (default).
   */
  open: e.bool,
  /**
   * Render the selected value.
   * You can only use it when the `native` prop is `false` (default).
   *
   * @param {any} value The `value` provided to the component.
   * @returns {ReactNode}
   */
  renderValue: e.func,
  /**
   * Props applied to the clickable div element.
   */
  SelectDisplayProps: e.object,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * The `input` value. Providing an empty string will select no options.
   * Set to an empty string `''` if you don't want any of the available options to be selected.
   *
   * If the value is an object it must have reference equality with the option in order to be selected.
   * If the value is not an object, the string representation must match with the string representation of the option in order to be selected.
   */
  value: e.oneOfType([e.oneOf([""]), e.any]),
  /**
   * The variant to use.
   * @default 'outlined'
   */
  variant: e.oneOf(["filled", "outlined", "standard"])
});
wo.muiName = "Select";
function or({
  description: o = "Connect your Slack workspace to receive notifications directly in Slack.",
  connectButtonText: t = "Connect Slack",
  editButtonText: n = "Edit Channel",
  disconnectButtonText: r = "Disconnect",
  saveButtonText: i = "Save",
  cancelButtonText: u = "Cancel",
  connectedText: a = "Slack notifications will be sent to:",
  selectChannelText: d = "Choose a channel or user to receive notifications:",
  destinationType: m = "any"
} = {}) {
  const v = at(it), l = v == null ? void 0 : v.getClient(), [h, x] = ge(), [g, I] = ge(), [S, w] = ge([]), [p, O] = ge(""), [s, f] = ge(!1), [R, T] = ge(null), [$, k] = ge(!1), j = Ro(async () => {
    if (l)
      try {
        f(!0);
        const b = await l.user.get();
        b.slackToken && x(b.slackToken), b.slackChannel && I(b.slackChannel);
      } catch (b) {
        console.error("Error fetching Slack status:", b);
      } finally {
        f(!1);
      }
  }, [l]), N = Ro(async () => {
    if (!l || !h) return [];
    try {
      f(!0), T(null);
      const b = await l.slack.getChannels(), F = [
        ...(b.channels || []).filter((L) => L.id && L.name).map((L) => ({
          id: L.id,
          name: L.name,
          type: "channel"
        })),
        ...(b.users || []).filter((L) => L.id && L.name).map((L) => ({
          id: L.id,
          name: L.name,
          type: "user"
        }))
      ];
      return w(F), F;
    } catch (b) {
      return console.error("Error loading channels and users:", b), T("Failed to load Slack channels and users. Please try again."), [];
    } finally {
      f(!1);
    }
  }, [l, h]);
  ko(() => {
    j();
  }, [j]), ko(() => {
    h && !g && !$ && (m === "me" ? (async () => {
      try {
        f(!0), T(null);
        const F = await (l == null ? void 0 : l.slack.getChannels()), L = F == null ? void 0 : F.me;
        if (L && L.name && l) {
          const P = `@${L.name}`;
          await l.slack.setChannel(P), I(P);
        } else
          T(
            "Unable to automatically set direct message. Please try again."
          );
      } catch (F) {
        console.error("Error setting direct message:", F), T("Failed to set direct message. Please try again.");
      } finally {
        f(!1);
      }
    })() : N());
  }, [
    h,
    g,
    $,
    N,
    m,
    l
  ]);
  const B = async () => {
    if (l)
      try {
        f(!0), T(null);
        const b = await l.slack.getOAuthUrl();
        window.location.href = b;
      } catch (b) {
        console.error("Error connecting to Slack:", b), T("Failed to connect to Slack. Please try again.");
      } finally {
        f(!1);
      }
  }, z = async () => {
    if (!(!l || !p))
      try {
        f(!0), T(null);
        const b = S.find((L) => L.id === p);
        if (!b) {
          T("Channel not found. Please try again.");
          return;
        }
        const F = `${b.type === "channel" ? "#" : "@"}${b.name}`;
        await l.slack.setChannel(F), I(F), k(!1), T(null);
      } catch (b) {
        console.error("Error saving channel:", b), T("Failed to save channel. Please try again.");
      } finally {
        f(!1);
      }
  }, W = async () => {
    if (l)
      try {
        f(!0), T(null), await l.identify({
          // @ts-expect-error - null is not assignable to type string
          slackToken: null,
          // @ts-expect-error - null is not assignable to type string
          slackChannel: null
        }), x(void 0), I(void 0), O(""), w([]), k(!1);
      } catch (b) {
        console.error("Error disconnecting Slack:", b), T("Failed to disconnect Slack. Please try again.");
      } finally {
        f(!1);
      }
  }, U = async () => {
    k(!0);
    let b = S;
    if (S.length === 0 && (b = await N()), g) {
      const F = g.startsWith("#"), L = g.substring(1), P = F ? "channel" : "user", ve = b.find(
        (ae) => ae.name === L && ae.type === P
      );
      ve && O(ve.id);
    }
  }, V = () => {
    k(!1), O("");
  };
  return l ? s && !h && !S.length ? /* @__PURE__ */ y(Re, { display: "flex", justifyContent: "center", alignItems: "center", p: 3, children: /* @__PURE__ */ y(Be, {}) }) : h ? m === "me" && h && !g ? /* @__PURE__ */ q(Re, { children: [
    R && /* @__PURE__ */ y(io, { severity: "error", sx: { mb: 2 }, children: R }),
    /* @__PURE__ */ y(Re, { display: "flex", justifyContent: "center", alignItems: "center", p: 3, children: /* @__PURE__ */ y(Be, {}) })
  ] }) : (!g || $) && m === "any" ? /* @__PURE__ */ q(Re, { children: [
    R && /* @__PURE__ */ y(io, { severity: "error", sx: { mb: 2 }, children: R }),
    s ? /* @__PURE__ */ y(Re, { display: "flex", justifyContent: "center", p: 2, children: /* @__PURE__ */ y(Be, {}) }) : /* @__PURE__ */ q(lo, { direction: "row", spacing: 2, alignItems: "center", children: [
      /* @__PURE__ */ y(Ge, { variant: "body2", color: "text.secondary", children: d }),
      /* @__PURE__ */ q(Go, { sx: { minWidth: 200 }, size: "small", children: [
        /* @__PURE__ */ y(Yo, { id: "slack-channel-label", children: "Channel or User" }),
        /* @__PURE__ */ y(
          wo,
          {
            labelId: "slack-channel-label",
            value: p,
            label: "Channel or User",
            onChange: (b) => O(b.target.value),
            children: S.sort((b, F) => b.type === F.type ? b.name.localeCompare(F.name) : b.type === "channel" ? -1 : 1).map((b) => /* @__PURE__ */ q(et, { value: b.id, children: [
              b.type === "channel" ? "#" : "@",
              b.name
            ] }, b.id))
          }
        )
      ] }),
      /* @__PURE__ */ y(
        ke,
        {
          variant: "contained",
          color: "primary",
          onClick: z,
          disabled: !p || s,
          children: i
        }
      ),
      $ && /* @__PURE__ */ y(
        ke,
        {
          variant: "outlined",
          onClick: V,
          disabled: s,
          children: u
        }
      ),
      /* @__PURE__ */ y(
        ke,
        {
          variant: "text",
          color: "error",
          onClick: W,
          disabled: s,
          size: "small",
          children: r
        }
      )
    ] })
  ] }) : /* @__PURE__ */ q(lo, { direction: "row", spacing: 2, alignItems: "center", children: [
    /* @__PURE__ */ y(Ge, { variant: "body2", color: "text.secondary", children: a }),
    /* @__PURE__ */ y(Ge, { variant: "body1", fontWeight: "medium", children: g }),
    m === "any" && /* @__PURE__ */ y(ke, { variant: "outlined", onClick: U, disabled: s, children: n }),
    /* @__PURE__ */ y(
      ke,
      {
        variant: "text",
        color: "error",
        onClick: W,
        disabled: s,
        children: r
      }
    )
  ] }) : /* @__PURE__ */ q(Re, { children: [
    R && /* @__PURE__ */ y(io, { severity: "error", sx: { mb: 2 }, children: R }),
    /* @__PURE__ */ q(lo, { direction: "row", spacing: 2, alignItems: "center", children: [
      /* @__PURE__ */ y(Ge, { variant: "body2", color: "text.secondary", children: o }),
      /* @__PURE__ */ y(
        ke,
        {
          variant: "contained",
          color: "primary",
          onClick: B,
          disabled: s,
          children: s ? /* @__PURE__ */ y(Be, { size: 24 }) : t
        }
      )
    ] })
  ] }) : null;
}
export {
  or as SlackConnect
};
