// @notificationapi/react/dist/components/Preferences/NotificationPreferencesInline.js
import { jsxs as o, jsx as t } from "react/jsx-runtime";
import { useContext as n } from "react";
import { Preferences as s } from "./Preferences.js";
import { NotificationAPIContext as m } from "../Provider/context.js";
import { W as f } from "../../assets/WebPushOptInMessage.js";
import { getThemeColors as l } from "../../utils/theme.js";
import { u as a } from "../../assets/ButtonBase.js";
import { D as p } from "../../assets/Divider.js";
function y(c) {
  const r = n(m), i = a(), e = l(i);
  return r ? /* @__PURE__ */ o(
    "div",
    {
      style: {
        borderRadius: 8,
        border: `1px solid ${e.border}`,
        backgroundColor: e.paper,
        color: e.text
      },
      children: [
        /* @__PURE__ */ t(s, {}),
        " ",
        r.webPushOptInMessage && /* @__PURE__ */ o("div", { children: [
          /* @__PURE__ */ t(
            p,
            {
              style: { margin: "10px 0", borderColor: e.divider }
            }
          ),
          /* @__PURE__ */ t(
            f,
            {
              hideAfterInteraction: !1,
              descriptionStyle: {
                flexDirection: "column",
                // Stack the elements vertically
                justifyContent: "flex-start",
                // Align items to the left
                fontSize: "14px",
                alignItems: "flex-start"
                // Align items to the left
              },
              buttonContainerStyle: {
                justifyContent: "flex-start",
                // Align buttons to the left
                alignItems: "flex-start",
                // Align buttons to the left
                marginTop: "10px"
                // Add some space between message and buttons
              }
            }
          )
        ] })
      ]
    }
  ) : null;
}
export {
  y as NotificationPreferencesInline
};
