// @notificationapi/react/dist/components/Preferences/NotificationPreferencesPopup.js
import { jsx as i, jsxs as $ } from "react/jsx-runtime";
import * as c from "react";
import { useContext as eo } from "react";
import { Preferences as ro } from "./Preferences.js";
import { NotificationAPIContext as to } from "../Provider/context.js";
import { W as ao } from "../../assets/WebPushOptInMessage.js";
import { getThemeColors as so } from "../../utils/theme.js";
import { u as j } from "../../assets/ButtonBase.js";
import { g as k, e as u, P as o } from "../../assets/createTheme.js";
import { g as T, s as f, m as E, u as S, c as g, a as P } from "../../assets/DefaultPropsProvider.js";
import { B as io, M as no, F as lo } from "../../assets/Modal.js";
import { P as F } from "../../assets/Paper.js";
import { u as po } from "../../assets/dividerClasses.js";
import { T as co } from "../../assets/Typography.js";
function uo(r) {
  return k("MuiDialog", r);
}
const D = T("MuiDialog", ["root", "scrollPaper", "scrollBody", "container", "paper", "paperScrollPaper", "paperScrollBody", "paperWidthFalse", "paperWidthXs", "paperWidthSm", "paperWidthMd", "paperWidthLg", "paperWidthXl", "paperFullWidth", "paperFullScreen"]), w = /* @__PURE__ */ c.createContext({});
process.env.NODE_ENV !== "production" && (w.displayName = "DialogContext");
const fo = f(io, {
  name: "MuiDialog",
  slot: "Backdrop",
  overrides: (r, e) => e.backdrop
})({
  // Improve scrollable dialog support.
  zIndex: -1
}), mo = (r) => {
  const {
    classes: e,
    scroll: t,
    maxWidth: a,
    fullWidth: s,
    fullScreen: n
  } = r, l = {
    root: ["root"],
    container: ["container", `scroll${u(t)}`],
    paper: ["paper", `paperScroll${u(t)}`, `paperWidth${u(String(a))}`, s && "paperFullWidth", n && "paperFullScreen"]
  };
  return P(l, uo, e);
}, go = f(no, {
  name: "MuiDialog",
  slot: "Root",
  overridesResolver: (r, e) => e.root
})({
  "@media print": {
    // Use !important to override the Modal inline-style.
    position: "absolute !important"
  }
}), bo = f("div", {
  name: "MuiDialog",
  slot: "Container",
  overridesResolver: (r, e) => {
    const {
      ownerState: t
    } = r;
    return [e.container, e[`scroll${u(t.scroll)}`]];
  }
})({
  height: "100%",
  "@media print": {
    height: "auto"
  },
  // We disable the focus ring for mouse, touch and keyboard users.
  outline: 0,
  variants: [{
    props: {
      scroll: "paper"
    },
    style: {
      display: "flex",
      justifyContent: "center",
      alignItems: "center"
    }
  }, {
    props: {
      scroll: "body"
    },
    style: {
      overflowY: "auto",
      overflowX: "hidden",
      textAlign: "center",
      "&::after": {
        content: '""',
        display: "inline-block",
        verticalAlign: "middle",
        height: "100%",
        width: "0"
      }
    }
  }]
}), xo = f(F, {
  name: "MuiDialog",
  slot: "Paper",
  overridesResolver: (r, e) => {
    const {
      ownerState: t
    } = r;
    return [e.paper, e[`scrollPaper${u(t.scroll)}`], e[`paperWidth${u(String(t.maxWidth))}`], t.fullWidth && e.paperFullWidth, t.fullScreen && e.paperFullScreen];
  }
})(E(({
  theme: r
}) => ({
  margin: 32,
  position: "relative",
  overflowY: "auto",
  "@media print": {
    overflowY: "visible",
    boxShadow: "none"
  },
  variants: [{
    props: {
      scroll: "paper"
    },
    style: {
      display: "flex",
      flexDirection: "column",
      maxHeight: "calc(100% - 64px)"
    }
  }, {
    props: {
      scroll: "body"
    },
    style: {
      display: "inline-block",
      verticalAlign: "middle",
      textAlign: "initial"
    }
  }, {
    props: ({
      ownerState: e
    }) => !e.maxWidth,
    style: {
      maxWidth: "calc(100% - 64px)"
    }
  }, {
    props: {
      maxWidth: "xs"
    },
    style: {
      maxWidth: r.breakpoints.unit === "px" ? Math.max(r.breakpoints.values.xs, 444) : `max(${r.breakpoints.values.xs}${r.breakpoints.unit}, 444px)`,
      [`&.${D.paperScrollBody}`]: {
        [r.breakpoints.down(Math.max(r.breakpoints.values.xs, 444) + 32 * 2)]: {
          maxWidth: "calc(100% - 64px)"
        }
      }
    }
  }, ...Object.keys(r.breakpoints.values).filter((e) => e !== "xs").map((e) => ({
    props: {
      maxWidth: e
    },
    style: {
      maxWidth: `${r.breakpoints.values[e]}${r.breakpoints.unit}`,
      [`&.${D.paperScrollBody}`]: {
        [r.breakpoints.down(r.breakpoints.values[e] + 32 * 2)]: {
          maxWidth: "calc(100% - 64px)"
        }
      }
    }
  })), {
    props: ({
      ownerState: e
    }) => e.fullWidth,
    style: {
      width: "calc(100% - 64px)"
    }
  }, {
    props: ({
      ownerState: e
    }) => e.fullScreen,
    style: {
      margin: 0,
      width: "100%",
      maxWidth: "100%",
      height: "100%",
      maxHeight: "none",
      borderRadius: 0,
      [`&.${D.paperScrollBody}`]: {
        margin: 0,
        maxWidth: "100%"
      }
    }
  }]
}))), I = /* @__PURE__ */ c.forwardRef(function(e, t) {
  const a = S({
    props: e,
    name: "MuiDialog"
  }), s = j(), n = {
    enter: s.transitions.duration.enteringScreen,
    exit: s.transitions.duration.leavingScreen
  }, {
    "aria-describedby": l,
    "aria-labelledby": p,
    "aria-modal": m = !0,
    BackdropComponent: x,
    BackdropProps: V,
    children: Y,
    className: _,
    disableEscapeKeyDown: W = !1,
    fullScreen: z = !1,
    fullWidth: K = !1,
    maxWidth: L = "sm",
    onBackdropClick: M,
    onClick: N,
    onClose: h,
    open: O,
    PaperComponent: X = F,
    PaperProps: R = {},
    scroll: H = "paper",
    TransitionComponent: q = lo,
    transitionDuration: B = n,
    TransitionProps: G,
    ...J
  } = a, b = {
    ...a,
    disableEscapeKeyDown: W,
    fullScreen: z,
    fullWidth: K,
    maxWidth: L,
    scroll: H
  }, y = mo(b), v = c.useRef(), Q = (d) => {
    v.current = d.target === d.currentTarget;
  }, Z = (d) => {
    N && N(d), v.current && (v.current = null, M && M(d), h && h(d, "backdropClick"));
  }, C = po(p), oo = c.useMemo(() => ({
    titleId: C
  }), [C]);
  return /* @__PURE__ */ i(go, {
    className: g(y.root, _),
    closeAfterTransition: !0,
    components: {
      Backdrop: fo
    },
    componentsProps: {
      backdrop: {
        transitionDuration: B,
        as: x,
        ...V
      }
    },
    disableEscapeKeyDown: W,
    onClose: h,
    open: O,
    ref: t,
    onClick: Z,
    ownerState: b,
    ...J,
    children: /* @__PURE__ */ i(q, {
      appear: !0,
      in: O,
      timeout: B,
      role: "presentation",
      ...G,
      children: /* @__PURE__ */ i(bo, {
        className: g(y.container),
        onMouseDown: Q,
        ownerState: b,
        children: /* @__PURE__ */ i(xo, {
          as: X,
          elevation: 24,
          role: "dialog",
          "aria-describedby": l,
          "aria-labelledby": C,
          "aria-modal": m,
          ...R,
          className: g(y.paper, R.className),
          ownerState: b,
          children: /* @__PURE__ */ i(w.Provider, {
            value: oo,
            children: Y
          })
        })
      })
    })
  });
});
process.env.NODE_ENV !== "production" && (I.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The id(s) of the element(s) that describe the dialog.
   */
  "aria-describedby": o.string,
  /**
   * The id(s) of the element(s) that label the dialog.
   */
  "aria-labelledby": o.string,
  /**
   * Informs assistive technologies that the element is modal.
   * It's added on the element with role="dialog".
   * @default true
   */
  "aria-modal": o.oneOfType([o.oneOf(["false", "true"]), o.bool]),
  /**
   * A backdrop component. This prop enables custom backdrop rendering.
   * @deprecated Use `slots.backdrop` instead. While this prop currently works, it will be removed in the next major version.
   * Use the `slots.backdrop` prop to make your application ready for the next version of Material UI.
   * @default styled(Backdrop, {
   *   name: 'MuiModal',
   *   slot: 'Backdrop',
   *   overridesResolver: (props, styles) => {
   *     return styles.backdrop;
   *   },
   * })({
   *   zIndex: -1,
   * })
   */
  BackdropComponent: o.elementType,
  /**
   * @ignore
   */
  BackdropProps: o.object,
  /**
   * Dialog children, usually the included sub-components.
   */
  children: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * If `true`, hitting escape will not fire the `onClose` callback.
   * @default false
   */
  disableEscapeKeyDown: o.bool,
  /**
   * If `true`, the dialog is full-screen.
   * @default false
   */
  fullScreen: o.bool,
  /**
   * If `true`, the dialog stretches to `maxWidth`.
   *
   * Notice that the dialog width grow is limited by the default margin.
   * @default false
   */
  fullWidth: o.bool,
  /**
   * Determine the max-width of the dialog.
   * The dialog width grows with the size of the screen.
   * Set to `false` to disable `maxWidth`.
   * @default 'sm'
   */
  maxWidth: o.oneOfType([o.oneOf(["xs", "sm", "md", "lg", "xl", !1]), o.string]),
  /**
   * Callback fired when the backdrop is clicked.
   * @deprecated Use the `onClose` prop with the `reason` argument to handle the `backdropClick` events.
   */
  onBackdropClick: o.func,
  /**
   * @ignore
   */
  onClick: o.func,
  /**
   * Callback fired when the component requests to be closed.
   *
   * @param {object} event The event source of the callback.
   * @param {string} reason Can be: `"escapeKeyDown"`, `"backdropClick"`.
   */
  onClose: o.func,
  /**
   * If `true`, the component is shown.
   */
  open: o.bool.isRequired,
  /**
   * The component used to render the body of the dialog.
   * @default Paper
   */
  PaperComponent: o.elementType,
  /**
   * Props applied to the [`Paper`](https://mui.com/material-ui/api/paper/) element.
   * @default {}
   */
  PaperProps: o.object,
  /**
   * Determine the container for scrolling the dialog.
   * @default 'paper'
   */
  scroll: o.oneOf(["body", "paper"]),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object]),
  /**
   * The component used for the transition.
   * [Follow this guide](https://mui.com/material-ui/transitions/#transitioncomponent-prop) to learn more about the requirements for this component.
   * @default Fade
   */
  TransitionComponent: o.elementType,
  /**
   * The duration for the transition, in milliseconds.
   * You may specify a single timeout for all transitions, or individually with an object.
   * @default {
   *   enter: theme.transitions.duration.enteringScreen,
   *   exit: theme.transitions.duration.leavingScreen,
   * }
   */
  transitionDuration: o.oneOfType([o.number, o.shape({
    appear: o.number,
    enter: o.number,
    exit: o.number
  })]),
  /**
   * Props applied to the transition element.
   * By default, the element is based on this [`Transition`](https://reactcommunity.org/react-transition-group/transition/) component.
   */
  TransitionProps: o.object
});
function ho(r) {
  return k("MuiDialogContent", r);
}
T("MuiDialogContent", ["root", "dividers"]);
function yo(r) {
  return k("MuiDialogTitle", r);
}
const vo = T("MuiDialogTitle", ["root"]), Co = (r) => {
  const {
    classes: e,
    dividers: t
  } = r;
  return P({
    root: ["root", t && "dividers"]
  }, ho, e);
}, Do = f("div", {
  name: "MuiDialogContent",
  slot: "Root",
  overridesResolver: (r, e) => {
    const {
      ownerState: t
    } = r;
    return [e.root, t.dividers && e.dividers];
  }
})(E(({
  theme: r
}) => ({
  flex: "1 1 auto",
  // Add iOS momentum scrolling for iOS < 13.0
  WebkitOverflowScrolling: "touch",
  overflowY: "auto",
  padding: "20px 24px",
  variants: [{
    props: ({
      ownerState: e
    }) => e.dividers,
    style: {
      padding: "16px 24px",
      borderTop: `1px solid ${(r.vars || r).palette.divider}`,
      borderBottom: `1px solid ${(r.vars || r).palette.divider}`
    }
  }, {
    props: ({
      ownerState: e
    }) => !e.dividers,
    style: {
      [`.${vo.root} + &`]: {
        paddingTop: 0
      }
    }
  }]
}))), U = /* @__PURE__ */ c.forwardRef(function(e, t) {
  const a = S({
    props: e,
    name: "MuiDialogContent"
  }), {
    className: s,
    dividers: n = !1,
    ...l
  } = a, p = {
    ...a,
    dividers: n
  }, m = Co(p);
  return /* @__PURE__ */ i(Do, {
    className: g(m.root, s),
    ownerState: p,
    ref: t,
    ...l
  });
});
process.env.NODE_ENV !== "production" && (U.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * Display the top and bottom dividers.
   * @default false
   */
  dividers: o.bool,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object])
});
const ko = (r) => {
  const {
    classes: e
  } = r;
  return P({
    root: ["root"]
  }, yo, e);
}, To = f(co, {
  name: "MuiDialogTitle",
  slot: "Root",
  overridesResolver: (r, e) => e.root
})({
  padding: "16px 24px",
  flex: "0 0 auto"
}), A = /* @__PURE__ */ c.forwardRef(function(e, t) {
  const a = S({
    props: e,
    name: "MuiDialogTitle"
  }), {
    className: s,
    id: n,
    ...l
  } = a, p = a, m = ko(p), {
    titleId: x = n
  } = c.useContext(w);
  return /* @__PURE__ */ i(To, {
    component: "h2",
    className: g(m.root, s),
    ownerState: p,
    ref: t,
    variant: "h6",
    id: n ?? x,
    ...l
  });
});
process.env.NODE_ENV !== "production" && (A.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * @ignore
   */
  id: o.string,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object])
});
function Io(r) {
  const e = eo(to), t = j(), a = so(t);
  if (!e)
    return null;
  const s = {
    open: r.open === void 0 ? !0 : r.open,
    onClose: r.onClose || (() => {
    }),
    collapse: r.collapse === void 0 ? !1 : r.collapse
  };
  return /* @__PURE__ */ $(
    I,
    {
      open: s.open,
      onClose: s.onClose,
      maxWidth: "sm",
      fullWidth: !0,
      PaperProps: {
        style: {
          borderRadius: 8,
          backgroundColor: a.paper,
          color: a.text
        }
      },
      "aria-hidden": !s.open,
      scroll: "body",
      children: [
        /* @__PURE__ */ i(A, { sx: { color: a.text }, children: "Notification Preferences" }),
        /* @__PURE__ */ $(
          U,
          {
            sx: { backgroundColor: a.paper, color: a.text },
            children: [
              /* @__PURE__ */ i(ro, {}),
              e.webPushOptInMessage && /* @__PURE__ */ i("div", { children: /* @__PURE__ */ i(
                ao,
                {
                  hideAfterInteraction: !1,
                  descriptionStyle: { fontSize: 12 }
                }
              ) })
            ]
          }
        )
      ]
    }
  );
}
export {
  Io as NotificationPreferencesPopup
};
