// @notificationapi/react/dist/components/Preferences/PreferenceInput.js
import { jsxs as x, jsx as n, Fragment as ne } from "react/jsx-runtime";
import { g as we, a as Ce } from "../../assets/channelUtils.js";
import { getThemeColors as ke } from "../../utils/theme.js";
import { B as Se, r as W, b as Re, u as xe } from "../../assets/ButtonBase.js";
import { T as _ } from "../../assets/Typography.js";
import * as v from "react";
import { g as L, P as o, e as w, a as A, o as Oe, n as $e } from "../../assets/createTheme.js";
import { g as q, s as C, r as ee, c as j, a as D, m as I, u as H } from "../../assets/DefaultPropsProvider.js";
import { c as Q } from "../../assets/createSimplePaletteValueFilter.js";
import { u as X, f as ue, c as Pe, S as Te } from "../../assets/Stack.js";
import { a as fe, u as Fe } from "../../assets/dividerClasses.js";
import { u as Ne } from "../../assets/resolveComponentProps.js";
import { c as he } from "../../assets/createSvgIcon.js";
import { D as Be } from "../../assets/Divider.js";
function Ie(e) {
  return L("PrivateSwitchBase", e);
}
q("PrivateSwitchBase", ["root", "checked", "disabled", "input", "edgeStart", "edgeEnd"]);
const ze = (e) => {
  const {
    classes: t,
    checked: r,
    disabled: s,
    edge: a
  } = e, c = {
    root: ["root", r && "checked", s && "disabled", a && `edge${w(a)}`],
    input: ["input"]
  };
  return D(c, Ie, t);
}, je = C(Se)({
  padding: 9,
  borderRadius: "50%",
  variants: [{
    props: {
      edge: "start",
      size: "small"
    },
    style: {
      marginLeft: -3
    }
  }, {
    props: ({
      edge: e,
      ownerState: t
    }) => e === "start" && t.size !== "small",
    style: {
      marginLeft: -12
    }
  }, {
    props: {
      edge: "end",
      size: "small"
    },
    style: {
      marginRight: -3
    }
  }, {
    props: ({
      edge: e,
      ownerState: t
    }) => e === "end" && t.size !== "small",
    style: {
      marginRight: -12
    }
  }]
}), Me = C("input", {
  shouldForwardProp: ee
})({
  cursor: "inherit",
  position: "absolute",
  opacity: 0,
  width: "100%",
  height: "100%",
  top: 0,
  left: 0,
  margin: 0,
  padding: 0,
  zIndex: 1
}), oe = /* @__PURE__ */ v.forwardRef(function(t, r) {
  const {
    autoFocus: s,
    checked: a,
    checkedIcon: c,
    className: i,
    defaultChecked: u,
    disabled: d,
    disableFocusRipple: m = !1,
    edge: f = !1,
    icon: p,
    id: b,
    inputProps: l,
    inputRef: R,
    name: k,
    onBlur: h,
    onChange: T,
    onFocus: g,
    readOnly: F,
    required: O = !1,
    tabIndex: M,
    type: S,
    value: G,
    ...K
  } = t, [z, U] = fe({
    controlled: a,
    default: !!u,
    name: "SwitchBase",
    state: "checked"
  }), $ = X(), Z = (N) => {
    g && g(N), $ && $.onFocus && $.onFocus(N);
  }, E = (N) => {
    h && h(N), $ && $.onBlur && $.onBlur(N);
  }, B = (N) => {
    if (N.nativeEvent.defaultPrevented)
      return;
    const ae = N.target.checked;
    U(ae), T && T(N, ae);
  };
  let P = d;
  $ && typeof P > "u" && (P = $.disabled);
  const ve = S === "checkbox" || S === "radio", J = {
    ...t,
    checked: z,
    disabled: P,
    disableFocusRipple: m,
    edge: f
  }, se = ze(J);
  return /* @__PURE__ */ x(je, {
    component: "span",
    className: j(se.root, i),
    centerRipple: !0,
    focusRipple: !m,
    disabled: P,
    tabIndex: null,
    role: void 0,
    onFocus: Z,
    onBlur: E,
    ownerState: J,
    ref: r,
    ...K,
    children: [/* @__PURE__ */ n(Me, {
      autoFocus: s,
      checked: a,
      defaultChecked: u,
      className: se.input,
      disabled: P,
      id: ve ? b : void 0,
      name: k,
      onChange: B,
      readOnly: F,
      ref: R,
      required: O,
      ownerState: J,
      tabIndex: M,
      type: S,
      ...S === "checkbox" && G === void 0 ? {} : {
        value: G
      },
      ...l
    }), z ? c : p]
  });
});
process.env.NODE_ENV !== "production" && (oe.propTypes = {
  /**
   * If `true`, the `input` element is focused during the first mount.
   */
  autoFocus: o.bool,
  /**
   * If `true`, the component is checked.
   */
  checked: o.bool,
  /**
   * The icon to display when the component is checked.
   */
  checkedIcon: o.node.isRequired,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * @ignore
   */
  defaultChecked: o.bool,
  /**
   * If `true`, the component is disabled.
   */
  disabled: o.bool,
  /**
   * If `true`, the  keyboard focus ripple is disabled.
   * @default false
   */
  disableFocusRipple: o.bool,
  /**
   * If given, uses a negative margin to counteract the padding on one
   * side (this is often helpful for aligning the left or right
   * side of the icon with content above or below, without ruining the border
   * size and shape).
   * @default false
   */
  edge: o.oneOf(["end", "start", !1]),
  /**
   * The icon to display when the component is unchecked.
   */
  icon: o.node.isRequired,
  /**
   * The id of the `input` element.
   */
  id: o.string,
  /**
   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes) applied to the `input` element.
   */
  inputProps: o.object,
  /**
   * Pass a ref to the `input` element.
   */
  inputRef: W,
  /*
   * @ignore
   */
  name: o.string,
  /**
   * @ignore
   */
  onBlur: o.func,
  /**
   * Callback fired when the state is changed.
   *
   * @param {object} event The event source of the callback.
   * You can pull out the new checked state by accessing `event.target.checked` (boolean).
   */
  onChange: o.func,
  /**
   * @ignore
   */
  onFocus: o.func,
  /**
   * It prevents the user from changing the value of the field
   * (not from interacting with the field).
   */
  readOnly: o.bool,
  /**
   * If `true`, the `input` element is required.
   */
  required: o.bool,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.object,
  /**
   * @ignore
   */
  tabIndex: o.oneOfType([o.number, o.string]),
  /**
   * The input component prop `type`.
   */
  type: o.string.isRequired,
  /**
   * The value of the component.
   */
  value: o.any
});
function Ee(e) {
  return L("MuiFormControlLabel", e);
}
const V = q("MuiFormControlLabel", ["root", "labelPlacementStart", "labelPlacementTop", "labelPlacementBottom", "disabled", "label", "error", "required", "asterisk"]), Le = (e) => {
  const {
    classes: t,
    disabled: r,
    labelPlacement: s,
    error: a,
    required: c
  } = e, i = {
    root: ["root", r && "disabled", `labelPlacement${w(s)}`, a && "error", c && "required"],
    label: ["label", r && "disabled"],
    asterisk: ["asterisk", a && "error"]
  };
  return D(i, Ee, t);
}, qe = C("label", {
  name: "MuiFormControlLabel",
  slot: "Root",
  overridesResolver: (e, t) => {
    const {
      ownerState: r
    } = e;
    return [{
      [`& .${V.label}`]: t.label
    }, t.root, t[`labelPlacement${w(r.labelPlacement)}`]];
  }
})(I(({
  theme: e
}) => ({
  display: "inline-flex",
  alignItems: "center",
  cursor: "pointer",
  // For correct alignment with the text.
  verticalAlign: "middle",
  WebkitTapHighlightColor: "transparent",
  marginLeft: -11,
  marginRight: 16,
  // used for row presentation of radio/checkbox
  [`&.${V.disabled}`]: {
    cursor: "default"
  },
  [`& .${V.label}`]: {
    [`&.${V.disabled}`]: {
      color: (e.vars || e).palette.text.disabled
    }
  },
  variants: [{
    props: {
      labelPlacement: "start"
    },
    style: {
      flexDirection: "row-reverse",
      marginRight: -11
    }
  }, {
    props: {
      labelPlacement: "top"
    },
    style: {
      flexDirection: "column-reverse"
    }
  }, {
    props: {
      labelPlacement: "bottom"
    },
    style: {
      flexDirection: "column"
    }
  }, {
    props: ({
      labelPlacement: t
    }) => t === "start" || t === "top" || t === "bottom",
    style: {
      marginLeft: 16
      // used for row presentation of radio/checkbox
    }
  }]
}))), De = C("span", {
  name: "MuiFormControlLabel",
  slot: "Asterisk",
  overridesResolver: (e, t) => t.asterisk
})(I(({
  theme: e
}) => ({
  [`&.${V.error}`]: {
    color: (e.vars || e).palette.error.main
  }
}))), me = /* @__PURE__ */ v.forwardRef(function(t, r) {
  const s = H({
    props: t,
    name: "MuiFormControlLabel"
  }), {
    checked: a,
    className: c,
    componentsProps: i = {},
    control: u,
    disabled: d,
    disableTypography: m,
    inputRef: f,
    label: p,
    labelPlacement: b = "end",
    name: l,
    onChange: R,
    required: k,
    slots: h = {},
    slotProps: T = {},
    value: g,
    ...F
  } = s, O = X(), M = d ?? u.props.disabled ?? (O == null ? void 0 : O.disabled), S = k ?? u.props.required, G = {
    disabled: M,
    required: S
  };
  ["checked", "name", "onChange", "value", "inputRef"].forEach((P) => {
    typeof u.props[P] > "u" && typeof s[P] < "u" && (G[P] = s[P]);
  });
  const K = ue({
    props: s,
    muiFormControl: O,
    states: ["error"]
  }), z = {
    ...s,
    disabled: M,
    labelPlacement: b,
    required: S,
    error: K.error
  }, U = Le(z), $ = {
    slots: h,
    slotProps: {
      ...i,
      ...T
    }
  }, [Z, E] = Re("typography", {
    elementType: _,
    externalForwardedProps: $,
    ownerState: z
  });
  let B = p;
  return B != null && B.type !== _ && !m && (B = /* @__PURE__ */ n(Z, {
    component: "span",
    ...E,
    className: j(U.label, E == null ? void 0 : E.className),
    children: B
  })), /* @__PURE__ */ x(qe, {
    className: j(U.root, c),
    ownerState: z,
    ref: r,
    ...F,
    children: [/* @__PURE__ */ v.cloneElement(u, G), S ? /* @__PURE__ */ x("div", {
      children: [B, /* @__PURE__ */ x(De, {
        ownerState: z,
        "aria-hidden": !0,
        className: U.asterisk,
        children: [" ", "*"]
      })]
    }) : B]
  });
});
process.env.NODE_ENV !== "production" && (me.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * If `true`, the component appears selected.
   */
  checked: o.bool,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * The props used for each slot inside.
   * @default {}
   * @deprecated use the `slotProps` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   */
  componentsProps: o.shape({
    typography: o.object
  }),
  /**
   * A control element. For instance, it can be a `Radio`, a `Switch` or a `Checkbox`.
   */
  control: o.element.isRequired,
  /**
   * If `true`, the control is disabled.
   */
  disabled: o.bool,
  /**
   * If `true`, the label is rendered as it is passed without an additional typography node.
   */
  disableTypography: o.bool,
  /**
   * Pass a ref to the `input` element.
   */
  inputRef: W,
  /**
   * A text or an element to be used in an enclosing label element.
   */
  label: o.node,
  /**
   * The position of the label.
   * @default 'end'
   */
  labelPlacement: o.oneOf(["bottom", "end", "start", "top"]),
  /**
   * @ignore
   */
  name: o.string,
  /**
   * Callback fired when the state is changed.
   *
   * @param {React.SyntheticEvent} event The event source of the callback.
   * You can pull out the new checked state by accessing `event.target.checked` (boolean).
   */
  onChange: o.func,
  /**
   * If `true`, the label will indicate that the `input` is required.
   */
  required: o.bool,
  /**
   * The props used for each slot inside.
   * @default {}
   */
  slotProps: o.shape({
    typography: o.oneOfType([o.func, o.object])
  }),
  /**
   * The components used for each slot inside.
   * @default {}
   */
  slots: o.shape({
    typography: o.elementType
  }),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object]),
  /**
   * The value of the component.
   */
  value: o.any
});
function Ge(e) {
  return L("MuiFormGroup", e);
}
q("MuiFormGroup", ["root", "row", "error"]);
const Ue = (e) => {
  const {
    classes: t,
    row: r,
    error: s
  } = e;
  return D({
    root: ["root", r && "row", s && "error"]
  }, Ge, t);
}, Ve = C("div", {
  name: "MuiFormGroup",
  slot: "Root",
  overridesResolver: (e, t) => {
    const {
      ownerState: r
    } = e;
    return [t.root, r.row && t.row];
  }
})({
  display: "flex",
  flexDirection: "column",
  flexWrap: "wrap",
  variants: [{
    props: {
      row: !0
    },
    style: {
      flexDirection: "row"
    }
  }]
}), be = /* @__PURE__ */ v.forwardRef(function(t, r) {
  const s = H({
    props: t,
    name: "MuiFormGroup"
  }), {
    className: a,
    row: c = !1,
    ...i
  } = s, u = X(), d = ue({
    props: s,
    muiFormControl: u,
    states: ["error"]
  }), m = {
    ...s,
    row: c,
    error: d.error
  }, f = Ue(m);
  return /* @__PURE__ */ n(Ve, {
    className: j(f.root, a),
    ownerState: m,
    ref: r,
    ...i
  });
});
process.env.NODE_ENV !== "production" && (be.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * Display group of elements in a compact row.
   * @default false
   */
  row: o.bool,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object])
});
const _e = he(/* @__PURE__ */ n("path", {
  d: "M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
}), "RadioButtonUnchecked"), Ae = he(/* @__PURE__ */ n("path", {
  d: "M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
}), "RadioButtonChecked"), We = C("span", {
  shouldForwardProp: ee
})({
  position: "relative",
  display: "flex"
}), He = C(_e)({
  // Scale applied to prevent dot misalignment in Safari
  transform: "scale(1)"
}), Xe = C(Ae)(I(({
  theme: e
}) => ({
  left: 0,
  position: "absolute",
  transform: "scale(0)",
  transition: e.transitions.create("transform", {
    easing: e.transitions.easing.easeIn,
    duration: e.transitions.duration.shortest
  }),
  variants: [{
    props: {
      checked: !0
    },
    style: {
      transform: "scale(1)",
      transition: e.transitions.create("transform", {
        easing: e.transitions.easing.easeOut,
        duration: e.transitions.duration.shortest
      })
    }
  }]
})));
function te(e) {
  const {
    checked: t = !1,
    classes: r = {},
    fontSize: s
  } = e, a = {
    ...e,
    checked: t
  };
  return /* @__PURE__ */ x(We, {
    className: r.root,
    ownerState: a,
    children: [/* @__PURE__ */ n(He, {
      fontSize: s,
      className: r.background,
      ownerState: a
    }), /* @__PURE__ */ n(Xe, {
      fontSize: s,
      className: r.dot,
      ownerState: a
    })]
  });
}
process.env.NODE_ENV !== "production" && (te.propTypes = {
  /**
   * If `true`, the component is checked.
   */
  checked: o.bool,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * The size of the component.
   * `small` is equivalent to the dense radio styling.
   */
  fontSize: o.oneOf(["small", "medium"])
});
const re = /* @__PURE__ */ v.createContext(void 0);
process.env.NODE_ENV !== "production" && (re.displayName = "RadioGroupContext");
function Ke() {
  return v.useContext(re);
}
function Ze(e) {
  return L("MuiRadio", e);
}
const ie = q("MuiRadio", ["root", "checked", "disabled", "colorPrimary", "colorSecondary", "sizeSmall"]), Je = (e) => {
  const {
    classes: t,
    color: r,
    size: s
  } = e, a = {
    root: ["root", `color${w(r)}`, s !== "medium" && `size${w(s)}`]
  };
  return {
    ...t,
    ...D(a, Ze, t)
  };
}, Qe = C(oe, {
  shouldForwardProp: (e) => ee(e) || e === "classes",
  name: "MuiRadio",
  slot: "Root",
  overridesResolver: (e, t) => {
    const {
      ownerState: r
    } = e;
    return [t.root, r.size !== "medium" && t[`size${w(r.size)}`], t[`color${w(r.color)}`]];
  }
})(I(({
  theme: e
}) => ({
  color: (e.vars || e).palette.text.secondary,
  [`&.${ie.disabled}`]: {
    color: (e.vars || e).palette.action.disabled
  },
  variants: [{
    props: {
      color: "default",
      disabled: !1,
      disableRipple: !1
    },
    style: {
      "&:hover": {
        backgroundColor: e.vars ? `rgba(${e.vars.palette.action.activeChannel} / ${e.vars.palette.action.hoverOpacity})` : A(e.palette.action.active, e.palette.action.hoverOpacity)
      }
    }
  }, ...Object.entries(e.palette).filter(Q()).map(([t]) => ({
    props: {
      color: t,
      disabled: !1,
      disableRipple: !1
    },
    style: {
      "&:hover": {
        backgroundColor: e.vars ? `rgba(${e.vars.palette[t].mainChannel} / ${e.vars.palette.action.hoverOpacity})` : A(e.palette[t].main, e.palette.action.hoverOpacity)
      }
    }
  })), ...Object.entries(e.palette).filter(Q()).map(([t]) => ({
    props: {
      color: t,
      disabled: !1
    },
    style: {
      [`&.${ie.checked}`]: {
        color: (e.vars || e).palette[t].main
      }
    }
  })), {
    // Should be last to override other colors
    props: {
      disableRipple: !1
    },
    style: {
      // Reset on touch devices, it doesn't add specificity
      "&:hover": {
        "@media (hover: none)": {
          backgroundColor: "transparent"
        }
      }
    }
  }]
})));
function Ye(e, t) {
  return typeof t == "object" && t !== null ? e === t : String(e) === String(t);
}
const le = /* @__PURE__ */ n(te, {
  checked: !0
}), ce = /* @__PURE__ */ n(te, {}), ge = /* @__PURE__ */ v.forwardRef(function(t, r) {
  const s = H({
    props: t,
    name: "MuiRadio"
  }), {
    checked: a,
    checkedIcon: c = le,
    color: i = "primary",
    icon: u = ce,
    name: d,
    onChange: m,
    size: f = "medium",
    className: p,
    disabled: b,
    disableRipple: l = !1,
    ...R
  } = s, k = X();
  let h = b;
  k && typeof h > "u" && (h = k.disabled), h ?? (h = !1);
  const T = {
    ...s,
    disabled: h,
    disableRipple: l,
    color: i,
    size: f
  }, g = Je(T), F = Ke();
  let O = a;
  const M = Pe(m, F && F.onChange);
  let S = d;
  return F && (typeof O > "u" && (O = Ye(F.value, s.value)), typeof S > "u" && (S = F.name)), /* @__PURE__ */ n(Qe, {
    type: "radio",
    icon: /* @__PURE__ */ v.cloneElement(u, {
      fontSize: ce.props.fontSize ?? f
    }),
    checkedIcon: /* @__PURE__ */ v.cloneElement(c, {
      fontSize: le.props.fontSize ?? f
    }),
    disabled: h,
    ownerState: T,
    classes: g,
    name: S,
    checked: O,
    onChange: M,
    ref: r,
    className: j(g.root, p),
    ...R
  });
});
process.env.NODE_ENV !== "production" && (ge.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * If `true`, the component is checked.
   */
  checked: o.bool,
  /**
   * The icon to display when the component is checked.
   * @default <RadioButtonIcon checked />
   */
  checkedIcon: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * The color of the component.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   * @default 'primary'
   */
  color: o.oneOfType([o.oneOf(["default", "primary", "secondary", "error", "info", "success", "warning"]), o.string]),
  /**
   * If `true`, the component is disabled.
   */
  disabled: o.bool,
  /**
   * If `true`, the ripple effect is disabled.
   * @default false
   */
  disableRipple: o.bool,
  /**
   * The icon to display when the component is unchecked.
   * @default <RadioButtonIcon />
   */
  icon: o.node,
  /**
   * The id of the `input` element.
   */
  id: o.string,
  /**
   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes) applied to the `input` element.
   */
  inputProps: o.object,
  /**
   * Pass a ref to the `input` element.
   */
  inputRef: W,
  /**
   * Name attribute of the `input` element.
   */
  name: o.string,
  /**
   * Callback fired when the state is changed.
   *
   * @param {React.ChangeEvent<HTMLInputElement>} event The event source of the callback.
   * You can pull out the new value by accessing `event.target.value` (string).
   * You can pull out the new checked state by accessing `event.target.checked` (boolean).
   */
  onChange: o.func,
  /**
   * If `true`, the `input` element is required.
   * @default false
   */
  required: o.bool,
  /**
   * The size of the component.
   * `small` is equivalent to the dense radio styling.
   * @default 'medium'
   */
  size: o.oneOfType([o.oneOf(["medium", "small"]), o.string]),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object]),
  /**
   * The value of the component. The DOM API casts this to a string.
   */
  value: o.any
});
function eo(e) {
  return L("MuiRadioGroup", e);
}
q("MuiRadioGroup", ["root", "row", "error"]);
const oo = (e) => {
  const {
    classes: t,
    row: r,
    error: s
  } = e;
  return D({
    root: ["root", r && "row", s && "error"]
  }, eo, t);
}, ye = /* @__PURE__ */ v.forwardRef(function(t, r) {
  const {
    // private
    // eslint-disable-next-line react/prop-types
    actions: s,
    children: a,
    className: c,
    defaultValue: i,
    name: u,
    onChange: d,
    value: m,
    ...f
  } = t, p = v.useRef(null), b = oo(t), [l, R] = fe({
    controlled: m,
    default: i,
    name: "RadioGroup"
  });
  v.useImperativeHandle(s, () => ({
    focus: () => {
      let g = p.current.querySelector("input:not(:disabled):checked");
      g || (g = p.current.querySelector("input:not(:disabled)")), g && g.focus();
    }
  }), []);
  const k = Ne(r, p), h = Fe(u), T = v.useMemo(() => ({
    name: h,
    onChange(g) {
      R(g.target.value), d && d(g, g.target.value);
    },
    value: l
  }), [h, d, R, l]);
  return /* @__PURE__ */ n(re.Provider, {
    value: T,
    children: /* @__PURE__ */ n(be, {
      role: "radiogroup",
      ref: k,
      className: j(b.root, c),
      ...f,
      children: a
    })
  });
});
process.env.NODE_ENV !== "production" && (ye.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: o.node,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * The default value. Use when the component is not controlled.
   */
  defaultValue: o.any,
  /**
   * The name used to reference the value of the control.
   * If you don't provide this prop, it falls back to a randomly generated name.
   */
  name: o.string,
  /**
   * Callback fired when a radio button is selected.
   *
   * @param {React.ChangeEvent<HTMLInputElement>} event The event source of the callback.
   * @param {string} value The value of the selected radio button.
   * You can pull out the new value by accessing `event.target.value` (string).
   */
  onChange: o.func,
  /**
   * Value of the selected radio button. The DOM API casts this to a string.
   */
  value: o.any
});
function to(e) {
  return L("MuiSwitch", e);
}
const y = q("MuiSwitch", ["root", "edgeStart", "edgeEnd", "switchBase", "colorPrimary", "colorSecondary", "sizeSmall", "sizeMedium", "checked", "disabled", "input", "thumb", "track"]), ro = (e) => {
  const {
    classes: t,
    edge: r,
    size: s,
    color: a,
    checked: c,
    disabled: i
  } = e, u = {
    root: ["root", r && `edge${w(r)}`, `size${w(s)}`],
    switchBase: ["switchBase", `color${w(a)}`, c && "checked", i && "disabled"],
    thumb: ["thumb"],
    track: ["track"],
    input: ["input"]
  }, d = D(u, to, t);
  return {
    ...t,
    // forward the disabled and checked classes to the SwitchBase
    ...d
  };
}, so = C("span", {
  name: "MuiSwitch",
  slot: "Root",
  overridesResolver: (e, t) => {
    const {
      ownerState: r
    } = e;
    return [t.root, r.edge && t[`edge${w(r.edge)}`], t[`size${w(r.size)}`]];
  }
})({
  display: "inline-flex",
  width: 34 + 12 * 2,
  height: 14 + 12 * 2,
  overflow: "hidden",
  padding: 12,
  boxSizing: "border-box",
  position: "relative",
  flexShrink: 0,
  zIndex: 0,
  // Reset the stacking context.
  verticalAlign: "middle",
  // For correct alignment with the text.
  "@media print": {
    colorAdjust: "exact"
  },
  variants: [{
    props: {
      edge: "start"
    },
    style: {
      marginLeft: -8
    }
  }, {
    props: {
      edge: "end"
    },
    style: {
      marginRight: -8
    }
  }, {
    props: {
      size: "small"
    },
    style: {
      width: 40,
      height: 24,
      padding: 7,
      [`& .${y.thumb}`]: {
        width: 16,
        height: 16
      },
      [`& .${y.switchBase}`]: {
        padding: 4,
        [`&.${y.checked}`]: {
          transform: "translateX(16px)"
        }
      }
    }
  }]
}), ao = C(oe, {
  name: "MuiSwitch",
  slot: "SwitchBase",
  overridesResolver: (e, t) => {
    const {
      ownerState: r
    } = e;
    return [t.switchBase, {
      [`& .${y.input}`]: t.input
    }, r.color !== "default" && t[`color${w(r.color)}`]];
  }
})(I(({
  theme: e
}) => ({
  position: "absolute",
  top: 0,
  left: 0,
  zIndex: 1,
  // Render above the focus ripple.
  color: e.vars ? e.vars.palette.Switch.defaultColor : `${e.palette.mode === "light" ? e.palette.common.white : e.palette.grey[300]}`,
  transition: e.transitions.create(["left", "transform"], {
    duration: e.transitions.duration.shortest
  }),
  [`&.${y.checked}`]: {
    transform: "translateX(20px)"
  },
  [`&.${y.disabled}`]: {
    color: e.vars ? e.vars.palette.Switch.defaultDisabledColor : `${e.palette.mode === "light" ? e.palette.grey[100] : e.palette.grey[600]}`
  },
  [`&.${y.checked} + .${y.track}`]: {
    opacity: 0.5
  },
  [`&.${y.disabled} + .${y.track}`]: {
    opacity: e.vars ? e.vars.opacity.switchTrackDisabled : `${e.palette.mode === "light" ? 0.12 : 0.2}`
  },
  [`& .${y.input}`]: {
    left: "-100%",
    width: "300%"
  }
})), I(({
  theme: e
}) => ({
  "&:hover": {
    backgroundColor: e.vars ? `rgba(${e.vars.palette.action.activeChannel} / ${e.vars.palette.action.hoverOpacity})` : A(e.palette.action.active, e.palette.action.hoverOpacity),
    // Reset on touch devices, it doesn't add specificity
    "@media (hover: none)": {
      backgroundColor: "transparent"
    }
  },
  variants: [...Object.entries(e.palette).filter(Q(["light"])).map(([t]) => ({
    props: {
      color: t
    },
    style: {
      [`&.${y.checked}`]: {
        color: (e.vars || e).palette[t].main,
        "&:hover": {
          backgroundColor: e.vars ? `rgba(${e.vars.palette[t].mainChannel} / ${e.vars.palette.action.hoverOpacity})` : A(e.palette[t].main, e.palette.action.hoverOpacity),
          "@media (hover: none)": {
            backgroundColor: "transparent"
          }
        },
        [`&.${y.disabled}`]: {
          color: e.vars ? e.vars.palette.Switch[`${t}DisabledColor`] : `${e.palette.mode === "light" ? Oe(e.palette[t].main, 0.62) : $e(e.palette[t].main, 0.55)}`
        }
      },
      [`&.${y.checked} + .${y.track}`]: {
        backgroundColor: (e.vars || e).palette[t].main
      }
    }
  }))]
}))), no = C("span", {
  name: "MuiSwitch",
  slot: "Track",
  overridesResolver: (e, t) => t.track
})(I(({
  theme: e
}) => ({
  height: "100%",
  width: "100%",
  borderRadius: 14 / 2,
  zIndex: -1,
  transition: e.transitions.create(["opacity", "background-color"], {
    duration: e.transitions.duration.shortest
  }),
  backgroundColor: e.vars ? e.vars.palette.common.onBackground : `${e.palette.mode === "light" ? e.palette.common.black : e.palette.common.white}`,
  opacity: e.vars ? e.vars.opacity.switchTrack : `${e.palette.mode === "light" ? 0.38 : 0.3}`
}))), io = C("span", {
  name: "MuiSwitch",
  slot: "Thumb",
  overridesResolver: (e, t) => t.thumb
})(I(({
  theme: e
}) => ({
  boxShadow: (e.vars || e).shadows[1],
  backgroundColor: "currentColor",
  width: 20,
  height: 20,
  borderRadius: "50%"
}))), Y = /* @__PURE__ */ v.forwardRef(function(t, r) {
  const s = H({
    props: t,
    name: "MuiSwitch"
  }), {
    className: a,
    color: c = "primary",
    edge: i = !1,
    size: u = "medium",
    sx: d,
    ...m
  } = s, f = {
    ...s,
    color: c,
    edge: i,
    size: u
  }, p = ro(f), b = /* @__PURE__ */ n(io, {
    className: p.thumb,
    ownerState: f
  });
  return /* @__PURE__ */ x(so, {
    className: j(p.root, a),
    sx: d,
    ownerState: f,
    children: [/* @__PURE__ */ n(ao, {
      type: "checkbox",
      icon: b,
      checkedIcon: b,
      ref: r,
      ownerState: f,
      ...m,
      classes: {
        ...p,
        root: p.switchBase
      }
    }), /* @__PURE__ */ n(no, {
      className: p.track,
      ownerState: f
    })]
  });
});
process.env.NODE_ENV !== "production" && (Y.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * If `true`, the component is checked.
   */
  checked: o.bool,
  /**
   * The icon to display when the component is checked.
   */
  checkedIcon: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * The color of the component.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   * @default 'primary'
   */
  color: o.oneOfType([o.oneOf(["default", "primary", "secondary", "error", "info", "success", "warning"]), o.string]),
  /**
   * The default checked state. Use when the component is not controlled.
   */
  defaultChecked: o.bool,
  /**
   * If `true`, the component is disabled.
   */
  disabled: o.bool,
  /**
   * If `true`, the ripple effect is disabled.
   * @default false
   */
  disableRipple: o.bool,
  /**
   * If given, uses a negative margin to counteract the padding on one
   * side (this is often helpful for aligning the left or right
   * side of the icon with content above or below, without ruining the border
   * size and shape).
   * @default false
   */
  edge: o.oneOf(["end", "start", !1]),
  /**
   * The icon to display when the component is unchecked.
   */
  icon: o.node,
  /**
   * The id of the `input` element.
   */
  id: o.string,
  /**
   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Attributes) applied to the `input` element.
   */
  inputProps: o.object,
  /**
   * Pass a ref to the `input` element.
   */
  inputRef: W,
  /**
   * Callback fired when the state is changed.
   *
   * @param {React.ChangeEvent<HTMLInputElement>} event The event source of the callback.
   * You can pull out the new value by accessing `event.target.value` (string).
   * You can pull out the new checked state by accessing `event.target.checked` (boolean).
   */
  onChange: o.func,
  /**
   * If `true`, the `input` element is required.
   * @default false
   */
  required: o.bool,
  /**
   * The size of the component.
   * `small` is equivalent to the dense switch styling.
   * @default 'medium'
   */
  size: o.oneOfType([o.oneOf(["medium", "small"]), o.string]),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object]),
  /**
   * The value of the component. The DOM API casts this to a string.
   * The browser uses "on" as the default value.
   */
  value: o.any
});
const lo = (e, t) => {
  const r = [
    "EMAIL",
    "INAPP_WEB",
    "SMS",
    "CALL",
    "PUSH",
    "WEB_PUSH",
    "SLACK"
  ];
  return r.indexOf(e) - r.indexOf(t);
}, de = (e, t) => {
  const r = ["off", "instant", "hourly", "daily", "weekly", "monthly"];
  return r.indexOf(e) - r.indexOf(t);
}, pe = (e) => ({
  off: "Off",
  instant: "Instant",
  hourly: "Hourly",
  daily: "Daily",
  weekly: "Weekly",
  monthly: "Monthly"
})[e], So = ({
  notification: e,
  preferences: t,
  updateDelivery: r,
  subNotificationId: s
}) => {
  const a = xe(), c = ke(a);
  return /* @__PURE__ */ n(ne, { children: e.channels.sort(lo).map((i, u) => {
    const d = t.find(
      (l) => l.notificationId === e.notificationId && l.channel === i
    );
    if (!d)
      return null;
    const m = we(i), f = Ce(i), p = Object.keys(
      e.options[i]
    ).filter(
      (l) => l !== "defaultDeliveryOption" && l !== "defaultDeliverOption"
    );
    let b;
    return p.length === 1 ? b = /* @__PURE__ */ n(_, { variant: "body1", sx: { color: c.text }, children: pe(d.delivery) }) : p.length === 2 && p.includes("off") ? b = /* @__PURE__ */ n(
      Y,
      {
        checked: d.delivery !== "off",
        onChange: (l, R) => {
          if (R) {
            const k = p.find((h) => h !== "off");
            r(
              e.notificationId,
              i,
              k,
              s
            );
          } else
            r(
              e.notificationId,
              i,
              "off",
              s
            );
        }
      }
    ) : b = /* @__PURE__ */ x(ne, { children: [
      /* @__PURE__ */ n(
        Y,
        {
          checked: d.delivery !== "off",
          onChange: (l, R) => {
            if (R) {
              const k = p.sort(de).find((h) => h !== "off");
              r(
                e.notificationId,
                i,
                k,
                s
              );
            } else
              r(
                e.notificationId,
                i,
                "off",
                s
              );
          }
        }
      ),
      /* @__PURE__ */ n(
        "div",
        {
          style: {
            width: "100%",
            marginTop: 8,
            maxHeight: d.delivery !== "off" ? 1e3 : 0,
            overflow: "hidden",
            transition: "max-height 0.3s ease"
          },
          children: /* @__PURE__ */ x("div", { children: [
            /* @__PURE__ */ n("div", { style: { marginTop: 20 }, children: /* @__PURE__ */ n(
              _,
              {
                variant: "body1",
                fontWeight: 600,
                sx: { color: c.text },
                children: "Choose frequency:"
              }
            ) }),
            /* @__PURE__ */ n(
              ye,
              {
                value: d.delivery,
                onChange: (l) => {
                  r(
                    e.notificationId,
                    i,
                    l.target.value,
                    s
                  );
                },
                children: /* @__PURE__ */ n(Te, { direction: "column", style: { paddingTop: 10 }, children: p.filter((l) => l !== "off").sort(de).map((l) => /* @__PURE__ */ n(
                  me,
                  {
                    control: /* @__PURE__ */ n(ge, {}),
                    value: l,
                    label: pe(l)
                  },
                  l
                )) })
              }
            )
          ] })
        }
      )
    ] }), /* @__PURE__ */ x("div", { children: [
      /* @__PURE__ */ x(
        "div",
        {
          style: {
            display: "flex",
            justifyContent: "space-between",
            flexWrap: "wrap",
            marginTop: u === 0 ? 0 : 12,
            marginBottom: u === e.channels.length - 1 ? 0 : 12
          },
          children: [
            /* @__PURE__ */ x(
              "div",
              {
                style: {
                  display: "flex",
                  alignItems: "center",
                  gap: 8
                },
                children: [
                  f,
                  /* @__PURE__ */ n(_, { variant: "body1", sx: { color: c.text }, children: m })
                ]
              }
            ),
            b
          ]
        }
      ),
      u !== e.channels.length - 1 && /* @__PURE__ */ n(Be, { sx: { borderColor: c.divider } })
    ] }, i);
  }) });
};
export {
  So as PreferenceInput
};
