// @notificationapi/react/dist/components/Preferences/Preferences.js
import { jsx as s, jsxs as G, Fragment as he } from "react/jsx-runtime";
import * as u from "react";
import { useContext as xe } from "react";
import { NotificationAPIContext as ye } from "../Provider/context.js";
import { PreferenceInput as ie } from "./PreferenceInput.js";
import { getThemeColors as ve } from "../../utils/theme.js";
import { u as le, a as Ce, e as Te, b as se, c as we, B as Ee } from "../../assets/ButtonBase.js";
import { r as Re } from "../../assets/index2.js";
import { g as q, P as e, p as Ae } from "../../assets/createTheme.js";
import { g as _, s as E, m as I, u as J, c as U, a as K } from "../../assets/DefaultPropsProvider.js";
import { P as Se } from "../../assets/Paper.js";
import { a as Ne } from "../../assets/dividerClasses.js";
import { T as Ie, g as ae } from "../../assets/utils.js";
import { u as De } from "../../assets/resolveComponentProps.js";
import { c as ke } from "../../assets/createSvgIcon.js";
import { T as ce } from "../../assets/Typography.js";
import { B as Me } from "../../assets/Box.js";
function Pe(o) {
  return q("MuiCollapse", o);
}
_("MuiCollapse", ["root", "horizontal", "vertical", "entered", "hidden", "wrapper", "wrapperInner"]);
const Oe = (o) => {
  const {
    orientation: t,
    classes: n
  } = o, c = {
    root: ["root", `${t}`],
    entered: ["entered"],
    hidden: ["hidden"],
    wrapper: ["wrapper", `${t}`],
    wrapperInner: ["wrapperInner", `${t}`]
  };
  return K(c, Pe, n);
}, $e = E("div", {
  name: "MuiCollapse",
  slot: "Root",
  overridesResolver: (o, t) => {
    const {
      ownerState: n
    } = o;
    return [t.root, t[n.orientation], n.state === "entered" && t.entered, n.state === "exited" && !n.in && n.collapsedSize === "0px" && t.hidden];
  }
})(I(({
  theme: o
}) => ({
  height: 0,
  overflow: "hidden",
  transition: o.transitions.create("height"),
  variants: [{
    props: {
      orientation: "horizontal"
    },
    style: {
      height: "auto",
      width: 0,
      transition: o.transitions.create("width")
    }
  }, {
    props: {
      state: "entered"
    },
    style: {
      height: "auto",
      overflow: "visible"
    }
  }, {
    props: {
      state: "entered",
      orientation: "horizontal"
    },
    style: {
      width: "auto"
    }
  }, {
    props: ({
      ownerState: t
    }) => t.state === "exited" && !t.in && t.collapsedSize === "0px",
    style: {
      visibility: "hidden"
    }
  }]
}))), ze = E("div", {
  name: "MuiCollapse",
  slot: "Wrapper",
  overridesResolver: (o, t) => t.wrapper
})({
  // Hack to get children with a negative margin to not falsify the height computation.
  display: "flex",
  width: "100%",
  variants: [{
    props: {
      orientation: "horizontal"
    },
    style: {
      width: "auto",
      height: "100%"
    }
  }]
}), je = E("div", {
  name: "MuiCollapse",
  slot: "WrapperInner",
  overridesResolver: (o, t) => t.wrapperInner
})({
  width: "100%",
  variants: [{
    props: {
      orientation: "horizontal"
    },
    style: {
      width: "auto",
      height: "100%"
    }
  }]
}), L = /* @__PURE__ */ u.forwardRef(function(t, n) {
  const c = J({
    props: t,
    name: "MuiCollapse"
  }), {
    addEndListener: i,
    children: p,
    className: l,
    collapsedSize: d = "0px",
    component: g,
    easing: C,
    in: m,
    onEnter: a,
    onEntered: x,
    onEntering: A,
    onExit: D,
    onExited: R,
    onExiting: T,
    orientation: f = "vertical",
    style: k,
    timeout: y = Ae.standard,
    // eslint-disable-next-line react/prop-types
    TransitionComponent: W = Ie,
    ...Q
  } = c, M = {
    ...c,
    orientation: f,
    collapsedSize: d
  }, v = Oe(M), P = le(), X = Ce(), w = u.useRef(null), O = u.useRef(), $ = typeof d == "number" ? `${d}px` : d, S = f === "horizontal", N = S ? "width" : "height", z = u.useRef(null), Y = De(n, z), j = (r) => (b) => {
    if (r) {
      const h = z.current;
      b === void 0 ? r(h) : r(h, b);
    }
  }, Z = () => w.current ? w.current[S ? "clientWidth" : "clientHeight"] : 0, pe = j((r, b) => {
    w.current && S && (w.current.style.position = "absolute"), r.style[N] = $, a && a(r, b);
  }), ue = j((r, b) => {
    const h = Z();
    w.current && S && (w.current.style.position = "");
    const {
      duration: B,
      easing: H
    } = ae({
      style: k,
      timeout: y,
      easing: C
    }, {
      mode: "enter"
    });
    if (y === "auto") {
      const re = P.transitions.getAutoHeightDuration(h);
      r.style.transitionDuration = `${re}ms`, O.current = re;
    } else
      r.style.transitionDuration = typeof B == "string" ? B : `${B}ms`;
    r.style[N] = `${h}px`, r.style.transitionTimingFunction = H, A && A(r, b);
  }), fe = j((r, b) => {
    r.style[N] = "auto", x && x(r, b);
  }), me = j((r) => {
    r.style[N] = `${Z()}px`, D && D(r);
  }), ge = j(R), be = j((r) => {
    const b = Z(), {
      duration: h,
      easing: B
    } = ae({
      style: k,
      timeout: y,
      easing: C
    }, {
      mode: "exit"
    });
    if (y === "auto") {
      const H = P.transitions.getAutoHeightDuration(b);
      r.style.transitionDuration = `${H}ms`, O.current = H;
    } else
      r.style.transitionDuration = typeof h == "string" ? h : `${h}ms`;
    r.style[N] = $, r.style.transitionTimingFunction = B, T && T(r);
  });
  return /* @__PURE__ */ s(W, {
    in: m,
    onEnter: pe,
    onEntered: fe,
    onEntering: ue,
    onExit: me,
    onExited: ge,
    onExiting: be,
    addEndListener: (r) => {
      y === "auto" && X.start(O.current || 0, r), i && i(z.current, r);
    },
    nodeRef: z,
    timeout: y === "auto" ? null : y,
    ...Q,
    children: (r, {
      ownerState: b,
      ...h
    }) => /* @__PURE__ */ s($e, {
      as: g,
      className: U(v.root, l, {
        entered: v.entered,
        exited: !m && $ === "0px" && v.hidden
      }[r]),
      style: {
        [S ? "minWidth" : "minHeight"]: $,
        ...k
      },
      ref: Y,
      ownerState: {
        ...M,
        state: r
      },
      ...h,
      children: /* @__PURE__ */ s(ze, {
        ownerState: {
          ...M,
          state: r
        },
        className: v.wrapper,
        ref: w,
        children: /* @__PURE__ */ s(je, {
          ownerState: {
            ...M,
            state: r
          },
          className: v.wrapperInner,
          children: p
        })
      })
    })
  });
});
process.env.NODE_ENV !== "production" && (L.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * Add a custom transition end trigger. Called with the transitioning DOM
   * node and a done callback. Allows for more fine grained transition end
   * logic. Note: Timeouts are still used as a fallback if provided.
   */
  addEndListener: e.func,
  /**
   * The content node to be collapsed.
   */
  children: e.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * The width (horizontal) or height (vertical) of the container when collapsed.
   * @default '0px'
   */
  collapsedSize: e.oneOfType([e.number, e.string]),
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: Te,
  /**
   * The transition timing function.
   * You may specify a single easing or a object containing enter and exit values.
   */
  easing: e.oneOfType([e.shape({
    enter: e.string,
    exit: e.string
  }), e.string]),
  /**
   * If `true`, the component will transition in.
   */
  in: e.bool,
  /**
   * @ignore
   */
  onEnter: e.func,
  /**
   * @ignore
   */
  onEntered: e.func,
  /**
   * @ignore
   */
  onEntering: e.func,
  /**
   * @ignore
   */
  onExit: e.func,
  /**
   * @ignore
   */
  onExited: e.func,
  /**
   * @ignore
   */
  onExiting: e.func,
  /**
   * The transition orientation.
   * @default 'vertical'
   */
  orientation: e.oneOf(["horizontal", "vertical"]),
  /**
   * @ignore
   */
  style: e.object,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * The duration for the transition, in milliseconds.
   * You may specify a single timeout for all transitions, or individually with an object.
   *
   * Set to 'auto' to automatically calculate transition time based on height.
   * @default duration.standard
   */
  timeout: e.oneOfType([e.oneOf(["auto"]), e.number, e.shape({
    appear: e.number,
    enter: e.number,
    exit: e.number
  })])
});
L && (L.muiSupportAuto = !0);
const ne = /* @__PURE__ */ u.createContext({});
process.env.NODE_ENV !== "production" && (ne.displayName = "AccordionContext");
function Ve(o) {
  return q("MuiAccordion", o);
}
const F = _("MuiAccordion", ["root", "heading", "rounded", "expanded", "disabled", "gutters", "region"]), We = (o) => {
  const {
    classes: t,
    square: n,
    expanded: c,
    disabled: i,
    disableGutters: p
  } = o;
  return K({
    root: ["root", !n && "rounded", c && "expanded", i && "disabled", !p && "gutters"],
    heading: ["heading"],
    region: ["region"]
  }, Ve, t);
}, Be = E(Se, {
  name: "MuiAccordion",
  slot: "Root",
  overridesResolver: (o, t) => {
    const {
      ownerState: n
    } = o;
    return [{
      [`& .${F.region}`]: t.region
    }, t.root, !n.square && t.rounded, !n.disableGutters && t.gutters];
  }
})(I(({
  theme: o
}) => {
  const t = {
    duration: o.transitions.duration.shortest
  };
  return {
    position: "relative",
    transition: o.transitions.create(["margin"], t),
    overflowAnchor: "none",
    // Keep the same scrolling position
    "&::before": {
      position: "absolute",
      left: 0,
      top: -1,
      right: 0,
      height: 1,
      content: '""',
      opacity: 1,
      backgroundColor: (o.vars || o).palette.divider,
      transition: o.transitions.create(["opacity", "background-color"], t)
    },
    "&:first-of-type": {
      "&::before": {
        display: "none"
      }
    },
    [`&.${F.expanded}`]: {
      "&::before": {
        opacity: 0
      },
      "&:first-of-type": {
        marginTop: 0
      },
      "&:last-of-type": {
        marginBottom: 0
      },
      "& + &": {
        "&::before": {
          display: "none"
        }
      }
    },
    [`&.${F.disabled}`]: {
      backgroundColor: (o.vars || o).palette.action.disabledBackground
    }
  };
}), I(({
  theme: o
}) => ({
  variants: [{
    props: (t) => !t.square,
    style: {
      borderRadius: 0,
      "&:first-of-type": {
        borderTopLeftRadius: (o.vars || o).shape.borderRadius,
        borderTopRightRadius: (o.vars || o).shape.borderRadius
      },
      "&:last-of-type": {
        borderBottomLeftRadius: (o.vars || o).shape.borderRadius,
        borderBottomRightRadius: (o.vars || o).shape.borderRadius,
        // Fix a rendering issue on Edge
        "@supports (-ms-ime-align: auto)": {
          borderBottomLeftRadius: 0,
          borderBottomRightRadius: 0
        }
      }
    }
  }, {
    props: (t) => !t.disableGutters,
    style: {
      [`&.${F.expanded}`]: {
        margin: "16px 0"
      }
    }
  }]
}))), Ge = E("h3", {
  name: "MuiAccordion",
  slot: "Heading",
  overridesResolver: (o, t) => t.heading
})({
  all: "unset"
}), ee = /* @__PURE__ */ u.forwardRef(function(t, n) {
  const c = J({
    props: t,
    name: "MuiAccordion"
  }), {
    children: i,
    className: p,
    defaultExpanded: l = !1,
    disabled: d = !1,
    disableGutters: g = !1,
    expanded: C,
    onChange: m,
    square: a = !1,
    slots: x = {},
    slotProps: A = {},
    TransitionComponent: D,
    TransitionProps: R,
    ...T
  } = c, [f, k] = Ne({
    controlled: C,
    default: l,
    name: "Accordion",
    state: "expanded"
  }), y = u.useCallback((Y) => {
    k(!f), m && m(Y, !f);
  }, [f, m, k]), [W, ...Q] = u.Children.toArray(i), M = u.useMemo(() => ({
    expanded: f,
    disabled: d,
    disableGutters: g,
    toggle: y
  }), [f, d, g, y]), v = {
    ...c,
    square: a,
    disabled: d,
    disableGutters: g,
    expanded: f
  }, P = We(v), X = {
    transition: D,
    ...x
  }, w = {
    transition: R,
    ...A
  }, O = {
    slots: X,
    slotProps: w
  }, [$, S] = se("heading", {
    elementType: Ge,
    externalForwardedProps: O,
    className: P.heading,
    ownerState: v
  }), [N, z] = se("transition", {
    elementType: L,
    externalForwardedProps: O,
    ownerState: v
  });
  return /* @__PURE__ */ G(Be, {
    className: U(P.root, p),
    ref: n,
    ownerState: v,
    square: a,
    ...T,
    children: [/* @__PURE__ */ s($, {
      ...S,
      children: /* @__PURE__ */ s(ne.Provider, {
        value: M,
        children: W
      })
    }), /* @__PURE__ */ s(N, {
      in: f,
      timeout: "auto",
      ...z,
      children: /* @__PURE__ */ s("div", {
        "aria-labelledby": W.props.id,
        id: W.props["aria-controls"],
        role: "region",
        className: P.region,
        children: Q
      })
    })]
  });
});
process.env.NODE_ENV !== "production" && (ee.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: we(e.node.isRequired, (o) => {
    const t = u.Children.toArray(o.children)[0];
    return Re.isFragment(t) ? new Error("MUI: The Accordion doesn't accept a Fragment as a child. Consider providing an array instead.") : /* @__PURE__ */ u.isValidElement(t) ? null : new Error("MUI: Expected the first child of Accordion to be a valid element.");
  }),
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * If `true`, expands the accordion by default.
   * @default false
   */
  defaultExpanded: e.bool,
  /**
   * If `true`, the component is disabled.
   * @default false
   */
  disabled: e.bool,
  /**
   * If `true`, it removes the margin between two expanded accordion items and the increase of height.
   * @default false
   */
  disableGutters: e.bool,
  /**
   * If `true`, expands the accordion, otherwise collapse it.
   * Setting this prop enables control over the accordion.
   */
  expanded: e.bool,
  /**
   * Callback fired when the expand/collapse state is changed.
   *
   * @param {React.SyntheticEvent} event The event source of the callback. **Warning**: This is a generic event not a change event.
   * @param {boolean} expanded The `expanded` state of the accordion.
   */
  onChange: e.func,
  /**
   * The props used for each slot inside.
   * @default {}
   */
  slotProps: e.shape({
    heading: e.oneOfType([e.func, e.object]),
    transition: e.oneOfType([e.func, e.object])
  }),
  /**
   * The components used for each slot inside.
   * @default {}
   */
  slots: e.shape({
    heading: e.elementType,
    transition: e.elementType
  }),
  /**
   * If `true`, rounded corners are disabled.
   * @default false
   */
  square: e.bool,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * The component used for the transition.
   * [Follow this guide](https://mui.com/material-ui/transitions/#transitioncomponent-prop) to learn more about the requirements for this component.
   * @deprecated Use `slots.transition` instead. This prop will be removed in v7. See [Migrating from deprecated APIs](/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   */
  TransitionComponent: e.elementType,
  /**
   * Props applied to the transition element.
   * By default, the element is based on this [`Transition`](https://reactcommunity.org/react-transition-group/transition/) component.
   * @deprecated Use `slotProps.transition` instead. This prop will be removed in v7. See [Migrating from deprecated APIs](/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   */
  TransitionProps: e.object
});
function Ue(o) {
  return q("MuiAccordionDetails", o);
}
_("MuiAccordionDetails", ["root"]);
const He = (o) => {
  const {
    classes: t
  } = o;
  return K({
    root: ["root"]
  }, Ue, t);
}, Fe = E("div", {
  name: "MuiAccordionDetails",
  slot: "Root",
  overridesResolver: (o, t) => t.root
})(I(({
  theme: o
}) => ({
  padding: o.spacing(1, 2, 2)
}))), oe = /* @__PURE__ */ u.forwardRef(function(t, n) {
  const c = J({
    props: t,
    name: "MuiAccordionDetails"
  }), {
    className: i,
    ...p
  } = c, l = c, d = He(l);
  return /* @__PURE__ */ s(Fe, {
    className: U(d.root, i),
    ref: n,
    ownerState: l,
    ...p
  });
});
process.env.NODE_ENV !== "production" && (oe.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: e.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object])
});
function Le(o) {
  return q("MuiAccordionSummary", o);
}
const V = _("MuiAccordionSummary", ["root", "expanded", "focusVisible", "disabled", "gutters", "contentGutters", "content", "expandIconWrapper"]), qe = (o) => {
  const {
    classes: t,
    expanded: n,
    disabled: c,
    disableGutters: i
  } = o;
  return K({
    root: ["root", n && "expanded", c && "disabled", !i && "gutters"],
    focusVisible: ["focusVisible"],
    content: ["content", n && "expanded", !i && "contentGutters"],
    expandIconWrapper: ["expandIconWrapper", n && "expanded"]
  }, Le, t);
}, _e = E(Ee, {
  name: "MuiAccordionSummary",
  slot: "Root",
  overridesResolver: (o, t) => t.root
})(I(({
  theme: o
}) => {
  const t = {
    duration: o.transitions.duration.shortest
  };
  return {
    display: "flex",
    width: "100%",
    minHeight: 48,
    padding: o.spacing(0, 2),
    transition: o.transitions.create(["min-height", "background-color"], t),
    [`&.${V.focusVisible}`]: {
      backgroundColor: (o.vars || o).palette.action.focus
    },
    [`&.${V.disabled}`]: {
      opacity: (o.vars || o).palette.action.disabledOpacity
    },
    [`&:hover:not(.${V.disabled})`]: {
      cursor: "pointer"
    },
    variants: [{
      props: (n) => !n.disableGutters,
      style: {
        [`&.${V.expanded}`]: {
          minHeight: 64
        }
      }
    }]
  };
})), Je = E("span", {
  name: "MuiAccordionSummary",
  slot: "Content",
  overridesResolver: (o, t) => t.content
})(I(({
  theme: o
}) => ({
  display: "flex",
  textAlign: "start",
  flexGrow: 1,
  margin: "12px 0",
  variants: [{
    props: (t) => !t.disableGutters,
    style: {
      transition: o.transitions.create(["margin"], {
        duration: o.transitions.duration.shortest
      }),
      [`&.${V.expanded}`]: {
        margin: "20px 0"
      }
    }
  }]
}))), Ke = E("span", {
  name: "MuiAccordionSummary",
  slot: "ExpandIconWrapper",
  overridesResolver: (o, t) => t.expandIconWrapper
})(I(({
  theme: o
}) => ({
  display: "flex",
  color: (o.vars || o).palette.action.active,
  transform: "rotate(0deg)",
  transition: o.transitions.create("transform", {
    duration: o.transitions.duration.shortest
  }),
  [`&.${V.expanded}`]: {
    transform: "rotate(180deg)"
  }
}))), te = /* @__PURE__ */ u.forwardRef(function(t, n) {
  const c = J({
    props: t,
    name: "MuiAccordionSummary"
  }), {
    children: i,
    className: p,
    expandIcon: l,
    focusVisibleClassName: d,
    onClick: g,
    ...C
  } = c, {
    disabled: m = !1,
    disableGutters: a,
    expanded: x,
    toggle: A
  } = u.useContext(ne), D = (f) => {
    A && A(f), g && g(f);
  }, R = {
    ...c,
    expanded: x,
    disabled: m,
    disableGutters: a
  }, T = qe(R);
  return /* @__PURE__ */ G(_e, {
    focusRipple: !1,
    disableRipple: !0,
    disabled: m,
    "aria-expanded": x,
    className: U(T.root, p),
    focusVisibleClassName: U(T.focusVisible, d),
    onClick: D,
    ref: n,
    ownerState: R,
    ...C,
    children: [/* @__PURE__ */ s(Je, {
      className: T.content,
      ownerState: R,
      children: i
    }), l && /* @__PURE__ */ s(Ke, {
      className: T.expandIconWrapper,
      ownerState: R,
      children: l
    })]
  });
});
process.env.NODE_ENV !== "production" && (te.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: e.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * The icon to display as the expand indicator.
   */
  expandIcon: e.node,
  /**
   * This prop can help identify which element has keyboard focus.
   * The class name will be applied when the element gains the focus through keyboard interaction.
   * It's a polyfill for the [CSS :focus-visible selector](https://drafts.csswg.org/selectors-4/#the-focus-visible-pseudo).
   * The rationale for using this feature [is explained here](https://github.com/WICG/focus-visible/blob/HEAD/explainer.md).
   * A [polyfill can be used](https://github.com/WICG/focus-visible) to apply a `focus-visible` class to other components
   * if needed.
   */
  focusVisibleClassName: e.string,
  /**
   * @ignore
   */
  onClick: e.func,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object])
});
const de = ke(/* @__PURE__ */ s("path", {
  d: "M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
}), "ExpandMore");
function mo() {
  const o = xe(ye), t = le(), n = ve(t);
  if (!o || !o.preferences)
    return null;
  const c = o.preferences.notifications.sort((i, p) => i.title.localeCompare(p.title)).map((i) => {
    var g, C, m;
    const p = (g = o.preferences) == null ? void 0 : g.preferences.filter(
      (a) => a.notificationId === i.notificationId && !a.subNotificationId
    ), l = (C = o.preferences) == null ? void 0 : C.preferences.filter(
      (a) => a.notificationId === i.notificationId && a.subNotificationId
    ), d = (m = o.preferences) == null ? void 0 : m.subNotifications.filter(
      (a) => l == null ? void 0 : l.find(
        (x) => x.subNotificationId === a.subNotificationId && x.notificationId === a.notificationId
      )
    );
    return /* @__PURE__ */ G(
      ee,
      {
        sx: {
          marginBottom: 0,
          marginTop: 0,
          backgroundColor: n.paper,
          color: n.text,
          "&:before": {
            backgroundColor: n.divider
          }
        },
        children: [
          /* @__PURE__ */ s(
            te,
            {
              expandIcon: /* @__PURE__ */ s(de, { sx: { color: n.text } }),
              sx: {
                backgroundColor: t.palette.mode === "dark" ? "rgba(255, 255, 255, 0.05)" : "rgba(0, 0, 0, 0.02)",
                flexDirection: "row-reverse",
                gap: 16,
                "&:hover": {
                  backgroundColor: t.palette.mode === "dark" ? "rgba(255, 255, 255, 0.08)" : "rgba(0, 0, 0, 0.04)"
                }
              },
              children: /* @__PURE__ */ s(ce, { variant: "body1", sx: { color: n.text }, children: i.title })
            }
          ),
          /* @__PURE__ */ s(
            oe,
            {
              sx: { backgroundColor: n.paper, color: n.text },
              children: /* @__PURE__ */ G(he, { children: [
                /* @__PURE__ */ s(
                  ie,
                  {
                    notification: i,
                    preferences: p || [],
                    updateDelivery: o.updateDelivery
                  },
                  i.notificationId
                ),
                d == null ? void 0 : d.map((a) => /* @__PURE__ */ G(
                  ee,
                  {
                    sx: {
                      marginTop: 12,
                      backgroundColor: n.paper,
                      color: n.text,
                      "&:before": {
                        backgroundColor: n.divider
                      }
                    },
                    children: [
                      /* @__PURE__ */ s(
                        te,
                        {
                          expandIcon: /* @__PURE__ */ s(de, { sx: { color: n.text } }),
                          sx: {
                            flexDirection: "row-reverse",
                            gap: 16,
                            backgroundColor: t.palette.mode === "dark" ? "rgba(255, 255, 255, 0.03)" : "rgba(0, 0, 0, 0.01)",
                            "&:hover": {
                              backgroundColor: t.palette.mode === "dark" ? "rgba(255, 255, 255, 0.06)" : "rgba(0, 0, 0, 0.02)"
                            }
                          },
                          children: /* @__PURE__ */ s(
                            ce,
                            {
                              variant: "body1",
                              sx: { color: n.text },
                              children: a.title
                            }
                          )
                        }
                      ),
                      /* @__PURE__ */ s(
                        oe,
                        {
                          sx: {
                            backgroundColor: n.paper,
                            color: n.text
                          },
                          children: /* @__PURE__ */ s(
                            ie,
                            {
                              notification: i,
                              preferences: l || [],
                              updateDelivery: o.updateDelivery,
                              subNotificationId: a.subNotificationId
                            },
                            a.subNotificationId
                          )
                        }
                      )
                    ]
                  },
                  `${a.notificationId}-${a.subNotificationId}`
                ))
              ] })
            }
          )
        ]
      },
      i.notificationId
    );
  });
  return /* @__PURE__ */ s(
    Me,
    {
      sx: {
        borderRadius: 2,
        overflow: "hidden",
        backgroundColor: n.paper,
        color: n.text
      },
      children: c
    }
  );
}
export {
  mo as Preferences
};
