// @notificationapi/react/dist/utils/debug.js
var s = Object.defineProperty;
var a = (t, o, e) => o in t ? s(t, o, { enumerable: !0, configurable: !0, writable: !0, value: e }) : t[o] = e;
var n = (t, o, e) => a(t, typeof o != "symbol" ? o + "" : o, e);
class c {
  log() {
  }
  warn() {
  }
  error() {
  }
  group() {
  }
  groupEnd() {
  }
  table() {
  }
}
class g {
  constructor() {
    n(this, "prefix", "[NotificationAPI Debug]");
  }
  log(o, e) {
    console.log(`${this.prefix} ${o}`, e || "");
  }
  warn(o, e) {
    console.warn(`${this.prefix} ${o}`, e || "");
  }
  error(o, e, r) {
    console.group(`${this.prefix} ERROR: ${o}`), e && console.error("Error:", e), r && console.error("Context:", r), e instanceof Error && e.stack && console.error("Stack trace:", e.stack), console.groupEnd();
  }
  group(o) {
    console.group(`${this.prefix} ${o}`);
  }
  groupEnd() {
    console.groupEnd();
  }
  table(o) {
    console.table(o);
  }
}
const l = (t) => t ? new g() : new c(), p = (t, o, e) => ({
  method: t,
  endpoint: o,
  params: e,
  timestamp: (/* @__PURE__ */ new Date()).toISOString()
}), u = (t, o) => ({
  event: t,
  data: o,
  timestamp: (/* @__PURE__ */ new Date()).toISOString()
}), f = (t, o, e) => ({
  stateName: t,
  oldValue: o,
  newValue: e,
  timestamp: (/* @__PURE__ */ new Date()).toISOString()
});
export {
  l as createDebugLogger,
  p as formatApiCall,
  f as formatStateChange,
  u as formatWebSocketEvent
};
