// @notificationapi/react/dist/utils/theme.js
import { c } from "../assets/createTheme.js";
const d = {
  background: "#ffffff",
  paper: "#ffffff",
  text: "#000000",
  textSecondary: "#666666",
  border: "#dcdcdc",
  icon: "#000000",
  divider: "#e0e0e0"
}, a = {
  background: "#1e1e1e",
  paper: "#2d2d2d",
  text: "#ffffff",
  textSecondary: "#b0b0b0",
  border: "#404040",
  icon: "#ffffff",
  divider: "#404040"
};
function n(r) {
  if (r && typeof r == "object" && "palette" in r)
    return r;
  let t = "light", e = {};
  r === "dark" || r === "light" ? t = r : r && typeof r == "object" && (t = r.mode || "light", e = r.colors || {});
  const o = { ...t === "dark" ? a : d, ...e }, i = {
    palette: {
      mode: t,
      background: {
        default: o.background,
        paper: o.paper
      },
      text: {
        primary: o.text,
        secondary: o.textSecondary
      },
      divider: o.divider
    },
    components: {
      MuiPaper: {
        styleOverrides: {
          root: {
            backgroundColor: o.paper,
            color: o.text
          }
        }
      },
      MuiPopover: {
        styleOverrides: {
          paper: {
            backgroundColor: o.paper,
            color: o.text
          }
        }
      },
      MuiDialog: {
        styleOverrides: {
          paper: {
            backgroundColor: o.paper,
            color: o.text
          }
        }
      },
      MuiList: {
        styleOverrides: {
          root: {
            backgroundColor: o.paper,
            color: o.text
          }
        }
      },
      MuiListItem: {
        styleOverrides: {
          root: {
            backgroundColor: o.paper,
            color: o.text
          }
        }
      },
      MuiDivider: {
        styleOverrides: {
          root: {
            borderColor: o.divider
          }
        }
      }
    }
  };
  return c(i);
}
function f(r) {
  const e = r.palette.mode === "dark" ? a : d;
  return {
    background: r.palette.background.default || e.background,
    paper: r.palette.background.paper || e.paper,
    text: r.palette.text.primary || e.text,
    textSecondary: r.palette.text.secondary || e.textSecondary,
    border: e.border,
    icon: e.icon,
    divider: r.palette.divider || e.divider
  };
}
export {
  n as createNotificationAPITheme,
  f as getThemeColors
};
