// @notificationapi/react/dist/assets/Box.js
import { s as u, c as x, C as h, P as e } from "./createTheme.js";
import { T } from "./DefaultPropsProvider2.js";
import { b as y, c as B, g as N } from "./DefaultPropsProvider.js";
import * as C from "react";
import { jsx as b } from "react/jsx-runtime";
import { u as g } from "./useTheme.js";
import { e as v } from "./Typography.js";
function E(n = {}) {
  const {
    themeId: t,
    defaultTheme: m,
    defaultClassName: a = "MuiBox-root",
    generateClassName: r
  } = n, c = y("div", {
    shouldForwardProp: (o) => o !== "theme" && o !== "sx" && o !== "as"
  })(u);
  return /* @__PURE__ */ C.forwardRef(function(f, l) {
    const s = g(m), {
      className: i,
      component: p = "div",
      ...d
    } = v(f);
    return /* @__PURE__ */ b(c, {
      as: p,
      ref: l,
      className: B(i, r ? r(a) : a),
      theme: t && s[t] || s,
      ...d
    });
  });
}
const O = N("MuiBox", ["root"]), P = x(), j = E({
  themeId: T,
  defaultTheme: P,
  defaultClassName: O.root,
  generateClassName: h.generate
});
process.env.NODE_ENV !== "production" && (j.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * @ignore
   */
  children: e.node,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: e.elementType,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object])
});
export {
  j as B
};
