// @notificationapi/react/dist/assets/Button.js
import * as S from "react";
import { g as D, e as s, n as k, o as W, P as t, a as B } from "./createTheme.js";
import { g as U, s as g, m as G, u as _, c as L, a as Z, r as X } from "./DefaultPropsProvider.js";
import { b as j, B as Y } from "./ButtonBase.js";
import { c as h } from "./createSimplePaletteValueFilter.js";
import { c as m } from "./createSvgIcon.js";
import { jsx as r, jsxs as q } from "react/jsx-runtime";
import { P as oo } from "./Paper.js";
import { I as to } from "./IconButton.js";
import { r as eo } from "./DefaultPropsProvider2.js";
function no(o) {
  return D("MuiAlert", o);
}
const V = U("MuiAlert", ["root", "action", "icon", "message", "filled", "colorSuccess", "colorInfo", "colorWarning", "colorError", "filledSuccess", "filledInfo", "filledWarning", "filledError", "outlined", "outlinedSuccess", "outlinedInfo", "outlinedWarning", "outlinedError", "standard", "standardSuccess", "standardInfo", "standardWarning", "standardError"]), ao = m(/* @__PURE__ */ r("path", {
  d: "M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
}), "SuccessOutlined"), ro = m(/* @__PURE__ */ r("path", {
  d: "M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
}), "ReportProblemOutlined"), so = m(/* @__PURE__ */ r("path", {
  d: "M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
}), "ErrorOutline"), io = m(/* @__PURE__ */ r("path", {
  d: "M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
}), "InfoOutlined"), lo = m(/* @__PURE__ */ r("path", {
  d: "M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
}), "Close"), co = (o) => {
  const {
    variant: n,
    color: a,
    severity: e,
    classes: i
  } = o, c = {
    root: ["root", `color${s(a || e)}`, `${n}${s(a || e)}`, `${n}`],
    icon: ["icon"],
    message: ["message"],
    action: ["action"]
  };
  return Z(c, no, i);
}, po = g(oo, {
  name: "MuiAlert",
  slot: "Root",
  overridesResolver: (o, n) => {
    const {
      ownerState: a
    } = o;
    return [n.root, n[a.variant], n[`${a.variant}${s(a.color || a.severity)}`]];
  }
})(G(({
  theme: o
}) => {
  const n = o.palette.mode === "light" ? k : W, a = o.palette.mode === "light" ? W : k;
  return {
    ...o.typography.body2,
    backgroundColor: "transparent",
    display: "flex",
    padding: "6px 16px",
    variants: [...Object.entries(o.palette).filter(h(["light"])).map(([e]) => ({
      props: {
        colorSeverity: e,
        variant: "standard"
      },
      style: {
        color: o.vars ? o.vars.palette.Alert[`${e}Color`] : n(o.palette[e].light, 0.6),
        backgroundColor: o.vars ? o.vars.palette.Alert[`${e}StandardBg`] : a(o.palette[e].light, 0.9),
        [`& .${V.icon}`]: o.vars ? {
          color: o.vars.palette.Alert[`${e}IconColor`]
        } : {
          color: o.palette[e].main
        }
      }
    })), ...Object.entries(o.palette).filter(h(["light"])).map(([e]) => ({
      props: {
        colorSeverity: e,
        variant: "outlined"
      },
      style: {
        color: o.vars ? o.vars.palette.Alert[`${e}Color`] : n(o.palette[e].light, 0.6),
        border: `1px solid ${(o.vars || o).palette[e].light}`,
        [`& .${V.icon}`]: o.vars ? {
          color: o.vars.palette.Alert[`${e}IconColor`]
        } : {
          color: o.palette[e].main
        }
      }
    })), ...Object.entries(o.palette).filter(h(["dark"])).map(([e]) => ({
      props: {
        colorSeverity: e,
        variant: "filled"
      },
      style: {
        fontWeight: o.typography.fontWeightMedium,
        ...o.vars ? {
          color: o.vars.palette.Alert[`${e}FilledColor`],
          backgroundColor: o.vars.palette.Alert[`${e}FilledBg`]
        } : {
          backgroundColor: o.palette.mode === "dark" ? o.palette[e].dark : o.palette[e].main,
          color: o.palette.getContrastText(o.palette[e].main)
        }
      }
    }))]
  };
})), uo = g("div", {
  name: "MuiAlert",
  slot: "Icon",
  overridesResolver: (o, n) => n.icon
})({
  marginRight: 12,
  padding: "7px 0",
  display: "flex",
  fontSize: 22,
  opacity: 0.9
}), vo = g("div", {
  name: "MuiAlert",
  slot: "Message",
  overridesResolver: (o, n) => n.message
})({
  padding: "8px 0",
  minWidth: 0,
  overflow: "auto"
}), F = g("div", {
  name: "MuiAlert",
  slot: "Action",
  overridesResolver: (o, n) => n.action
})({
  display: "flex",
  alignItems: "flex-start",
  padding: "4px 0 0 16px",
  marginLeft: "auto",
  marginRight: -8
}), H = {
  success: /* @__PURE__ */ r(ao, {
    fontSize: "inherit"
  }),
  warning: /* @__PURE__ */ r(ro, {
    fontSize: "inherit"
  }),
  error: /* @__PURE__ */ r(so, {
    fontSize: "inherit"
  }),
  info: /* @__PURE__ */ r(io, {
    fontSize: "inherit"
  })
}, go = /* @__PURE__ */ S.forwardRef(function(n, a) {
  const e = _({
    props: n,
    name: "MuiAlert"
  }), {
    action: i,
    children: c,
    className: p,
    closeText: f = "Close",
    color: y,
    components: C = {},
    componentsProps: A = {},
    icon: z,
    iconMapping: M = H,
    onClose: I,
    role: $ = "alert",
    severity: b = "success",
    slotProps: T = {},
    slots: R = {},
    variant: O = "standard",
    ...w
  } = e, l = {
    ...e,
    color: y,
    severity: b,
    variant: O,
    colorSeverity: y || b
  }, d = co(l), u = {
    slots: {
      closeButton: C.CloseButton,
      closeIcon: C.CloseIcon,
      ...R
    },
    slotProps: {
      ...A,
      ...T
    }
  }, [v, E] = j("closeButton", {
    elementType: to,
    externalForwardedProps: u,
    ownerState: l
  }), [N, P] = j("closeIcon", {
    elementType: lo,
    externalForwardedProps: u,
    ownerState: l
  });
  return /* @__PURE__ */ q(po, {
    role: $,
    elevation: 0,
    ownerState: l,
    className: L(d.root, p),
    ref: a,
    ...w,
    children: [z !== !1 ? /* @__PURE__ */ r(uo, {
      ownerState: l,
      className: d.icon,
      children: z || M[b] || H[b]
    }) : null, /* @__PURE__ */ r(vo, {
      ownerState: l,
      className: d.message,
      children: c
    }), i != null ? /* @__PURE__ */ r(F, {
      ownerState: l,
      className: d.action,
      children: i
    }) : null, i == null && I ? /* @__PURE__ */ r(F, {
      ownerState: l,
      className: d.action,
      children: /* @__PURE__ */ r(v, {
        size: "small",
        "aria-label": f,
        title: f,
        color: "inherit",
        onClick: I,
        ...E,
        children: /* @__PURE__ */ r(N, {
          fontSize: "small",
          ...P
        })
      })
    }) : null]
  });
});
process.env.NODE_ENV !== "production" && (go.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The action to display. It renders after the message, at the end of the alert.
   */
  action: t.node,
  /**
   * The content of the component.
   */
  children: t.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: t.object,
  /**
   * @ignore
   */
  className: t.string,
  /**
   * Override the default label for the *close popup* icon button.
   *
   * For localization purposes, you can use the provided [translations](https://mui.com/material-ui/guides/localization/).
   * @default 'Close'
   */
  closeText: t.string,
  /**
   * The color of the component. Unless provided, the value is taken from the `severity` prop.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   */
  color: t.oneOfType([t.oneOf(["error", "info", "success", "warning"]), t.string]),
  /**
   * The components used for each slot inside.
   *
   * @deprecated use the `slots` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  components: t.shape({
    CloseButton: t.elementType,
    CloseIcon: t.elementType
  }),
  /**
   * The extra props for the slot components.
   * You can override the existing props or add new ones.
   *
   * @deprecated use the `slotProps` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  componentsProps: t.shape({
    closeButton: t.object,
    closeIcon: t.object
  }),
  /**
   * Override the icon displayed before the children.
   * Unless provided, the icon is mapped to the value of the `severity` prop.
   * Set to `false` to remove the `icon`.
   */
  icon: t.node,
  /**
   * The component maps the `severity` prop to a range of different icons,
   * for instance success to `<SuccessOutlined>`.
   * If you wish to change this mapping, you can provide your own.
   * Alternatively, you can use the `icon` prop to override the icon displayed.
   */
  iconMapping: t.shape({
    error: t.node,
    info: t.node,
    success: t.node,
    warning: t.node
  }),
  /**
   * Callback fired when the component requests to be closed.
   * When provided and no `action` prop is set, a close icon button is displayed that triggers the callback when clicked.
   * @param {React.SyntheticEvent} event The event source of the callback.
   */
  onClose: t.func,
  /**
   * The ARIA role attribute of the element.
   * @default 'alert'
   */
  role: t.string,
  /**
   * The severity of the alert. This defines the color and icon used.
   * @default 'success'
   */
  severity: t.oneOfType([t.oneOf(["error", "info", "success", "warning"]), t.string]),
  /**
   * The props used for each slot inside.
   * @default {}
   */
  slotProps: t.shape({
    closeButton: t.oneOfType([t.func, t.object]),
    closeIcon: t.oneOfType([t.func, t.object])
  }),
  /**
   * The components used for each slot inside.
   * @default {}
   */
  slots: t.shape({
    closeButton: t.elementType,
    closeIcon: t.elementType
  }),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: t.oneOfType([t.arrayOf(t.oneOfType([t.func, t.object, t.bool])), t.func, t.object]),
  /**
   * The variant to use.
   * @default 'standard'
   */
  variant: t.oneOfType([t.oneOf(["filled", "outlined", "standard"]), t.string])
});
function fo(o) {
  return D("MuiButton", o);
}
const x = U("MuiButton", ["root", "text", "textInherit", "textPrimary", "textSecondary", "textSuccess", "textError", "textInfo", "textWarning", "outlined", "outlinedInherit", "outlinedPrimary", "outlinedSecondary", "outlinedSuccess", "outlinedError", "outlinedInfo", "outlinedWarning", "contained", "containedInherit", "containedPrimary", "containedSecondary", "containedSuccess", "containedError", "containedInfo", "containedWarning", "disableElevation", "focusVisible", "disabled", "colorInherit", "colorPrimary", "colorSecondary", "colorSuccess", "colorError", "colorInfo", "colorWarning", "textSizeSmall", "textSizeMedium", "textSizeLarge", "outlinedSizeSmall", "outlinedSizeMedium", "outlinedSizeLarge", "containedSizeSmall", "containedSizeMedium", "containedSizeLarge", "sizeMedium", "sizeSmall", "sizeLarge", "fullWidth", "startIcon", "endIcon", "icon", "iconSizeSmall", "iconSizeMedium", "iconSizeLarge"]), J = /* @__PURE__ */ S.createContext({});
process.env.NODE_ENV !== "production" && (J.displayName = "ButtonGroupContext");
const K = /* @__PURE__ */ S.createContext(void 0);
process.env.NODE_ENV !== "production" && (K.displayName = "ButtonGroupButtonContext");
const yo = (o) => {
  const {
    color: n,
    disableElevation: a,
    fullWidth: e,
    size: i,
    variant: c,
    classes: p
  } = o, f = {
    root: ["root", c, `${c}${s(n)}`, `size${s(i)}`, `${c}Size${s(i)}`, `color${s(n)}`, a && "disableElevation", e && "fullWidth"],
    label: ["label"],
    startIcon: ["icon", "startIcon", `iconSize${s(i)}`],
    endIcon: ["icon", "endIcon", `iconSize${s(i)}`]
  }, y = Z(f, fo, p);
  return {
    ...p,
    // forward the focused, disabled, etc. classes to the ButtonBase
    ...y
  };
}, Q = [{
  props: {
    size: "small"
  },
  style: {
    "& > *:nth-of-type(1)": {
      fontSize: 18
    }
  }
}, {
  props: {
    size: "medium"
  },
  style: {
    "& > *:nth-of-type(1)": {
      fontSize: 20
    }
  }
}, {
  props: {
    size: "large"
  },
  style: {
    "& > *:nth-of-type(1)": {
      fontSize: 22
    }
  }
}], bo = g(Y, {
  shouldForwardProp: (o) => X(o) || o === "classes",
  name: "MuiButton",
  slot: "Root",
  overridesResolver: (o, n) => {
    const {
      ownerState: a
    } = o;
    return [n.root, n[a.variant], n[`${a.variant}${s(a.color)}`], n[`size${s(a.size)}`], n[`${a.variant}Size${s(a.size)}`], a.color === "inherit" && n.colorInherit, a.disableElevation && n.disableElevation, a.fullWidth && n.fullWidth];
  }
})(G(({
  theme: o
}) => {
  const n = o.palette.mode === "light" ? o.palette.grey[300] : o.palette.grey[800], a = o.palette.mode === "light" ? o.palette.grey.A100 : o.palette.grey[700];
  return {
    ...o.typography.button,
    minWidth: 64,
    padding: "6px 16px",
    border: 0,
    borderRadius: (o.vars || o).shape.borderRadius,
    transition: o.transitions.create(["background-color", "box-shadow", "border-color", "color"], {
      duration: o.transitions.duration.short
    }),
    "&:hover": {
      textDecoration: "none"
    },
    [`&.${x.disabled}`]: {
      color: (o.vars || o).palette.action.disabled
    },
    variants: [{
      props: {
        variant: "contained"
      },
      style: {
        color: "var(--variant-containedColor)",
        backgroundColor: "var(--variant-containedBg)",
        boxShadow: (o.vars || o).shadows[2],
        "&:hover": {
          boxShadow: (o.vars || o).shadows[4],
          // Reset on touch devices, it doesn't add specificity
          "@media (hover: none)": {
            boxShadow: (o.vars || o).shadows[2]
          }
        },
        "&:active": {
          boxShadow: (o.vars || o).shadows[8]
        },
        [`&.${x.focusVisible}`]: {
          boxShadow: (o.vars || o).shadows[6]
        },
        [`&.${x.disabled}`]: {
          color: (o.vars || o).palette.action.disabled,
          boxShadow: (o.vars || o).shadows[0],
          backgroundColor: (o.vars || o).palette.action.disabledBackground
        }
      }
    }, {
      props: {
        variant: "outlined"
      },
      style: {
        padding: "5px 15px",
        border: "1px solid currentColor",
        borderColor: "var(--variant-outlinedBorder, currentColor)",
        backgroundColor: "var(--variant-outlinedBg)",
        color: "var(--variant-outlinedColor)",
        [`&.${x.disabled}`]: {
          border: `1px solid ${(o.vars || o).palette.action.disabledBackground}`
        }
      }
    }, {
      props: {
        variant: "text"
      },
      style: {
        padding: "6px 8px",
        color: "var(--variant-textColor)",
        backgroundColor: "var(--variant-textBg)"
      }
    }, ...Object.entries(o.palette).filter(h()).map(([e]) => ({
      props: {
        color: e
      },
      style: {
        "--variant-textColor": (o.vars || o).palette[e].main,
        "--variant-outlinedColor": (o.vars || o).palette[e].main,
        "--variant-outlinedBorder": o.vars ? `rgba(${o.vars.palette[e].mainChannel} / 0.5)` : B(o.palette[e].main, 0.5),
        "--variant-containedColor": (o.vars || o).palette[e].contrastText,
        "--variant-containedBg": (o.vars || o).palette[e].main,
        "@media (hover: hover)": {
          "&:hover": {
            "--variant-containedBg": (o.vars || o).palette[e].dark,
            "--variant-textBg": o.vars ? `rgba(${o.vars.palette[e].mainChannel} / ${o.vars.palette.action.hoverOpacity})` : B(o.palette[e].main, o.palette.action.hoverOpacity),
            "--variant-outlinedBorder": (o.vars || o).palette[e].main,
            "--variant-outlinedBg": o.vars ? `rgba(${o.vars.palette[e].mainChannel} / ${o.vars.palette.action.hoverOpacity})` : B(o.palette[e].main, o.palette.action.hoverOpacity)
          }
        }
      }
    })), {
      props: {
        color: "inherit"
      },
      style: {
        color: "inherit",
        borderColor: "currentColor",
        "--variant-containedBg": o.vars ? o.vars.palette.Button.inheritContainedBg : n,
        "@media (hover: hover)": {
          "&:hover": {
            "--variant-containedBg": o.vars ? o.vars.palette.Button.inheritContainedHoverBg : a,
            "--variant-textBg": o.vars ? `rgba(${o.vars.palette.text.primaryChannel} / ${o.vars.palette.action.hoverOpacity})` : B(o.palette.text.primary, o.palette.action.hoverOpacity),
            "--variant-outlinedBg": o.vars ? `rgba(${o.vars.palette.text.primaryChannel} / ${o.vars.palette.action.hoverOpacity})` : B(o.palette.text.primary, o.palette.action.hoverOpacity)
          }
        }
      }
    }, {
      props: {
        size: "small",
        variant: "text"
      },
      style: {
        padding: "4px 5px",
        fontSize: o.typography.pxToRem(13)
      }
    }, {
      props: {
        size: "large",
        variant: "text"
      },
      style: {
        padding: "8px 11px",
        fontSize: o.typography.pxToRem(15)
      }
    }, {
      props: {
        size: "small",
        variant: "outlined"
      },
      style: {
        padding: "3px 9px",
        fontSize: o.typography.pxToRem(13)
      }
    }, {
      props: {
        size: "large",
        variant: "outlined"
      },
      style: {
        padding: "7px 21px",
        fontSize: o.typography.pxToRem(15)
      }
    }, {
      props: {
        size: "small",
        variant: "contained"
      },
      style: {
        padding: "4px 10px",
        fontSize: o.typography.pxToRem(13)
      }
    }, {
      props: {
        size: "large",
        variant: "contained"
      },
      style: {
        padding: "8px 22px",
        fontSize: o.typography.pxToRem(15)
      }
    }, {
      props: {
        disableElevation: !0
      },
      style: {
        boxShadow: "none",
        "&:hover": {
          boxShadow: "none"
        },
        [`&.${x.focusVisible}`]: {
          boxShadow: "none"
        },
        "&:active": {
          boxShadow: "none"
        },
        [`&.${x.disabled}`]: {
          boxShadow: "none"
        }
      }
    }, {
      props: {
        fullWidth: !0
      },
      style: {
        width: "100%"
      }
    }]
  };
})), xo = g("span", {
  name: "MuiButton",
  slot: "StartIcon",
  overridesResolver: (o, n) => {
    const {
      ownerState: a
    } = o;
    return [n.startIcon, n[`iconSize${s(a.size)}`]];
  }
})({
  display: "inherit",
  marginRight: 8,
  marginLeft: -4,
  variants: [{
    props: {
      size: "small"
    },
    style: {
      marginLeft: -2
    }
  }, ...Q]
}), So = g("span", {
  name: "MuiButton",
  slot: "EndIcon",
  overridesResolver: (o, n) => {
    const {
      ownerState: a
    } = o;
    return [n.endIcon, n[`iconSize${s(a.size)}`]];
  }
})({
  display: "inherit",
  marginRight: -4,
  marginLeft: 8,
  variants: [{
    props: {
      size: "small"
    },
    style: {
      marginRight: -2
    }
  }, ...Q]
}), Co = /* @__PURE__ */ S.forwardRef(function(n, a) {
  const e = S.useContext(J), i = S.useContext(K), c = eo(e, n), p = _({
    props: c,
    name: "MuiButton"
  }), {
    children: f,
    color: y = "primary",
    component: C = "button",
    className: A,
    disabled: z = !1,
    disableElevation: M = !1,
    disableFocusRipple: I = !1,
    endIcon: $,
    focusVisibleClassName: b,
    fullWidth: T = !1,
    size: R = "medium",
    startIcon: O,
    type: w,
    variant: l = "text",
    ...d
  } = p, u = {
    ...p,
    color: y,
    component: C,
    disabled: z,
    disableElevation: M,
    disableFocusRipple: I,
    fullWidth: T,
    size: R,
    type: w,
    variant: l
  }, v = yo(u), E = O && /* @__PURE__ */ r(xo, {
    className: v.startIcon,
    ownerState: u,
    children: O
  }), N = $ && /* @__PURE__ */ r(So, {
    className: v.endIcon,
    ownerState: u,
    children: $
  }), P = i || "";
  return /* @__PURE__ */ q(bo, {
    ownerState: u,
    className: L(e.className, v.root, A, P),
    component: C,
    disabled: z,
    focusRipple: !I,
    focusVisibleClassName: L(v.focusVisible, b),
    ref: a,
    type: w,
    ...d,
    classes: v,
    children: [E, f, N]
  });
});
process.env.NODE_ENV !== "production" && (Co.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: t.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: t.object,
  /**
   * @ignore
   */
  className: t.string,
  /**
   * The color of the component.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   * @default 'primary'
   */
  color: t.oneOfType([t.oneOf(["inherit", "primary", "secondary", "success", "error", "info", "warning"]), t.string]),
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: t.elementType,
  /**
   * If `true`, the component is disabled.
   * @default false
   */
  disabled: t.bool,
  /**
   * If `true`, no elevation is used.
   * @default false
   */
  disableElevation: t.bool,
  /**
   * If `true`, the  keyboard focus ripple is disabled.
   * @default false
   */
  disableFocusRipple: t.bool,
  /**
   * If `true`, the ripple effect is disabled.
   *
   * ⚠️ Without a ripple there is no styling for :focus-visible by default. Be sure
   * to highlight the element by applying separate styles with the `.Mui-focusVisible` class.
   * @default false
   */
  disableRipple: t.bool,
  /**
   * Element placed after the children.
   */
  endIcon: t.node,
  /**
   * @ignore
   */
  focusVisibleClassName: t.string,
  /**
   * If `true`, the button will take up the full width of its container.
   * @default false
   */
  fullWidth: t.bool,
  /**
   * The URL to link to when the button is clicked.
   * If defined, an `a` element will be used as the root node.
   */
  href: t.string,
  /**
   * The size of the component.
   * `small` is equivalent to the dense button styling.
   * @default 'medium'
   */
  size: t.oneOfType([t.oneOf(["small", "medium", "large"]), t.string]),
  /**
   * Element placed before the children.
   */
  startIcon: t.node,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: t.oneOfType([t.arrayOf(t.oneOfType([t.func, t.object, t.bool])), t.func, t.object]),
  /**
   * @ignore
   */
  type: t.oneOfType([t.oneOf(["button", "reset", "submit"]), t.string]),
  /**
   * The variant to use.
   * @default 'text'
   */
  variant: t.oneOfType([t.oneOf(["contained", "outlined", "text"]), t.string])
});
export {
  go as A,
  Co as B
};
