// @notificationapi/react/dist/assets/ButtonBase.js
var Le = Object.defineProperty;
var Be = (e, t, n) => t in e ? Le(e, t, { enumerable: !0, configurable: !0, writable: !0, value: n }) : e[t] = n;
var K = (e, t, n) => Be(e, typeof t != "symbol" ? t + "" : t, n);
import * as p from "react";
import q, { Children as je, isValidElement as G, cloneElement as Z } from "react";
import { d as ke, _ as $e, c as x, g as de, s as re, u as he, a as Fe } from "./DefaultPropsProvider.js";
import { T as _e } from "./DefaultPropsProvider2.js";
import { u as Ue } from "./useTheme.js";
import { r as ze, m as Ae, u as te, a as Ye } from "./resolveComponentProps.js";
import { P as o, g as Xe } from "./createTheme.js";
import { jsx as _, jsxs as Ke } from "react/jsx-runtime";
import { k as se } from "./useThemeWithoutDefault.js";
const qe = typeof window < "u" ? p.useLayoutEffect : p.useEffect;
function He(e, t) {
  return process.env.NODE_ENV === "production" ? () => null : function(...i) {
    return e(...i) || t(...i);
  };
}
function We(e) {
  const {
    prototype: t = {}
  } = e;
  return !!t.isReactComponent;
}
function Ge(e, t, n, i, r) {
  const s = e[t], u = r || t;
  if (s == null || // When server-side rendering React doesn't warn either.
  // This is not an accurate check for SSR.
  // This is only in place for emotion compat.
  // TODO: Revisit once https://github.com/facebook/react/issues/20047 is resolved.
  typeof window > "u")
    return null;
  let c;
  return typeof s == "function" && !We(s) && (c = "Did you accidentally provide a plain function component instead?"), c !== void 0 ? new Error(`Invalid ${i} \`${u}\` supplied to \`${n}\`. Expected an element type that can hold a ref. ${c} For more information see https://mui.com/r/caveat-with-refs-guide`) : null;
}
const Ze = He(o.elementType, Ge), Je = o.oneOfType([o.func, o.object]);
function J(e) {
  const t = p.useRef(e);
  return qe(() => {
    t.current = e;
  }), p.useRef((...n) => (
    // @ts-expect-error hide `this`
    (0, t.current)(...n)
  )).current;
}
const ce = {};
function me(e, t) {
  const n = p.useRef(ce);
  return n.current === ce && (n.current = e(t)), n;
}
const Qe = [];
function et(e) {
  p.useEffect(e, Qe);
}
class ie {
  constructor() {
    K(this, "currentId", null);
    K(this, "clear", () => {
      this.currentId !== null && (clearTimeout(this.currentId), this.currentId = null);
    });
    K(this, "disposeEffect", () => this.clear);
  }
  static create() {
    return new ie();
  }
  /**
   * Executes `fn` after `delay`, clearing any previously scheduled call.
   */
  start(t, n) {
    this.clear(), this.currentId = setTimeout(() => {
      this.currentId = null, n();
    }, t);
  }
}
function tt() {
  const e = me(ie.create).current;
  return et(e.disposeEffect), e;
}
function pe(e) {
  try {
    return e.matches(":focus-visible");
  } catch {
    process.env.NODE_ENV !== "production" && !/jsdom/.test(window.navigator.userAgent) && console.warn(["MUI: The `:focus-visible` pseudo class is not supported in this browser.", "Some components rely on this feature to work properly."].join(`
`));
  }
  return !1;
}
function Vt() {
  const e = Ue(ke);
  return process.env.NODE_ENV !== "production" && p.useDebugValue(e), e[_e] || e;
}
function nt(e, t) {
  if (e == null) return {};
  var n = {};
  for (var i in e) if ({}.hasOwnProperty.call(e, i)) {
    if (t.indexOf(i) !== -1) continue;
    n[i] = e[i];
  }
  return n;
}
function ne(e, t) {
  return ne = Object.setPrototypeOf ? Object.setPrototypeOf.bind() : function(n, i) {
    return n.__proto__ = i, n;
  }, ne(e, t);
}
function ot(e, t) {
  e.prototype = Object.create(t.prototype), e.prototype.constructor = e, ne(e, t);
}
const fe = q.createContext(null);
function rt(e) {
  if (e === void 0) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
  return e;
}
function ue(e, t) {
  var n = function(s) {
    return t && G(s) ? t(s) : s;
  }, i = /* @__PURE__ */ Object.create(null);
  return e && je.map(e, function(r) {
    return r;
  }).forEach(function(r) {
    i[r.key] = n(r);
  }), i;
}
function st(e, t) {
  e = e || {}, t = t || {};
  function n(h) {
    return h in t ? t[h] : e[h];
  }
  var i = /* @__PURE__ */ Object.create(null), r = [];
  for (var s in e)
    s in t ? r.length && (i[s] = r, r = []) : r.push(s);
  var u, c = {};
  for (var l in t) {
    if (i[l])
      for (u = 0; u < i[l].length; u++) {
        var f = i[l][u];
        c[i[l][u]] = n(f);
      }
    c[l] = n(l);
  }
  for (u = 0; u < r.length; u++)
    c[r[u]] = n(r[u]);
  return c;
}
function $(e, t, n) {
  return n[t] != null ? n[t] : e.props[t];
}
function it(e, t) {
  return ue(e.children, function(n) {
    return Z(n, {
      onExited: t.bind(null, n),
      in: !0,
      appear: $(n, "appear", e),
      enter: $(n, "enter", e),
      exit: $(n, "exit", e)
    });
  });
}
function ut(e, t, n) {
  var i = ue(e.children), r = st(t, i);
  return Object.keys(r).forEach(function(s) {
    var u = r[s];
    if (G(u)) {
      var c = s in t, l = s in i, f = t[s], h = G(f) && !f.props.in;
      l && (!c || h) ? r[s] = Z(u, {
        onExited: n.bind(null, u),
        in: !0,
        exit: $(u, "exit", e),
        enter: $(u, "enter", e)
      }) : !l && c && !h ? r[s] = Z(u, {
        in: !1
      }) : l && c && G(f) && (r[s] = Z(u, {
        onExited: n.bind(null, u),
        in: f.props.in,
        exit: $(u, "exit", e),
        enter: $(u, "enter", e)
      }));
    }
  }), r;
}
var at = Object.values || function(e) {
  return Object.keys(e).map(function(t) {
    return e[t];
  });
}, lt = {
  component: "div",
  childFactory: function(t) {
    return t;
  }
}, ae = /* @__PURE__ */ function(e) {
  ot(t, e);
  function t(i, r) {
    var s;
    s = e.call(this, i, r) || this;
    var u = s.handleExited.bind(rt(s));
    return s.state = {
      contextValue: {
        isMounting: !0
      },
      handleExited: u,
      firstRender: !0
    }, s;
  }
  var n = t.prototype;
  return n.componentDidMount = function() {
    this.mounted = !0, this.setState({
      contextValue: {
        isMounting: !1
      }
    });
  }, n.componentWillUnmount = function() {
    this.mounted = !1;
  }, t.getDerivedStateFromProps = function(r, s) {
    var u = s.children, c = s.handleExited, l = s.firstRender;
    return {
      children: l ? it(r, c) : ut(r, u, c),
      firstRender: !1
    };
  }, n.handleExited = function(r, s) {
    var u = ue(this.props.children);
    r.key in u || (r.props.onExited && r.props.onExited(s), this.mounted && this.setState(function(c) {
      var l = $e({}, c.children);
      return delete l[r.key], {
        children: l
      };
    }));
  }, n.render = function() {
    var r = this.props, s = r.component, u = r.childFactory, c = nt(r, ["component", "childFactory"]), l = this.state.contextValue, f = at(this.state.children).map(u);
    return delete c.appear, delete c.enter, delete c.exit, s === null ? /* @__PURE__ */ q.createElement(fe.Provider, {
      value: l
    }, f) : /* @__PURE__ */ q.createElement(fe.Provider, {
      value: l
    }, /* @__PURE__ */ q.createElement(s, c, f));
  }, t;
}(q.Component);
ae.propTypes = process.env.NODE_ENV !== "production" ? {
  /**
   * `<TransitionGroup>` renders a `<div>` by default. You can change this
   * behavior by providing a `component` prop.
   * If you use React v16+ and would like to avoid a wrapping `<div>` element
   * you can pass in `component={null}`. This is useful if the wrapping div
   * borks your css styles.
   */
  component: o.any,
  /**
   * A set of `<Transition>` components, that are toggled `in` and out as they
   * leave. the `<TransitionGroup>` will inject specific transition props, so
   * remember to spread them through if you are wrapping the `<Transition>` as
   * with our `<Fade>` example.
   *
   * While this component is meant for multiple `Transition` or `CSSTransition`
   * children, sometimes you may want to have a single transition child with
   * content that you want to be transitioned out and in when you change it
   * (e.g. routes, images etc.) In that case you can change the `key` prop of
   * the transition child as you change its content, this will cause
   * `TransitionGroup` to transition the child out and back in.
   */
  children: o.node,
  /**
   * A convenience prop that enables or disables appear animations
   * for all children. Note that specifying this will override any defaults set
   * on individual children Transitions.
   */
  appear: o.bool,
  /**
   * A convenience prop that enables or disables enter animations
   * for all children. Note that specifying this will override any defaults set
   * on individual children Transitions.
   */
  enter: o.bool,
  /**
   * A convenience prop that enables or disables exit animations
   * for all children. Note that specifying this will override any defaults set
   * on individual children Transitions.
   */
  exit: o.bool,
  /**
   * You may need to apply reactive updates to a child as it is exiting.
   * This is generally done by using `cloneElement` however in the case of an exiting
   * child the element has already been removed and not accessible to the consumer.
   *
   * If you do need to update a child as it leaves you can provide a `childFactory`
   * to wrap every child, even the ones that are leaving.
   *
   * @type Function(child: ReactElement) -> ReactElement
   */
  childFactory: o.func
} : {};
ae.defaultProps = lt;
function It(e, t) {
  const {
    className: n,
    elementType: i,
    ownerState: r,
    externalForwardedProps: s,
    internalForwardedProps: u,
    ...c
  } = t, {
    component: l,
    slots: f = {
      [e]: void 0
    },
    slotProps: h = {
      [e]: void 0
    },
    ...m
  } = s, y = f[e] || i, T = ze(h[e], r), {
    props: {
      component: g,
      ...w
    },
    internalRef: M
  } = Ae({
    className: n,
    ...c,
    externalForwardedProps: e === "root" ? m : void 0,
    externalSlotProps: T
  }), D = te(M, T == null ? void 0 : T.ref, t.ref), L = e === "root" ? g || l : g, S = Ye(y, {
    ...e === "root" && !l && !f[e] && u,
    ...e !== "root" && !f[e] && u,
    ...w,
    ...L && {
      as: L
    },
    ref: D
  }, r);
  return [y, S];
}
class Q {
  constructor() {
    K(this, "mountEffect", () => {
      this.shouldMount && !this.didMount && this.ref.current !== null && (this.didMount = !0, this.mounted.resolve());
    });
    this.ref = {
      current: null
    }, this.mounted = null, this.didMount = !1, this.shouldMount = !1, this.setShouldMount = null;
  }
  /** React ref to the ripple instance */
  /** If the ripple component should be mounted */
  /** Promise that resolves when the ripple component is mounted */
  /** If the ripple component has been mounted */
  /** React state hook setter */
  static create() {
    return new Q();
  }
  static use() {
    const t = me(Q.create).current, [n, i] = p.useState(!1);
    return t.shouldMount = n, t.setShouldMount = i, p.useEffect(t.mountEffect, [n]), t;
  }
  mount() {
    return this.mounted || (this.mounted = pt(), this.shouldMount = !0, this.setShouldMount(this.shouldMount)), this.mounted;
  }
  /* Ripple API */
  start(...t) {
    this.mount().then(() => {
      var n;
      return (n = this.ref.current) == null ? void 0 : n.start(...t);
    });
  }
  stop(...t) {
    this.mount().then(() => {
      var n;
      return (n = this.ref.current) == null ? void 0 : n.stop(...t);
    });
  }
  pulsate(...t) {
    this.mount().then(() => {
      var n;
      return (n = this.ref.current) == null ? void 0 : n.pulsate(...t);
    });
  }
}
function ct() {
  return Q.use();
}
function pt() {
  let e, t;
  const n = new Promise((i, r) => {
    e = i, t = r;
  });
  return n.resolve = e, n.reject = t, n;
}
function be(e) {
  const {
    className: t,
    classes: n,
    pulsate: i = !1,
    rippleX: r,
    rippleY: s,
    rippleSize: u,
    in: c,
    onExited: l,
    timeout: f
  } = e, [h, m] = p.useState(!1), y = x(t, n.ripple, n.rippleVisible, i && n.ripplePulsate), T = {
    width: u,
    height: u,
    top: -(u / 2) + s,
    left: -(u / 2) + r
  }, g = x(n.child, h && n.childLeaving, i && n.childPulsate);
  return !c && !h && m(!0), p.useEffect(() => {
    if (!c && l != null) {
      const w = setTimeout(l, f);
      return () => {
        clearTimeout(w);
      };
    }
  }, [l, c, f]), /* @__PURE__ */ _("span", {
    className: y,
    style: T,
    children: /* @__PURE__ */ _("span", {
      className: g
    })
  });
}
process.env.NODE_ENV !== "production" && (be.propTypes = {
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object.isRequired,
  className: o.string,
  /**
   * @ignore - injected from TransitionGroup
   */
  in: o.bool,
  /**
   * @ignore - injected from TransitionGroup
   */
  onExited: o.func,
  /**
   * If `true`, the ripple pulsates, typically indicating the keyboard focus state of an element.
   */
  pulsate: o.bool,
  /**
   * Diameter of the ripple.
   */
  rippleSize: o.number,
  /**
   * Horizontal position of the ripple center.
   */
  rippleX: o.number,
  /**
   * Vertical position of the ripple center.
   */
  rippleY: o.number,
  /**
   * exit delay
   */
  timeout: o.number.isRequired
});
const R = de("MuiTouchRipple", ["root", "ripple", "rippleVisible", "ripplePulsate", "child", "childLeaving", "childPulsate"]), oe = 550, ft = 80, dt = se`
  0% {
    transform: scale(0);
    opacity: 0.1;
  }

  100% {
    transform: scale(1);
    opacity: 0.3;
  }
`, ht = se`
  0% {
    opacity: 1;
  }

  100% {
    opacity: 0;
  }
`, mt = se`
  0% {
    transform: scale(1);
  }

  50% {
    transform: scale(0.92);
  }

  100% {
    transform: scale(1);
  }
`, bt = re("span", {
  name: "MuiTouchRipple",
  slot: "Root"
})({
  overflow: "hidden",
  pointerEvents: "none",
  position: "absolute",
  zIndex: 0,
  top: 0,
  right: 0,
  bottom: 0,
  left: 0,
  borderRadius: "inherit"
}), gt = re(be, {
  name: "MuiTouchRipple",
  slot: "Ripple"
})`
  opacity: 0;
  position: absolute;

  &.${R.rippleVisible} {
    opacity: 0.3;
    transform: scale(1);
    animation-name: ${dt};
    animation-duration: ${oe}ms;
    animation-timing-function: ${({
  theme: e
}) => e.transitions.easing.easeInOut};
  }

  &.${R.ripplePulsate} {
    animation-duration: ${({
  theme: e
}) => e.transitions.duration.shorter}ms;
  }

  & .${R.child} {
    opacity: 1;
    display: block;
    width: 100%;
    height: 100%;
    border-radius: 50%;
    background-color: currentColor;
  }

  & .${R.childLeaving} {
    opacity: 0;
    animation-name: ${ht};
    animation-duration: ${oe}ms;
    animation-timing-function: ${({
  theme: e
}) => e.transitions.easing.easeInOut};
  }

  & .${R.childPulsate} {
    position: absolute;
    /* @noflip */
    left: 0px;
    top: 0;
    animation-name: ${mt};
    animation-duration: 2500ms;
    animation-timing-function: ${({
  theme: e
}) => e.transitions.easing.easeInOut};
    animation-iteration-count: infinite;
    animation-delay: 200ms;
  }
`, ge = /* @__PURE__ */ p.forwardRef(function(t, n) {
  const i = he({
    props: t,
    name: "MuiTouchRipple"
  }), {
    center: r = !1,
    classes: s = {},
    className: u,
    ...c
  } = i, [l, f] = p.useState([]), h = p.useRef(0), m = p.useRef(null);
  p.useEffect(() => {
    m.current && (m.current(), m.current = null);
  }, [l]);
  const y = p.useRef(!1), T = tt(), g = p.useRef(null), w = p.useRef(null), M = p.useCallback((d) => {
    const {
      pulsate: C,
      rippleX: E,
      rippleY: F,
      rippleSize: B,
      cb: U
    } = d;
    f((P) => [...P, /* @__PURE__ */ _(gt, {
      classes: {
        ripple: x(s.ripple, R.ripple),
        rippleVisible: x(s.rippleVisible, R.rippleVisible),
        ripplePulsate: x(s.ripplePulsate, R.ripplePulsate),
        child: x(s.child, R.child),
        childLeaving: x(s.childLeaving, R.childLeaving),
        childPulsate: x(s.childPulsate, R.childPulsate)
      },
      timeout: oe,
      pulsate: C,
      rippleX: E,
      rippleY: F,
      rippleSize: B
    }, h.current)]), h.current += 1, m.current = U;
  }, [s]), D = p.useCallback((d = {}, C = {}, E = () => {
  }) => {
    const {
      pulsate: F = !1,
      center: B = r || C.pulsate,
      fakeElement: U = !1
      // For test purposes
    } = C;
    if ((d == null ? void 0 : d.type) === "mousedown" && y.current) {
      y.current = !1;
      return;
    }
    (d == null ? void 0 : d.type) === "touchstart" && (y.current = !0);
    const P = U ? null : w.current, O = P ? P.getBoundingClientRect() : {
      width: 0,
      height: 0,
      left: 0,
      top: 0
    };
    let V, N, I;
    if (B || d === void 0 || d.clientX === 0 && d.clientY === 0 || !d.clientX && !d.touches)
      V = Math.round(O.width / 2), N = Math.round(O.height / 2);
    else {
      const {
        clientX: z,
        clientY: j
      } = d.touches && d.touches.length > 0 ? d.touches[0] : d;
      V = Math.round(z - O.left), N = Math.round(j - O.top);
    }
    if (B)
      I = Math.sqrt((2 * O.width ** 2 + O.height ** 2) / 3), I % 2 === 0 && (I += 1);
    else {
      const z = Math.max(Math.abs((P ? P.clientWidth : 0) - V), V) * 2 + 2, j = Math.max(Math.abs((P ? P.clientHeight : 0) - N), N) * 2 + 2;
      I = Math.sqrt(z ** 2 + j ** 2);
    }
    d != null && d.touches ? g.current === null && (g.current = () => {
      M({
        pulsate: F,
        rippleX: V,
        rippleY: N,
        rippleSize: I,
        cb: E
      });
    }, T.start(ft, () => {
      g.current && (g.current(), g.current = null);
    })) : M({
      pulsate: F,
      rippleX: V,
      rippleY: N,
      rippleSize: I,
      cb: E
    });
  }, [r, M, T]), L = p.useCallback(() => {
    D({}, {
      pulsate: !0
    });
  }, [D]), S = p.useCallback((d, C) => {
    if (T.clear(), (d == null ? void 0 : d.type) === "touchend" && g.current) {
      g.current(), g.current = null, T.start(0, () => {
        S(d, C);
      });
      return;
    }
    g.current = null, f((E) => E.length > 0 ? E.slice(1) : E), m.current = C;
  }, [T]);
  return p.useImperativeHandle(n, () => ({
    pulsate: L,
    start: D,
    stop: S
  }), [L, D, S]), /* @__PURE__ */ _(bt, {
    className: x(R.root, s.root, u),
    ref: w,
    ...c,
    children: /* @__PURE__ */ _(ae, {
      component: null,
      exit: !0,
      children: l
    })
  });
});
process.env.NODE_ENV !== "production" && (ge.propTypes = {
  /**
   * If `true`, the ripple starts at the center of the component
   * rather than at the point of interaction.
   */
  center: o.bool,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string
});
function yt(e) {
  return Xe("MuiButtonBase", e);
}
const Tt = de("MuiButtonBase", ["root", "disabled", "focusVisible"]), Rt = (e) => {
  const {
    disabled: t,
    focusVisible: n,
    focusVisibleClassName: i,
    classes: r
  } = e, u = Fe({
    root: ["root", t && "disabled", n && "focusVisible"]
  }, yt, r);
  return n && i && (u.root += ` ${i}`), u;
}, Mt = re("button", {
  name: "MuiButtonBase",
  slot: "Root",
  overridesResolver: (e, t) => t.root
})({
  display: "inline-flex",
  alignItems: "center",
  justifyContent: "center",
  position: "relative",
  boxSizing: "border-box",
  WebkitTapHighlightColor: "transparent",
  backgroundColor: "transparent",
  // Reset default value
  // We disable the focus ring for mouse, touch and keyboard users.
  outline: 0,
  border: 0,
  margin: 0,
  // Remove the margin in Safari
  borderRadius: 0,
  padding: 0,
  // Remove the padding in Firefox
  cursor: "pointer",
  userSelect: "none",
  verticalAlign: "middle",
  MozAppearance: "none",
  // Reset
  WebkitAppearance: "none",
  // Reset
  textDecoration: "none",
  // So we take precedent over the style of a native <a /> element.
  color: "inherit",
  "&::-moz-focus-inner": {
    borderStyle: "none"
    // Remove Firefox dotted outline.
  },
  [`&.${Tt.disabled}`]: {
    pointerEvents: "none",
    // Disable link interactions
    cursor: "default"
  },
  "@media print": {
    colorAdjust: "exact"
  }
}), Et = /* @__PURE__ */ p.forwardRef(function(t, n) {
  const i = he({
    props: t,
    name: "MuiButtonBase"
  }), {
    action: r,
    centerRipple: s = !1,
    children: u,
    className: c,
    component: l = "button",
    disabled: f = !1,
    disableRipple: h = !1,
    disableTouchRipple: m = !1,
    focusRipple: y = !1,
    focusVisibleClassName: T,
    LinkComponent: g = "a",
    onBlur: w,
    onClick: M,
    onContextMenu: D,
    onDragLeave: L,
    onFocus: S,
    onFocusVisible: d,
    onKeyDown: C,
    onKeyUp: E,
    onMouseDown: F,
    onMouseLeave: B,
    onMouseUp: U,
    onTouchEnd: P,
    onTouchMove: O,
    onTouchStart: V,
    tabIndex: N = 0,
    TouchRippleProps: I,
    touchRippleRef: z,
    type: j,
    ...A
  } = i, Y = p.useRef(null), b = ct(), ye = te(b.ref, z), [k, H] = p.useState(!1);
  f && k && H(!1), p.useImperativeHandle(r, () => ({
    focusVisible: () => {
      H(!0), Y.current.focus();
    }
  }), []);
  const Te = b.shouldMount && !h && !f;
  p.useEffect(() => {
    k && y && !h && b.pulsate();
  }, [h, y, k, b]);
  const Re = v(b, "start", F, m), Me = v(b, "stop", D, m), Ee = v(b, "stop", L, m), Ce = v(b, "stop", U, m), Pe = v(b, "stop", (a) => {
    k && a.preventDefault(), B && B(a);
  }, m), xe = v(b, "start", V, m), we = v(b, "stop", P, m), Ne = v(b, "stop", O, m), ve = v(b, "stop", (a) => {
    pe(a.target) || H(!1), w && w(a);
  }, !1), De = J((a) => {
    Y.current || (Y.current = a.currentTarget), pe(a.target) && (H(!0), d && d(a)), S && S(a);
  }), ee = () => {
    const a = Y.current;
    return l && l !== "button" && !(a.tagName === "A" && a.href);
  }, Se = J((a) => {
    y && !a.repeat && k && a.key === " " && b.stop(a, () => {
      b.start(a);
    }), a.target === a.currentTarget && ee() && a.key === " " && a.preventDefault(), C && C(a), a.target === a.currentTarget && ee() && a.key === "Enter" && !f && (a.preventDefault(), M && M(a));
  }), Oe = J((a) => {
    y && a.key === " " && k && !a.defaultPrevented && b.stop(a, () => {
      b.pulsate(a);
    }), E && E(a), M && a.target === a.currentTarget && ee() && a.key === " " && !a.defaultPrevented && M(a);
  });
  let W = l;
  W === "button" && (A.href || A.to) && (W = g);
  const X = {};
  W === "button" ? (X.type = j === void 0 ? "button" : j, X.disabled = f) : (!A.href && !A.to && (X.role = "button"), f && (X["aria-disabled"] = f));
  const Ve = te(n, Y), le = {
    ...i,
    centerRipple: s,
    component: l,
    disabled: f,
    disableRipple: h,
    disableTouchRipple: m,
    focusRipple: y,
    tabIndex: N,
    focusVisible: k
  }, Ie = Rt(le);
  return /* @__PURE__ */ Ke(Mt, {
    as: W,
    className: x(Ie.root, c),
    ownerState: le,
    onBlur: ve,
    onClick: M,
    onContextMenu: Me,
    onFocus: De,
    onKeyDown: Se,
    onKeyUp: Oe,
    onMouseDown: Re,
    onMouseLeave: Pe,
    onMouseUp: Ce,
    onDragLeave: Ee,
    onTouchEnd: we,
    onTouchMove: Ne,
    onTouchStart: xe,
    ref: Ve,
    tabIndex: f ? -1 : N,
    type: j,
    ...X,
    ...A,
    children: [u, Te ? /* @__PURE__ */ _(ge, {
      ref: ye,
      center: s,
      ...I
    }) : null]
  });
});
function v(e, t, n, i = !1) {
  return J((r) => (n && n(r), i || e[t](r), !0));
}
process.env.NODE_ENV !== "production" && (Et.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * A ref for imperative actions.
   * It currently only supports `focusVisible()` action.
   */
  action: Je,
  /**
   * If `true`, the ripples are centered.
   * They won't start at the cursor interaction position.
   * @default false
   */
  centerRipple: o.bool,
  /**
   * The content of the component.
   */
  children: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: Ze,
  /**
   * If `true`, the component is disabled.
   * @default false
   */
  disabled: o.bool,
  /**
   * If `true`, the ripple effect is disabled.
   *
   * ⚠️ Without a ripple there is no styling for :focus-visible by default. Be sure
   * to highlight the element by applying separate styles with the `.Mui-focusVisible` class.
   * @default false
   */
  disableRipple: o.bool,
  /**
   * If `true`, the touch ripple effect is disabled.
   * @default false
   */
  disableTouchRipple: o.bool,
  /**
   * If `true`, the base button will have a keyboard focus ripple.
   * @default false
   */
  focusRipple: o.bool,
  /**
   * This prop can help identify which element has keyboard focus.
   * The class name will be applied when the element gains the focus through keyboard interaction.
   * It's a polyfill for the [CSS :focus-visible selector](https://drafts.csswg.org/selectors-4/#the-focus-visible-pseudo).
   * The rationale for using this feature [is explained here](https://github.com/WICG/focus-visible/blob/HEAD/explainer.md).
   * A [polyfill can be used](https://github.com/WICG/focus-visible) to apply a `focus-visible` class to other components
   * if needed.
   */
  focusVisibleClassName: o.string,
  /**
   * @ignore
   */
  href: o.any,
  /**
   * The component used to render a link when the `href` prop is provided.
   * @default 'a'
   */
  LinkComponent: o.elementType,
  /**
   * @ignore
   */
  onBlur: o.func,
  /**
   * @ignore
   */
  onClick: o.func,
  /**
   * @ignore
   */
  onContextMenu: o.func,
  /**
   * @ignore
   */
  onDragLeave: o.func,
  /**
   * @ignore
   */
  onFocus: o.func,
  /**
   * Callback fired when the component is focused with a keyboard.
   * We trigger a `onFocus` callback too.
   */
  onFocusVisible: o.func,
  /**
   * @ignore
   */
  onKeyDown: o.func,
  /**
   * @ignore
   */
  onKeyUp: o.func,
  /**
   * @ignore
   */
  onMouseDown: o.func,
  /**
   * @ignore
   */
  onMouseLeave: o.func,
  /**
   * @ignore
   */
  onMouseUp: o.func,
  /**
   * @ignore
   */
  onTouchEnd: o.func,
  /**
   * @ignore
   */
  onTouchMove: o.func,
  /**
   * @ignore
   */
  onTouchStart: o.func,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object]),
  /**
   * @default 0
   */
  tabIndex: o.number,
  /**
   * Props applied to the `TouchRipple` element.
   */
  TouchRippleProps: o.object,
  /**
   * A ref that points to the `TouchRipple` element.
   */
  touchRippleRef: o.oneOfType([o.func, o.shape({
    current: o.shape({
      pulsate: o.func.isRequired,
      start: o.func.isRequired,
      stop: o.func.isRequired
    })
  })]),
  /**
   * @ignore
   */
  type: o.oneOfType([o.oneOf(["button", "reset", "submit"]), o.string])
});
export {
  Et as B,
  fe as T,
  ot as _,
  tt as a,
  It as b,
  He as c,
  qe as d,
  Ze as e,
  J as f,
  nt as g,
  ie as h,
  pe as i,
  ne as j,
  rt as k,
  Je as r,
  Vt as u
};
