// @notificationapi/react/dist/assets/createTheme.js
import * as Rr from "react";
const $e = {
  black: "#000",
  white: "#fff"
}, ge = {
  50: "#ffebee",
  100: "#ffcdd2",
  200: "#ef9a9a",
  300: "#e57373",
  400: "#ef5350",
  500: "#f44336",
  600: "#e53935",
  700: "#d32f2f",
  800: "#c62828",
  900: "#b71c1c",
  A100: "#ff8a80",
  A200: "#ff5252",
  A400: "#ff1744",
  A700: "#d50000"
}, me = {
  50: "#f3e5f5",
  100: "#e1bee7",
  200: "#ce93d8",
  300: "#ba68c8",
  400: "#ab47bc",
  500: "#9c27b0",
  600: "#8e24aa",
  700: "#7b1fa2",
  800: "#6a1b9a",
  900: "#4a148c",
  A100: "#ea80fc",
  A200: "#e040fb",
  A400: "#d500f9",
  A700: "#aa00ff"
}, he = {
  50: "#e3f2fd",
  100: "#bbdefb",
  200: "#90caf9",
  300: "#64b5f6",
  400: "#42a5f5",
  500: "#2196f3",
  600: "#1e88e5",
  700: "#1976d2",
  800: "#1565c0",
  900: "#0d47a1",
  A100: "#82b1ff",
  A200: "#448aff",
  A400: "#2979ff",
  A700: "#2962ff"
}, ye = {
  50: "#e1f5fe",
  100: "#b3e5fc",
  200: "#81d4fa",
  300: "#4fc3f7",
  400: "#29b6f6",
  500: "#03a9f4",
  600: "#039be5",
  700: "#0288d1",
  800: "#0277bd",
  900: "#01579b",
  A100: "#80d8ff",
  A200: "#40c4ff",
  A400: "#00b0ff",
  A700: "#0091ea"
}, be = {
  50: "#e8f5e9",
  100: "#c8e6c9",
  200: "#a5d6a7",
  300: "#81c784",
  400: "#66bb6a",
  500: "#4caf50",
  600: "#43a047",
  700: "#388e3c",
  800: "#2e7d32",
  900: "#1b5e20",
  A100: "#b9f6ca",
  A200: "#69f0ae",
  A400: "#00e676",
  A700: "#00c853"
}, Se = {
  50: "#fff3e0",
  100: "#ffe0b2",
  200: "#ffcc80",
  300: "#ffb74d",
  400: "#ffa726",
  500: "#ff9800",
  600: "#fb8c00",
  700: "#f57c00",
  800: "#ef6c00",
  900: "#e65100",
  A100: "#ffd180",
  A200: "#ffab40",
  A400: "#ff9100",
  A700: "#ff6d00"
}, qr = {
  50: "#fafafa",
  100: "#f5f5f5",
  200: "#eeeeee",
  300: "#e0e0e0",
  400: "#bdbdbd",
  500: "#9e9e9e",
  600: "#757575",
  700: "#616161",
  800: "#424242",
  900: "#212121",
  A100: "#f5f5f5",
  A200: "#eeeeee",
  A400: "#bdbdbd",
  A700: "#616161"
};
function ue(e, ...r) {
  const t = new URL(`https://mui.com/production-error/?code=${e}`);
  return r.forEach((n) => t.searchParams.append("args[]", n)), `Minified MUI error #${e}; visit ${t} for the full message.`;
}
var qn = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function Kr(e) {
  return e && e.__esModule && Object.prototype.hasOwnProperty.call(e, "default") ? e.default : e;
}
var rr = { exports: {} }, Pe = { exports: {} }, D = {};
/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var dr;
function Hr() {
  if (dr) return D;
  dr = 1;
  var e = typeof Symbol == "function" && Symbol.for, r = e ? Symbol.for("react.element") : 60103, t = e ? Symbol.for("react.portal") : 60106, n = e ? Symbol.for("react.fragment") : 60107, i = e ? Symbol.for("react.strict_mode") : 60108, a = e ? Symbol.for("react.profiler") : 60114, s = e ? Symbol.for("react.provider") : 60109, c = e ? Symbol.for("react.context") : 60110, u = e ? Symbol.for("react.async_mode") : 60111, d = e ? Symbol.for("react.concurrent_mode") : 60111, h = e ? Symbol.for("react.forward_ref") : 60112, p = e ? Symbol.for("react.suspense") : 60113, f = e ? Symbol.for("react.suspense_list") : 60120, S = e ? Symbol.for("react.memo") : 60115, T = e ? Symbol.for("react.lazy") : 60116, b = e ? Symbol.for("react.block") : 60121, $ = e ? Symbol.for("react.fundamental") : 60117, A = e ? Symbol.for("react.responder") : 60118, Y = e ? Symbol.for("react.scope") : 60119;
  function w(m) {
    if (typeof m == "object" && m !== null) {
      var q = m.$$typeof;
      switch (q) {
        case r:
          switch (m = m.type, m) {
            case u:
            case d:
            case n:
            case a:
            case i:
            case p:
              return m;
            default:
              switch (m = m && m.$$typeof, m) {
                case c:
                case h:
                case T:
                case S:
                case s:
                  return m;
                default:
                  return q;
              }
          }
        case t:
          return q;
      }
    }
  }
  function E(m) {
    return w(m) === d;
  }
  return D.AsyncMode = u, D.ConcurrentMode = d, D.ContextConsumer = c, D.ContextProvider = s, D.Element = r, D.ForwardRef = h, D.Fragment = n, D.Lazy = T, D.Memo = S, D.Portal = t, D.Profiler = a, D.StrictMode = i, D.Suspense = p, D.isAsyncMode = function(m) {
    return E(m) || w(m) === u;
  }, D.isConcurrentMode = E, D.isContextConsumer = function(m) {
    return w(m) === c;
  }, D.isContextProvider = function(m) {
    return w(m) === s;
  }, D.isElement = function(m) {
    return typeof m == "object" && m !== null && m.$$typeof === r;
  }, D.isForwardRef = function(m) {
    return w(m) === h;
  }, D.isFragment = function(m) {
    return w(m) === n;
  }, D.isLazy = function(m) {
    return w(m) === T;
  }, D.isMemo = function(m) {
    return w(m) === S;
  }, D.isPortal = function(m) {
    return w(m) === t;
  }, D.isProfiler = function(m) {
    return w(m) === a;
  }, D.isStrictMode = function(m) {
    return w(m) === i;
  }, D.isSuspense = function(m) {
    return w(m) === p;
  }, D.isValidElementType = function(m) {
    return typeof m == "string" || typeof m == "function" || m === n || m === d || m === a || m === i || m === p || m === f || typeof m == "object" && m !== null && (m.$$typeof === T || m.$$typeof === S || m.$$typeof === s || m.$$typeof === c || m.$$typeof === h || m.$$typeof === $ || m.$$typeof === A || m.$$typeof === Y || m.$$typeof === b);
  }, D.typeOf = w, D;
}
var M = {};
/** @license React v16.13.1
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var pr;
function Qr() {
  return pr || (pr = 1, process.env.NODE_ENV !== "production" && function() {
    var e = typeof Symbol == "function" && Symbol.for, r = e ? Symbol.for("react.element") : 60103, t = e ? Symbol.for("react.portal") : 60106, n = e ? Symbol.for("react.fragment") : 60107, i = e ? Symbol.for("react.strict_mode") : 60108, a = e ? Symbol.for("react.profiler") : 60114, s = e ? Symbol.for("react.provider") : 60109, c = e ? Symbol.for("react.context") : 60110, u = e ? Symbol.for("react.async_mode") : 60111, d = e ? Symbol.for("react.concurrent_mode") : 60111, h = e ? Symbol.for("react.forward_ref") : 60112, p = e ? Symbol.for("react.suspense") : 60113, f = e ? Symbol.for("react.suspense_list") : 60120, S = e ? Symbol.for("react.memo") : 60115, T = e ? Symbol.for("react.lazy") : 60116, b = e ? Symbol.for("react.block") : 60121, $ = e ? Symbol.for("react.fundamental") : 60117, A = e ? Symbol.for("react.responder") : 60118, Y = e ? Symbol.for("react.scope") : 60119;
    function w(y) {
      return typeof y == "string" || typeof y == "function" || // Note: its typeof might be other than 'symbol' or 'number' if it's a polyfill.
      y === n || y === d || y === a || y === i || y === p || y === f || typeof y == "object" && y !== null && (y.$$typeof === T || y.$$typeof === S || y.$$typeof === s || y.$$typeof === c || y.$$typeof === h || y.$$typeof === $ || y.$$typeof === A || y.$$typeof === Y || y.$$typeof === b);
    }
    function E(y) {
      if (typeof y == "object" && y !== null) {
        var ne = y.$$typeof;
        switch (ne) {
          case r:
            var ke = y.type;
            switch (ke) {
              case u:
              case d:
              case n:
              case a:
              case i:
              case p:
                return ke;
              default:
                var fr = ke && ke.$$typeof;
                switch (fr) {
                  case c:
                  case h:
                  case T:
                  case S:
                  case s:
                    return fr;
                  default:
                    return ne;
                }
            }
          case t:
            return ne;
        }
      }
    }
    var m = u, q = d, X = c, ie = s, K = r, o = h, C = n, j = T, L = S, re = t, ce = a, te = i, pe = p, xe = !1;
    function Ye(y) {
      return xe || (xe = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 17+. Update your code to use ReactIs.isConcurrentMode() instead. It has the exact same API.")), g(y) || E(y) === u;
    }
    function g(y) {
      return E(y) === d;
    }
    function v(y) {
      return E(y) === c;
    }
    function _(y) {
      return E(y) === s;
    }
    function k(y) {
      return typeof y == "object" && y !== null && y.$$typeof === r;
    }
    function O(y) {
      return E(y) === h;
    }
    function R(y) {
      return E(y) === n;
    }
    function x(y) {
      return E(y) === T;
    }
    function P(y) {
      return E(y) === S;
    }
    function I(y) {
      return E(y) === t;
    }
    function N(y) {
      return E(y) === a;
    }
    function B(y) {
      return E(y) === i;
    }
    function H(y) {
      return E(y) === p;
    }
    M.AsyncMode = m, M.ConcurrentMode = q, M.ContextConsumer = X, M.ContextProvider = ie, M.Element = K, M.ForwardRef = o, M.Fragment = C, M.Lazy = j, M.Memo = L, M.Portal = re, M.Profiler = ce, M.StrictMode = te, M.Suspense = pe, M.isAsyncMode = Ye, M.isConcurrentMode = g, M.isContextConsumer = v, M.isContextProvider = _, M.isElement = k, M.isForwardRef = O, M.isFragment = R, M.isLazy = x, M.isMemo = P, M.isPortal = I, M.isProfiler = N, M.isStrictMode = B, M.isSuspense = H, M.isValidElementType = w, M.typeOf = E;
  }()), M;
}
var gr;
function Ir() {
  return gr || (gr = 1, process.env.NODE_ENV === "production" ? Pe.exports = Hr() : Pe.exports = Qr()), Pe.exports;
}
/*
object-assign
(c) Sindre Sorhus
@license MIT
*/
var qe, mr;
function Jr() {
  if (mr) return qe;
  mr = 1;
  var e = Object.getOwnPropertySymbols, r = Object.prototype.hasOwnProperty, t = Object.prototype.propertyIsEnumerable;
  function n(a) {
    if (a == null)
      throw new TypeError("Object.assign cannot be called with null or undefined");
    return Object(a);
  }
  function i() {
    try {
      if (!Object.assign)
        return !1;
      var a = new String("abc");
      if (a[5] = "de", Object.getOwnPropertyNames(a)[0] === "5")
        return !1;
      for (var s = {}, c = 0; c < 10; c++)
        s["_" + String.fromCharCode(c)] = c;
      var u = Object.getOwnPropertyNames(s).map(function(h) {
        return s[h];
      });
      if (u.join("") !== "0123456789")
        return !1;
      var d = {};
      return "abcdefghijklmnopqrst".split("").forEach(function(h) {
        d[h] = h;
      }), Object.keys(Object.assign({}, d)).join("") === "abcdefghijklmnopqrst";
    } catch {
      return !1;
    }
  }
  return qe = i() ? Object.assign : function(a, s) {
    for (var c, u = n(a), d, h = 1; h < arguments.length; h++) {
      c = Object(arguments[h]);
      for (var p in c)
        r.call(c, p) && (u[p] = c[p]);
      if (e) {
        d = e(c);
        for (var f = 0; f < d.length; f++)
          t.call(c, d[f]) && (u[d[f]] = c[d[f]]);
      }
    }
    return u;
  }, qe;
}
var Ke, hr;
function or() {
  if (hr) return Ke;
  hr = 1;
  var e = "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED";
  return Ke = e, Ke;
}
var He, yr;
function Br() {
  return yr || (yr = 1, He = Function.call.bind(Object.prototype.hasOwnProperty)), He;
}
var Qe, br;
function Xr() {
  if (br) return Qe;
  br = 1;
  var e = function() {
  };
  if (process.env.NODE_ENV !== "production") {
    var r = or(), t = {}, n = Br();
    e = function(a) {
      var s = "Warning: " + a;
      typeof console < "u" && console.error(s);
      try {
        throw new Error(s);
      } catch {
      }
    };
  }
  function i(a, s, c, u, d) {
    if (process.env.NODE_ENV !== "production") {
      for (var h in a)
        if (n(a, h)) {
          var p;
          try {
            if (typeof a[h] != "function") {
              var f = Error(
                (u || "React class") + ": " + c + " type `" + h + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof a[h] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`."
              );
              throw f.name = "Invariant Violation", f;
            }
            p = a[h](s, h, u, c, null, r);
          } catch (T) {
            p = T;
          }
          if (p && !(p instanceof Error) && e(
            (u || "React class") + ": type specification of " + c + " `" + h + "` is invalid; the type checker function must return `null` or an `Error` but returned a " + typeof p + ". You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument)."
          ), p instanceof Error && !(p.message in t)) {
            t[p.message] = !0;
            var S = d ? d() : "";
            e(
              "Failed " + c + " type: " + p.message + (S ?? "")
            );
          }
        }
    }
  }
  return i.resetWarningCache = function() {
    process.env.NODE_ENV !== "production" && (t = {});
  }, Qe = i, Qe;
}
var Je, vr;
function Zr() {
  if (vr) return Je;
  vr = 1;
  var e = Ir(), r = Jr(), t = or(), n = Br(), i = Xr(), a = function() {
  };
  process.env.NODE_ENV !== "production" && (a = function(c) {
    var u = "Warning: " + c;
    typeof console < "u" && console.error(u);
    try {
      throw new Error(u);
    } catch {
    }
  });
  function s() {
    return null;
  }
  return Je = function(c, u) {
    var d = typeof Symbol == "function" && Symbol.iterator, h = "@@iterator";
    function p(g) {
      var v = g && (d && g[d] || g[h]);
      if (typeof v == "function")
        return v;
    }
    var f = "<<anonymous>>", S = {
      array: A("array"),
      bigint: A("bigint"),
      bool: A("boolean"),
      func: A("function"),
      number: A("number"),
      object: A("object"),
      string: A("string"),
      symbol: A("symbol"),
      any: Y(),
      arrayOf: w,
      element: E(),
      elementType: m(),
      instanceOf: q,
      node: o(),
      objectOf: ie,
      oneOf: X,
      oneOfType: K,
      shape: j,
      exact: L
    };
    function T(g, v) {
      return g === v ? g !== 0 || 1 / g === 1 / v : g !== g && v !== v;
    }
    function b(g, v) {
      this.message = g, this.data = v && typeof v == "object" ? v : {}, this.stack = "";
    }
    b.prototype = Error.prototype;
    function $(g) {
      if (process.env.NODE_ENV !== "production")
        var v = {}, _ = 0;
      function k(R, x, P, I, N, B, H) {
        if (I = I || f, B = B || P, H !== t) {
          if (u) {
            var y = new Error(
              "Calling PropTypes validators directly is not supported by the `prop-types` package. Use `PropTypes.checkPropTypes()` to call them. Read more at http://fb.me/use-check-prop-types"
            );
            throw y.name = "Invariant Violation", y;
          } else if (process.env.NODE_ENV !== "production" && typeof console < "u") {
            var ne = I + ":" + P;
            !v[ne] && // Avoid spamming the console because they are often not actionable except for lib authors
            _ < 3 && (a(
              "You are manually calling a React.PropTypes validation function for the `" + B + "` prop on `" + I + "`. This is deprecated and will throw in the standalone `prop-types` package. You may be seeing this warning due to a third-party PropTypes library. See https://fb.me/react-warning-dont-call-proptypes for details."
            ), v[ne] = !0, _++);
          }
        }
        return x[P] == null ? R ? x[P] === null ? new b("The " + N + " `" + B + "` is marked as required " + ("in `" + I + "`, but its value is `null`.")) : new b("The " + N + " `" + B + "` is marked as required in " + ("`" + I + "`, but its value is `undefined`.")) : null : g(x, P, I, N, B);
      }
      var O = k.bind(null, !1);
      return O.isRequired = k.bind(null, !0), O;
    }
    function A(g) {
      function v(_, k, O, R, x, P) {
        var I = _[k], N = te(I);
        if (N !== g) {
          var B = pe(I);
          return new b(
            "Invalid " + R + " `" + x + "` of type " + ("`" + B + "` supplied to `" + O + "`, expected ") + ("`" + g + "`."),
            { expectedType: g }
          );
        }
        return null;
      }
      return $(v);
    }
    function Y() {
      return $(s);
    }
    function w(g) {
      function v(_, k, O, R, x) {
        if (typeof g != "function")
          return new b("Property `" + x + "` of component `" + O + "` has invalid PropType notation inside arrayOf.");
        var P = _[k];
        if (!Array.isArray(P)) {
          var I = te(P);
          return new b("Invalid " + R + " `" + x + "` of type " + ("`" + I + "` supplied to `" + O + "`, expected an array."));
        }
        for (var N = 0; N < P.length; N++) {
          var B = g(P, N, O, R, x + "[" + N + "]", t);
          if (B instanceof Error)
            return B;
        }
        return null;
      }
      return $(v);
    }
    function E() {
      function g(v, _, k, O, R) {
        var x = v[_];
        if (!c(x)) {
          var P = te(x);
          return new b("Invalid " + O + " `" + R + "` of type " + ("`" + P + "` supplied to `" + k + "`, expected a single ReactElement."));
        }
        return null;
      }
      return $(g);
    }
    function m() {
      function g(v, _, k, O, R) {
        var x = v[_];
        if (!e.isValidElementType(x)) {
          var P = te(x);
          return new b("Invalid " + O + " `" + R + "` of type " + ("`" + P + "` supplied to `" + k + "`, expected a single ReactElement type."));
        }
        return null;
      }
      return $(g);
    }
    function q(g) {
      function v(_, k, O, R, x) {
        if (!(_[k] instanceof g)) {
          var P = g.name || f, I = Ye(_[k]);
          return new b("Invalid " + R + " `" + x + "` of type " + ("`" + I + "` supplied to `" + O + "`, expected ") + ("instance of `" + P + "`."));
        }
        return null;
      }
      return $(v);
    }
    function X(g) {
      if (!Array.isArray(g))
        return process.env.NODE_ENV !== "production" && (arguments.length > 1 ? a(
          "Invalid arguments supplied to oneOf, expected an array, got " + arguments.length + " arguments. A common mistake is to write oneOf(x, y, z) instead of oneOf([x, y, z])."
        ) : a("Invalid argument supplied to oneOf, expected an array.")), s;
      function v(_, k, O, R, x) {
        for (var P = _[k], I = 0; I < g.length; I++)
          if (T(P, g[I]))
            return null;
        var N = JSON.stringify(g, function(H, y) {
          var ne = pe(y);
          return ne === "symbol" ? String(y) : y;
        });
        return new b("Invalid " + R + " `" + x + "` of value `" + String(P) + "` " + ("supplied to `" + O + "`, expected one of " + N + "."));
      }
      return $(v);
    }
    function ie(g) {
      function v(_, k, O, R, x) {
        if (typeof g != "function")
          return new b("Property `" + x + "` of component `" + O + "` has invalid PropType notation inside objectOf.");
        var P = _[k], I = te(P);
        if (I !== "object")
          return new b("Invalid " + R + " `" + x + "` of type " + ("`" + I + "` supplied to `" + O + "`, expected an object."));
        for (var N in P)
          if (n(P, N)) {
            var B = g(P, N, O, R, x + "." + N, t);
            if (B instanceof Error)
              return B;
          }
        return null;
      }
      return $(v);
    }
    function K(g) {
      if (!Array.isArray(g))
        return process.env.NODE_ENV !== "production" && a("Invalid argument supplied to oneOfType, expected an instance of array."), s;
      for (var v = 0; v < g.length; v++) {
        var _ = g[v];
        if (typeof _ != "function")
          return a(
            "Invalid argument supplied to oneOfType. Expected an array of check functions, but received " + xe(_) + " at index " + v + "."
          ), s;
      }
      function k(O, R, x, P, I) {
        for (var N = [], B = 0; B < g.length; B++) {
          var H = g[B], y = H(O, R, x, P, I, t);
          if (y == null)
            return null;
          y.data && n(y.data, "expectedType") && N.push(y.data.expectedType);
        }
        var ne = N.length > 0 ? ", expected one of type [" + N.join(", ") + "]" : "";
        return new b("Invalid " + P + " `" + I + "` supplied to " + ("`" + x + "`" + ne + "."));
      }
      return $(k);
    }
    function o() {
      function g(v, _, k, O, R) {
        return re(v[_]) ? null : new b("Invalid " + O + " `" + R + "` supplied to " + ("`" + k + "`, expected a ReactNode."));
      }
      return $(g);
    }
    function C(g, v, _, k, O) {
      return new b(
        (g || "React class") + ": " + v + " type `" + _ + "." + k + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + O + "`."
      );
    }
    function j(g) {
      function v(_, k, O, R, x) {
        var P = _[k], I = te(P);
        if (I !== "object")
          return new b("Invalid " + R + " `" + x + "` of type `" + I + "` " + ("supplied to `" + O + "`, expected `object`."));
        for (var N in g) {
          var B = g[N];
          if (typeof B != "function")
            return C(O, R, x, N, pe(B));
          var H = B(P, N, O, R, x + "." + N, t);
          if (H)
            return H;
        }
        return null;
      }
      return $(v);
    }
    function L(g) {
      function v(_, k, O, R, x) {
        var P = _[k], I = te(P);
        if (I !== "object")
          return new b("Invalid " + R + " `" + x + "` of type `" + I + "` " + ("supplied to `" + O + "`, expected `object`."));
        var N = r({}, _[k], g);
        for (var B in N) {
          var H = g[B];
          if (n(g, B) && typeof H != "function")
            return C(O, R, x, B, pe(H));
          if (!H)
            return new b(
              "Invalid " + R + " `" + x + "` key `" + B + "` supplied to `" + O + "`.\nBad object: " + JSON.stringify(_[k], null, "  ") + `
Valid keys: ` + JSON.stringify(Object.keys(g), null, "  ")
            );
          var y = H(P, B, O, R, x + "." + B, t);
          if (y)
            return y;
        }
        return null;
      }
      return $(v);
    }
    function re(g) {
      switch (typeof g) {
        case "number":
        case "string":
        case "undefined":
          return !0;
        case "boolean":
          return !g;
        case "object":
          if (Array.isArray(g))
            return g.every(re);
          if (g === null || c(g))
            return !0;
          var v = p(g);
          if (v) {
            var _ = v.call(g), k;
            if (v !== g.entries) {
              for (; !(k = _.next()).done; )
                if (!re(k.value))
                  return !1;
            } else
              for (; !(k = _.next()).done; ) {
                var O = k.value;
                if (O && !re(O[1]))
                  return !1;
              }
          } else
            return !1;
          return !0;
        default:
          return !1;
      }
    }
    function ce(g, v) {
      return g === "symbol" ? !0 : v ? v["@@toStringTag"] === "Symbol" || typeof Symbol == "function" && v instanceof Symbol : !1;
    }
    function te(g) {
      var v = typeof g;
      return Array.isArray(g) ? "array" : g instanceof RegExp ? "object" : ce(v, g) ? "symbol" : v;
    }
    function pe(g) {
      if (typeof g > "u" || g === null)
        return "" + g;
      var v = te(g);
      if (v === "object") {
        if (g instanceof Date)
          return "date";
        if (g instanceof RegExp)
          return "regexp";
      }
      return v;
    }
    function xe(g) {
      var v = pe(g);
      switch (v) {
        case "array":
        case "object":
          return "an " + v;
        case "boolean":
        case "date":
        case "regexp":
          return "a " + v;
        default:
          return v;
      }
    }
    function Ye(g) {
      return !g.constructor || !g.constructor.name ? f : g.constructor.name;
    }
    return S.checkPropTypes = i, S.resetWarningCache = i.resetWarningCache, S.PropTypes = S, S;
  }, Je;
}
var Xe, Sr;
function et() {
  if (Sr) return Xe;
  Sr = 1;
  var e = or();
  function r() {
  }
  function t() {
  }
  return t.resetWarningCache = r, Xe = function() {
    function n(s, c, u, d, h, p) {
      if (p !== e) {
        var f = new Error(
          "Calling PropTypes validators directly is not supported by the `prop-types` package. Use PropTypes.checkPropTypes() to call them. Read more at http://fb.me/use-check-prop-types"
        );
        throw f.name = "Invariant Violation", f;
      }
    }
    n.isRequired = n;
    function i() {
      return n;
    }
    var a = {
      array: n,
      bigint: n,
      bool: n,
      func: n,
      number: n,
      object: n,
      string: n,
      symbol: n,
      any: n,
      arrayOf: i,
      element: n,
      elementType: n,
      instanceOf: i,
      node: n,
      objectOf: i,
      oneOf: i,
      oneOfType: i,
      shape: i,
      exact: i,
      checkPropTypes: t,
      resetWarningCache: r
    };
    return a.PropTypes = a, a;
  }, Xe;
}
if (process.env.NODE_ENV !== "production") {
  var rt = Ir(), tt = !0;
  rr.exports = Zr()(rt.isElement, tt);
} else
  rr.exports = et()();
var nt = rr.exports;
const Ce = /* @__PURE__ */ Kr(nt);
function le(e) {
  if (typeof e != "object" || e === null)
    return !1;
  const r = Object.getPrototypeOf(e);
  return (r === null || r === Object.prototype || Object.getPrototypeOf(r) === null) && !(Symbol.toStringTag in e) && !(Symbol.iterator in e);
}
function jr(e) {
  if (/* @__PURE__ */ Rr.isValidElement(e) || !le(e))
    return e;
  const r = {};
  return Object.keys(e).forEach((t) => {
    r[t] = jr(e[t]);
  }), r;
}
function Q(e, r, t = {
  clone: !0
}) {
  const n = t.clone ? {
    ...e
  } : e;
  return le(e) && le(r) && Object.keys(r).forEach((i) => {
    /* @__PURE__ */ Rr.isValidElement(r[i]) ? n[i] = r[i] : le(r[i]) && // Avoid prototype pollution
    Object.prototype.hasOwnProperty.call(e, i) && le(e[i]) ? n[i] = Q(e[i], r[i], t) : t.clone ? n[i] = le(r[i]) ? jr(r[i]) : r[i] : n[i] = r[i];
  }), n;
}
const ot = (e) => {
  const r = Object.keys(e).map((t) => ({
    key: t,
    val: e[t]
  })) || [];
  return r.sort((t, n) => t.val - n.val), r.reduce((t, n) => ({
    ...t,
    [n.key]: n.val
  }), {});
};
function it(e) {
  const {
    // The breakpoint **start** at this value.
    // For instance with the first breakpoint xs: [xs, sm).
    values: r = {
      xs: 0,
      // phone
      sm: 600,
      // tablet
      md: 900,
      // small laptop
      lg: 1200,
      // desktop
      xl: 1536
      // large screen
    },
    unit: t = "px",
    step: n = 5,
    ...i
  } = e, a = ot(r), s = Object.keys(a);
  function c(f) {
    return `@media (min-width:${typeof r[f] == "number" ? r[f] : f}${t})`;
  }
  function u(f) {
    return `@media (max-width:${(typeof r[f] == "number" ? r[f] : f) - n / 100}${t})`;
  }
  function d(f, S) {
    const T = s.indexOf(S);
    return `@media (min-width:${typeof r[f] == "number" ? r[f] : f}${t}) and (max-width:${(T !== -1 && typeof r[s[T]] == "number" ? r[s[T]] : S) - n / 100}${t})`;
  }
  function h(f) {
    return s.indexOf(f) + 1 < s.length ? d(f, s[s.indexOf(f) + 1]) : c(f);
  }
  function p(f) {
    const S = s.indexOf(f);
    return S === 0 ? c(s[1]) : S === s.length - 1 ? u(s[S]) : d(f, s[s.indexOf(f) + 1]).replace("@media", "@media not all and");
  }
  return {
    keys: s,
    values: a,
    up: c,
    down: u,
    between: d,
    only: h,
    not: p,
    unit: t,
    ...i
  };
}
function at(e, r) {
  if (!e.containerQueries)
    return r;
  const t = Object.keys(r).filter((n) => n.startsWith("@container")).sort((n, i) => {
    var s, c;
    const a = /min-width:\s*([0-9.]+)/;
    return +(((s = n.match(a)) == null ? void 0 : s[1]) || 0) - +(((c = i.match(a)) == null ? void 0 : c[1]) || 0);
  });
  return t.length ? t.reduce((n, i) => {
    const a = r[i];
    return delete n[i], n[i] = a, n;
  }, {
    ...r
  }) : r;
}
function st(e, r) {
  return r === "@" || r.startsWith("@") && (e.some((t) => r.startsWith(`@${t}`)) || !!r.match(/^@\d/));
}
function ct(e, r) {
  const t = r.match(/^@([^/]+)?\/?(.+)?$/);
  if (!t) {
    if (process.env.NODE_ENV !== "production")
      throw new Error(process.env.NODE_ENV !== "production" ? `MUI: The provided shorthand ${`(${r})`} is invalid. The format should be \`@<breakpoint | number>\` or \`@<breakpoint | number>/<container>\`.
For example, \`@sm\` or \`@600\` or \`@40rem/sidebar\`.` : ue(18, `(${r})`));
    return null;
  }
  const [, n, i] = t, a = Number.isNaN(+n) ? n || 0 : +n;
  return e.containerQueries(i).up(a);
}
function lt(e) {
  const r = (a, s) => a.replace("@media", s ? `@container ${s}` : "@container");
  function t(a, s) {
    a.up = (...c) => r(e.breakpoints.up(...c), s), a.down = (...c) => r(e.breakpoints.down(...c), s), a.between = (...c) => r(e.breakpoints.between(...c), s), a.only = (...c) => r(e.breakpoints.only(...c), s), a.not = (...c) => {
      const u = r(e.breakpoints.not(...c), s);
      return u.includes("not all and") ? u.replace("not all and ", "").replace("min-width:", "width<").replace("max-width:", "width>").replace("and", "or") : u;
    };
  }
  const n = {}, i = (a) => (t(n, a), n);
  return t(i), {
    ...e,
    containerQueries: i
  };
}
const ut = {
  borderRadius: 4
}, de = process.env.NODE_ENV !== "production" ? Ce.oneOfType([Ce.number, Ce.string, Ce.object, Ce.array]) : {};
function we(e, r) {
  return r ? Q(e, r, {
    clone: !1
    // No need to clone deep, it's way faster.
  }) : e;
}
const Be = {
  xs: 0,
  // phone
  sm: 600,
  // tablet
  md: 900,
  // small laptop
  lg: 1200,
  // desktop
  xl: 1536
  // large screen
}, Cr = {
  // Sorted ASC by size. That's important.
  // It can't be configured as it's used statically for propTypes.
  keys: ["xs", "sm", "md", "lg", "xl"],
  up: (e) => `@media (min-width:${Be[e]}px)`
}, ft = {
  containerQueries: (e) => ({
    up: (r) => {
      let t = typeof r == "number" ? r : Be[r] || r;
      return typeof t == "number" && (t = `${t}px`), e ? `@container ${e} (min-width:${t})` : `@container (min-width:${t})`;
    }
  })
};
function se(e, r, t) {
  const n = e.theme || {};
  if (Array.isArray(r)) {
    const a = n.breakpoints || Cr;
    return r.reduce((s, c, u) => (s[a.up(a.keys[u])] = t(r[u]), s), {});
  }
  if (typeof r == "object") {
    const a = n.breakpoints || Cr;
    return Object.keys(r).reduce((s, c) => {
      if (st(a.keys, c)) {
        const u = ct(n.containerQueries ? n : ft, c);
        u && (s[u] = t(r[c], c));
      } else if (Object.keys(a.values || Be).includes(c)) {
        const u = a.up(c);
        s[u] = t(r[c], c);
      } else {
        const u = c;
        s[u] = r[u];
      }
      return s;
    }, {});
  }
  return t(r);
}
function Dr(e = {}) {
  var t;
  return ((t = e.keys) == null ? void 0 : t.reduce((n, i) => {
    const a = e.up(i);
    return n[a] = {}, n;
  }, {})) || {};
}
function Mr(e, r) {
  return e.reduce((t, n) => {
    const i = t[n];
    return (!i || Object.keys(i).length === 0) && delete t[n], t;
  }, r);
}
function Kn(e, ...r) {
  const t = Dr(e), n = [t, ...r].reduce((i, a) => Q(i, a), {});
  return Mr(Object.keys(t), n);
}
function dt(e, r) {
  if (typeof e != "object")
    return {};
  const t = {}, n = Object.keys(r);
  return Array.isArray(e) ? n.forEach((i, a) => {
    a < e.length && (t[i] = !0);
  }) : n.forEach((i) => {
    e[i] != null && (t[i] = !0);
  }), t;
}
function Hn({
  values: e,
  breakpoints: r,
  base: t
}) {
  const n = t || dt(e, r), i = Object.keys(n);
  if (i.length === 0)
    return e;
  let a;
  return i.reduce((s, c, u) => (Array.isArray(e) ? (s[c] = e[u] != null ? e[u] : e[a], a = u) : typeof e == "object" ? (s[c] = e[c] != null ? e[c] : e[a], a = c) : s[c] = e, s), {});
}
function Nr(e) {
  if (typeof e != "string")
    throw new Error(process.env.NODE_ENV !== "production" ? "MUI: `capitalize(string)` expects a string argument." : ue(7));
  return e.charAt(0).toUpperCase() + e.slice(1);
}
function je(e, r, t = !0) {
  if (!r || typeof r != "string")
    return null;
  if (e && e.vars && t) {
    const n = `vars.${r}`.split(".").reduce((i, a) => i && i[a] ? i[a] : null, e);
    if (n != null)
      return n;
  }
  return r.split(".").reduce((n, i) => n && n[i] != null ? n[i] : null, e);
}
function Ie(e, r, t, n = t) {
  let i;
  return typeof e == "function" ? i = e(t) : Array.isArray(e) ? i = e[t] || n : i = je(e, t) || n, r && (i = r(i, n, e)), i;
}
function G(e) {
  const {
    prop: r,
    cssProperty: t = e.prop,
    themeKey: n,
    transform: i
  } = e, a = (s) => {
    if (s[r] == null)
      return null;
    const c = s[r], u = s.theme, d = je(u, n) || {};
    return se(s, c, (p) => {
      let f = Ie(d, i, p);
      return p === f && typeof p == "string" && (f = Ie(d, i, `${r}${p === "default" ? "" : Nr(p)}`, p)), t === !1 ? f : {
        [t]: f
      };
    });
  };
  return a.propTypes = process.env.NODE_ENV !== "production" ? {
    [r]: de
  } : {}, a.filterProps = [r], a;
}
function pt(e) {
  const r = {};
  return (t) => (r[t] === void 0 && (r[t] = e(t)), r[t]);
}
const gt = {
  m: "margin",
  p: "padding"
}, mt = {
  t: "Top",
  r: "Right",
  b: "Bottom",
  l: "Left",
  x: ["Left", "Right"],
  y: ["Top", "Bottom"]
}, Tr = {
  marginX: "mx",
  marginY: "my",
  paddingX: "px",
  paddingY: "py"
}, ht = pt((e) => {
  if (e.length > 2)
    if (Tr[e])
      e = Tr[e];
    else
      return [e];
  const [r, t] = e.split(""), n = gt[r], i = mt[t] || "";
  return Array.isArray(i) ? i.map((a) => n + a) : [n + i];
}), De = ["m", "mt", "mr", "mb", "ml", "mx", "my", "margin", "marginTop", "marginRight", "marginBottom", "marginLeft", "marginX", "marginY", "marginInline", "marginInlineStart", "marginInlineEnd", "marginBlock", "marginBlockStart", "marginBlockEnd"], Me = ["p", "pt", "pr", "pb", "pl", "px", "py", "padding", "paddingTop", "paddingRight", "paddingBottom", "paddingLeft", "paddingX", "paddingY", "paddingInline", "paddingInlineStart", "paddingInlineEnd", "paddingBlock", "paddingBlockStart", "paddingBlockEnd"], yt = [...De, ...Me];
function Oe(e, r, t, n) {
  const i = je(e, r, !0) ?? t;
  return typeof i == "number" || typeof i == "string" ? (a) => typeof a == "string" ? a : (process.env.NODE_ENV !== "production" && typeof a != "number" && console.error(`MUI: Expected ${n} argument to be a number or a string, got ${a}.`), typeof i == "string" ? `calc(${a} * ${i})` : i * a) : Array.isArray(i) ? (a) => {
    if (typeof a == "string")
      return a;
    const s = Math.abs(a);
    process.env.NODE_ENV !== "production" && (Number.isInteger(s) ? s > i.length - 1 && console.error([`MUI: The value provided (${s}) overflows.`, `The supported values are: ${JSON.stringify(i)}.`, `${s} > ${i.length - 1}, you need to add the missing values.`].join(`
`)) : console.error([`MUI: The \`theme.${r}\` array type cannot be combined with non integer values.You should either use an integer value that can be used as index, or define the \`theme.${r}\` as a number.`].join(`
`)));
    const c = i[s];
    return a >= 0 ? c : typeof c == "number" ? -c : `-${c}`;
  } : typeof i == "function" ? i : (process.env.NODE_ENV !== "production" && console.error([`MUI: The \`theme.${r}\` value (${i}) is invalid.`, "It should be a number, an array or a function."].join(`
`)), () => {
  });
}
function ir(e) {
  return Oe(e, "spacing", 8, "spacing");
}
function Ae(e, r) {
  return typeof r == "string" || r == null ? r : e(r);
}
function bt(e, r) {
  return (t) => e.reduce((n, i) => (n[i] = Ae(r, t), n), {});
}
function vt(e, r, t, n) {
  if (!r.includes(t))
    return null;
  const i = ht(t), a = bt(i, n), s = e[t];
  return se(e, s, a);
}
function Wr(e, r) {
  const t = ir(e.theme);
  return Object.keys(e).map((n) => vt(e, r, n, t)).reduce(we, {});
}
function V(e) {
  return Wr(e, De);
}
V.propTypes = process.env.NODE_ENV !== "production" ? De.reduce((e, r) => (e[r] = de, e), {}) : {};
V.filterProps = De;
function z(e) {
  return Wr(e, Me);
}
z.propTypes = process.env.NODE_ENV !== "production" ? Me.reduce((e, r) => (e[r] = de, e), {}) : {};
z.filterProps = Me;
process.env.NODE_ENV !== "production" && yt.reduce((e, r) => (e[r] = de, e), {});
function Ur(e = 8, r = ir({
  spacing: e
})) {
  if (e.mui)
    return e;
  const t = (...n) => (process.env.NODE_ENV !== "production" && (n.length <= 4 || console.error(`MUI: Too many arguments provided, expected between 0 and 4, got ${n.length}`)), (n.length === 0 ? [1] : n).map((a) => {
    const s = r(a);
    return typeof s == "number" ? `${s}px` : s;
  }).join(" "));
  return t.mui = !0, t;
}
function Ne(...e) {
  const r = e.reduce((n, i) => (i.filterProps.forEach((a) => {
    n[a] = i;
  }), n), {}), t = (n) => Object.keys(n).reduce((i, a) => r[a] ? we(i, r[a](n)) : i, {});
  return t.propTypes = process.env.NODE_ENV !== "production" ? e.reduce((n, i) => Object.assign(n, i.propTypes), {}) : {}, t.filterProps = e.reduce((n, i) => n.concat(i.filterProps), []), t;
}
function Z(e) {
  return typeof e != "number" ? e : `${e}px solid`;
}
function ee(e, r) {
  return G({
    prop: e,
    themeKey: "borders",
    transform: r
  });
}
const St = ee("border", Z), Ct = ee("borderTop", Z), Tt = ee("borderRight", Z), Et = ee("borderBottom", Z), wt = ee("borderLeft", Z), $t = ee("borderColor"), Ot = ee("borderTopColor"), At = ee("borderRightColor"), xt = ee("borderBottomColor"), kt = ee("borderLeftColor"), Pt = ee("outline", Z), _t = ee("outlineColor"), We = (e) => {
  if (e.borderRadius !== void 0 && e.borderRadius !== null) {
    const r = Oe(e.theme, "shape.borderRadius", 4, "borderRadius"), t = (n) => ({
      borderRadius: Ae(r, n)
    });
    return se(e, e.borderRadius, t);
  }
  return null;
};
We.propTypes = process.env.NODE_ENV !== "production" ? {
  borderRadius: de
} : {};
We.filterProps = ["borderRadius"];
Ne(St, Ct, Tt, Et, wt, $t, Ot, At, xt, kt, We, Pt, _t);
const Ue = (e) => {
  if (e.gap !== void 0 && e.gap !== null) {
    const r = Oe(e.theme, "spacing", 8, "gap"), t = (n) => ({
      gap: Ae(r, n)
    });
    return se(e, e.gap, t);
  }
  return null;
};
Ue.propTypes = process.env.NODE_ENV !== "production" ? {
  gap: de
} : {};
Ue.filterProps = ["gap"];
const Fe = (e) => {
  if (e.columnGap !== void 0 && e.columnGap !== null) {
    const r = Oe(e.theme, "spacing", 8, "columnGap"), t = (n) => ({
      columnGap: Ae(r, n)
    });
    return se(e, e.columnGap, t);
  }
  return null;
};
Fe.propTypes = process.env.NODE_ENV !== "production" ? {
  columnGap: de
} : {};
Fe.filterProps = ["columnGap"];
const Le = (e) => {
  if (e.rowGap !== void 0 && e.rowGap !== null) {
    const r = Oe(e.theme, "spacing", 8, "rowGap"), t = (n) => ({
      rowGap: Ae(r, n)
    });
    return se(e, e.rowGap, t);
  }
  return null;
};
Le.propTypes = process.env.NODE_ENV !== "production" ? {
  rowGap: de
} : {};
Le.filterProps = ["rowGap"];
const Rt = G({
  prop: "gridColumn"
}), It = G({
  prop: "gridRow"
}), Bt = G({
  prop: "gridAutoFlow"
}), jt = G({
  prop: "gridAutoColumns"
}), Dt = G({
  prop: "gridAutoRows"
}), Mt = G({
  prop: "gridTemplateColumns"
}), Nt = G({
  prop: "gridTemplateRows"
}), Wt = G({
  prop: "gridTemplateAreas"
}), Ut = G({
  prop: "gridArea"
});
Ne(Ue, Fe, Le, Rt, It, Bt, jt, Dt, Mt, Nt, Wt, Ut);
function ve(e, r) {
  return r === "grey" ? r : e;
}
const Ft = G({
  prop: "color",
  themeKey: "palette",
  transform: ve
}), Lt = G({
  prop: "bgcolor",
  cssProperty: "backgroundColor",
  themeKey: "palette",
  transform: ve
}), Vt = G({
  prop: "backgroundColor",
  themeKey: "palette",
  transform: ve
});
Ne(Ft, Lt, Vt);
function J(e) {
  return e <= 1 && e !== 0 ? `${e * 100}%` : e;
}
const zt = G({
  prop: "width",
  transform: J
}), ar = (e) => {
  if (e.maxWidth !== void 0 && e.maxWidth !== null) {
    const r = (t) => {
      var i, a, s, c, u;
      const n = ((s = (a = (i = e.theme) == null ? void 0 : i.breakpoints) == null ? void 0 : a.values) == null ? void 0 : s[t]) || Be[t];
      return n ? ((u = (c = e.theme) == null ? void 0 : c.breakpoints) == null ? void 0 : u.unit) !== "px" ? {
        maxWidth: `${n}${e.theme.breakpoints.unit}`
      } : {
        maxWidth: n
      } : {
        maxWidth: J(t)
      };
    };
    return se(e, e.maxWidth, r);
  }
  return null;
};
ar.filterProps = ["maxWidth"];
const Gt = G({
  prop: "minWidth",
  transform: J
}), Yt = G({
  prop: "height",
  transform: J
}), qt = G({
  prop: "maxHeight",
  transform: J
}), Kt = G({
  prop: "minHeight",
  transform: J
});
G({
  prop: "size",
  cssProperty: "width",
  transform: J
});
G({
  prop: "size",
  cssProperty: "height",
  transform: J
});
const Ht = G({
  prop: "boxSizing"
});
Ne(zt, ar, Gt, Yt, qt, Kt, Ht);
const Ve = {
  // borders
  border: {
    themeKey: "borders",
    transform: Z
  },
  borderTop: {
    themeKey: "borders",
    transform: Z
  },
  borderRight: {
    themeKey: "borders",
    transform: Z
  },
  borderBottom: {
    themeKey: "borders",
    transform: Z
  },
  borderLeft: {
    themeKey: "borders",
    transform: Z
  },
  borderColor: {
    themeKey: "palette"
  },
  borderTopColor: {
    themeKey: "palette"
  },
  borderRightColor: {
    themeKey: "palette"
  },
  borderBottomColor: {
    themeKey: "palette"
  },
  borderLeftColor: {
    themeKey: "palette"
  },
  outline: {
    themeKey: "borders",
    transform: Z
  },
  outlineColor: {
    themeKey: "palette"
  },
  borderRadius: {
    themeKey: "shape.borderRadius",
    style: We
  },
  // palette
  color: {
    themeKey: "palette",
    transform: ve
  },
  bgcolor: {
    themeKey: "palette",
    cssProperty: "backgroundColor",
    transform: ve
  },
  backgroundColor: {
    themeKey: "palette",
    transform: ve
  },
  // spacing
  p: {
    style: z
  },
  pt: {
    style: z
  },
  pr: {
    style: z
  },
  pb: {
    style: z
  },
  pl: {
    style: z
  },
  px: {
    style: z
  },
  py: {
    style: z
  },
  padding: {
    style: z
  },
  paddingTop: {
    style: z
  },
  paddingRight: {
    style: z
  },
  paddingBottom: {
    style: z
  },
  paddingLeft: {
    style: z
  },
  paddingX: {
    style: z
  },
  paddingY: {
    style: z
  },
  paddingInline: {
    style: z
  },
  paddingInlineStart: {
    style: z
  },
  paddingInlineEnd: {
    style: z
  },
  paddingBlock: {
    style: z
  },
  paddingBlockStart: {
    style: z
  },
  paddingBlockEnd: {
    style: z
  },
  m: {
    style: V
  },
  mt: {
    style: V
  },
  mr: {
    style: V
  },
  mb: {
    style: V
  },
  ml: {
    style: V
  },
  mx: {
    style: V
  },
  my: {
    style: V
  },
  margin: {
    style: V
  },
  marginTop: {
    style: V
  },
  marginRight: {
    style: V
  },
  marginBottom: {
    style: V
  },
  marginLeft: {
    style: V
  },
  marginX: {
    style: V
  },
  marginY: {
    style: V
  },
  marginInline: {
    style: V
  },
  marginInlineStart: {
    style: V
  },
  marginInlineEnd: {
    style: V
  },
  marginBlock: {
    style: V
  },
  marginBlockStart: {
    style: V
  },
  marginBlockEnd: {
    style: V
  },
  // display
  displayPrint: {
    cssProperty: !1,
    transform: (e) => ({
      "@media print": {
        display: e
      }
    })
  },
  display: {},
  overflow: {},
  textOverflow: {},
  visibility: {},
  whiteSpace: {},
  // flexbox
  flexBasis: {},
  flexDirection: {},
  flexWrap: {},
  justifyContent: {},
  alignItems: {},
  alignContent: {},
  order: {},
  flex: {},
  flexGrow: {},
  flexShrink: {},
  alignSelf: {},
  justifyItems: {},
  justifySelf: {},
  // grid
  gap: {
    style: Ue
  },
  rowGap: {
    style: Le
  },
  columnGap: {
    style: Fe
  },
  gridColumn: {},
  gridRow: {},
  gridAutoFlow: {},
  gridAutoColumns: {},
  gridAutoRows: {},
  gridTemplateColumns: {},
  gridTemplateRows: {},
  gridTemplateAreas: {},
  gridArea: {},
  // positions
  position: {},
  zIndex: {
    themeKey: "zIndex"
  },
  top: {},
  right: {},
  bottom: {},
  left: {},
  // shadows
  boxShadow: {
    themeKey: "shadows"
  },
  // sizing
  width: {
    transform: J
  },
  maxWidth: {
    style: ar
  },
  minWidth: {
    transform: J
  },
  height: {
    transform: J
  },
  maxHeight: {
    transform: J
  },
  minHeight: {
    transform: J
  },
  boxSizing: {},
  // typography
  font: {
    themeKey: "font"
  },
  fontFamily: {
    themeKey: "typography"
  },
  fontSize: {
    themeKey: "typography"
  },
  fontStyle: {
    themeKey: "typography"
  },
  fontWeight: {
    themeKey: "typography"
  },
  letterSpacing: {},
  textTransform: {},
  lineHeight: {},
  textAlign: {},
  typography: {
    cssProperty: !1,
    themeKey: "typography"
  }
};
function Qt(...e) {
  const r = e.reduce((n, i) => n.concat(Object.keys(i)), []), t = new Set(r);
  return e.every((n) => t.size === Object.keys(n).length);
}
function Jt(e, r) {
  return typeof e == "function" ? e(r) : e;
}
function Xt() {
  function e(t, n, i, a) {
    const s = {
      [t]: n,
      theme: i
    }, c = a[t];
    if (!c)
      return {
        [t]: n
      };
    const {
      cssProperty: u = t,
      themeKey: d,
      transform: h,
      style: p
    } = c;
    if (n == null)
      return null;
    if (d === "typography" && n === "inherit")
      return {
        [t]: n
      };
    const f = je(i, d) || {};
    return p ? p(s) : se(s, n, (T) => {
      let b = Ie(f, h, T);
      return T === b && typeof T == "string" && (b = Ie(f, h, `${t}${T === "default" ? "" : Nr(T)}`, T)), u === !1 ? b : {
        [u]: b
      };
    });
  }
  function r(t) {
    const {
      sx: n,
      theme: i = {}
    } = t || {};
    if (!n)
      return null;
    const a = i.unstable_sxConfig ?? Ve;
    function s(c) {
      let u = c;
      if (typeof c == "function")
        u = c(i);
      else if (typeof c != "object")
        return c;
      if (!u)
        return null;
      const d = Dr(i.breakpoints), h = Object.keys(d);
      let p = d;
      return Object.keys(u).forEach((f) => {
        const S = Jt(u[f], i);
        if (S != null)
          if (typeof S == "object")
            if (a[f])
              p = we(p, e(f, S, i, a));
            else {
              const T = se({
                theme: i
              }, S, (b) => ({
                [f]: b
              }));
              Qt(T, S) ? p[f] = r({
                sx: S,
                theme: i
              }) : p = we(p, T);
            }
          else
            p = we(p, e(f, S, i, a));
      }), at(i, Mr(h, p));
    }
    return Array.isArray(n) ? n.map(s) : s(n);
  }
  return r;
}
const ze = Xt();
ze.filterProps = ["sx"];
function Zt(e, r) {
  var n;
  const t = this;
  if (t.vars) {
    if (!((n = t.colorSchemes) != null && n[e]) || typeof t.getColorSchemeSelector != "function")
      return {};
    let i = t.getColorSchemeSelector(e);
    return i === "&" ? r : ((i.includes("data-") || i.includes(".")) && (i = `*:where(${i.replace(/\s*&$/, "")}) &`), {
      [i]: r
    });
  }
  return t.palette.mode === e ? r : {};
}
function en(e = {}, ...r) {
  const {
    breakpoints: t = {},
    palette: n = {},
    spacing: i,
    shape: a = {},
    ...s
  } = e, c = it(t), u = Ur(i);
  let d = Q({
    breakpoints: c,
    direction: "ltr",
    components: {},
    // Inject component definitions.
    palette: {
      mode: "light",
      ...n
    },
    spacing: u,
    shape: {
      ...ut,
      ...a
    }
  }, s);
  return d = lt(d), d.applyStyles = Zt, d = r.reduce((h, p) => Q(h, p), d), d.unstable_sxConfig = {
    ...Ve,
    ...s == null ? void 0 : s.unstable_sxConfig
  }, d.unstable_sx = function(p) {
    return ze({
      sx: p,
      theme: this
    });
  }, d;
}
const Er = (e) => e, rn = () => {
  let e = Er;
  return {
    configure(r) {
      e = r;
    },
    generate(r) {
      return e(r);
    },
    reset() {
      e = Er;
    }
  };
}, tn = rn(), nn = {
  active: "active",
  checked: "checked",
  completed: "completed",
  disabled: "disabled",
  error: "error",
  expanded: "expanded",
  focused: "focused",
  focusVisible: "focusVisible",
  open: "open",
  readOnly: "readOnly",
  required: "required",
  selected: "selected"
};
function on(e, r, t = "Mui") {
  const n = nn[r];
  return n ? `${t}-${n}` : `${tn.generate(e)}-${r}`;
}
function an(e, r = Number.MIN_SAFE_INTEGER, t = Number.MAX_SAFE_INTEGER) {
  return Math.max(r, Math.min(e, t));
}
function sr(e, r = 0, t = 1) {
  return process.env.NODE_ENV !== "production" && (e < r || e > t) && console.error(`MUI: The value provided ${e} is out of range [${r}, ${t}].`), an(e, r, t);
}
function sn(e) {
  e = e.slice(1);
  const r = new RegExp(`.{1,${e.length >= 6 ? 2 : 1}}`, "g");
  let t = e.match(r);
  return t && t[0].length === 1 && (t = t.map((n) => n + n)), process.env.NODE_ENV !== "production" && e.length !== e.trim().length && console.error(`MUI: The color: "${e}" is invalid. Make sure the color input doesn't contain leading/trailing space.`), t ? `rgb${t.length === 4 ? "a" : ""}(${t.map((n, i) => i < 3 ? parseInt(n, 16) : Math.round(parseInt(n, 16) / 255 * 1e3) / 1e3).join(", ")})` : "";
}
function fe(e) {
  if (e.type)
    return e;
  if (e.charAt(0) === "#")
    return fe(sn(e));
  const r = e.indexOf("("), t = e.substring(0, r);
  if (!["rgb", "rgba", "hsl", "hsla", "color"].includes(t))
    throw new Error(process.env.NODE_ENV !== "production" ? `MUI: Unsupported \`${e}\` color.
The following formats are supported: #nnn, #nnnnnn, rgb(), rgba(), hsl(), hsla(), color().` : ue(9, e));
  let n = e.substring(r + 1, e.length - 1), i;
  if (t === "color") {
    if (n = n.split(" "), i = n.shift(), n.length === 4 && n[3].charAt(0) === "/" && (n[3] = n[3].slice(1)), !["srgb", "display-p3", "a98-rgb", "prophoto-rgb", "rec-2020"].includes(i))
      throw new Error(process.env.NODE_ENV !== "production" ? `MUI: unsupported \`${i}\` color space.
The following color spaces are supported: srgb, display-p3, a98-rgb, prophoto-rgb, rec-2020.` : ue(10, i));
  } else
    n = n.split(",");
  return n = n.map((a) => parseFloat(a)), {
    type: t,
    values: n,
    colorSpace: i
  };
}
const cn = (e) => {
  const r = fe(e);
  return r.values.slice(0, 3).map((t, n) => r.type.includes("hsl") && n !== 0 ? `${t}%` : t).join(" ");
}, Te = (e, r) => {
  try {
    return cn(e);
  } catch {
    return r && process.env.NODE_ENV !== "production" && console.warn(r), e;
  }
};
function Ge(e) {
  const {
    type: r,
    colorSpace: t
  } = e;
  let {
    values: n
  } = e;
  return r.includes("rgb") ? n = n.map((i, a) => a < 3 ? parseInt(i, 10) : i) : r.includes("hsl") && (n[1] = `${n[1]}%`, n[2] = `${n[2]}%`), r.includes("color") ? n = `${t} ${n.join(" ")}` : n = `${n.join(", ")}`, `${r}(${n})`;
}
function Fr(e) {
  e = fe(e);
  const {
    values: r
  } = e, t = r[0], n = r[1] / 100, i = r[2] / 100, a = n * Math.min(i, 1 - i), s = (d, h = (d + t / 30) % 12) => i - a * Math.max(Math.min(h - 3, 9 - h, 1), -1);
  let c = "rgb";
  const u = [Math.round(s(0) * 255), Math.round(s(8) * 255), Math.round(s(4) * 255)];
  return e.type === "hsla" && (c += "a", u.push(r[3])), Ge({
    type: c,
    values: u
  });
}
function tr(e) {
  e = fe(e);
  let r = e.type === "hsl" || e.type === "hsla" ? fe(Fr(e)).values : e.values;
  return r = r.map((t) => (e.type !== "color" && (t /= 255), t <= 0.03928 ? t / 12.92 : ((t + 0.055) / 1.055) ** 2.4)), Number((0.2126 * r[0] + 0.7152 * r[1] + 0.0722 * r[2]).toFixed(3));
}
function wr(e, r) {
  const t = tr(e), n = tr(r);
  return (Math.max(t, n) + 0.05) / (Math.min(t, n) + 0.05);
}
function ln(e, r) {
  return e = fe(e), r = sr(r), (e.type === "rgb" || e.type === "hsl") && (e.type += "a"), e.type === "color" ? e.values[3] = `/${r}` : e.values[3] = r, Ge(e);
}
function _e(e, r, t) {
  try {
    return ln(e, r);
  } catch {
    return e;
  }
}
function cr(e, r) {
  if (e = fe(e), r = sr(r), e.type.includes("hsl"))
    e.values[2] *= 1 - r;
  else if (e.type.includes("rgb") || e.type.includes("color"))
    for (let t = 0; t < 3; t += 1)
      e.values[t] *= 1 - r;
  return Ge(e);
}
function W(e, r, t) {
  try {
    return cr(e, r);
  } catch {
    return e;
  }
}
function lr(e, r) {
  if (e = fe(e), r = sr(r), e.type.includes("hsl"))
    e.values[2] += (100 - e.values[2]) * r;
  else if (e.type.includes("rgb"))
    for (let t = 0; t < 3; t += 1)
      e.values[t] += (255 - e.values[t]) * r;
  else if (e.type.includes("color"))
    for (let t = 0; t < 3; t += 1)
      e.values[t] += (1 - e.values[t]) * r;
  return Ge(e);
}
function U(e, r, t) {
  try {
    return lr(e, r);
  } catch {
    return e;
  }
}
function un(e, r = 0.15) {
  return tr(e) > 0.5 ? cr(e, r) : lr(e, r);
}
function Re(e, r, t) {
  try {
    return un(e, r);
  } catch {
    return e;
  }
}
function fn(e = "") {
  function r(...n) {
    if (!n.length)
      return "";
    const i = n[0];
    return typeof i == "string" && !i.match(/(#|\(|\)|(-?(\d*\.)?\d+)(px|em|%|ex|ch|rem|vw|vh|vmin|vmax|cm|mm|in|pt|pc))|^(-?(\d*\.)?\d+)$|(\d+ \d+ \d+)/) ? `, var(--${e ? `${e}-` : ""}${i}${r(...n.slice(1))})` : `, ${i}`;
  }
  return (n, ...i) => `var(--${e ? `${e}-` : ""}${n}${r(...i)})`;
}
const $r = (e, r, t, n = []) => {
  let i = e;
  r.forEach((a, s) => {
    s === r.length - 1 ? Array.isArray(i) ? i[Number(a)] = t : i && typeof i == "object" && (i[a] = t) : i && typeof i == "object" && (i[a] || (i[a] = n.includes(a) ? [] : {}), i = i[a]);
  });
}, dn = (e, r, t) => {
  function n(i, a = [], s = []) {
    Object.entries(i).forEach(([c, u]) => {
      (!t || t && !t([...a, c])) && u != null && (typeof u == "object" && Object.keys(u).length > 0 ? n(u, [...a, c], Array.isArray(u) ? [...s, c] : s) : r([...a, c], u, s));
    });
  }
  n(e);
}, pn = (e, r) => typeof r == "number" ? ["lineHeight", "fontWeight", "opacity", "zIndex"].some((n) => e.includes(n)) || e[e.length - 1].toLowerCase().includes("opacity") ? r : `${r}px` : r;
function Ze(e, r) {
  const {
    prefix: t,
    shouldSkipGeneratingVar: n
  } = r || {}, i = {}, a = {}, s = {};
  return dn(
    e,
    (c, u, d) => {
      if ((typeof u == "string" || typeof u == "number") && (!n || !n(c, u))) {
        const h = `--${t ? `${t}-` : ""}${c.join("-")}`, p = pn(c, u);
        Object.assign(i, {
          [h]: p
        }), $r(a, c, `var(${h})`, d), $r(s, c, `var(${h}, ${p})`, d);
      }
    },
    (c) => c[0] === "vars"
    // skip 'vars/*' paths
  ), {
    css: i,
    vars: a,
    varsWithDefaults: s
  };
}
function gn(e, r = {}) {
  const {
    getSelector: t = $,
    disableCssColorScheme: n,
    colorSchemeSelector: i
  } = r, {
    colorSchemes: a = {},
    components: s,
    defaultColorScheme: c = "light",
    ...u
  } = e, {
    vars: d,
    css: h,
    varsWithDefaults: p
  } = Ze(u, r);
  let f = p;
  const S = {}, {
    [c]: T,
    ...b
  } = a;
  if (Object.entries(b || {}).forEach(([w, E]) => {
    const {
      vars: m,
      css: q,
      varsWithDefaults: X
    } = Ze(E, r);
    f = Q(f, X), S[w] = {
      css: q,
      vars: m
    };
  }), T) {
    const {
      css: w,
      vars: E,
      varsWithDefaults: m
    } = Ze(T, r);
    f = Q(f, m), S[c] = {
      css: w,
      vars: E
    };
  }
  function $(w, E) {
    var q, X;
    let m = i;
    if (i === "class" && (m = ".%s"), i === "data" && (m = "[data-%s]"), i != null && i.startsWith("data-") && !i.includes("%s") && (m = `[${i}="%s"]`), w) {
      if (m === "media")
        return e.defaultColorScheme === w ? ":root" : {
          [`@media (prefers-color-scheme: ${((X = (q = a[w]) == null ? void 0 : q.palette) == null ? void 0 : X.mode) || w})`]: {
            ":root": E
          }
        };
      if (m)
        return e.defaultColorScheme === w ? `:root, ${m.replace("%s", String(w))}` : m.replace("%s", String(w));
    }
    return ":root";
  }
  return {
    vars: f,
    generateThemeVars: () => {
      let w = {
        ...d
      };
      return Object.entries(S).forEach(([, {
        vars: E
      }]) => {
        w = Q(w, E);
      }), w;
    },
    generateStyleSheets: () => {
      var ie, K;
      const w = [], E = e.defaultColorScheme || "light";
      function m(o, C) {
        Object.keys(C).length && w.push(typeof o == "string" ? {
          [o]: {
            ...C
          }
        } : o);
      }
      m(t(void 0, {
        ...h
      }), h);
      const {
        [E]: q,
        ...X
      } = S;
      if (q) {
        const {
          css: o
        } = q, C = (K = (ie = a[E]) == null ? void 0 : ie.palette) == null ? void 0 : K.mode, j = !n && C ? {
          colorScheme: C,
          ...o
        } : {
          ...o
        };
        m(t(E, {
          ...j
        }), j);
      }
      return Object.entries(X).forEach(([o, {
        css: C
      }]) => {
        var re, ce;
        const j = (ce = (re = a[o]) == null ? void 0 : re.palette) == null ? void 0 : ce.mode, L = !n && j ? {
          colorScheme: j,
          ...C
        } : {
          ...C
        };
        m(t(o, {
          ...L
        }), L);
      }), w;
    }
  };
}
function mn(e) {
  return function(t) {
    return e === "media" ? (process.env.NODE_ENV !== "production" && t !== "light" && t !== "dark" && console.error(`MUI: @media (prefers-color-scheme) supports only 'light' or 'dark', but receive '${t}'.`), `@media (prefers-color-scheme: ${t})`) : e ? e.startsWith("data-") && !e.includes("%s") ? `[${e}="${t}"] &` : e === "class" ? `.${t} &` : e === "data" ? `[data-${t}] &` : `${e.replace("%s", t)} &` : "&";
  };
}
function Lr() {
  return {
    // The colors used to style the text.
    text: {
      // The most important text.
      primary: "rgba(0, 0, 0, 0.87)",
      // Secondary text.
      secondary: "rgba(0, 0, 0, 0.6)",
      // Disabled text have even lower visual prominence.
      disabled: "rgba(0, 0, 0, 0.38)"
    },
    // The color used to divide different elements.
    divider: "rgba(0, 0, 0, 0.12)",
    // The background colors used to style the surfaces.
    // Consistency between these values is important.
    background: {
      paper: $e.white,
      default: $e.white
    },
    // The colors used to style the action elements.
    action: {
      // The color of an active action like an icon button.
      active: "rgba(0, 0, 0, 0.54)",
      // The color of an hovered action.
      hover: "rgba(0, 0, 0, 0.04)",
      hoverOpacity: 0.04,
      // The color of a selected action.
      selected: "rgba(0, 0, 0, 0.08)",
      selectedOpacity: 0.08,
      // The color of a disabled action.
      disabled: "rgba(0, 0, 0, 0.26)",
      // The background color of a disabled action.
      disabledBackground: "rgba(0, 0, 0, 0.12)",
      disabledOpacity: 0.38,
      focus: "rgba(0, 0, 0, 0.12)",
      focusOpacity: 0.12,
      activatedOpacity: 0.12
    }
  };
}
const hn = Lr();
function Vr() {
  return {
    text: {
      primary: $e.white,
      secondary: "rgba(255, 255, 255, 0.7)",
      disabled: "rgba(255, 255, 255, 0.5)",
      icon: "rgba(255, 255, 255, 0.5)"
    },
    divider: "rgba(255, 255, 255, 0.12)",
    background: {
      paper: "#121212",
      default: "#121212"
    },
    action: {
      active: $e.white,
      hover: "rgba(255, 255, 255, 0.08)",
      hoverOpacity: 0.08,
      selected: "rgba(255, 255, 255, 0.16)",
      selectedOpacity: 0.16,
      disabled: "rgba(255, 255, 255, 0.3)",
      disabledBackground: "rgba(255, 255, 255, 0.12)",
      disabledOpacity: 0.38,
      focus: "rgba(255, 255, 255, 0.12)",
      focusOpacity: 0.12,
      activatedOpacity: 0.24
    }
  };
}
const Or = Vr();
function Ar(e, r, t, n) {
  const i = n.light || n, a = n.dark || n * 1.5;
  e[r] || (e.hasOwnProperty(t) ? e[r] = e[t] : r === "light" ? e.light = lr(e.main, i) : r === "dark" && (e.dark = cr(e.main, a)));
}
function yn(e = "light") {
  return e === "dark" ? {
    main: he[200],
    light: he[50],
    dark: he[400]
  } : {
    main: he[700],
    light: he[400],
    dark: he[800]
  };
}
function bn(e = "light") {
  return e === "dark" ? {
    main: me[200],
    light: me[50],
    dark: me[400]
  } : {
    main: me[500],
    light: me[300],
    dark: me[700]
  };
}
function vn(e = "light") {
  return e === "dark" ? {
    main: ge[500],
    light: ge[300],
    dark: ge[700]
  } : {
    main: ge[700],
    light: ge[400],
    dark: ge[800]
  };
}
function Sn(e = "light") {
  return e === "dark" ? {
    main: ye[400],
    light: ye[300],
    dark: ye[700]
  } : {
    main: ye[700],
    light: ye[500],
    dark: ye[900]
  };
}
function Cn(e = "light") {
  return e === "dark" ? {
    main: be[400],
    light: be[300],
    dark: be[700]
  } : {
    main: be[800],
    light: be[500],
    dark: be[900]
  };
}
function Tn(e = "light") {
  return e === "dark" ? {
    main: Se[400],
    light: Se[300],
    dark: Se[700]
  } : {
    main: "#ed6c02",
    // closest to orange[800] that pass 3:1.
    light: Se[500],
    dark: Se[900]
  };
}
function ur(e) {
  const {
    mode: r = "light",
    contrastThreshold: t = 3,
    tonalOffset: n = 0.2,
    ...i
  } = e, a = e.primary || yn(r), s = e.secondary || bn(r), c = e.error || vn(r), u = e.info || Sn(r), d = e.success || Cn(r), h = e.warning || Tn(r);
  function p(b) {
    const $ = wr(b, Or.text.primary) >= t ? Or.text.primary : hn.text.primary;
    if (process.env.NODE_ENV !== "production") {
      const A = wr(b, $);
      A < 3 && console.error([`MUI: The contrast ratio of ${A}:1 for ${$} on ${b}`, "falls below the WCAG recommended absolute minimum contrast ratio of 3:1.", "https://www.w3.org/TR/2008/REC-WCAG20-20081211/#visual-audio-contrast-contrast"].join(`
`));
    }
    return $;
  }
  const f = ({
    color: b,
    name: $,
    mainShade: A = 500,
    lightShade: Y = 300,
    darkShade: w = 700
  }) => {
    if (b = {
      ...b
    }, !b.main && b[A] && (b.main = b[A]), !b.hasOwnProperty("main"))
      throw new Error(process.env.NODE_ENV !== "production" ? `MUI: The color${$ ? ` (${$})` : ""} provided to augmentColor(color) is invalid.
The color object needs to have a \`main\` property or a \`${A}\` property.` : ue(11, $ ? ` (${$})` : "", A));
    if (typeof b.main != "string")
      throw new Error(process.env.NODE_ENV !== "production" ? `MUI: The color${$ ? ` (${$})` : ""} provided to augmentColor(color) is invalid.
\`color.main\` should be a string, but \`${JSON.stringify(b.main)}\` was provided instead.

Did you intend to use one of the following approaches?

import { green } from "@mui/material/colors";

const theme1 = createTheme({ palette: {
  primary: green,
} });

const theme2 = createTheme({ palette: {
  primary: { main: green[500] },
} });` : ue(12, $ ? ` (${$})` : "", JSON.stringify(b.main)));
    return Ar(b, "light", Y, n), Ar(b, "dark", w, n), b.contrastText || (b.contrastText = p(b.main)), b;
  };
  let S;
  return r === "light" ? S = Lr() : r === "dark" && (S = Vr()), process.env.NODE_ENV !== "production" && (S || console.error(`MUI: The palette mode \`${r}\` is not supported.`)), Q({
    // A collection of common colors.
    common: {
      ...$e
    },
    // prevent mutable object.
    // The palette mode, can be light or dark.
    mode: r,
    // The colors used to represent primary interface elements for a user.
    primary: f({
      color: a,
      name: "primary"
    }),
    // The colors used to represent secondary interface elements for a user.
    secondary: f({
      color: s,
      name: "secondary",
      mainShade: "A400",
      lightShade: "A200",
      darkShade: "A700"
    }),
    // The colors used to represent interface elements that the user should be made aware of.
    error: f({
      color: c,
      name: "error"
    }),
    // The colors used to represent potentially dangerous actions or important messages.
    warning: f({
      color: h,
      name: "warning"
    }),
    // The colors used to present information to the user that is neutral and not necessarily important.
    info: f({
      color: u,
      name: "info"
    }),
    // The colors used to indicate the successful completion of an action that user triggered.
    success: f({
      color: d,
      name: "success"
    }),
    // The grey colors.
    grey: qr,
    // Used by `getContrastText()` to maximize the contrast between
    // the background and the text.
    contrastThreshold: t,
    // Takes a background color and returns the text color that maximizes the contrast.
    getContrastText: p,
    // Generate a rich color object.
    augmentColor: f,
    // Used by the functions below to shift a color's luminance by approximately
    // two indexes within its tonal palette.
    // E.g., shift from Red 500 to Red 300 or Red 700.
    tonalOffset: n,
    // The light and dark mode object.
    ...S
  }, i);
}
function En(e) {
  const r = {};
  return Object.entries(e).forEach((n) => {
    const [i, a] = n;
    typeof a == "object" && (r[i] = `${a.fontStyle ? `${a.fontStyle} ` : ""}${a.fontVariant ? `${a.fontVariant} ` : ""}${a.fontWeight ? `${a.fontWeight} ` : ""}${a.fontStretch ? `${a.fontStretch} ` : ""}${a.fontSize || ""}${a.lineHeight ? `/${a.lineHeight} ` : ""}${a.fontFamily || ""}`);
  }), r;
}
function wn(e, r) {
  return {
    toolbar: {
      minHeight: 56,
      [e.up("xs")]: {
        "@media (orientation: landscape)": {
          minHeight: 48
        }
      },
      [e.up("sm")]: {
        minHeight: 64
      }
    },
    ...r
  };
}
function $n(e) {
  return Math.round(e * 1e5) / 1e5;
}
const xr = {
  textTransform: "uppercase"
}, kr = '"Roboto", "Helvetica", "Arial", sans-serif';
function On(e, r) {
  const {
    fontFamily: t = kr,
    // The default font size of the Material Specification.
    fontSize: n = 14,
    // px
    fontWeightLight: i = 300,
    fontWeightRegular: a = 400,
    fontWeightMedium: s = 500,
    fontWeightBold: c = 700,
    // Tell MUI what's the font-size on the html element.
    // 16px is the default font-size used by browsers.
    htmlFontSize: u = 16,
    // Apply the CSS properties to all the variants.
    allVariants: d,
    pxToRem: h,
    ...p
  } = typeof r == "function" ? r(e) : r;
  process.env.NODE_ENV !== "production" && (typeof n != "number" && console.error("MUI: `fontSize` is required to be a number."), typeof u != "number" && console.error("MUI: `htmlFontSize` is required to be a number."));
  const f = n / 14, S = h || (($) => `${$ / u * f}rem`), T = ($, A, Y, w, E) => ({
    fontFamily: t,
    fontWeight: $,
    fontSize: S(A),
    // Unitless following https://meyerweb.com/eric/thoughts/2006/02/08/unitless-line-heights/
    lineHeight: Y,
    // The letter spacing was designed for the Roboto font-family. Using the same letter-spacing
    // across font-families can cause issues with the kerning.
    ...t === kr ? {
      letterSpacing: `${$n(w / A)}em`
    } : {},
    ...E,
    ...d
  }), b = {
    h1: T(i, 96, 1.167, -1.5),
    h2: T(i, 60, 1.2, -0.5),
    h3: T(a, 48, 1.167, 0),
    h4: T(a, 34, 1.235, 0.25),
    h5: T(a, 24, 1.334, 0),
    h6: T(s, 20, 1.6, 0.15),
    subtitle1: T(a, 16, 1.75, 0.15),
    subtitle2: T(s, 14, 1.57, 0.1),
    body1: T(a, 16, 1.5, 0.15),
    body2: T(a, 14, 1.43, 0.15),
    button: T(s, 14, 1.75, 0.4, xr),
    caption: T(a, 12, 1.66, 0.4),
    overline: T(a, 12, 2.66, 1, xr),
    // TODO v6: Remove handling of 'inherit' variant from the theme as it is already handled in Material UI's Typography component. Also, remember to remove the associated types.
    inherit: {
      fontFamily: "inherit",
      fontWeight: "inherit",
      fontSize: "inherit",
      lineHeight: "inherit",
      letterSpacing: "inherit"
    }
  };
  return Q({
    htmlFontSize: u,
    pxToRem: S,
    fontFamily: t,
    fontSize: n,
    fontWeightLight: i,
    fontWeightRegular: a,
    fontWeightMedium: s,
    fontWeightBold: c,
    ...b
  }, p, {
    clone: !1
    // No need to clone deep
  });
}
const An = 0.2, xn = 0.14, kn = 0.12;
function F(...e) {
  return [`${e[0]}px ${e[1]}px ${e[2]}px ${e[3]}px rgba(0,0,0,${An})`, `${e[4]}px ${e[5]}px ${e[6]}px ${e[7]}px rgba(0,0,0,${xn})`, `${e[8]}px ${e[9]}px ${e[10]}px ${e[11]}px rgba(0,0,0,${kn})`].join(",");
}
const Pn = ["none", F(0, 2, 1, -1, 0, 1, 1, 0, 0, 1, 3, 0), F(0, 3, 1, -2, 0, 2, 2, 0, 0, 1, 5, 0), F(0, 3, 3, -2, 0, 3, 4, 0, 0, 1, 8, 0), F(0, 2, 4, -1, 0, 4, 5, 0, 0, 1, 10, 0), F(0, 3, 5, -1, 0, 5, 8, 0, 0, 1, 14, 0), F(0, 3, 5, -1, 0, 6, 10, 0, 0, 1, 18, 0), F(0, 4, 5, -2, 0, 7, 10, 1, 0, 2, 16, 1), F(0, 5, 5, -3, 0, 8, 10, 1, 0, 3, 14, 2), F(0, 5, 6, -3, 0, 9, 12, 1, 0, 3, 16, 2), F(0, 6, 6, -3, 0, 10, 14, 1, 0, 4, 18, 3), F(0, 6, 7, -4, 0, 11, 15, 1, 0, 4, 20, 3), F(0, 7, 8, -4, 0, 12, 17, 2, 0, 5, 22, 4), F(0, 7, 8, -4, 0, 13, 19, 2, 0, 5, 24, 4), F(0, 7, 9, -4, 0, 14, 21, 2, 0, 5, 26, 4), F(0, 8, 9, -5, 0, 15, 22, 2, 0, 6, 28, 5), F(0, 8, 10, -5, 0, 16, 24, 2, 0, 6, 30, 5), F(0, 8, 11, -5, 0, 17, 26, 2, 0, 6, 32, 5), F(0, 9, 11, -5, 0, 18, 28, 2, 0, 7, 34, 6), F(0, 9, 12, -6, 0, 19, 29, 2, 0, 7, 36, 6), F(0, 10, 13, -6, 0, 20, 31, 3, 0, 8, 38, 7), F(0, 10, 13, -6, 0, 21, 33, 3, 0, 8, 40, 7), F(0, 10, 14, -6, 0, 22, 35, 3, 0, 8, 42, 7), F(0, 11, 14, -7, 0, 23, 36, 3, 0, 9, 44, 8), F(0, 11, 15, -7, 0, 24, 38, 3, 0, 9, 46, 8)], _n = {
  // This is the most common easing curve.
  easeInOut: "cubic-bezier(0.4, 0, 0.2, 1)",
  // Objects enter the screen at full velocity from off-screen and
  // slowly decelerate to a resting point.
  easeOut: "cubic-bezier(0.0, 0, 0.2, 1)",
  // Objects leave the screen at full velocity. They do not decelerate when off-screen.
  easeIn: "cubic-bezier(0.4, 0, 1, 1)",
  // The sharp curve is used by objects that may return to the screen at any time.
  sharp: "cubic-bezier(0.4, 0, 0.6, 1)"
}, Rn = {
  shortest: 150,
  shorter: 200,
  short: 250,
  // most basic recommended timing
  standard: 300,
  // this is to be used in complex animations
  complex: 375,
  // recommended when something is entering screen
  enteringScreen: 225,
  // recommended when something is leaving screen
  leavingScreen: 195
};
function Pr(e) {
  return `${Math.round(e)}ms`;
}
function In(e) {
  if (!e)
    return 0;
  const r = e / 36;
  return Math.min(Math.round((4 + 15 * r ** 0.25 + r / 5) * 10), 3e3);
}
function Bn(e) {
  const r = {
    ..._n,
    ...e.easing
  }, t = {
    ...Rn,
    ...e.duration
  };
  return {
    getAutoHeightDuration: In,
    create: (i = ["all"], a = {}) => {
      const {
        duration: s = t.standard,
        easing: c = r.easeInOut,
        delay: u = 0,
        ...d
      } = a;
      if (process.env.NODE_ENV !== "production") {
        const h = (f) => typeof f == "string", p = (f) => !Number.isNaN(parseFloat(f));
        !h(i) && !Array.isArray(i) && console.error('MUI: Argument "props" must be a string or Array.'), !p(s) && !h(s) && console.error(`MUI: Argument "duration" must be a number or a string but found ${s}.`), h(c) || console.error('MUI: Argument "easing" must be a string.'), !p(u) && !h(u) && console.error('MUI: Argument "delay" must be a number or a string.'), typeof a != "object" && console.error(["MUI: Secong argument of transition.create must be an object.", "Arguments should be either `create('prop1', options)` or `create(['prop1', 'prop2'], options)`"].join(`
`)), Object.keys(d).length !== 0 && console.error(`MUI: Unrecognized argument(s) [${Object.keys(d).join(",")}].`);
      }
      return (Array.isArray(i) ? i : [i]).map((h) => `${h} ${typeof s == "string" ? s : Pr(s)} ${c} ${typeof u == "string" ? u : Pr(u)}`).join(",");
    },
    ...e,
    easing: r,
    duration: t
  };
}
const jn = {
  mobileStepper: 1e3,
  fab: 1050,
  speedDial: 1050,
  appBar: 1100,
  drawer: 1200,
  modal: 1300,
  snackbar: 1400,
  tooltip: 1500
};
function Dn(e) {
  return le(e) || typeof e > "u" || typeof e == "string" || typeof e == "boolean" || typeof e == "number" || Array.isArray(e);
}
function zr(e = {}) {
  const r = {
    ...e
  };
  function t(n) {
    const i = Object.entries(n);
    for (let a = 0; a < i.length; a++) {
      const [s, c] = i[a];
      !Dn(c) || s.startsWith("unstable_") ? delete n[s] : le(c) && (n[s] = {
        ...c
      }, t(n[s]));
    }
  }
  return t(r), `import { unstable_createBreakpoints as createBreakpoints, createTransitions } from '@mui/material/styles';

const theme = ${JSON.stringify(r, null, 2)};

theme.breakpoints = createBreakpoints(theme.breakpoints || {});
theme.transitions = createTransitions(theme.transitions || {});

export default theme;`;
}
function nr(e = {}, ...r) {
  const {
    breakpoints: t,
    mixins: n = {},
    spacing: i,
    palette: a = {},
    transitions: s = {},
    typography: c = {},
    shape: u,
    ...d
  } = e;
  if (e.vars)
    throw new Error(process.env.NODE_ENV !== "production" ? "MUI: `vars` is a private field used for CSS variables support.\nPlease use another name." : ue(20));
  const h = ur(a), p = en(e);
  let f = Q(p, {
    mixins: wn(p.breakpoints, n),
    palette: h,
    // Don't use [...shadows] until you've verified its transpiled code is not invoking the iterator protocol.
    shadows: Pn.slice(),
    typography: On(h, c),
    transitions: Bn(s),
    zIndex: {
      ...jn
    }
  });
  if (f = Q(f, d), f = r.reduce((S, T) => Q(S, T), f), process.env.NODE_ENV !== "production") {
    const S = ["active", "checked", "completed", "disabled", "error", "expanded", "focused", "focusVisible", "required", "selected"], T = (b, $) => {
      let A;
      for (A in b) {
        const Y = b[A];
        if (S.includes(A) && Object.keys(Y).length > 0) {
          if (process.env.NODE_ENV !== "production") {
            const w = on("", A);
            console.error([`MUI: The \`${$}\` component increases the CSS specificity of the \`${A}\` internal state.`, "You can not override it like this: ", JSON.stringify(b, null, 2), "", `Instead, you need to use the '&.${w}' syntax:`, JSON.stringify({
              root: {
                [`&.${w}`]: Y
              }
            }, null, 2), "", "https://mui.com/r/state-classes-guide"].join(`
`));
          }
          b[A] = {};
        }
      }
    };
    Object.keys(f.components).forEach((b) => {
      const $ = f.components[b].styleOverrides;
      $ && b.startsWith("Mui") && T($, b);
    });
  }
  return f.unstable_sxConfig = {
    ...Ve,
    ...d == null ? void 0 : d.unstable_sxConfig
  }, f.unstable_sx = function(T) {
    return ze({
      sx: T,
      theme: this
    });
  }, f.toRuntimeSource = zr, f;
}
function Mn(e) {
  let r;
  return e < 1 ? r = 5.11916 * e ** 2 : r = 4.5 * Math.log(e + 1) + 2, Math.round(r * 10) / 1e3;
}
const Nn = [...Array(25)].map((e, r) => {
  if (r === 0)
    return "none";
  const t = Mn(r);
  return `linear-gradient(rgba(255 255 255 / ${t}), rgba(255 255 255 / ${t}))`;
});
function Gr(e) {
  return {
    inputPlaceholder: e === "dark" ? 0.5 : 0.42,
    inputUnderline: e === "dark" ? 0.7 : 0.42,
    switchTrackDisabled: e === "dark" ? 0.2 : 0.12,
    switchTrack: e === "dark" ? 0.3 : 0.38
  };
}
function Yr(e) {
  return e === "dark" ? Nn : [];
}
function Wn(e) {
  const {
    palette: r = {
      mode: "light"
    },
    // need to cast to avoid module augmentation test
    opacity: t,
    overlays: n,
    ...i
  } = e, a = ur(r);
  return {
    palette: a,
    opacity: {
      ...Gr(a.mode),
      ...t
    },
    overlays: n || Yr(a.mode),
    ...i
  };
}
function Un(e) {
  var r;
  return !!e[0].match(/(cssVarPrefix|colorSchemeSelector|rootSelector|typography|mixins|breakpoints|direction|transitions)/) || !!e[0].match(/sxConfig$/) || // ends with sxConfig
  e[0] === "palette" && !!((r = e[1]) != null && r.match(/(mode|contrastThreshold|tonalOffset)/));
}
const Fn = (e) => [...[...Array(25)].map((r, t) => `--${e ? `${e}-` : ""}overlays-${t}`), `--${e ? `${e}-` : ""}palette-AppBar-darkBg`, `--${e ? `${e}-` : ""}palette-AppBar-darkColor`], Ln = (e) => (r, t) => {
  const n = e.rootSelector || ":root", i = e.colorSchemeSelector;
  let a = i;
  if (i === "class" && (a = ".%s"), i === "data" && (a = "[data-%s]"), i != null && i.startsWith("data-") && !i.includes("%s") && (a = `[${i}="%s"]`), e.defaultColorScheme === r) {
    if (r === "dark") {
      const s = {};
      return Fn(e.cssVarPrefix).forEach((c) => {
        s[c] = t[c], delete t[c];
      }), a === "media" ? {
        [n]: t,
        "@media (prefers-color-scheme: dark)": {
          [n]: s
        }
      } : a ? {
        [a.replace("%s", r)]: s,
        [`${n}, ${a.replace("%s", r)}`]: t
      } : {
        [n]: {
          ...t,
          ...s
        }
      };
    }
    if (a && a !== "media")
      return `${n}, ${a.replace("%s", String(r))}`;
  } else if (r) {
    if (a === "media")
      return {
        [`@media (prefers-color-scheme: ${String(r)})`]: {
          [n]: t
        }
      };
    if (a)
      return a.replace("%s", String(r));
  }
  return n;
};
function Vn(e, r) {
  r.forEach((t) => {
    e[t] || (e[t] = {});
  });
}
function l(e, r, t) {
  !e[r] && t && (e[r] = t);
}
function Ee(e) {
  return typeof e != "string" || !e.startsWith("hsl") ? e : Fr(e);
}
function ae(e, r) {
  `${r}Channel` in e || (e[`${r}Channel`] = Te(Ee(e[r]), `MUI: Can't create \`palette.${r}Channel\` because \`palette.${r}\` is not one of these formats: #nnn, #nnnnnn, rgb(), rgba(), hsl(), hsla(), color().
To suppress this warning, you need to explicitly provide the \`palette.${r}Channel\` as a string (in rgb format, for example "12 12 12") or undefined if you want to remove the channel token.`));
}
function zn(e) {
  return typeof e == "number" ? `${e}px` : typeof e == "string" || typeof e == "function" || Array.isArray(e) ? e : "8px";
}
const oe = (e) => {
  try {
    return e();
  } catch {
  }
}, Gn = (e = "mui") => fn(e);
function er(e, r, t, n) {
  if (!r)
    return;
  r = r === !0 ? {} : r;
  const i = n === "dark" ? "dark" : "light";
  if (!t) {
    e[n] = Wn({
      ...r,
      palette: {
        mode: i,
        ...r == null ? void 0 : r.palette
      }
    });
    return;
  }
  const {
    palette: a,
    ...s
  } = nr({
    ...t,
    palette: {
      mode: i,
      ...r == null ? void 0 : r.palette
    }
  });
  return e[n] = {
    ...r,
    palette: a,
    opacity: {
      ...Gr(i),
      ...r == null ? void 0 : r.opacity
    },
    overlays: (r == null ? void 0 : r.overlays) || Yr(i)
  }, s;
}
function Yn(e = {}, ...r) {
  const {
    colorSchemes: t = {
      light: !0
    },
    defaultColorScheme: n,
    disableCssColorScheme: i = !1,
    cssVarPrefix: a = "mui",
    shouldSkipGeneratingVar: s = Un,
    colorSchemeSelector: c = t.light && t.dark ? "media" : void 0,
    rootSelector: u = ":root",
    ...d
  } = e, h = Object.keys(t)[0], p = n || (t.light && h !== "light" ? "light" : h), f = Gn(a), {
    [p]: S,
    light: T,
    dark: b,
    ...$
  } = t, A = {
    ...$
  };
  let Y = S;
  if ((p === "dark" && !("dark" in t) || p === "light" && !("light" in t)) && (Y = !0), !Y)
    throw new Error(process.env.NODE_ENV !== "production" ? `MUI: The \`colorSchemes.${p}\` option is either missing or invalid.` : ue(21, p));
  const w = er(A, Y, d, p);
  T && !A.light && er(A, T, void 0, "light"), b && !A.dark && er(A, b, void 0, "dark");
  let E = {
    defaultColorScheme: p,
    ...w,
    cssVarPrefix: a,
    colorSchemeSelector: c,
    rootSelector: u,
    getCssVar: f,
    colorSchemes: A,
    font: {
      ...En(w.typography),
      ...w.font
    },
    spacing: zn(d.spacing)
  };
  Object.keys(E.colorSchemes).forEach((K) => {
    const o = E.colorSchemes[K].palette, C = (j) => {
      const L = j.split("-"), re = L[1], ce = L[2];
      return f(j, o[re][ce]);
    };
    if (o.mode === "light" && (l(o.common, "background", "#fff"), l(o.common, "onBackground", "#000")), o.mode === "dark" && (l(o.common, "background", "#000"), l(o.common, "onBackground", "#fff")), Vn(o, ["Alert", "AppBar", "Avatar", "Button", "Chip", "FilledInput", "LinearProgress", "Skeleton", "Slider", "SnackbarContent", "SpeedDialAction", "StepConnector", "StepContent", "Switch", "TableCell", "Tooltip"]), o.mode === "light") {
      l(o.Alert, "errorColor", W(o.error.light, 0.6)), l(o.Alert, "infoColor", W(o.info.light, 0.6)), l(o.Alert, "successColor", W(o.success.light, 0.6)), l(o.Alert, "warningColor", W(o.warning.light, 0.6)), l(o.Alert, "errorFilledBg", C("palette-error-main")), l(o.Alert, "infoFilledBg", C("palette-info-main")), l(o.Alert, "successFilledBg", C("palette-success-main")), l(o.Alert, "warningFilledBg", C("palette-warning-main")), l(o.Alert, "errorFilledColor", oe(() => o.getContrastText(o.error.main))), l(o.Alert, "infoFilledColor", oe(() => o.getContrastText(o.info.main))), l(o.Alert, "successFilledColor", oe(() => o.getContrastText(o.success.main))), l(o.Alert, "warningFilledColor", oe(() => o.getContrastText(o.warning.main))), l(o.Alert, "errorStandardBg", U(o.error.light, 0.9)), l(o.Alert, "infoStandardBg", U(o.info.light, 0.9)), l(o.Alert, "successStandardBg", U(o.success.light, 0.9)), l(o.Alert, "warningStandardBg", U(o.warning.light, 0.9)), l(o.Alert, "errorIconColor", C("palette-error-main")), l(o.Alert, "infoIconColor", C("palette-info-main")), l(o.Alert, "successIconColor", C("palette-success-main")), l(o.Alert, "warningIconColor", C("palette-warning-main")), l(o.AppBar, "defaultBg", C("palette-grey-100")), l(o.Avatar, "defaultBg", C("palette-grey-400")), l(o.Button, "inheritContainedBg", C("palette-grey-300")), l(o.Button, "inheritContainedHoverBg", C("palette-grey-A100")), l(o.Chip, "defaultBorder", C("palette-grey-400")), l(o.Chip, "defaultAvatarColor", C("palette-grey-700")), l(o.Chip, "defaultIconColor", C("palette-grey-700")), l(o.FilledInput, "bg", "rgba(0, 0, 0, 0.06)"), l(o.FilledInput, "hoverBg", "rgba(0, 0, 0, 0.09)"), l(o.FilledInput, "disabledBg", "rgba(0, 0, 0, 0.12)"), l(o.LinearProgress, "primaryBg", U(o.primary.main, 0.62)), l(o.LinearProgress, "secondaryBg", U(o.secondary.main, 0.62)), l(o.LinearProgress, "errorBg", U(o.error.main, 0.62)), l(o.LinearProgress, "infoBg", U(o.info.main, 0.62)), l(o.LinearProgress, "successBg", U(o.success.main, 0.62)), l(o.LinearProgress, "warningBg", U(o.warning.main, 0.62)), l(o.Skeleton, "bg", `rgba(${C("palette-text-primaryChannel")} / 0.11)`), l(o.Slider, "primaryTrack", U(o.primary.main, 0.62)), l(o.Slider, "secondaryTrack", U(o.secondary.main, 0.62)), l(o.Slider, "errorTrack", U(o.error.main, 0.62)), l(o.Slider, "infoTrack", U(o.info.main, 0.62)), l(o.Slider, "successTrack", U(o.success.main, 0.62)), l(o.Slider, "warningTrack", U(o.warning.main, 0.62));
      const j = Re(o.background.default, 0.8);
      l(o.SnackbarContent, "bg", j), l(o.SnackbarContent, "color", oe(() => o.getContrastText(j))), l(o.SpeedDialAction, "fabHoverBg", Re(o.background.paper, 0.15)), l(o.StepConnector, "border", C("palette-grey-400")), l(o.StepContent, "border", C("palette-grey-400")), l(o.Switch, "defaultColor", C("palette-common-white")), l(o.Switch, "defaultDisabledColor", C("palette-grey-100")), l(o.Switch, "primaryDisabledColor", U(o.primary.main, 0.62)), l(o.Switch, "secondaryDisabledColor", U(o.secondary.main, 0.62)), l(o.Switch, "errorDisabledColor", U(o.error.main, 0.62)), l(o.Switch, "infoDisabledColor", U(o.info.main, 0.62)), l(o.Switch, "successDisabledColor", U(o.success.main, 0.62)), l(o.Switch, "warningDisabledColor", U(o.warning.main, 0.62)), l(o.TableCell, "border", U(_e(o.divider, 1), 0.88)), l(o.Tooltip, "bg", _e(o.grey[700], 0.92));
    }
    if (o.mode === "dark") {
      l(o.Alert, "errorColor", U(o.error.light, 0.6)), l(o.Alert, "infoColor", U(o.info.light, 0.6)), l(o.Alert, "successColor", U(o.success.light, 0.6)), l(o.Alert, "warningColor", U(o.warning.light, 0.6)), l(o.Alert, "errorFilledBg", C("palette-error-dark")), l(o.Alert, "infoFilledBg", C("palette-info-dark")), l(o.Alert, "successFilledBg", C("palette-success-dark")), l(o.Alert, "warningFilledBg", C("palette-warning-dark")), l(o.Alert, "errorFilledColor", oe(() => o.getContrastText(o.error.dark))), l(o.Alert, "infoFilledColor", oe(() => o.getContrastText(o.info.dark))), l(o.Alert, "successFilledColor", oe(() => o.getContrastText(o.success.dark))), l(o.Alert, "warningFilledColor", oe(() => o.getContrastText(o.warning.dark))), l(o.Alert, "errorStandardBg", W(o.error.light, 0.9)), l(o.Alert, "infoStandardBg", W(o.info.light, 0.9)), l(o.Alert, "successStandardBg", W(o.success.light, 0.9)), l(o.Alert, "warningStandardBg", W(o.warning.light, 0.9)), l(o.Alert, "errorIconColor", C("palette-error-main")), l(o.Alert, "infoIconColor", C("palette-info-main")), l(o.Alert, "successIconColor", C("palette-success-main")), l(o.Alert, "warningIconColor", C("palette-warning-main")), l(o.AppBar, "defaultBg", C("palette-grey-900")), l(o.AppBar, "darkBg", C("palette-background-paper")), l(o.AppBar, "darkColor", C("palette-text-primary")), l(o.Avatar, "defaultBg", C("palette-grey-600")), l(o.Button, "inheritContainedBg", C("palette-grey-800")), l(o.Button, "inheritContainedHoverBg", C("palette-grey-700")), l(o.Chip, "defaultBorder", C("palette-grey-700")), l(o.Chip, "defaultAvatarColor", C("palette-grey-300")), l(o.Chip, "defaultIconColor", C("palette-grey-300")), l(o.FilledInput, "bg", "rgba(255, 255, 255, 0.09)"), l(o.FilledInput, "hoverBg", "rgba(255, 255, 255, 0.13)"), l(o.FilledInput, "disabledBg", "rgba(255, 255, 255, 0.12)"), l(o.LinearProgress, "primaryBg", W(o.primary.main, 0.5)), l(o.LinearProgress, "secondaryBg", W(o.secondary.main, 0.5)), l(o.LinearProgress, "errorBg", W(o.error.main, 0.5)), l(o.LinearProgress, "infoBg", W(o.info.main, 0.5)), l(o.LinearProgress, "successBg", W(o.success.main, 0.5)), l(o.LinearProgress, "warningBg", W(o.warning.main, 0.5)), l(o.Skeleton, "bg", `rgba(${C("palette-text-primaryChannel")} / 0.13)`), l(o.Slider, "primaryTrack", W(o.primary.main, 0.5)), l(o.Slider, "secondaryTrack", W(o.secondary.main, 0.5)), l(o.Slider, "errorTrack", W(o.error.main, 0.5)), l(o.Slider, "infoTrack", W(o.info.main, 0.5)), l(o.Slider, "successTrack", W(o.success.main, 0.5)), l(o.Slider, "warningTrack", W(o.warning.main, 0.5));
      const j = Re(o.background.default, 0.98);
      l(o.SnackbarContent, "bg", j), l(o.SnackbarContent, "color", oe(() => o.getContrastText(j))), l(o.SpeedDialAction, "fabHoverBg", Re(o.background.paper, 0.15)), l(o.StepConnector, "border", C("palette-grey-600")), l(o.StepContent, "border", C("palette-grey-600")), l(o.Switch, "defaultColor", C("palette-grey-300")), l(o.Switch, "defaultDisabledColor", C("palette-grey-600")), l(o.Switch, "primaryDisabledColor", W(o.primary.main, 0.55)), l(o.Switch, "secondaryDisabledColor", W(o.secondary.main, 0.55)), l(o.Switch, "errorDisabledColor", W(o.error.main, 0.55)), l(o.Switch, "infoDisabledColor", W(o.info.main, 0.55)), l(o.Switch, "successDisabledColor", W(o.success.main, 0.55)), l(o.Switch, "warningDisabledColor", W(o.warning.main, 0.55)), l(o.TableCell, "border", W(_e(o.divider, 1), 0.68)), l(o.Tooltip, "bg", _e(o.grey[700], 0.92));
    }
    ae(o.background, "default"), ae(o.background, "paper"), ae(o.common, "background"), ae(o.common, "onBackground"), ae(o, "divider"), Object.keys(o).forEach((j) => {
      const L = o[j];
      j !== "tonalOffset" && L && typeof L == "object" && (L.main && l(o[j], "mainChannel", Te(Ee(L.main))), L.light && l(o[j], "lightChannel", Te(Ee(L.light))), L.dark && l(o[j], "darkChannel", Te(Ee(L.dark))), L.contrastText && l(o[j], "contrastTextChannel", Te(Ee(L.contrastText))), j === "text" && (ae(o[j], "primary"), ae(o[j], "secondary")), j === "action" && (L.active && ae(o[j], "active"), L.selected && ae(o[j], "selected")));
    });
  }), E = r.reduce((K, o) => Q(K, o), E);
  const m = {
    prefix: a,
    disableCssColorScheme: i,
    shouldSkipGeneratingVar: s,
    getSelector: Ln(E)
  }, {
    vars: q,
    generateThemeVars: X,
    generateStyleSheets: ie
  } = gn(E, m);
  return E.vars = q, Object.entries(E.colorSchemes[E.defaultColorScheme]).forEach(([K, o]) => {
    E[K] = o;
  }), E.generateThemeVars = X, E.generateStyleSheets = ie, E.generateSpacing = function() {
    return Ur(d.spacing, ir(this));
  }, E.getColorSchemeSelector = mn(c), E.spacing = E.generateSpacing(), E.shouldSkipGeneratingVar = s, E.unstable_sxConfig = {
    ...Ve,
    ...d == null ? void 0 : d.unstable_sxConfig
  }, E.unstable_sx = function(o) {
    return ze({
      sx: o,
      theme: this
    });
  }, E.toRuntimeSource = zr, E;
}
function _r(e, r, t) {
  e.colorSchemes && t && (e.colorSchemes[r] = {
    ...t !== !0 && t,
    palette: ur({
      ...t === !0 ? {} : t.palette,
      mode: r
    })
    // cast type to skip module augmentation test
  });
}
function Qn(e = {}, ...r) {
  const {
    palette: t,
    cssVariables: n = !1,
    colorSchemes: i = t ? void 0 : {
      light: !0
    },
    defaultColorScheme: a = t == null ? void 0 : t.mode,
    ...s
  } = e, c = a || "light", u = i == null ? void 0 : i[c], d = {
    ...i,
    ...t ? {
      [c]: {
        ...typeof u != "boolean" && u,
        palette: t
      }
    } : void 0
  };
  if (n === !1) {
    if (!("colorSchemes" in e))
      return nr(e, ...r);
    let h = t;
    "palette" in e || d[c] && (d[c] !== !0 ? h = d[c].palette : c === "dark" && (h = {
      mode: "dark"
    }));
    const p = nr({
      ...e,
      palette: h
    }, ...r);
    return p.defaultColorScheme = c, p.colorSchemes = d, p.palette.mode === "light" && (p.colorSchemes.light = {
      ...d.light !== !0 && d.light,
      palette: p.palette
    }, _r(p, "dark", d.dark)), p.palette.mode === "dark" && (p.colorSchemes.dark = {
      ...d.dark !== !0 && d.dark,
      palette: p.palette
    }, _r(p, "light", d.light)), p;
  }
  return !t && !("light" in d) && c === "light" && (d.light = !0), Yn({
    ...s,
    colorSchemes: d,
    defaultColorScheme: c,
    ...typeof n != "boolean" && n
  }, ...r);
}
export {
  tn as C,
  Ce as P,
  ln as a,
  en as b,
  Qn as c,
  Mn as d,
  Nr as e,
  Ve as f,
  on as g,
  se as h,
  le as i,
  ir as j,
  Q as k,
  Ae as l,
  Kn as m,
  cr as n,
  lr as o,
  Rn as p,
  Kr as q,
  Hn as r,
  ze as s,
  On as t,
  ue as u,
  qn as v
};
