// @notificationapi/react/dist/assets/DefaultPropsProvider.js
import { g as ae, b as se, s as q, e as le, i as ue, c as ce, P as H } from "./createTheme.js";
import { w as fe, a as de, g as me, s as ee, b as pe, d as ye, i as he, T as ge, e as Se } from "./DefaultPropsProvider2.js";
import * as M from "react";
import "react/jsx-runtime";
function Y() {
  return Y = Object.assign ? Object.assign.bind() : function(e) {
    for (var n = 1; n < arguments.length; n++) {
      var r = arguments[n];
      for (var o in r) ({}).hasOwnProperty.call(r, o) && (e[o] = r[o]);
    }
    return e;
  }, Y.apply(null, arguments);
}
function Ee(e) {
  var n = /* @__PURE__ */ Object.create(null);
  return function(r) {
    return n[r] === void 0 && (n[r] = e(r)), n[r];
  };
}
var ve = /^((children|dangerouslySetInnerHTML|key|ref|autoFocus|defaultValue|defaultChecked|innerHTML|suppressContentEditableWarning|suppressHydrationWarning|valueLink|abbr|accept|acceptCharset|accessKey|action|allow|allowUserMedia|allowPaymentRequest|allowFullScreen|allowTransparency|alt|async|autoComplete|autoPlay|capture|cellPadding|cellSpacing|challenge|charSet|checked|cite|classID|className|cols|colSpan|content|contentEditable|contextMenu|controls|controlsList|coords|crossOrigin|data|dateTime|decoding|default|defer|dir|disabled|disablePictureInPicture|disableRemotePlayback|download|draggable|encType|enterKeyHint|fetchpriority|fetchPriority|form|formAction|formEncType|formMethod|formNoValidate|formTarget|frameBorder|headers|height|hidden|high|href|hrefLang|htmlFor|httpEquiv|id|inputMode|integrity|is|keyParams|keyType|kind|label|lang|list|loading|loop|low|marginHeight|marginWidth|max|maxLength|media|mediaGroup|method|min|minLength|multiple|muted|name|nonce|noValidate|open|optimum|pattern|placeholder|playsInline|poster|preload|profile|radioGroup|readOnly|referrerPolicy|rel|required|reversed|role|rows|rowSpan|sandbox|scope|scoped|scrolling|seamless|selected|shape|size|sizes|slot|span|spellCheck|src|srcDoc|srcLang|srcSet|start|step|style|summary|tabIndex|target|title|translate|type|useMap|value|width|wmode|wrap|about|datatype|inlist|prefix|property|resource|typeof|vocab|autoCapitalize|autoCorrect|autoSave|color|incremental|fallback|inert|itemProp|itemScope|itemType|itemID|itemRef|on|option|results|security|unselectable|accentHeight|accumulate|additive|alignmentBaseline|allowReorder|alphabetic|amplitude|arabicForm|ascent|attributeName|attributeType|autoReverse|azimuth|baseFrequency|baselineShift|baseProfile|bbox|begin|bias|by|calcMode|capHeight|clip|clipPathUnits|clipPath|clipRule|colorInterpolation|colorInterpolationFilters|colorProfile|colorRendering|contentScriptType|contentStyleType|cursor|cx|cy|d|decelerate|descent|diffuseConstant|direction|display|divisor|dominantBaseline|dur|dx|dy|edgeMode|elevation|enableBackground|end|exponent|externalResourcesRequired|fill|fillOpacity|fillRule|filter|filterRes|filterUnits|floodColor|floodOpacity|focusable|fontFamily|fontSize|fontSizeAdjust|fontStretch|fontStyle|fontVariant|fontWeight|format|from|fr|fx|fy|g1|g2|glyphName|glyphOrientationHorizontal|glyphOrientationVertical|glyphRef|gradientTransform|gradientUnits|hanging|horizAdvX|horizOriginX|ideographic|imageRendering|in|in2|intercept|k|k1|k2|k3|k4|kernelMatrix|kernelUnitLength|kerning|keyPoints|keySplines|keyTimes|lengthAdjust|letterSpacing|lightingColor|limitingConeAngle|local|markerEnd|markerMid|markerStart|markerHeight|markerUnits|markerWidth|mask|maskContentUnits|maskUnits|mathematical|mode|numOctaves|offset|opacity|operator|order|orient|orientation|origin|overflow|overlinePosition|overlineThickness|panose1|paintOrder|pathLength|patternContentUnits|patternTransform|patternUnits|pointerEvents|points|pointsAtX|pointsAtY|pointsAtZ|preserveAlpha|preserveAspectRatio|primitiveUnits|r|radius|refX|refY|renderingIntent|repeatCount|repeatDur|requiredExtensions|requiredFeatures|restart|result|rotate|rx|ry|scale|seed|shapeRendering|slope|spacing|specularConstant|specularExponent|speed|spreadMethod|startOffset|stdDeviation|stemh|stemv|stitchTiles|stopColor|stopOpacity|strikethroughPosition|strikethroughThickness|string|stroke|strokeDasharray|strokeDashoffset|strokeLinecap|strokeLinejoin|strokeMiterlimit|strokeOpacity|strokeWidth|surfaceScale|systemLanguage|tableValues|targetX|targetY|textAnchor|textDecoration|textRendering|textLength|to|transform|u1|u2|underlinePosition|underlineThickness|unicode|unicodeBidi|unicodeRange|unitsPerEm|vAlphabetic|vHanging|vIdeographic|vMathematical|values|vectorEffect|version|vertAdvY|vertOriginX|vertOriginY|viewBox|viewTarget|visibility|widths|wordSpacing|writingMode|x|xHeight|x1|x2|xChannelSelector|xlinkActuate|xlinkArcrole|xlinkHref|xlinkRole|xlinkShow|xlinkTitle|xlinkType|xmlBase|xmlns|xmlnsXlink|xmlLang|xmlSpace|y|y1|y2|yChannelSelector|z|zoomAndPan|for|class|autofocus)|(([Dd][Aa][Tt][Aa]|[Aa][Rr][Ii][Aa]|x)-.*))$/, _e = /* @__PURE__ */ Ee(
  function(e) {
    return ve.test(e) || e.charCodeAt(0) === 111 && e.charCodeAt(1) === 110 && e.charCodeAt(2) < 91;
  }
  /* Z+1 */
), Te = !1, Pe = _e, Re = function(n) {
  return n !== "theme";
}, W = function(n) {
  return typeof n == "string" && // 96 is one less than the char code
  // for "a" so this is checking that
  // it's a lowercase character
  n.charCodeAt(0) > 96 ? Pe : Re;
}, X = function(n, r, o) {
  var i;
  if (r) {
    var a = r.shouldForwardProp;
    i = n.__emotion_forwardProp && a ? function(s) {
      return n.__emotion_forwardProp(s) && a(s);
    } : a;
  }
  return typeof i != "function" && o && (i = n.__emotion_forwardProp), i;
}, be = function(n) {
  var r = n.cache, o = n.serialized, i = n.isStringTag;
  return pe(r, o, i), ye(function() {
    return he(r, o, i);
  }), null;
}, Ce = function e(n, r) {
  var o = n.__emotion_real === n, i = o && n.__emotion_base || n, a, s;
  r !== void 0 && (a = r.label, s = r.target);
  var l = X(n, r, o), h = l || W(i), d = !h("as");
  return function() {
    var f = arguments, p = o && n.__emotion_styles !== void 0 ? n.__emotion_styles.slice(0) : [];
    if (a !== void 0 && p.push("label:" + a + ";"), f[0] == null || f[0].raw === void 0)
      p.push.apply(p, f);
    else {
      var v = f[0];
      p.push(v[0]);
      for (var C = f.length, y = 1; y < C; y++)
        p.push(f[y], v[y]);
    }
    var t = fe(function(m, E, w) {
      var O = d && m.as || i, A = "", S = [], g = m;
      if (m.theme == null) {
        g = {};
        for (var P in m)
          g[P] = m[P];
        g.theme = M.useContext(de);
      }
      typeof m.className == "string" ? A = me(E.registered, S, m.className) : m.className != null && (A = m.className + " ");
      var _ = ee(p.concat(S), E.registered, g);
      A += E.key + "-" + _.name, s !== void 0 && (A += " " + s);
      var I = d && l === void 0 ? W(O) : h, x = {};
      for (var R in m)
        d && R === "as" || I(R) && (x[R] = m[R]);
      return x.className = A, w && (x.ref = w), /* @__PURE__ */ M.createElement(M.Fragment, null, /* @__PURE__ */ M.createElement(be, {
        cache: E,
        serialized: _,
        isStringTag: typeof O == "string"
      }), /* @__PURE__ */ M.createElement(O, x));
    });
    return t.displayName = a !== void 0 ? a : "Styled(" + (typeof i == "string" ? i : i.displayName || i.name || "Component") + ")", t.defaultProps = n.defaultProps, t.__emotion_real = t, t.__emotion_base = i, t.__emotion_styles = p, t.__emotion_forwardProp = l, Object.defineProperty(t, "toString", {
      value: function() {
        return s === void 0 && Te ? "NO_COMPONENT_SELECTOR" : "." + s;
      }
    }), t.withComponent = function(m, E) {
      var w = e(m, Y({}, r, E, {
        shouldForwardProp: X(t, E, !0)
      }));
      return w.apply(void 0, p);
    }, t;
  };
}, we = [
  "a",
  "abbr",
  "address",
  "area",
  "article",
  "aside",
  "audio",
  "b",
  "base",
  "bdi",
  "bdo",
  "big",
  "blockquote",
  "body",
  "br",
  "button",
  "canvas",
  "caption",
  "cite",
  "code",
  "col",
  "colgroup",
  "data",
  "datalist",
  "dd",
  "del",
  "details",
  "dfn",
  "dialog",
  "div",
  "dl",
  "dt",
  "em",
  "embed",
  "fieldset",
  "figcaption",
  "figure",
  "footer",
  "form",
  "h1",
  "h2",
  "h3",
  "h4",
  "h5",
  "h6",
  "head",
  "header",
  "hgroup",
  "hr",
  "html",
  "i",
  "iframe",
  "img",
  "input",
  "ins",
  "kbd",
  "keygen",
  "label",
  "legend",
  "li",
  "link",
  "main",
  "map",
  "mark",
  "marquee",
  "menu",
  "menuitem",
  "meta",
  "meter",
  "nav",
  "noscript",
  "object",
  "ol",
  "optgroup",
  "option",
  "output",
  "p",
  "param",
  "picture",
  "pre",
  "progress",
  "q",
  "rp",
  "rt",
  "ruby",
  "s",
  "samp",
  "script",
  "section",
  "select",
  "small",
  "source",
  "span",
  "strong",
  "style",
  "sub",
  "summary",
  "sup",
  "table",
  "tbody",
  "td",
  "textarea",
  "tfoot",
  "th",
  "thead",
  "time",
  "title",
  "tr",
  "track",
  "u",
  "ul",
  "var",
  "video",
  "wbr",
  // SVG
  "circle",
  "clipPath",
  "defs",
  "ellipse",
  "foreignObject",
  "g",
  "image",
  "line",
  "linearGradient",
  "mask",
  "path",
  "pattern",
  "polygon",
  "polyline",
  "radialGradient",
  "rect",
  "stop",
  "svg",
  "text",
  "tspan"
], D = Ce.bind(null);
we.forEach(function(e) {
  D[e] = D(e);
});
/**
 * @mui/styled-engine v6.3.1
 *
 * @license MIT
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
function Ae(e, n) {
  const r = D(e, n);
  return process.env.NODE_ENV !== "production" ? (...o) => {
    const i = typeof e == "string" ? `"${e}"` : "component";
    return o.length === 0 ? console.error([`MUI: Seems like you called \`styled(${i})()\` without a \`style\` argument.`, 'You must provide a `styles` argument: `styled("div")(styleYouForgotToPass)`.'].join(`
`)) : o.some((a) => a === void 0) && console.error(`MUI: the styled(${i})(...args) API requires all its args to be defined.`), r(...o);
  } : r;
}
function xe(e, n) {
  Array.isArray(e.__emotion_styles) && (e.__emotion_styles = n(e.__emotion_styles));
}
const B = [];
function G(e) {
  return B[0] = e, ee(B);
}
function te(e) {
  var n, r, o = "";
  if (typeof e == "string" || typeof e == "number") o += e;
  else if (typeof e == "object") if (Array.isArray(e)) {
    var i = e.length;
    for (n = 0; n < i; n++) e[n] && (r = te(e[n])) && (o && (o += " "), o += r);
  } else for (r in e) e[r] && (o && (o += " "), o += r);
  return o;
}
function Ge() {
  for (var e, n, r = 0, o = "", i = arguments.length; r < i; r++) (e = arguments[r]) && (n = te(e)) && (o && (o += " "), o += n);
  return o;
}
function Ke(e, n, r = "Mui") {
  const o = {};
  return n.forEach((i) => {
    o[i] = ae(e, i, r);
  }), o;
}
var z = { exports: {} }, u = {};
/**
 * @license React
 * react-is.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var K;
function ke() {
  if (K) return u;
  K = 1;
  var e = Symbol.for("react.transitional.element"), n = Symbol.for("react.portal"), r = Symbol.for("react.fragment"), o = Symbol.for("react.strict_mode"), i = Symbol.for("react.profiler"), a = Symbol.for("react.consumer"), s = Symbol.for("react.context"), l = Symbol.for("react.forward_ref"), h = Symbol.for("react.suspense"), d = Symbol.for("react.suspense_list"), f = Symbol.for("react.memo"), p = Symbol.for("react.lazy"), v = Symbol.for("react.offscreen"), C = Symbol.for("react.client.reference");
  function y(t) {
    if (typeof t == "object" && t !== null) {
      var m = t.$$typeof;
      switch (m) {
        case e:
          switch (t = t.type, t) {
            case r:
            case i:
            case o:
            case h:
            case d:
              return t;
            default:
              switch (t = t && t.$$typeof, t) {
                case s:
                case l:
                case p:
                case f:
                  return t;
                case a:
                  return t;
                default:
                  return m;
              }
          }
        case n:
          return m;
      }
    }
  }
  return u.ContextConsumer = a, u.ContextProvider = s, u.Element = e, u.ForwardRef = l, u.Fragment = r, u.Lazy = p, u.Memo = f, u.Portal = n, u.Profiler = i, u.StrictMode = o, u.Suspense = h, u.SuspenseList = d, u.isContextConsumer = function(t) {
    return y(t) === a;
  }, u.isContextProvider = function(t) {
    return y(t) === s;
  }, u.isElement = function(t) {
    return typeof t == "object" && t !== null && t.$$typeof === e;
  }, u.isForwardRef = function(t) {
    return y(t) === l;
  }, u.isFragment = function(t) {
    return y(t) === r;
  }, u.isLazy = function(t) {
    return y(t) === p;
  }, u.isMemo = function(t) {
    return y(t) === f;
  }, u.isPortal = function(t) {
    return y(t) === n;
  }, u.isProfiler = function(t) {
    return y(t) === i;
  }, u.isStrictMode = function(t) {
    return y(t) === o;
  }, u.isSuspense = function(t) {
    return y(t) === h;
  }, u.isSuspenseList = function(t) {
    return y(t) === d;
  }, u.isValidElementType = function(t) {
    return typeof t == "string" || typeof t == "function" || t === r || t === i || t === o || t === h || t === d || t === v || typeof t == "object" && t !== null && (t.$$typeof === p || t.$$typeof === f || t.$$typeof === s || t.$$typeof === a || t.$$typeof === l || t.$$typeof === C || t.getModuleId !== void 0);
  }, u.typeOf = y, u;
}
var c = {};
/**
 * @license React
 * react-is.development.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Z;
function Oe() {
  return Z || (Z = 1, process.env.NODE_ENV !== "production" && function() {
    function e(t) {
      if (typeof t == "object" && t !== null) {
        var m = t.$$typeof;
        switch (m) {
          case n:
            switch (t = t.type, t) {
              case o:
              case a:
              case i:
              case d:
              case f:
                return t;
              default:
                switch (t = t && t.$$typeof, t) {
                  case l:
                  case h:
                  case v:
                  case p:
                    return t;
                  case s:
                    return t;
                  default:
                    return m;
                }
            }
          case r:
            return m;
        }
      }
    }
    var n = Symbol.for("react.transitional.element"), r = Symbol.for("react.portal"), o = Symbol.for("react.fragment"), i = Symbol.for("react.strict_mode"), a = Symbol.for("react.profiler"), s = Symbol.for("react.consumer"), l = Symbol.for("react.context"), h = Symbol.for("react.forward_ref"), d = Symbol.for("react.suspense"), f = Symbol.for("react.suspense_list"), p = Symbol.for("react.memo"), v = Symbol.for("react.lazy"), C = Symbol.for("react.offscreen"), y = Symbol.for("react.client.reference");
    c.ContextConsumer = s, c.ContextProvider = l, c.Element = n, c.ForwardRef = h, c.Fragment = o, c.Lazy = v, c.Memo = p, c.Portal = r, c.Profiler = a, c.StrictMode = i, c.Suspense = d, c.SuspenseList = f, c.isContextConsumer = function(t) {
      return e(t) === s;
    }, c.isContextProvider = function(t) {
      return e(t) === l;
    }, c.isElement = function(t) {
      return typeof t == "object" && t !== null && t.$$typeof === n;
    }, c.isForwardRef = function(t) {
      return e(t) === h;
    }, c.isFragment = function(t) {
      return e(t) === o;
    }, c.isLazy = function(t) {
      return e(t) === v;
    }, c.isMemo = function(t) {
      return e(t) === p;
    }, c.isPortal = function(t) {
      return e(t) === r;
    }, c.isProfiler = function(t) {
      return e(t) === a;
    }, c.isStrictMode = function(t) {
      return e(t) === i;
    }, c.isSuspense = function(t) {
      return e(t) === d;
    }, c.isSuspenseList = function(t) {
      return e(t) === f;
    }, c.isValidElementType = function(t) {
      return typeof t == "string" || typeof t == "function" || t === o || t === a || t === i || t === d || t === f || t === C || typeof t == "object" && t !== null && (t.$$typeof === v || t.$$typeof === p || t.$$typeof === l || t.$$typeof === s || t.$$typeof === h || t.$$typeof === y || t.getModuleId !== void 0);
    }, c.typeOf = e;
  }()), c;
}
process.env.NODE_ENV === "production" ? z.exports = ke() : z.exports = Oe();
var J = z.exports;
function re(e, n = "") {
  return e.displayName || e.name || n;
}
function Q(e, n, r) {
  const o = re(n);
  return e.displayName || (o !== "" ? `${r}(${o})` : r);
}
function Fe(e) {
  if (e != null) {
    if (typeof e == "string")
      return e;
    if (typeof e == "function")
      return re(e, "Component");
    if (typeof e == "object")
      switch (e.$$typeof) {
        case J.ForwardRef:
          return Q(e, e.render, "ForwardRef");
        case J.Memo:
          return Q(e, e.type, "memo");
        default:
          return;
      }
  }
}
function ne(e) {
  const {
    variants: n,
    ...r
  } = e, o = {
    variants: n,
    style: G(r),
    isProcessed: !0
  };
  return o.style === r || n && n.forEach((i) => {
    typeof i.style != "function" && (i.style = G(i.style));
  }), o;
}
const Ne = se();
function L(e) {
  return e !== "ownerState" && e !== "theme" && e !== "sx" && e !== "as";
}
function Me(e) {
  return e ? (n, r) => r[e] : null;
}
function $e(e, n, r) {
  e.theme = De(e.theme) ? r : e.theme[n] || e.theme;
}
function $(e, n) {
  const r = typeof n == "function" ? n(e) : n;
  if (Array.isArray(r))
    return r.flatMap((o) => $(e, o));
  if (Array.isArray(r == null ? void 0 : r.variants)) {
    let o;
    if (r.isProcessed)
      o = r.style;
    else {
      const {
        variants: i,
        ...a
      } = r;
      o = a;
    }
    return oe(e, r.variants, [o]);
  }
  return r != null && r.isProcessed ? r.style : r;
}
function oe(e, n, r = []) {
  var i;
  let o;
  e: for (let a = 0; a < n.length; a += 1) {
    const s = n[a];
    if (typeof s.props == "function") {
      if (o ?? (o = {
        ...e,
        ...e.ownerState,
        ownerState: e.ownerState
      }), !s.props(o))
        continue;
    } else
      for (const l in s.props)
        if (e[l] !== s.props[l] && ((i = e.ownerState) == null ? void 0 : i[l]) !== s.props[l])
          continue e;
    typeof s.style == "function" ? (o ?? (o = {
      ...e,
      ...e.ownerState,
      ownerState: e.ownerState
    }), r.push(s.style(o))) : r.push(s.style);
  }
  return r;
}
function Ie(e = {}) {
  const {
    themeId: n,
    defaultTheme: r = Ne,
    rootShouldForwardProp: o = L,
    slotShouldForwardProp: i = L
  } = e;
  function a(l) {
    $e(l, n, r);
  }
  return (l, h = {}) => {
    xe(l, (S) => S.filter((g) => g !== q));
    const {
      name: d,
      slot: f,
      skipVariantsResolver: p,
      skipSx: v,
      // TODO v6: remove `lowercaseFirstLetter()` in the next major release
      // For more details: https://github.com/mui/material-ui/pull/37908
      overridesResolver: C = Me(ie(f)),
      ...y
    } = h, t = p !== void 0 ? p : (
      // TODO v6: remove `Root` in the next major release
      // For more details: https://github.com/mui/material-ui/pull/37908
      f && f !== "Root" && f !== "root" || !1
    ), m = v || !1;
    let E = L;
    f === "Root" || f === "root" ? E = o : f ? E = i : ze(l) && (E = void 0);
    const w = Ae(l, {
      shouldForwardProp: E,
      label: Ye(d, f),
      ...y
    }), O = (S) => {
      if (typeof S == "function" && S.__emotion_real !== S)
        return function(P) {
          return $(P, S);
        };
      if (ue(S)) {
        const g = ne(S);
        return g.variants ? function(_) {
          return $(_, g);
        } : g.style;
      }
      return S;
    }, A = (...S) => {
      const g = [], P = S.map(O), _ = [];
      if (g.push(a), d && C && _.push(function(b) {
        var N, U;
        const T = (U = (N = b.theme.components) == null ? void 0 : N[d]) == null ? void 0 : U.styleOverrides;
        if (!T)
          return null;
        const F = {};
        for (const V in T)
          F[V] = $(b, T[V]);
        return C(b, F);
      }), d && !t && _.push(function(b) {
        var F, N;
        const k = b.theme, T = (N = (F = k == null ? void 0 : k.components) == null ? void 0 : F[d]) == null ? void 0 : N.variants;
        return T ? oe(b, T) : null;
      }), m || _.push(q), Array.isArray(P[0])) {
        const R = P.shift(), b = new Array(g.length).fill(""), k = new Array(_.length).fill("");
        let T;
        T = [...b, ...R, ...k], T.raw = [...b, ...R.raw, ...k], g.unshift(T);
      }
      const I = [...g, ...P, ..._], x = w(...I);
      return l.muiName && (x.muiName = l.muiName), process.env.NODE_ENV !== "production" && (x.displayName = Le(d, f, l)), x;
    };
    return w.withConfig && (A.withConfig = w.withConfig), A;
  };
}
function Le(e, n, r) {
  return e ? `${e}${le(n || "")}` : `Styled(${Fe(r)})`;
}
function Ye(e, n) {
  let r;
  return process.env.NODE_ENV !== "production" && e && (r = `${e}-${ie(n || "Root")}`), r;
}
function De(e) {
  for (const n in e)
    return !1;
  return !0;
}
function ze(e) {
  return typeof e == "string" && // 96 is one less than the char code
  // for "a" so this is checking that
  // it's a lowercase character
  e.charCodeAt(0) > 96;
}
function ie(e) {
  return e && e.charAt(0).toLowerCase() + e.slice(1);
}
function Ze(e, n, r = void 0) {
  const o = {};
  for (const i in e) {
    const a = e[i];
    let s = "", l = !0;
    for (let h = 0; h < a.length; h += 1) {
      const d = a[h];
      d && (s += (l === !0 ? "" : " ") + n(d), l = !1, r && r[d] && (s += " " + r[d]));
    }
    o[i] = s;
  }
  return o;
}
const j = {
  theme: void 0
};
function Ue(e) {
  let n, r;
  return function(i) {
    let a = n;
    return (a === void 0 || i.theme !== r) && (j.theme = i.theme, a = ne(e(j)), n = a, r = i.theme), a;
  };
}
const Ve = ce();
function qe(e) {
  return e !== "ownerState" && e !== "theme" && e !== "sx" && e !== "as";
}
const He = (e) => qe(e) && e !== "classes", Je = Ie({
  themeId: ge,
  defaultTheme: Ve,
  rootShouldForwardProp: He
}), Qe = Ue;
process.env.NODE_ENV !== "production" && (H.node, H.object.isRequired);
function je(e) {
  return Se(e);
}
export {
  Y as _,
  Ze as a,
  Ae as b,
  Ge as c,
  Ve as d,
  Ie as e,
  qe as f,
  Ke as g,
  Qe as m,
  D as n,
  He as r,
  Je as s,
  je as u
};
