// @notificationapi/react/dist/assets/DefaultPropsProvider2.js
import * as $ from "react";
import { forwardRef as $e, useContext as ke } from "react";
import { P as se } from "./createTheme.js";
import { jsx as Pe } from "react/jsx-runtime";
const Ar = "$$material";
var Ae = !1;
function Oe(e) {
  if (e.sheet)
    return e.sheet;
  for (var r = 0; r < document.styleSheets.length; r++)
    if (document.styleSheets[r].ownerNode === e)
      return document.styleSheets[r];
}
function Te(e) {
  var r = document.createElement("style");
  return r.setAttribute("data-emotion", e.key), e.nonce !== void 0 && r.setAttribute("nonce", e.nonce), r.appendChild(document.createTextNode("")), r.setAttribute("data-s", ""), r;
}
var Ne = /* @__PURE__ */ function() {
  function e(t) {
    var n = this;
    this._insertTag = function(a) {
      var s;
      n.tags.length === 0 ? n.insertionPoint ? s = n.insertionPoint.nextSibling : n.prepend ? s = n.container.firstChild : s = n.before : s = n.tags[n.tags.length - 1].nextSibling, n.container.insertBefore(a, s), n.tags.push(a);
    }, this.isSpeedy = t.speedy === void 0 ? !Ae : t.speedy, this.tags = [], this.ctr = 0, this.nonce = t.nonce, this.key = t.key, this.container = t.container, this.prepend = t.prepend, this.insertionPoint = t.insertionPoint, this.before = null;
  }
  var r = e.prototype;
  return r.hydrate = function(n) {
    n.forEach(this._insertTag);
  }, r.insert = function(n) {
    this.ctr % (this.isSpeedy ? 65e3 : 1) === 0 && this._insertTag(Te(this));
    var a = this.tags[this.tags.length - 1];
    if (this.isSpeedy) {
      var s = Oe(a);
      try {
        s.insertRule(n, s.cssRules.length);
      } catch {
      }
    } else
      a.appendChild(document.createTextNode(n));
    this.ctr++;
  }, r.flush = function() {
    this.tags.forEach(function(n) {
      var a;
      return (a = n.parentNode) == null ? void 0 : a.removeChild(n);
    }), this.tags = [], this.ctr = 0;
  }, e;
}(), x = "-ms-", Y = "-moz-", f = "-webkit-", he = "comm", te = "rule", ne = "decl", Re = "@import", le = "@keyframes", Ie = "@layer", _e = Math.abs, q = String.fromCharCode, Me = Object.assign;
function De(e, r) {
  return y(e, 0) ^ 45 ? (((r << 2 ^ y(e, 0)) << 2 ^ y(e, 1)) << 2 ^ y(e, 2)) << 2 ^ y(e, 3) : 0;
}
function pe(e) {
  return e.trim();
}
function Le(e, r) {
  return (e = r.exec(e)) ? e[0] : e;
}
function u(e, r, t) {
  return e.replace(r, t);
}
function Q(e, r) {
  return e.indexOf(r);
}
function y(e, r) {
  return e.charCodeAt(r) | 0;
}
function W(e, r, t) {
  return e.slice(r, t);
}
function A(e) {
  return e.length;
}
function ae(e) {
  return e.length;
}
function V(e, r) {
  return r.push(e), e;
}
function We(e, r) {
  return e.map(r).join("");
}
var U = 1, M = 1, me = 0, E = 0, v = 0, D = "";
function Z(e, r, t, n, a, s, i) {
  return { value: e, root: r, parent: t, type: n, props: a, children: s, line: U, column: M, length: i, return: "" };
}
function L(e, r) {
  return Me(Z("", null, null, "", null, null, 0), e, { length: -e.length }, r);
}
function Fe() {
  return v;
}
function Ge() {
  return v = E > 0 ? y(D, --E) : 0, M--, v === 10 && (M = 1, U--), v;
}
function S() {
  return v = E < me ? y(D, E++) : 0, M++, v === 10 && (M = 1, U++), v;
}
function T() {
  return y(D, E);
}
function H() {
  return E;
}
function j(e, r) {
  return W(D, e, r);
}
function F(e) {
  switch (e) {
    // \0 \t \n \r \s whitespace token
    case 0:
    case 9:
    case 10:
    case 13:
    case 32:
      return 5;
    // ! + , / > @ ~ isolate token
    case 33:
    case 43:
    case 44:
    case 47:
    case 62:
    case 64:
    case 126:
    // ; { } breakpoint token
    case 59:
    case 123:
    case 125:
      return 4;
    // : accompanied token
    case 58:
      return 3;
    // " ' ( [ opening delimit token
    case 34:
    case 39:
    case 40:
    case 91:
      return 2;
    // ) ] closing delimit token
    case 41:
    case 93:
      return 1;
  }
  return 0;
}
function be(e) {
  return U = M = 1, me = A(D = e), E = 0, [];
}
function ve(e) {
  return D = "", e;
}
function K(e) {
  return pe(j(E - 1, X(e === 91 ? e + 2 : e === 40 ? e + 1 : e)));
}
function je(e) {
  for (; (v = T()) && v < 33; )
    S();
  return F(e) > 2 || F(v) > 3 ? "" : " ";
}
function ze(e, r) {
  for (; --r && S() && !(v < 48 || v > 102 || v > 57 && v < 65 || v > 70 && v < 97); )
    ;
  return j(e, H() + (r < 6 && T() == 32 && S() == 32));
}
function X(e) {
  for (; S(); )
    switch (v) {
      // ] ) " '
      case e:
        return E;
      // " '
      case 34:
      case 39:
        e !== 34 && e !== 39 && X(v);
        break;
      // (
      case 40:
        e === 41 && X(e);
        break;
      // \
      case 92:
        S();
        break;
    }
  return E;
}
function Ve(e, r) {
  for (; S() && e + v !== 57; )
    if (e + v === 84 && T() === 47)
      break;
  return "/*" + j(r, E - 1) + "*" + q(e === 47 ? e : S());
}
function He(e) {
  for (; !F(T()); )
    S();
  return j(e, E);
}
function Ke(e) {
  return ve(B("", null, null, null, [""], e = be(e), 0, [0], e));
}
function B(e, r, t, n, a, s, i, c, o) {
  for (var d = 0, l = 0, m = i, N = 0, R = 0, C = 0, p = 1, w = 1, b = 1, g = 0, k = "", z = a, I = s, P = n, h = k; w; )
    switch (C = g, g = S()) {
      // (
      case 40:
        if (C != 108 && y(h, m - 1) == 58) {
          Q(h += u(K(g), "&", "&\f"), "&\f") != -1 && (b = -1);
          break;
        }
      // " ' [
      case 34:
      case 39:
      case 91:
        h += K(g);
        break;
      // \t \n \r \s
      case 9:
      case 10:
      case 13:
      case 32:
        h += je(C);
        break;
      // \
      case 92:
        h += ze(H() - 1, 7);
        continue;
      // /
      case 47:
        switch (T()) {
          case 42:
          case 47:
            V(Be(Ve(S(), H()), r, t), o);
            break;
          default:
            h += "/";
        }
        break;
      // {
      case 123 * p:
        c[d++] = A(h) * b;
      // } ; \0
      case 125 * p:
      case 59:
      case 0:
        switch (g) {
          // \0 }
          case 0:
          case 125:
            w = 0;
          // ;
          case 59 + l:
            b == -1 && (h = u(h, /\f/g, "")), R > 0 && A(h) - m && V(R > 32 ? ce(h + ";", n, t, m - 1) : ce(u(h, " ", "") + ";", n, t, m - 2), o);
            break;
          // @ ;
          case 59:
            h += ";";
          // { rule/at-rule
          default:
            if (V(P = ie(h, r, t, d, l, a, c, k, z = [], I = [], m), s), g === 123)
              if (l === 0)
                B(h, r, P, P, z, s, m, c, I);
              else
                switch (N === 99 && y(h, 3) === 110 ? 100 : N) {
                  // d l m s
                  case 100:
                  case 108:
                  case 109:
                  case 115:
                    B(e, P, P, n && V(ie(e, P, P, 0, 0, a, c, k, a, z = [], m), I), a, I, m, c, n ? z : I);
                    break;
                  default:
                    B(h, P, P, P, [""], I, 0, c, I);
                }
        }
        d = l = R = 0, p = b = 1, k = h = "", m = i;
        break;
      // :
      case 58:
        m = 1 + A(h), R = C;
      default:
        if (p < 1) {
          if (g == 123)
            --p;
          else if (g == 125 && p++ == 0 && Ge() == 125)
            continue;
        }
        switch (h += q(g), g * p) {
          // &
          case 38:
            b = l > 0 ? 1 : (h += "\f", -1);
            break;
          // ,
          case 44:
            c[d++] = (A(h) - 1) * b, b = 1;
            break;
          // @
          case 64:
            T() === 45 && (h += K(S())), N = T(), l = m = A(k = h += He(H())), g++;
            break;
          // -
          case 45:
            C === 45 && A(h) == 2 && (p = 0);
        }
    }
  return s;
}
function ie(e, r, t, n, a, s, i, c, o, d, l) {
  for (var m = a - 1, N = a === 0 ? s : [""], R = ae(N), C = 0, p = 0, w = 0; C < n; ++C)
    for (var b = 0, g = W(e, m + 1, m = _e(p = i[C])), k = e; b < R; ++b)
      (k = pe(p > 0 ? N[b] + " " + g : u(g, /&\f/g, N[b]))) && (o[w++] = k);
  return Z(e, r, t, a === 0 ? te : c, o, d, l);
}
function Be(e, r, t) {
  return Z(e, r, t, he, q(Fe()), W(e, 2, -2), 0);
}
function ce(e, r, t, n) {
  return Z(e, r, t, ne, W(e, 0, n), W(e, n + 1, -1), n);
}
function _(e, r) {
  for (var t = "", n = ae(e), a = 0; a < n; a++)
    t += r(e[a], a, e, r) || "";
  return t;
}
function Ye(e, r, t, n) {
  switch (e.type) {
    case Ie:
      if (e.children.length) break;
    case Re:
    case ne:
      return e.return = e.return || e.value;
    case he:
      return "";
    case le:
      return e.return = e.value + "{" + _(e.children, n) + "}";
    case te:
      e.value = e.props.join(",");
  }
  return A(t = _(e.children, n)) ? e.return = e.value + "{" + t + "}" : "";
}
function qe(e) {
  var r = ae(e);
  return function(t, n, a, s) {
    for (var i = "", c = 0; c < r; c++)
      i += e[c](t, n, a, s) || "";
    return i;
  };
}
function Ue(e) {
  return function(r) {
    r.root || (r = r.return) && e(r);
  };
}
var Ze = function(r, t, n) {
  for (var a = 0, s = 0; a = s, s = T(), a === 38 && s === 12 && (t[n] = 1), !F(s); )
    S();
  return j(r, E);
}, Je = function(r, t) {
  var n = -1, a = 44;
  do
    switch (F(a)) {
      case 0:
        a === 38 && T() === 12 && (t[n] = 1), r[n] += Ze(E - 1, t, n);
        break;
      case 2:
        r[n] += K(a);
        break;
      case 4:
        if (a === 44) {
          r[++n] = T() === 58 ? "&\f" : "", t[n] = r[n].length;
          break;
        }
      // fallthrough
      default:
        r[n] += q(a);
    }
  while (a = S());
  return r;
}, Qe = function(r, t) {
  return ve(Je(be(r), t));
}, oe = /* @__PURE__ */ new WeakMap(), Xe = function(r) {
  if (!(r.type !== "rule" || !r.parent || // positive .length indicates that this rule contains pseudo
  // negative .length indicates that this rule has been already prefixed
  r.length < 1)) {
    for (var t = r.value, n = r.parent, a = r.column === n.column && r.line === n.line; n.type !== "rule"; )
      if (n = n.parent, !n) return;
    if (!(r.props.length === 1 && t.charCodeAt(0) !== 58 && !oe.get(n)) && !a) {
      oe.set(r, !0);
      for (var s = [], i = Qe(t, s), c = n.props, o = 0, d = 0; o < i.length; o++)
        for (var l = 0; l < c.length; l++, d++)
          r.props[d] = s[o] ? i[o].replace(/&\f/g, c[l]) : c[l] + " " + i[o];
    }
  }
}, er = function(r) {
  if (r.type === "decl") {
    var t = r.value;
    // charcode for l
    t.charCodeAt(0) === 108 && // charcode for b
    t.charCodeAt(2) === 98 && (r.return = "", r.value = "");
  }
};
function ye(e, r) {
  switch (De(e, r)) {
    // color-adjust
    case 5103:
      return f + "print-" + e + e;
    // animation, animation-(delay|direction|duration|fill-mode|iteration-count|name|play-state|timing-function)
    case 5737:
    case 4201:
    case 3177:
    case 3433:
    case 1641:
    case 4457:
    case 2921:
    // text-decoration, filter, clip-path, backface-visibility, column, box-decoration-break
    case 5572:
    case 6356:
    case 5844:
    case 3191:
    case 6645:
    case 3005:
    // mask, mask-image, mask-(mode|clip|size), mask-(repeat|origin), mask-position, mask-composite,
    case 6391:
    case 5879:
    case 5623:
    case 6135:
    case 4599:
    case 4855:
    // background-clip, columns, column-(count|fill|gap|rule|rule-color|rule-style|rule-width|span|width)
    case 4215:
    case 6389:
    case 5109:
    case 5365:
    case 5621:
    case 3829:
      return f + e + e;
    // appearance, user-select, transform, hyphens, text-size-adjust
    case 5349:
    case 4246:
    case 4810:
    case 6968:
    case 2756:
      return f + e + Y + e + x + e + e;
    // flex, flex-direction
    case 6828:
    case 4268:
      return f + e + x + e + e;
    // order
    case 6165:
      return f + e + x + "flex-" + e + e;
    // align-items
    case 5187:
      return f + e + u(e, /(\w+).+(:[^]+)/, f + "box-$1$2" + x + "flex-$1$2") + e;
    // align-self
    case 5443:
      return f + e + x + "flex-item-" + u(e, /flex-|-self/, "") + e;
    // align-content
    case 4675:
      return f + e + x + "flex-line-pack" + u(e, /align-content|flex-|-self/, "") + e;
    // flex-shrink
    case 5548:
      return f + e + x + u(e, "shrink", "negative") + e;
    // flex-basis
    case 5292:
      return f + e + x + u(e, "basis", "preferred-size") + e;
    // flex-grow
    case 6060:
      return f + "box-" + u(e, "-grow", "") + f + e + x + u(e, "grow", "positive") + e;
    // transition
    case 4554:
      return f + u(e, /([^-])(transform)/g, "$1" + f + "$2") + e;
    // cursor
    case 6187:
      return u(u(u(e, /(zoom-|grab)/, f + "$1"), /(image-set)/, f + "$1"), e, "") + e;
    // background, background-image
    case 5495:
    case 3959:
      return u(e, /(image-set\([^]*)/, f + "$1$`$1");
    // justify-content
    case 4968:
      return u(u(e, /(.+:)(flex-)?(.*)/, f + "box-pack:$3" + x + "flex-pack:$3"), /s.+-b[^;]+/, "justify") + f + e + e;
    // (margin|padding)-inline-(start|end)
    case 4095:
    case 3583:
    case 4068:
    case 2532:
      return u(e, /(.+)-inline(.+)/, f + "$1$2") + e;
    // (min|max)?(width|height|inline-size|block-size)
    case 8116:
    case 7059:
    case 5753:
    case 5535:
    case 5445:
    case 5701:
    case 4933:
    case 4677:
    case 5533:
    case 5789:
    case 5021:
    case 4765:
      if (A(e) - 1 - r > 6) switch (y(e, r + 1)) {
        // (m)ax-content, (m)in-content
        case 109:
          if (y(e, r + 4) !== 45) break;
        // (f)ill-available, (f)it-content
        case 102:
          return u(e, /(.+:)(.+)-([^]+)/, "$1" + f + "$2-$3$1" + Y + (y(e, r + 3) == 108 ? "$3" : "$2-$3")) + e;
        // (s)tretch
        case 115:
          return ~Q(e, "stretch") ? ye(u(e, "stretch", "fill-available"), r) + e : e;
      }
      break;
    // position: sticky
    case 4949:
      if (y(e, r + 1) !== 115) break;
    // display: (flex|inline-flex)
    case 6444:
      switch (y(e, A(e) - 3 - (~Q(e, "!important") && 10))) {
        // stic(k)y
        case 107:
          return u(e, ":", ":" + f) + e;
        // (inline-)?fl(e)x
        case 101:
          return u(e, /(.+:)([^;!]+)(;|!.+)?/, "$1" + f + (y(e, 14) === 45 ? "inline-" : "") + "box$3$1" + f + "$2$3$1" + x + "$2box$3") + e;
      }
      break;
    // writing-mode
    case 5936:
      switch (y(e, r + 11)) {
        // vertical-l(r)
        case 114:
          return f + e + x + u(e, /[svh]\w+-[tblr]{2}/, "tb") + e;
        // vertical-r(l)
        case 108:
          return f + e + x + u(e, /[svh]\w+-[tblr]{2}/, "tb-rl") + e;
        // horizontal(-)tb
        case 45:
          return f + e + x + u(e, /[svh]\w+-[tblr]{2}/, "lr") + e;
      }
      return f + e + x + e + e;
  }
  return e;
}
var rr = function(r, t, n, a) {
  if (r.length > -1 && !r.return) switch (r.type) {
    case ne:
      r.return = ye(r.value, r.length);
      break;
    case le:
      return _([L(r, {
        value: u(r.value, "@", "@" + f)
      })], a);
    case te:
      if (r.length) return We(r.props, function(s) {
        switch (Le(s, /(::plac\w+|:read-\w+)/)) {
          // :read-(only|write)
          case ":read-only":
          case ":read-write":
            return _([L(r, {
              props: [u(s, /:(read-\w+)/, ":" + Y + "$1")]
            })], a);
          // :placeholder
          case "::placeholder":
            return _([L(r, {
              props: [u(s, /:(plac\w+)/, ":" + f + "input-$1")]
            }), L(r, {
              props: [u(s, /:(plac\w+)/, ":" + Y + "$1")]
            }), L(r, {
              props: [u(s, /:(plac\w+)/, x + "input-$1")]
            })], a);
        }
        return "";
      });
  }
}, tr = [rr], nr = function(r) {
  var t = r.key;
  if (t === "css") {
    var n = document.querySelectorAll("style[data-emotion]:not([data-s])");
    Array.prototype.forEach.call(n, function(p) {
      var w = p.getAttribute("data-emotion");
      w.indexOf(" ") !== -1 && (document.head.appendChild(p), p.setAttribute("data-s", ""));
    });
  }
  var a = r.stylisPlugins || tr, s = {}, i, c = [];
  i = r.container || document.head, Array.prototype.forEach.call(
    // this means we will ignore elements which don't have a space in them which
    // means that the style elements we're looking at are only Emotion 11 server-rendered style elements
    document.querySelectorAll('style[data-emotion^="' + t + ' "]'),
    function(p) {
      for (var w = p.getAttribute("data-emotion").split(" "), b = 1; b < w.length; b++)
        s[w[b]] = !0;
      c.push(p);
    }
  );
  var o, d = [Xe, er];
  {
    var l, m = [Ye, Ue(function(p) {
      l.insert(p);
    })], N = qe(d.concat(a, m)), R = function(w) {
      return _(Ke(w), N);
    };
    o = function(w, b, g, k) {
      l = g, R(w ? w + "{" + b.styles + "}" : b.styles), k && (C.inserted[b.name] = !0);
    };
  }
  var C = {
    key: t,
    sheet: new Ne({
      key: t,
      container: i,
      nonce: r.nonce,
      speedy: r.speedy,
      prepend: r.prepend,
      insertionPoint: r.insertionPoint
    }),
    nonce: r.nonce,
    inserted: s,
    registered: {},
    insert: o
  };
  return C.sheet.hydrate(c), C;
}, ar = !0;
function sr(e, r, t) {
  var n = "";
  return t.split(" ").forEach(function(a) {
    e[a] !== void 0 ? r.push(e[a] + ";") : a && (n += a + " ");
  }), n;
}
var ge = function(r, t, n) {
  var a = r.key + "-" + t.name;
  // we only need to add the styles to the registered cache if the
  // class name could be used further down
  // the tree but if it's a string tag, we know it won't
  // so we don't have to add it to registered cache.
  // this improves memory usage since we can avoid storing the whole style string
  (n === !1 || // we need to always store it if we're in compat mode and
  // in node since emotion-server relies on whether a style is in
  // the registered cache to know whether a style is global or not
  // also, note that this check will be dead code eliminated in the browser
  ar === !1) && r.registered[a] === void 0 && (r.registered[a] = t.styles);
}, ir = function(r, t, n) {
  ge(r, t, n);
  var a = r.key + "-" + t.name;
  if (r.inserted[t.name] === void 0) {
    var s = t;
    do
      r.insert(t === s ? "." + a : "", s, r.sheet, !0), s = s.next;
    while (s !== void 0);
  }
};
function cr(e) {
  for (var r = 0, t, n = 0, a = e.length; a >= 4; ++n, a -= 4)
    t = e.charCodeAt(n) & 255 | (e.charCodeAt(++n) & 255) << 8 | (e.charCodeAt(++n) & 255) << 16 | (e.charCodeAt(++n) & 255) << 24, t = /* Math.imul(k, m): */
    (t & 65535) * 1540483477 + ((t >>> 16) * 59797 << 16), t ^= /* k >>> r: */
    t >>> 24, r = /* Math.imul(k, m): */
    (t & 65535) * 1540483477 + ((t >>> 16) * 59797 << 16) ^ /* Math.imul(h, m): */
    (r & 65535) * 1540483477 + ((r >>> 16) * 59797 << 16);
  switch (a) {
    case 3:
      r ^= (e.charCodeAt(n + 2) & 255) << 16;
    case 2:
      r ^= (e.charCodeAt(n + 1) & 255) << 8;
    case 1:
      r ^= e.charCodeAt(n) & 255, r = /* Math.imul(h, m): */
      (r & 65535) * 1540483477 + ((r >>> 16) * 59797 << 16);
  }
  return r ^= r >>> 13, r = /* Math.imul(h, m): */
  (r & 65535) * 1540483477 + ((r >>> 16) * 59797 << 16), ((r ^ r >>> 15) >>> 0).toString(36);
}
var or = {
  animationIterationCount: 1,
  aspectRatio: 1,
  borderImageOutset: 1,
  borderImageSlice: 1,
  borderImageWidth: 1,
  boxFlex: 1,
  boxFlexGroup: 1,
  boxOrdinalGroup: 1,
  columnCount: 1,
  columns: 1,
  flex: 1,
  flexGrow: 1,
  flexPositive: 1,
  flexShrink: 1,
  flexNegative: 1,
  flexOrder: 1,
  gridRow: 1,
  gridRowEnd: 1,
  gridRowSpan: 1,
  gridRowStart: 1,
  gridColumn: 1,
  gridColumnEnd: 1,
  gridColumnSpan: 1,
  gridColumnStart: 1,
  msGridRow: 1,
  msGridRowSpan: 1,
  msGridColumn: 1,
  msGridColumnSpan: 1,
  fontWeight: 1,
  lineHeight: 1,
  opacity: 1,
  order: 1,
  orphans: 1,
  scale: 1,
  tabSize: 1,
  widows: 1,
  zIndex: 1,
  zoom: 1,
  WebkitLineClamp: 1,
  // SVG-related properties
  fillOpacity: 1,
  floodOpacity: 1,
  stopOpacity: 1,
  strokeDasharray: 1,
  strokeDashoffset: 1,
  strokeMiterlimit: 1,
  strokeOpacity: 1,
  strokeWidth: 1
};
function fr(e) {
  var r = /* @__PURE__ */ Object.create(null);
  return function(t) {
    return r[t] === void 0 && (r[t] = e(t)), r[t];
  };
}
var ur = !1, dr = /[A-Z]|^ms/g, hr = /_EMO_([^_]+?)_([^]*?)_EMO_/g, xe = function(r) {
  return r.charCodeAt(1) === 45;
}, fe = function(r) {
  return r != null && typeof r != "boolean";
}, J = /* @__PURE__ */ fr(function(e) {
  return xe(e) ? e : e.replace(dr, "-$&").toLowerCase();
}), ue = function(r, t) {
  switch (r) {
    case "animation":
    case "animationName":
      if (typeof t == "string")
        return t.replace(hr, function(n, a, s) {
          return O = {
            name: a,
            styles: s,
            next: O
          }, a;
        });
  }
  return or[r] !== 1 && !xe(r) && typeof t == "number" && t !== 0 ? t + "px" : t;
}, lr = "Component selectors can only be used in conjunction with @emotion/babel-plugin, the swc Emotion plugin, or another Emotion-aware compiler transform.";
function G(e, r, t) {
  if (t == null)
    return "";
  var n = t;
  if (n.__emotion_styles !== void 0)
    return n;
  switch (typeof t) {
    case "boolean":
      return "";
    case "object": {
      var a = t;
      if (a.anim === 1)
        return O = {
          name: a.name,
          styles: a.styles,
          next: O
        }, a.name;
      var s = t;
      if (s.styles !== void 0) {
        var i = s.next;
        if (i !== void 0)
          for (; i !== void 0; )
            O = {
              name: i.name,
              styles: i.styles,
              next: O
            }, i = i.next;
        var c = s.styles + ";";
        return c;
      }
      return pr(e, r, t);
    }
    case "function": {
      if (e !== void 0) {
        var o = O, d = t(e);
        return O = o, G(e, r, d);
      }
      break;
    }
  }
  var l = t;
  if (r == null)
    return l;
  var m = r[l];
  return m !== void 0 ? m : l;
}
function pr(e, r, t) {
  var n = "";
  if (Array.isArray(t))
    for (var a = 0; a < t.length; a++)
      n += G(e, r, t[a]) + ";";
  else
    for (var s in t) {
      var i = t[s];
      if (typeof i != "object") {
        var c = i;
        r != null && r[c] !== void 0 ? n += s + "{" + r[c] + "}" : fe(c) && (n += J(s) + ":" + ue(s, c) + ";");
      } else {
        if (s === "NO_COMPONENT_SELECTOR" && ur)
          throw new Error(lr);
        if (Array.isArray(i) && typeof i[0] == "string" && (r == null || r[i[0]] === void 0))
          for (var o = 0; o < i.length; o++)
            fe(i[o]) && (n += J(s) + ":" + ue(s, i[o]) + ";");
        else {
          var d = G(e, r, i);
          switch (s) {
            case "animation":
            case "animationName": {
              n += J(s) + ":" + d + ";";
              break;
            }
            default:
              n += s + "{" + d + "}";
          }
        }
      }
    }
  return n;
}
var de = /label:\s*([^\s;{]+)\s*(;|$)/g, O;
function mr(e, r, t) {
  if (e.length === 1 && typeof e[0] == "object" && e[0] !== null && e[0].styles !== void 0)
    return e[0];
  var n = !0, a = "";
  O = void 0;
  var s = e[0];
  if (s == null || s.raw === void 0)
    n = !1, a += G(t, r, s);
  else {
    var i = s;
    a += i[0];
  }
  for (var c = 1; c < e.length; c++)
    if (a += G(t, r, e[c]), n) {
      var o = s;
      a += o[c];
    }
  de.lastIndex = 0;
  for (var d = "", l; (l = de.exec(a)) !== null; )
    d += "-" + l[1];
  var m = cr(a) + d;
  return {
    name: m,
    styles: a,
    next: O
  };
}
var br = function(r) {
  return r();
}, we = $.useInsertionEffect ? $.useInsertionEffect : !1, vr = we || br, Or = we || $.useLayoutEffect, yr = !1, Ee = /* @__PURE__ */ $.createContext(
  // we're doing this to avoid preconstruct's dead code elimination in this one case
  // because this module is primarily intended for the browser and node
  // but it's also required in react native and similar environments sometimes
  // and we could have a special build just for that
  // but this is much easier and the native packages
  // might use a different theme context in the future anyway
  typeof HTMLElement < "u" ? /* @__PURE__ */ nr({
    key: "css"
  }) : null
);
Ee.Provider;
var gr = function(r) {
  return /* @__PURE__ */ $e(function(t, n) {
    var a = ke(Ee);
    return r(t, a, n);
  });
}, xr = /* @__PURE__ */ $.createContext({}), Ce = {}.hasOwnProperty, ee = "__EMOTION_TYPE_PLEASE_DO_NOT_USE__", Tr = function(r, t) {
  var n = {};
  for (var a in t)
    Ce.call(t, a) && (n[a] = t[a]);
  return n[ee] = r, n;
}, wr = function(r) {
  var t = r.cache, n = r.serialized, a = r.isStringTag;
  return ge(t, n, a), vr(function() {
    return ir(t, n, a);
  }), null;
}, Er = /* @__PURE__ */ gr(function(e, r, t) {
  var n = e.css;
  typeof n == "string" && r.registered[n] !== void 0 && (n = r.registered[n]);
  var a = e[ee], s = [n], i = "";
  typeof e.className == "string" ? i = sr(r.registered, s, e.className) : e.className != null && (i = e.className + " ");
  var c = mr(s, void 0, $.useContext(xr));
  i += r.key + "-" + c.name;
  var o = {};
  for (var d in e)
    Ce.call(e, d) && d !== "css" && d !== ee && !yr && (o[d] = e[d]);
  return o.className = i, t && (o.ref = t), /* @__PURE__ */ $.createElement($.Fragment, null, /* @__PURE__ */ $.createElement(wr, {
    cache: r,
    serialized: c,
    isStringTag: typeof a == "string"
  }), /* @__PURE__ */ $.createElement(a, o));
}), Nr = Er;
function re(e, r) {
  const t = {
    ...r
  };
  for (const n in e)
    if (Object.prototype.hasOwnProperty.call(e, n)) {
      const a = n;
      if (a === "components" || a === "slots")
        t[a] = {
          ...e[a],
          ...t[a]
        };
      else if (a === "componentsProps" || a === "slotProps") {
        const s = e[a], i = r[a];
        if (!i)
          t[a] = s || {};
        else if (!s)
          t[a] = i;
        else {
          t[a] = {
            ...i
          };
          for (const c in s)
            if (Object.prototype.hasOwnProperty.call(s, c)) {
              const o = c;
              t[a][o] = re(s[o], i[o]);
            }
        }
      } else t[a] === void 0 && (t[a] = e[a]);
    }
  return t;
}
const Se = /* @__PURE__ */ $.createContext(void 0);
function Cr({
  value: e,
  children: r
}) {
  return /* @__PURE__ */ Pe(Se.Provider, {
    value: e,
    children: r
  });
}
process.env.NODE_ENV !== "production" && (Cr.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │ To update them, edit the TypeScript types and run `pnpm proptypes`. │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * @ignore
   */
  children: se.node,
  /**
   * @ignore
   */
  value: se.object
});
function Sr(e) {
  const {
    theme: r,
    name: t,
    props: n
  } = e;
  if (!r || !r.components || !r.components[t])
    return n;
  const a = r.components[t];
  return a.defaultProps ? re(a.defaultProps, n) : !a.styleOverrides && !a.variants ? re(a, n) : n;
}
function Rr({
  props: e,
  name: r
}) {
  const t = $.useContext(Se);
  return Sr({
    props: e,
    name: r,
    theme: {
      components: t
    }
  });
}
export {
  Cr as D,
  Nr as E,
  Ar as T,
  xr as a,
  ge as b,
  Tr as c,
  vr as d,
  Rr as e,
  sr as g,
  Ce as h,
  ir as i,
  re as r,
  mr as s,
  Or as u,
  gr as w
};
