// @notificationapi/react/dist/assets/dividerClasses.js
import * as n from "react";
import { g as h } from "./DefaultPropsProvider.js";
import { g } from "./createTheme.js";
let l = 0;
function m(e) {
  const [t, o] = n.useState(e), s = e || t;
  return n.useEffect(() => {
    t == null && (l += 1, o(`mui-${l}`));
  }, [t]), s;
}
const p = {
  ...n
}, u = p.useId;
function v(e) {
  if (u !== void 0) {
    const t = u();
    return e ?? t;
  }
  return m(e);
}
function C({
  controlled: e,
  default: t,
  name: o,
  state: s = "value"
}) {
  const {
    current: i
  } = n.useRef(e !== void 0), [c, a] = n.useState(t), d = i ? e : c;
  if (process.env.NODE_ENV !== "production") {
    n.useEffect(() => {
      i !== (e !== void 0) && console.error([`MUI: A component is changing the ${i ? "" : "un"}controlled ${s} state of ${o} to be ${i ? "un" : ""}controlled.`, "Elements should not switch from uncontrolled to controlled (or vice versa).", `Decide between using a controlled or uncontrolled ${o} element for the lifetime of the component.`, "The nature of the state is determined during the first render. It's considered controlled if the value is not `undefined`.", "More info: https://fb.me/react-controlled-components"].join(`
`));
    }, [s, o, e]);
    const {
      current: r
    } = n.useRef(t);
    n.useEffect(() => {
      !i && !Object.is(r, t) && console.error([`MUI: A component is changing the default ${s} state of an uncontrolled ${o} after being initialized. To suppress this warning opt to use a controlled ${o}.`].join(`
`));
    }, [JSON.stringify(t)]);
  }
  const f = n.useCallback((r) => {
    i || a(r);
  }, []);
  return [d, f];
}
function $(e) {
  return g("MuiDivider", e);
}
const w = h("MuiDivider", ["root", "absolute", "fullWidth", "inset", "middle", "flexItem", "light", "vertical", "withChildren", "withChildrenVertical", "textAlignRight", "textAlignLeft", "wrapper", "wrapperVertical"]);
export {
  C as a,
  w as d,
  $ as g,
  v as u
};
