// @notificationapi/react/dist/assets/List.js
import * as i from "react";
import { g as y, P as s } from "./createTheme.js";
import { g as h, s as L, u as v, c as x, a as N } from "./DefaultPropsProvider.js";
import { jsx as C, jsxs as P } from "react/jsx-runtime";
function S(e, o) {
  var t, n, a;
  return /* @__PURE__ */ i.isValidElement(e) && o.indexOf(
    // For server components `muiName` is avaialble in element.type._payload.value.muiName
    // relevant info - https://github.com/facebook/react/blob/2807d781a08db8e9873687fccc25c0f12b4fb3d4/packages/react/src/ReactLazy.js#L45
    // eslint-disable-next-line no-underscore-dangle
    e.type.muiName ?? ((a = (n = (t = e.type) == null ? void 0 : t._payload) == null ? void 0 : n.value) == null ? void 0 : a.muiName)
  ) !== -1;
}
function U(e) {
  return typeof e == "string";
}
const c = /* @__PURE__ */ i.createContext({});
process.env.NODE_ENV !== "production" && (c.displayName = "ListContext");
function T(e) {
  return y("MuiList", e);
}
h("MuiList", ["root", "padding", "dense", "subheader"]);
const E = (e) => {
  const {
    classes: o,
    disablePadding: t,
    dense: n,
    subheader: a
  } = e;
  return N({
    root: ["root", !t && "padding", n && "dense", a && "subheader"]
  }, T, o);
}, M = L("ul", {
  name: "MuiList",
  slot: "Root",
  overridesResolver: (e, o) => {
    const {
      ownerState: t
    } = e;
    return [o.root, !t.disablePadding && o.padding, t.dense && o.dense, t.subheader && o.subheader];
  }
})({
  listStyle: "none",
  margin: 0,
  padding: 0,
  position: "relative",
  variants: [{
    props: ({
      ownerState: e
    }) => !e.disablePadding,
    style: {
      paddingTop: 8,
      paddingBottom: 8
    }
  }, {
    props: ({
      ownerState: e
    }) => e.subheader,
    style: {
      paddingTop: 0
    }
  }]
}), O = /* @__PURE__ */ i.forwardRef(function(o, t) {
  const n = v({
    props: o,
    name: "MuiList"
  }), {
    children: a,
    className: d,
    component: l = "ul",
    dense: r = !1,
    disablePadding: u = !1,
    subheader: m,
    ...f
  } = n, b = i.useMemo(() => ({
    dense: r
  }), [r]), p = {
    ...n,
    component: l,
    dense: r,
    disablePadding: u
  }, g = E(p);
  return /* @__PURE__ */ C(c.Provider, {
    value: b,
    children: /* @__PURE__ */ P(M, {
      as: l,
      className: x(g.root, d),
      ref: t,
      ownerState: p,
      ...f,
      children: [m, a]
    })
  });
});
process.env.NODE_ENV !== "production" && (O.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: s.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: s.object,
  /**
   * @ignore
   */
  className: s.string,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: s.elementType,
  /**
   * If `true`, compact vertical padding designed for keyboard and mouse input is used for
   * the list and list items.
   * The prop is available to descendant components as the `dense` context.
   * @default false
   */
  dense: s.bool,
  /**
   * If `true`, vertical padding is removed from the list.
   * @default false
   */
  disablePadding: s.bool,
  /**
   * The content of the subheader, normally `ListSubheader`.
   */
  subheader: s.node,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: s.oneOfType([s.arrayOf(s.oneOfType([s.func, s.object, s.bool])), s.func, s.object])
});
export {
  O as L,
  S as a,
  c as b,
  U as i
};
