// @notificationapi/react/dist/assets/Modal.js
import * as f from "react";
import { P as e, g as le } from "./createTheme.js";
import { g as de, s as J, u as ue, c as X, a as fe, m as ke } from "./DefaultPropsProvider.js";
import { o as j, e as Q, g as pe, P as Pe, H as Ne } from "./Portal.js";
import { u as Y, e as Ce } from "./resolveComponentProps.js";
import { u as Se, b as H, f as oe } from "./ButtonBase.js";
import { c as re } from "./Stack.js";
import { jsx as A, jsxs as be } from "react/jsx-runtime";
import { T as we, r as Ie, g as se } from "./utils.js";
import { e as Oe } from "./exactProp.js";
function _(t) {
  return j(t).defaultView || window;
}
function Fe(t = window) {
  const n = t.document.documentElement.clientWidth;
  return t.innerWidth - n;
}
const Me = {
  entering: {
    opacity: 1
  },
  entered: {
    opacity: 1
  }
}, me = /* @__PURE__ */ f.forwardRef(function(n, r) {
  const o = Se(), i = {
    enter: o.transitions.duration.enteringScreen,
    exit: o.transitions.duration.leavingScreen
  }, {
    addEndListener: s,
    appear: a = !0,
    children: c,
    easing: p,
    in: y,
    onEnter: P,
    onEntered: g,
    onEntering: v,
    onExit: h,
    onExited: b,
    onExiting: I,
    style: T,
    timeout: k = i,
    // eslint-disable-next-line react/prop-types
    TransitionComponent: N = we,
    ...l
  } = n, m = f.useRef(null), C = Y(m, pe(c), r), x = (E) => (d) => {
    if (E) {
      const u = m.current;
      d === void 0 ? E(u) : E(u, d);
    }
  }, R = x(v), S = x((E, d) => {
    Ie(E);
    const u = se({
      style: T,
      timeout: k,
      easing: p
    }, {
      mode: "enter"
    });
    E.style.webkitTransition = o.transitions.create("opacity", u), E.style.transition = o.transitions.create("opacity", u), P && P(E, d);
  }), w = x(g), F = x(I), D = x((E) => {
    const d = se({
      style: T,
      timeout: k,
      easing: p
    }, {
      mode: "exit"
    });
    E.style.webkitTransition = o.transitions.create("opacity", d), E.style.transition = o.transitions.create("opacity", d), h && h(E);
  }), B = x(b);
  return /* @__PURE__ */ A(N, {
    appear: a,
    in: y,
    nodeRef: m,
    onEnter: S,
    onEntered: w,
    onEntering: R,
    onExit: D,
    onExited: B,
    onExiting: F,
    addEndListener: (E) => {
      s && s(m.current, E);
    },
    timeout: k,
    ...l,
    children: (E, {
      ownerState: d,
      ...u
    }) => /* @__PURE__ */ f.cloneElement(c, {
      style: {
        opacity: 0,
        visibility: E === "exited" && !y ? "hidden" : void 0,
        ...Me[E],
        ...T,
        ...c.props.style
      },
      ref: C,
      ...u
    })
  });
});
process.env.NODE_ENV !== "production" && (me.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * Add a custom transition end trigger. Called with the transitioning DOM
   * node and a done callback. Allows for more fine grained transition end
   * logic. Note: Timeouts are still used as a fallback if provided.
   */
  addEndListener: e.func,
  /**
   * Perform the enter transition when it first mounts if `in` is also `true`.
   * Set this to `false` to disable this behavior.
   * @default true
   */
  appear: e.bool,
  /**
   * A single child content element.
   */
  children: Q.isRequired,
  /**
   * The transition timing function.
   * You may specify a single easing or a object containing enter and exit values.
   */
  easing: e.oneOfType([e.shape({
    enter: e.string,
    exit: e.string
  }), e.string]),
  /**
   * If `true`, the component will transition in.
   */
  in: e.bool,
  /**
   * @ignore
   */
  onEnter: e.func,
  /**
   * @ignore
   */
  onEntered: e.func,
  /**
   * @ignore
   */
  onEntering: e.func,
  /**
   * @ignore
   */
  onExit: e.func,
  /**
   * @ignore
   */
  onExited: e.func,
  /**
   * @ignore
   */
  onExiting: e.func,
  /**
   * @ignore
   */
  style: e.object,
  /**
   * The duration for the transition, in milliseconds.
   * You may specify a single timeout for all transitions, or individually with an object.
   * @default {
   *   enter: theme.transitions.duration.enteringScreen,
   *   exit: theme.transitions.duration.leavingScreen,
   * }
   */
  timeout: e.oneOfType([e.number, e.shape({
    appear: e.number,
    enter: e.number,
    exit: e.number
  })])
});
function Ae(t) {
  return le("MuiBackdrop", t);
}
de("MuiBackdrop", ["root", "invisible"]);
const De = (t) => {
  const {
    classes: n,
    invisible: r
  } = t;
  return fe({
    root: ["root", r && "invisible"]
  }, Ae, n);
}, Be = J("div", {
  name: "MuiBackdrop",
  slot: "Root",
  overridesResolver: (t, n) => {
    const {
      ownerState: r
    } = t;
    return [n.root, r.invisible && n.invisible];
  }
})({
  position: "fixed",
  display: "flex",
  alignItems: "center",
  justifyContent: "center",
  right: 0,
  bottom: 0,
  top: 0,
  left: 0,
  backgroundColor: "rgba(0, 0, 0, 0.5)",
  WebkitTapHighlightColor: "transparent",
  variants: [{
    props: {
      invisible: !0
    },
    style: {
      backgroundColor: "transparent"
    }
  }]
}), he = /* @__PURE__ */ f.forwardRef(function(n, r) {
  const o = ue({
    props: n,
    name: "MuiBackdrop"
  }), {
    children: i,
    className: s,
    component: a = "div",
    invisible: c = !1,
    open: p,
    components: y = {},
    componentsProps: P = {},
    slotProps: g = {},
    slots: v = {},
    TransitionComponent: h,
    transitionDuration: b,
    ...I
  } = o, T = {
    ...o,
    component: a,
    invisible: c
  }, k = De(T), N = {
    transition: h,
    root: y.Root,
    ...v
  }, l = {
    ...P,
    ...g
  }, m = {
    slots: N,
    slotProps: l
  }, [C, x] = H("root", {
    elementType: Be,
    externalForwardedProps: m,
    className: X(k.root, s),
    ownerState: T
  }), [R, S] = H("transition", {
    elementType: me,
    externalForwardedProps: m,
    ownerState: T
  });
  return /* @__PURE__ */ A(R, {
    in: p,
    timeout: b,
    ...I,
    ...S,
    children: /* @__PURE__ */ A(C, {
      "aria-hidden": !0,
      ...x,
      classes: k,
      ref: r,
      children: i
    })
  });
});
process.env.NODE_ENV !== "production" && (he.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: e.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: e.elementType,
  /**
   * The components used for each slot inside.
   *
   * @deprecated Use the `slots` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  components: e.shape({
    Root: e.elementType
  }),
  /**
   * The extra props for the slot components.
   * You can override the existing props or add new ones.
   *
   * @deprecated Use the `slotProps` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  componentsProps: e.shape({
    root: e.object
  }),
  /**
   * If `true`, the backdrop is invisible.
   * It can be used when rendering a popover or a custom select component.
   * @default false
   */
  invisible: e.bool,
  /**
   * If `true`, the component is shown.
   */
  open: e.bool.isRequired,
  /**
   * The props used for each slot inside.
   * @default {}
   */
  slotProps: e.shape({
    root: e.oneOfType([e.func, e.object]),
    transition: e.oneOfType([e.func, e.object])
  }),
  /**
   * The components used for each slot inside.
   * @default {}
   */
  slots: e.shape({
    root: e.elementType,
    transition: e.elementType
  }),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * The component used for the transition.
   * [Follow this guide](https://mui.com/material-ui/transitions/#transitioncomponent-prop) to learn more about the requirements for this component.
   * @default Fade
   * @deprecated Use `slots.transition` instead. This prop will be removed in v7. See [Migrating from deprecated APIs](/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   */
  TransitionComponent: e.elementType,
  /**
   * The duration for the transition, in milliseconds.
   * You may specify a single timeout for all transitions, or individually with an object.
   */
  transitionDuration: e.oneOfType([e.number, e.shape({
    appear: e.number,
    enter: e.number,
    exit: e.number
  })])
});
function Le(t) {
  const n = j(t);
  return n.body === t ? _(t).innerWidth > n.documentElement.clientWidth : t.scrollHeight > t.clientHeight;
}
function W(t, n) {
  n ? t.setAttribute("aria-hidden", "true") : t.removeAttribute("aria-hidden");
}
function ie(t) {
  return parseInt(_(t).getComputedStyle(t).paddingRight, 10) || 0;
}
function je(t) {
  const r = ["TEMPLATE", "SCRIPT", "STYLE", "LINK", "MAP", "META", "NOSCRIPT", "PICTURE", "COL", "COLGROUP", "PARAM", "SLOT", "SOURCE", "TRACK"].includes(t.tagName), o = t.tagName === "INPUT" && t.getAttribute("type") === "hidden";
  return r || o;
}
function ae(t, n, r, o, i) {
  const s = [n, r, ...o];
  [].forEach.call(t.children, (a) => {
    const c = !s.includes(a), p = !je(a);
    c && p && W(a, i);
  });
}
function G(t, n) {
  let r = -1;
  return t.some((o, i) => n(o) ? (r = i, !0) : !1), r;
}
function Ue(t, n) {
  const r = [], o = t.container;
  if (!n.disableScrollLock) {
    if (Le(o)) {
      const a = Fe(_(o));
      r.push({
        value: o.style.paddingRight,
        property: "padding-right",
        el: o
      }), o.style.paddingRight = `${ie(o) + a}px`;
      const c = j(o).querySelectorAll(".mui-fixed");
      [].forEach.call(c, (p) => {
        r.push({
          value: p.style.paddingRight,
          property: "padding-right",
          el: p
        }), p.style.paddingRight = `${ie(p) + a}px`;
      });
    }
    let s;
    if (o.parentNode instanceof DocumentFragment)
      s = j(o).body;
    else {
      const a = o.parentElement, c = _(o);
      s = (a == null ? void 0 : a.nodeName) === "HTML" && c.getComputedStyle(a).overflowY === "scroll" ? a : o;
    }
    r.push({
      value: s.style.overflow,
      property: "overflow",
      el: s
    }, {
      value: s.style.overflowX,
      property: "overflow-x",
      el: s
    }, {
      value: s.style.overflowY,
      property: "overflow-y",
      el: s
    }), s.style.overflow = "hidden";
  }
  return () => {
    r.forEach(({
      value: s,
      el: a,
      property: c
    }) => {
      s ? a.style.setProperty(c, s) : a.style.removeProperty(c);
    });
  };
}
function Ke(t) {
  const n = [];
  return [].forEach.call(t.children, (r) => {
    r.getAttribute("aria-hidden") === "true" && n.push(r);
  }), n;
}
class We {
  constructor() {
    this.modals = [], this.containers = [];
  }
  add(n, r) {
    let o = this.modals.indexOf(n);
    if (o !== -1)
      return o;
    o = this.modals.length, this.modals.push(n), n.modalRef && W(n.modalRef, !1);
    const i = Ke(r);
    ae(r, n.mount, n.modalRef, i, !0);
    const s = G(this.containers, (a) => a.container === r);
    return s !== -1 ? (this.containers[s].modals.push(n), o) : (this.containers.push({
      modals: [n],
      container: r,
      restore: null,
      hiddenSiblings: i
    }), o);
  }
  mount(n, r) {
    const o = G(this.containers, (s) => s.modals.includes(n)), i = this.containers[o];
    i.restore || (i.restore = Ue(i, r));
  }
  remove(n, r = !0) {
    const o = this.modals.indexOf(n);
    if (o === -1)
      return o;
    const i = G(this.containers, (a) => a.modals.includes(n)), s = this.containers[i];
    if (s.modals.splice(s.modals.indexOf(n), 1), this.modals.splice(o, 1), s.modals.length === 0)
      s.restore && s.restore(), n.modalRef && W(n.modalRef, r), ae(s.container, n.mount, n.modalRef, s.hiddenSiblings, !1), this.containers.splice(i, 1);
    else {
      const a = s.modals[s.modals.length - 1];
      a.modalRef && W(a.modalRef, !1);
    }
    return o;
  }
  isTopModal(n) {
    return this.modals.length > 0 && this.modals[this.modals.length - 1] === n;
  }
}
const qe = ["input", "select", "textarea", "a[href]", "button", "[tabindex]", "audio[controls]", "video[controls]", '[contenteditable]:not([contenteditable="false"])'].join(",");
function Ve(t) {
  const n = parseInt(t.getAttribute("tabindex") || "", 10);
  return Number.isNaN(n) ? t.contentEditable === "true" || (t.nodeName === "AUDIO" || t.nodeName === "VIDEO" || t.nodeName === "DETAILS") && t.getAttribute("tabindex") === null ? 0 : t.tabIndex : n;
}
function ze(t) {
  if (t.tagName !== "INPUT" || t.type !== "radio" || !t.name)
    return !1;
  const n = (o) => t.ownerDocument.querySelector(`input[type="radio"]${o}`);
  let r = n(`[name="${t.name}"]:checked`);
  return r || (r = n(`[name="${t.name}"]`)), r !== t;
}
function He(t) {
  return !(t.disabled || t.tagName === "INPUT" && t.type === "hidden" || ze(t));
}
function _e(t) {
  const n = [], r = [];
  return Array.from(t.querySelectorAll(qe)).forEach((o, i) => {
    const s = Ve(o);
    s === -1 || !He(o) || (s === 0 ? n.push(o) : r.push({
      documentOrder: i,
      tabIndex: s,
      node: o
    }));
  }), r.sort((o, i) => o.tabIndex === i.tabIndex ? o.documentOrder - i.documentOrder : o.tabIndex - i.tabIndex).map((o) => o.node).concat(n);
}
function $e() {
  return !0;
}
function $(t) {
  const {
    children: n,
    disableAutoFocus: r = !1,
    disableEnforceFocus: o = !1,
    disableRestoreFocus: i = !1,
    getTabbable: s = _e,
    isEnabled: a = $e,
    open: c
  } = t, p = f.useRef(!1), y = f.useRef(null), P = f.useRef(null), g = f.useRef(null), v = f.useRef(null), h = f.useRef(!1), b = f.useRef(null), I = Y(pe(n), b), T = f.useRef(null);
  f.useEffect(() => {
    !c || !b.current || (h.current = !r);
  }, [r, c]), f.useEffect(() => {
    if (!c || !b.current)
      return;
    const l = j(b.current);
    return b.current.contains(l.activeElement) || (b.current.hasAttribute("tabIndex") || (process.env.NODE_ENV !== "production" && console.error(["MUI: The modal content node does not accept focus.", 'For the benefit of assistive technologies, the tabIndex of the node is being set to "-1".'].join(`
`)), b.current.setAttribute("tabIndex", "-1")), h.current && b.current.focus()), () => {
      i || (g.current && g.current.focus && (p.current = !0, g.current.focus()), g.current = null);
    };
  }, [c]), f.useEffect(() => {
    if (!c || !b.current)
      return;
    const l = j(b.current), m = (R) => {
      T.current = R, !(o || !a() || R.key !== "Tab") && l.activeElement === b.current && R.shiftKey && (p.current = !0, P.current && P.current.focus());
    }, C = () => {
      var w, F;
      const R = b.current;
      if (R === null)
        return;
      if (!l.hasFocus() || !a() || p.current) {
        p.current = !1;
        return;
      }
      if (R.contains(l.activeElement) || o && l.activeElement !== y.current && l.activeElement !== P.current)
        return;
      if (l.activeElement !== v.current)
        v.current = null;
      else if (v.current !== null)
        return;
      if (!h.current)
        return;
      let S = [];
      if ((l.activeElement === y.current || l.activeElement === P.current) && (S = s(b.current)), S.length > 0) {
        const D = !!((w = T.current) != null && w.shiftKey && ((F = T.current) == null ? void 0 : F.key) === "Tab"), B = S[0], U = S[S.length - 1];
        typeof B != "string" && typeof U != "string" && (D ? U.focus() : B.focus());
      } else
        R.focus();
    };
    l.addEventListener("focusin", C), l.addEventListener("keydown", m, !0);
    const x = setInterval(() => {
      l.activeElement && l.activeElement.tagName === "BODY" && C();
    }, 50);
    return () => {
      clearInterval(x), l.removeEventListener("focusin", C), l.removeEventListener("keydown", m, !0);
    };
  }, [r, o, i, a, c, s]);
  const k = (l) => {
    g.current === null && (g.current = l.relatedTarget), h.current = !0, v.current = l.target;
    const m = n.props.onFocus;
    m && m(l);
  }, N = (l) => {
    g.current === null && (g.current = l.relatedTarget), h.current = !0;
  };
  return /* @__PURE__ */ be(f.Fragment, {
    children: [/* @__PURE__ */ A("div", {
      tabIndex: c ? 0 : -1,
      onFocus: N,
      ref: y,
      "data-testid": "sentinelStart"
    }), /* @__PURE__ */ f.cloneElement(n, {
      ref: I,
      onFocus: k
    }), /* @__PURE__ */ A("div", {
      tabIndex: c ? 0 : -1,
      onFocus: N,
      ref: P,
      "data-testid": "sentinelEnd"
    })]
  });
}
process.env.NODE_ENV !== "production" && ($.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │ To update them, edit the TypeScript types and run `pnpm proptypes`. │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * A single child content element.
   */
  children: Q,
  /**
   * If `true`, the focus trap will not automatically shift focus to itself when it opens, and
   * replace it to the last focused element when it closes.
   * This also works correctly with any focus trap children that have the `disableAutoFocus` prop.
   *
   * Generally this should never be set to `true` as it makes the focus trap less
   * accessible to assistive technologies, like screen readers.
   * @default false
   */
  disableAutoFocus: e.bool,
  /**
   * If `true`, the focus trap will not prevent focus from leaving the focus trap while open.
   *
   * Generally this should never be set to `true` as it makes the focus trap less
   * accessible to assistive technologies, like screen readers.
   * @default false
   */
  disableEnforceFocus: e.bool,
  /**
   * If `true`, the focus trap will not restore focus to previously focused element once
   * focus trap is hidden or unmounted.
   * @default false
   */
  disableRestoreFocus: e.bool,
  /**
   * Returns an array of ordered tabbable nodes (i.e. in tab order) within the root.
   * For instance, you can provide the "tabbable" npm dependency.
   * @param {HTMLElement} root
   */
  getTabbable: e.func,
  /**
   * This prop extends the `open` prop.
   * It allows to toggle the open state without having to wait for a rerender when changing the `open` prop.
   * This prop should be memoized.
   * It can be used to support multiple focus trap mounted at the same time.
   * @default function defaultIsEnabled(): boolean {
   *   return true;
   * }
   */
  isEnabled: e.func,
  /**
   * If `true`, focus is locked.
   */
  open: e.bool.isRequired
});
process.env.NODE_ENV !== "production" && ($.propTypes = Oe($.propTypes));
function Ye(t) {
  return typeof t == "function" ? t() : t;
}
function Ge(t) {
  return t ? t.props.hasOwnProperty("in") : !1;
}
const ce = () => {
}, z = new We();
function Xe(t) {
  const {
    container: n,
    disableEscapeKeyDown: r = !1,
    disableScrollLock: o = !1,
    closeAfterTransition: i = !1,
    onTransitionEnter: s,
    onTransitionExited: a,
    children: c,
    onClose: p,
    open: y,
    rootRef: P
  } = t, g = f.useRef({}), v = f.useRef(null), h = f.useRef(null), b = Y(h, P), [I, T] = f.useState(!y), k = Ge(c);
  let N = !0;
  (t["aria-hidden"] === "false" || t["aria-hidden"] === !1) && (N = !1);
  const l = () => j(v.current), m = () => (g.current.modalRef = h.current, g.current.mount = v.current, g.current), C = () => {
    z.mount(m(), {
      disableScrollLock: o
    }), h.current && (h.current.scrollTop = 0);
  }, x = oe(() => {
    const d = Ye(n) || l().body;
    z.add(m(), d), h.current && C();
  }), R = () => z.isTopModal(m()), S = oe((d) => {
    v.current = d, d && (y && R() ? C() : h.current && W(h.current, N));
  }), w = f.useCallback(() => {
    z.remove(m(), N);
  }, [N]);
  f.useEffect(() => () => {
    w();
  }, [w]), f.useEffect(() => {
    y ? x() : (!k || !i) && w();
  }, [y, w, k, i, x]);
  const F = (d) => (u) => {
    var O;
    (O = d.onKeyDown) == null || O.call(d, u), !(u.key !== "Escape" || u.which === 229 || // Wait until IME is settled.
    !R()) && (r || (u.stopPropagation(), p && p(u, "escapeKeyDown")));
  }, D = (d) => (u) => {
    var O;
    (O = d.onClick) == null || O.call(d, u), u.target === u.currentTarget && p && p(u, "backdropClick");
  };
  return {
    getRootProps: (d = {}) => {
      const u = Ce(t);
      delete u.onTransitionEnter, delete u.onTransitionExited;
      const O = {
        ...u,
        ...d
      };
      return {
        /*
         * Marking an element with the role presentation indicates to assistive technology
         * that this element should be ignored; it exists to support the web application and
         * is not meant for humans to interact with directly.
         * https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-static-element-interactions.md
         */
        role: "presentation",
        ...O,
        onKeyDown: F(O),
        ref: b
      };
    },
    getBackdropProps: (d = {}) => {
      const u = d;
      return {
        "aria-hidden": !0,
        ...u,
        onClick: D(u),
        open: y
      };
    },
    getTransitionProps: () => {
      const d = () => {
        T(!1), s && s();
      }, u = () => {
        T(!0), a && a(), i && w();
      };
      return {
        onEnter: re(d, (c == null ? void 0 : c.props.onEnter) ?? ce),
        onExited: re(u, (c == null ? void 0 : c.props.onExited) ?? ce)
      };
    },
    rootRef: b,
    portalRef: S,
    isTopModal: R,
    exited: I,
    hasTransition: k
  };
}
function Je(t) {
  return le("MuiModal", t);
}
de("MuiModal", ["root", "hidden", "backdrop"]);
const Qe = (t) => {
  const {
    open: n,
    exited: r,
    classes: o
  } = t;
  return fe({
    root: ["root", !n && r && "hidden"],
    backdrop: ["backdrop"]
  }, Je, o);
}, Ze = J("div", {
  name: "MuiModal",
  slot: "Root",
  overridesResolver: (t, n) => {
    const {
      ownerState: r
    } = t;
    return [n.root, !r.open && r.exited && n.hidden];
  }
})(ke(({
  theme: t
}) => ({
  position: "fixed",
  zIndex: (t.vars || t).zIndex.modal,
  right: 0,
  bottom: 0,
  top: 0,
  left: 0,
  variants: [{
    props: ({
      ownerState: n
    }) => !n.open && n.exited,
    style: {
      visibility: "hidden"
    }
  }]
}))), et = J(he, {
  name: "MuiModal",
  slot: "Backdrop",
  overridesResolver: (t, n) => n.backdrop
})({
  zIndex: -1
}), tt = /* @__PURE__ */ f.forwardRef(function(n, r) {
  const o = ue({
    name: "MuiModal",
    props: n
  }), {
    BackdropComponent: i = et,
    BackdropProps: s,
    classes: a,
    className: c,
    closeAfterTransition: p = !1,
    children: y,
    container: P,
    component: g,
    components: v = {},
    componentsProps: h = {},
    disableAutoFocus: b = !1,
    disableEnforceFocus: I = !1,
    disableEscapeKeyDown: T = !1,
    disablePortal: k = !1,
    disableRestoreFocus: N = !1,
    disableScrollLock: l = !1,
    hideBackdrop: m = !1,
    keepMounted: C = !1,
    onBackdropClick: x,
    onClose: R,
    onTransitionEnter: S,
    onTransitionExited: w,
    open: F,
    slotProps: D = {},
    slots: B = {},
    // eslint-disable-next-line react/prop-types
    theme: U,
    ...E
  } = o, d = {
    ...o,
    closeAfterTransition: p,
    disableAutoFocus: b,
    disableEnforceFocus: I,
    disableEscapeKeyDown: T,
    disablePortal: k,
    disableRestoreFocus: N,
    disableScrollLock: l,
    hideBackdrop: m,
    keepMounted: C
  }, {
    getRootProps: u,
    getBackdropProps: O,
    getTransitionProps: Ee,
    portalRef: ge,
    isTopModal: ye,
    exited: Z,
    hasTransition: ee
  } = Xe({
    ...d,
    rootRef: r
  }), K = {
    ...d,
    exited: Z
  }, M = Qe(K), q = {};
  if (y.props.tabIndex === void 0 && (q.tabIndex = "-1"), ee) {
    const {
      onEnter: L,
      onExited: V
    } = Ee();
    q.onEnter = L, q.onExited = V;
  }
  const te = {
    ...E,
    slots: {
      root: v.Root,
      backdrop: v.Backdrop,
      ...B
    },
    slotProps: {
      ...h,
      ...D
    }
  }, [Te, xe] = H("root", {
    elementType: Ze,
    externalForwardedProps: te,
    getSlotProps: u,
    additionalProps: {
      ref: r,
      as: g
    },
    ownerState: K,
    className: X(c, M == null ? void 0 : M.root, !K.open && K.exited && (M == null ? void 0 : M.hidden))
  }), [Re, ne] = H("backdrop", {
    elementType: i,
    externalForwardedProps: te,
    additionalProps: s,
    getSlotProps: (L) => O({
      ...L,
      onClick: (V) => {
        x && x(V), L != null && L.onClick && L.onClick(V);
      }
    }),
    className: X(s == null ? void 0 : s.className, M == null ? void 0 : M.backdrop),
    ownerState: K
  }), ve = Y(s == null ? void 0 : s.ref, ne.ref);
  return !C && !F && (!ee || Z) ? null : /* @__PURE__ */ A(Pe, {
    ref: ge,
    container: P,
    disablePortal: k,
    children: /* @__PURE__ */ be(Te, {
      ...xe,
      children: [!m && i ? /* @__PURE__ */ A(Re, {
        ...ne,
        ref: ve
      }) : null, /* @__PURE__ */ A($, {
        disableEnforceFocus: I,
        disableAutoFocus: b,
        disableRestoreFocus: N,
        isEnabled: ye,
        open: F,
        children: /* @__PURE__ */ f.cloneElement(y, q)
      })]
    })
  });
});
process.env.NODE_ENV !== "production" && (tt.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * A backdrop component. This prop enables custom backdrop rendering.
   * @deprecated Use `slots.backdrop` instead. While this prop currently works, it will be removed in the next major version.
   * Use the `slots.backdrop` prop to make your application ready for the next version of Material UI.
   * @default styled(Backdrop, {
   *   name: 'MuiModal',
   *   slot: 'Backdrop',
   *   overridesResolver: (props, styles) => {
   *     return styles.backdrop;
   *   },
   * })({
   *   zIndex: -1,
   * })
   */
  BackdropComponent: e.elementType,
  /**
   * Props applied to the [`Backdrop`](https://mui.com/material-ui/api/backdrop/) element.
   * @deprecated Use `slotProps.backdrop` instead.
   */
  BackdropProps: e.object,
  /**
   * A single child content element.
   */
  children: Q.isRequired,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * When set to true the Modal waits until a nested Transition is completed before closing.
   * @default false
   */
  closeAfterTransition: e.bool,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: e.elementType,
  /**
   * The components used for each slot inside.
   *
   * @deprecated Use the `slots` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  components: e.shape({
    Backdrop: e.elementType,
    Root: e.elementType
  }),
  /**
   * The extra props for the slot components.
   * You can override the existing props or add new ones.
   *
   * @deprecated Use the `slotProps` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   *
   * @default {}
   */
  componentsProps: e.shape({
    backdrop: e.oneOfType([e.func, e.object]),
    root: e.oneOfType([e.func, e.object])
  }),
  /**
   * An HTML element or function that returns one.
   * The `container` will have the portal children appended to it.
   *
   * You can also provide a callback, which is called in a React layout effect.
   * This lets you set the container from a ref, and also makes server-side rendering possible.
   *
   * By default, it uses the body of the top-level document object,
   * so it's simply `document.body` most of the time.
   */
  container: e.oneOfType([Ne, e.func]),
  /**
   * If `true`, the modal will not automatically shift focus to itself when it opens, and
   * replace it to the last focused element when it closes.
   * This also works correctly with any modal children that have the `disableAutoFocus` prop.
   *
   * Generally this should never be set to `true` as it makes the modal less
   * accessible to assistive technologies, like screen readers.
   * @default false
   */
  disableAutoFocus: e.bool,
  /**
   * If `true`, the modal will not prevent focus from leaving the modal while open.
   *
   * Generally this should never be set to `true` as it makes the modal less
   * accessible to assistive technologies, like screen readers.
   * @default false
   */
  disableEnforceFocus: e.bool,
  /**
   * If `true`, hitting escape will not fire the `onClose` callback.
   * @default false
   */
  disableEscapeKeyDown: e.bool,
  /**
   * The `children` will be under the DOM hierarchy of the parent component.
   * @default false
   */
  disablePortal: e.bool,
  /**
   * If `true`, the modal will not restore focus to previously focused element once
   * modal is hidden or unmounted.
   * @default false
   */
  disableRestoreFocus: e.bool,
  /**
   * Disable the scroll lock behavior.
   * @default false
   */
  disableScrollLock: e.bool,
  /**
   * If `true`, the backdrop is not rendered.
   * @default false
   */
  hideBackdrop: e.bool,
  /**
   * Always keep the children in the DOM.
   * This prop can be useful in SEO situation or
   * when you want to maximize the responsiveness of the Modal.
   * @default false
   */
  keepMounted: e.bool,
  /**
   * Callback fired when the backdrop is clicked.
   * @deprecated Use the `onClose` prop with the `reason` argument to handle the `backdropClick` events.
   */
  onBackdropClick: e.func,
  /**
   * Callback fired when the component requests to be closed.
   * The `reason` parameter can optionally be used to control the response to `onClose`.
   *
   * @param {object} event The event source of the callback.
   * @param {string} reason Can be: `"escapeKeyDown"`, `"backdropClick"`.
   */
  onClose: e.func,
  /**
   * A function called when a transition enters.
   */
  onTransitionEnter: e.func,
  /**
   * A function called when a transition has exited.
   */
  onTransitionExited: e.func,
  /**
   * If `true`, the component is shown.
   */
  open: e.bool.isRequired,
  /**
   * The props used for each slot inside the Modal.
   * @default {}
   */
  slotProps: e.shape({
    backdrop: e.oneOfType([e.func, e.object]),
    root: e.oneOfType([e.func, e.object])
  }),
  /**
   * The components used for each slot inside the Modal.
   * Either a string to use a HTML element or a component.
   * @default {}
   */
  slots: e.shape({
    backdrop: e.elementType,
    root: e.elementType
  }),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object])
});
export {
  he as B,
  me as F,
  tt as M,
  Fe as g,
  _ as o
};
