// @notificationapi/react/dist/assets/Notification.js
import { jsx as _, jsxs as ut } from "react/jsx-runtime";
import { g as ai, s as $t, m as ui, u as li, c as ci, a as fi, n as hi } from "./DefaultPropsProvider.js";
import { getThemeColors as pi } from "../utils/theme.js";
import { b as di, u as mi } from "./ButtonBase.js";
import * as mt from "react";
import De, { useMemo as ne, useCallback as gi, useRef as Kt, useState as yi, useEffect as er } from "react";
import { g as wi, P as p, v as dn, q as vi } from "./createTheme.js";
import { c as mn } from "./createSvgIcon.js";
import { T as tr } from "./Typography.js";
import { I as bi } from "./IconButton.js";
import { B as Ti } from "./Badge.js";
const Oi = mn(/* @__PURE__ */ _("path", {
  d: "M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
}), "Person");
function Si(t) {
  return wi("MuiAvatar", t);
}
ai("MuiAvatar", ["root", "colorDefault", "circular", "rounded", "square", "img", "fallback"]);
const ki = (t) => {
  const {
    classes: e,
    variant: r,
    colorDefault: n
  } = t;
  return fi({
    root: ["root", r, n && "colorDefault"],
    img: ["img"],
    fallback: ["fallback"]
  }, Si, e);
}, xi = $t("div", {
  name: "MuiAvatar",
  slot: "Root",
  overridesResolver: (t, e) => {
    const {
      ownerState: r
    } = t;
    return [e.root, e[r.variant], r.colorDefault && e.colorDefault];
  }
})(ui(({
  theme: t
}) => ({
  position: "relative",
  display: "flex",
  alignItems: "center",
  justifyContent: "center",
  flexShrink: 0,
  width: 40,
  height: 40,
  fontFamily: t.typography.fontFamily,
  fontSize: t.typography.pxToRem(20),
  lineHeight: 1,
  borderRadius: "50%",
  overflow: "hidden",
  userSelect: "none",
  variants: [{
    props: {
      variant: "rounded"
    },
    style: {
      borderRadius: (t.vars || t).shape.borderRadius
    }
  }, {
    props: {
      variant: "square"
    },
    style: {
      borderRadius: 0
    }
  }, {
    props: {
      colorDefault: !0
    },
    style: {
      color: (t.vars || t).palette.background.default,
      ...t.vars ? {
        backgroundColor: t.vars.palette.Avatar.defaultBg
      } : {
        backgroundColor: t.palette.grey[400],
        ...t.applyStyles("dark", {
          backgroundColor: t.palette.grey[600]
        })
      }
    }
  }]
}))), Li = $t("img", {
  name: "MuiAvatar",
  slot: "Img",
  overridesResolver: (t, e) => e.img
})({
  width: "100%",
  height: "100%",
  textAlign: "center",
  // Handle non-square image.
  objectFit: "cover",
  // Hide alt text.
  color: "transparent",
  // Hide the image broken icon, only works on Chrome.
  textIndent: 1e4
}), Ai = $t(Oi, {
  name: "MuiAvatar",
  slot: "Fallback",
  overridesResolver: (t, e) => e.fallback
})({
  width: "75%",
  height: "75%"
});
function Fi({
  crossOrigin: t,
  referrerPolicy: e,
  src: r,
  srcSet: n
}) {
  const [i, s] = mt.useState(!1);
  return mt.useEffect(() => {
    if (!r && !n)
      return;
    s(!1);
    let o = !0;
    const a = new Image();
    return a.onload = () => {
      o && s("loaded");
    }, a.onerror = () => {
      o && s("error");
    }, a.crossOrigin = t, a.referrerPolicy = e, a.src = r, n && (a.srcset = n), () => {
      o = !1;
    };
  }, [t, e, r, n]), i;
}
const gn = /* @__PURE__ */ mt.forwardRef(function(e, r) {
  const n = li({
    props: e,
    name: "MuiAvatar"
  }), {
    alt: i,
    children: s,
    className: o,
    component: a = "div",
    slots: l = {},
    slotProps: c = {},
    imgProps: f,
    sizes: d,
    src: g,
    srcSet: w,
    variant: F = "circular",
    ...k
  } = n;
  let P = null;
  const O = {
    ...n,
    component: a,
    variant: F
  }, D = Fi({
    ...f,
    ...typeof c.img == "function" ? c.img(O) : c.img,
    src: g,
    srcSet: w
  }), N = g || w, E = N && D !== "error";
  O.colorDefault = !E, delete O.ownerState;
  const v = ki(O), [ee, pe] = di("img", {
    className: v.img,
    elementType: Li,
    externalForwardedProps: {
      slots: l,
      slotProps: {
        img: {
          ...f,
          ...c.img
        }
      }
    },
    additionalProps: {
      alt: i,
      src: g,
      srcSet: w,
      sizes: d
    },
    ownerState: O
  });
  return E ? P = /* @__PURE__ */ _(ee, {
    ...pe
  }) : s || s === 0 ? P = s : N && i ? P = i[0] : P = /* @__PURE__ */ _(Ai, {
    ownerState: O,
    className: v.fallback
  }), /* @__PURE__ */ _(xi, {
    as: a,
    className: ci(v.root, o),
    ref: r,
    ...k,
    ownerState: O,
    children: P
  });
});
process.env.NODE_ENV !== "production" && (gn.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * Used in combination with `src` or `srcSet` to
   * provide an alt attribute for the rendered `img` element.
   */
  alt: p.string,
  /**
   * Used to render icon or text elements inside the Avatar if `src` is not set.
   * This can be an element, or just a string.
   */
  children: p.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: p.object,
  /**
   * @ignore
   */
  className: p.string,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: p.elementType,
  /**
   * [Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#attributes) applied to the `img` element if the component is used to display an image.
   * It can be used to listen for the loading error event.
   * @deprecated Use `slotProps.img` instead. This prop will be removed in v7. See [Migrating from deprecated APIs](/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   */
  imgProps: p.object,
  /**
   * The `sizes` attribute for the `img` element.
   */
  sizes: p.string,
  /**
   * The props used for each slot inside.
   * @default {}
   */
  slotProps: p.shape({
    img: p.oneOfType([p.func, p.object])
  }),
  /**
   * The components used for each slot inside.
   * @default {}
   */
  slots: p.shape({
    img: p.elementType
  }),
  /**
   * The `src` attribute for the `img` element.
   */
  src: p.string,
  /**
   * The `srcSet` attribute for the `img` element.
   * Use this attribute for responsive image display.
   */
  srcSet: p.string,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: p.oneOfType([p.arrayOf(p.oneOfType([p.func, p.object, p.bool])), p.func, p.object]),
  /**
   * The shape of the avatar.
   * @default 'circular'
   */
  variant: p.oneOfType([p.oneOf(["circular", "rounded", "square"]), p.string])
});
const Di = mn(/* @__PURE__ */ _("path", {
  d: "M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
}), "Check");
class C {
  constructor(e, r, n, i, s) {
    this.kind = e, this.input = r, this.begin = n, this.end = i, this.file = s;
  }
  getText() {
    return this.input.slice(this.begin, this.end);
  }
  getPosition() {
    let [e, r] = [1, 1];
    for (let n = 0; n < this.begin; n++)
      this.input[n] === `
` ? (e++, r = 1) : r++;
    return [e, r];
  }
  size() {
    return this.end - this.begin;
  }
}
class ce {
  liquidMethodMissing(e) {
  }
}
const _i = Object.prototype.toString, rr = String.prototype.toLowerCase, It = Object.hasOwnProperty;
function j(t) {
  return typeof t == "string";
}
function x(t) {
  return typeof t == "function";
}
function Ri(t) {
  return t && x(t.then);
}
function Ye(t) {
  return t && x(t.next) && x(t.throw) && x(t.return);
}
function Pi(t) {
  return t.replace(/[-/\\^$*+?.()|[\]{}]/g, "\\$&");
}
function h(t) {
  return t = m(t), j(t) ? t : z(t) ? "" : A(t) ? t.map((e) => h(e)).join("") : String(t);
}
function jt(t) {
  return t = m(t), A(t) ? t : j(t) && t.length > 0 ? [t] : Ni(t) ? Array.from(t) : nt(t) ? Object.keys(t).map((e) => [e, t[e]]) : [];
}
function L(t) {
  return t = m(t), z(t) ? [] : A(t) ? t : [t];
}
function m(t) {
  return t instanceof ce && x(t.valueOf) ? t.valueOf() : t;
}
function nr(t) {
  return t = Number(t), isNaN(t) ? 0 : t;
}
function zt(t) {
  return typeof t == "number";
}
function yn(t) {
  return t && x(t.toLiquid) ? yn(t.toLiquid()) : t;
}
function z(t) {
  return t == null;
}
function Ei(t) {
  return t === void 0;
}
function A(t) {
  return _i.call(t) === "[object Array]";
}
function Ni(t) {
  return nt(t) && Symbol.iterator in t;
}
function ir(t, e) {
  t = t || {};
  for (const r in t)
    if (It.call(t, r) && e(t[r], r, t) === !1)
      break;
  return t;
}
function wn(t) {
  return t[t.length - 1];
}
function nt(t) {
  const e = typeof t;
  return t !== null && (e === "object" || e === "function");
}
function vn(t, e, r = 1) {
  const n = [];
  for (let i = t; i < e; i += r)
    n.push(i);
  return n;
}
function _e(t, e, r = " ") {
  return bn(t, e, r, (n, i) => i + n);
}
function $i(t, e, r = " ") {
  return bn(t, e, r, (n, i) => n + i);
}
function bn(t, e, r, n) {
  t = String(t);
  let i = e - t.length;
  for (; i-- > 0; )
    t = n(t, r);
  return t;
}
function Tn(t) {
  return t;
}
function Ii(t) {
  return [...t].some((r) => r >= "a" && r <= "z") ? t.toUpperCase() : t.toLowerCase();
}
function ji(t, e) {
  return t.length > e ? t.slice(0, e - 3) + "..." : t;
}
function sr(t, e) {
  return t == null && e == null ? 0 : t == null ? 1 : e == null || (t = rr.call(t), e = rr.call(e), t < e) ? -1 : t > e ? 1 : 0;
}
function I(t) {
  return function(...e) {
    return t.call(this, ...e.map(m));
  };
}
function Mt(t) {
  return t.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
}
const or = "__liquidClass__";
class fe extends Error {
  constructor(e, r) {
    super(typeof e == "string" ? e : e.message), this.context = "", typeof e != "string" && Object.defineProperty(this, "originalError", { value: e, enumerable: !1 }), Object.defineProperty(this, "token", { value: r, enumerable: !1 }), Object.defineProperty(this, or, { value: "LiquidError", enumerable: !1 });
  }
  update() {
    Object.defineProperty(this, "context", { value: Bi(this.token), enumerable: !1 }), this.message = Hi(this.message, this.token), this.stack = this.message + `
` + this.context + `
` + this.stack, this.originalError && (this.stack += `
From ` + this.originalError.stack);
  }
  static is(e) {
    return (e == null ? void 0 : e[or]) === "LiquidError";
  }
}
class zi extends fe {
  constructor(e, r) {
    super(e, r), this.name = "TokenizationError", super.update();
  }
}
class Mi extends fe {
  constructor(e, r) {
    super(e, r), this.name = "ParseError", this.message = e.message, super.update();
  }
}
class Ci extends fe {
  constructor(e, r) {
    super(e, r.token), this.name = "RenderError", this.message = e.message, super.update();
  }
  static is(e) {
    return e.name === "RenderError";
  }
}
class On extends fe {
  constructor(e) {
    super(e[0], e[0].token), this.errors = e, this.name = "LiquidErrors";
    const r = e.length > 1 ? "s" : "";
    this.message = `${e.length} error${r} found`, super.update();
  }
  static is(e) {
    return e.name === "LiquidErrors";
  }
}
class qi extends fe {
  constructor(e, r) {
    super(e, r), this.name = "UndefinedVariableError", this.message = e.message, super.update();
  }
}
class Ui extends Error {
  constructor(e) {
    super(`undefined variable: ${e}`), this.name = "InternalUndefinedVariableError", this.variableName = e;
  }
}
class Vi extends Error {
  constructor(e) {
    super(e), this.name = "AssertionError", this.message = e + "";
  }
}
function Bi(t) {
  const [e, r] = t.getPosition(), n = t.input.split(`
`), i = Math.max(e - 2, 1), s = Math.min(e + 3, n.length);
  return vn(i, s + 1).map((a) => {
    const l = a === e ? ">> " : "   ", c = _e(String(a), String(s).length);
    let f = `${l}${c}| `;
    const d = a === e ? `
` + _e("^", r + f.length) : "";
    return f += n[a - 1], f += d, f;
  }).join(`
`);
}
function Hi(t, e) {
  e.file && (t += `, file:${e.file}`);
  const [r, n] = e.getPosition();
  return t += `, line:${r}, col:${n}`, t;
}
const y = [0, 0, 0, 0, 0, 0, 0, 0, 0, 20, 4, 4, 4, 20, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 20, 2, 8, 0, 0, 0, 0, 8, 0, 0, 0, 64, 0, 65, 0, 0, 33, 33, 33, 33, 33, 33, 33, 33, 33, 33, 0, 0, 2, 2, 2, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0], Wi = 1, ve = 4, ar = 8, Sn = 16, Yi = 32, Ji = 64, Qi = 128;
function Be(t) {
  const e = t.charCodeAt(0);
  return e >= 128 ? !y[e] : !!(y[e] & Wi);
}
y[160] = y[5760] = y[6158] = y[8192] = y[8193] = y[8194] = y[8195] = y[8196] = y[8197] = y[8198] = y[8199] = y[8200] = y[8201] = y[8202] = y[8232] = y[8233] = y[8239] = y[8287] = y[12288] = ve;
y[8220] = y[8221] = Qi;
function R(t, e) {
  if (!t) {
    const r = typeof e == "function" ? e() : e || `expect ${t} to be true`;
    throw new Vi(r);
  }
}
function Je(t, e = `unexpected ${JSON.stringify(t)}`) {
  R(!t, e);
}
class Zi extends ce {
  equals(e) {
    return z(m(e));
  }
  gt() {
    return !1;
  }
  geq() {
    return !1;
  }
  lt() {
    return !1;
  }
  leq() {
    return !1;
  }
  valueOf() {
    return null;
  }
}
class be extends ce {
  equals(e) {
    return e instanceof be ? !1 : (e = m(e), j(e) || A(e) ? e.length === 0 : nt(e) ? Object.keys(e).length === 0 : !1);
  }
  gt() {
    return !1;
  }
  geq() {
    return !1;
  }
  lt() {
    return !1;
  }
  leq() {
    return !1;
  }
  valueOf() {
    return "";
  }
  static is(e) {
    return e instanceof be;
  }
}
class Ct extends be {
  equals(e) {
    return e === !1 || z(m(e)) ? !0 : j(e) ? /^\s*$/.test(e) : super.equals(e);
  }
  static is(e) {
    return e instanceof Ct;
  }
}
class qt extends ce {
  constructor(e, r, n) {
    super(), this.i = 0, this.length = e, this.name = `${n}-${r}`;
  }
  next() {
    this.i++;
  }
  index0() {
    return this.i;
  }
  index() {
    return this.i + 1;
  }
  first() {
    return this.i === 0;
  }
  last() {
    return this.i === this.length - 1;
  }
  rindex() {
    return this.length - this.i;
  }
  rindex0() {
    return this.length - this.i - 1;
  }
  valueOf() {
    return JSON.stringify(this);
  }
}
class ur extends ce {
  constructor(e = () => "") {
    super(), this.superBlockRender = e;
  }
  /**
   * Provide parent access in child block by
   * {{ block.super }}
   */
  super() {
    return this.superBlockRender();
  }
}
function V(t) {
  return t && x(t.equals) && x(t.gt) && x(t.geq) && x(t.lt) && x(t.leq);
}
const lr = new Zi(), kn = {
  true: !0,
  false: !1,
  nil: lr,
  null: lr,
  empty: new be(),
  blank: new Ct()
};
function cr(t) {
  const e = {};
  for (const [r, n] of Object.entries(t)) {
    let i = e;
    for (let s = 0; s < r.length; s++) {
      const o = r[s];
      i[o] = i[o] || {}, s === r.length - 1 && Be(r[s]) && (i[o].needBoundary = !0), i = i[o];
    }
    i.data = n, i.end = !0;
  }
  return e;
}
var Qe = function() {
  return Qe = Object.assign || function(e) {
    for (var r, n = 1, i = arguments.length; n < i; n++) {
      r = arguments[n];
      for (var s in r) Object.prototype.hasOwnProperty.call(r, s) && (e[s] = r[s]);
    }
    return e;
  }, Qe.apply(this, arguments);
};
function q(t, e, r, n) {
  function i(s) {
    return s instanceof r ? s : new r(function(o) {
      o(s);
    });
  }
  return new (r || (r = Promise))(function(s, o) {
    function a(f) {
      try {
        c(n.next(f));
      } catch (d) {
        o(d);
      }
    }
    function l(f) {
      try {
        c(n.throw(f));
      } catch (d) {
        o(d);
      }
    }
    function c(f) {
      f.done ? s(f.value) : i(f.value).then(a, l);
    }
    c((n = n.apply(t, [])).next());
  });
}
function Z(t) {
  return q(this, void 0, void 0, function* () {
    if (!Ye(t))
      return t;
    let e, r = !1, n = "next";
    do {
      const i = t[n](e);
      r = i.done, e = i.value, n = "next";
      try {
        Ye(e) && (e = Z(e)), Ri(e) && (e = yield e);
      } catch (s) {
        n = "throw", e = s;
      }
    } while (!r);
    return e;
  });
}
function G(t) {
  if (!Ye(t))
    return t;
  let e, r = !1, n = "next";
  do {
    const i = t[n](e);
    if (r = i.done, e = i.value, n = "next", Ye(e))
      try {
        e = G(e);
      } catch (s) {
        n = "throw", e = s;
      }
  } while (!r);
  return e;
}
const Gi = /%([-_0^#:]+)?(\d+)?([EO])?(.)/;
function Xi(t) {
  return [31, Ki(t) ? 29 : 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
}
function xn(t) {
  let e = 0;
  for (let r = 0; r < t.getMonth(); ++r)
    e += Xi(t)[r];
  return e + t.getDate();
}
function fr(t, e) {
  const r = xn(t) + (e - t.getDay()), i = 7 - new Date(t.getFullYear(), 0, 1).getDay() + e;
  return String(Math.floor((r - i) / 7) + 1);
}
function Ki(t) {
  const e = t.getFullYear();
  return !!((e & 3) === 0 && (e % 100 || e % 400 === 0 && e));
}
function es(t) {
  const e = t.getDate();
  if ([11, 12, 13].includes(e))
    return "th";
  switch (e % 10) {
    case 1:
      return "st";
    case 2:
      return "nd";
    case 3:
      return "rd";
    default:
      return "th";
  }
}
function ts(t) {
  return parseInt(t.getFullYear().toString().substring(0, 2), 10);
}
const rs = {
  d: 2,
  e: 2,
  H: 2,
  I: 2,
  j: 3,
  k: 2,
  l: 2,
  L: 3,
  m: 2,
  M: 2,
  S: 2,
  U: 2,
  W: 2
}, ns = new Set("aAbBceklpP");
function hr(t, e) {
  const r = Math.abs(t.getTimezoneOffset()), n = Math.floor(r / 60), i = r % 60;
  return (t.getTimezoneOffset() > 0 ? "-" : "+") + _e(n, 2, "0") + (e.flags[":"] ? ":" : "") + _e(i, 2, "0");
}
const gt = {
  a: (t) => t.getShortWeekdayName(),
  A: (t) => t.getLongWeekdayName(),
  b: (t) => t.getShortMonthName(),
  B: (t) => t.getLongMonthName(),
  c: (t) => t.toLocaleString(),
  C: (t) => ts(t),
  d: (t) => t.getDate(),
  e: (t) => t.getDate(),
  H: (t) => t.getHours(),
  I: (t) => String(t.getHours() % 12 || 12),
  j: (t) => xn(t),
  k: (t) => t.getHours(),
  l: (t) => String(t.getHours() % 12 || 12),
  L: (t) => t.getMilliseconds(),
  m: (t) => t.getMonth() + 1,
  M: (t) => t.getMinutes(),
  N: (t, e) => {
    const r = Number(e.width) || 9, n = String(t.getMilliseconds()).slice(0, r);
    return $i(n, r, "0");
  },
  p: (t) => t.getHours() < 12 ? "AM" : "PM",
  P: (t) => t.getHours() < 12 ? "am" : "pm",
  q: (t) => es(t),
  s: (t) => Math.round(t.getTime() / 1e3),
  S: (t) => t.getSeconds(),
  u: (t) => t.getDay() || 7,
  U: (t) => fr(t, 0),
  w: (t) => t.getDay(),
  W: (t) => fr(t, 1),
  x: (t) => t.toLocaleDateString(),
  X: (t) => t.toLocaleTimeString(),
  y: (t) => t.getFullYear().toString().slice(2, 4),
  Y: (t) => t.getFullYear(),
  z: hr,
  Z: (t, e) => t.getTimeZoneName() || hr(t, e),
  t: () => "	",
  n: () => `
`,
  "%": () => "%"
};
gt.h = gt.b;
function He(t, e) {
  let r = "", n = e, i;
  for (; i = Gi.exec(n); )
    r += n.slice(0, i.index), n = n.slice(i.index + i[0].length), r += is(t, i);
  return r + n;
}
function is(t, e) {
  const [r, n = "", i, s, o] = e, a = gt[o];
  if (!a)
    return r;
  const l = {};
  for (const g of n)
    l[g] = !0;
  let c = String(a(t, { flags: l, width: i, modifier: s })), f = ns.has(o) ? " " : "0", d = i || rs[o] || 0;
  return l["^"] ? c = c.toUpperCase() : l["#"] && (c = Ii(c)), l._ ? f = " " : l[0] && (f = "0"), l["-"] && (d = 0), _e(c, d, f);
}
function Ln() {
  return typeof Intl < "u" ? Intl.DateTimeFormat : void 0;
}
const ss = 6e4, os = /([zZ]|([+-])(\d{2}):(\d{2}))$/, An = [
  "January",
  "February",
  "March",
  "April",
  "May",
  "June",
  "July",
  "August",
  "September",
  "October",
  "November",
  "December"
], as = An.map((t) => t.slice(0, 3)), Fn = [
  "Sunday",
  "Monday",
  "Tuesday",
  "Wednesday",
  "Thursday",
  "Friday",
  "Saturday"
], us = Fn.map((t) => t.slice(0, 3));
class U {
  constructor(e, r, n) {
    this.locale = r, this.DateTimeFormat = Ln(), this.date = new Date(e), this.timezoneFixed = n !== void 0, n === void 0 && (n = this.date.getTimezoneOffset()), this.timezoneOffset = j(n) ? U.getTimezoneOffset(n, this.date) : n, this.timezoneName = j(n) ? n : "";
    const i = (this.date.getTimezoneOffset() - this.timezoneOffset) * ss, s = this.date.getTime() + i;
    this.displayDate = new Date(s);
  }
  getTime() {
    return this.displayDate.getTime();
  }
  getMilliseconds() {
    return this.displayDate.getMilliseconds();
  }
  getSeconds() {
    return this.displayDate.getSeconds();
  }
  getMinutes() {
    return this.displayDate.getMinutes();
  }
  getHours() {
    return this.displayDate.getHours();
  }
  getDay() {
    return this.displayDate.getDay();
  }
  getDate() {
    return this.displayDate.getDate();
  }
  getMonth() {
    return this.displayDate.getMonth();
  }
  getFullYear() {
    return this.displayDate.getFullYear();
  }
  toLocaleString(e, r) {
    return r != null && r.timeZone ? this.date.toLocaleString(e, r) : this.displayDate.toLocaleString(e, r);
  }
  toLocaleTimeString(e) {
    return this.displayDate.toLocaleTimeString(e);
  }
  toLocaleDateString(e) {
    return this.displayDate.toLocaleDateString(e);
  }
  getTimezoneOffset() {
    return this.timezoneOffset;
  }
  getTimeZoneName() {
    if (this.timezoneFixed)
      return this.timezoneName;
    if (this.DateTimeFormat)
      return this.DateTimeFormat().resolvedOptions().timeZone;
  }
  getLongMonthName() {
    var e;
    return (e = this.format({ month: "long" })) !== null && e !== void 0 ? e : An[this.getMonth()];
  }
  getShortMonthName() {
    var e;
    return (e = this.format({ month: "short" })) !== null && e !== void 0 ? e : as[this.getMonth()];
  }
  getLongWeekdayName() {
    var e;
    return (e = this.format({ weekday: "long" })) !== null && e !== void 0 ? e : Fn[this.displayDate.getDay()];
  }
  getShortWeekdayName() {
    var e;
    return (e = this.format({ weekday: "short" })) !== null && e !== void 0 ? e : us[this.displayDate.getDay()];
  }
  valid() {
    return !isNaN(this.getTime());
  }
  format(e) {
    return this.DateTimeFormat && this.DateTimeFormat(this.locale, e).format(this.displayDate);
  }
  /**
   * Create a Date object fixed to it's declared Timezone. Both
   * - 2021-08-06T02:29:00.000Z and
   * - 2021-08-06T02:29:00.000+08:00
   * will always be displayed as
   * - 2021-08-06 02:29:00
   * regardless timezoneOffset in JavaScript realm
   *
   * The implementation hack:
   * Instead of calling `.getMonth()`/`.getUTCMonth()` respect to `preserveTimezones`,
   * we create a different Date to trick strftime, it's both simpler and more performant.
   * Given that a template is expected to be parsed fewer times than rendered.
   */
  static createDateFixedToTimezone(e, r) {
    const n = e.match(os);
    if (n && n[1] === "Z")
      return new U(+new Date(e), r, 0);
    if (n && n[2] && n[3] && n[4]) {
      const [, , i, s, o] = n, a = (i === "+" ? -1 : 1) * (parseInt(s, 10) * 60 + parseInt(o, 10));
      return new U(+new Date(e), r, a);
    }
    return new U(e, r);
  }
  static getTimezoneOffset(e, r) {
    const n = r.toLocaleString("en-US", { timeZone: e }), i = r.toLocaleString("en-US", { timeZone: "UTC" }), s = new Date(n);
    return (+new Date(i) - +s) / (60 * 1e3);
  }
}
class yt {
  constructor(e, r) {
    this.base = 0, this.message = `${e} limit exceeded`, this.limit = r;
  }
  use(e) {
    e = nr(e), R(this.base + e <= this.limit, this.message), this.base += e;
  }
  check(e) {
    e = nr(e), R(e <= this.limit, this.message);
  }
}
class Ut extends C {
  constructor(e, [r, n], i, s, o, a, l, c) {
    super(e, i, s, o, c), this.trimLeft = !1, this.trimRight = !1;
    const f = i[r] === "-", d = i[n - 1] === "-";
    let g = f ? r + 1 : r, w = d ? n - 1 : n;
    for (; g < w && y[i.charCodeAt(g)] & ve; )
      g++;
    for (; w > g && y[i.charCodeAt(w - 1)] & ve; )
      w--;
    this.contentRange = [g, w], this.trimLeft = f || a, this.trimRight = d || l;
  }
  get content() {
    return this.input.slice(this.contentRange[0], this.contentRange[1]);
  }
}
class pr extends Ut {
  constructor(e, r, n, i, s) {
    const { trimTagLeft: o, trimTagRight: a, tagDelimiterLeft: l, tagDelimiterRight: c } = i, [f, d] = [r + l.length, n - c.length];
    super(b.Tag, [f, d], e, r, n, o, a, s), this.tokenizer = new Q(e, i.operators, s, this.contentRange), this.name = this.tokenizer.readTagName(), this.tokenizer.assert(this.name, "illegal tag syntax, tag name expected"), this.tokenizer.skipBlank();
  }
  get args() {
    return this.tokenizer.input.slice(this.tokenizer.p, this.contentRange[1]);
  }
}
class ls extends Ut {
  constructor(e, r, n, i, s) {
    const { trimOutputLeft: o, trimOutputRight: a, outputDelimiterLeft: l, outputDelimiterRight: c } = i, f = [r + l.length, n - c.length];
    super(b.Output, f, e, r, n, o, a, s);
  }
}
class dr extends C {
  constructor(e, r, n, i) {
    super(b.HTML, e, r, n, i), this.input = e, this.begin = r, this.end = n, this.file = i, this.trimLeft = 0, this.trimRight = 0;
  }
  getContent() {
    return this.input.slice(this.begin + this.trimLeft, this.end - this.trimRight);
  }
}
class cs extends C {
  constructor(e, r, n, i) {
    super(b.Number, e, r, n, i), this.input = e, this.begin = r, this.end = n, this.file = i, this.content = Number(this.getText());
  }
}
class lt extends C {
  constructor(e, r, n, i) {
    super(b.Word, e, r, n, i), this.input = e, this.begin = r, this.end = n, this.file = i, this.content = this.getText();
  }
}
class fs extends C {
  constructor(e, r, n, i) {
    super(b.Literal, e, r, n, i), this.input = e, this.begin = r, this.end = n, this.file = i, this.literal = this.getText(), this.content = kn[this.literal];
  }
}
const mr = {
  "==": 2,
  "!=": 2,
  ">": 2,
  "<": 2,
  ">=": 2,
  "<=": 2,
  contains: 2,
  not: 1,
  and: 0,
  or: 0
}, hs = {
  "==": 0,
  "!=": 0,
  ">": 0,
  "<": 0,
  ">=": 0,
  "<=": 0,
  contains: 0,
  not: 1,
  and: 0,
  or: 0
  /* OperatorType.Binary */
};
class ps extends C {
  constructor(e, r, n, i) {
    super(b.Operator, e, r, n, i), this.input = e, this.begin = r, this.end = n, this.file = i, this.operator = this.getText();
  }
  getPrecedence() {
    const e = this.getText();
    return e in mr ? mr[e] : 1;
  }
}
class gr extends C {
  constructor(e, r, n, i, s, o) {
    super(b.PropertyAccess, n, i, s, o), this.variable = e, this.props = r;
  }
}
class Dn extends C {
  constructor(e, r, n, i, s, o) {
    super(b.Filter, n, i, s, o), this.name = e, this.args = r;
  }
}
class ds extends C {
  constructor(e, r, n, i, s, o) {
    super(b.Hash, e, r, n, o), this.input = e, this.begin = r, this.end = n, this.name = i, this.value = s, this.file = o;
  }
}
const ms = /[\da-fA-F]/, yr = /[0-7]/, wr = {
  b: "\b",
  f: "\f",
  n: `
`,
  r: "\r",
  t: "	",
  v: "\v"
};
function vr(t) {
  const e = t.charCodeAt(0);
  return e >= 97 ? e - 87 : e >= 65 ? e - 55 : e - 48;
}
function gs(t) {
  let e = "";
  for (let r = 1; r < t.length - 1; r++) {
    if (t[r] !== "\\") {
      e += t[r];
      continue;
    }
    if (wr[t[r + 1]] !== void 0)
      e += wr[t[++r]];
    else if (t[r + 1] === "u") {
      let n = 0, i = r + 2;
      for (; i <= r + 5 && ms.test(t[i]); )
        n = n * 16 + vr(t[i++]);
      r = i - 1, e += String.fromCharCode(n);
    } else if (!yr.test(t[r + 1]))
      e += t[++r];
    else {
      let n = r + 1, i = 0;
      for (; n <= r + 3 && yr.test(t[n]); )
        i = i * 8 + vr(t[n++]);
      r = n - 1, e += String.fromCharCode(i);
    }
  }
  return e;
}
class ys extends C {
  constructor(e, r, n, i) {
    super(b.Quoted, e, r, n, i), this.input = e, this.begin = r, this.end = n, this.file = i, this.content = gs(this.getText());
  }
}
class ws extends C {
  constructor(e, r, n, i, s, o) {
    super(b.Range, e, r, n, o), this.input = e, this.begin = r, this.end = n, this.lhs = i, this.rhs = s, this.file = o;
  }
}
class vs extends Ut {
  constructor(e, r, n, i, s) {
    super(b.Tag, [r, n], e, r, n, !1, !1, s), this.tokenizer = new Q(e, i.operators, s, this.contentRange), this.name = this.tokenizer.readTagName(), this.tokenizer.assert(this.name, "illegal liquid tag syntax"), this.tokenizer.skipBlank(), this.args = this.tokenizer.remaining();
  }
}
class bs extends C {
  constructor(e, r, n, i, s, o) {
    super(b.FilteredValue, n, i, s, o), this.initial = e, this.filters = r, this.input = n, this.begin = i, this.end = s, this.file = o;
  }
}
const Ts = {
  now: () => Date.now()
};
function _n() {
  return typeof global == "object" && global.performance || typeof window == "object" && window.performance || Ts;
}
class Os {
  constructor() {
    this.buffer = "";
  }
  write(e) {
    this.buffer += h(e);
  }
}
class Ss {
  constructor() {
    throw this.buffer = "", this.stream = null, new Error("streaming not supported in browser");
  }
}
class ks {
  constructor() {
    this.buffer = "";
  }
  write(e) {
    e = m(e), typeof e != "string" && this.buffer === "" ? this.buffer = e : this.buffer = h(this.buffer) + h(e);
  }
}
class xs {
  renderTemplatesToNodeStream(e, r) {
    const n = new Ss();
    return Promise.resolve().then(() => Z(this.renderTemplates(e, r, n))).then(() => n.end(), (i) => n.error(i)), n.stream;
  }
  *renderTemplates(e, r, n) {
    n || (n = r.opts.keepOutputType ? new ks() : new Os());
    const i = [];
    for (const s of e) {
      r.renderLimit.check(_n().now());
      try {
        const o = yield s.render(r, n);
        if (o && n.write(o), n.break || n.continue)
          break;
      } catch (o) {
        const a = fe.is(o) ? o : new Ci(o, s);
        if (r.opts.catchAllErrors)
          i.push(a);
        else
          throw a;
      }
    }
    if (i.length)
      throw new On(i);
    return n.buffer;
  }
}
class Ls {
  constructor(e) {
    this.postfix = [...Ds(e)];
  }
  *evaluate(e, r) {
    R(e, "unable to evaluate: context not defined");
    const n = [];
    for (const i of this.postfix)
      if ($n(i)) {
        const s = n.pop();
        let o;
        if (hs[i.operator] === 1)
          o = yield e.opts.operators[i.operator](s, e);
        else {
          const a = n.pop();
          o = yield e.opts.operators[i.operator](a, s, e);
        }
        n.push(o);
      } else
        n.push(yield S(i, e, r));
    return n[0];
  }
  valid() {
    return !!this.postfix.length;
  }
}
function* S(t, e, r = !1) {
  if (t) {
    if ("content" in t)
      return t.content;
    if (mo(t))
      return yield As(t, e, r);
    if (go(t))
      return yield Fs(t, e);
  }
}
function* As(t, e, r) {
  const n = [];
  for (const i of t.props)
    n.push(yield S(i, e, !1));
  try {
    if (t.variable) {
      const i = yield S(t.variable, e, r);
      return yield e._getFromScope(i, n);
    } else
      return yield e._get(n);
  } catch (i) {
    if (r && i.name === "InternalUndefinedVariableError")
      return null;
    throw new qi(i, t);
  }
}
function Rn(t) {
  return t.content;
}
function* Fs(t, e) {
  const r = yield S(t.lhs, e), n = yield S(t.rhs, e);
  return vn(+r, +n + 1);
}
function* Ds(t) {
  const e = [];
  for (const r of t)
    if ($n(r)) {
      for (; e.length && e[e.length - 1].getPrecedence() > r.getPrecedence(); )
        yield e.pop();
      e.push(r);
    } else
      yield r;
  for (; e.length; )
    yield e.pop();
}
function se(t, e) {
  return !it(t, e);
}
function it(t, e) {
  return t = m(t), e.opts.jsTruthy ? !t : t === !1 || t === void 0 || t === null;
}
const _s = {
  "==": H,
  "!=": (t, e) => !H(t, e),
  ">": (t, e) => V(t) ? t.gt(e) : V(e) ? e.lt(t) : m(t) > m(e),
  "<": (t, e) => V(t) ? t.lt(e) : V(e) ? e.gt(t) : m(t) < m(e),
  ">=": (t, e) => V(t) ? t.geq(e) : V(e) ? e.leq(t) : m(t) >= m(e),
  "<=": (t, e) => V(t) ? t.leq(e) : V(e) ? e.geq(t) : m(t) <= m(e),
  contains: (t, e) => (t = m(t), A(t) ? t.some((r) => H(r, e)) : x(t == null ? void 0 : t.indexOf) ? t.indexOf(m(e)) > -1 : !1),
  not: (t, e) => it(m(t), e),
  and: (t, e, r) => se(m(t), r) && se(m(e), r),
  or: (t, e, r) => se(m(t), r) || se(m(e), r)
};
function H(t, e) {
  return V(t) ? t.equals(e) : V(e) ? e.equals(t) : (t = m(t), e = m(e), A(t) ? A(e) && Rs(t, e) : t === e);
}
function Rs(t, e) {
  return t.length !== e.length ? !1 : !t.some((r, n) => !H(r, e[n]));
}
function Ps(t, e) {
  return t.some((r) => H(r, e));
}
class ct {
  constructor(e, r, n, i) {
    this.key = e, this.value = r, this.next = n, this.prev = i;
  }
}
class br {
  constructor(e, r = 0) {
    this.limit = e, this.size = r, this.cache = {}, this.head = new ct("HEAD", null, null, null), this.tail = new ct("TAIL", null, null, null), this.head.next = this.tail, this.tail.prev = this.head;
  }
  write(e, r) {
    if (this.cache[e])
      this.cache[e].value = r;
    else {
      const n = new ct(e, r, this.head.next, this.head);
      this.head.next.prev = n, this.head.next = n, this.cache[e] = n, this.size++, this.ensureLimit();
    }
  }
  read(e) {
    if (!this.cache[e])
      return;
    const { value: r } = this.cache[e];
    return this.remove(e), this.write(e, r), r;
  }
  remove(e) {
    const r = this.cache[e];
    r.prev.next = r.next, r.next.prev = r.prev, delete this.cache[e], this.size--;
  }
  clear() {
    this.head.next = this.tail, this.tail.prev = this.head, this.size = 0, this.cache = {};
  }
  ensureLimit() {
    this.size > this.limit && this.remove(this.tail.prev.key);
  }
}
function Pn(t, e) {
  const r = document.createElement("base");
  r.href = t;
  const n = document.getElementsByTagName("head")[0];
  n.insertBefore(r, n.firstChild);
  const i = document.createElement("a");
  i.href = e;
  const s = i.href;
  return n.removeChild(r), s;
}
function Es(t, e, r) {
  return t.length && wn(t) !== "/" && (t += "/"), Pn(t, e).replace(/^(\w+:\/\/[^/]+)(\/[^?]+)/, (i, s, o) => {
    const a = o.split("/").pop();
    return /\.\w+$/.test(a) ? i : s + o + r;
  });
}
function Ns(t) {
  return q(this, void 0, void 0, function* () {
    return new Promise((e, r) => {
      const n = new XMLHttpRequest();
      n.onload = () => {
        n.status >= 200 && n.status < 300 ? e(n.responseText) : r(new Error(n.statusText));
      }, n.onerror = () => {
        r(new Error("An error occurred whilst receiving the response."));
      }, n.open("GET", t), n.send();
    });
  });
}
function $s(t) {
  const e = new XMLHttpRequest();
  if (e.open("GET", t, !1), e.send(), e.status < 200 || e.status >= 300)
    throw new Error(e.statusText);
  return e.responseText;
}
function Is(t) {
  return q(this, void 0, void 0, function* () {
    return !0;
  });
}
function js(t) {
  return !0;
}
function zs(t) {
  return Pn(t, ".");
}
const Ms = "/";
var Cs = /* @__PURE__ */ Object.freeze({
  __proto__: null,
  resolve: Es,
  readFile: Ns,
  readFileSync: $s,
  exists: Is,
  existsSync: js,
  dirname: zs,
  sep: Ms
});
function qs(t, e, ...r) {
  return t = m(t), A(t) || j(t) ? t.length ? t : e : t === !1 && new Map(r).get("allow_false") ? !1 : it(t, this.context) ? e : t;
}
function Tr(t, e = 0) {
  return JSON.stringify(t, null, e);
}
function Us(t, e = 0) {
  const r = [];
  return JSON.stringify(t, function(n, i) {
    if (typeof i != "object" || i === null)
      return i;
    for (; r.length > 0 && r[r.length - 1] !== this; )
      r.pop();
    return r.includes(i) ? "[Circular]" : (r.push(i), i);
  }, e);
}
function Vs(t) {
  return Number(t);
}
const Bs = {
  raw: !0,
  handler: Tn
};
var En = {
  default: qs,
  raw: Bs,
  jsonify: Tr,
  to_integer: Vs,
  json: Tr,
  inspect: Us
};
const Hs = {
  "&": "&amp;",
  "<": "&lt;",
  ">": "&gt;",
  '"': "&#34;",
  "'": "&#39;"
}, Ws = {
  "&amp;": "&",
  "&lt;": "<",
  "&gt;": ">",
  "&#34;": '"',
  "&#39;": "'"
};
function st(t) {
  return t = h(t), this.context.memoryLimit.use(t.length), t.replace(/&|<|>|"|'/g, (e) => Hs[e]);
}
function Ys(t) {
  return st.call(this, t);
}
function Js(t) {
  return t = h(t), this.context.memoryLimit.use(t.length), t.replace(/&(amp|lt|gt|#34|#39);/g, (e) => Ws[e]);
}
function Qs(t) {
  return st.call(this, Js.call(this, t));
}
function Zs(t) {
  const e = h(t);
  return this.context.memoryLimit.use(e.length), e.replace(/\r?\n/gm, `<br />
`);
}
function Gs(t) {
  const e = h(t);
  return this.context.memoryLimit.use(e.length), e.replace(/<script[\s\S]*?<\/script>|<style[\s\S]*?<\/style>|<.*?>|<!--[\s\S]*?-->/g, "");
}
var Xs = /* @__PURE__ */ Object.freeze({
  __proto__: null,
  escape: st,
  xml_escape: Ys,
  escape_once: Qs,
  newline_to_br: Zs,
  strip_html: Gs
});
class Ks {
  constructor(e) {
    this.mapping = e, this.sep = "/";
  }
  exists(e) {
    return q(this, void 0, void 0, function* () {
      return this.existsSync(e);
    });
  }
  existsSync(e) {
    return !z(this.mapping[e]);
  }
  readFile(e) {
    return q(this, void 0, void 0, function* () {
      return this.readFileSync(e);
    });
  }
  readFileSync(e) {
    const r = this.mapping[e];
    if (z(r))
      throw new Error(`ENOENT: ${e}`);
    return r;
  }
  dirname(e) {
    const r = e.split(this.sep);
    return r.pop(), r.join(this.sep);
  }
  resolve(e, r, n) {
    if (r += n, e === ".")
      return r;
    const i = e.split(/\/+/);
    for (const s of r.split(this.sep))
      s === "." || s === "" || (s === ".." ? (i.length > 1 || i[0] !== "") && i.pop() : i.push(s));
    return i.join(this.sep);
  }
}
const de = {
  root: ["."],
  layouts: ["."],
  partials: ["."],
  relativeReference: !0,
  jekyllInclude: !1,
  keyValueSeparator: ":",
  cache: void 0,
  extname: "",
  fs: Cs,
  dynamicPartials: !0,
  jsTruthy: !1,
  dateFormat: "%A, %B %-e, %Y at %-l:%M %P %z",
  locale: "",
  trimTagRight: !1,
  trimTagLeft: !1,
  trimOutputRight: !1,
  trimOutputLeft: !1,
  greedy: !0,
  tagDelimiterLeft: "{%",
  tagDelimiterRight: "%}",
  outputDelimiterLeft: "{{",
  outputDelimiterRight: "}}",
  preserveTimezones: !1,
  strictFilters: !1,
  strictVariables: !1,
  ownPropertyOnly: !0,
  lenientIf: !1,
  globals: {},
  keepOutputType: !1,
  operators: _s,
  memoryLimit: 1 / 0,
  parseLimit: 1 / 0,
  renderLimit: 1 / 0
};
function eo(t) {
  var e, r;
  if (t.hasOwnProperty("root") && (t.hasOwnProperty("partials") || (t.partials = t.root), t.hasOwnProperty("layouts") || (t.layouts = t.root)), t.hasOwnProperty("cache")) {
    let n;
    typeof t.cache == "number" ? n = t.cache > 0 ? new br(t.cache) : void 0 : typeof t.cache == "object" ? n = t.cache : n = t.cache ? new br(1024) : void 0, t.cache = n;
  }
  return t = Object.assign(Object.assign(Object.assign({}, de), t.jekyllInclude ? { dynamicPartials: !1 } : {}), t), (!t.fs.dirname || !t.fs.sep) && t.relativeReference && (console.warn("[LiquidJS] `fs.dirname` and `fs.sep` are required for relativeReference, set relativeReference to `false` to suppress this warning"), t.relativeReference = !1), t.root = We(t.root), t.partials = We(t.partials), t.layouts = We(t.layouts), t.outputEscape = t.outputEscape && to(t.outputEscape), t.locale || (t.locale = (r = (e = Ln()) === null || e === void 0 ? void 0 : e().resolvedOptions().locale) !== null && r !== void 0 ? r : "en-US"), t.templates && (t.fs = new Ks(t.templates), t.relativeReference = !0, t.root = t.partials = t.layouts = "."), t;
}
function to(t) {
  return t === "escape" ? st : t === "json" ? En.json : (R(x(t), "`outputEscape` need to be of type string or function"), t);
}
function We(t) {
  let e = [];
  return A(t) && (e = t), j(t) && (e = [t]), e;
}
function ro(t, e) {
  let r = !1;
  for (let n = 0; n < t.length; n++) {
    const i = t[n];
    fo(i) && (!r && i.trimLeft && no(t[n - 1], e.greedy), he(i) && (i.name === "raw" ? r = !0 : i.name === "endraw" && (r = !1)), !r && i.trimRight && io(t[n + 1], e.greedy));
  }
}
function no(t, e) {
  if (!t || !Bt(t))
    return;
  const r = e ? ve : Sn;
  for (; y[t.input.charCodeAt(t.end - 1 - t.trimRight)] & r; )
    t.trimRight++;
}
function io(t, e) {
  if (!t || !Bt(t))
    return;
  const r = e ? ve : Sn;
  for (; y[t.input.charCodeAt(t.begin + t.trimLeft)] & r; )
    t.trimLeft++;
  t.input.charAt(t.begin + t.trimLeft) === `
` && t.trimLeft++;
}
class Q {
  constructor(e, r = de.operators, n, i) {
    this.input = e, this.file = n, this.rawBeginAt = -1, this.p = i ? i[0] : 0, this.N = i ? i[1] : e.length, this.opTrie = cr(r), this.literalTrie = cr(kn);
  }
  readExpression() {
    return new Ls(this.readExpressionTokens());
  }
  *readExpressionTokens() {
    for (; this.p < this.N; ) {
      const e = this.readOperator();
      if (e) {
        yield e;
        continue;
      }
      const r = this.readValue();
      if (r) {
        yield r;
        continue;
      }
      return;
    }
  }
  readOperator() {
    this.skipBlank();
    const e = this.matchTrie(this.opTrie);
    if (e !== -1)
      return new ps(this.input, this.p, this.p = e, this.file);
  }
  matchTrie(e) {
    let r = e, n = this.p, i;
    for (; r[this.input[n]] && n < this.N; )
      r = r[this.input[n++]], r.end && (i = r);
    return !i || i.needBoundary && Be(this.peek(n - this.p)) ? -1 : n;
  }
  readFilteredValue() {
    const e = this.p, r = this.readExpression();
    this.assert(r.valid(), `invalid value expression: ${this.snapshot()}`);
    const n = this.readFilters();
    return new bs(r, n, this.input, e, this.p, this.file);
  }
  readFilters() {
    const e = [];
    for (; ; ) {
      const r = this.readFilter();
      if (!r)
        return e;
      e.push(r);
    }
  }
  readFilter() {
    if (this.skipBlank(), this.end())
      return null;
    this.assert(this.read() === "|", 'expected "|" before filter');
    const e = this.readIdentifier();
    if (!e.size())
      return this.assert(this.end(), "expected filter name"), null;
    const r = [];
    if (this.skipBlank(), this.peek() === ":")
      do {
        ++this.p;
        const n = this.readFilterArg();
        n && r.push(n), this.skipBlank(), this.assert(this.end() || this.peek() === "," || this.peek() === "|", () => `unexpected character ${this.snapshot()}`);
      } while (this.peek() === ",");
    else if (!(this.peek() === "|" || this.end())) throw this.error('expected ":" after filter name');
    return new Dn(e.getText(), r, this.input, e.begin, this.p, this.file);
  }
  readFilterArg() {
    const e = this.readValue();
    if (!e)
      return;
    if (this.skipBlank(), this.peek() !== ":")
      return e;
    ++this.p;
    const r = this.readValue();
    return [e.getText(), r];
  }
  readTopLevelTokens(e = de) {
    const r = [];
    for (; this.p < this.N; ) {
      const n = this.readTopLevelToken(e);
      r.push(n);
    }
    return ro(r, e), r;
  }
  readTopLevelToken(e) {
    const { tagDelimiterLeft: r, outputDelimiterLeft: n } = e;
    return this.rawBeginAt > -1 ? this.readEndrawOrRawContent(e) : this.match(r) ? this.readTagToken(e) : this.match(n) ? this.readOutputToken(e) : this.readHTMLToken([r, n]);
  }
  readHTMLToken(e) {
    const r = this.p;
    for (; this.p < this.N && !e.some((n) => this.match(n)); )
      ++this.p;
    return new dr(this.input, r, this.p, this.file);
  }
  readTagToken(e) {
    const { file: r, input: n } = this, i = this.p;
    if (this.readToDelimiter(e.tagDelimiterRight) === -1)
      throw this.error(`tag ${this.snapshot(i)} not closed`, i);
    const s = new pr(n, i, this.p, e, r);
    return s.name === "raw" && (this.rawBeginAt = i), s;
  }
  readToDelimiter(e, r = !1) {
    for (this.skipBlank(); this.p < this.N; ) {
      if (r && this.peekType() & ar) {
        this.readQuoted();
        continue;
      }
      if (++this.p, this.rmatch(e))
        return this.p;
    }
    return -1;
  }
  readOutputToken(e = de) {
    const { file: r, input: n } = this, { outputDelimiterRight: i } = e, s = this.p;
    if (this.readToDelimiter(i, !0) === -1)
      throw this.error(`output ${this.snapshot(s)} not closed`, s);
    return new ls(n, s, this.p, e, r);
  }
  readEndrawOrRawContent(e) {
    const { tagDelimiterLeft: r, tagDelimiterRight: n } = e, i = this.p;
    let s = this.readTo(r) - r.length;
    for (; this.p < this.N; ) {
      if (this.readIdentifier().getText() !== "endraw") {
        s = this.readTo(r) - r.length;
        continue;
      }
      for (; this.p <= this.N; ) {
        if (this.rmatch(n)) {
          const o = this.p;
          return i === s ? (this.rawBeginAt = -1, new pr(this.input, i, o, e, this.file)) : (this.p = s, new dr(this.input, i, s, this.file));
        }
        if (this.rmatch(r))
          break;
        this.p++;
      }
    }
    throw this.error(`raw ${this.snapshot(this.rawBeginAt)} not closed`, i);
  }
  readLiquidTagTokens(e = de) {
    const r = [];
    for (; this.p < this.N; ) {
      const n = this.readLiquidTagToken(e);
      n && r.push(n);
    }
    return r;
  }
  readLiquidTagToken(e) {
    if (this.skipBlank(), this.end())
      return;
    const r = this.p;
    this.readToDelimiter(`
`);
    const n = this.p;
    return new vs(this.input, r, n, e, this.file);
  }
  error(e, r = this.p) {
    return new zi(e, new lt(this.input, r, this.N, this.file));
  }
  assert(e, r, n) {
    if (!e)
      throw this.error(typeof r == "function" ? r() : r, n);
  }
  snapshot(e = this.p) {
    return JSON.stringify(ji(this.input.slice(e, this.N), 32));
  }
  /**
   * @deprecated use #readIdentifier instead
   */
  readWord() {
    return this.readIdentifier();
  }
  readIdentifier() {
    this.skipBlank();
    const e = this.p;
    for (; !this.end() && Be(this.peek()); )
      ++this.p;
    return new lt(this.input, e, this.p, this.file);
  }
  readNonEmptyIdentifier() {
    const e = this.readIdentifier();
    return e.size() ? e : void 0;
  }
  readTagName() {
    return this.skipBlank(), this.input[this.p] === "#" ? this.input.slice(this.p, ++this.p) : this.readIdentifier().getText();
  }
  readHashes(e) {
    const r = [];
    for (; ; ) {
      const n = this.readHash(e);
      if (!n)
        return r;
      r.push(n);
    }
  }
  readHash(e) {
    this.skipBlank(), this.peek() === "," && ++this.p;
    const r = this.p, n = this.readNonEmptyIdentifier();
    if (!n)
      return;
    let i;
    this.skipBlank();
    const s = j(e) ? e : e ? "=" : ":";
    return this.peek() === s && (++this.p, i = this.readValue()), new ds(this.input, r, this.p, n, i, this.file);
  }
  remaining() {
    return this.input.slice(this.p, this.N);
  }
  advance(e = 1) {
    this.p += e;
  }
  end() {
    return this.p >= this.N;
  }
  read() {
    return this.input[this.p++];
  }
  readTo(e) {
    for (; this.p < this.N; )
      if (++this.p, this.rmatch(e))
        return this.p;
    return -1;
  }
  readValue() {
    this.skipBlank();
    const e = this.p, r = this.readLiteral() || this.readQuoted() || this.readRange() || this.readNumber(), n = this.readProperties(!r);
    return n.length ? new gr(r, n, this.input, e, this.p) : r;
  }
  readScopeValue() {
    this.skipBlank();
    const e = this.p, r = this.readProperties();
    if (r.length)
      return new gr(void 0, r, this.input, e, this.p);
  }
  readProperties(e = !0) {
    const r = [];
    for (; ; ) {
      if (this.peek() === "[") {
        this.p++;
        const n = this.readValue() || new lt(this.input, this.p, this.p, this.file);
        this.assert(this.readTo("]") !== -1, "[ not closed"), r.push(n);
        continue;
      }
      if (e && !r.length) {
        const n = this.readNonEmptyIdentifier();
        if (n) {
          r.push(n);
          continue;
        }
      }
      if (this.peek() === "." && this.peek(1) !== ".") {
        this.p++;
        const n = this.readNonEmptyIdentifier();
        if (!n)
          break;
        r.push(n);
        continue;
      }
      break;
    }
    return r;
  }
  readNumber() {
    this.skipBlank();
    let e = !1, r = !1, n = 0;
    for (this.peekType() & Ji && n++; this.p + n <= this.N; )
      if (this.peekType(n) & Yi)
        r = !0, n++;
      else if (this.peek(n) === "." && this.peek(n + 1) !== ".") {
        if (e || !r)
          return;
        e = !0, n++;
      } else
        break;
    if (r && !Be(this.peek(n))) {
      const i = new cs(this.input, this.p, this.p + n, this.file);
      return this.advance(n), i;
    }
  }
  readLiteral() {
    this.skipBlank();
    const e = this.matchTrie(this.literalTrie);
    if (e === -1)
      return;
    const r = new fs(this.input, this.p, e, this.file);
    return this.p = e, r;
  }
  readRange() {
    this.skipBlank();
    const e = this.p;
    if (this.peek() !== "(")
      return;
    ++this.p;
    const r = this.readValueOrThrow();
    this.p += 2;
    const n = this.readValueOrThrow();
    return ++this.p, new ws(this.input, e, this.p, r, n, this.file);
  }
  readValueOrThrow() {
    const e = this.readValue();
    return this.assert(e, () => `unexpected token ${this.snapshot()}, value expected`), e;
  }
  readQuoted() {
    this.skipBlank();
    const e = this.p;
    if (!(this.peekType() & ar))
      return;
    ++this.p;
    let r = !1;
    for (; this.p < this.N && (++this.p, !(this.input[this.p - 1] === this.input[e] && !r)); )
      r ? r = !1 : this.input[this.p - 1] === "\\" && (r = !0);
    return new ys(this.input, e, this.p, this.file);
  }
  *readFileNameTemplate(e) {
    const { outputDelimiterLeft: r } = e, n = [",", " ", r], i = new Set(n);
    for (; this.p < this.N && !i.has(this.peek()); )
      yield this.match(r) ? this.readOutputToken(e) : this.readHTMLToken(n);
  }
  match(e) {
    for (let r = 0; r < e.length; r++)
      if (e[r] !== this.input[this.p + r])
        return !1;
    return !0;
  }
  rmatch(e) {
    for (let r = 0; r < e.length; r++)
      if (e[e.length - 1 - r] !== this.input[this.p - 1 - r])
        return !1;
    return !0;
  }
  peekType(e = 0) {
    return this.p + e >= this.N ? 0 : y[this.input.charCodeAt(this.p + e)];
  }
  peek(e = 0) {
    return this.p + e >= this.N ? "" : this.input[this.p + e];
  }
  skipBlank() {
    for (; this.peekType() & ve; )
      ++this.p;
  }
}
class so {
  constructor(e, r) {
    this.handlers = {}, this.stopRequested = !1, this.tokens = e, this.parseToken = r;
  }
  on(e, r) {
    return this.handlers[e] = r, this;
  }
  trigger(e, r) {
    const n = this.handlers[e];
    return n ? (n.call(this, r), !0) : !1;
  }
  start() {
    this.trigger("start");
    let e;
    for (; !this.stopRequested && (e = this.tokens.shift()); ) {
      if (this.trigger("token", e) || he(e) && this.trigger(`tag:${e.name}`, e))
        continue;
      const r = this.parseToken(e, this.tokens);
      this.trigger("template", r);
    }
    return this.stopRequested || this.trigger("end"), this;
  }
  stop() {
    return this.stopRequested = !0, this;
  }
}
class Vt {
  constructor(e) {
    this.token = e;
  }
}
class T extends Vt {
  constructor(e, r, n) {
    super(e), this.name = e.name, this.liquid = n, this.tokenizer = e.tokenizer;
  }
}
class Oe {
  constructor(e, r) {
    this.hash = {};
    const n = new Q(e, {});
    for (const i of n.readHashes(r))
      this.hash[i.name.content] = i.value;
  }
  *render(e) {
    const r = {};
    for (const n of Object.keys(this.hash))
      r[n] = this.hash[n] === void 0 ? !0 : yield S(this.hash[n], e);
    return r;
  }
}
function oo(t) {
  return class extends T {
    constructor(e, r, n) {
      super(e, r, n), x(t.parse) && t.parse.call(this, e, r);
    }
    *render(e, r) {
      const n = yield new Oe(this.token.args, e.opts.keyValueSeparator).render(e);
      return yield t.render.call(this, e, r, n);
    }
  };
}
function ao(t) {
  return A(t);
}
class Nn {
  constructor(e, r, n) {
    this.token = e, this.name = e.name, this.handler = x(r) ? r : x(r == null ? void 0 : r.handler) ? r.handler : Tn, this.raw = !x(r) && !!(r != null && r.raw), this.args = e.args, this.liquid = n;
  }
  *render(e, r) {
    const n = [];
    for (const i of this.args)
      ao(i) ? n.push([i[0], yield S(i[1], r)]) : n.push(yield S(i, r));
    return yield this.handler.apply({ context: r, token: this.token, liquid: this.liquid }, [e, ...n]);
  }
}
class M {
  /**
   * @param str the value to be valuated, eg.: "foobar" | truncate: 3
   */
  constructor(e, r) {
    this.filters = [];
    const n = typeof e == "string" ? new Q(e, r.options.operators).readFilteredValue() : e;
    this.initial = n.initial, this.filters = n.filters.map((i) => new Nn(i, this.getFilter(r, i.name), r));
  }
  *value(e, r) {
    r = r || e.opts.lenientIf && this.filters.length > 0 && this.filters[0].name === "default";
    let n = yield this.initial.evaluate(e, r);
    for (const i of this.filters)
      n = yield i.render(n, e);
    return n;
  }
  getFilter(e, r) {
    const n = e.filters[r];
    return R(n || !e.options.strictFilters, () => `undefined filter: ${r}`), n;
  }
}
class uo extends Vt {
  constructor(e, r) {
    var n;
    super(e);
    const i = new Q(e.input, r.options.operators, e.file, e.contentRange);
    this.value = new M(i.readFilteredValue(), r);
    const s = this.value.filters, o = r.options.outputEscape;
    if (!(!((n = s[s.length - 1]) === null || n === void 0) && n.raw) && o) {
      const a = new Dn(toString.call(o), [], "", 0, 0);
      s.push(new Nn(a, o, r));
    }
  }
  *render(e, r) {
    const n = yield this.value.value(e, !1);
    r.write(n);
  }
}
class lo extends Vt {
  constructor(e) {
    super(e), this.str = e.getContent();
  }
  *render(e, r) {
    r.write(this.str);
  }
}
var ue;
(function(t) {
  t.Partials = "partials", t.Layouts = "layouts", t.Root = "root";
})(ue || (ue = {}));
class co {
  constructor(e) {
    if (this.options = e, e.relativeReference) {
      const r = e.fs.sep;
      R(r, "`fs.sep` is required for relative reference");
      const n = new RegExp(["." + r, ".." + r, "./", "../"].map((i) => Pi(i)).join("|"));
      this.shouldLoadRelative = (i) => n.test(i);
    } else
      this.shouldLoadRelative = (r) => !1;
    this.contains = this.options.fs.contains || (() => !0);
  }
  *lookup(e, r, n, i) {
    const { fs: s } = this.options, o = this.options[r];
    for (const a of this.candidates(e, o, i, r !== ue.Root))
      if (n ? s.existsSync(a) : yield s.exists(a))
        return a;
    throw this.lookupError(e, o);
  }
  *candidates(e, r, n, i) {
    const { fs: s, extname: o } = this.options;
    if (this.shouldLoadRelative(e) && n) {
      const a = s.resolve(this.dirname(n), e, o);
      for (const l of r)
        if (!i || this.contains(l, a)) {
          yield a;
          break;
        }
    }
    for (const a of r) {
      const l = s.resolve(a, e, o);
      (!i || this.contains(a, l)) && (yield l);
    }
    if (s.fallback !== void 0) {
      const a = s.fallback(e);
      a !== void 0 && (yield a);
    }
  }
  dirname(e) {
    const r = this.options.fs;
    return R(r.dirname, "`fs.dirname` is required for relative reference"), r.dirname(e);
  }
  lookupError(e, r) {
    const n = new Error("ENOENT");
    return n.message = `ENOENT: Failed to lookup "${e}" in "${r}"`, n.code = "ENOENT", n;
  }
}
class te {
  constructor(e) {
    this.liquid = e, this.cache = this.liquid.options.cache, this.fs = this.liquid.options.fs, this.parseFile = this.cache ? this._parseFileCached : this._parseFile, this.loader = new co(this.liquid.options), this.parseLimit = new yt("parse length", e.options.parseLimit);
  }
  parse(e, r) {
    e = String(e), this.parseLimit.use(e.length);
    const i = new Q(e, this.liquid.options.operators, r).readTopLevelTokens(this.liquid.options);
    return this.parseTokens(i);
  }
  parseTokens(e) {
    let r;
    const n = [], i = [];
    for (; r = e.shift(); )
      try {
        n.push(this.parseToken(r, e));
      } catch (s) {
        if (this.liquid.options.catchAllErrors)
          i.push(s);
        else
          throw s;
      }
    if (i.length)
      throw new On(i);
    return n;
  }
  parseToken(e, r) {
    try {
      if (he(e)) {
        const n = this.liquid.tags[e.name];
        return R(n, `tag "${e.name}" not found`), new n(e, r, this.liquid, this);
      }
      return ho(e) ? new uo(e, this.liquid) : new lo(e);
    } catch (n) {
      throw fe.is(n) ? n : new Mi(n, e);
    }
  }
  parseStream(e) {
    return new so(e, (r, n) => this.parseToken(r, n));
  }
  *_parseFileCached(e, r, n = ue.Root, i) {
    const s = this.cache, o = this.loader.shouldLoadRelative(e) ? i + "," + e : n + ":" + e, a = yield s.read(o);
    if (a)
      return a;
    const l = this._parseFile(e, r, n, i), c = r ? yield l : Z(l);
    s.write(o, c);
    try {
      return yield c;
    } catch (f) {
      throw s.remove(o), f;
    }
  }
  *_parseFile(e, r, n = ue.Root, i) {
    const s = yield this.loader.lookup(e, n, r, i);
    return this.parse(r ? this.fs.readFileSync(s) : yield this.fs.readFile(s), s);
  }
}
var b;
(function(t) {
  t[t.Number = 1] = "Number", t[t.Literal = 2] = "Literal", t[t.Tag = 4] = "Tag", t[t.Output = 8] = "Output", t[t.HTML = 16] = "HTML", t[t.Filter = 32] = "Filter", t[t.Hash = 64] = "Hash", t[t.PropertyAccess = 128] = "PropertyAccess", t[t.Word = 256] = "Word", t[t.Range = 512] = "Range", t[t.Quoted = 1024] = "Quoted", t[t.Operator = 2048] = "Operator", t[t.FilteredValue = 4096] = "FilteredValue", t[t.Delimited = 12] = "Delimited";
})(b || (b = {}));
function fo(t) {
  return !!(K(t) & b.Delimited);
}
function $n(t) {
  return K(t) === b.Operator;
}
function Bt(t) {
  return K(t) === b.HTML;
}
function ho(t) {
  return K(t) === b.Output;
}
function he(t) {
  return K(t) === b.Tag;
}
function po(t) {
  return K(t) === b.Quoted;
}
function mo(t) {
  return K(t) === b.PropertyAccess;
}
function go(t) {
  return K(t) === b.Range;
}
function K(t) {
  return t ? t.kind : -1;
}
class ie {
  constructor(e = {}, r = de, n = {}, { memoryLimit: i, renderLimit: s } = {}) {
    var o, a, l, c, f;
    this.scopes = [{}], this.registers = {}, this.sync = !!n.sync, this.opts = r, this.globals = (o = n.globals) !== null && o !== void 0 ? o : r.globals, this.environments = nt(e) ? e : Object(e), this.strictVariables = (a = n.strictVariables) !== null && a !== void 0 ? a : this.opts.strictVariables, this.ownPropertyOnly = (l = n.ownPropertyOnly) !== null && l !== void 0 ? l : r.ownPropertyOnly, this.memoryLimit = i ?? new yt("memory alloc", (c = n.memoryLimit) !== null && c !== void 0 ? c : r.memoryLimit), this.renderLimit = s ?? new yt("template render", _n().now() + ((f = n.renderLimit) !== null && f !== void 0 ? f : r.renderLimit));
  }
  getRegister(e) {
    return this.registers[e] = this.registers[e] || {};
  }
  setRegister(e, r) {
    return this.registers[e] = r;
  }
  saveRegister(...e) {
    return e.map((r) => [r, this.getRegister(r)]);
  }
  restoreRegister(e) {
    return e.forEach(([r, n]) => this.setRegister(r, n));
  }
  getAll() {
    return [this.globals, this.environments, ...this.scopes].reduce((e, r) => Qe(e, r), {});
  }
  /**
   * @deprecated use `_get()` or `getSync()` instead
   */
  get(e) {
    return this.getSync(e);
  }
  getSync(e) {
    return G(this._get(e));
  }
  *_get(e) {
    const r = this.findScope(e[0]);
    return yield this._getFromScope(r, e);
  }
  /**
   * @deprecated use `_get()` instead
   */
  getFromScope(e, r) {
    return G(this._getFromScope(e, r));
  }
  *_getFromScope(e, r, n = this.strictVariables) {
    j(r) && (r = r.split("."));
    for (let i = 0; i < r.length; i++)
      if (e = yield yo(e, r[i], this.ownPropertyOnly), n && Ei(e))
        throw new Ui(r.slice(0, i + 1).join("."));
    return e;
  }
  push(e) {
    return this.scopes.push(e);
  }
  pop() {
    return this.scopes.pop();
  }
  bottom() {
    return this.scopes[0];
  }
  spawn(e = {}) {
    return new ie(e, this.opts, {
      sync: this.sync,
      globals: this.globals,
      strictVariables: this.strictVariables
    }, {
      renderLimit: this.renderLimit,
      memoryLimit: this.memoryLimit
    });
  }
  findScope(e) {
    for (let r = this.scopes.length - 1; r >= 0; r--) {
      const n = this.scopes[r];
      if (e in n)
        return n;
    }
    return e in this.environments ? this.environments : this.globals;
  }
}
function yo(t, e, r) {
  if (t = yn(t), e = m(e), z(t))
    return t;
  if (A(t) && e < 0)
    return t[t.length + +e];
  const n = wo(t, e, r);
  return n === void 0 && t instanceof ce ? t.liquidMethodMissing(e) : x(n) ? n.call(t) : e === "size" ? To(t) : e === "first" ? vo(t) : e === "last" ? bo(t) : n;
}
function wo(t, e, r) {
  if (!(r && !It.call(t, e) && !(t instanceof ce)))
    return t[e];
}
function vo(t) {
  return A(t) ? t[0] : t.first;
}
function bo(t) {
  return A(t) ? t[t.length - 1] : t.last;
}
function To(t) {
  if (It.call(t, "size") || t.size !== void 0)
    return t.size;
  if (A(t) || j(t))
    return t.length;
  if (typeof t == "object")
    return Object.keys(t).length;
}
var oe;
(function(t) {
  t[t.OUTPUT = 0] = "OUTPUT", t[t.STORE = 1] = "STORE";
})(oe || (oe = {}));
const Oo = I(Math.abs), So = I(Math.max), ko = I(Math.min), xo = I(Math.ceil), Lo = I((t, e, r = !1) => r ? Math.floor(t / e) : t / e), Ao = I(Math.floor), Fo = I((t, e) => t - e), Do = I((t, e) => t % e), _o = I((t, e) => t * e);
function Ro(t, e = 0) {
  t = m(t), e = m(e);
  const r = Math.pow(10, e);
  return Math.round(t * r) / r;
}
function Po(t, e) {
  return t = m(t), e = m(e), Number(t) + Number(e);
}
var Eo = /* @__PURE__ */ Object.freeze({
  __proto__: null,
  abs: Oo,
  at_least: So,
  at_most: ko,
  ceil: xo,
  divided_by: Lo,
  floor: Ao,
  minus: Fo,
  modulo: Do,
  times: _o,
  round: Ro,
  plus: Po
});
const No = (t) => decodeURIComponent(h(t)).replace(/\+/g, " "), $o = (t) => encodeURIComponent(h(t)).replace(/%20/g, "+"), Io = (t) => encodeURIComponent(h(t)).replace(/%20/g, "+").replace(/[!'()*]/g, (e) => "%" + e.charCodeAt(0).toString(16).toUpperCase()), jo = (t) => encodeURI(h(t)).replace(/%5B/g, "[").replace(/%5D/g, "]"), Or = /[^\p{M}\p{L}\p{Nd}]+/ug, zo = {
  raw: /\s+/g,
  default: Or,
  pretty: /[^\p{M}\p{L}\p{Nd}._~!$&'()+,;=@]+/ug,
  ascii: /[^A-Za-z0-9]+/g,
  latin: Or,
  none: null
};
function Mo(t, e = "default", r = !1) {
  t = h(t);
  const n = zo[e];
  return n && (e === "latin" && (t = Co(t)), t = t.replace(n, "-").replace(/^-|-$/g, "")), r ? t : t.toLowerCase();
}
function Co(t) {
  return t.replace(/[àáâãäå]/g, "a").replace(/[æ]/g, "ae").replace(/[ç]/g, "c").replace(/[èéêë]/g, "e").replace(/[ìíîï]/g, "i").replace(/[ð]/g, "d").replace(/[ñ]/g, "n").replace(/[òóôõöø]/g, "o").replace(/[ùúûü]/g, "u").replace(/[ýÿ]/g, "y").replace(/[ß]/g, "ss").replace(/[œ]/g, "oe").replace(/[þ]/g, "th").replace(/[ẞ]/g, "SS").replace(/[Œ]/g, "OE").replace(/[Þ]/g, "TH");
}
var qo = /* @__PURE__ */ Object.freeze({
  __proto__: null,
  url_decode: No,
  url_encode: $o,
  cgi_escape: Io,
  uri_escape: jo,
  slugify: Mo
});
const Uo = I(function(t, e) {
  const r = L(t), n = z(e) ? " " : h(e), i = r.length * (1 + n.length);
  return this.context.memoryLimit.use(i), r.join(n);
}), Vo = I((t) => A(t) ? wn(t) : ""), Bo = I((t) => A(t) ? t[0] : ""), Ho = I(function(t) {
  const e = L(t);
  return this.context.memoryLimit.use(e.length), [...e].reverse();
});
function* Wo(t, e) {
  const r = [], n = L(t);
  this.context.memoryLimit.use(n.length);
  for (const i of n)
    r.push([
      i,
      e ? yield this.context._getFromScope(i, h(e).split("."), !1) : i
    ]);
  return r.sort((i, s) => {
    const o = i[1], a = s[1];
    return o < a ? -1 : o > a ? 1 : 0;
  }).map((i) => i[0]);
}
function Yo(t, e) {
  const r = h(e), n = e === void 0 ? sr : (s, o) => sr(s[r], o[r]), i = L(t);
  return this.context.memoryLimit.use(i.length), [...i].sort(n);
}
const Jo = (t) => t && t.length || 0;
function* Qo(t, e) {
  const r = [], n = L(t);
  this.context.memoryLimit.use(n.length);
  for (const i of n)
    r.push(yield this.context._getFromScope(i, h(e), !1));
  return r;
}
function* Zo(t, e) {
  let r = 0;
  const n = L(t);
  for (const i of n) {
    const s = Number(e ? yield this.context._getFromScope(i, h(e), !1) : i);
    r += Number.isNaN(s) ? 0 : s;
  }
  return r;
}
function Go(t) {
  const e = L(t);
  return this.context.memoryLimit.use(e.length), e.filter((r) => !z(m(r)));
}
function In(t, e = []) {
  const r = L(t), n = L(e);
  return this.context.memoryLimit.use(r.length + n.length), r.concat(n);
}
function Xo(t, e) {
  return In.call(this, t, [e]);
}
function Ko(t, e) {
  const r = L(t);
  this.context.memoryLimit.use(r.length);
  const n = [...r];
  return n.unshift(e), n;
}
function ea(t) {
  const e = [...L(t)];
  return e.pop(), e;
}
function ta(t) {
  const e = L(t);
  this.context.memoryLimit.use(e.length);
  const r = [...e];
  return r.shift(), r;
}
function ra(t, e, r = 1) {
  return t = m(t), z(t) ? [] : (A(t) || (t = h(t)), e = e < 0 ? t.length + e : e, this.context.memoryLimit.use(r), t.slice(e, e + r));
}
function* na(t, e, r) {
  const n = [];
  t = L(t), this.context.memoryLimit.use(t.length);
  const i = new Q(h(e)).readScopeValue();
  for (const o of t)
    n.push(yield S(i, this.context.spawn(o)));
  const s = this.context.opts.jekyllWhere ? (o) => be.is(r) ? H(o, r) : A(o) ? Ps(o, r) : H(o, r) : (o) => H(o, r);
  return t.filter((o, a) => r === void 0 ? se(n[a], this.context) : s(n[a]));
}
function* ia(t, e, r) {
  const n = [], i = new M(h(r), this.liquid), s = L(t);
  this.context.memoryLimit.use(s.length);
  for (const o of s)
    (yield i.value(this.context.spawn({ [e]: o }))) && n.push(o);
  return n;
}
function* sa(t, e) {
  const r = /* @__PURE__ */ new Map();
  t = L(t);
  const n = new Q(h(e)).readScopeValue();
  this.context.memoryLimit.use(t.length);
  for (const i of t) {
    const s = yield S(n, this.context.spawn(i));
    r.has(s) || r.set(s, []), r.get(s).push(i);
  }
  return [...r.entries()].map(([i, s]) => ({ name: i, items: s }));
}
function* oa(t, e, r) {
  const n = /* @__PURE__ */ new Map(), i = new M(h(r), this.liquid);
  t = L(t), this.context.memoryLimit.use(t.length);
  for (const s of t) {
    const o = yield i.value(this.context.spawn({ [e]: s }));
    n.has(o) || n.set(o, []), n.get(o).push(s);
  }
  return [...n.entries()].map(([s, o]) => ({ name: s, items: o }));
}
function* aa(t, e, r) {
  const n = new Q(h(e)).readScopeValue(), i = L(t);
  for (const s of i) {
    const o = yield S(n, this.context.spawn(s));
    if (H(o, r))
      return s;
  }
}
function* ua(t, e, r) {
  const n = new M(h(r), this.liquid), i = L(t);
  for (const s of i)
    if (yield n.value(this.context.spawn({ [e]: s })))
      return s;
}
function la(t) {
  return t = L(t), this.context.memoryLimit.use(t.length), [...new Set(t)];
}
function ca(t, e = 1) {
  if (t = m(t), z(t))
    return [];
  A(t) || (t = h(t)), this.context.memoryLimit.use(e);
  const r = [...t].sort(() => Math.random() - 0.5);
  return e === 1 ? r[0] : r.slice(0, e);
}
var fa = /* @__PURE__ */ Object.freeze({
  __proto__: null,
  join: Uo,
  last: Vo,
  first: Bo,
  reverse: Ho,
  sort: Wo,
  sort_natural: Yo,
  size: Jo,
  map: Qo,
  sum: Zo,
  compact: Go,
  concat: In,
  push: Xo,
  unshift: Ko,
  pop: ea,
  shift: ta,
  slice: ra,
  where: na,
  where_exp: ia,
  group_by: sa,
  group_by_exp: oa,
  find: aa,
  find_exp: ua,
  uniq: la,
  sample: ca
});
function Ht(t, e, r) {
  var n, i, s;
  const o = ((n = t == null ? void 0 : t.length) !== null && n !== void 0 ? n : 0) + ((i = e == null ? void 0 : e.length) !== null && i !== void 0 ? i : 0) + ((s = r == null ? void 0 : r.length) !== null && s !== void 0 ? s : 0);
  this.context.memoryLimit.use(o);
  const a = zn(t, this.context.opts, r);
  return a ? (e = m(e), e = z(e) ? this.context.opts.dateFormat : h(e), He(a, e)) : t;
}
function ha(t) {
  return Ht.call(this, t, "%Y-%m-%dT%H:%M:%S%:z");
}
function pa(t) {
  return Ht.call(this, t, "%a, %d %b %Y %H:%M:%S %z");
}
function da(t, e, r) {
  return jn.call(this, t, "%b", e, r);
}
function ma(t, e, r) {
  return jn.call(this, t, "%B", e, r);
}
function jn(t, e, r, n) {
  const i = zn(t, this.context.opts);
  if (!i)
    return t;
  if (r === "ordinal") {
    const s = i.getDate();
    return n === "US" ? He(i, `${e} ${s}%q, %Y`) : He(i, `${s}%q ${e} %Y`);
  }
  return He(i, `%d ${e} %Y`);
}
function zn(t, e, r) {
  let n;
  const i = r ?? e.timezoneOffset, s = e.locale;
  return t = m(t), t === "now" || t === "today" ? n = new U(Date.now(), s, i) : zt(t) ? n = new U(t * 1e3, s, i) : j(t) ? /^\d+$/.test(t) ? n = new U(+t * 1e3, s, i) : e.preserveTimezones && r === void 0 ? n = U.createDateFixedToTimezone(t, s) : n = new U(t, s, i) : n = new U(t, s, i), n.valid() ? n : void 0;
}
var ga = /* @__PURE__ */ Object.freeze({
  __proto__: null,
  date: Ht,
  date_to_xmlschema: ha,
  date_to_rfc822: pa,
  date_to_string: da,
  date_to_long_string: ma
});
const ft = /[\u4E00-\u9FFF\uF900-\uFAFF\u3400-\u4DBF\u3040-\u309F\u30A0-\u30FF\uAC00-\uD7AF]/gu, Sr = /[^\u4E00-\u9FFF\uF900-\uFAFF\u3400-\u4DBF\u3040-\u309F\u30A0-\u30FF\uAC00-\uD7AF\s]+/gu;
function ya(t, e) {
  R(arguments.length === 2, "append expect 2 arguments");
  const r = h(t), n = h(e);
  return this.context.memoryLimit.use(r.length + n.length), r + n;
}
function wa(t, e) {
  R(arguments.length === 2, "prepend expect 2 arguments");
  const r = h(t), n = h(e);
  return this.context.memoryLimit.use(r.length + n.length), n + r;
}
function va(t, e) {
  const r = h(t);
  return this.context.memoryLimit.use(r.length), e ? (e = Mt(h(e)), r.replace(new RegExp(`^[${e}]+`, "g"), "")) : r.replace(/^\s+/, "");
}
function ba(t) {
  const e = h(t);
  return this.context.memoryLimit.use(e.length), e.toLowerCase();
}
function Ta(t) {
  const e = h(t);
  return this.context.memoryLimit.use(e.length), h(e).toUpperCase();
}
function Oa(t, e) {
  const r = h(t);
  return this.context.memoryLimit.use(r.length), r.split(h(e)).join("");
}
function Sa(t, e) {
  const r = h(t);
  return this.context.memoryLimit.use(r.length), r.replace(h(e), "");
}
function ka(t, e) {
  const r = h(t);
  this.context.memoryLimit.use(r.length);
  const n = h(e), i = r.lastIndexOf(n);
  return i === -1 ? r : r.substring(0, i) + r.substring(i + n.length);
}
function xa(t, e) {
  return t = h(t), this.context.memoryLimit.use(t.length), e ? (e = Mt(h(e)), t.replace(new RegExp(`[${e}]+$`, "g"), "")) : t.replace(/\s+$/, "");
}
function La(t, e) {
  const r = h(t);
  this.context.memoryLimit.use(r.length);
  const n = r.split(h(e));
  for (; n.length && n[n.length - 1] === ""; )
    n.pop();
  return n;
}
function Aa(t, e) {
  const r = h(t);
  return this.context.memoryLimit.use(r.length), e ? (e = Mt(h(e)), r.replace(new RegExp(`^[${e}]+`, "g"), "").replace(new RegExp(`[${e}]+$`, "g"), "")) : r.trim();
}
function Fa(t) {
  const e = h(t);
  return this.context.memoryLimit.use(e.length), e.replace(/\r?\n/gm, "");
}
function Da(t) {
  return t = h(t), this.context.memoryLimit.use(t.length), t.charAt(0).toUpperCase() + t.slice(1).toLowerCase();
}
function _a(t, e, r) {
  const n = h(t);
  return this.context.memoryLimit.use(n.length), n.split(h(e)).join(r);
}
function Ra(t, e, r) {
  const n = h(t);
  return this.context.memoryLimit.use(n.length), n.replace(h(e), r);
}
function Pa(t, e, r) {
  const n = h(t);
  this.context.memoryLimit.use(n.length);
  const i = h(e), s = n.lastIndexOf(i);
  if (s === -1)
    return n;
  const o = h(r);
  return n.substring(0, s) + o + n.substring(s + i.length);
}
function Ea(t, e = 50, r = "...") {
  const n = h(t);
  return this.context.memoryLimit.use(n.length), n.length <= e ? t : n.substring(0, e - r.length) + r;
}
function Na(t, e = 15, r = "...") {
  const n = h(t);
  this.context.memoryLimit.use(n.length);
  const i = n.split(/\s+/);
  e <= 0 && (e = 1);
  let s = i.slice(0, e).join(" ");
  return i.length >= e && (s += r), s;
}
function $a(t) {
  const e = h(t);
  return this.context.memoryLimit.use(e.length), e.replace(/\s+/g, " ");
}
function Ia(t, e) {
  const r = h(t);
  if (this.context.memoryLimit.use(r.length), t = r.trim(), !t)
    return 0;
  switch (e) {
    case "cjk":
      return (t.match(ft) || []).length + (t.match(Sr) || []).length;
    case "auto":
      return ft.test(t) ? t.match(ft).length + (t.match(Sr) || []).length : t.split(/\s+/).length;
    default:
      return t.split(/\s+/).length;
  }
}
function ja(t, e = "and") {
  switch (this.context.memoryLimit.use(t.length), t.length) {
    case 0:
      return "";
    case 1:
      return t[0];
    case 2:
      return `${t[0]} ${e} ${t[1]}`;
    default:
      return `${t.slice(0, -1).join(", ")}, ${e} ${t[t.length - 1]}`;
  }
}
var za = /* @__PURE__ */ Object.freeze({
  __proto__: null,
  append: ya,
  prepend: wa,
  lstrip: va,
  downcase: ba,
  upcase: Ta,
  remove: Oa,
  remove_first: Sa,
  remove_last: ka,
  rstrip: xa,
  split: La,
  strip: Aa,
  strip_newlines: Fa,
  capitalize: Da,
  replace: _a,
  replace_first: Ra,
  replace_last: Pa,
  truncate: Ea,
  truncatewords: Na,
  normalize_whitespace: $a,
  number_of_words: Ia,
  array_to_sentence_string: ja
});
const Ma = Object.assign(Object.assign(Object.assign(Object.assign(Object.assign(Object.assign(Object.assign({}, Xs), Eo), qo), fa), ga), za), En);
class Ca extends T {
  constructor(e, r, n) {
    super(e, r, n), this.key = this.tokenizer.readIdentifier().content, this.tokenizer.assert(this.key, "expected variable name"), this.tokenizer.skipBlank(), this.tokenizer.assert(this.tokenizer.peek() === "=", 'expected "="'), this.tokenizer.advance(), this.value = new M(this.tokenizer.readFilteredValue(), this.liquid);
  }
  *render(e) {
    e.bottom()[this.key] = yield this.value.value(e, this.liquid.options.lenientIf);
  }
}
const kr = ["offset", "limit", "reversed"];
class qa extends T {
  constructor(e, r, n, i) {
    super(e, r, n);
    const s = this.tokenizer.readIdentifier(), o = this.tokenizer.readIdentifier(), a = this.tokenizer.readValue();
    if (!s.size() || o.content !== "in" || !a)
      throw new Error(`illegal tag: ${e.getText()}`);
    this.variable = s.content, this.collection = a, this.hash = new Oe(this.tokenizer.remaining(), n.options.keyValueSeparator), this.templates = [], this.elseTemplates = [];
    let l;
    const c = i.parseStream(r).on("start", () => l = this.templates).on("tag:else", (f) => {
      Je(f.args), l = this.elseTemplates;
    }).on("tag:endfor", (f) => {
      Je(f.args), c.stop();
    }).on("template", (f) => l.push(f)).on("end", () => {
      throw new Error(`tag ${e.getText()} not closed`);
    });
    c.start();
  }
  *render(e, r) {
    const n = this.liquid.renderer;
    let i = jt(yield S(this.collection, e));
    if (!i.length) {
      yield n.renderTemplates(this.elseTemplates, e, r);
      return;
    }
    const s = "continue-" + this.variable + "-" + this.collection.getText();
    e.push({ continue: e.getRegister(s) });
    const o = yield this.hash.render(e);
    e.pop(), i = (this.liquid.options.orderedFilterParameters ? Object.keys(o).filter((c) => kr.includes(c)) : kr.filter((c) => o[c] !== void 0)).reduce((c, f) => f === "offset" ? Va(c, o.offset) : f === "limit" ? Ba(c, o.limit) : Ua(c), i), e.setRegister(s, (o.offset || 0) + i.length);
    const l = { forloop: new qt(i.length, this.collection.getText(), this.variable) };
    e.push(l);
    for (const c of i) {
      if (l[this.variable] = c, yield n.renderTemplates(this.templates, e, r), r.break) {
        r.break = !1;
        break;
      }
      r.continue = !1, l.forloop.next();
    }
    e.pop();
  }
}
function Ua(t) {
  return [...t].reverse();
}
function Va(t, e) {
  return t.slice(e);
}
function Ba(t, e) {
  return t.slice(0, e);
}
class Ha extends T {
  constructor(e, r, n, i) {
    for (super(e, r, n), this.templates = [], this.variable = this.readVariableName(); r.length; ) {
      const s = r.shift();
      if (he(s) && s.name === "endcapture")
        return;
      this.templates.push(i.parseToken(s, r));
    }
    throw new Error(`tag ${e.getText()} not closed`);
  }
  *render(e) {
    const n = yield this.liquid.renderer.renderTemplates(this.templates, e);
    e.bottom()[this.variable] = n;
  }
  readVariableName() {
    const e = this.tokenizer.readIdentifier().content;
    if (e)
      return e;
    const r = this.tokenizer.readQuoted();
    if (r)
      return Rn(r);
    throw this.tokenizer.error("invalid capture name");
  }
}
class Wa extends T {
  constructor(e, r, n, i) {
    super(e, r, n), this.branches = [], this.elseTemplates = [], this.value = new M(this.tokenizer.readFilteredValue(), this.liquid), this.elseTemplates = [];
    let s = [], o = 0;
    const a = i.parseStream(r).on("tag:when", (l) => {
      if (o > 0)
        return;
      s = [];
      const c = [];
      for (; !l.tokenizer.end(); )
        c.push(l.tokenizer.readValueOrThrow()), l.tokenizer.skipBlank(), l.tokenizer.peek() === "," ? l.tokenizer.readTo(",") : l.tokenizer.readTo("or");
      this.branches.push({
        values: c,
        templates: s
      });
    }).on("tag:else", () => {
      o++, s = this.elseTemplates;
    }).on("tag:endcase", () => a.stop()).on("template", (l) => {
      (s !== this.elseTemplates || o === 1) && s.push(l);
    }).on("end", () => {
      throw new Error(`tag ${e.getText()} not closed`);
    });
    a.start();
  }
  *render(e, r) {
    const n = this.liquid.renderer, i = m(yield this.value.value(e, e.opts.lenientIf));
    let s = !1;
    for (const o of this.branches)
      for (const a of o.values) {
        const l = yield S(a, e, e.opts.lenientIf);
        if (H(i, l)) {
          yield n.renderTemplates(o.templates, e, r), s = !0;
          break;
        }
      }
    s || (yield n.renderTemplates(this.elseTemplates, e, r));
  }
}
class Ya extends T {
  constructor(e, r, n) {
    for (super(e, r, n); r.length; ) {
      const i = r.shift();
      if (he(i) && i.name === "endcomment")
        return;
    }
    throw new Error(`tag ${e.getText()} not closed`);
  }
  render() {
  }
}
class Ja extends T {
  constructor(e, r, n, i) {
    super(e, r, n);
    const s = this.tokenizer;
    for (this.file = Wt(s, this.liquid, i), this.currentFile = e.file; !s.end(); ) {
      s.skipBlank();
      const o = s.p, a = s.readIdentifier();
      if ((a.content === "with" || a.content === "for") && (s.skipBlank(), s.peek() !== ":")) {
        const l = s.readValue();
        if (l) {
          const c = s.p, f = s.readIdentifier();
          let d;
          f.content === "as" ? d = s.readIdentifier() : s.p = c, this[a.content] = { value: l, alias: d && d.content }, s.skipBlank(), s.peek() === "," && s.advance();
          continue;
        }
      }
      s.p = o;
      break;
    }
    this.hash = new Oe(s.remaining(), n.options.keyValueSeparator);
  }
  *render(e, r) {
    const { liquid: n, hash: i } = this, s = yield Yt(this.file, e, n);
    R(s, () => `illegal file path "${s}"`);
    const o = e.spawn(), a = o.bottom();
    if (Qe(a, yield i.render(e)), this.with) {
      const { value: l, alias: c } = this.with;
      a[c || s] = yield S(l, e);
    }
    if (this.for) {
      const { value: l, alias: c } = this.for, f = jt(yield S(l, e));
      a.forloop = new qt(f.length, l.getText(), c);
      for (const d of f) {
        a[c] = d;
        const g = yield n._parsePartialFile(s, o.sync, this.currentFile);
        yield n.renderer.renderTemplates(g, o, r), a.forloop.next();
      }
    } else {
      const l = yield n._parsePartialFile(s, o.sync, this.currentFile);
      yield n.renderer.renderTemplates(l, o, r);
    }
  }
}
function Wt(t, e, r) {
  if (e.options.dynamicPartials) {
    const s = t.readValue();
    if (t.assert(s, "illegal file path"), s.getText() === "none")
      return;
    if (po(s)) {
      const o = r.parse(Rn(s));
      return xr(o);
    }
    return s;
  }
  const n = [...t.readFileNameTemplate(e.options)], i = xr(r.parseTokens(n));
  return i === "none" ? void 0 : i;
}
function xr(t) {
  return t.length === 1 && Bt(t[0].token) ? t[0].token.getContent() : t;
}
function* Yt(t, e, r) {
  return typeof t == "string" ? t : Array.isArray(t) ? r.renderer.renderTemplates(t, e) : yield S(t, e);
}
class Qa extends T {
  constructor(e, r, n, i) {
    super(e, r, n);
    const { tokenizer: s } = e;
    this.file = Wt(s, this.liquid, i), this.currentFile = e.file;
    const o = s.p;
    s.readIdentifier().content === "with" ? (s.skipBlank(), s.peek() !== ":" ? this.withVar = s.readValue() : s.p = o) : s.p = o, this.hash = new Oe(s.remaining(), n.options.jekyllInclude || n.options.keyValueSeparator);
  }
  *render(e, r) {
    const { liquid: n, hash: i, withVar: s } = this, { renderer: o } = n, a = yield Yt(this.file, e, n);
    R(a, () => `illegal file path "${a}"`);
    const l = e.saveRegister("blocks", "blockMode");
    e.setRegister("blocks", {}), e.setRegister("blockMode", oe.OUTPUT);
    const c = yield i.render(e);
    s && (c[a] = yield S(s, e));
    const f = yield n._parsePartialFile(a, e.sync, this.currentFile);
    e.push(e.opts.jekyllInclude ? { include: c } : c), yield o.renderTemplates(f, e, r), e.pop(), e.restoreRegister(l);
  }
}
class Za extends T {
  constructor(e, r, n) {
    super(e, r, n), this.variable = this.tokenizer.readIdentifier().content;
  }
  render(e, r) {
    const n = e.environments;
    zt(n[this.variable]) || (n[this.variable] = 0), r.write(h(--n[this.variable]));
  }
}
class Ga extends T {
  constructor(e, r, n) {
    super(e, r, n), this.candidates = [];
    const i = this.tokenizer.readValue();
    for (this.tokenizer.skipBlank(), i && (this.tokenizer.peek() === ":" ? (this.group = i, this.tokenizer.advance()) : this.candidates.push(i)); !this.tokenizer.end(); ) {
      const s = this.tokenizer.readValue();
      s && this.candidates.push(s), this.tokenizer.readTo(",");
    }
    this.tokenizer.assert(this.candidates.length, () => `empty candidates: "${e.getText()}"`);
  }
  *render(e, r) {
    const i = `cycle:${yield S(this.group, e)}:` + this.candidates.join(","), s = e.getRegister("cycle");
    let o = s[i];
    o === void 0 && (o = s[i] = 0);
    const a = this.candidates[o];
    return o = (o + 1) % this.candidates.length, s[i] = o, yield S(a, e);
  }
}
class Xa extends T {
  constructor(e, r, n, i) {
    super(e, r, n), this.branches = [];
    let s = [];
    i.parseStream(r).on("start", () => this.branches.push({
      value: new M(e.args, this.liquid),
      templates: s = []
    })).on("tag:elsif", (o) => {
      R(!this.elseTemplates, "unexpected elsif after else"), this.branches.push({
        value: new M(o.args, this.liquid),
        templates: s = []
      });
    }).on("tag:else", (o) => {
      Je(o.args), R(!this.elseTemplates, "duplicated else"), s = this.elseTemplates = [];
    }).on("tag:endif", function(o) {
      Je(o.args), this.stop();
    }).on("template", (o) => s.push(o)).on("end", () => {
      throw new Error(`tag ${e.getText()} not closed`);
    }).start();
  }
  *render(e, r) {
    const n = this.liquid.renderer;
    for (const { value: i, templates: s } of this.branches) {
      const o = yield i.value(e, e.opts.lenientIf);
      if (se(o, e)) {
        yield n.renderTemplates(s, e, r);
        return;
      }
    }
    yield n.renderTemplates(this.elseTemplates || [], e, r);
  }
}
class Ka extends T {
  constructor(e, r, n) {
    super(e, r, n), this.variable = this.tokenizer.readIdentifier().content;
  }
  render(e, r) {
    const n = e.environments;
    zt(n[this.variable]) || (n[this.variable] = 0);
    const i = n[this.variable];
    n[this.variable]++, r.write(h(i));
  }
}
class eu extends T {
  constructor(e, r, n, i) {
    super(e, r, n), this.file = Wt(this.tokenizer, this.liquid, i), this.currentFile = e.file, this.args = new Oe(this.tokenizer.remaining(), n.options.keyValueSeparator), this.templates = i.parseTokens(r);
  }
  *render(e, r) {
    const { liquid: n, args: i, file: s } = this, { renderer: o } = n;
    if (s === void 0) {
      e.setRegister("blockMode", oe.OUTPUT), yield o.renderTemplates(this.templates, e, r);
      return;
    }
    const a = yield Yt(this.file, e, n);
    R(a, () => `illegal file path "${a}"`);
    const l = yield n._parseLayoutFile(a, e.sync, this.currentFile);
    e.setRegister("blockMode", oe.STORE);
    const c = yield o.renderTemplates(this.templates, e), f = e.getRegister("blocks");
    f[""] === void 0 && (f[""] = (d, g) => g.write(c)), e.setRegister("blockMode", oe.OUTPUT), e.push(yield i.render(e)), yield o.renderTemplates(l, e, r), e.pop();
  }
}
class tu extends T {
  constructor(e, r, n, i) {
    super(e, r, n), this.templates = [];
    const s = /\w+/.exec(e.args);
    for (this.block = s ? s[0] : ""; r.length; ) {
      const o = r.shift();
      if (he(o) && o.name === "endblock")
        return;
      const a = i.parseToken(o, r);
      this.templates.push(a);
    }
    throw new Error(`tag ${e.getText()} not closed`);
  }
  *render(e, r) {
    const n = this.getBlockRender(e);
    e.getRegister("blockMode") === oe.STORE ? e.getRegister("blocks")[this.block] = n : yield n(new ur(), r);
  }
  getBlockRender(e) {
    const { liquid: r, templates: n } = this, i = e.getRegister("blocks")[this.block], s = function* (o, a) {
      e.push({ block: o }), yield r.renderer.renderTemplates(n, e, a), e.pop();
    };
    return i ? (o, a) => i(new ur(() => s(o, a)), a) : s;
  }
}
class ru extends T {
  constructor(e, r, n) {
    for (super(e, r, n), this.tokens = []; r.length; ) {
      const i = r.shift();
      if (he(i) && i.name === "endraw")
        return;
      this.tokens.push(i);
    }
    throw new Error(`tag ${e.getText()} not closed`);
  }
  render() {
    return this.tokens.map((e) => e.getText()).join("");
  }
}
class nu extends qt {
  constructor(e, r, n, i) {
    super(e, n, i), this.length = e, this.cols = r;
  }
  row() {
    return Math.floor(this.i / this.cols) + 1;
  }
  col0() {
    return this.i % this.cols;
  }
  col() {
    return this.col0() + 1;
  }
  col_first() {
    return this.col0() === 0;
  }
  col_last() {
    return this.col() === this.cols;
  }
}
class iu extends T {
  constructor(e, r, n, i) {
    super(e, r, n);
    const s = this.tokenizer.readIdentifier();
    this.tokenizer.skipBlank();
    const o = this.tokenizer.readIdentifier(), a = this.tokenizer.readValue();
    if (o.content !== "in" || !a)
      throw new Error(`illegal tag: ${e.getText()}`);
    this.variable = s.content, this.collection = a, this.args = new Oe(this.tokenizer.remaining(), n.options.keyValueSeparator), this.templates = [];
    let l;
    const c = i.parseStream(r).on("start", () => l = this.templates).on("tag:endtablerow", () => c.stop()).on("template", (f) => l.push(f)).on("end", () => {
      throw new Error(`tag ${e.getText()} not closed`);
    });
    c.start();
  }
  *render(e, r) {
    let n = jt(yield S(this.collection, e));
    const i = yield this.args.render(e), s = i.offset || 0, o = i.limit === void 0 ? n.length : i.limit;
    n = n.slice(s, s + o);
    const a = i.cols || n.length, l = this.liquid.renderer, c = new nu(n.length, a, this.collection.getText(), this.variable), f = { tablerowloop: c };
    e.push(f);
    for (let d = 0; d < n.length; d++, c.next())
      f[this.variable] = n[d], c.col0() === 0 && (c.row() !== 1 && r.write("</tr>"), r.write(`<tr class="row${c.row()}">`)), r.write(`<td class="col${c.col()}">`), yield l.renderTemplates(this.templates, e, r), r.write("</td>");
    n.length && r.write("</tr>"), e.pop();
  }
}
class su extends T {
  constructor(e, r, n, i) {
    super(e, r, n), this.branches = [], this.elseTemplates = [];
    let s = [], o = 0;
    i.parseStream(r).on("start", () => this.branches.push({
      value: new M(e.args, this.liquid),
      test: it,
      templates: s = []
    })).on("tag:elsif", (a) => {
      if (o > 0) {
        s = [];
        return;
      }
      this.branches.push({
        value: new M(a.args, this.liquid),
        test: se,
        templates: s = []
      });
    }).on("tag:else", () => {
      o++, s = this.elseTemplates;
    }).on("tag:endunless", function() {
      this.stop();
    }).on("template", (a) => {
      (s !== this.elseTemplates || o === 1) && s.push(a);
    }).on("end", () => {
      throw new Error(`tag ${e.getText()} not closed`);
    }).start();
  }
  *render(e, r) {
    const n = this.liquid.renderer;
    for (const { value: i, test: s, templates: o } of this.branches) {
      const a = yield i.value(e, e.opts.lenientIf);
      if (s(a, e)) {
        yield n.renderTemplates(o, e, r);
        return;
      }
    }
    yield n.renderTemplates(this.elseTemplates, e, r);
  }
}
class ou extends T {
  render(e, r) {
    r.break = !0;
  }
}
class au extends T {
  render(e, r) {
    r.continue = !0;
  }
}
class uu extends T {
  constructor(e, r, n) {
    super(e, r, n), this.tokenizer.skipBlank(), this.tokenizer.end() || (this.value = new M(this.tokenizer.readFilteredValue(), this.liquid));
  }
  *render(e, r) {
    if (!this.value)
      return;
    const n = yield this.value.value(e, !1);
    r.write(n);
  }
}
class lu extends T {
  constructor(e, r, n, i) {
    super(e, r, n);
    const s = this.tokenizer.readLiquidTagTokens(this.liquid.options);
    this.templates = i.parseTokens(s);
  }
  *render(e, r) {
    yield this.liquid.renderer.renderTemplates(this.templates, e, r);
  }
}
class cu extends T {
  constructor(e, r, n) {
    if (super(e, r, n), e.args.search(/\n\s*[^#\s]/g) !== -1)
      throw new Error("every line of an inline comment must start with a '#' character");
  }
  render() {
  }
}
const fu = {
  assign: Ca,
  for: qa,
  capture: Ha,
  case: Wa,
  comment: Ya,
  include: Qa,
  render: Ja,
  decrement: Za,
  increment: Ka,
  cycle: Ga,
  if: Xa,
  layout: eu,
  block: tu,
  raw: ru,
  tablerow: iu,
  unless: su,
  break: ou,
  continue: au,
  echo: uu,
  liquid: lu,
  "#": cu
};
class Jt {
  constructor(e = {}) {
    this.renderer = new xs(), this.filters = {}, this.tags = {}, this.options = eo(e), this.parser = new te(this), ir(fu, (r, n) => this.registerTag(n, r)), ir(Ma, (r, n) => this.registerFilter(n, r));
  }
  parse(e, r) {
    return new te(this).parse(e, r);
  }
  _render(e, r, n) {
    const i = r instanceof ie ? r : new ie(r, this.options, n);
    return this.renderer.renderTemplates(e, i);
  }
  render(e, r, n) {
    return q(this, void 0, void 0, function* () {
      return Z(this._render(e, r, Object.assign(Object.assign({}, n), { sync: !1 })));
    });
  }
  renderSync(e, r, n) {
    return G(this._render(e, r, Object.assign(Object.assign({}, n), { sync: !0 })));
  }
  renderToNodeStream(e, r, n = {}) {
    const i = new ie(r, this.options, n);
    return this.renderer.renderTemplatesToNodeStream(e, i);
  }
  _parseAndRender(e, r, n) {
    const i = this.parse(e);
    return this._render(i, r, n);
  }
  parseAndRender(e, r, n) {
    return q(this, void 0, void 0, function* () {
      return Z(this._parseAndRender(e, r, Object.assign(Object.assign({}, n), { sync: !1 })));
    });
  }
  parseAndRenderSync(e, r, n) {
    return G(this._parseAndRender(e, r, Object.assign(Object.assign({}, n), { sync: !0 })));
  }
  _parsePartialFile(e, r, n) {
    return new te(this).parseFile(e, r, ue.Partials, n);
  }
  _parseLayoutFile(e, r, n) {
    return new te(this).parseFile(e, r, ue.Layouts, n);
  }
  _parseFile(e, r, n, i) {
    return new te(this).parseFile(e, r, n, i);
  }
  parseFile(e, r) {
    return q(this, void 0, void 0, function* () {
      return Z(new te(this).parseFile(e, !1, r));
    });
  }
  parseFileSync(e, r) {
    return G(new te(this).parseFile(e, !0, r));
  }
  *_renderFile(e, r, n) {
    const i = yield this._parseFile(e, n.sync, n.lookupType);
    return yield this._render(i, r, n);
  }
  renderFile(e, r, n) {
    return q(this, void 0, void 0, function* () {
      return Z(this._renderFile(e, r, Object.assign(Object.assign({}, n), { sync: !1 })));
    });
  }
  renderFileSync(e, r, n) {
    return G(this._renderFile(e, r, Object.assign(Object.assign({}, n), { sync: !0 })));
  }
  renderFileToNodeStream(e, r, n) {
    return q(this, void 0, void 0, function* () {
      const i = yield this.parseFile(e);
      return this.renderToNodeStream(i, r, n);
    });
  }
  _evalValue(e, r) {
    const n = new M(e, this), i = r instanceof ie ? r : new ie(r, this.options);
    return n.value(i);
  }
  evalValue(e, r) {
    return q(this, void 0, void 0, function* () {
      return Z(this._evalValue(e, r));
    });
  }
  evalValueSync(e, r) {
    return G(this._evalValue(e, r));
  }
  registerFilter(e, r) {
    this.filters[e] = r;
  }
  registerTag(e, r) {
    this.tags[e] = x(r) ? r : oo(r);
  }
  plugin(e) {
    return e.call(this, Jt);
  }
  express() {
    const e = this;
    let r = !0;
    return function(n, i, s) {
      if (r) {
        r = !1;
        const o = We(this.root);
        e.options.root.unshift(...o), e.options.layouts.unshift(...o), e.options.partials.unshift(...o);
      }
      e.renderFile(n, i).then((o) => s(null, o), s);
    };
  }
}
var Mn = "en", Qt = {}, wt = {};
function Cn() {
  return Mn;
}
function hu(t) {
  Mn = t;
}
function pu(t) {
  return Qt[t];
}
function du(t) {
  if (!t)
    throw new Error("No locale data passed");
  Qt[t.locale] = t, wt[t.locale.toLowerCase()] = t.locale;
}
function Lr(t) {
  if (Qt[t])
    return t;
  if (wt[t.toLowerCase()])
    return wt[t.toLowerCase()];
}
function qn(t) {
  var e = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, r = e.localeMatcher || "lookup";
  switch (r) {
    case "lookup":
      return Ar(t);
    // "best fit" locale matching is not supported.
    // https://github.com/catamphetamine/relative-time-format/issues/2
    case "best fit":
      return Ar(t);
    default:
      throw new RangeError('Invalid "localeMatcher" option: '.concat(r));
  }
}
function Ar(t) {
  var e = Lr(t);
  if (e)
    return e;
  for (var r = t.split("-"); t.length > 1; ) {
    r.pop(), t = r.join("-");
    var n = Lr(t);
    if (n)
      return n;
  }
}
var u = {
  af: function(e) {
    return e == 1 ? "one" : "other";
  },
  am: function(e) {
    return e >= 0 && e <= 1 ? "one" : "other";
  },
  ar: function(e) {
    var r = String(e).split("."), n = Number(r[0]) == e, i = n && r[0].slice(-2);
    return e == 0 ? "zero" : e == 1 ? "one" : e == 2 ? "two" : i >= 3 && i <= 10 ? "few" : i >= 11 && i <= 99 ? "many" : "other";
  },
  ast: function(e) {
    var r = String(e).split("."), n = !r[1];
    return e == 1 && n ? "one" : "other";
  },
  be: function(e) {
    var r = String(e).split("."), n = Number(r[0]) == e, i = n && r[0].slice(-1), s = n && r[0].slice(-2);
    return i == 1 && s != 11 ? "one" : i >= 2 && i <= 4 && (s < 12 || s > 14) ? "few" : n && i == 0 || i >= 5 && i <= 9 || s >= 11 && s <= 14 ? "many" : "other";
  },
  br: function(e) {
    var r = String(e).split("."), n = Number(r[0]) == e, i = n && r[0].slice(-1), s = n && r[0].slice(-2), o = n && r[0].slice(-6);
    return i == 1 && s != 11 && s != 71 && s != 91 ? "one" : i == 2 && s != 12 && s != 72 && s != 92 ? "two" : (i == 3 || i == 4 || i == 9) && (s < 10 || s > 19) && (s < 70 || s > 79) && (s < 90 || s > 99) ? "few" : e != 0 && n && o == 0 ? "many" : "other";
  },
  bs: function(e) {
    var r = String(e).split("."), n = r[0], i = r[1] || "", s = !r[1], o = n.slice(-1), a = n.slice(-2), l = i.slice(-1), c = i.slice(-2);
    return s && o == 1 && a != 11 || l == 1 && c != 11 ? "one" : s && o >= 2 && o <= 4 && (a < 12 || a > 14) || l >= 2 && l <= 4 && (c < 12 || c > 14) ? "few" : "other";
  },
  ca: function(e) {
    var r = String(e).split("."), n = r[0], i = !r[1], s = n.slice(-6);
    return e == 1 && i ? "one" : n != 0 && s == 0 && i ? "many" : "other";
  },
  ceb: function(e) {
    var r = String(e).split("."), n = r[0], i = r[1] || "", s = !r[1], o = n.slice(-1), a = i.slice(-1);
    return s && (n == 1 || n == 2 || n == 3) || s && o != 4 && o != 6 && o != 9 || !s && a != 4 && a != 6 && a != 9 ? "one" : "other";
  },
  cs: function(e) {
    var r = String(e).split("."), n = r[0], i = !r[1];
    return e == 1 && i ? "one" : n >= 2 && n <= 4 && i ? "few" : i ? "other" : "many";
  },
  cy: function(e) {
    return e == 0 ? "zero" : e == 1 ? "one" : e == 2 ? "two" : e == 3 ? "few" : e == 6 ? "many" : "other";
  },
  da: function(e) {
    var r = String(e).split("."), n = r[0], i = Number(r[0]) == e;
    return e == 1 || !i && (n == 0 || n == 1) ? "one" : "other";
  },
  dsb: function(e) {
    var r = String(e).split("."), n = r[0], i = r[1] || "", s = !r[1], o = n.slice(-2), a = i.slice(-2);
    return s && o == 1 || a == 1 ? "one" : s && o == 2 || a == 2 ? "two" : s && (o == 3 || o == 4) || a == 3 || a == 4 ? "few" : "other";
  },
  dz: function(e) {
    return "other";
  },
  es: function(e) {
    var r = String(e).split("."), n = r[0], i = !r[1], s = n.slice(-6);
    return e == 1 ? "one" : n != 0 && s == 0 && i ? "many" : "other";
  },
  ff: function(e) {
    return e >= 0 && e < 2 ? "one" : "other";
  },
  fr: function(e) {
    var r = String(e).split("."), n = r[0], i = !r[1], s = n.slice(-6);
    return e >= 0 && e < 2 ? "one" : n != 0 && s == 0 && i ? "many" : "other";
  },
  ga: function(e) {
    var r = String(e).split("."), n = Number(r[0]) == e;
    return e == 1 ? "one" : e == 2 ? "two" : n && e >= 3 && e <= 6 ? "few" : n && e >= 7 && e <= 10 ? "many" : "other";
  },
  gd: function(e) {
    var r = String(e).split("."), n = Number(r[0]) == e;
    return e == 1 || e == 11 ? "one" : e == 2 || e == 12 ? "two" : n && e >= 3 && e <= 10 || n && e >= 13 && e <= 19 ? "few" : "other";
  },
  he: function(e) {
    var r = String(e).split("."), n = r[0], i = !r[1];
    return n == 1 && i || n == 0 && !i ? "one" : n == 2 && i ? "two" : "other";
  },
  is: function(e) {
    var r = String(e).split("."), n = r[0], i = (r[1] || "").replace(/0+$/, ""), s = Number(r[0]) == e, o = n.slice(-1), a = n.slice(-2);
    return s && o == 1 && a != 11 || i % 10 == 1 && i % 100 != 11 ? "one" : "other";
  },
  ksh: function(e) {
    return e == 0 ? "zero" : e == 1 ? "one" : "other";
  },
  lt: function(e) {
    var r = String(e).split("."), n = r[1] || "", i = Number(r[0]) == e, s = i && r[0].slice(-1), o = i && r[0].slice(-2);
    return s == 1 && (o < 11 || o > 19) ? "one" : s >= 2 && s <= 9 && (o < 11 || o > 19) ? "few" : n != 0 ? "many" : "other";
  },
  lv: function(e) {
    var r = String(e).split("."), n = r[1] || "", i = n.length, s = Number(r[0]) == e, o = s && r[0].slice(-1), a = s && r[0].slice(-2), l = n.slice(-2), c = n.slice(-1);
    return s && o == 0 || a >= 11 && a <= 19 || i == 2 && l >= 11 && l <= 19 ? "zero" : o == 1 && a != 11 || i == 2 && c == 1 && l != 11 || i != 2 && c == 1 ? "one" : "other";
  },
  mk: function(e) {
    var r = String(e).split("."), n = r[0], i = r[1] || "", s = !r[1], o = n.slice(-1), a = n.slice(-2), l = i.slice(-1), c = i.slice(-2);
    return s && o == 1 && a != 11 || l == 1 && c != 11 ? "one" : "other";
  },
  mt: function(e) {
    var r = String(e).split("."), n = Number(r[0]) == e, i = n && r[0].slice(-2);
    return e == 1 ? "one" : e == 2 ? "two" : e == 0 || i >= 3 && i <= 10 ? "few" : i >= 11 && i <= 19 ? "many" : "other";
  },
  pa: function(e) {
    return e == 0 || e == 1 ? "one" : "other";
  },
  pl: function(e) {
    var r = String(e).split("."), n = r[0], i = !r[1], s = n.slice(-1), o = n.slice(-2);
    return e == 1 && i ? "one" : i && s >= 2 && s <= 4 && (o < 12 || o > 14) ? "few" : i && n != 1 && (s == 0 || s == 1) || i && s >= 5 && s <= 9 || i && o >= 12 && o <= 14 ? "many" : "other";
  },
  pt: function(e) {
    var r = String(e).split("."), n = r[0], i = !r[1], s = n.slice(-6);
    return n == 0 || n == 1 ? "one" : n != 0 && s == 0 && i ? "many" : "other";
  },
  ro: function(e) {
    var r = String(e).split("."), n = !r[1], i = Number(r[0]) == e, s = i && r[0].slice(-2);
    return e == 1 && n ? "one" : !n || e == 0 || e != 1 && s >= 1 && s <= 19 ? "few" : "other";
  },
  ru: function(e) {
    var r = String(e).split("."), n = r[0], i = !r[1], s = n.slice(-1), o = n.slice(-2);
    return i && s == 1 && o != 11 ? "one" : i && s >= 2 && s <= 4 && (o < 12 || o > 14) ? "few" : i && s == 0 || i && s >= 5 && s <= 9 || i && o >= 11 && o <= 14 ? "many" : "other";
  },
  se: function(e) {
    return e == 1 ? "one" : e == 2 ? "two" : "other";
  },
  si: function(e) {
    var r = String(e).split("."), n = r[0], i = r[1] || "";
    return e == 0 || e == 1 || n == 0 && i == 1 ? "one" : "other";
  },
  sl: function(e) {
    var r = String(e).split("."), n = r[0], i = !r[1], s = n.slice(-2);
    return i && s == 1 ? "one" : i && s == 2 ? "two" : i && (s == 3 || s == 4) || !i ? "few" : "other";
  }
};
u.as = u.am;
u.az = u.af;
u.bg = u.af;
u.bn = u.am;
u.brx = u.af;
u.ce = u.af;
u.chr = u.af;
u.de = u.ast;
u.ee = u.af;
u.el = u.af;
u.en = u.ast;
u.et = u.ast;
u.eu = u.af;
u.fa = u.am;
u.fi = u.ast;
u.fil = u.ceb;
u.fo = u.af;
u.fur = u.af;
u.fy = u.ast;
u.gl = u.ast;
u.gu = u.am;
u.ha = u.af;
u.hi = u.am;
u.hr = u.bs;
u.hsb = u.dsb;
u.hu = u.af;
u.hy = u.ff;
u.ia = u.ast;
u.id = u.dz;
u.ig = u.dz;
u.it = u.ca;
u.ja = u.dz;
u.jgo = u.af;
u.jv = u.dz;
u.ka = u.af;
u.kea = u.dz;
u.kk = u.af;
u.kl = u.af;
u.km = u.dz;
u.kn = u.am;
u.ko = u.dz;
u.ks = u.af;
u.ku = u.af;
u.ky = u.af;
u.lb = u.af;
u.lkt = u.dz;
u.lo = u.dz;
u.ml = u.af;
u.mn = u.af;
u.mr = u.af;
u.ms = u.dz;
u.my = u.dz;
u.nb = u.af;
u.ne = u.af;
u.nl = u.ast;
u.nn = u.af;
u.no = u.af;
u.or = u.af;
u.pcm = u.am;
u.ps = u.af;
u.rm = u.af;
u.sah = u.dz;
u.sc = u.ast;
u.sd = u.af;
u.sk = u.cs;
u.so = u.af;
u.sq = u.af;
u.sr = u.bs;
u.su = u.dz;
u.sv = u.ast;
u.sw = u.ast;
u.ta = u.af;
u.te = u.af;
u.th = u.dz;
u.ti = u.pa;
u.tk = u.af;
u.to = u.dz;
u.tr = u.af;
u.ug = u.af;
u.uk = u.ru;
u.ur = u.ast;
u.uz = u.af;
u.vi = u.dz;
u.wae = u.af;
u.wo = u.dz;
u.xh = u.af;
u.yi = u.ast;
u.yo = u.dz;
u.yue = u.dz;
u.zh = u.dz;
u.zu = u.am;
function Fr(t) {
  return t === "pt-PT" ? t : gu(t);
}
var mu = /^([a-z0-9]+)/i;
function gu(t) {
  var e = t.match(mu);
  if (!e)
    throw new TypeError("Invalid locale: ".concat(t));
  return e[1];
}
function yu(t, e) {
  if (!(t instanceof e))
    throw new TypeError("Cannot call a class as a function");
}
function Dr(t, e) {
  for (var r = 0; r < e.length; r++) {
    var n = e[r];
    n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), Object.defineProperty(t, n.key, n);
  }
}
function wu(t, e, r) {
  return e && Dr(t.prototype, e), r && Dr(t, r), Object.defineProperty(t, "prototype", { writable: !1 }), t;
}
var vt = /* @__PURE__ */ function() {
  function t(e, r) {
    yu(this, t);
    var n = t.supportedLocalesOf(e);
    if (n.length === 0)
      throw new RangeError("Unsupported locale: " + e);
    if (r && r.type !== "cardinal")
      throw new RangeError('Only "cardinal" "type" is supported');
    this.$ = u[Fr(n[0])];
  }
  return wu(t, [{
    key: "select",
    value: function(r) {
      return this.$(r);
    }
  }], [{
    key: "supportedLocalesOf",
    value: function(r) {
      return typeof r == "string" && (r = [r]), r.filter(function(n) {
        return u[Fr(n)];
      });
    }
  }]), t;
}();
function bt(t) {
  "@babel/helpers - typeof";
  return bt = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(e) {
    return typeof e;
  } : function(e) {
    return e && typeof Symbol == "function" && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
  }, bt(t);
}
function _r(t, e) {
  var r = Object.keys(t);
  if (Object.getOwnPropertySymbols) {
    var n = Object.getOwnPropertySymbols(t);
    e && (n = n.filter(function(i) {
      return Object.getOwnPropertyDescriptor(t, i).enumerable;
    })), r.push.apply(r, n);
  }
  return r;
}
function Rr(t) {
  for (var e = 1; e < arguments.length; e++) {
    var r = arguments[e] != null ? arguments[e] : {};
    e % 2 ? _r(Object(r), !0).forEach(function(n) {
      vu(t, n, r[n]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(r)) : _r(Object(r)).forEach(function(n) {
      Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(r, n));
    });
  }
  return t;
}
function vu(t, e, r) {
  return e in t ? Object.defineProperty(t, e, { value: r, enumerable: !0, configurable: !0, writable: !0 }) : t[e] = r, t;
}
function Pr(t, e) {
  return Su(t) || Ou(t, e) || Tu(t, e) || bu();
}
function bu() {
  throw new TypeError(`Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`);
}
function Tu(t, e) {
  if (t) {
    if (typeof t == "string") return Er(t, e);
    var r = Object.prototype.toString.call(t).slice(8, -1);
    if (r === "Object" && t.constructor && (r = t.constructor.name), r === "Map" || r === "Set") return Array.from(t);
    if (r === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r)) return Er(t, e);
  }
}
function Er(t, e) {
  (e == null || e > t.length) && (e = t.length);
  for (var r = 0, n = new Array(e); r < e; r++)
    n[r] = t[r];
  return n;
}
function Ou(t, e) {
  var r = t == null ? null : typeof Symbol < "u" && t[Symbol.iterator] || t["@@iterator"];
  if (r != null) {
    var n = [], i = !0, s = !1, o, a;
    try {
      for (r = r.call(t); !(i = (o = r.next()).done) && (n.push(o.value), !(e && n.length === e)); i = !0)
        ;
    } catch (l) {
      s = !0, a = l;
    } finally {
      try {
        !i && r.return != null && r.return();
      } finally {
        if (s) throw a;
      }
    }
    return n;
  }
}
function Su(t) {
  if (Array.isArray(t)) return t;
}
function ku(t, e) {
  if (!(t instanceof e))
    throw new TypeError("Cannot call a class as a function");
}
function xu(t, e) {
  for (var r = 0; r < e.length; r++) {
    var n = e[r];
    n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), Object.defineProperty(t, n.key, n);
  }
}
function Lu(t, e, r) {
  return e && xu(t.prototype, e), Object.defineProperty(t, "prototype", { writable: !1 }), t;
}
var Au = ["second", "minute", "hour", "day", "week", "month", "quarter", "year"], Fu = ["auto", "always"], Du = ["long", "short", "narrow"], _u = ["lookup", "best fit"], X = /* @__PURE__ */ function() {
  function t() {
    var e = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : [], r = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {};
    ku(this, t);
    var n = r.numeric, i = r.style, s = r.localeMatcher;
    if (this.numeric = "always", this.style = "long", this.localeMatcher = "lookup", n !== void 0) {
      if (Fu.indexOf(n) < 0)
        throw new RangeError('Invalid "numeric" option: '.concat(n));
      this.numeric = n;
    }
    if (i !== void 0) {
      if (Du.indexOf(i) < 0)
        throw new RangeError('Invalid "style" option: '.concat(i));
      this.style = i;
    }
    if (s !== void 0) {
      if (_u.indexOf(s) < 0)
        throw new RangeError('Invalid "localeMatcher" option: '.concat(s));
      this.localeMatcher = s;
    }
    if (typeof e == "string" && (e = [e]), e.push(Cn()), this.locale = t.supportedLocalesOf(e, {
      localeMatcher: this.localeMatcher
    })[0], !this.locale)
      throw new Error("No supported locale was found");
    vt.supportedLocalesOf(this.locale).length > 0 ? this.pluralRules = new vt(this.locale) : console.warn('"'.concat(this.locale, '" locale is not supported')), typeof Intl < "u" && Intl.NumberFormat ? (this.numberFormat = new Intl.NumberFormat(this.locale), this.numberingSystem = this.numberFormat.resolvedOptions().numberingSystem) : this.numberingSystem = "latn", this.locale = qn(this.locale, {
      localeMatcher: this.localeMatcher
    });
  }
  return Lu(t, [{
    key: "format",
    value: function() {
      var r = Nr(arguments), n = Pr(r, 2), i = n[0], s = n[1];
      return this.getRule(i, s).replace("{0}", this.formatNumber(Math.abs(i)));
    }
    /**
     * Formats time `number` in `units` (either in past or in future).
     * @param {number} number - Time interval value.
     * @param {string} unit - Time interval measurement unit.
     * @return {Object[]} The parts (`{ type, value, unit? }`).
     * @throws {RangeError} If unit is not one of "second", "minute", "hour", "day", "week", "month", "quarter".
     * @example
     * // Version 1 (deprecated).
     * // Returns [
     * //   { type: "literal", value: "in " },
     * //   { type: "day", value: "100" },
     * //   { type: "literal", value: " days" }
     * // ]
     * rtf.formatToParts(100, "day")
     * //
     * // Version 2.
     * // Returns [
     * //   { type: "literal", value: "in " },
     * //   { type: "integer", value: "100", unit: "day" },
     * //   { type: "literal", value: " days" }
     * // ]
     * rtf.formatToParts(100, "day")
     */
  }, {
    key: "formatToParts",
    value: function() {
      var r = Nr(arguments), n = Pr(r, 2), i = n[0], s = n[1], o = this.getRule(i, s), a = o.indexOf("{0}");
      if (a < 0)
        return [{
          type: "literal",
          value: o
        }];
      var l = [];
      return a > 0 && l.push({
        type: "literal",
        value: o.slice(0, a)
      }), l = l.concat(this.formatNumberToParts(Math.abs(i)).map(function(c) {
        return Rr(Rr({}, c), {}, {
          unit: s
        });
      })), a + 3 < o.length - 1 && l.push({
        type: "literal",
        value: o.slice(a + 3)
      }), l;
    }
    /**
     * Returns formatting rule for `value` in `units` (either in past or in future).
     * @param {number} value - Time interval value.
     * @param {string} unit - Time interval measurement unit.
     * @return {string}
     * @throws {RangeError} If unit is not one of "second", "minute", "hour", "day", "week", "month", "quarter".
     * @example
     * // Returns "{0} days ago"
     * getRule(-2, "day")
     */
  }, {
    key: "getRule",
    value: function(r, n) {
      var i = pu(this.locale)[this.style][n];
      if (typeof i == "string")
        return i;
      if (this.numeric === "auto") {
        if (r === -2 || r === -1) {
          var s = i["previous".concat(r === -1 ? "" : "-" + Math.abs(r))];
          if (s)
            return s;
        } else if (r === 1 || r === 2) {
          var o = i["next".concat(r === 1 ? "" : "-" + Math.abs(r))];
          if (o)
            return o;
        } else if (r === 0 && i.current)
          return i.current;
      }
      var a = i[$u(r) ? "past" : "future"];
      if (typeof a == "string")
        return a;
      var l = this.pluralRules && this.pluralRules.select(Math.abs(r)) || "other";
      return a[l] || a.other;
    }
    /**
     * Formats a number into a string.
     * Uses `Intl.NumberFormat` when available.
     * @param  {number} number
     * @return {string}
     */
  }, {
    key: "formatNumber",
    value: function(r) {
      return this.numberFormat ? this.numberFormat.format(r) : String(r);
    }
    /**
     * Formats a number into a list of parts.
     * Uses `Intl.NumberFormat` when available.
     * @param  {number} number
     * @return {object[]}
     */
  }, {
    key: "formatNumberToParts",
    value: function(r) {
      return this.numberFormat && this.numberFormat.formatToParts ? this.numberFormat.formatToParts(r) : [{
        type: "integer",
        value: this.formatNumber(r)
      }];
    }
    /**
     * Returns a new object with properties reflecting the locale and date and time formatting options computed during initialization of this DateTimeFormat object.
     * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/DateTimeFormat/resolvedOptions
     * @return {Object}
     */
  }, {
    key: "resolvedOptions",
    value: function() {
      return {
        locale: this.locale,
        style: this.style,
        numeric: this.numeric,
        numberingSystem: this.numberingSystem
      };
    }
  }]), t;
}();
X.supportedLocalesOf = function(t) {
  var e = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {};
  if (typeof t == "string")
    t = [t];
  else if (!Array.isArray(t))
    throw new TypeError('Invalid "locales" argument');
  return t.filter(function(r) {
    return qn(r, e);
  });
};
X.addLocale = du;
X.setDefaultLocale = hu;
X.getDefaultLocale = Cn;
X.PluralRules = vt;
var ht = 'Invalid "unit" argument';
function Ru(t) {
  if (bt(t) === "symbol")
    throw new TypeError(ht);
  if (typeof t != "string")
    throw new RangeError("".concat(ht, ": ").concat(t));
  if (t[t.length - 1] === "s" && (t = t.slice(0, t.length - 1)), Au.indexOf(t) < 0)
    throw new RangeError("".concat(ht, ": ").concat(t));
  return t;
}
var Pu = 'Invalid "number" argument';
function Eu(t) {
  if (t = Number(t), Number.isFinite && !Number.isFinite(t))
    throw new RangeError("".concat(Pu, ": ").concat(t));
  return t;
}
function Nu(t) {
  return 1 / t === -1 / 0;
}
function $u(t) {
  return t < 0 || t === 0 && Nu(t);
}
function Nr(t) {
  if (t.length < 2)
    throw new TypeError('"unit" argument is required');
  return [Eu(t[0]), Ru(t[1])];
}
function Ze(t) {
  "@babel/helpers - typeof";
  return Ze = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(e) {
    return typeof e;
  } : function(e) {
    return e && typeof Symbol == "function" && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
  }, Ze(t);
}
function Iu(t, e) {
  if (!(t instanceof e))
    throw new TypeError("Cannot call a class as a function");
}
function ju(t, e) {
  for (var r = 0; r < e.length; r++) {
    var n = e[r];
    n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), Object.defineProperty(t, n.key, n);
  }
}
function zu(t, e, r) {
  return e && ju(t.prototype, e), Object.defineProperty(t, "prototype", { writable: !1 }), t;
}
var $r = /* @__PURE__ */ function() {
  function t() {
    Iu(this, t), this.cache = {};
  }
  return zu(t, [{
    key: "get",
    value: function() {
      for (var r = this.cache, n = arguments.length, i = new Array(n), s = 0; s < n; s++)
        i[s] = arguments[s];
      for (var o = 0, a = i; o < a.length; o++) {
        var l = a[o];
        if (Ze(r) !== "object")
          return;
        r = r[l];
      }
      return r;
    }
  }, {
    key: "put",
    value: function() {
      for (var r = arguments.length, n = new Array(r), i = 0; i < r; i++)
        n[i] = arguments[i];
      for (var s = n.pop(), o = n.pop(), a = this.cache, l = 0, c = n; l < c.length; l++) {
        var f = c[l];
        Ze(a[f]) !== "object" && (a[f] = {}), a = a[f];
      }
      return a[o] = s;
    }
  }]), t;
}();
function Tt(t) {
  "@babel/helpers - typeof";
  return Tt = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(e) {
    return typeof e;
  } : function(e) {
    return e && typeof Symbol == "function" && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
  }, Tt(t);
}
function Mu(t, e) {
  var r = typeof Symbol < "u" && t[Symbol.iterator] || t["@@iterator"];
  if (r) return (r = r.call(t)).next.bind(r);
  if (Array.isArray(t) || (r = Cu(t)) || e) {
    r && (t = r);
    var n = 0;
    return function() {
      return n >= t.length ? { done: !0 } : { done: !1, value: t[n++] };
    };
  }
  throw new TypeError(`Invalid attempt to iterate non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`);
}
function Cu(t, e) {
  if (t) {
    if (typeof t == "string") return Ir(t, e);
    var r = Object.prototype.toString.call(t).slice(8, -1);
    if (r === "Object" && t.constructor && (r = t.constructor.name), r === "Map" || r === "Set") return Array.from(t);
    if (r === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r)) return Ir(t, e);
  }
}
function Ir(t, e) {
  (e == null || e > t.length) && (e = t.length);
  for (var r = 0, n = new Array(e); r < e; r++)
    n[r] = t[r];
  return n;
}
function qu(t, e) {
  for (var r = Mu(t), n; !(n = r()).done; ) {
    var i = n.value;
    if (e(i))
      return i;
    for (var s = i.split("-"); s.length > 1; )
      if (s.pop(), i = s.join("-"), e(i))
        return i;
  }
  throw new Error("No locale data has been registered for any of the locales: ".concat(t.join(", ")));
}
function Uu() {
  var t = (typeof Intl > "u" ? "undefined" : Tt(Intl)) === "object";
  return t && typeof Intl.DateTimeFormat == "function";
}
function Ot(t) {
  "@babel/helpers - typeof";
  return Ot = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(e) {
    return typeof e;
  } : function(e) {
    return e && typeof Symbol == "function" && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
  }, Ot(t);
}
function Vu(t) {
  return Hu(t) && (Array.isArray(t.steps) || // `gradation` property is deprecated: it has been renamed to `steps`.
  Array.isArray(t.gradation) || // `flavour` property is deprecated: it has been renamed to `labels`.
  Array.isArray(t.flavour) || typeof t.flavour == "string" || Array.isArray(t.labels) || typeof t.labels == "string" || // `units` property is deprecated.
  Array.isArray(t.units) || // `custom` property is deprecated.
  typeof t.custom == "function");
}
var Bu = {}.constructor;
function Hu(t) {
  return Ot(t) !== void 0 && t !== null && t.constructor === Bu;
}
var Y = 60, Ge = 60 * Y, ae = 24 * Ge, St = 7 * ae, kt = 30.44 * ae, Un = 146097 / 400 * ae;
function ye(t) {
  switch (t) {
    case "second":
      return 1;
    case "minute":
      return Y;
    case "hour":
      return Ge;
    case "day":
      return ae;
    case "week":
      return St;
    case "month":
      return kt;
    case "year":
      return Un;
  }
}
function Vn(t) {
  return t.factor !== void 0 ? t.factor : ye(t.unit || t.formatAs) || 1;
}
function Re(t) {
  switch (t) {
    case "floor":
      return Math.floor;
    default:
      return Math.round;
  }
}
function Zt(t) {
  switch (t) {
    case "floor":
      return 1;
    default:
      return 0.5;
  }
}
function xt(t) {
  "@babel/helpers - typeof";
  return xt = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(e) {
    return typeof e;
  } : function(e) {
    return e && typeof Symbol == "function" && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
  }, xt(t);
}
function Bn(t, e) {
  var r = e.prevStep, n = e.timestamp, i = e.now, s = e.future, o = e.round, a;
  return r && (r.id || r.unit) && (a = t["threshold_for_".concat(r.id || r.unit)]), a === void 0 && t.threshold !== void 0 && (a = t.threshold, typeof a == "function" && (a = a(i, s))), a === void 0 && (a = t.minTime), xt(a) === "object" && (r && r.id && a[r.id] !== void 0 ? a = a[r.id] : a = a.default), typeof a == "function" && (a = a(n, {
    future: s,
    getMinTimeForUnit: function(c, f) {
      return jr(c, f || r && r.formatAs, {
        round: o
      });
    }
  })), a === void 0 && t.test && (t.test(n, {
    now: i,
    future: s
  }) ? a = 0 : a = 9007199254740991), a === void 0 && (r ? t.formatAs && r.formatAs && (a = jr(t.formatAs, r.formatAs, {
    round: o
  })) : a = 0), a === void 0 && console.warn("[javascript-time-ago] A step should specify `minTime`:\n" + JSON.stringify(t, null, 2)), a;
}
function jr(t, e, r) {
  var n = r.round, i = ye(t), s;
  if (e === "now" ? s = ye(t) : s = ye(e), i !== void 0 && s !== void 0)
    return i - s * (1 - Zt(n));
}
function zr(t, e) {
  var r = Object.keys(t);
  if (Object.getOwnPropertySymbols) {
    var n = Object.getOwnPropertySymbols(t);
    e && (n = n.filter(function(i) {
      return Object.getOwnPropertyDescriptor(t, i).enumerable;
    })), r.push.apply(r, n);
  }
  return r;
}
function Wu(t) {
  for (var e = 1; e < arguments.length; e++) {
    var r = arguments[e] != null ? arguments[e] : {};
    e % 2 ? zr(Object(r), !0).forEach(function(n) {
      Yu(t, n, r[n]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(r)) : zr(Object(r)).forEach(function(n) {
      Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(r, n));
    });
  }
  return t;
}
function Yu(t, e, r) {
  return e in t ? Object.defineProperty(t, e, { value: r, enumerable: !0, configurable: !0, writable: !0 }) : t[e] = r, t;
}
function Ju(t, e, r) {
  var n = r.now, i = r.future, s = r.round, o = r.units;
  t = Zu(t, o);
  var a = Qu(t, e, {
    now: n,
    future: i,
    round: s
  });
  {
    if (a) {
      var l = t[t.indexOf(a) - 1], c = t[t.indexOf(a) + 1];
      return [l, a, c];
    }
    return [void 0, void 0, t[0]];
  }
}
function Qu(t, e, r) {
  var n = r.now, i = r.future, s = r.round;
  if (t.length !== 0) {
    var o = Hn(t, e, {
      now: n,
      future: i || e < 0,
      round: s
    });
    if (o !== -1) {
      var a = t[o];
      if (a.granularity) {
        var l = Re(s)(Math.abs(e) / Vn(a) / a.granularity) * a.granularity;
        if (l === 0 && o > 0)
          return t[o - 1];
      }
      return a;
    }
  }
}
function Hn(t, e, r) {
  var n = arguments.length > 3 && arguments[3] !== void 0 ? arguments[3] : 0, i = Bn(t[n], Wu({
    prevStep: t[n - 1],
    timestamp: r.now - e * 1e3
  }, r));
  return i === void 0 || Math.abs(e) < i ? n - 1 : n === t.length - 1 ? n : Hn(t, e, r, n + 1);
}
function Zu(t, e) {
  return t.filter(function(r) {
    var n = r.unit, i = r.formatAs;
    return n = n || i, n ? e.indexOf(n) >= 0 : !0;
  });
}
function Gu(t, e, r) {
  var n = r.now, i = r.round;
  if (ye(t)) {
    var s = ye(t) * 1e3, o = e > n, a = Math.abs(e - n), l = Re(i)(a / s) * s;
    return o ? l > 0 ? a - l + Ku(i, s) : a - l + 1 : -(a - l) + Xu(i, s);
  }
}
function Xu(t, e) {
  return Zt(t) * e;
}
function Ku(t, e) {
  return (1 - Zt(t)) * e + 1;
}
var el = 365 * 24 * 60 * 60 * 1e3, Wn = 1e3 * el;
function tl(t, e, r) {
  var n = r.prevStep, i = r.nextStep, s = r.now, o = r.future, a = r.round, l = t.getTime ? t.getTime() : t, c = function(F) {
    return Gu(F, l, {
      now: s,
      round: a
    });
  }, f = nl(o ? e : i, l, {
    future: o,
    now: s,
    round: a,
    prevStep: o ? n : e
    // isFirstStep: future && isFirstStep
  });
  if (f !== void 0) {
    var d;
    if (e && (e.getTimeToNextUpdate && (d = e.getTimeToNextUpdate(l, {
      getTimeToNextUpdateForUnit: c,
      getRoundFunction: Re,
      now: s,
      future: o,
      round: a
    })), d === void 0)) {
      var g = e.unit || e.formatAs;
      g && (d = c(g));
    }
    return d === void 0 ? f : Math.min(d, f);
  }
}
function rl(t, e, r) {
  var n = r.now, i = r.future, s = r.round, o = r.prevStep, a = Bn(t, {
    timestamp: e,
    now: n,
    future: i,
    round: s,
    prevStep: o
  });
  if (a !== void 0)
    return i ? e - a * 1e3 + 1 : a === 0 && e === n ? Wn : e + a * 1e3;
}
function nl(t, e, r) {
  var n = r.now, i = r.future, s = r.round, o = r.prevStep;
  if (t) {
    var a = rl(t, e, {
      now: n,
      future: i,
      round: s,
      prevStep: o
    });
    return a === void 0 ? void 0 : a - n;
  } else
    return i ? e - n + 1 : Wn;
}
var Yn = {};
function me(t) {
  return Yn[t];
}
function Jn(t) {
  if (!t)
    throw new Error("[javascript-time-ago] No locale data passed.");
  Yn[t.locale] = t;
}
const il = [{
  formatAs: "now"
}, {
  formatAs: "second"
}, {
  formatAs: "minute"
}, {
  formatAs: "hour"
}, {
  formatAs: "day"
}, {
  formatAs: "week"
}, {
  formatAs: "month"
}, {
  formatAs: "year"
}], Lt = {
  steps: il,
  labels: "long"
};
function Mr(t, e) {
  var r = Object.keys(t);
  if (Object.getOwnPropertySymbols) {
    var n = Object.getOwnPropertySymbols(t);
    e && (n = n.filter(function(i) {
      return Object.getOwnPropertyDescriptor(t, i).enumerable;
    })), r.push.apply(r, n);
  }
  return r;
}
function Cr(t) {
  for (var e = 1; e < arguments.length; e++) {
    var r = arguments[e] != null ? arguments[e] : {};
    e % 2 ? Mr(Object(r), !0).forEach(function(n) {
      sl(t, n, r[n]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(r)) : Mr(Object(r)).forEach(function(n) {
      Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(r, n));
    });
  }
  return t;
}
function sl(t, e, r) {
  return e in t ? Object.defineProperty(t, e, { value: r, enumerable: !0, configurable: !0, writable: !0 }) : t[e] = r, t;
}
const At = Cr(Cr({}, Lt), {}, {
  // Skip "seconds".
  steps: Lt.steps.filter(function(t) {
    return t.formatAs !== "second";
  })
}), Qn = [{
  // This step returns the amount of seconds
  // by dividing the amount of seconds by `1`.
  factor: 1,
  // "now" labels are used for formatting the output.
  unit: "now"
}, {
  // When the language doesn't support `now` unit,
  // the first step is ignored, and it uses this `second` unit.
  threshold: 1,
  // `threshold_for_now` should be the same as `threshold` on minutes.
  threshold_for_now: 45.5,
  // This step returns the amount of seconds
  // by dividing the amount of seconds by `1`.
  factor: 1,
  // "second" labels are used for formatting the output.
  unit: "second"
}, {
  // `threshold` should be the same as `threshold_for_now` on seconds.
  threshold: 45.5,
  // Return the amount of minutes by dividing the amount
  // of seconds by the amount of seconds in a minute.
  factor: Y,
  // "minute" labels are used for formatting the output.
  unit: "minute"
}, {
  // This step is effective starting from 2.5 minutes.
  threshold: 2.5 * Y,
  // Allow only 5-minute increments of minutes starting from 2.5 minutes.
  // `granularity` — (advanced) Time interval value "granularity".
  // For example, it could be set to `5` for minutes to allow only 5-minute increments
  // when formatting time intervals: `0 minutes`, `5 minutes`, `10 minutes`, etc.
  // Perhaps this feature will be removed because there seem to be no use cases
  // of it in the real world.
  granularity: 5,
  // Return the amount of minutes by dividing the amount
  // of seconds by the amount of seconds in a minute.
  factor: Y,
  // "minute" labels are used for formatting the output.
  unit: "minute"
}, {
  // This step is effective starting from 22.5 minutes.
  threshold: 22.5 * Y,
  // Return the amount of minutes by dividing the amount
  // of seconds by the amount of seconds in  half-an-hour.
  factor: 0.5 * Ge,
  // "half-hour" labels are used for formatting the output.
  // (if available, which is no longer the case)
  unit: "half-hour"
}, {
  // This step is effective starting from 42.5 minutes.
  threshold: 42.5 * Y,
  threshold_for_minute: 52.5 * Y,
  // Return the amount of minutes by dividing the amount
  // of seconds by the amount of seconds in an hour.
  factor: Ge,
  // "hour" labels are used for formatting the output.
  unit: "hour"
}, {
  // This step is effective starting from 20.5 hours.
  threshold: 20.5 / 24 * ae,
  // Return the amount of minutes by dividing the amount
  // of seconds by the amount of seconds in a day.
  factor: ae,
  // "day" labels are used for formatting the output.
  unit: "day"
}, {
  // This step is effective starting from 5.5 days.
  threshold: 5.5 * ae,
  // Return the amount of minutes by dividing the amount
  // of seconds by the amount of seconds in a week.
  factor: St,
  // "week" labels are used for formatting the output.
  unit: "week"
}, {
  // This step is effective starting from 3.5 weeks.
  threshold: 3.5 * St,
  // Return the amount of minutes by dividing the amount
  // of seconds by the amount of seconds in a month.
  factor: kt,
  // "month" labels are used for formatting the output.
  unit: "month"
}, {
  // This step is effective starting from 10.5 months.
  threshold: 10.5 * kt,
  // Return the amount of minutes by dividing the amount
  // of seconds by the amount of seconds in a year.
  factor: Un,
  // "year" labels are used for formatting the output.
  unit: "year"
}], qr = {
  gradation: Qn,
  flavour: "long",
  units: ["now", "minute", "hour", "day", "week", "month", "year"]
}, ol = {
  gradation: Qn,
  flavour: "long-time",
  units: ["now", "minute", "hour", "day", "week", "month", "year"]
};
function Zn(t) {
  return t instanceof Date ? t : new Date(t);
}
var Ft = [{
  formatAs: "second"
}, {
  formatAs: "minute"
}, {
  formatAs: "hour"
}], B = {}, al = {
  minTime: function(e, r) {
    r.future;
    var n = r.getMinTimeForUnit;
    return n("day");
  },
  format: function(e, r) {
    return B[r] || (B[r] = {}), B[r].dayMonth || (B[r].dayMonth = new Intl.DateTimeFormat(r, {
      month: "short",
      day: "numeric"
    })), B[r].dayMonth.format(Zn(e));
  }
}, ul = {
  minTime: function(e, r) {
    var n = r.future;
    if (n) {
      var i = new Date(new Date(e).getFullYear(), 0).getTime() - 1;
      return (e - i) / 1e3;
    } else {
      var s = new Date(new Date(e).getFullYear() + 1, 0).getTime();
      return (s - e) / 1e3;
    }
  },
  format: function(e, r) {
    return B[r] || (B[r] = {}), B[r].dayMonthYear || (B[r].dayMonthYear = new Intl.DateTimeFormat(r, {
      year: "numeric",
      month: "short",
      day: "numeric"
    })), B[r].dayMonthYear.format(Zn(e));
  }
};
Uu() ? Ft.push(al, ul) : Ft.push({
  formatAs: "day"
}, {
  formatAs: "week"
}, {
  formatAs: "month"
}, {
  formatAs: "year"
});
const le = {
  steps: Ft,
  labels: [
    // "mini" labels are only defined for a few languages.
    "mini",
    // "short-time" labels are only defined for a few languages.
    "short-time",
    // "narrow" and "short" labels are defined for all languages.
    // "narrow" labels can sometimes be weird (like "+5d."),
    // but "short" labels have the " ago" part, so "narrow" seem
    // more appropriate.
    // "short" labels would have been more appropriate if they
    // didn't have the " ago" part, hence the "short-time" above.
    "narrow",
    // Since "narrow" labels are always present, "short" element
    // of this array can be removed.
    "short"
  ]
};
function Ur(t, e) {
  var r = Object.keys(t);
  if (Object.getOwnPropertySymbols) {
    var n = Object.getOwnPropertySymbols(t);
    e && (n = n.filter(function(i) {
      return Object.getOwnPropertyDescriptor(t, i).enumerable;
    })), r.push.apply(r, n);
  }
  return r;
}
function Vr(t) {
  for (var e = 1; e < arguments.length; e++) {
    var r = arguments[e] != null ? arguments[e] : {};
    e % 2 ? Ur(Object(r), !0).forEach(function(n) {
      ll(t, n, r[n]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(r)) : Ur(Object(r)).forEach(function(n) {
      Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(r, n));
    });
  }
  return t;
}
function ll(t, e, r) {
  return e in t ? Object.defineProperty(t, e, { value: r, enumerable: !0, configurable: !0, writable: !0 }) : t[e] = r, t;
}
const cl = Vr(Vr({}, le), {}, {
  // Add "now".
  steps: [{
    formatAs: "now"
  }].concat(le.steps)
});
function Br(t, e) {
  var r = Object.keys(t);
  if (Object.getOwnPropertySymbols) {
    var n = Object.getOwnPropertySymbols(t);
    e && (n = n.filter(function(i) {
      return Object.getOwnPropertyDescriptor(t, i).enumerable;
    })), r.push.apply(r, n);
  }
  return r;
}
function Hr(t) {
  for (var e = 1; e < arguments.length; e++) {
    var r = arguments[e] != null ? arguments[e] : {};
    e % 2 ? Br(Object(r), !0).forEach(function(n) {
      fl(t, n, r[n]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(r)) : Br(Object(r)).forEach(function(n) {
      Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(r, n));
    });
  }
  return t;
}
function fl(t, e, r) {
  return e in t ? Object.defineProperty(t, e, { value: r, enumerable: !0, configurable: !0, writable: !0 }) : t[e] = r, t;
}
const Dt = Hr(Hr({}, le), {}, {
  // Skip "seconds".
  steps: le.steps.filter(function(t) {
    return t.formatAs !== "second";
  })
});
function Wr(t, e) {
  var r = Object.keys(t);
  if (Object.getOwnPropertySymbols) {
    var n = Object.getOwnPropertySymbols(t);
    e && (n = n.filter(function(i) {
      return Object.getOwnPropertyDescriptor(t, i).enumerable;
    })), r.push.apply(r, n);
  }
  return r;
}
function Yr(t) {
  for (var e = 1; e < arguments.length; e++) {
    var r = arguments[e] != null ? arguments[e] : {};
    e % 2 ? Wr(Object(r), !0).forEach(function(n) {
      hl(t, n, r[n]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(r)) : Wr(Object(r)).forEach(function(n) {
      Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(r, n));
    });
  }
  return t;
}
function hl(t, e, r) {
  return e in t ? Object.defineProperty(t, e, { value: r, enumerable: !0, configurable: !0, writable: !0 }) : t[e] = r, t;
}
const pl = Yr(Yr({}, Dt), {}, {
  // Add "now".
  steps: [{
    formatAs: "now"
  }].concat(Dt.steps)
});
function Jr(t, e) {
  var r = Object.keys(t);
  if (Object.getOwnPropertySymbols) {
    var n = Object.getOwnPropertySymbols(t);
    e && (n = n.filter(function(i) {
      return Object.getOwnPropertyDescriptor(t, i).enumerable;
    })), r.push.apply(r, n);
  }
  return r;
}
function Ce(t) {
  for (var e = 1; e < arguments.length; e++) {
    var r = arguments[e] != null ? arguments[e] : {};
    e % 2 ? Jr(Object(r), !0).forEach(function(n) {
      dl(t, n, r[n]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(r)) : Jr(Object(r)).forEach(function(n) {
      Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(r, n));
    });
  }
  return t;
}
function dl(t, e, r) {
  return e in t ? Object.defineProperty(t, e, { value: r, enumerable: !0, configurable: !0, writable: !0 }) : t[e] = r, t;
}
const ml = Ce(Ce({}, le), {}, {
  // Skip "seconds".
  steps: le.steps.filter(function(t) {
    return t.formatAs !== "second";
  }).map(function(t) {
    return t.formatAs === "minute" ? Ce(Ce({}, t), {}, {
      minTime: Y
    }) : t;
  })
}), Pe = {
  steps: [{
    formatAs: "second"
  }, {
    formatAs: "minute"
  }, {
    formatAs: "hour"
  }, {
    formatAs: "day"
  }, {
    formatAs: "month"
  }, {
    formatAs: "year"
  }],
  labels: [
    // "mini" labels are only defined for a few languages.
    "mini",
    // "short-time" labels are only defined for a few languages.
    "short-time",
    // "narrow" and "short" labels are defined for all languages.
    // "narrow" labels can sometimes be weird (like "+5d."),
    // but "short" labels have the " ago" part, so "narrow" seem
    // more appropriate.
    // "short" labels would have been more appropriate if they
    // didn't have the " ago" part, hence the "short-time" above.
    "narrow",
    // Since "narrow" labels are always present, "short" element
    // of this array can be removed.
    "short"
  ]
};
function Qr(t, e) {
  var r = Object.keys(t);
  if (Object.getOwnPropertySymbols) {
    var n = Object.getOwnPropertySymbols(t);
    e && (n = n.filter(function(i) {
      return Object.getOwnPropertyDescriptor(t, i).enumerable;
    })), r.push.apply(r, n);
  }
  return r;
}
function Zr(t) {
  for (var e = 1; e < arguments.length; e++) {
    var r = arguments[e] != null ? arguments[e] : {};
    e % 2 ? Qr(Object(r), !0).forEach(function(n) {
      gl(t, n, r[n]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(r)) : Qr(Object(r)).forEach(function(n) {
      Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(r, n));
    });
  }
  return t;
}
function gl(t, e, r) {
  return e in t ? Object.defineProperty(t, e, { value: r, enumerable: !0, configurable: !0, writable: !0 }) : t[e] = r, t;
}
const yl = Zr(Zr({}, Pe), {}, {
  // Add "now".
  steps: [{
    formatAs: "now"
  }].concat(Pe.steps)
});
function Gr(t, e) {
  var r = Object.keys(t);
  if (Object.getOwnPropertySymbols) {
    var n = Object.getOwnPropertySymbols(t);
    e && (n = n.filter(function(i) {
      return Object.getOwnPropertyDescriptor(t, i).enumerable;
    })), r.push.apply(r, n);
  }
  return r;
}
function Xr(t) {
  for (var e = 1; e < arguments.length; e++) {
    var r = arguments[e] != null ? arguments[e] : {};
    e % 2 ? Gr(Object(r), !0).forEach(function(n) {
      wl(t, n, r[n]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(r)) : Gr(Object(r)).forEach(function(n) {
      Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(r, n));
    });
  }
  return t;
}
function wl(t, e, r) {
  return e in t ? Object.defineProperty(t, e, { value: r, enumerable: !0, configurable: !0, writable: !0 }) : t[e] = r, t;
}
const _t = Xr(Xr({}, Pe), {}, {
  // Skip "seconds".
  steps: Pe.steps.filter(function(t) {
    return t.formatAs !== "second";
  })
});
function Kr(t, e) {
  var r = Object.keys(t);
  if (Object.getOwnPropertySymbols) {
    var n = Object.getOwnPropertySymbols(t);
    e && (n = n.filter(function(i) {
      return Object.getOwnPropertyDescriptor(t, i).enumerable;
    })), r.push.apply(r, n);
  }
  return r;
}
function en(t) {
  for (var e = 1; e < arguments.length; e++) {
    var r = arguments[e] != null ? arguments[e] : {};
    e % 2 ? Kr(Object(r), !0).forEach(function(n) {
      vl(t, n, r[n]);
    }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(r)) : Kr(Object(r)).forEach(function(n) {
      Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(r, n));
    });
  }
  return t;
}
function vl(t, e, r) {
  return e in t ? Object.defineProperty(t, e, { value: r, enumerable: !0, configurable: !0, writable: !0 }) : t[e] = r, t;
}
const bl = en(en({}, _t), {}, {
  // Add "now".
  steps: [{
    formatAs: "now"
  }].concat(_t.steps)
});
function Tl(t) {
  switch (t) {
    // "default" style name is deprecated.
    case "default":
    case "round":
      return Lt;
    case "round-minute":
      return At;
    case "approximate":
      return qr;
    // "time" style name is deprecated.
    case "time":
    case "approximate-time":
      return ol;
    case "mini":
      return Pe;
    case "mini-now":
      return yl;
    case "mini-minute":
      return _t;
    case "mini-minute-now":
      return bl;
    case "twitter":
      return le;
    case "twitter-now":
      return cl;
    case "twitter-minute":
      return Dt;
    case "twitter-minute-now":
      return pl;
    case "twitter-first-minute":
      return ml;
    default:
      return qr;
  }
}
function Xe(t) {
  "@babel/helpers - typeof";
  return Xe = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(e) {
    return typeof e;
  } : function(e) {
    return e && typeof Symbol == "function" && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
  }, Xe(t);
}
function Ol(t, e) {
  var r = typeof Symbol < "u" && t[Symbol.iterator] || t["@@iterator"];
  if (r) return (r = r.call(t)).next.bind(r);
  if (Array.isArray(t) || (r = Gn(t)) || e) {
    r && (t = r);
    var n = 0;
    return function() {
      return n >= t.length ? { done: !0 } : { done: !1, value: t[n++] };
    };
  }
  throw new TypeError(`Invalid attempt to iterate non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`);
}
function Sl(t, e) {
  return Ll(t) || xl(t, e) || Gn(t, e) || kl();
}
function kl() {
  throw new TypeError(`Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`);
}
function Gn(t, e) {
  if (t) {
    if (typeof t == "string") return tn(t, e);
    var r = Object.prototype.toString.call(t).slice(8, -1);
    if (r === "Object" && t.constructor && (r = t.constructor.name), r === "Map" || r === "Set") return Array.from(t);
    if (r === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r)) return tn(t, e);
  }
}
function tn(t, e) {
  (e == null || e > t.length) && (e = t.length);
  for (var r = 0, n = new Array(e); r < e; r++)
    n[r] = t[r];
  return n;
}
function xl(t, e) {
  var r = t == null ? null : typeof Symbol < "u" && t[Symbol.iterator] || t["@@iterator"];
  if (r != null) {
    var n = [], i = !0, s = !1, o, a;
    try {
      for (r = r.call(t); !(i = (o = r.next()).done) && (n.push(o.value), !(e && n.length === e)); i = !0)
        ;
    } catch (l) {
      s = !0, a = l;
    } finally {
      try {
        !i && r.return != null && r.return();
      } finally {
        if (s) throw a;
      }
    }
    return n;
  }
}
function Ll(t) {
  if (Array.isArray(t)) return t;
}
function Al(t, e) {
  if (!(t instanceof e))
    throw new TypeError("Cannot call a class as a function");
}
function Fl(t, e) {
  for (var r = 0; r < e.length; r++) {
    var n = e[r];
    n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), Object.defineProperty(t, n.key, n);
  }
}
function Dl(t, e, r) {
  return e && Fl(t.prototype, e), Object.defineProperty(t, "prototype", { writable: !1 }), t;
}
var $ = /* @__PURE__ */ function() {
  function t() {
    var e = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : [], r = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, n = r.polyfill;
    Al(this, t), typeof e == "string" && (e = [e]), this.locale = qu(e.concat(t.getDefaultLocale()), me), typeof Intl < "u" && Intl.NumberFormat && (this.numberFormat = new Intl.NumberFormat(this.locale)), n === !1 ? (this.IntlRelativeTimeFormat = Intl.RelativeTimeFormat, this.IntlPluralRules = Intl.PluralRules) : (this.IntlRelativeTimeFormat = X, this.IntlPluralRules = X.PluralRules), this.relativeTimeFormatCache = new $r(), this.pluralRulesCache = new $r();
  }
  return Dl(t, [{
    key: "format",
    value: function(r, n, i) {
      i || (n && !Nl(n) ? (i = n, n = void 0) : i = {}), n || (n = At), typeof n == "string" && (n = Tl(n));
      var s = _l(r), o = this.getLabels(n.flavour || n.labels), a = o.labels, l = o.labelsType, c;
      n.now !== void 0 && (c = n.now), c === void 0 && i.now !== void 0 && (c = i.now), c === void 0 && (c = Date.now());
      var f = (c - s) / 1e3, d = i.future || f < 0, g = El(a, me(this.locale).now, me(this.locale).long, d);
      if (n.custom) {
        var w = n.custom({
          now: c,
          date: new Date(s),
          time: s,
          elapsed: f,
          locale: this.locale
        });
        if (w !== void 0)
          return w;
      }
      var F = Pl(
        // Controlling `style.steps` through `style.units` seems to be deprecated:
        // create a new custom `style` instead.
        n.units,
        a,
        g
      ), k = i.round || n.round, P = Ju(
        // "gradation" is a legacy name for "steps".
        // For historical reasons, "approximate" steps are used by default.
        // In the next major version, there'll be no default for `steps`.
        n.gradation || n.steps || At.steps,
        f,
        {
          now: c,
          units: F,
          round: k,
          future: d,
          getNextStep: !0
        }
      ), O = Sl(P, 3), D = O[0], N = O[1], E = O[2], v = this.formatDateForStep(s, N, f, {
        labels: a,
        labelsType: l,
        nowLabel: g,
        now: c,
        future: d,
        round: k
      }) || "";
      if (i.getTimeToNextUpdate) {
        var ee = tl(s, N, {
          nextStep: E,
          prevStep: D,
          now: c,
          future: d,
          round: k
        });
        return [v, ee];
      }
      return v;
    }
  }, {
    key: "formatDateForStep",
    value: function(r, n, i, s) {
      var o = this, a = s.labels, l = s.labelsType, c = s.nowLabel, f = s.now, d = s.future, g = s.round;
      if (n) {
        if (n.format)
          return n.format(r, this.locale, {
            formatAs: function(O, D) {
              return o.formatValue(D, O, {
                labels: a,
                future: d
              });
            },
            now: f,
            future: d
          });
        var w = n.unit || n.formatAs;
        if (!w)
          throw new Error("[javascript-time-ago] Each step must define either `formatAs` or `format()`. Step: ".concat(JSON.stringify(n)));
        if (w === "now")
          return c;
        var F = Math.abs(i) / Vn(n);
        n.granularity && (F = Re(g)(F / n.granularity) * n.granularity);
        var k = -1 * Math.sign(i) * Re(g)(F);
        switch (k === 0 && (d ? k = 0 : k = -0), l) {
          case "long":
          case "short":
          case "narrow":
            return this.getFormatter(l).format(k, w);
          default:
            return this.formatValue(k, w, {
              labels: a,
              future: d
            });
        }
      }
    }
    /**
     * Mimicks what `Intl.RelativeTimeFormat` does for additional locale styles.
     * @param  {number} value
     * @param  {string} unit
     * @param  {object} options.labels — Relative time labels.
     * @param  {boolean} [options.future] — Tells how to format value `0`: as "future" (`true`) or "past" (`false`). Is `false` by default, but should have been `true` actually.
     * @return {string}
     */
  }, {
    key: "formatValue",
    value: function(r, n, i) {
      var s = i.labels, o = i.future;
      return this.getFormattingRule(s, n, r, {
        future: o
      }).replace("{0}", this.formatNumber(Math.abs(r)));
    }
    /**
     * Returns formatting rule for `value` in `units` (either in past or in future).
     * @param {object} formattingRules — Relative time labels for different units.
     * @param {string} unit - Time interval measurement unit.
     * @param {number} value - Time interval value.
     * @param  {boolean} [options.future] — Tells how to format value `0`: as "future" (`true`) or "past" (`false`). Is `false` by default.
     * @return {string}
     * @example
     * // Returns "{0} days ago"
     * getFormattingRule(en.long, "day", -2, 'en')
     */
  }, {
    key: "getFormattingRule",
    value: function(r, n, i, s) {
      var o = s.future;
      if (this.locale, r = r[n], typeof r == "string")
        return r;
      var a = i === 0 ? o ? "future" : "past" : i < 0 ? "past" : "future", l = r[a] || r;
      if (typeof l == "string")
        return l;
      var c = this.getPluralRules().select(Math.abs(i));
      return l[c] || l.other;
    }
    /**
     * Formats a number into a string.
     * Uses `Intl.NumberFormat` when available.
     * @param  {number} number
     * @return {string}
     */
  }, {
    key: "formatNumber",
    value: function(r) {
      return this.numberFormat ? this.numberFormat.format(r) : String(r);
    }
    /**
     * Returns an `Intl.RelativeTimeFormat` for a given `labelsType`.
     * @param {string} labelsType
     * @return {object} `Intl.RelativeTimeFormat` instance
     */
  }, {
    key: "getFormatter",
    value: function(r) {
      return this.relativeTimeFormatCache.get(this.locale, r) || this.relativeTimeFormatCache.put(this.locale, r, new this.IntlRelativeTimeFormat(this.locale, {
        style: r
      }));
    }
    /**
     * Returns an `Intl.PluralRules` instance.
     * @return {object} `Intl.PluralRules` instance
     */
  }, {
    key: "getPluralRules",
    value: function() {
      return this.pluralRulesCache.get(this.locale) || this.pluralRulesCache.put(this.locale, new this.IntlPluralRules(this.locale));
    }
    /**
     * Gets localized labels for this type of labels.
     *
     * @param {(string|string[])} labelsType - Relative date/time labels type.
     *                                     If it's an array then all label types are tried
     *                                     until a suitable one is found.
     *
     * @returns {Object} Returns an object of shape { labelsType, labels }
     */
  }, {
    key: "getLabels",
    value: function() {
      var r = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : [];
      typeof r == "string" && (r = [r]), r = r.map(function(a) {
        switch (a) {
          case "tiny":
          case "mini-time":
            return "mini";
          default:
            return a;
        }
      }), r = r.concat("long");
      for (var n = me(this.locale), i = Ol(r), s; !(s = i()).done; ) {
        var o = s.value;
        if (n[o])
          return {
            labelsType: o,
            labels: n[o]
          };
      }
    }
  }]), t;
}(), Xn = "en";
$.getDefaultLocale = function() {
  return Xn;
};
$.setDefaultLocale = function(t) {
  return Xn = t;
};
$.addDefaultLocale = function(t) {
  if (rn)
    return console.error("[javascript-time-ago] `TimeAgo.addDefaultLocale()` can only be called once. To add other locales, use `TimeAgo.addLocale()`.");
  rn = !0, $.setDefaultLocale(t.locale), $.addLocale(t);
};
var rn;
$.addLocale = function(t) {
  Jn(t), X.addLocale(t);
};
$.locale = $.addLocale;
$.addLabels = function(t, e, r) {
  var n = me(t);
  n || (Jn({
    locale: t
  }), n = me(t)), n[e] = r;
};
function _l(t) {
  if (t.constructor === Date || Rl(t))
    return t.getTime();
  if (typeof t == "number")
    return t;
  throw new Error("Unsupported relative time formatter input: ".concat(Xe(t), ", ").concat(t));
}
function Rl(t) {
  return Xe(t) === "object" && typeof t.getTime == "function";
}
function Pl(t, e, r) {
  var n = Object.keys(e);
  return r && n.push("now"), t && (n = t.filter(function(i) {
    return i === "now" || n.indexOf(i) >= 0;
  })), n;
}
function El(t, e, r, n) {
  var i = t.now || e && e.now;
  if (i)
    return typeof i == "string" ? i : n ? i.future : i.past;
  if (r && r.second && r.second.current)
    return r.second.current;
}
function Nl(t) {
  return typeof t == "string" || Vu(t);
}
const Kn = {
  locale: "en",
  long: {
    year: {
      previous: "last year",
      current: "this year",
      next: "next year",
      past: {
        one: "{0} year ago",
        other: "{0} years ago"
      },
      future: {
        one: "in {0} year",
        other: "in {0} years"
      }
    },
    quarter: {
      previous: "last quarter",
      current: "this quarter",
      next: "next quarter",
      past: {
        one: "{0} quarter ago",
        other: "{0} quarters ago"
      },
      future: {
        one: "in {0} quarter",
        other: "in {0} quarters"
      }
    },
    month: {
      previous: "last month",
      current: "this month",
      next: "next month",
      past: {
        one: "{0} month ago",
        other: "{0} months ago"
      },
      future: {
        one: "in {0} month",
        other: "in {0} months"
      }
    },
    week: {
      previous: "last week",
      current: "this week",
      next: "next week",
      past: {
        one: "{0} week ago",
        other: "{0} weeks ago"
      },
      future: {
        one: "in {0} week",
        other: "in {0} weeks"
      }
    },
    day: {
      previous: "yesterday",
      current: "today",
      next: "tomorrow",
      past: {
        one: "{0} day ago",
        other: "{0} days ago"
      },
      future: {
        one: "in {0} day",
        other: "in {0} days"
      }
    },
    hour: {
      current: "this hour",
      past: {
        one: "{0} hour ago",
        other: "{0} hours ago"
      },
      future: {
        one: "in {0} hour",
        other: "in {0} hours"
      }
    },
    minute: {
      current: "this minute",
      past: {
        one: "{0} minute ago",
        other: "{0} minutes ago"
      },
      future: {
        one: "in {0} minute",
        other: "in {0} minutes"
      }
    },
    second: {
      current: "now",
      past: {
        one: "{0} second ago",
        other: "{0} seconds ago"
      },
      future: {
        one: "in {0} second",
        other: "in {0} seconds"
      }
    }
  },
  short: {
    year: {
      previous: "last yr.",
      current: "this yr.",
      next: "next yr.",
      past: "{0} yr. ago",
      future: "in {0} yr."
    },
    quarter: {
      previous: "last qtr.",
      current: "this qtr.",
      next: "next qtr.",
      past: {
        one: "{0} qtr. ago",
        other: "{0} qtrs. ago"
      },
      future: {
        one: "in {0} qtr.",
        other: "in {0} qtrs."
      }
    },
    month: {
      previous: "last mo.",
      current: "this mo.",
      next: "next mo.",
      past: "{0} mo. ago",
      future: "in {0} mo."
    },
    week: {
      previous: "last wk.",
      current: "this wk.",
      next: "next wk.",
      past: "{0} wk. ago",
      future: "in {0} wk."
    },
    day: {
      previous: "yesterday",
      current: "today",
      next: "tomorrow",
      past: {
        one: "{0} day ago",
        other: "{0} days ago"
      },
      future: {
        one: "in {0} day",
        other: "in {0} days"
      }
    },
    hour: {
      current: "this hour",
      past: "{0} hr. ago",
      future: "in {0} hr."
    },
    minute: {
      current: "this minute",
      past: "{0} min. ago",
      future: "in {0} min."
    },
    second: {
      current: "now",
      past: "{0} sec. ago",
      future: "in {0} sec."
    }
  },
  narrow: {
    year: {
      previous: "last yr.",
      current: "this yr.",
      next: "next yr.",
      past: "{0}y ago",
      future: "in {0}y"
    },
    quarter: {
      previous: "last qtr.",
      current: "this qtr.",
      next: "next qtr.",
      past: "{0}q ago",
      future: "in {0}q"
    },
    month: {
      previous: "last mo.",
      current: "this mo.",
      next: "next mo.",
      past: "{0}mo ago",
      future: "in {0}mo"
    },
    week: {
      previous: "last wk.",
      current: "this wk.",
      next: "next wk.",
      past: "{0}w ago",
      future: "in {0}w"
    },
    day: {
      previous: "yesterday",
      current: "today",
      next: "tomorrow",
      past: "{0}d ago",
      future: "in {0}d"
    },
    hour: {
      current: "this hour",
      past: "{0}h ago",
      future: "in {0}h"
    },
    minute: {
      current: "this minute",
      past: "{0}m ago",
      future: "in {0}m"
    },
    second: {
      current: "now",
      past: "{0}s ago",
      future: "in {0}s"
    }
  },
  now: {
    now: {
      current: "now",
      future: "in a moment",
      past: "just now"
    }
  },
  mini: {
    year: "{0}yr",
    month: "{0}mo",
    week: "{0}wk",
    day: "{0}d",
    hour: "{0}h",
    minute: "{0}m",
    second: "{0}s",
    now: "now"
  },
  "short-time": {
    year: "{0} yr.",
    month: "{0} mo.",
    week: "{0} wk.",
    day: {
      one: "{0} day",
      other: "{0} days"
    },
    hour: "{0} hr.",
    minute: "{0} min.",
    second: "{0} sec."
  },
  "long-time": {
    year: {
      one: "{0} year",
      other: "{0} years"
    },
    month: {
      one: "{0} month",
      other: "{0} months"
    },
    week: {
      one: "{0} week",
      other: "{0} weeks"
    },
    day: {
      one: "{0} day",
      other: "{0} days"
    },
    hour: {
      one: "{0} hour",
      other: "{0} hours"
    },
    minute: {
      one: "{0} minute",
      other: "{0} minutes"
    },
    second: {
      one: "{0} second",
      other: "{0} seconds"
    }
  }
};
var $l = ["date", "verboseDate", "tooltip", "children"];
function Rt() {
  return Rt = Object.assign || function(t) {
    for (var e = 1; e < arguments.length; e++) {
      var r = arguments[e];
      for (var n in r)
        Object.prototype.hasOwnProperty.call(r, n) && (t[n] = r[n]);
    }
    return t;
  }, Rt.apply(this, arguments);
}
function Il(t, e) {
  if (t == null) return {};
  var r = jl(t, e), n, i;
  if (Object.getOwnPropertySymbols) {
    var s = Object.getOwnPropertySymbols(t);
    for (i = 0; i < s.length; i++)
      n = s[i], !(e.indexOf(n) >= 0) && Object.prototype.propertyIsEnumerable.call(t, n) && (r[n] = t[n]);
  }
  return r;
}
function jl(t, e) {
  if (t == null) return {};
  var r = {}, n = Object.keys(t), i, s;
  for (s = 0; s < n.length; s++)
    i = n[s], !(e.indexOf(i) >= 0) && (r[i] = t[i]);
  return r;
}
function Ke(t, e) {
  var r = t.date, n = t.verboseDate, i = t.tooltip, s = t.children, o = Il(t, $l), a = ne(function() {
    return r.toISOString();
  }, [r]);
  return /* @__PURE__ */ De.createElement("time", Rt({
    ref: e
  }, o, {
    dateTime: a,
    title: i ? n : void 0
  }), s);
}
Ke = /* @__PURE__ */ De.forwardRef(Ke);
Ke.propTypes = {
  date: p.instanceOf(Date).isRequired,
  verboseDate: p.string,
  tooltip: p.bool.isRequired,
  children: p.string.isRequired
};
var nn = Number.isNaN || function(e) {
  return typeof e == "number" && e !== e;
};
function zl(t, e) {
  return !!(t === e || nn(t) && nn(e));
}
function Ml(t, e) {
  if (t.length !== e.length)
    return !1;
  for (var r = 0; r < t.length; r++)
    if (!zl(t[r], e[r]))
      return !1;
  return !0;
}
function Cl(t, e) {
  e === void 0 && (e = Ml);
  var r = null;
  function n() {
    for (var i = [], s = 0; s < arguments.length; s++)
      i[s] = arguments[s];
    if (r && r.lastThis === this && e(i, r.lastArgs))
      return r.lastResult;
    var o = t.apply(this, i);
    return r = {
      lastResult: o,
      lastArgs: i,
      lastThis: this
    }, o;
  }
  return n.clear = function() {
    r = null;
  }, n;
}
function Pt(t) {
  "@babel/helpers - typeof";
  return Pt = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(e) {
    return typeof e;
  } : function(e) {
    return e && typeof Symbol == "function" && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
  }, Pt(t);
}
function ql(t) {
  if (ei())
    return Intl.DateTimeFormat.supportedLocalesOf(t)[0];
}
function ei() {
  var t = (typeof Intl > "u" ? "undefined" : Pt(Intl)) === "object";
  return t && typeof Intl.DateTimeFormat == "function";
}
function et(t) {
  "@babel/helpers - typeof";
  return et = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(e) {
    return typeof e;
  } : function(e) {
    return e && typeof Symbol == "function" && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
  }, et(t);
}
function Ul(t, e) {
  if (!(t instanceof e))
    throw new TypeError("Cannot call a class as a function");
}
function Vl(t, e) {
  for (var r = 0; r < e.length; r++) {
    var n = e[r];
    n.enumerable = n.enumerable || !1, n.configurable = !0, "value" in n && (n.writable = !0), Object.defineProperty(t, n.key, n);
  }
}
function Bl(t, e, r) {
  return e && Vl(t.prototype, e), Object.defineProperty(t, "prototype", { writable: !1 }), t;
}
var ti = /* @__PURE__ */ function() {
  function t() {
    Ul(this, t), this.cache = {};
  }
  return Bl(t, [{
    key: "get",
    value: function() {
      for (var r = this.cache, n = arguments.length, i = new Array(n), s = 0; s < n; s++)
        i[s] = arguments[s];
      for (var o = 0, a = i; o < a.length; o++) {
        var l = a[o];
        if (et(r) !== "object")
          return;
        r = r[l];
      }
      return r;
    }
  }, {
    key: "put",
    value: function() {
      for (var r = arguments.length, n = new Array(r), i = 0; i < r; i++)
        n[i] = arguments[i];
      for (var s = n.pop(), o = n.pop(), a = this.cache, l = 0, c = n; l < c.length; l++) {
        var f = c[l];
        et(a[f]) !== "object" && (a[f] = {}), a = a[f];
      }
      return a[o] = s;
    }
  }]), t;
}(), sn = new ti(), Hl = ei(), Wl = function(e) {
  return e.toString();
};
function Yl(t, e) {
  if (!Hl)
    return Wl;
  var r = Ql(t), n = JSON.stringify(e), i = sn.get(String(r), n) || sn.put(String(r), n, new Intl.DateTimeFormat(r, e));
  return function(s) {
    return i.format(s);
  };
}
const Jl = Cl(Yl);
var pt = {};
function Ql(t) {
  var e = t.toString();
  return pt[e] ? pt[e] : pt[e] = ql(t);
}
function Et(t) {
  "@babel/helpers - typeof";
  return Et = typeof Symbol == "function" && typeof Symbol.iterator == "symbol" ? function(e) {
    return typeof e;
  } : function(e) {
    return e && typeof Symbol == "function" && e.constructor === Symbol && e !== Symbol.prototype ? "symbol" : typeof e;
  }, Et(t);
}
function Zl(t) {
  return Gl(t) ? t : new Date(t);
}
function Gl(t) {
  return t instanceof Date || Xl(t);
}
function Xl(t) {
  return Et(t) === "object" && typeof t.getTime == "function";
}
var on = new ti();
function Kl(t, e) {
  var r = e.polyfill, n = String(t) + ":" + String(r);
  return on.get(n) || on.put(n, new $(t, {
    polyfill: r
  }));
}
var ot = { exports: {} }, Ae = { exports: {} };
(function() {
  var t, e, r, n, i, s;
  typeof performance < "u" && performance !== null && performance.now ? Ae.exports = function() {
    return performance.now();
  } : typeof process < "u" && process !== null && process.hrtime ? (Ae.exports = function() {
    return (t() - i) / 1e6;
  }, e = process.hrtime, t = function() {
    var o;
    return o = e(), o[0] * 1e9 + o[1];
  }, n = t(), s = process.uptime() * 1e9, i = n - s) : Date.now ? (Ae.exports = function() {
    return Date.now() - r;
  }, r = Date.now()) : (Ae.exports = function() {
    return (/* @__PURE__ */ new Date()).getTime() - r;
  }, r = (/* @__PURE__ */ new Date()).getTime());
}).call(dn);
var ec = Ae.exports, tc = ec, J = typeof window > "u" ? dn : window, qe = ["moz", "webkit"], we = "AnimationFrame", Te = J["request" + we], Ee = J["cancel" + we] || J["cancelRequest" + we];
for (var Le = 0; !Te && Le < qe.length; Le++)
  Te = J[qe[Le] + "Request" + we], Ee = J[qe[Le] + "Cancel" + we] || J[qe[Le] + "CancelRequest" + we];
if (!Te || !Ee) {
  var dt = 0, an = 0, re = [], rc = 1e3 / 60;
  Te = function(t) {
    if (re.length === 0) {
      var e = tc(), r = Math.max(0, rc - (e - dt));
      dt = r + e, setTimeout(function() {
        var n = re.slice(0);
        re.length = 0;
        for (var i = 0; i < n.length; i++)
          if (!n[i].cancelled)
            try {
              n[i].callback(dt);
            } catch (s) {
              setTimeout(function() {
                throw s;
              }, 0);
            }
      }, Math.round(r));
    }
    return re.push({
      handle: ++an,
      callback: t,
      cancelled: !1
    }), an;
  }, Ee = function(t) {
    for (var e = 0; e < re.length; e++)
      re[e].handle === t && (re[e].cancelled = !0);
  };
}
ot.exports = function(t) {
  return Te.call(J, t);
};
ot.exports.cancel = function() {
  Ee.apply(J, arguments);
};
ot.exports.polyfill = function(t) {
  t || (t = J), t.requestAnimationFrame = Te, t.cancelAnimationFrame = Ee;
};
var nc = ot.exports;
const un = /* @__PURE__ */ vi(nc);
function ic(t, e) {
  if (t.length === 0)
    return 0;
  for (var r = 0, n = t.length - 1, i; r <= n; ) {
    i = Math.floor((n + r) / 2);
    var s = e(t[i]);
    if (s === 0)
      return i;
    if (s < 0) {
      if (r = i + 1, r > n)
        return r;
    } else if (n = i - 1, n < r)
      return r;
  }
}
function sc(t, e) {
  return lc(t) || uc(t, e) || ac(t, e) || oc();
}
function oc() {
  throw new TypeError(`Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`);
}
function ac(t, e) {
  if (t) {
    if (typeof t == "string") return ln(t, e);
    var r = Object.prototype.toString.call(t).slice(8, -1);
    if (r === "Object" && t.constructor && (r = t.constructor.name), r === "Map" || r === "Set") return Array.from(t);
    if (r === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r)) return ln(t, e);
  }
}
function ln(t, e) {
  (e == null || e > t.length) && (e = t.length);
  for (var r = 0, n = new Array(e); r < e; r++)
    n[r] = t[r];
  return n;
}
function uc(t, e) {
  var r = t == null ? null : typeof Symbol < "u" && t[Symbol.iterator] || t["@@iterator"];
  if (r != null) {
    var n = [], i = !0, s = !1, o, a;
    try {
      for (r = r.call(t); !(i = (o = r.next()).done) && (n.push(o.value), !(e && n.length === e)); i = !0)
        ;
    } catch (l) {
      s = !0, a = l;
    } finally {
      try {
        !i && r.return != null && r.return();
      } finally {
        if (s) throw a;
      }
    }
    return n;
  }
}
function lc(t) {
  if (Array.isArray(t)) return t;
}
const cc = {
  instances: [],
  add: function(e) {
    var r = this, n = this.instances.length === 0;
    return ri(this.instances, e), n && this.start(), {
      stop: function() {
        ni(r.instances, e), r.instances.length === 0 && r.stop();
      },
      forceUpdate: function() {
        cn(e, r.instances);
      }
    };
  },
  tick: function() {
    for (var e = Date.now(); ; ) {
      var r = this.instances[0];
      if (e >= r.nextUpdateTime)
        cn(r, this.instances);
      else
        break;
    }
  },
  scheduleNextTick: function() {
    var e = this;
    this.scheduledTick = un(function() {
      e.tick(), e.scheduleNextTick();
    });
  },
  start: function() {
    this.scheduleNextTick();
  },
  stop: function() {
    un.cancel(this.scheduledTick);
  }
};
function fc(t) {
  var e = t.getNextValue(), r = sc(e, 2), n = r[0], i = r[1];
  t.setValue(n), t.nextUpdateTime = i;
}
function cn(t, e) {
  fc(t), ni(e, t), ri(e, t);
}
function ri(t, e) {
  var r = hc(t, e);
  t.splice(r, 0, e);
}
function ni(t, e) {
  var r = t.indexOf(e);
  t.splice(r, 1);
}
function hc(t, e) {
  var r = e.nextUpdateTime;
  return ic(t, function(n) {
    return n.nextUpdateTime === r ? 0 : n.nextUpdateTime > r ? 1 : -1;
  });
}
function Ue(t, e) {
  return gc(t) || mc(t, e) || dc(t, e) || pc();
}
function pc() {
  throw new TypeError(`Invalid attempt to destructure non-iterable instance.
In order to be iterable, non-array objects must have a [Symbol.iterator]() method.`);
}
function dc(t, e) {
  if (t) {
    if (typeof t == "string") return fn(t, e);
    var r = Object.prototype.toString.call(t).slice(8, -1);
    if (r === "Object" && t.constructor && (r = t.constructor.name), r === "Map" || r === "Set") return Array.from(t);
    if (r === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(r)) return fn(t, e);
  }
}
function fn(t, e) {
  (e == null || e > t.length) && (e = t.length);
  for (var r = 0, n = new Array(e); r < e; r++)
    n[r] = t[r];
  return n;
}
function mc(t, e) {
  var r = t == null ? null : typeof Symbol < "u" && t[Symbol.iterator] || t["@@iterator"];
  if (r != null) {
    var n = [], i = !0, s = !1, o, a;
    try {
      for (r = r.call(t); !(i = (o = r.next()).done) && (n.push(o.value), !(e && n.length === e)); i = !0)
        ;
    } catch (l) {
      s = !0, a = l;
    } finally {
      try {
        !i && r.return != null && r.return();
      } finally {
        if (s) throw a;
      }
    }
    return n;
  }
}
function gc(t) {
  if (Array.isArray(t)) return t;
}
function yc(t) {
  var e = t.date, r = t.future, n = t.locale, i = t.locales, s = t.timeStyle, o = t.round, a = t.minTimeLeft, l = t.formatVerboseDate, c = t.verboseDateFormat, f = c === void 0 ? bc : c, d = t.updateInterval, g = t.tick, w = g === void 0 ? !0 : g, F = t.now, k = t.timeOffset, P = k === void 0 ? 0 : k, O = t.polyfill, D = ne(function() {
    return n && (i = [n]), i.concat($.getDefaultLocale());
  }, [n, i]), N = ne(function() {
    return Kl(D, {
      polyfill: O
    });
  }, [D, O]);
  e = ne(function() {
    return Zl(e);
  }, [e]);
  var E = gi(function() {
    var W = (F || Date.now()) - P, xe;
    if (r && W >= e.getTime() && (W = e.getTime(), xe = !0), a !== void 0) {
      var ze = e.getTime() - a * 1e3;
      W > ze && (W = ze, xe = !0);
    }
    var si = N.format(e, s, {
      getTimeToNextUpdate: !0,
      now: W,
      future: r,
      round: o
    }), Xt = Ue(si, 2), oi = Xt[0], Me = Xt[1];
    return xe ? Me = vc : Me = d || Me || 60 * 1e3, [oi, W + Me];
  }, [e, r, s, d, o, a, N, F]), v = Kt();
  v.current = E;
  var ee = ne(E, []), pe = Ue(ee, 2), Se = pe[0], at = pe[1], Ne = yi(Se), $e = Ue(Ne, 2), Ie = $e[0], ke = $e[1], je = Kt();
  er(function() {
    if (w)
      return je.current = cc.add({
        getNextValue: function() {
          return v.current();
        },
        setValue: ke,
        nextUpdateTime: at
      }), function() {
        return je.current.stop();
      };
  }, [w]), er(function() {
    if (je.current)
      je.current.forceUpdate();
    else {
      var W = E(), xe = Ue(W, 1), ze = xe[0];
      ke(ze);
    }
  }, [E]);
  var Gt = ne(function() {
    return Jl(D, f);
  }, [D, f]), ii = ne(function() {
    return l ? l(e) : Gt(e);
  }, [e, l, Gt]);
  return {
    date: e,
    formattedDate: Ie,
    verboseDate: ii
  };
}
var wc = 365 * 24 * 60 * 60 * 1e3, vc = 1e3 * wc, bc = {
  weekday: "long",
  day: "numeric",
  month: "long",
  year: "numeric",
  hour: "numeric",
  minute: "2-digit",
  second: "2-digit"
  // timeZoneName: 'short'
}, Nt = p.oneOfType, hn = p.arrayOf, ge = p.string, pn = p.number, Fe = p.shape, Ve = p.func, Tc = Nt([Fe({
  minTime: pn,
  formatAs: ge.isRequired
}), Fe({
  test: Ve,
  formatAs: ge.isRequired
}), Fe({
  minTime: pn,
  format: Ve.isRequired
}), Fe({
  test: Ve,
  format: Ve.isRequired
})]), Oc = Nt([
  // Not using `oneOf()` here, because that way
  // this package wouldn't support some hypothetical
  // new styles added to `javascript-time-ago` in some future.
  ge,
  Fe({
    steps: hn(Tc).isRequired,
    labels: Nt([ge, hn(ge)]).isRequired,
    round: ge
  })
]), Sc = ["date", "future", "timeStyle", "round", "minTimeLeft", "locale", "locales", "formatVerboseDate", "verboseDateFormat", "updateInterval", "tick", "now", "timeOffset", "polyfill", "tooltip", "component", "container", "wrapperComponent", "wrapperProps"];
function tt() {
  return tt = Object.assign || function(t) {
    for (var e = 1; e < arguments.length; e++) {
      var r = arguments[e];
      for (var n in r)
        Object.prototype.hasOwnProperty.call(r, n) && (t[n] = r[n]);
    }
    return t;
  }, tt.apply(this, arguments);
}
function kc(t, e) {
  if (t == null) return {};
  var r = xc(t, e), n, i;
  if (Object.getOwnPropertySymbols) {
    var s = Object.getOwnPropertySymbols(t);
    for (i = 0; i < s.length; i++)
      n = s[i], !(e.indexOf(n) >= 0) && Object.prototype.propertyIsEnumerable.call(t, n) && (r[n] = t[n]);
  }
  return r;
}
function xc(t, e) {
  if (t == null) return {};
  var r = {}, n = Object.keys(t), i, s;
  for (s = 0; s < n.length; s++)
    i = n[s], !(e.indexOf(i) >= 0) && (r[i] = t[i]);
  return r;
}
function rt(t) {
  var e = t.date, r = t.future, n = t.timeStyle, i = t.round, s = t.minTimeLeft, o = t.locale, a = t.locales, l = a === void 0 ? [] : a, c = t.formatVerboseDate, f = t.verboseDateFormat, d = t.updateInterval, g = t.tick, w = t.now, F = t.timeOffset, k = t.polyfill, P = t.tooltip, O = P === void 0 ? !0 : P, D = t.component, N = D === void 0 ? Ke : D, E = t.container, v = t.wrapperComponent, ee = t.wrapperProps, pe = kc(t, Sc), Se = yc({
    date: e,
    future: r,
    timeStyle: n,
    round: i,
    minTimeLeft: s,
    locale: o,
    locales: l,
    formatVerboseDate: c,
    verboseDateFormat: f,
    updateInterval: d,
    tick: g,
    now: w,
    timeOffset: F,
    polyfill: k
  }), at = Se.date, Ne = Se.verboseDate, $e = Se.formattedDate, Ie = /* @__PURE__ */ De.createElement(N, tt({
    date: at,
    verboseDate: Ne,
    tooltip: O
  }, pe), $e), ke = v || E;
  return ke ? /* @__PURE__ */ De.createElement(ke, tt({}, ee, {
    verboseDate: Ne
  }), Ie) : Ie;
}
rt.propTypes = {
  // `date: Date` or `timestamp: number`.
  // E.g. `new Date()` or `1355972400000`.
  date: p.oneOfType([p.instanceOf(Date), p.number]).isRequired,
  // Preferred locale.
  // Is 'en' by default.
  // E.g. 'ru-RU'.
  locale: p.string,
  // Alternatively to `locale`, one could pass `locales`:
  // A list of preferred locales (ordered).
  // Will choose the first supported locale from the list.
  // E.g. `['ru-RU', 'en-GB']`.
  locales: p.arrayOf(p.string),
  // If set to `true`, then will stop at "zero point"
  // when going from future dates to past dates.
  // In other words, even if the `date` has passed,
  // it will still render as if `date` is `now`.
  future: p.bool,
  // Date/time formatting style.
  // See `javascript-time-ago` docs on "Styles" for more info.
  // E.g. 'round', 'round-minute', 'twitter', 'twitter-first-minute'.
  timeStyle: Oc,
  // `round` parameter of `javascript-time-ago`.
  // See `javascript-time-ago` docs on "Rounding" for more info.
  // Examples: "round", "floor".
  round: p.string,
  // If specified, the time won't "tick" past this threshold (in seconds).
  // For example, if `minTimeLeft` is `60 * 60`
  // then the time won't "tick" past "in 1 hour".
  minTimeLeft: p.number,
  // A React component to render the relative time label.
  // Receives properties:
  // * date: Date — The date.
  // * verboseDate: string — Formatted verbose date.
  // * tooltip: boolean — The `tooltip` property of `<ReactTimeAgo/>` component.
  // * children: string — The relative time label.
  // * All "unknown" properties that have been passed to `<ReactTimeAgo/>` are passed through to this component.
  component: p.elementType,
  // Whether to use HTML `tooltip` attribute to show a verbose date tooltip.
  // Is `true` by default.
  // Can be set to `false` to disable the native HTML `tooltip`.
  tooltip: p.bool,
  // Verbose date formatter.
  // By default it's `(date) => new Intl.DateTimeFormat(locale, {…}).format(date)`.
  formatVerboseDate: p.func,
  // `Intl.DateTimeFormat` format for formatting verbose date.
  // See `Intl.DateTimeFormat` docs for more info.
  verboseDateFormat: p.object,
  // (deprecated)
  // How often the component refreshes itself.
  // When not provided, will use `getNextTimeToUpdate()` feature
  // of `javascript-time-ago` styles to determine the update interval.
  updateInterval: p.oneOfType([p.number, p.arrayOf(p.shape({
    threshold: p.number,
    interval: p.number.isRequired
  }))]),
  // (deprecated).
  // Set to `false` to disable automatic refresh of the component.
  // Is `true` by default.
  // I guess no one actually turns auto-update off, so this parameter is deprecated.
  tick: p.bool,
  // Allows setting a custom baseline for relative time measurement.
  // https://gitlab.com/catamphetamine/react-time-ago/-/issues/4
  now: p.number,
  // Allows offsetting the `date` by an arbitrary amount of milliseconds.
  // https://gitlab.com/catamphetamine/react-time-ago/-/issues/4
  timeOffset: p.number,
  // Pass `false` to use native `Intl.RelativeTimeFormat` / `Intl.PluralRules`
  // instead of the polyfilled ones in `javascript-time-ago`.
  polyfill: p.bool,
  // (advanced)
  // A React Component to wrap the resulting `<time/>` React Element.
  // Receives `verboseDate` and `children` properties.
  // Also receives `wrapperProps`, if they're passed.
  // `verboseDate` can be used for displaying verbose date label
  // in an "on mouse over" (or "on touch") tooltip.
  // See the "Tooltip" readme section for more info.
  // Another example could be having `wrapperComponent`
  // being rerendered every time the component refreshes itself.
  wrapperComponent: p.elementType,
  // Custom `props` passed to `wrapperComponent`.
  wrapperProps: p.object
};
rt = /* @__PURE__ */ De.memo(rt);
$.addDefaultLocale(Kn);
$.addLocale(Kn);
const Lc = hi.div`
  cursor: ${(t) => t.$redirect ? "pointer" : "default"};

  &:hover {
    background: ${(t) => t.$hoverBackground} !important;
    border-radius: 8px !important;
  }

  & .notification-archive-button {
    visibility: hidden;
    transition: visibility 0s;
  }

  &:hover .notification-archive-button {
    visibility: ${(t) => t.$archived ? "hidden" : "visible"};
  }
`, jc = (t) => {
  var O, D, N, E;
  const e = mi(), r = pi(e);
  if (t.renderer)
    return t.renderer(t.notifications);
  const n = new Jt(), i = t.notifications.length, s = t.notifications[i - 1], o = (D = (O = s.deliveryOptions) == null ? void 0 : O.instant) != null && D.batching ? (N = s.template) == null ? void 0 : N.batch : ((E = s.template) == null ? void 0 : E.instant) ?? {
    title: s.title,
    redirectURL: s.redirectURL,
    imageURL: s.imageURL
  };
  let a = { _items: [] };
  t.notifications.forEach((v) => {
    a = {
      ...a,
      ...v.parameters
    };
  }), a._items = t.notifications.map((v) => ({ ...v.parameters, _items: void 0 }));
  const l = n.parseAndRenderSync((o == null ? void 0 : o.title) ?? "", a), c = n.parseAndRenderSync(
    (o == null ? void 0 : o.redirectURL) ?? "",
    a
  ), f = n.parseAndRenderSync(
    (o == null ? void 0 : o.imageURL) ?? "",
    a
  ), d = t.notifications.every((v) => v.seen), g = t.notifications.every((v) => v.opened), w = t.notifications.every((v) => v.archived), F = t.notifications[i - 1].date, k = t.notifications.map((v) => v.id), P = e.palette.mode === "dark" ? "rgba(255, 255, 255, 0.08)" : "rgba(0, 0, 0, 0.04)";
  return /* @__PURE__ */ ut(
    Lc,
    {
      $redirect: !!c,
      $seen: d || !!g,
      $archived: !!w,
      $hoverBackground: P,
      onClick: () => {
        t.markAsClicked(k), c && (t.newTab ? window.open(c, "_blank") : window.location.href = c);
      },
      style: {
        padding: "16px 6px 16px 0",
        background: r.paper,
        color: r.text,
        position: "relative",
        display: "flex",
        alignItems: "center",
        width: "100%"
      },
      children: [
        /* @__PURE__ */ _("div", { children: /* @__PURE__ */ _(
          gn,
          {
            src: f,
            sizes: "32",
            style: {
              marginRight: 8,
              marginLeft: 12
            },
            variant: t.imageShape === "circle" ? "circular" : "rounded"
          }
        ) }),
        /* @__PURE__ */ ut(
          "div",
          {
            style: {
              flexGrow: 1
            },
            children: [
              /* @__PURE__ */ _("div", { children: /* @__PURE__ */ _(
                tr,
                {
                  variant: "body2",
                  fontWeight: w ? 300 : 400,
                  style: {
                    whiteSpace: "pre-line",
                    color: r.text
                  },
                  children: /* @__PURE__ */ _("span", { dangerouslySetInnerHTML: { __html: l } })
                }
              ) }),
              /* @__PURE__ */ _("div", { children: /* @__PURE__ */ _(
                tr,
                {
                  variant: "body2",
                  fontWeight: 300,
                  style: {
                    fontSize: 12,
                    color: r.textSecondary
                  },
                  children: /* @__PURE__ */ _(rt, { date: new Date(F).getTime(), locale: "en-US" })
                }
              ) })
            ]
          }
        ),
        /* @__PURE__ */ ut(
          "div",
          {
            style: {
              position: "relative",
              minWidth: 52,
              height: 32,
              display: "flex",
              alignItems: "center",
              justifyContent: "flex-end",
              paddingRight: 12
            },
            children: [
              /* @__PURE__ */ _(
                bi,
                {
                  className: "notification-archive-button",
                  size: "small",
                  onClick: (v) => (t.markAsArchived(k), v.preventDefault(), v.stopPropagation(), !1),
                  style: {
                    position: "absolute",
                    left: 0
                  },
                  children: /* @__PURE__ */ _(Di, { fontSize: "small" })
                }
              ),
              /* @__PURE__ */ _(
                Ti,
                {
                  variant: "dot",
                  color: "error",
                  className: "notification-highlight",
                  style: {
                    visibility: w ? "hidden" : "visible"
                  }
                }
              )
            ]
          }
        )
      ]
    }
  );
};
export {
  Jt as L,
  jc as N
};
