// @notificationapi/react/dist/assets/Popover.js
import * as c from "react";
import { g as Pe, P as e } from "./createTheme.js";
import { g as ve, s as te, u as ye, c as Y, a as be } from "./DefaultPropsProvider.js";
import { i as Te } from "./List.js";
import { b as X, r as Ee, c as Oe, e as Re } from "./ButtonBase.js";
import { jsx as z } from "react/jsx-runtime";
import { M as we, o as J } from "./Modal.js";
import { P as xe } from "./Paper.js";
import { o as K, H as Q } from "./Portal.js";
import { u as Ce } from "./resolveComponentProps.js";
import { i as Me } from "./integerPropType.js";
import { G as Ne } from "./Grow.js";
function je(t, o = 166) {
  let n;
  function P(...R) {
    const a = () => {
      t.apply(this, R);
    };
    clearTimeout(n), n = setTimeout(a, o);
  }
  return P.clear = () => {
    clearTimeout(n);
  }, P;
}
function ze(t) {
  return Pe("MuiPopover", t);
}
ve("MuiPopover", ["root", "paper"]);
function Z(t, o) {
  let n = 0;
  return typeof o == "number" ? n = o : o === "center" ? n = t.height / 2 : o === "bottom" && (n = t.height), n;
}
function ee(t, o) {
  let n = 0;
  return typeof o == "number" ? n = o : o === "center" ? n = t.width / 2 : o === "right" && (n = t.width), n;
}
function oe(t) {
  return [t.horizontal, t.vertical].map((o) => typeof o == "number" ? `${o}px` : o).join(" ");
}
function O(t) {
  return typeof t == "function" ? t() : t;
}
const De = (t) => {
  const {
    classes: o
  } = t;
  return be({
    root: ["root"],
    paper: ["paper"]
  }, ze, o);
}, ke = te(we, {
  name: "MuiPopover",
  slot: "Root",
  overridesResolver: (t, o) => o.root
})({}), Se = te(xe, {
  name: "MuiPopover",
  slot: "Paper",
  overridesResolver: (t, o) => o.paper
})({
  position: "absolute",
  overflowY: "auto",
  overflowX: "hidden",
  // So we see the popover when it's empty.
  // It's most likely on issue on userland.
  minWidth: 16,
  minHeight: 16,
  maxWidth: "calc(100% - 32px)",
  maxHeight: "calc(100% - 32px)",
  // We disable the focus ring for mouse, touch and keyboard users.
  outline: 0
}), Ue = /* @__PURE__ */ c.forwardRef(function(o, n) {
  const P = ye({
    props: o,
    name: "MuiPopover"
  }), {
    action: R,
    anchorEl: a,
    anchorOrigin: v = {
      vertical: "top",
      horizontal: "left"
    },
    anchorPosition: w,
    anchorReference: y = "anchorEl",
    children: ne,
    className: re,
    container: ie,
    elevation: D = 8,
    marginThreshold: l = 16,
    open: u,
    PaperProps: se = {},
    slots: ae = {},
    slotProps: E = {},
    transformOrigin: b = {
      vertical: "top",
      horizontal: "left"
    },
    TransitionComponent: x = Ne,
    transitionDuration: C = "auto",
    TransitionProps: {
      onEntering: k,
      ...S
    } = {},
    disableScrollLock: M = !1,
    ...le
  } = P, h = (E == null ? void 0 : E.paper) ?? se, N = c.useRef(), j = {
    ...P,
    anchorOrigin: v,
    anchorReference: y,
    elevation: D,
    marginThreshold: l,
    externalPaperSlotProps: h,
    transformOrigin: b,
    TransitionComponent: x,
    transitionDuration: C,
    TransitionProps: S
  }, U = De(j), H = c.useCallback(() => {
    if (y === "anchorPosition")
      return process.env.NODE_ENV !== "production" && (w || console.error('MUI: You need to provide a `anchorPosition` prop when using <Popover anchorReference="anchorPosition" />.')), w;
    const r = O(a), i = r && r.nodeType === 1 ? r : K(N.current).body, s = i.getBoundingClientRect();
    if (process.env.NODE_ENV !== "production") {
      const d = i.getBoundingClientRect();
      process.env.NODE_ENV !== "test" && d.top === 0 && d.left === 0 && d.right === 0 && d.bottom === 0 && console.warn(["MUI: The `anchorEl` prop provided to the component is invalid.", "The anchor element should be part of the document layout.", "Make sure the element is present in the document or that it's not display none."].join(`
`));
    }
    return {
      top: s.top + Z(s, v.vertical),
      left: s.left + ee(s, v.horizontal)
    };
  }, [a, v.horizontal, v.vertical, w, y]), I = c.useCallback((r) => ({
    vertical: Z(r, b.vertical),
    horizontal: ee(r, b.horizontal)
  }), [b.horizontal, b.vertical]), L = c.useCallback((r) => {
    const i = {
      width: r.offsetWidth,
      height: r.offsetHeight
    }, s = I(i);
    if (y === "none")
      return {
        top: null,
        left: null,
        transformOrigin: oe(s)
      };
    const d = H();
    let m = d.top - s.vertical, g = d.left - s.horizontal;
    const B = m + i.height, $ = g + i.width, F = J(O(a)), T = F.innerHeight - l, G = F.innerWidth - l;
    if (l !== null && m < l) {
      const p = m - l;
      m -= p, s.vertical += p;
    } else if (l !== null && B > T) {
      const p = B - T;
      m -= p, s.vertical += p;
    }
    if (process.env.NODE_ENV !== "production" && i.height > T && i.height && T && console.error(["MUI: The popover component is too tall.", `Some part of it can not be seen on the screen (${i.height - T}px).`, "Please consider adding a `max-height` to improve the user-experience."].join(`
`)), l !== null && g < l) {
      const p = g - l;
      g -= p, s.horizontal += p;
    } else if ($ > G) {
      const p = $ - G;
      g -= p, s.horizontal += p;
    }
    return {
      top: `${Math.round(m)}px`,
      left: `${Math.round(g)}px`,
      transformOrigin: oe(s)
    };
  }, [a, y, H, I, l]), [pe, V] = c.useState(u), f = c.useCallback(() => {
    const r = N.current;
    if (!r)
      return;
    const i = L(r);
    i.top !== null && r.style.setProperty("top", i.top), i.left !== null && (r.style.left = i.left), r.style.transformOrigin = i.transformOrigin, V(!0);
  }, [L]);
  c.useEffect(() => (M && window.addEventListener("scroll", f), () => window.removeEventListener("scroll", f)), [a, M, f]);
  const ce = (r, i) => {
    k && k(r, i), f();
  }, fe = () => {
    V(!1);
  };
  c.useEffect(() => {
    u && f();
  }), c.useImperativeHandle(R, () => u ? {
    updatePosition: () => {
      f();
    }
  } : null, [u, f]), c.useEffect(() => {
    if (!u)
      return;
    const r = je(() => {
      f();
    }), i = J(a);
    return i.addEventListener("resize", r), () => {
      r.clear(), i.removeEventListener("resize", r);
    };
  }, [a, u, f]);
  let q = C;
  C === "auto" && !x.muiSupportAuto && (q = void 0);
  const ue = ie || (a ? K(O(a)).body : void 0), W = {
    slots: ae,
    slotProps: {
      ...E,
      paper: h
    }
  }, [he, A] = X("paper", {
    elementType: Se,
    externalForwardedProps: W,
    additionalProps: {
      elevation: D,
      className: Y(U.paper, h == null ? void 0 : h.className),
      style: pe ? h.style : {
        ...h.style,
        opacity: 0
      }
    },
    ownerState: j
  }), [_, {
    slotProps: de,
    ...me
  }] = X("root", {
    elementType: ke,
    externalForwardedProps: W,
    additionalProps: {
      slotProps: {
        backdrop: {
          invisible: !0
        }
      },
      container: ue,
      open: u
    },
    ownerState: j,
    className: Y(U.root, re)
  }), ge = Ce(N, A.ref);
  return /* @__PURE__ */ z(_, {
    ...me,
    ...!Te(_) && {
      slotProps: de,
      disableScrollLock: M
    },
    ...le,
    ref: n,
    children: /* @__PURE__ */ z(x, {
      appear: !0,
      in: u,
      onEntering: ce,
      onExited: fe,
      timeout: q,
      ...S,
      children: /* @__PURE__ */ z(he, {
        ...A,
        ref: ge,
        children: ne
      })
    })
  });
});
process.env.NODE_ENV !== "production" && (Ue.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * A ref for imperative actions.
   * It currently only supports updatePosition() action.
   */
  action: Ee,
  /**
   * An HTML element, [PopoverVirtualElement](https://mui.com/material-ui/react-popover/#virtual-element),
   * or a function that returns either.
   * It's used to set the position of the popover.
   */
  anchorEl: Oe(e.oneOfType([Q, e.func]), (t) => {
    if (t.open && (!t.anchorReference || t.anchorReference === "anchorEl")) {
      const o = O(t.anchorEl);
      if (o && o.nodeType === 1) {
        const n = o.getBoundingClientRect();
        if (process.env.NODE_ENV !== "test" && n.top === 0 && n.left === 0 && n.right === 0 && n.bottom === 0)
          return new Error(["MUI: The `anchorEl` prop provided to the component is invalid.", "The anchor element should be part of the document layout.", "Make sure the element is present in the document or that it's not display none."].join(`
`));
      } else
        return new Error(["MUI: The `anchorEl` prop provided to the component is invalid.", `It should be an Element or PopoverVirtualElement instance but it's \`${o}\` instead.`].join(`
`));
    }
    return null;
  }),
  /**
   * This is the point on the anchor where the popover's
   * `anchorEl` will attach to. This is not used when the
   * anchorReference is 'anchorPosition'.
   *
   * Options:
   * vertical: [top, center, bottom];
   * horizontal: [left, center, right].
   * @default {
   *   vertical: 'top',
   *   horizontal: 'left',
   * }
   */
  anchorOrigin: e.shape({
    horizontal: e.oneOfType([e.oneOf(["center", "left", "right"]), e.number]).isRequired,
    vertical: e.oneOfType([e.oneOf(["bottom", "center", "top"]), e.number]).isRequired
  }),
  /**
   * This is the position that may be used to set the position of the popover.
   * The coordinates are relative to the application's client area.
   */
  anchorPosition: e.shape({
    left: e.number.isRequired,
    top: e.number.isRequired
  }),
  /**
   * This determines which anchor prop to refer to when setting
   * the position of the popover.
   * @default 'anchorEl'
   */
  anchorReference: e.oneOf(["anchorEl", "anchorPosition", "none"]),
  /**
   * A backdrop component. This prop enables custom backdrop rendering.
   * @deprecated Use `slotProps.root.slots.backdrop` instead. While this prop currently works, it will be removed in the next major version.
   * Use the `slotProps.root.slots.backdrop` prop to make your application ready for the next version of Material UI.
   * @default styled(Backdrop, {
   *   name: 'MuiModal',
   *   slot: 'Backdrop',
   *   overridesResolver: (props, styles) => {
   *     return styles.backdrop;
   *   },
   * })({
   *   zIndex: -1,
   * })
   */
  BackdropComponent: e.elementType,
  /**
   * Props applied to the [`Backdrop`](/material-ui/api/backdrop/) element.
   * @deprecated Use `slotProps.root.slotProps.backdrop` instead.
   */
  BackdropProps: e.object,
  /**
   * The content of the component.
   */
  children: e.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: e.object,
  /**
   * @ignore
   */
  className: e.string,
  /**
   * An HTML element, component instance, or function that returns either.
   * The `container` will passed to the Modal component.
   *
   * By default, it uses the body of the anchorEl's top-level document object,
   * so it's simply `document.body` most of the time.
   */
  container: e.oneOfType([Q, e.func]),
  /**
   * Disable the scroll lock behavior.
   * @default false
   */
  disableScrollLock: e.bool,
  /**
   * The elevation of the popover.
   * @default 8
   */
  elevation: Me,
  /**
   * Specifies how close to the edge of the window the popover can appear.
   * If null, the popover will not be constrained by the window.
   * @default 16
   */
  marginThreshold: e.number,
  /**
   * Callback fired when the component requests to be closed.
   * The `reason` parameter can optionally be used to control the response to `onClose`.
   */
  onClose: e.func,
  /**
   * If `true`, the component is shown.
   */
  open: e.bool.isRequired,
  /**
   * Props applied to the [`Paper`](https://mui.com/material-ui/api/paper/) element.
   *
   * This prop is an alias for `slotProps.paper` and will be overriden by it if both are used.
   * @deprecated Use `slotProps.paper` instead.
   *
   * @default {}
   */
  PaperProps: e.shape({
    component: Re
  }),
  /**
   * The props used for each slot inside.
   * @default {}
   */
  slotProps: e.shape({
    paper: e.oneOfType([e.func, e.object]),
    root: e.oneOfType([e.func, e.object])
  }),
  /**
   * The components used for each slot inside.
   * @default {}
   */
  slots: e.shape({
    paper: e.elementType,
    root: e.elementType
  }),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * This is the point on the popover which
   * will attach to the anchor's origin.
   *
   * Options:
   * vertical: [top, center, bottom, x(px)];
   * horizontal: [left, center, right, x(px)].
   * @default {
   *   vertical: 'top',
   *   horizontal: 'left',
   * }
   */
  transformOrigin: e.shape({
    horizontal: e.oneOfType([e.oneOf(["center", "left", "right"]), e.number]).isRequired,
    vertical: e.oneOfType([e.oneOf(["bottom", "center", "top"]), e.number]).isRequired
  }),
  /**
   * The component used for the transition.
   * [Follow this guide](https://mui.com/material-ui/transitions/#transitioncomponent-prop) to learn more about the requirements for this component.
   * @default Grow
   */
  TransitionComponent: e.elementType,
  /**
   * Set to 'auto' to automatically calculate transition time based on height.
   * @default 'auto'
   */
  transitionDuration: e.oneOfType([e.oneOf(["auto"]), e.number, e.shape({
    appear: e.number,
    enter: e.number,
    exit: e.number
  })]),
  /**
   * Props applied to the transition element.
   * By default, the element is based on this [`Transition`](https://reactcommunity.org/react-transition-group/transition/) component.
   * @default {}
   */
  TransitionProps: e.object
});
export {
  Ue as P,
  Se as a,
  je as d
};
