// @notificationapi/react/dist/assets/Portal.js
import { P as c } from "./createTheme.js";
import { c as m, d as f } from "./ButtonBase.js";
import * as u from "react";
import { a as P } from "./utils.js";
import { u as T, s as d } from "./resolveComponentProps.js";
import { e as h } from "./exactProp.js";
function R(e) {
  const {
    prototype: n = {}
  } = e;
  return !!n.isReactComponent;
}
function E(e, n, i, o, s) {
  const t = e[n], r = s || n;
  if (t == null || // When server-side rendering React doesn't warn either.
  // This is not an accurate check for SSR.
  // This is only in place for Emotion compat.
  // TODO: Revisit once https://github.com/facebook/react/issues/20047 is resolved.
  typeof window > "u")
    return null;
  let a;
  const l = t.type;
  return typeof l == "function" && !R(l) && (a = "Did you accidentally use a plain function component for an element instead?"), a !== void 0 ? new Error(`Invalid ${o} \`${r}\` supplied to \`${i}\`. Expected an element that can hold a ref. ${a} For more information see https://mui.com/r/caveat-with-refs-guide`) : null;
}
const v = m(c.element, E);
v.isRequired = m(c.element.isRequired, E);
function N(e, n, i, o, s) {
  if (process.env.NODE_ENV === "production")
    return null;
  const t = e[n], r = s || n;
  return t == null ? null : t && t.nodeType !== 1 ? new Error(`Invalid ${o} \`${r}\` supplied to \`${i}\`. Expected an HTMLElement.`) : null;
}
function C(e) {
  return e && e.ownerDocument || document;
}
function g(e) {
  var n;
  return parseInt(u.version, 10) >= 19 ? ((n = e == null ? void 0 : e.props) == null ? void 0 : n.ref) || null : (e == null ? void 0 : e.ref) || null;
}
function w(e) {
  return typeof e == "function" ? e() : e;
}
const p = /* @__PURE__ */ u.forwardRef(function(n, i) {
  const {
    children: o,
    container: s,
    disablePortal: t = !1
  } = n, [r, a] = u.useState(null), l = T(/* @__PURE__ */ u.isValidElement(o) ? g(o) : null, i);
  if (f(() => {
    t || a(w(s) || document.body);
  }, [s, t]), f(() => {
    if (r && !t)
      return d(i, r), () => {
        d(i, null);
      };
  }, [i, r, t]), t) {
    if (/* @__PURE__ */ u.isValidElement(o)) {
      const y = {
        ref: l
      };
      return /* @__PURE__ */ u.cloneElement(o, y);
    }
    return o;
  }
  return r && /* @__PURE__ */ P.createPortal(o, r);
});
process.env.NODE_ENV !== "production" && (p.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │ To update them, edit the TypeScript types and run `pnpm proptypes`. │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The children to render into the `container`.
   */
  children: c.node,
  /**
   * An HTML element or function that returns one.
   * The `container` will have the portal children appended to it.
   *
   * You can also provide a callback, which is called in a React layout effect.
   * This lets you set the container from a ref, and also makes server-side rendering possible.
   *
   * By default, it uses the body of the top-level document object,
   * so it's simply `document.body` most of the time.
   */
  container: c.oneOfType([N, c.func]),
  /**
   * The `children` will be under the DOM hierarchy of the parent component.
   * @default false
   */
  disablePortal: c.bool
});
process.env.NODE_ENV !== "production" && (p.propTypes = h(p.propTypes));
export {
  N as H,
  p as P,
  v as e,
  g,
  C as o
};
