// @notificationapi/react/dist/assets/resolveComponentProps.js
import * as k from "react";
import { c as N } from "./DefaultPropsProvider.js";
function O(n, s) {
  typeof n == "function" ? n(s) : n && (n.current = s);
}
function W(...n) {
  return k.useMemo(() => n.every((s) => s == null) ? null : (s) => {
    n.forEach((e) => {
      O(e, s);
    });
  }, n);
}
function R(n) {
  return typeof n == "string";
}
function Z(n, s, e) {
  return n === void 0 || R(n) ? s : {
    ...s,
    ownerState: {
      ...s.ownerState,
      ...e
    }
  };
}
function C(n, s = []) {
  if (n === void 0)
    return {};
  const e = {};
  return Object.keys(n).filter((t) => t.match(/^on[A-Z]/) && typeof n[t] == "function" && !s.includes(t)).forEach((t) => {
    e[t] = n[t];
  }), e;
}
function g(n) {
  if (n === void 0)
    return {};
  const s = {};
  return Object.keys(n).filter((e) => !(e.match(/^on[A-Z]/) && typeof n[e] == "function")).forEach((e) => {
    s[e] = n[e];
  }), s;
}
function M(n) {
  const {
    getSlotProps: s,
    additionalProps: e,
    externalSlotProps: t,
    externalForwardedProps: c,
    className: m
  } = n;
  if (!s) {
    const a = N(e == null ? void 0 : e.className, m, c == null ? void 0 : c.className, t == null ? void 0 : t.className), h = {
      ...e == null ? void 0 : e.style,
      ...c == null ? void 0 : c.style,
      ...t == null ? void 0 : t.style
    }, i = {
      ...e,
      ...c,
      ...t
    };
    return a.length > 0 && (i.className = a), Object.keys(h).length > 0 && (i.style = h), {
      props: i,
      internalRef: void 0
    };
  }
  const E = C({
    ...c,
    ...t
  }), v = g(t), H = g(c), f = s(E), y = N(f == null ? void 0 : f.className, e == null ? void 0 : e.className, m, c == null ? void 0 : c.className, t == null ? void 0 : t.className), l = {
    ...f == null ? void 0 : f.style,
    ...e == null ? void 0 : e.style,
    ...c == null ? void 0 : c.style,
    ...t == null ? void 0 : t.style
  }, u = {
    ...f,
    ...e,
    ...H,
    ...v
  };
  return y.length > 0 && (u.className = y), Object.keys(l).length > 0 && (u.style = l), {
    props: u,
    internalRef: f.ref
  };
}
function j(n, s, e) {
  return typeof n == "function" ? n(s, e) : n;
}
export {
  Z as a,
  C as e,
  M as m,
  j as r,
  O as s,
  W as u
};
