// @notificationapi/react/dist/assets/Stack.js
import * as l from "react";
import { P as e, b as S, h as g, r as m, j as V, k as N, m as D, g as E, l as h } from "./createTheme.js";
import { e as R, c as B, a as M, s as w, u as G } from "./DefaultPropsProvider.js";
import { jsx as U } from "react/jsx-runtime";
import { e as _ } from "./Typography.js";
import { r as $ } from "./DefaultPropsProvider2.js";
import { u as A } from "./useTheme.js";
const L = R();
function q(o) {
  const {
    theme: r,
    name: n,
    props: t
  } = o;
  return !r || !r.components || !r.components[n] || !r.components[n].defaultProps ? t : $(r.components[n].defaultProps, t);
}
function z({
  props: o,
  name: r,
  defaultTheme: n,
  themeId: t
}) {
  let s = A(n);
  return t && (s = s[t] || s), q({
    theme: s,
    name: r,
    props: o
  });
}
function se(...o) {
  return o.reduce((r, n) => n == null ? r : function(...s) {
    r.apply(this, s), n.apply(this, s);
  }, () => {
  });
}
const H = S(), I = L("div", {
  name: "MuiStack",
  slot: "Root",
  overridesResolver: (o, r) => r.root
});
function J(o) {
  return z({
    props: o,
    name: "MuiStack",
    defaultTheme: H
  });
}
function K(o, r) {
  const n = l.Children.toArray(o).filter(Boolean);
  return n.reduce((t, s, i) => (t.push(s), i < n.length - 1 && t.push(/* @__PURE__ */ l.cloneElement(r, {
    key: `separator-${i}`
  })), t), []);
}
const Q = (o) => ({
  row: "Left",
  "row-reverse": "Right",
  column: "Top",
  "column-reverse": "Bottom"
})[o], W = ({
  ownerState: o,
  theme: r
}) => {
  let n = {
    display: "flex",
    flexDirection: "column",
    ...g({
      theme: r
    }, m({
      values: o.direction,
      breakpoints: r.breakpoints.values
    }), (t) => ({
      flexDirection: t
    }))
  };
  if (o.spacing) {
    const t = V(r), s = Object.keys(r.breakpoints.values).reduce((c, a) => ((typeof o.spacing == "object" && o.spacing[a] != null || typeof o.direction == "object" && o.direction[a] != null) && (c[a] = !0), c), {}), i = m({
      values: o.direction,
      base: s
    }), u = m({
      values: o.spacing,
      base: s
    });
    typeof i == "object" && Object.keys(i).forEach((c, a, p) => {
      if (!i[c]) {
        const f = a > 0 ? i[p[a - 1]] : "column";
        i[c] = f;
      }
    }), n = N(n, g({
      theme: r
    }, u, (c, a) => o.useFlexGap ? {
      gap: h(t, c)
    } : {
      // The useFlexGap={false} implement relies on each child to give up control of the margin.
      // We need to reset the margin to avoid double spacing.
      "& > :not(style):not(style)": {
        margin: 0
      },
      "& > :not(style) ~ :not(style)": {
        [`margin${Q(a ? i[a] : o.direction)}`]: h(t, c)
      }
    }));
  }
  return n = D(r.breakpoints, n), n;
};
function X(o = {}) {
  const {
    // This will allow adding custom styled fn (for example for custom sx style function)
    createStyledComponent: r = I,
    useThemeProps: n = J,
    componentName: t = "MuiStack"
  } = o, s = () => M({
    root: ["root"]
  }, (c) => E(t, c), {}), i = r(W), u = /* @__PURE__ */ l.forwardRef(function(c, a) {
    const p = n(c), y = _(p), {
      component: f = "div",
      direction: b = "column",
      spacing: C = 0,
      divider: v,
      children: O,
      className: j,
      useFlexGap: k = !1,
      ...x
    } = y, P = {
      direction: b,
      spacing: C,
      useFlexGap: k
    }, F = s();
    return /* @__PURE__ */ U(i, {
      as: f,
      ownerState: P,
      ref: a,
      className: B(F.root, j),
      ...x,
      children: v ? K(O, v) : O
    });
  });
  return process.env.NODE_ENV !== "production" && (u.propTypes = {
    children: e.node,
    direction: e.oneOfType([e.oneOf(["column-reverse", "column", "row-reverse", "row"]), e.arrayOf(e.oneOf(["column-reverse", "column", "row-reverse", "row"])), e.object]),
    divider: e.node,
    spacing: e.oneOfType([e.arrayOf(e.oneOfType([e.number, e.string])), e.number, e.object, e.string]),
    sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object])
  }), u;
}
function ce({
  props: o,
  states: r,
  muiFormControl: n
}) {
  return r.reduce((t, s) => (t[s] = o[s], n && typeof o[s] > "u" && (t[s] = n[s]), t), {});
}
const T = /* @__PURE__ */ l.createContext(void 0);
process.env.NODE_ENV !== "production" && (T.displayName = "FormControlContext");
function ie() {
  return l.useContext(T);
}
const Y = X({
  createStyledComponent: w("div", {
    name: "MuiStack",
    slot: "Root",
    overridesResolver: (o, r) => r.root
  }),
  useThemeProps: (o) => G({
    props: o,
    name: "MuiStack"
  })
});
process.env.NODE_ENV !== "production" && (Y.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: e.node,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: e.elementType,
  /**
   * Defines the `flex-direction` style property.
   * It is applied for all screen sizes.
   * @default 'column'
   */
  direction: e.oneOfType([e.oneOf(["column-reverse", "column", "row-reverse", "row"]), e.arrayOf(e.oneOf(["column-reverse", "column", "row-reverse", "row"])), e.object]),
  /**
   * Add an element between each child.
   */
  divider: e.node,
  /**
   * Defines the space between immediate children.
   * @default 0
   */
  spacing: e.oneOfType([e.arrayOf(e.oneOfType([e.number, e.string])), e.number, e.object, e.string]),
  /**
   * The system prop, which allows defining system overrides as well as additional CSS styles.
   */
  sx: e.oneOfType([e.arrayOf(e.oneOfType([e.func, e.object, e.bool])), e.func, e.object]),
  /**
   * If `true`, the CSS flexbox `gap` is used instead of applying `margin` to children.
   *
   * While CSS `gap` removes the [known limitations](https://mui.com/joy-ui/react-stack/#limitations),
   * it is not fully supported in some browsers. We recommend checking https://caniuse.com/?search=flex%20gap before using this flag.
   *
   * To enable this flag globally, follow the [theme's default props](https://mui.com/material-ui/customization/theme-components/#default-props) configuration.
   * @default false
   */
  useFlexGap: e.bool
});
export {
  T as F,
  Y as S,
  se as c,
  ce as f,
  ie as u
};
