// @notificationapi/react/dist/assets/Typography.js
import * as j from "react";
import { P as t, f as C, i as B, g as E, e as u } from "./createTheme.js";
import { d as w, g as N, s as W, m as M, u as D, c as R, a as _ } from "./DefaultPropsProvider.js";
import { jsx as l } from "react/jsx-runtime";
import { T as $ } from "./DefaultPropsProvider2.js";
import { u as A } from "./useTheme.js";
import { G } from "./GlobalStyles.js";
import { c as V } from "./createSimplePaletteValueFilter.js";
function x({
  styles: r,
  themeId: o,
  defaultTheme: e = {}
}) {
  const n = A(e), a = typeof r == "function" ? r(o && n[o] || n) : r;
  return /* @__PURE__ */ l(G, {
    styles: a
  });
}
process.env.NODE_ENV !== "production" && (x.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │ To update them, edit the TypeScript types and run `pnpm proptypes`. │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * @ignore
   */
  defaultTheme: t.object,
  /**
   * @ignore
   */
  styles: t.oneOfType([t.array, t.func, t.number, t.object, t.string, t.bool]),
  /**
   * @ignore
   */
  themeId: t.string
});
const U = (r) => {
  var n;
  const o = {
    systemProps: {},
    otherProps: {}
  }, e = ((n = r == null ? void 0 : r.theme) == null ? void 0 : n.unstable_sxConfig) ?? C;
  return Object.keys(r).forEach((a) => {
    e[a] ? o.systemProps[a] = r[a] : o.otherProps[a] = r[a];
  }), o;
};
function H(r) {
  const {
    sx: o,
    ...e
  } = r, {
    systemProps: n,
    otherProps: a
  } = U(e);
  let s;
  return Array.isArray(o) ? s = [n, ...o] : typeof o == "function" ? s = (...p) => {
    const i = o(...p);
    return B(i) ? {
      ...n,
      ...i
    } : n;
  } : s = {
    ...n,
    ...o
  }, {
    ...a,
    sx: s
  };
}
function T(r) {
  return /* @__PURE__ */ l(x, {
    ...r,
    defaultTheme: w,
    themeId: $
  });
}
process.env.NODE_ENV !== "production" && (T.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The styles you want to apply globally.
   */
  styles: t.oneOfType([t.array, t.func, t.number, t.object, t.string, t.bool])
});
function rt(r) {
  return function(e) {
    return (
      // Pigment CSS `globalCss` support callback with theme inside an object but `GlobalStyles` support theme as a callback value.
      /* @__PURE__ */ l(T, {
        styles: typeof r == "function" ? (n) => r({
          theme: n,
          ...e
        }) : r
      })
    );
  };
}
function I() {
  return H;
}
function k(r) {
  return E("MuiTypography", r);
}
N("MuiTypography", ["root", "h1", "h2", "h3", "h4", "h5", "h6", "subtitle1", "subtitle2", "body1", "body2", "inherit", "button", "caption", "overline", "alignLeft", "alignRight", "alignCenter", "alignJustify", "noWrap", "gutterBottom", "paragraph"]);
const z = {
  primary: !0,
  secondary: !0,
  error: !0,
  info: !0,
  success: !0,
  warning: !0,
  textPrimary: !0,
  textSecondary: !0,
  textDisabled: !0
}, F = I(), J = (r) => {
  const {
    align: o,
    gutterBottom: e,
    noWrap: n,
    paragraph: a,
    variant: s,
    classes: p
  } = r, i = {
    root: ["root", s, r.align !== "inherit" && `align${u(o)}`, e && "gutterBottom", n && "noWrap", a && "paragraph"]
  };
  return _(i, k, p);
}, L = W("span", {
  name: "MuiTypography",
  slot: "Root",
  overridesResolver: (r, o) => {
    const {
      ownerState: e
    } = r;
    return [o.root, e.variant && o[e.variant], e.align !== "inherit" && o[`align${u(e.align)}`], e.noWrap && o.noWrap, e.gutterBottom && o.gutterBottom, e.paragraph && o.paragraph];
  }
})(M(({
  theme: r
}) => {
  var o;
  return {
    margin: 0,
    variants: [{
      props: {
        variant: "inherit"
      },
      style: {
        // Some elements, like <button> on Chrome have default font that doesn't inherit, reset this.
        font: "inherit",
        lineHeight: "inherit",
        letterSpacing: "inherit"
      }
    }, ...Object.entries(r.typography).filter(([e, n]) => e !== "inherit" && n && typeof n == "object").map(([e, n]) => ({
      props: {
        variant: e
      },
      style: n
    })), ...Object.entries(r.palette).filter(V()).map(([e]) => ({
      props: {
        color: e
      },
      style: {
        color: (r.vars || r).palette[e].main
      }
    })), ...Object.entries(((o = r.palette) == null ? void 0 : o.text) || {}).filter(([, e]) => typeof e == "string").map(([e]) => ({
      props: {
        color: `text${u(e)}`
      },
      style: {
        color: (r.vars || r).palette.text[e]
      }
    })), {
      props: ({
        ownerState: e
      }) => e.align !== "inherit",
      style: {
        textAlign: "var(--Typography-textAlign)"
      }
    }, {
      props: ({
        ownerState: e
      }) => e.noWrap,
      style: {
        overflow: "hidden",
        textOverflow: "ellipsis",
        whiteSpace: "nowrap"
      }
    }, {
      props: ({
        ownerState: e
      }) => e.gutterBottom,
      style: {
        marginBottom: "0.35em"
      }
    }, {
      props: ({
        ownerState: e
      }) => e.paragraph,
      style: {
        marginBottom: 16
      }
    }]
  };
})), d = {
  h1: "h1",
  h2: "h2",
  h3: "h3",
  h4: "h4",
  h5: "h5",
  h6: "h6",
  subtitle1: "h6",
  subtitle2: "h6",
  body1: "p",
  body2: "p",
  inherit: "p"
}, q = /* @__PURE__ */ j.forwardRef(function(o, e) {
  const {
    color: n,
    ...a
  } = D({
    props: o,
    name: "MuiTypography"
  }), s = !z[n], p = F({
    ...a,
    ...s && {
      color: n
    }
  }), {
    align: i = "inherit",
    className: y,
    component: h,
    gutterBottom: v = !1,
    noWrap: O = !1,
    paragraph: g = !1,
    variant: c = "body1",
    variantMapping: f = d,
    ...m
  } = p, b = {
    ...p,
    align: i,
    color: n,
    className: y,
    component: h,
    gutterBottom: v,
    noWrap: O,
    paragraph: g,
    variant: c,
    variantMapping: f
  }, S = h || (g ? "p" : f[c] || d[c]) || "span", P = J(b);
  return /* @__PURE__ */ l(L, {
    as: S,
    ref: e,
    className: R(P.root, y),
    ...m,
    ownerState: b,
    style: {
      ...i !== "inherit" && {
        "--Typography-textAlign": i
      },
      ...m.style
    }
  });
});
process.env.NODE_ENV !== "production" && (q.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * Set the text-align on the component.
   * @default 'inherit'
   */
  align: t.oneOf(["center", "inherit", "justify", "left", "right"]),
  /**
   * The content of the component.
   */
  children: t.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: t.object,
  /**
   * @ignore
   */
  className: t.string,
  /**
   * The color of the component.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   */
  color: t.oneOfType([t.oneOf(["primary", "secondary", "success", "error", "info", "warning", "textPrimary", "textSecondary", "textDisabled"]), t.string]),
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: t.elementType,
  /**
   * If `true`, the text will have a bottom margin.
   * @default false
   */
  gutterBottom: t.bool,
  /**
   * If `true`, the text will not wrap, but instead will truncate with a text overflow ellipsis.
   *
   * Note that text overflow can only happen with block or inline-block level elements
   * (the element needs to have a width in order to overflow).
   * @default false
   */
  noWrap: t.bool,
  /**
   * If `true`, the element will be a paragraph element.
   * @default false
   * @deprecated Use the `component` prop instead. This prop will be removed in v7. See [Migrating from deprecated APIs](https://mui.com/material-ui/migration/migrating-from-deprecated-apis/) for more details.
   */
  paragraph: t.bool,
  /**
   * @ignore
   */
  style: t.object,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: t.oneOfType([t.arrayOf(t.oneOfType([t.func, t.object, t.bool])), t.func, t.object]),
  /**
   * Applies the theme typography styles.
   * @default 'body1'
   */
  variant: t.oneOfType([t.oneOf(["body1", "body2", "button", "caption", "h1", "h2", "h3", "h4", "h5", "h6", "inherit", "overline", "subtitle1", "subtitle2"]), t.string]),
  /**
   * The component maps the variant prop to a range of different HTML element types.
   * For instance, subtitle1 to `<h6>`.
   * If you wish to change that mapping, you can provide your own.
   * Alternatively, you can use the `component` prop.
   * @default {
   *   h1: 'h1',
   *   h2: 'h2',
   *   h3: 'h3',
   *   h4: 'h4',
   *   h5: 'h5',
   *   h6: 'h6',
   *   subtitle1: 'h6',
   *   subtitle2: 'h6',
   *   body1: 'p',
   *   body2: 'p',
   *   inherit: 'p',
   * }
   */
  variantMapping: t.object
});
export {
  q as T,
  H as e,
  rt as g
};
