// @notificationapi/react/dist/assets/useThemeWithoutDefault.js
import { s as L, h as ue, E as le, c as me, w as ye, a as z, u as I, i as de } from "./DefaultPropsProvider2.js";
import * as b from "react";
var h = { exports: {} }, o = {};
/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var F;
function Ee() {
  if (F) return o;
  F = 1;
  var r = typeof Symbol == "function" && Symbol.for, n = r ? Symbol.for("react.element") : 60103, c = r ? Symbol.for("react.portal") : 60106, f = r ? Symbol.for("react.fragment") : 60107, l = r ? Symbol.for("react.strict_mode") : 60108, i = r ? Symbol.for("react.profiler") : 60114, s = r ? Symbol.for("react.provider") : 60109, y = r ? Symbol.for("react.context") : 60110, d = r ? Symbol.for("react.async_mode") : 60111, E = r ? Symbol.for("react.concurrent_mode") : 60111, v = r ? Symbol.for("react.forward_ref") : 60112, p = r ? Symbol.for("react.suspense") : 60113, R = r ? Symbol.for("react.suspense_list") : 60120, S = r ? Symbol.for("react.memo") : 60115, _ = r ? Symbol.for("react.lazy") : 60116, x = r ? Symbol.for("react.block") : 60121, P = r ? Symbol.for("react.fundamental") : 60117, $ = r ? Symbol.for("react.responder") : 60118, A = r ? Symbol.for("react.scope") : 60119;
  function m(e) {
    if (typeof e == "object" && e !== null) {
      var T = e.$$typeof;
      switch (T) {
        case n:
          switch (e = e.type, e) {
            case d:
            case E:
            case f:
            case i:
            case l:
            case p:
              return e;
            default:
              switch (e = e && e.$$typeof, e) {
                case y:
                case v:
                case _:
                case S:
                case s:
                  return e;
                default:
                  return T;
              }
          }
        case c:
          return T;
      }
    }
  }
  function u(e) {
    return m(e) === E;
  }
  return o.AsyncMode = d, o.ConcurrentMode = E, o.ContextConsumer = y, o.ContextProvider = s, o.Element = n, o.ForwardRef = v, o.Fragment = f, o.Lazy = _, o.Memo = S, o.Portal = c, o.Profiler = i, o.StrictMode = l, o.Suspense = p, o.isAsyncMode = function(e) {
    return u(e) || m(e) === d;
  }, o.isConcurrentMode = u, o.isContextConsumer = function(e) {
    return m(e) === y;
  }, o.isContextProvider = function(e) {
    return m(e) === s;
  }, o.isElement = function(e) {
    return typeof e == "object" && e !== null && e.$$typeof === n;
  }, o.isForwardRef = function(e) {
    return m(e) === v;
  }, o.isFragment = function(e) {
    return m(e) === f;
  }, o.isLazy = function(e) {
    return m(e) === _;
  }, o.isMemo = function(e) {
    return m(e) === S;
  }, o.isPortal = function(e) {
    return m(e) === c;
  }, o.isProfiler = function(e) {
    return m(e) === i;
  }, o.isStrictMode = function(e) {
    return m(e) === l;
  }, o.isSuspense = function(e) {
    return m(e) === p;
  }, o.isValidElementType = function(e) {
    return typeof e == "string" || typeof e == "function" || e === f || e === E || e === i || e === l || e === p || e === R || typeof e == "object" && e !== null && (e.$$typeof === _ || e.$$typeof === S || e.$$typeof === s || e.$$typeof === y || e.$$typeof === v || e.$$typeof === P || e.$$typeof === $ || e.$$typeof === A || e.$$typeof === x);
  }, o.typeOf = m, o;
}
var a = {};
/** @license React v16.13.1
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Y;
function ve() {
  return Y || (Y = 1, process.env.NODE_ENV !== "production" && function() {
    var r = typeof Symbol == "function" && Symbol.for, n = r ? Symbol.for("react.element") : 60103, c = r ? Symbol.for("react.portal") : 60106, f = r ? Symbol.for("react.fragment") : 60107, l = r ? Symbol.for("react.strict_mode") : 60108, i = r ? Symbol.for("react.profiler") : 60114, s = r ? Symbol.for("react.provider") : 60109, y = r ? Symbol.for("react.context") : 60110, d = r ? Symbol.for("react.async_mode") : 60111, E = r ? Symbol.for("react.concurrent_mode") : 60111, v = r ? Symbol.for("react.forward_ref") : 60112, p = r ? Symbol.for("react.suspense") : 60113, R = r ? Symbol.for("react.suspense_list") : 60120, S = r ? Symbol.for("react.memo") : 60115, _ = r ? Symbol.for("react.lazy") : 60116, x = r ? Symbol.for("react.block") : 60121, P = r ? Symbol.for("react.fundamental") : 60117, $ = r ? Symbol.for("react.responder") : 60118, A = r ? Symbol.for("react.scope") : 60119;
    function m(t) {
      return typeof t == "string" || typeof t == "function" || // Note: its typeof might be other than 'symbol' or 'number' if it's a polyfill.
      t === f || t === E || t === i || t === l || t === p || t === R || typeof t == "object" && t !== null && (t.$$typeof === _ || t.$$typeof === S || t.$$typeof === s || t.$$typeof === y || t.$$typeof === v || t.$$typeof === P || t.$$typeof === $ || t.$$typeof === A || t.$$typeof === x);
    }
    function u(t) {
      if (typeof t == "object" && t !== null) {
        var M = t.$$typeof;
        switch (M) {
          case n:
            var C = t.type;
            switch (C) {
              case d:
              case E:
              case f:
              case i:
              case l:
              case p:
                return C;
              default:
                var w = C && C.$$typeof;
                switch (w) {
                  case y:
                  case v:
                  case _:
                  case S:
                  case s:
                    return w;
                  default:
                    return M;
                }
            }
          case c:
            return M;
        }
      }
    }
    var e = d, T = E, q = y, V = s, U = n, X = v, J = f, W = _, j = S, G = c, B = i, K = l, Z = p, g = !1;
    function H(t) {
      return g || (g = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 17+. Update your code to use ReactIs.isConcurrentMode() instead. It has the exact same API.")), O(t) || u(t) === d;
    }
    function O(t) {
      return u(t) === E;
    }
    function Q(t) {
      return u(t) === y;
    }
    function ee(t) {
      return u(t) === s;
    }
    function re(t) {
      return typeof t == "object" && t !== null && t.$$typeof === n;
    }
    function te(t) {
      return u(t) === v;
    }
    function ne(t) {
      return u(t) === f;
    }
    function oe(t) {
      return u(t) === _;
    }
    function ae(t) {
      return u(t) === S;
    }
    function se(t) {
      return u(t) === c;
    }
    function ie(t) {
      return u(t) === i;
    }
    function ce(t) {
      return u(t) === l;
    }
    function fe(t) {
      return u(t) === p;
    }
    a.AsyncMode = e, a.ConcurrentMode = T, a.ContextConsumer = q, a.ContextProvider = V, a.Element = U, a.ForwardRef = X, a.Fragment = J, a.Lazy = W, a.Memo = j, a.Portal = G, a.Profiler = B, a.StrictMode = K, a.Suspense = Z, a.isAsyncMode = H, a.isConcurrentMode = O, a.isContextConsumer = Q, a.isContextProvider = ee, a.isElement = re, a.isForwardRef = te, a.isFragment = ne, a.isLazy = oe, a.isMemo = ae, a.isPortal = se, a.isProfiler = ie, a.isStrictMode = ce, a.isSuspense = fe, a.isValidElementType = m, a.typeOf = u;
  }()), a;
}
process.env.NODE_ENV === "production" ? h.exports = Ee() : h.exports = ve();
var pe = h.exports, D = pe, Se = {
  $$typeof: !0,
  render: !0,
  defaultProps: !0,
  displayName: !0,
  propTypes: !0
}, _e = {
  $$typeof: !0,
  compare: !0,
  defaultProps: !0,
  displayName: !0,
  propTypes: !0,
  type: !0
}, k = {};
k[D.ForwardRef] = Se;
k[D.Memo] = _e;
var N = function(n, c) {
  var f = arguments;
  if (c == null || !ue.call(c, "css"))
    return b.createElement.apply(void 0, f);
  var l = f.length, i = new Array(l);
  i[0] = le, i[1] = me(n, c);
  for (var s = 2; s < l; s++)
    i[s] = f[s];
  return b.createElement.apply(null, i);
};
(function(r) {
  var n;
  n || (n = r.JSX || (r.JSX = {}));
})(N || (N = {}));
var Re = /* @__PURE__ */ ye(function(r, n) {
  var c = r.styles, f = L([c], void 0, b.useContext(z)), l = b.useRef();
  return I(function() {
    var i = n.key + "-global", s = new n.sheet.constructor({
      key: i,
      nonce: n.sheet.nonce,
      container: n.sheet.container,
      speedy: n.sheet.isSpeedy
    }), y = !1, d = document.querySelector('style[data-emotion="' + i + " " + f.name + '"]');
    return n.sheet.tags.length && (s.before = n.sheet.tags[0]), d !== null && (y = !0, d.setAttribute("data-emotion", i), s.hydrate([d])), l.current = [s, y], function() {
      s.flush();
    };
  }, [n]), I(function() {
    var i = l.current, s = i[0], y = i[1];
    if (y) {
      i[1] = !1;
      return;
    }
    if (f.next !== void 0 && de(n, f.next, !0), s.tags.length) {
      var d = s.tags[s.tags.length - 1].nextElementSibling;
      s.before = d, s.flush();
    }
    n.insert("", f, s, !1);
  }, [n, f.name]), null;
});
function Te() {
  for (var r = arguments.length, n = new Array(r), c = 0; c < r; c++)
    n[c] = arguments[c];
  return L(n);
}
function xe() {
  var r = Te.apply(void 0, arguments), n = "animation-" + r.name;
  return {
    name: n,
    styles: "@keyframes " + n + "{" + r.styles + "}",
    anim: 1,
    toString: function() {
      return "_EMO_" + this.name + "_" + this.styles + "_EMO_";
    }
  };
}
function be(r) {
  return Object.keys(r).length === 0;
}
function Pe(r = null) {
  const n = b.useContext(z);
  return !n || be(n) ? r : n;
}
export {
  Re as G,
  Te as c,
  xe as k,
  Pe as u
};
