// @notificationapi/react/dist/assets/utils.js
import { _ as k, g as I, T as A } from "./ButtonBase.js";
import { q as U, P as c } from "./createTheme.js";
import T from "react";
var N = { exports: {} }, y = {};
/**
 * @license React
 * react-dom.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var C;
function j() {
  if (C) return y;
  C = 1;
  var p = T;
  function g(i) {
    var t = "https://react.dev/errors/" + i;
    if (1 < arguments.length) {
      t += "?args[]=" + encodeURIComponent(arguments[1]);
      for (var d = 2; d < arguments.length; d++)
        t += "&args[]=" + encodeURIComponent(arguments[d]);
    }
    return "Minified React error #" + i + "; visit " + t + " for the full message or use the non-minified dev environment for full errors and additional helpful warnings.";
  }
  function l() {
  }
  var a = {
    d: {
      f: l,
      r: function() {
        throw Error(g(522));
      },
      D: l,
      C: l,
      L: l,
      m: l,
      X: l,
      S: l,
      M: l
    },
    p: 0,
    findDOMNode: null
  }, o = Symbol.for("react.portal");
  function n(i, t, d) {
    var r = 3 < arguments.length && arguments[3] !== void 0 ? arguments[3] : null;
    return {
      $$typeof: o,
      key: r == null ? null : "" + r,
      children: i,
      containerInfo: t,
      implementation: d
    };
  }
  var s = p.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE;
  function f(i, t) {
    if (i === "font") return "";
    if (typeof t == "string")
      return t === "use-credentials" ? t : "";
  }
  return y.__DOM_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE = a, y.createPortal = function(i, t) {
    var d = 2 < arguments.length && arguments[2] !== void 0 ? arguments[2] : null;
    if (!t || t.nodeType !== 1 && t.nodeType !== 9 && t.nodeType !== 11)
      throw Error(g(299));
    return n(i, t, null, d);
  }, y.flushSync = function(i) {
    var t = s.T, d = a.p;
    try {
      if (s.T = null, a.p = 2, i) return i();
    } finally {
      s.T = t, a.p = d, a.d.f();
    }
  }, y.preconnect = function(i, t) {
    typeof i == "string" && (t ? (t = t.crossOrigin, t = typeof t == "string" ? t === "use-credentials" ? t : "" : void 0) : t = null, a.d.C(i, t));
  }, y.prefetchDNS = function(i) {
    typeof i == "string" && a.d.D(i);
  }, y.preinit = function(i, t) {
    if (typeof i == "string" && t && typeof t.as == "string") {
      var d = t.as, r = f(d, t.crossOrigin), e = typeof t.integrity == "string" ? t.integrity : void 0, u = typeof t.fetchPriority == "string" ? t.fetchPriority : void 0;
      d === "style" ? a.d.S(
        i,
        typeof t.precedence == "string" ? t.precedence : void 0,
        {
          crossOrigin: r,
          integrity: e,
          fetchPriority: u
        }
      ) : d === "script" && a.d.X(i, {
        crossOrigin: r,
        integrity: e,
        fetchPriority: u,
        nonce: typeof t.nonce == "string" ? t.nonce : void 0
      });
    }
  }, y.preinitModule = function(i, t) {
    if (typeof i == "string")
      if (typeof t == "object" && t !== null) {
        if (t.as == null || t.as === "script") {
          var d = f(
            t.as,
            t.crossOrigin
          );
          a.d.M(i, {
            crossOrigin: d,
            integrity: typeof t.integrity == "string" ? t.integrity : void 0,
            nonce: typeof t.nonce == "string" ? t.nonce : void 0
          });
        }
      } else t == null && a.d.M(i);
  }, y.preload = function(i, t) {
    if (typeof i == "string" && typeof t == "object" && t !== null && typeof t.as == "string") {
      var d = t.as, r = f(d, t.crossOrigin);
      a.d.L(i, d, {
        crossOrigin: r,
        integrity: typeof t.integrity == "string" ? t.integrity : void 0,
        nonce: typeof t.nonce == "string" ? t.nonce : void 0,
        type: typeof t.type == "string" ? t.type : void 0,
        fetchPriority: typeof t.fetchPriority == "string" ? t.fetchPriority : void 0,
        referrerPolicy: typeof t.referrerPolicy == "string" ? t.referrerPolicy : void 0,
        imageSrcSet: typeof t.imageSrcSet == "string" ? t.imageSrcSet : void 0,
        imageSizes: typeof t.imageSizes == "string" ? t.imageSizes : void 0,
        media: typeof t.media == "string" ? t.media : void 0
      });
    }
  }, y.preloadModule = function(i, t) {
    if (typeof i == "string")
      if (t) {
        var d = f(t.as, t.crossOrigin);
        a.d.m(i, {
          as: typeof t.as == "string" && t.as !== "script" ? t.as : void 0,
          crossOrigin: d,
          integrity: typeof t.integrity == "string" ? t.integrity : void 0
        });
      } else a.d.m(i);
  }, y.requestFormReset = function(i) {
    a.d.r(i);
  }, y.unstable_batchedUpdates = function(i, t) {
    return i(t);
  }, y.useFormState = function(i, t, d) {
    return s.H.useFormState(i, t, d);
  }, y.useFormStatus = function() {
    return s.H.useHostTransitionStatus();
  }, y.version = "19.0.0", y;
}
var m = {};
/**
 * @license React
 * react-dom.development.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var w;
function H() {
  return w || (w = 1, process.env.NODE_ENV !== "production" && function() {
    function p() {
    }
    function g(r) {
      return "" + r;
    }
    function l(r, e, u) {
      var h = 3 < arguments.length && arguments[3] !== void 0 ? arguments[3] : null;
      try {
        g(h);
        var b = !1;
      } catch {
        b = !0;
      }
      return b && (console.error(
        "The provided key is an unsupported type %s. This value must be coerced to a string before using it here.",
        typeof Symbol == "function" && Symbol.toStringTag && h[Symbol.toStringTag] || h.constructor.name || "Object"
      ), g(h)), {
        $$typeof: t,
        key: h == null ? null : "" + h,
        children: r,
        containerInfo: e,
        implementation: u
      };
    }
    function a(r, e) {
      if (r === "font") return "";
      if (typeof e == "string")
        return e === "use-credentials" ? e : "";
    }
    function o(r) {
      return r === null ? "`null`" : r === void 0 ? "`undefined`" : r === "" ? "an empty string" : 'something with type "' + typeof r + '"';
    }
    function n(r) {
      return r === null ? "`null`" : r === void 0 ? "`undefined`" : r === "" ? "an empty string" : typeof r == "string" ? JSON.stringify(r) : typeof r == "number" ? "`" + r + "`" : 'something with type "' + typeof r + '"';
    }
    function s() {
      var r = d.H;
      return r === null && console.error(
        `Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://react.dev/link/invalid-hook-call for tips about how to debug and fix this problem.`
      ), r;
    }
    typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ < "u" && typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStart == "function" && __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStart(Error());
    var f = T, i = {
      d: {
        f: p,
        r: function() {
          throw Error(
            "Invalid form element. requestFormReset must be passed a form that was rendered by React."
          );
        },
        D: p,
        C: p,
        L: p,
        m: p,
        X: p,
        S: p,
        M: p
      },
      p: 0,
      findDOMNode: null
    }, t = Symbol.for("react.portal"), d = f.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE;
    typeof Map == "function" && Map.prototype != null && typeof Map.prototype.forEach == "function" && typeof Set == "function" && Set.prototype != null && typeof Set.prototype.clear == "function" && typeof Set.prototype.forEach == "function" || console.error(
      "React depends on Map and Set built-in types. Make sure that you load a polyfill in older browsers. https://reactjs.org/link/react-polyfills"
    ), m.__DOM_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE = i, m.createPortal = function(r, e) {
      var u = 2 < arguments.length && arguments[2] !== void 0 ? arguments[2] : null;
      if (!e || e.nodeType !== 1 && e.nodeType !== 9 && e.nodeType !== 11)
        throw Error("Target container is not a DOM element.");
      return l(r, e, null, u);
    }, m.flushSync = function(r) {
      var e = d.T, u = i.p;
      try {
        if (d.T = null, i.p = 2, r)
          return r();
      } finally {
        d.T = e, i.p = u, i.d.f() && console.error(
          "flushSync was called from inside a lifecycle method. React cannot flush when React is already rendering. Consider moving this call to a scheduler task or micro task."
        );
      }
    }, m.preconnect = function(r, e) {
      typeof r == "string" && r ? e != null && typeof e != "object" ? console.error(
        "ReactDOM.preconnect(): Expected the `options` argument (second) to be an object but encountered %s instead. The only supported option at this time is `crossOrigin` which accepts a string.",
        n(e)
      ) : e != null && typeof e.crossOrigin != "string" && console.error(
        "ReactDOM.preconnect(): Expected the `crossOrigin` option (second argument) to be a string but encountered %s instead. Try removing this option or passing a string value instead.",
        o(e.crossOrigin)
      ) : console.error(
        "ReactDOM.preconnect(): Expected the `href` argument (first) to be a non-empty string but encountered %s instead.",
        o(r)
      ), typeof r == "string" && (e ? (e = e.crossOrigin, e = typeof e == "string" ? e === "use-credentials" ? e : "" : void 0) : e = null, i.d.C(r, e));
    }, m.prefetchDNS = function(r) {
      if (typeof r != "string" || !r)
        console.error(
          "ReactDOM.prefetchDNS(): Expected the `href` argument (first) to be a non-empty string but encountered %s instead.",
          o(r)
        );
      else if (1 < arguments.length) {
        var e = arguments[1];
        typeof e == "object" && e.hasOwnProperty("crossOrigin") ? console.error(
          "ReactDOM.prefetchDNS(): Expected only one argument, `href`, but encountered %s as a second argument instead. This argument is reserved for future options and is currently disallowed. It looks like the you are attempting to set a crossOrigin property for this DNS lookup hint. Browsers do not perform DNS queries using CORS and setting this attribute on the resource hint has no effect. Try calling ReactDOM.prefetchDNS() with just a single string argument, `href`.",
          n(e)
        ) : console.error(
          "ReactDOM.prefetchDNS(): Expected only one argument, `href`, but encountered %s as a second argument instead. This argument is reserved for future options and is currently disallowed. Try calling ReactDOM.prefetchDNS() with just a single string argument, `href`.",
          n(e)
        );
      }
      typeof r == "string" && i.d.D(r);
    }, m.preinit = function(r, e) {
      if (typeof r == "string" && r ? e == null || typeof e != "object" ? console.error(
        "ReactDOM.preinit(): Expected the `options` argument (second) to be an object with an `as` property describing the type of resource to be preinitialized but encountered %s instead.",
        n(e)
      ) : e.as !== "style" && e.as !== "script" && console.error(
        'ReactDOM.preinit(): Expected the `as` property in the `options` argument (second) to contain a valid value describing the type of resource to be preinitialized but encountered %s instead. Valid values for `as` are "style" and "script".',
        n(e.as)
      ) : console.error(
        "ReactDOM.preinit(): Expected the `href` argument (first) to be a non-empty string but encountered %s instead.",
        o(r)
      ), typeof r == "string" && e && typeof e.as == "string") {
        var u = e.as, h = a(u, e.crossOrigin), b = typeof e.integrity == "string" ? e.integrity : void 0, x = typeof e.fetchPriority == "string" ? e.fetchPriority : void 0;
        u === "style" ? i.d.S(
          r,
          typeof e.precedence == "string" ? e.precedence : void 0,
          {
            crossOrigin: h,
            integrity: b,
            fetchPriority: x
          }
        ) : u === "script" && i.d.X(r, {
          crossOrigin: h,
          integrity: b,
          fetchPriority: x,
          nonce: typeof e.nonce == "string" ? e.nonce : void 0
        });
      }
    }, m.preinitModule = function(r, e) {
      var u = "";
      if (typeof r == "string" && r || (u += " The `href` argument encountered was " + o(r) + "."), e !== void 0 && typeof e != "object" ? u += " The `options` argument encountered was " + o(e) + "." : e && "as" in e && e.as !== "script" && (u += " The `as` option encountered was " + n(e.as) + "."), u)
        console.error(
          "ReactDOM.preinitModule(): Expected up to two arguments, a non-empty `href` string and, optionally, an `options` object with a valid `as` property.%s",
          u
        );
      else
        switch (u = e && typeof e.as == "string" ? e.as : "script", u) {
          case "script":
            break;
          default:
            u = n(u), console.error(
              'ReactDOM.preinitModule(): Currently the only supported "as" type for this function is "script" but received "%s" instead. This warning was generated for `href` "%s". In the future other module types will be supported, aligning with the import-attributes proposal. Learn more here: (https://github.com/tc39/proposal-import-attributes)',
              u,
              r
            );
        }
      typeof r == "string" && (typeof e == "object" && e !== null ? (e.as == null || e.as === "script") && (u = a(
        e.as,
        e.crossOrigin
      ), i.d.M(r, {
        crossOrigin: u,
        integrity: typeof e.integrity == "string" ? e.integrity : void 0,
        nonce: typeof e.nonce == "string" ? e.nonce : void 0
      })) : e == null && i.d.M(r));
    }, m.preload = function(r, e) {
      var u = "";
      if (typeof r == "string" && r || (u += " The `href` argument encountered was " + o(r) + "."), e == null || typeof e != "object" ? u += " The `options` argument encountered was " + o(e) + "." : typeof e.as == "string" && e.as || (u += " The `as` option encountered was " + o(e.as) + "."), u && console.error(
        'ReactDOM.preload(): Expected two arguments, a non-empty `href` string and an `options` object with an `as` property valid for a `<link rel="preload" as="..." />` tag.%s',
        u
      ), typeof r == "string" && typeof e == "object" && e !== null && typeof e.as == "string") {
        u = e.as;
        var h = a(
          u,
          e.crossOrigin
        );
        i.d.L(r, u, {
          crossOrigin: h,
          integrity: typeof e.integrity == "string" ? e.integrity : void 0,
          nonce: typeof e.nonce == "string" ? e.nonce : void 0,
          type: typeof e.type == "string" ? e.type : void 0,
          fetchPriority: typeof e.fetchPriority == "string" ? e.fetchPriority : void 0,
          referrerPolicy: typeof e.referrerPolicy == "string" ? e.referrerPolicy : void 0,
          imageSrcSet: typeof e.imageSrcSet == "string" ? e.imageSrcSet : void 0,
          imageSizes: typeof e.imageSizes == "string" ? e.imageSizes : void 0,
          media: typeof e.media == "string" ? e.media : void 0
        });
      }
    }, m.preloadModule = function(r, e) {
      var u = "";
      typeof r == "string" && r || (u += " The `href` argument encountered was " + o(r) + "."), e !== void 0 && typeof e != "object" ? u += " The `options` argument encountered was " + o(e) + "." : e && "as" in e && typeof e.as != "string" && (u += " The `as` option encountered was " + o(e.as) + "."), u && console.error(
        'ReactDOM.preloadModule(): Expected two arguments, a non-empty `href` string and, optionally, an `options` object with an `as` property valid for a `<link rel="modulepreload" as="..." />` tag.%s',
        u
      ), typeof r == "string" && (e ? (u = a(
        e.as,
        e.crossOrigin
      ), i.d.m(r, {
        as: typeof e.as == "string" && e.as !== "script" ? e.as : void 0,
        crossOrigin: u,
        integrity: typeof e.integrity == "string" ? e.integrity : void 0
      })) : i.d.m(r));
    }, m.requestFormReset = function(r) {
      i.d.r(r);
    }, m.unstable_batchedUpdates = function(r, e) {
      return r(e);
    }, m.useFormState = function(r, e, u) {
      return s().useFormState(r, e, u);
    }, m.useFormStatus = function() {
      return s().useHostTransitionStatus();
    }, m.version = "19.0.0", typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ < "u" && typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStop == "function" && __REACT_DEVTOOLS_GLOBAL_HOOK__.registerInternalModuleStop(Error());
  }()), m;
}
function P() {
  if (!(typeof __REACT_DEVTOOLS_GLOBAL_HOOK__ > "u" || typeof __REACT_DEVTOOLS_GLOBAL_HOOK__.checkDCE != "function")) {
    if (process.env.NODE_ENV !== "production")
      throw new Error("^_^");
    try {
      __REACT_DEVTOOLS_GLOBAL_HOOK__.checkDCE(P);
    } catch (p) {
      console.error(p);
    }
  }
}
process.env.NODE_ENV === "production" ? (P(), N.exports = j()) : N.exports = H();
var G = N.exports;
const D = /* @__PURE__ */ U(G), L = {
  disabled: !1
};
var V = process.env.NODE_ENV !== "production" ? c.oneOfType([c.number, c.shape({
  enter: c.number,
  exit: c.number,
  appear: c.number
}).isRequired]) : null;
process.env.NODE_ENV !== "production" && c.oneOfType([c.string, c.shape({
  enter: c.string,
  exit: c.string,
  active: c.string
}), c.shape({
  enter: c.string,
  enterDone: c.string,
  enterActive: c.string,
  exit: c.string,
  exitDone: c.string,
  exitActive: c.string
})]);
var F = function(g) {
  return g.scrollTop;
}, R = "unmounted", v = "exited", O = "entering", S = "entered", M = "exiting", E = /* @__PURE__ */ function(p) {
  k(g, p);
  function g(a, o) {
    var n;
    n = p.call(this, a, o) || this;
    var s = o, f = s && !s.isMounting ? a.enter : a.appear, i;
    return n.appearStatus = null, a.in ? f ? (i = v, n.appearStatus = O) : i = S : a.unmountOnExit || a.mountOnEnter ? i = R : i = v, n.state = {
      status: i
    }, n.nextCallback = null, n;
  }
  g.getDerivedStateFromProps = function(o, n) {
    var s = o.in;
    return s && n.status === R ? {
      status: v
    } : null;
  };
  var l = g.prototype;
  return l.componentDidMount = function() {
    this.updateStatus(!0, this.appearStatus);
  }, l.componentDidUpdate = function(o) {
    var n = null;
    if (o !== this.props) {
      var s = this.state.status;
      this.props.in ? s !== O && s !== S && (n = O) : (s === O || s === S) && (n = M);
    }
    this.updateStatus(!1, n);
  }, l.componentWillUnmount = function() {
    this.cancelNextCallback();
  }, l.getTimeouts = function() {
    var o = this.props.timeout, n, s, f;
    return n = s = f = o, o != null && typeof o != "number" && (n = o.exit, s = o.enter, f = o.appear !== void 0 ? o.appear : s), {
      exit: n,
      enter: s,
      appear: f
    };
  }, l.updateStatus = function(o, n) {
    if (o === void 0 && (o = !1), n !== null)
      if (this.cancelNextCallback(), n === O) {
        if (this.props.unmountOnExit || this.props.mountOnEnter) {
          var s = this.props.nodeRef ? this.props.nodeRef.current : D.findDOMNode(this);
          s && F(s);
        }
        this.performEnter(o);
      } else
        this.performExit();
    else this.props.unmountOnExit && this.state.status === v && this.setState({
      status: R
    });
  }, l.performEnter = function(o) {
    var n = this, s = this.props.enter, f = this.context ? this.context.isMounting : o, i = this.props.nodeRef ? [f] : [D.findDOMNode(this), f], t = i[0], d = i[1], r = this.getTimeouts(), e = f ? r.appear : r.enter;
    if (!o && !s || L.disabled) {
      this.safeSetState({
        status: S
      }, function() {
        n.props.onEntered(t);
      });
      return;
    }
    this.props.onEnter(t, d), this.safeSetState({
      status: O
    }, function() {
      n.props.onEntering(t, d), n.onTransitionEnd(e, function() {
        n.safeSetState({
          status: S
        }, function() {
          n.props.onEntered(t, d);
        });
      });
    });
  }, l.performExit = function() {
    var o = this, n = this.props.exit, s = this.getTimeouts(), f = this.props.nodeRef ? void 0 : D.findDOMNode(this);
    if (!n || L.disabled) {
      this.safeSetState({
        status: v
      }, function() {
        o.props.onExited(f);
      });
      return;
    }
    this.props.onExit(f), this.safeSetState({
      status: M
    }, function() {
      o.props.onExiting(f), o.onTransitionEnd(s.exit, function() {
        o.safeSetState({
          status: v
        }, function() {
          o.props.onExited(f);
        });
      });
    });
  }, l.cancelNextCallback = function() {
    this.nextCallback !== null && (this.nextCallback.cancel(), this.nextCallback = null);
  }, l.safeSetState = function(o, n) {
    n = this.setNextCallback(n), this.setState(o, n);
  }, l.setNextCallback = function(o) {
    var n = this, s = !0;
    return this.nextCallback = function(f) {
      s && (s = !1, n.nextCallback = null, o(f));
    }, this.nextCallback.cancel = function() {
      s = !1;
    }, this.nextCallback;
  }, l.onTransitionEnd = function(o, n) {
    this.setNextCallback(n);
    var s = this.props.nodeRef ? this.props.nodeRef.current : D.findDOMNode(this), f = o == null && !this.props.addEndListener;
    if (!s || f) {
      setTimeout(this.nextCallback, 0);
      return;
    }
    if (this.props.addEndListener) {
      var i = this.props.nodeRef ? [this.nextCallback] : [s, this.nextCallback], t = i[0], d = i[1];
      this.props.addEndListener(t, d);
    }
    o != null && setTimeout(this.nextCallback, o);
  }, l.render = function() {
    var o = this.state.status;
    if (o === R)
      return null;
    var n = this.props, s = n.children;
    n.in, n.mountOnEnter, n.unmountOnExit, n.appear, n.enter, n.exit, n.timeout, n.addEndListener, n.onEnter, n.onEntering, n.onEntered, n.onExit, n.onExiting, n.onExited, n.nodeRef;
    var f = I(n, ["children", "in", "mountOnEnter", "unmountOnExit", "appear", "enter", "exit", "timeout", "addEndListener", "onEnter", "onEntering", "onEntered", "onExit", "onExiting", "onExited", "nodeRef"]);
    return (
      // allows for nested Transitions
      /* @__PURE__ */ T.createElement(A.Provider, {
        value: null
      }, typeof s == "function" ? s(o, f) : T.cloneElement(T.Children.only(s), f))
    );
  }, g;
}(T.Component);
E.contextType = A;
E.propTypes = process.env.NODE_ENV !== "production" ? {
  /**
   * A React reference to DOM element that need to transition:
   * https://stackoverflow.com/a/51127130/4671932
   *
   *   - When `nodeRef` prop is used, `node` is not passed to callback functions
   *      (e.g. `onEnter`) because user already has direct access to the node.
   *   - When changing `key` prop of `Transition` in a `TransitionGroup` a new
   *     `nodeRef` need to be provided to `Transition` with changed `key` prop
   *     (see
   *     [test/CSSTransition-test.js](https://github.com/reactjs/react-transition-group/blob/13435f897b3ab71f6e19d724f145596f5910581c/test/CSSTransition-test.js#L362-L437)).
   */
  nodeRef: c.shape({
    current: typeof Element > "u" ? c.any : function(p, g, l, a, o, n) {
      var s = p[g];
      return c.instanceOf(s && "ownerDocument" in s ? s.ownerDocument.defaultView.Element : Element)(p, g, l, a, o, n);
    }
  }),
  /**
   * A `function` child can be used instead of a React element. This function is
   * called with the current transition status (`'entering'`, `'entered'`,
   * `'exiting'`, `'exited'`), which can be used to apply context
   * specific props to a component.
   *
   * ```jsx
   * <Transition in={this.state.in} timeout={150}>
   *   {state => (
   *     <MyComponent className={`fade fade-${state}`} />
   *   )}
   * </Transition>
   * ```
   */
  children: c.oneOfType([c.func.isRequired, c.element.isRequired]).isRequired,
  /**
   * Show the component; triggers the enter or exit states
   */
  in: c.bool,
  /**
   * By default the child component is mounted immediately along with
   * the parent `Transition` component. If you want to "lazy mount" the component on the
   * first `in={true}` you can set `mountOnEnter`. After the first enter transition the component will stay
   * mounted, even on "exited", unless you also specify `unmountOnExit`.
   */
  mountOnEnter: c.bool,
  /**
   * By default the child component stays mounted after it reaches the `'exited'` state.
   * Set `unmountOnExit` if you'd prefer to unmount the component after it finishes exiting.
   */
  unmountOnExit: c.bool,
  /**
   * By default the child component does not perform the enter transition when
   * it first mounts, regardless of the value of `in`. If you want this
   * behavior, set both `appear` and `in` to `true`.
   *
   * > **Note**: there are no special appear states like `appearing`/`appeared`, this prop
   * > only adds an additional enter transition. However, in the
   * > `<CSSTransition>` component that first enter transition does result in
   * > additional `.appear-*` classes, that way you can choose to style it
   * > differently.
   */
  appear: c.bool,
  /**
   * Enable or disable enter transitions.
   */
  enter: c.bool,
  /**
   * Enable or disable exit transitions.
   */
  exit: c.bool,
  /**
   * The duration of the transition, in milliseconds.
   * Required unless `addEndListener` is provided.
   *
   * You may specify a single timeout for all transitions:
   *
   * ```jsx
   * timeout={500}
   * ```
   *
   * or individually:
   *
   * ```jsx
   * timeout={{
   *  appear: 500,
   *  enter: 300,
   *  exit: 500,
   * }}
   * ```
   *
   * - `appear` defaults to the value of `enter`
   * - `enter` defaults to `0`
   * - `exit` defaults to `0`
   *
   * @type {number | { enter?: number, exit?: number, appear?: number }}
   */
  timeout: function(g) {
    var l = V;
    g.addEndListener || (l = l.isRequired);
    for (var a = arguments.length, o = new Array(a > 1 ? a - 1 : 0), n = 1; n < a; n++)
      o[n - 1] = arguments[n];
    return l.apply(void 0, [g].concat(o));
  },
  /**
   * Add a custom transition end trigger. Called with the transitioning
   * DOM node and a `done` callback. Allows for more fine grained transition end
   * logic. Timeouts are still used as a fallback if provided.
   *
   * **Note**: when `nodeRef` prop is passed, `node` is not passed.
   *
   * ```jsx
   * addEndListener={(node, done) => {
   *   // use the css transitionend event to mark the finish of a transition
   *   node.addEventListener('transitionend', done, false);
   * }}
   * ```
   */
  addEndListener: c.func,
  /**
   * Callback fired before the "entering" status is applied. An extra parameter
   * `isAppearing` is supplied to indicate if the enter stage is occurring on the initial mount
   *
   * **Note**: when `nodeRef` prop is passed, `node` is not passed.
   *
   * @type Function(node: HtmlElement, isAppearing: bool) -> void
   */
  onEnter: c.func,
  /**
   * Callback fired after the "entering" status is applied. An extra parameter
   * `isAppearing` is supplied to indicate if the enter stage is occurring on the initial mount
   *
   * **Note**: when `nodeRef` prop is passed, `node` is not passed.
   *
   * @type Function(node: HtmlElement, isAppearing: bool)
   */
  onEntering: c.func,
  /**
   * Callback fired after the "entered" status is applied. An extra parameter
   * `isAppearing` is supplied to indicate if the enter stage is occurring on the initial mount
   *
   * **Note**: when `nodeRef` prop is passed, `node` is not passed.
   *
   * @type Function(node: HtmlElement, isAppearing: bool) -> void
   */
  onEntered: c.func,
  /**
   * Callback fired before the "exiting" status is applied.
   *
   * **Note**: when `nodeRef` prop is passed, `node` is not passed.
   *
   * @type Function(node: HtmlElement) -> void
   */
  onExit: c.func,
  /**
   * Callback fired after the "exiting" status is applied.
   *
   * **Note**: when `nodeRef` prop is passed, `node` is not passed.
   *
   * @type Function(node: HtmlElement) -> void
   */
  onExiting: c.func,
  /**
   * Callback fired after the "exited" status is applied.
   *
   * **Note**: when `nodeRef` prop is passed, `node` is not passed
   *
   * @type Function(node: HtmlElement) -> void
   */
  onExited: c.func
} : {};
function _() {
}
E.defaultProps = {
  in: !1,
  mountOnEnter: !1,
  unmountOnExit: !1,
  appear: !1,
  enter: !0,
  exit: !0,
  onEnter: _,
  onEntering: _,
  onEntered: _,
  onExit: _,
  onExiting: _,
  onExited: _
};
E.UNMOUNTED = R;
E.EXITED = v;
E.ENTERING = O;
E.ENTERED = S;
E.EXITING = M;
const W = (p) => p.scrollTop;
function Y(p, g) {
  const {
    timeout: l,
    easing: a,
    style: o = {}
  } = p;
  return {
    duration: o.transitionDuration ?? (typeof l == "number" ? l : l[g.mode] || 0),
    easing: o.transitionTimingFunction ?? (typeof a == "object" ? a[g.mode] : a),
    delay: o.transitionDelay
  };
}
export {
  D as R,
  E as T,
  G as a,
  Y as g,
  W as r
};
