// @notificationapi/react/dist/assets/WebPushOptInMessage.js
import { jsx as t, jsxs as s } from "react/jsx-runtime";
import { useContext as g } from "react";
import { NotificationAPIContext as x } from "../components/Provider/context.js";
import { c as S } from "./createSvgIcon.js";
import { A as C, B as o } from "./Button.js";
const b = S(/* @__PURE__ */ t("path", {
  d: "M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2m6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56M12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96M4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2s.06 1.34.14 2zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56m2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8M12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96M14.34 14H9.66c-.09-.66-.16-1.32-.16-2s.07-1.35.16-2h4.68c.09.65.16 1.32.16 2s-.07 1.34-.16 2m.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56M16.36 14c.08-.66.14-1.32.14-2s-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2z"
}), "LanguageOutlined"), P = ({
  hideAfterInteraction: n,
  alertContainerStyle: i,
  descriptionStyle: c,
  buttonContainerStyle: l,
  yesButtonStyle: a,
  cancelButtonStyle: r,
  description: p,
  icon: u
}) => {
  const e = g(x);
  if (!e)
    return null;
  const d = {
    marginBottom: "10px",
    borderRadius: "5px",
    padding: "10px",
    fontSize: "12px",
    // Smaller font size
    margin: "10px auto",
    // Center if width is more than 500px
    display: "flex",
    alignItems: "center",
    // Align items vertically
    ...i
  }, h = {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    // Align items vertically
    fontSize: "12px",
    width: "100%",
    ...c
  }, y = {
    display: "flex",
    alignItems: "center",
    ...l
  }, m = {
    marginLeft: "10px",
    marginRight: "10px",
    fontSize: "10px",
    ...a
  }, f = {
    fontSize: "10px",
    ...r
  };
  return /* @__PURE__ */ t(
    C,
    {
      severity: "info",
      icon: u ?? /* @__PURE__ */ t(b, { type: "text" }),
      style: d,
      children: /* @__PURE__ */ s("div", { style: h, children: [
        /* @__PURE__ */ t("span", { children: p ?? "Would you like to enable web push notifications to stay updated?" }),
        /* @__PURE__ */ s("div", { style: y, children: [
          /* @__PURE__ */ t(
            o,
            {
              style: m,
              onClick: () => {
                n && localStorage.setItem("hideWebPushOptInMessage", "true"), e.setWebPushOptIn(!0), e.setWebPushOptInMessage(!1);
              },
              children: "Yes"
            }
          ),
          /* @__PURE__ */ t(
            o,
            {
              style: f,
              onClick: () => {
                n && localStorage.setItem("hideWebPushOptInMessage", "true"), e.setWebPushOptInMessage(!1);
              },
              children: "No"
            }
          )
        ] })
      ] })
    }
  );
};
export {
  b as L,
  P as W
};
